
## Monday, April 27

To do:

* ~~Finish PIPITS soil fungal analysis.~~
* ~~Edit next week's *Refuge Notebook* article.~~

```shell
## Trying to figure out why denoising didn't work.
## The denoise command I issued the other day yielded an output file with 0 lines.
## Command: vsearch --cluster_unoise out_funits/intermediate/derep.ITS2.sizefiltered.fasta --sizein --sizeout --minsize 3 --centroids out_funits/ITS_denoised.fasta

source activate pipits_env
cd ~/Documents/FWS/2019_black_spruce_inventory/data/soil_fungi
vsearch --cluster_unoise out_funits/intermediate/derep.ITS2.sizefiltered.fasta --sizein --sizeout --minsize 3 --centroids out_funits/ITS_denoised.fasta
vsearch v2.14.2_linux_x86_64, 7.8GB RAM, 2 cores
https://github.com/torognes/vsearch

Reading file out_funits/intermediate/derep.ITS2.sizefiltered.fasta 100% 
3733528 nt in 21662 seqs, min 102, max 263, avg 172
minsize 3: 304142 sequences discarded.
Masking 100% 
Sorting by abundance 100%
Counting k-mers 100% 
Clustering 100% 
Sorting clusters 100%
Writing clusters 100% 
Clusters: 917 Size min 3, max 65239, avg 23.6
Singletons: 0, 0.0% of seqs, 0.0% of clusters
## That seemed to work.

## Trying to rereplicate
pipits_rereplicate -i out_funits/ITS_denoised.fasta -o out_funits/ITS_denoised_rerep.fasta --uc out_funits/intermediate/derep.uc
Error: Size values in headers added by USEARCH != number of sequences in .uc
65239 3136

## Ok, I see what happened. unoise clustered the sequences, and now the clusters are bigger than in the --uc file. ## How can I deal with this?
```

I am editing the `pipits_process` script, renaming it `pipits_process_ESV`. I am replacing with the OTU clustering section below.

```python
    ##################
    # OTU clustering #
    ##################

    logger(ENDC + "Picking OTUs [VSEARCH]" + ENDC, logging_file, display = True)
    cmd = " ".join([VSEARCH, 
                    "--cluster_fast", tmpDir + "/input_nr.fasta", 
                    "--id", options.VSEARCH_id,
                    "--centroids", tmpDir + "/input_nr_otus.fasta",
                    "--uc", tmpDir + "/input_nr_otus.uc",
                    "--threads", options.threads])
    run_cmd(cmd, logging_file, options.verbose)
```
Replacement:

```python
    ##################
    # Denoising      #
    ##################

    logger(ENDC + "Denoising [VSEARCH]" + ENDC, logging_file, display = True)
    cmd = " ".join([VSEARCH, 
                    "--cluster_unoise", tmpDir + "/input_nr.fasta", 
                    "--minsize 3",
                    "--centroids", tmpDir + "/input_nr_otus.fasta",
                    "--uc", tmpDir + "/input_nr_otus.uc",
                    "--threads", options.threads])
    run_cmd(cmd, logging_file, options.verbose)
```

Trying it out.

```shell
pipits_process_ESV -i out_funits/ITS.fasta -o out_process -v -r

## That worked!!
```

Complete relevant I/O from today:

```shell
(base) matt@line:~/Documents/FWS/2019_black_spruce_inventory/data/soil_fungi$ source activate pipits_env
(pipits_env) matt@line:~/Documents/FWS/2019_black_spruce_inventory/data/soil_fungi$ cd ~/Documents/FWS/2019_black_spruce_inventory/data/soil_fungi
(pipits_env) matt@line:~/Documents/FWS/2019_black_spruce_inventory/data/soil_fungi$ ls
2020-04-24-0635_io.txt  out_funits  out_seqprep  raw_fastq  readpairslist.txt
(pipits_env) matt@line:~/Documents/FWS/2019_black_spruce_inventory/data/soil_fungi$ vsearch --cluster_unoise out_funits/intermediate/derep.ITS2.sizefiltered.fasta --sizein --sizeout --minsize 3 --centroids out_funits/ITS_denoised.fasta
vsearch v2.14.2_linux_x86_64, 7.8GB RAM, 2 cores
https://github.com/torognes/vsearch

Reading file out_funits/intermediate/derep.ITS2.sizefiltered.fasta 100%  
3733528 nt in 21662 seqs, min 102, max 263, avg 172
minsize 3: 304142 sequences discarded.
Masking 100% 
Sorting by abundance 100%
Counting k-mers 100% 
Clustering 100%  
Sorting clusters 100%
Writing clusters 100% 
Clusters: 917 Size min 3, max 65239, avg 23.6
Singletons: 0, 0.0% of seqs, 0.0% of clusters
(pipits_env) matt@line:~/Documents/FWS/2019_black_spruce_inventory/data/soil_fungi$ ls -al out_funits
total 131980
drwxrwxr-x 3 matt matt      4096 Apr 25 07:36 .
drwxrwxr-x 5 matt matt      4096 Apr 25 10:52 ..
drwxrwxr-x 2 matt matt      4096 Apr 25 04:49 intermediate
-rw-rw-r-- 1 matt matt    178365 Apr 27 06:03 ITS_denoised.fasta
-rw-rw-r-- 1 matt matt         0 Apr 25 07:36 ITS_denoised_rerep.fasta
-rw-rw-r-- 1 matt matt 134937545 Apr 25 04:50 ITS.fasta
-rw-rw-r-- 1 matt matt      3490 Apr 25 04:50 output.log
-rw-rw-r-- 1 matt matt       228 Apr 25 04:50 summary.log
-rw-rw-r-- 1 matt matt      7068 Apr 25 04:50 versions.log
(pipits_env) matt@line:~/Documents/FWS/2019_black_spruce_inventory/data/soil_fungi$ head ITS_denoised.fasta
head: cannot open 'ITS_denoised.fasta' for reading: No such file or directory
(pipits_env) matt@line:~/Documents/FWS/2019_black_spruce_inventory/data/soil_fungi$ head out_funits/ITS_denoised.fasta
>3069-1_320;size=65239
CTGCCCTCAAGCACGGCTTGTGTGTTGGGCTCCGTCCCCCCGGGGACGGGTCCGAAAGGCAGCGGCGGCACCGAGTCCGG
TCCTCGAGCGTATGGGGCTTTGTCACCCGCTCTGTAGGCCCGGCCGGCGCCAGCCGACAACCAATCATCCTTTTCA
>3074-1_997;size=22437
AACTCTCACCCTCTAGCTTTCTTAATCGTGGCTAGCGGCGTGGACGTGAGCGCTGCTGCTTTGTTGCGGCTCGCTCGAAA
TGCATTAGCAGACCCTTTTCGTAATCGGTTCCACTCAACGTGATAAGTATTTCGTTGAGGACAGTTGCAGCAATGCGGCT
GGCCGGGATAAGAAAGGCATAGTTGTCAGCTTCTAATCGCCCTTGGGCAATTTTTTATG
>3288-1_526;size=21465
AACTCTCAACCTATAGATTTTTGTTAATCTTGTAGGCTTGGACTTGGAAGTCTTGCTGGCTGAATAAGCCGGCTCTTCTG
AAATGTATTAGCAGATCCTCTTTGTAATTGGTTCCTACTCAACGTGATAAGTATTTCGTTGAGGACATTTCCTTTATTGG
(pipits_env) matt@line:~/Documents/FWS/2019_black_spruce_inventory/data/soil_fungi$ pipits_rereplicate -i out_funits/ITS_denoised.fasta -o out_funits/ITS_denoised_rerep.fasta --uc out_funits/intermediate/derep.uc
Error: Size values in headers added by USEARCH != number of sequences in .uc
65239 3136
(pipits_env) matt@line:~/Documents/FWS/2019_black_spruce_inventory/data/soil_fungi$ ls
2020-04-24-0635_io.txt  out_funits  out_seqprep  pipits_process_ESV  raw_fastq  readpairslist.txt
(pipits_env) matt@line:~/Documents/FWS/2019_black_spruce_inventory/data/soil_fungi$ Trying it out.
Trying: command not found
(pipits_env) matt@line:~/Documents/FWS/2019_black_spruce_inventory/data/soil_fungi$ 
(pipits_env) matt@line:~/Documents/FWS/2019_black_spruce_inventory/data/soil_fungi$ ```shell
> ./pipits_process_ESV -i out_funits/ITS.fasta -o out_process -v -r
> ```

Command 'shell' not found, did you mean:

  command 'spell' from deb spell
  command 'qshell' from deb qtile
  command 'shelr' from deb shelr
  command 'jshell' from deb openjdk-11-jdk-headless
  command 'bshell' from deb avahi-ui-utils
  command 'pshell' from deb python-pyramid

Try: sudo apt install <deb name>

usage: PIPITS_PROCESS: Sequences to OTU Table [-h] -i <FILE> [-o <DIR>]
                                              [-d <FLOAT>] [-c <FLOAT>]
                                              [-l <TXT>] [--includeuniqueseqs]
                                              [-r] [-v] [-t <INT>]
                                              [--Xms <INT>] [--Xmx <INT>]
                                              [--warcup]
                                              [--unite {04.02.2020,02.02.2019,01.12.2017,28.06.2017}]
PIPITS_PROCESS: Sequences to OTU Table: error: unrecognized arguments: 2.5, the PIPITS Project https://github.com/hsgweon/pipits --------------------------------- 2020-04-27 07:08:27 pipits_process started 2020-04-27 07:08:27 Generating a sample list from the input sequences python: can't open file '/home/matt/Documents/FWS/2019_black_spruce_inventory/data/soil_fungi/pipits_getsamplelistfromfasta': [Errno 2] No such file or directory 2020-04-27 07:08:27 Error: None zero returncode: python /home/matt/Documents/FWS/2019_black_spruce_inventory/data/soil_fungi/pipits_getsamplelistfromfasta
(pipits_env) matt@line:~/Documents/FWS/2019_black_spruce_inventory/data/soil_fungi$ 
(pipits_env) matt@line:~/Documents/FWS/2019_black_spruce_inventory/data/soil_fungi$ ./pipits_process_ESV -i out_funits/ITS.fasta -o out_process -v -r
pipits_process 2.5, the PIPITS Project
https://github.com/hsgweon/pipits
---------------------------------

2020-04-27 07:10:02 pipits_process started
2020-04-27 07:10:02 Generating a sample list from the input sequences
python: can't open file '/home/matt/Documents/FWS/2019_black_spruce_inventory/data/soil_fungi/pipits_getsamplelistfromfasta': [Errno 2] No such file or directory
2020-04-27 07:10:02 Error: None zero returncode: python /home/matt/Documents/FWS/2019_black_spruce_inventory/data/soil_fungi/pipits_getsamplelistfromfasta -i out_funits/ITS.fasta -o out_process/sampleIDs.txt
(pipits_env) matt@line:~/Documents/FWS/2019_black_spruce_inventory/data/soil_fungi$ pipits_process_ESV -i out_funits/ITS.fasta -o out_process -v -r
pipits_process 2.5, the PIPITS Project
https://github.com/hsgweon/pipits
---------------------------------

2020-04-27 07:11:12 pipits_process started
2020-04-27 07:11:12 Generating a sample list from the input sequences
2020-04-27 07:11:15 Downloading UNITE trained database, version: 04.02.2020
[###################################################################################################################]100% | 275.1 KiB/s | 131112800 of 131112800 | Time:  0:07:45
File size: 125.04 MB
2020-04-27 07:19:05 ... Unpacking
2020-04-27 07:19:15 ... done
2020-04-27 07:19:15 Downloading WARCUP trained database: 
[#####################################################################################################################]100% | 292.9 KiB/s | 17880783 of 17880783 | Time:  0:00:59
File size: 17.05 MB
2020-04-27 07:20:19 ... Unpacking
2020-04-27 07:20:20 ... done
2020-04-27 07:20:20 Downloading UCHIME database for chimera filtering: 
[#######################################################################################################################]100% | 200.0 KiB/s | 4578703 of 4578703 | Time:  0:00:22
File size: 4.37 MB
2020-04-27 07:20:46 ... Unpacking
2020-04-27 07:20:47 ... done
2020-04-27 07:20:47 Dereplicating and removing unique sequences prior to picking OTUs
vsearch v2.14.2_linux_x86_64, 7.8GB RAM, 2 cores
https://github.com/torognes/vsearch

Dereplicating file out_funits/ITS.fasta 100%
124639323 nt in 704602 seqs, min 101, max 332, avg 177
Sorting 100%
75623 unique sequences, avg cluster 9.3, median 1, max 72255
Writing output file 100%
19852 uniques written, 55771 clusters discarded (73.7%)
2020-04-27 07:20:49 Denoising [VSEARCH]
vsearch v2.14.2_linux_x86_64, 7.8GB RAM, 2 cores
https://github.com/torognes/vsearch

Reading file out_process/intermediate/input_nr.fasta 100%
2138936 nt in 11839 seqs, min 101, max 277, avg 181
minsize 3: 8013 sequences discarded.
Masking 100%
Sorting by abundance 100%
Counting k-mers 100%
Clustering 100%
Sorting clusters 100%
Writing clusters 100%
Clusters: 3413 Size min 1, max 449, avg 3.5
Singletons: 2868, 24.2% of seqs, 84.0% of clusters
2020-04-27 07:20:57 Removing chimeras [VSEARCH]
vsearch v2.14.2_linux_x86_64, 7.8GB RAM, 2 cores
https://github.com/torognes/vsearch

Reading file pipits_db/uchime_reference_dataset_28.06.2017/uchime_reference_dataset_28.06.2017.fasta 100%
16786547 nt in 30555 seqs, min 146, max 2570, avg 549
Masking 100%
Counting k-mers 100%
Creating k-mer index 100%
Detecting chimeras 100%
Found 1069 (31.3%) chimeras, 2327 (68.2%) non-chimeras,
and 17 (0.5%) borderline sequences in 3413 unique sequences.
Taking abundance information into account, this corresponds to
12349 (2.3%) chimeras, 526932 (97.4%) non-chimeras,
and 1902 (0.4%) borderline sequences in 541183 total sequences.
2020-04-27 07:21:19 Renaming OTUs
2020-04-27 07:21:19 Mapping reads onto centroids [VSEARCH]
vsearch v2.14.2_linux_x86_64, 7.8GB RAM, 2 cores
https://github.com/torognes/vsearch

Reading file out_process/intermediate/input_nr_otus_nonchimeras_relabelled.fasta 100%
397037 nt in 2327 seqs, min 102, max 277, avg 171
Masking 100%
Counting k-mers 100%
Creating k-mer index 100%
Searching 100%
Matching unique query sequences: 682959 of 704602 (96.93%)
2020-04-27 07:27:06 Making OTU table
2020-04-27 07:27:11 Converting classic tabular OTU into a BIOM format [BIOM]
2020-04-27 07:27:15 Assigning taxonomy with UNITE [RDP Classifier]
2020-04-27 07:38:25 Reformatting RDP_Classifier output
2020-04-27 07:38:25 Adding assignment to OTU table [BIOM]
2020-04-27 07:38:30 Converting OTU table with taxa assignment into a BIOM format [BIOM]
2020-04-27 07:38:32 Phylotyping OTU table
2020-04-27 07:38:39 	Number of reads used to generate OTU table: 682959
2020-04-27 07:38:39 	Number of OTUs:                             2327
2020-04-27 07:38:39 	Number of phylotypes:                       784
2020-04-27 07:38:39 	Number of samples:                          26
2020-04-27 07:38:39 Done - Resulting files are in "out_process" directory
2020-04-27 07:38:39 pipits_process ended successfully.
```

I worked on editing a *Refuge Notebook* article on biotech life for next week, getting a revised draft back to the author.

Arctos is still down.

Update: Arctos is up as of 12:39!

I updated content for the Kenai NWR content in update to the [Kenai Peninsula Cooperative Weed Management Area](http://www.kenaiweeds.org/) strategic plan.

