
## Monday, March 23

```
## Downloaded ASV Sequences from Bowser et al. (2020)

Bowser, M. L. 2020. Supplementary material 5. ASV sequences. doi.org/10.3897/BDJ.8.e50124.suppl5

oo_371419.fas

cd SC*/ml-jg
source activate SCVUCv4.3

## Assign taxonomy from Bowser (2020).
vsearch --usearch_global chimera.denoised.nonchimeras.taxon \
--db 2020-03-21-0600_oo_371419.fas \
--id .90 \
--iddef 0 \
--blast6out 2020-03-21-0626_vsearch_Bowser_2020_matches.txt

## Assign taxonomy based on DS-BOWSER BOLD project.
vsearch --usearch_global chimera.denoised.nonchimeras.taxon \
--db 2020-03-21-0611_DS-BOWSER.fas \
--id .90 \
--iddef 0 \
--blast6out 2020-03-21-0629_vsearch_DS-BOWSER_matches.txt

## Assign taxonomy based on AKTerrInvCOILib.
vsearch --usearch_global chimera.denoised.nonchimeras.taxon \
--db 2019-03-25-0832_clusters.fas \
--id .90 \
--iddef 0 \
--blast6out 2020-03-21-0639_vsearch_AKTerrInvCOILib_matches.txt
```


```r
setwd("D:/projects/blackfish/2020_blackfish_diet/2020-03_SCVUC/ml-jg")

results1 <- read.csv("results.csv", stringsAsFactors=FALSE)

length(levels(as.factor(results1$ml.jg_GlobalESV)))
[1] 212

aklib <- read.delim("2020-03-23-0659_vsearch_AKTerrInvCOILib_matches.txt", header=FALSE, stringsAsFactors=FALSE)

names(results1)[1] <- "ESV_name"
results1$ESV_name <- gsub("ml-jg_", "", results1$ESV_name)

## column names from http://www.metagenomics.wiki/tools/blast/blastn-output-format-6
names(aklib)[1:3] <- c("ESV_name", "AKlibprocessid", "pident")
```

I accidentally overwrote most of the rest of that R script.

I attended the 11:00 biology check-in telecon. 

I prepared data for Rob Massengill, but I accidentally overwrote the R script. I had downloaded occurrence data from the Vogel Lake vicinity from GBIF [@gbiforg_gbif_2020-3], Arctos, and iNaturalist, then processed these.

