
## Thursday, April 23

Last night I downloaded and installed Miniconda 3 and PIPITS 2.5.

To do:

* Slikok occupancy?
* Review Rob's pike treatment plan.

I spent much of the morning setting up and testing a PIPITS 2.5 environment [@gweon_pipits_2015] (<https://github.com/hsgweon/pipits>) and working on some tweaks to run it using exact sequence variant methods instead of the OTU clustering method that is the default.

Testing, following directions at <https://github.com/hsgweon/pipits>.

```shell
cd ~/Documents/FWS

wget https://sourceforge.net/projects/pipits/files/PIPITS_TESTDATA/pipits_test.tar.gz -O pipits_test.tar.gz
tar xvfz pipits_test.tar.gz

cd pipits_test
source activate pipits_env
pispino_createreadpairslist -i rawdata -o readpairslist.txt
pispino_seqprep -i rawdata -o out_seqprep -l readpairslist.txt
pipits_funits -i out_seqprep/prepped.fasta -o out_funits -x ITS2 -v -r
pipits_process -i out_funits/ITS.fasta -o out_process -v -r

```

That worked.

Now I would like to use an exact sequence variant filtering method and avoid the OTU clustering.

I looked at how this was done in the SCVUC pipeline
shell:
		"vsearch --cluster_unoise {input} --sizein --sizeout --minsize {config[VSEARCH_DENOISE][minsize]} --centroids {output} --log {log}"
VSEARCH_DENOISE:
    minsize: 3

```shell
vsearch --cluster_unoise out_funits/ITS.fasta --sizein --sizeout --minsize 3 --centroids out_funits/ITS_denoised.fasta
```

That got rid of all the sequences. Trying `--minsize 2`.

```shell
rm out_funits/ITS_denoised.fasta
vsearch --cluster_unoise out_funits/ITS.fasta --sizein --sizeout --minsize 2 --centroids out_funits/ITS_denoised.fasta
```

That also discarded all sequences.

I edited the input FASTA, manually making more copies of one sequence.

```shell
rm out_funits/ITS_denoised.fasta
vsearch --cluster_unoise out_funits/ITS_edited.fasta --sizein --sizeout --minsize 3 --centroids out_funits/ITS_denoised.fasta
```

Oh, I think I see. The `-sizein` and `--sizeout` options are for dereplicated input and output.

```shell
rm out_funits/ITS_denoised.fasta
vsearch --cluster_unoise out_funits/ITS.fasta --minsize 3 --centroids out_funits/ITS_denoised.fasta
```

That still filtered out all sequences. Trying again to make sure I did not mess anything up. Deleted products of ITS and processing steps.

```shell
pipits_funits -i out_seqprep/prepped.fasta -o out_funits -x ITS2 -v -r
```

Ok, I see that the dereplicated file after ITS sequencing is out_funits/intermediate/derep.ITS2.sizefiltered.fasta

```shell
vsearch --cluster_unoise out_funits/intermediate/derep.ITS2.sizefiltered.fasta --sizein --sizeout --minsize 3 --centroids out_funits/ITS_denoised.fasta
```

Need to rereplicate now.

```shell
pipits_rereplicate -i out_funits/ITS_denoised.fasta -o out_funits/ITS.fasta --uc out_funits/intermediate/derep.uc
```
Now trying the rest with no OTU clustering.

```shell
pipits_process -i out_funits/ITS.fasta -o out_process -v -r -d 1.00
```

That worked! Output looked good.

Later in the day, trying this out on 2019 black spruce soil fungal samples.

```shell
cd ~/Documents/FWS/2019_black_spruce_inventory/data/soil_fungi
source activate pipits_env
## renaming files.
cd raw_fastq
rename -n 's/6202-soifun//' *
rename 's/6202-soifun//' *
rename -n 's/-ITS3mix-ITS4ngs//' *
rename 's/-ITS3mix-ITS4ngs//' *
cd ..
pispino_createreadpairslist -i raw_fastq -o readpairslist.txt
pispino_seqprep -i raw_fastq -o out_seqprep -l readpairslist.txt
pipits_funits -i out_seqprep/prepped.fasta -o out_funits -x ITS2 -v -r
```

I worked some on adding geographic extents to the metadata of the snowshow hare data in preparation for the data management early adopters' meeting.

### Data management early adopters' meeting at 13:00

* Still no specific homework.

Transferring 2019 black spruce soil fungi raw metagenomic data from the server over VPN to my laptop took a long time, but it worked in the end. I initiated a PIPITS analysis of this dataset.

Pretty complete I/O from today:

```shell
(pipits_env) matt@line:~/Documents/FWS/2019_black_spruce_inventory/data/soil_fungi$ cd raw_fastq
(pipits_env) matt@line:~/Documents/FWS/2019_black_spruce_inventory/data/soil_fungi/raw_fastq$ rename -n 's/6202-soifun//' *
rename(6202-soifun3069-1-ITS3mix-ITS4ngs_R1.fastq, 3069-1-ITS3mix-ITS4ngs_R1.fastq)
rename(6202-soifun3069-1-ITS3mix-ITS4ngs_R2.fastq, 3069-1-ITS3mix-ITS4ngs_R2.fastq)
rename(6202-soifun3069-2-ITS3mix-ITS4ngs_R1.fastq, 3069-2-ITS3mix-ITS4ngs_R1.fastq)
rename(6202-soifun3069-2-ITS3mix-ITS4ngs_R2.fastq, 3069-2-ITS3mix-ITS4ngs_R2.fastq)
rename(6202-soifun3072-1-ITS3mix-ITS4ngs_R1.fastq, 3072-1-ITS3mix-ITS4ngs_R1.fastq)
rename(6202-soifun3072-1-ITS3mix-ITS4ngs_R2.fastq, 3072-1-ITS3mix-ITS4ngs_R2.fastq)
rename(6202-soifun3072-2-ITS3mix-ITS4ngs_R1.fastq, 3072-2-ITS3mix-ITS4ngs_R1.fastq)
rename(6202-soifun3072-2-ITS3mix-ITS4ngs_R2.fastq, 3072-2-ITS3mix-ITS4ngs_R2.fastq)
rename(6202-soifun3074-1-ITS3mix-ITS4ngs_R1.fastq, 3074-1-ITS3mix-ITS4ngs_R1.fastq)
rename(6202-soifun3074-1-ITS3mix-ITS4ngs_R2.fastq, 3074-1-ITS3mix-ITS4ngs_R2.fastq)
rename(6202-soifun3074-2-ITS3mix-ITS4ngs_R1.fastq, 3074-2-ITS3mix-ITS4ngs_R1.fastq)
rename(6202-soifun3074-2-ITS3mix-ITS4ngs_R2.fastq, 3074-2-ITS3mix-ITS4ngs_R2.fastq)
rename(6202-soifun3092-1-ITS3mix-ITS4ngs_R1.fastq, 3092-1-ITS3mix-ITS4ngs_R1.fastq)
rename(6202-soifun3092-1-ITS3mix-ITS4ngs_R2.fastq, 3092-1-ITS3mix-ITS4ngs_R2.fastq)
rename(6202-soifun3092-2-ITS3mix-ITS4ngs_R1.fastq, 3092-2-ITS3mix-ITS4ngs_R1.fastq)
rename(6202-soifun3092-2-ITS3mix-ITS4ngs_R2.fastq, 3092-2-ITS3mix-ITS4ngs_R2.fastq)
rename(6202-soifun3140-1-ITS3mix-ITS4ngs_R1.fastq, 3140-1-ITS3mix-ITS4ngs_R1.fastq)
rename(6202-soifun3140-1-ITS3mix-ITS4ngs_R2.fastq, 3140-1-ITS3mix-ITS4ngs_R2.fastq)
rename(6202-soifun3140-2-ITS3mix-ITS4ngs_R1.fastq, 3140-2-ITS3mix-ITS4ngs_R1.fastq)
rename(6202-soifun3140-2-ITS3mix-ITS4ngs_R2.fastq, 3140-2-ITS3mix-ITS4ngs_R2.fastq)
rename(6202-soifun3202-1-ITS3mix-ITS4ngs_R1.fastq, 3202-1-ITS3mix-ITS4ngs_R1.fastq)
rename(6202-soifun3202-1-ITS3mix-ITS4ngs_R2.fastq, 3202-1-ITS3mix-ITS4ngs_R2.fastq)
rename(6202-soifun3202-2-ITS3mix-ITS4ngs_R1.fastq, 3202-2-ITS3mix-ITS4ngs_R1.fastq)
rename(6202-soifun3202-2-ITS3mix-ITS4ngs_R2.fastq, 3202-2-ITS3mix-ITS4ngs_R2.fastq)
rename(6202-soifun3288-1-ITS3mix-ITS4ngs_R1.fastq, 3288-1-ITS3mix-ITS4ngs_R1.fastq)
rename(6202-soifun3288-1-ITS3mix-ITS4ngs_R2.fastq, 3288-1-ITS3mix-ITS4ngs_R2.fastq)
rename(6202-soifun3288-2-ITS3mix-ITS4ngs_R1.fastq, 3288-2-ITS3mix-ITS4ngs_R1.fastq)
rename(6202-soifun3288-2-ITS3mix-ITS4ngs_R2.fastq, 3288-2-ITS3mix-ITS4ngs_R2.fastq)
rename(6202-soifun5555-1-ITS3mix-ITS4ngs_R1.fastq, 5555-1-ITS3mix-ITS4ngs_R1.fastq)
rename(6202-soifun5555-1-ITS3mix-ITS4ngs_R2.fastq, 5555-1-ITS3mix-ITS4ngs_R2.fastq)
rename(6202-soifun5555-2-ITS3mix-ITS4ngs_R1.fastq, 5555-2-ITS3mix-ITS4ngs_R1.fastq)
rename(6202-soifun5555-2-ITS3mix-ITS4ngs_R2.fastq, 5555-2-ITS3mix-ITS4ngs_R2.fastq)
rename(6202-soifunMG07-1-ITS3mix-ITS4ngs_R1.fastq, MG07-1-ITS3mix-ITS4ngs_R1.fastq)
rename(6202-soifunMG07-1-ITS3mix-ITS4ngs_R2.fastq, MG07-1-ITS3mix-ITS4ngs_R2.fastq)
rename(6202-soifunMG07-2-ITS3mix-ITS4ngs_R1.fastq, MG07-2-ITS3mix-ITS4ngs_R1.fastq)
rename(6202-soifunMG07-2-ITS3mix-ITS4ngs_R2.fastq, MG07-2-ITS3mix-ITS4ngs_R2.fastq)
rename(6202-soifunMG19-1-ITS3mix-ITS4ngs_R1.fastq, MG19-1-ITS3mix-ITS4ngs_R1.fastq)
rename(6202-soifunMG19-1-ITS3mix-ITS4ngs_R2.fastq, MG19-1-ITS3mix-ITS4ngs_R2.fastq)
rename(6202-soifunMG19-2-ITS3mix-ITS4ngs_R1.fastq, MG19-2-ITS3mix-ITS4ngs_R1.fastq)
rename(6202-soifunMG19-2-ITS3mix-ITS4ngs_R2.fastq, MG19-2-ITS3mix-ITS4ngs_R2.fastq)
rename(6202-soifunMG24-1-ITS3mix-ITS4ngs_R1.fastq, MG24-1-ITS3mix-ITS4ngs_R1.fastq)
rename(6202-soifunMG24-1-ITS3mix-ITS4ngs_R2.fastq, MG24-1-ITS3mix-ITS4ngs_R2.fastq)
rename(6202-soifunMG24-2-ITS3mix-ITS4ngs_R1.fastq, MG24-2-ITS3mix-ITS4ngs_R1.fastq)
rename(6202-soifunMG24-2-ITS3mix-ITS4ngs_R2.fastq, MG24-2-ITS3mix-ITS4ngs_R2.fastq)
rename(6202-soifunMG39-1-ITS3mix-ITS4ngs_R1.fastq, MG39-1-ITS3mix-ITS4ngs_R1.fastq)
rename(6202-soifunMG39-1-ITS3mix-ITS4ngs_R2.fastq, MG39-1-ITS3mix-ITS4ngs_R2.fastq)
rename(6202-soifunMG39-2-ITS3mix-ITS4ngs_R1.fastq, MG39-2-ITS3mix-ITS4ngs_R1.fastq)
rename(6202-soifunMG39-2-ITS3mix-ITS4ngs_R2.fastq, MG39-2-ITS3mix-ITS4ngs_R2.fastq)
rename(6202-soifunSK22-1-ITS3mix-ITS4ngs_R1.fastq, SK22-1-ITS3mix-ITS4ngs_R1.fastq)
rename(6202-soifunSK22-1-ITS3mix-ITS4ngs_R2.fastq, SK22-1-ITS3mix-ITS4ngs_R2.fastq)
rename(6202-soifunSK22-2-ITS3mix-ITS4ngs_R1.fastq, SK22-2-ITS3mix-ITS4ngs_R1.fastq)
rename(6202-soifunSK22-2-ITS3mix-ITS4ngs_R2.fastq, SK22-2-ITS3mix-ITS4ngs_R2.fastq)
(pipits_env) matt@line:~/Documents/FWS/2019_black_spruce_inventory/data/soil_fungi/raw_fastq$ rename 's/6202-soifun//' *
(pipits_env) matt@line:~/Documents/FWS/2019_black_spruce_inventory/data/soil_fungi/raw_fastq$ ls
3069-1-ITS3mix-ITS4ngs_R1.fastq  3140-2-ITS3mix-ITS4ngs_R1.fastq  MG19-1-ITS3mix-ITS4ngs_R1.fastq
3069-1-ITS3mix-ITS4ngs_R2.fastq  3140-2-ITS3mix-ITS4ngs_R2.fastq  MG19-1-ITS3mix-ITS4ngs_R2.fastq
3069-2-ITS3mix-ITS4ngs_R1.fastq  3202-1-ITS3mix-ITS4ngs_R1.fastq  MG19-2-ITS3mix-ITS4ngs_R1.fastq
3069-2-ITS3mix-ITS4ngs_R2.fastq  3202-1-ITS3mix-ITS4ngs_R2.fastq  MG19-2-ITS3mix-ITS4ngs_R2.fastq
3072-1-ITS3mix-ITS4ngs_R1.fastq  3202-2-ITS3mix-ITS4ngs_R1.fastq  MG24-1-ITS3mix-ITS4ngs_R1.fastq
3072-1-ITS3mix-ITS4ngs_R2.fastq  3202-2-ITS3mix-ITS4ngs_R2.fastq  MG24-1-ITS3mix-ITS4ngs_R2.fastq
3072-2-ITS3mix-ITS4ngs_R1.fastq  3288-1-ITS3mix-ITS4ngs_R1.fastq  MG24-2-ITS3mix-ITS4ngs_R1.fastq
3072-2-ITS3mix-ITS4ngs_R2.fastq  3288-1-ITS3mix-ITS4ngs_R2.fastq  MG24-2-ITS3mix-ITS4ngs_R2.fastq
3074-1-ITS3mix-ITS4ngs_R1.fastq  3288-2-ITS3mix-ITS4ngs_R1.fastq  MG39-1-ITS3mix-ITS4ngs_R1.fastq
3074-1-ITS3mix-ITS4ngs_R2.fastq  3288-2-ITS3mix-ITS4ngs_R2.fastq  MG39-1-ITS3mix-ITS4ngs_R2.fastq
3074-2-ITS3mix-ITS4ngs_R1.fastq  5555-1-ITS3mix-ITS4ngs_R1.fastq  MG39-2-ITS3mix-ITS4ngs_R1.fastq
3074-2-ITS3mix-ITS4ngs_R2.fastq  5555-1-ITS3mix-ITS4ngs_R2.fastq  MG39-2-ITS3mix-ITS4ngs_R2.fastq
3092-1-ITS3mix-ITS4ngs_R1.fastq  5555-2-ITS3mix-ITS4ngs_R1.fastq  SK22-1-ITS3mix-ITS4ngs_R1.fastq
3092-1-ITS3mix-ITS4ngs_R2.fastq  5555-2-ITS3mix-ITS4ngs_R2.fastq  SK22-1-ITS3mix-ITS4ngs_R2.fastq
3092-2-ITS3mix-ITS4ngs_R1.fastq  MG07-1-ITS3mix-ITS4ngs_R1.fastq  SK22-2-ITS3mix-ITS4ngs_R1.fastq
3092-2-ITS3mix-ITS4ngs_R2.fastq  MG07-1-ITS3mix-ITS4ngs_R2.fastq  SK22-2-ITS3mix-ITS4ngs_R2.fastq
3140-1-ITS3mix-ITS4ngs_R1.fastq  MG07-2-ITS3mix-ITS4ngs_R1.fastq
3140-1-ITS3mix-ITS4ngs_R2.fastq  MG07-2-ITS3mix-ITS4ngs_R2.fastq
(pipits_env) matt@line:~/Documents/FWS/2019_black_spruce_inventory/data/soil_fungi/raw_fastq$ rename -n 's/-ITS3mix-ITS4ngs//' *
rename(3069-1-ITS3mix-ITS4ngs_R1.fastq, 3069-1_R1.fastq)
rename(3069-1-ITS3mix-ITS4ngs_R2.fastq, 3069-1_R2.fastq)
rename(3069-2-ITS3mix-ITS4ngs_R1.fastq, 3069-2_R1.fastq)
rename(3069-2-ITS3mix-ITS4ngs_R2.fastq, 3069-2_R2.fastq)
rename(3072-1-ITS3mix-ITS4ngs_R1.fastq, 3072-1_R1.fastq)
rename(3072-1-ITS3mix-ITS4ngs_R2.fastq, 3072-1_R2.fastq)
rename(3072-2-ITS3mix-ITS4ngs_R1.fastq, 3072-2_R1.fastq)
rename(3072-2-ITS3mix-ITS4ngs_R2.fastq, 3072-2_R2.fastq)
rename(3074-1-ITS3mix-ITS4ngs_R1.fastq, 3074-1_R1.fastq)
rename(3074-1-ITS3mix-ITS4ngs_R2.fastq, 3074-1_R2.fastq)
rename(3074-2-ITS3mix-ITS4ngs_R1.fastq, 3074-2_R1.fastq)
rename(3074-2-ITS3mix-ITS4ngs_R2.fastq, 3074-2_R2.fastq)
rename(3092-1-ITS3mix-ITS4ngs_R1.fastq, 3092-1_R1.fastq)
rename(3092-1-ITS3mix-ITS4ngs_R2.fastq, 3092-1_R2.fastq)
rename(3092-2-ITS3mix-ITS4ngs_R1.fastq, 3092-2_R1.fastq)
rename(3092-2-ITS3mix-ITS4ngs_R2.fastq, 3092-2_R2.fastq)
rename(3140-1-ITS3mix-ITS4ngs_R1.fastq, 3140-1_R1.fastq)
rename(3140-1-ITS3mix-ITS4ngs_R2.fastq, 3140-1_R2.fastq)
rename(3140-2-ITS3mix-ITS4ngs_R1.fastq, 3140-2_R1.fastq)
rename(3140-2-ITS3mix-ITS4ngs_R2.fastq, 3140-2_R2.fastq)
rename(3202-1-ITS3mix-ITS4ngs_R1.fastq, 3202-1_R1.fastq)
rename(3202-1-ITS3mix-ITS4ngs_R2.fastq, 3202-1_R2.fastq)
rename(3202-2-ITS3mix-ITS4ngs_R1.fastq, 3202-2_R1.fastq)
rename(3202-2-ITS3mix-ITS4ngs_R2.fastq, 3202-2_R2.fastq)
rename(3288-1-ITS3mix-ITS4ngs_R1.fastq, 3288-1_R1.fastq)
rename(3288-1-ITS3mix-ITS4ngs_R2.fastq, 3288-1_R2.fastq)
rename(3288-2-ITS3mix-ITS4ngs_R1.fastq, 3288-2_R1.fastq)
rename(3288-2-ITS3mix-ITS4ngs_R2.fastq, 3288-2_R2.fastq)
rename(5555-1-ITS3mix-ITS4ngs_R1.fastq, 5555-1_R1.fastq)
rename(5555-1-ITS3mix-ITS4ngs_R2.fastq, 5555-1_R2.fastq)
rename(5555-2-ITS3mix-ITS4ngs_R1.fastq, 5555-2_R1.fastq)
rename(5555-2-ITS3mix-ITS4ngs_R2.fastq, 5555-2_R2.fastq)
rename(MG07-1-ITS3mix-ITS4ngs_R1.fastq, MG07-1_R1.fastq)
rename(MG07-1-ITS3mix-ITS4ngs_R2.fastq, MG07-1_R2.fastq)
rename(MG07-2-ITS3mix-ITS4ngs_R1.fastq, MG07-2_R1.fastq)
rename(MG07-2-ITS3mix-ITS4ngs_R2.fastq, MG07-2_R2.fastq)
rename(MG19-1-ITS3mix-ITS4ngs_R1.fastq, MG19-1_R1.fastq)
rename(MG19-1-ITS3mix-ITS4ngs_R2.fastq, MG19-1_R2.fastq)
rename(MG19-2-ITS3mix-ITS4ngs_R1.fastq, MG19-2_R1.fastq)
rename(MG19-2-ITS3mix-ITS4ngs_R2.fastq, MG19-2_R2.fastq)
rename(MG24-1-ITS3mix-ITS4ngs_R1.fastq, MG24-1_R1.fastq)
rename(MG24-1-ITS3mix-ITS4ngs_R2.fastq, MG24-1_R2.fastq)
rename(MG24-2-ITS3mix-ITS4ngs_R1.fastq, MG24-2_R1.fastq)
rename(MG24-2-ITS3mix-ITS4ngs_R2.fastq, MG24-2_R2.fastq)
rename(MG39-1-ITS3mix-ITS4ngs_R1.fastq, MG39-1_R1.fastq)
rename(MG39-1-ITS3mix-ITS4ngs_R2.fastq, MG39-1_R2.fastq)
rename(MG39-2-ITS3mix-ITS4ngs_R1.fastq, MG39-2_R1.fastq)
rename(MG39-2-ITS3mix-ITS4ngs_R2.fastq, MG39-2_R2.fastq)
rename(SK22-1-ITS3mix-ITS4ngs_R1.fastq, SK22-1_R1.fastq)
rename(SK22-1-ITS3mix-ITS4ngs_R2.fastq, SK22-1_R2.fastq)
rename(SK22-2-ITS3mix-ITS4ngs_R1.fastq, SK22-2_R1.fastq)
rename(SK22-2-ITS3mix-ITS4ngs_R2.fastq, SK22-2_R2.fastq)
(pipits_env) matt@line:~/Documents/FWS/2019_black_spruce_inventory/data/soil_fungi/raw_fastq$ rename 's/-ITS3mix-ITS4ngs//' *
(pipits_env) matt@line:~/Documents/FWS/2019_black_spruce_inventory/data/soil_fungi/raw_fastq$ ls
3069-1_R1.fastq  3074-1_R2.fastq  3140-2_R1.fastq  3288-2_R2.fastq  MG19-1_R1.fastq  MG39-1_R2.fastq
3069-1_R2.fastq  3074-2_R1.fastq  3140-2_R2.fastq  5555-1_R1.fastq  MG19-1_R2.fastq  MG39-2_R1.fastq
3069-2_R1.fastq  3074-2_R2.fastq  3202-1_R1.fastq  5555-1_R2.fastq  MG19-2_R1.fastq  MG39-2_R2.fastq
3069-2_R2.fastq  3092-1_R1.fastq  3202-1_R2.fastq  5555-2_R1.fastq  MG19-2_R2.fastq  SK22-1_R1.fastq
3072-1_R1.fastq  3092-1_R2.fastq  3202-2_R1.fastq  5555-2_R2.fastq  MG24-1_R1.fastq  SK22-1_R2.fastq
3072-1_R2.fastq  3092-2_R1.fastq  3202-2_R2.fastq  MG07-1_R1.fastq  MG24-1_R2.fastq  SK22-2_R1.fastq
3072-2_R1.fastq  3092-2_R2.fastq  3288-1_R1.fastq  MG07-1_R2.fastq  MG24-2_R1.fastq  SK22-2_R2.fastq
3072-2_R2.fastq  3140-1_R1.fastq  3288-1_R2.fastq  MG07-2_R1.fastq  MG24-2_R2.fastq
3074-1_R1.fastq  3140-1_R2.fastq  3288-2_R1.fastq  MG07-2_R2.fastq  MG39-1_R1.fastq
(pipits_env) matt@line:~/Documents/FWS/2019_black_spruce_inventory/data/soil_fungi/raw_fastq$ pispino_createreadpairslist -i raw_fastq -o readpairslist.txt
Generating a read-pair list file from the input directory...
Traceback (most recent call last):
  File "/home/matt/miniconda3/envs/pipits_env/bin/pispino_createreadpairslist", line 164, in <module>
    make_read_pairs_list(options)
  File "/home/matt/miniconda3/envs/pipits_env/bin/pispino_createreadpairslist", line 37, in make_read_pairs_list
    for file in os.listdir(options.inputdir):
FileNotFoundError: [Errno 2] No such file or directory: 'raw_fastq'
(pipits_env) matt@line:~/Documents/FWS/2019_black_spruce_inventory/data/soil_fungi/raw_fastq$ cd ..
(pipits_env) matt@line:~/Documents/FWS/2019_black_spruce_inventory/data/soil_fungi$ pispino_createreadpairslist -i raw_fastq -o readpairslist.txt
Generating a read-pair list file from the input directory...
Done - "readpairslist.txt" created
(pipits_env) matt@line:~/Documents/FWS/2019_black_spruce_inventory/data/soil_fungi$ pispino_seqprep -i rawdata -o out_seqprep -l readpairslist.txt
2020-04-23 22:22:54 pispino_seqprep started
2020-04-23 22:22:54 ERROR: Cannot find "rawdata" directory. Ensure you have the correct name of the input directory.
(pipits_env) matt@line:~/Documents/FWS/2019_black_spruce_inventory/data/soil_fungi$ pispino_seqprep -i raw_fastq -o out_seqprep -l readpairslist.txt
2020-04-23 22:23:16 pispino_seqprep started
2020-04-23 22:23:16 Checking listfile
2020-04-23 22:23:16 ... done
2020-04-23 22:23:16 Counting sequences in rawdata
2020-04-23 22:23:17 ... number of reads: 748853
2020-04-23 22:23:17 Reindexing forward reads
2020-04-23 22:23:25 ... done
2020-04-23 22:23:25 Reindexing reverse reads
2020-04-23 22:23:32 ... done
2020-04-23 22:23:32 Joining paired-end reads [VSEARCH]
2020-04-23 22:24:09 ... number of joined reads: 710194
2020-04-23 22:24:09 Quality filtering [FASTX]
2020-04-23 22:31:43 ... number of quality filtered reads: 708943
2020-04-23 22:31:43 Converting FASTQ to FASTA [FASTX] (also removing reads with "N" nucleotide if specified with "--FASTX-n")
2020-04-23 22:35:34 ... number of prepped sequences: 708943
2020-04-23 22:35:34 Merging into a single file
2020-04-23 22:35:37 ... done
2020-04-23 22:35:37 Cleaning temporary directory
2020-04-23 22:35:38 ... done
2020-04-23 22:35:38 Done - pispino_seqprep completed (Resulting file: out_seqprep/prepped.fasta)
(pipits_env) matt@line:~/Documents/FWS/2019_black_spruce_inventory/data/soil_fungi$ pipits_funits -i out_seqprep/prepped.fasta -o out_funits -x ITS2 -v -r
pipits_funits 2.5, the PIPITS Project
https://github.com/hsgweon/pipits
---------------------------------

2020-04-23 22:37:33 pipits_funits started
2020-04-23 22:37:33 Checking input FASTA for illegal characters
2020-04-23 22:37:34 ... done
2020-04-23 22:37:34 Counting input sequences
2020-04-23 22:37:35 ... number of input sequences: 708943
2020-04-23 22:37:35 Dereplicating sequences for efficiency
vsearch v2.14.2_linux_x86_64, 7.8GB RAM, 2 cores
https://github.com/torognes/vsearch

Dereplicating file out_seqprep/prepped.fasta 100%
257686201 nt in 708943 seqs, min 112, max 530, avg 363
Sorting 100%
328449 unique sequences, avg cluster 2.2, median 1, max 3136
Writing output file 100%
Writing uc file, first part 100%
Writing uc file, second part 100%
2020-04-23 22:37:41 ... done
2020-04-23 22:37:41 Counting dereplicated sequences
2020-04-23 22:37:41 ... number of dereplicated sequences: 328449
2020-04-23 22:37:41 Extracting ITS2 from sequences [ITSx]
ITSx -- Identifies ITS sequences and extracts the ITS region
by Johan Bengtsson-Palme et al., University of Gothenburg
Version: 1.1b1
-----------------------------------------------------------------
Thu Apr 23 22:37:41 2020 : Preparing HMM database (should be quick)...
Thu Apr 23 22:37:42 2020 : Checking and handling input sequence data (should not take long)...
Thu Apr 23 22:37:50 2020 : Comparing sequences to HMM database (this may take a long while)...
```
