
## Tuesday, March 17

To do:

* *Refuge Notebook* catch-up.
* Early adopter data management homework.
* Blackfish diet analysis.
* Sandpiper Lake *Elodea* project PUPs.

I fiddled around with the first pair of FASTQ files I had uploaded to mBRAVE. I failed to figure out how to merge the paired FASTQ files. Maybe this functionality is not fully available yet on mBRAVE. I saw that Braukmann et al. [-@braukmann_metabarcoding_2019] used QIIME to pair all of their sequences before uploading them to mBRAVE. Stolz [-@stolz_nestling_2019] used mBRAVE, but with single-end reads. I could not find any other references where methods using mBRAVE were detailed. There appears to be no help, guide, or manual on mBRAVE right now.

I fiddled around with parameters for a while. Selecting "MERGED PAIRED END READS" turned paired end merging on. I am not sure about the trimming and masking. I turned primer masking on and selected trimming the first 26 bp, both of which might remove the primers. Run parameters are included below.

1. Trimming\
Trim Front: 26 bp\
Trim End: 0 bp\
Trim Length: 500 bp\
Primer Masking: On
2. Filtering\
Min QV: 10 qv\
Min Length: 150 bp\
Max Bases with Low QV ( <20): 4.0%\
Max Bases with Ultra Low QV ( <10): 1.0%
3. Other Parameters\
Pre-Clustering Threshold: None\
ID Distance Threshold: 3.0%\
Exclude From OTU Threshold: 3.0%\
Minimum OTU Size: 1OTU Threshold: 0.0%
4. Paired End (optional - Illumina instruments only)\
Paired End Merging: On\
Assembler Min Overlap: 25 bp\
Assembler Max Substitution: 5 bp 

That failed to run on the first filtering steps. I turned primer masking off. This failed, also.

If I cannot get this to work, then I may be trying the [SCVUC metabarcode pipeline](https://github.com/Hajibabaei-Lab/SCVUC_COI_metabarcode_pipeline) used by Hajibabaei et al. [-@hajibabaei_coi_2019b]. This will require some setup to make this happen on Yeti.

For the data managmenet early adopters' homework, I downloaded the directory structure template from McCrea, which generally follows the structure described by the Alaska Region Data Stewardship Team [-@ak_region_data_stewardship_team_alaska_2020].

I worked on organizing files and authoring metadata for the snowshoe hare project, part of the data management homework.

```r
## Assemble plot location data.
data_dir <- "../data/raw_data/geodata"
file_list <- dir(data_dir)

## Select only .txt files.
file_list <- file_list[grepl("\\.txt", file_list)]

plot_data <- read.csv(paste0(data_dir, "/", file_list[1]), stringsAsFactors=FALSE)

dim(plot_data)
[1] 50  3
## Whoa, there should be 49 rows.
dim(unique(plot_data))
[1] 50  3
## All rows are unique.
length(levels(as.factor(plot_data$Ident)))
[1] 49 ## The number of unique labels is right.

plot_data <- plot_data[order(plot_data$Ident),]
plot_data

## There are two locations for plot CF-43.

## Plotting.
plot(plot_data$Longitude, plot_data$Latitude)
text(plot_data$Longitude, 
 plot_data$Latitude,
 labels=plot_data$Ident
 )
## One of the CF-43's is over by CF-49 far to the west of where it should be.

## Removing the western CF-43.
sl <- (plot_data$Ident=="CF-43") & (plot_data$Longitude < -150.603)
plot_data <- plot_data[!sl,]

## Plotting.
plot(plot_data$Longitude, plot_data$Latitude)
text(plot_data$Longitude, 
 plot_data$Latitude,
 labels=plot_data$Ident
 )

plot_data$grid_name <- gsub("\\.txt", "", gsub("_", " ", file_list[1]))

## Now loop through the rest of the files.
for (this_grid in 2:length(file_list))
 {
 plot_data_temp <- read.csv(paste0(data_dir, "/", file_list[this_grid]),
  stringsAsFactors=FALSE
  )
 plot_data_temp <- plot_data_temp[order(plot_data_temp$Ident),]
 plot_data_temp$grid_name <- gsub("\\.txt", "", gsub("_", " ", file_list[this_grid]))
 plot_data <- rbind(plot_data, plot_data_temp)
 }
 
names(plot_data)[1:3] <- c("plot_name", "longitude", "latitude")
plot_data <- plot_data[,c(4,1,3,2)]

## Now it is time to start reorganizing the observation data.

## Starting with the first sheet, Swanson River.
obs <- read.csv("../data/raw_data/observations/2020-03-17-2151_Swanson_River.csv", stringsAsFactors=FALSE)
obs$plot_name <- paste0("SR-", obs$plot_name)
obs$habitat <- tolower(obs$habitat)

## I will join this to the plot data later.
plot_hab <- obs[,c(1:2)]

obs <- obs[,c(1,3:38)]

library(reshape2)
obs <- melt(obs, id.vars="plot_name")
names(obs)[2:3] <- c("year", "pellet_count")
obs$year <- as.character(obs$year)
obs$year <- gsub("X", "", obs$year)
obs$year <- as.integer(obs$year)

obs_all <- obs

## Oil field.
obs <- read.csv("../data/raw_data/observations/2020-03-17-2218_Oil_field.csv", stringsAsFactors=FALSE)
obs$plot_name <- paste0("OF-", obs$plot_name)
obs$habitat <- tolower(obs$habitat)

## For joining to the plot data later.
plot_hab <- rbind(plot_hab, obs[,c(1:2)])

obs <- obs[,c(1,3:ncol(obs))]

obs <- melt(obs, id.vars="plot_name")
names(obs)[2:3] <- c("year", "pellet_count")
obs$year <- as.character(obs$year)
obs$year <- gsub("X", "", obs$year)
obs$year <- as.integer(obs$year)

obs_all <- rbind(obs_all, obs)

## Campfire.
obs <- read.csv("../data/raw_data/observations/2020-03-17-2229_Campfire.csv", stringsAsFactors=FALSE)
obs$plot_name <- paste0("CF-", obs$plot_name)
obs$habitat <- tolower(obs$habitat)

## For joining to the plot data later.
plot_hab <- rbind(plot_hab, obs[,c(1:2)])

obs <- obs[,c(1,3:ncol(obs))]

obs <- melt(obs, id.vars="plot_name")
names(obs)[2:3] <- c("year", "pellet_count")
obs$year <- as.character(obs$year)
obs$year <- gsub("X", "", obs$year)
obs$year <- as.integer(obs$year)

obs_all <- rbind(obs_all, obs)

## Skilak.
obs <- read.csv("../data/raw_data/observations/2020-03-18-0729_Skilak.csv", stringsAsFactors=FALSE)
obs$plot_name <- paste0("SK-", obs$plot_name)
obs$habitat <- tolower(obs$habitat)

## For joining to the plot data later.
plot_hab <- rbind(plot_hab, obs[,c(1:2)])

obs <- obs[,c(1,3:ncol(obs))]

obs <- melt(obs, id.vars="plot_name")
names(obs)[2:3] <- c("year", "pellet_count")
obs$year <- as.character(obs$year)
obs$year <- gsub("X", "", obs$year)
obs$year <- as.integer(obs$year)

obs_all <- rbind(obs_all, obs)

## Funny River.
obs <- read.csv("../data/raw_data/observations/2020-03-18-0734_Funny_River.csv", stringsAsFactors=FALSE)
obs$plot_name <- paste0("FR-", obs$plot_name)
obs$habitat <- tolower(obs$habitat)

## For joining to the plot data later.
plot_hab <- rbind(plot_hab, obs[,c(1:2)])

obs <- obs[,c(1,3:ncol(obs))]

obs <- melt(obs, id.vars="plot_name")
names(obs)[2:3] <- c("year", "pellet_count")
obs$year <- as.character(obs$year)
obs$year <- gsub("X", "", obs$year)
obs$year <- as.integer(obs$year)

obs_all <- rbind(obs_all, obs)

## Funny River New.
obs <- read.csv("../data/raw_data/observations/2020-03-18-0738_Funny_River_New.csv", stringsAsFactors=FALSE)
obs$plot_name <- paste0("FRN-", obs$plot_name)
obs$habitat <- tolower(obs$habitat)

plot_hab <- rbind(plot_hab, obs[,c(1:2)])

obs <- obs[,c(1,3:ncol(obs))]

obs <- melt(obs, id.vars="plot_name")
names(obs)[2:3] <- c("year", "pellet_count")
obs$year <- as.character(obs$year)
obs$year <- gsub("X", "", obs$year)
obs$year <- as.integer(obs$year)

obs_all <- rbind(obs_all, obs)

## Skilak New.
obs <- read.csv("../data/raw_data/observations/2020-03-18-0742_Skilak_New.csv", stringsAsFactors=FALSE)
obs$plot_name <- paste0("SKN-", obs$plot_name)
obs$habitat <- tolower(obs$habitat)

plot_hab <- rbind(plot_hab, obs[,c(1:2)])

obs <- obs[,c(1,3:ncol(obs))]

obs <- melt(obs, id.vars="plot_name")
names(obs)[2:3] <- c("year", "pellet_count")
obs$year <- as.character(obs$year)
obs$year <- gsub("X", "", obs$year)
obs$year <- as.integer(obs$year)

obs_all <- rbind(obs_all, obs)

## Checking before joining...
levels(as.factor(plot_hab$habitat))
[1] "black spruce" "mixed mature" "reburn mixed"
## Wow, that is really clean!

## Now conforming some of the plot names.
plot_data$plot_name <- gsub("FRB ", "FRN-", plot_data$plot_name)
plot_data$plot_name <- gsub("FRN-0", "FRN-", plot_data$plot_name)

plot_data_2 <- merge(x=plot_data, y=plot_hab, all.x=TRUE)

## Saving here.
write.csv(plot_data_2, "../data/final_data/geodata/2020-03-18_snowshoe_hare_plot_data.csv", row.names=FALSE)

write.csv(obs_all, "../data/final_data/observations/2020-03-18_snowshoe_hare_pellet_counts.csv", row.names=FALSE)
```



 


