
## Thursday, April 23

Last night I downloaded and installed Miniconda 3 and PIPITS 2.5.

To do:

* Slikok occupancy?
* Review Rob's pike treatment plan.

I spent much of the morning setting up and testing a PIPITS 2.5 environment [@gweon_pipits_2015] (<https://github.com/hsgweon/pipits>) and working on some tweaks to run it using exact sequence variant methods instead of the OTU clustering method that is the default.

Testing, following directions at <https://github.com/hsgweon/pipits>.

```shell
cd ~/Documents/FWS

wget https://sourceforge.net/projects/pipits/files/PIPITS_TESTDATA/pipits_test.tar.gz -O pipits_test.tar.gz
tar xvfz pipits_test.tar.gz

cd pipits_test
source activate pipits_env
pispino_createreadpairslist -i rawdata -o readpairslist.txt
pispino_seqprep -i rawdata -o out_seqprep -l readpairslist.txt
pipits_funits -i out_seqprep/prepped.fasta -o out_funits -x ITS2 -v -r
pipits_process -i out_funits/ITS.fasta -o out_process -v -r

```

That worked.

Now I would like to use an exact sequence variant filtering method and avoid the OTU clustering.

I looked at how this was done in the SCVUC pipeline
shell:
		"vsearch --cluster_unoise {input} --sizein --sizeout --minsize {config[VSEARCH_DENOISE][minsize]} --centroids {output} --log {log}"
VSEARCH_DENOISE:
    minsize: 3

```shell
vsearch --cluster_unoise out_funits/ITS.fasta --sizein --sizeout --minsize 3 --centroids out_funits/ITS_denoised.fasta
```

That got rid of all the sequences. Trying `--minsize 2`.

```shell
rm out_funits/ITS_denoised.fasta
vsearch --cluster_unoise out_funits/ITS.fasta --sizein --sizeout --minsize 2 --centroids out_funits/ITS_denoised.fasta
```

That also discarded all sequences.

I edited the input FASTA, manually making more copies of one sequence.

```shell
rm out_funits/ITS_denoised.fasta
vsearch --cluster_unoise out_funits/ITS_edited.fasta --sizein --sizeout --minsize 3 --centroids out_funits/ITS_denoised.fasta
```

Oh, I think I see. The `-sizein` and `--sizeout` options are for dereplicated input and output.

```shell
rm out_funits/ITS_denoised.fasta
vsearch --cluster_unoise out_funits/ITS.fasta --minsize 3 --centroids out_funits/ITS_denoised.fasta
```

That still filtered out all sequences. Trying again to make sure I did not mess anything up. Deleted products of ITS and processing steps.

```shell
pipits_funits -i out_seqprep/prepped.fasta -o out_funits -x ITS2 -v -r
```

Ok, I see that the dereplicated file after ITS sequencing is out_funits/intermediate/derep.ITS2.sizefiltered.fasta

```shell
vsearch --cluster_unoise out_funits/intermediate/derep.ITS2.sizefiltered.fasta --sizein --sizeout --minsize 3 --centroids out_funits/ITS_denoised.fasta
```

Need to rereplicate now.

```shell
pipits_rereplicate -i out_funits/ITS_denoised.fasta -o out_funits/ITS.fasta --uc out_funits/intermediate/derep.uc
```
Now trying the rest with no OTU clustering.

```shell
pipits_process -i out_funits/ITS.fasta -o out_process -v -r -d 1.00
```

That worked! Output looked good.

Later in the day, trying this out on 2019 black spruce soil fungal samples.

```shell
cd ~/Documents/FWS/2019_black_spruce_inventory/data/soil_fungi
source activate pipits_env
## renaming files.
cd raw_fastq
rename -n 's/6202-soifun//' *
rename 's/6202-soifun//' *
rename -n 's/-ITS3mix-ITS4ngs//' *
rename 's/-ITS3mix-ITS4ngs//' *
cd ..
pispino_createreadpairslist -i raw_fastq -o readpairslist.txt
pispino_seqprep -i raw_fastq -o out_seqprep -l readpairslist.txt
pipits_funits -i out_seqprep/prepped.fasta -o out_funits -x ITS2 -v -r
```

I worked some on adding geographic extents to the metadata of the snowshow hare data in preparation for the data management early adopters' meeting.

### Data management early adopters' meeting at 13:00

* Still no specific homework.

Transferring 2019 black spruce soil fungi raw metagenomic data from the server over VPN to my laptop took a long time, but it worked in the end. I initiated a PIPITS analysis of this dataset.
