
## Saturday, April 25

```shell
## Denoising.
vsearch --cluster_unoise out_funits/intermediate/derep.ITS2.sizefiltered.fasta --sizein --sizeout --minsize 3 --centroids out_funits/ITS_denoised.fasta
## rereplicate.
pipits_rereplicate -i out_funits/ITS_denoised.fasta -o out_funits/ITS_denoised_rerep.fasta --uc out_funits/intermediate/derep.uc
## That did not work. Trying cluster_unoise on the ITS.fasta file.
vsearch --cluster_unoise out_funits/ITS.fasta --minsize 3 --centroids out_funits/ITS_denoised.fasta
## That did not work.
```

Complete I/O from today:

```shell
    Sat Apr 25 04:39:37 2020 : Fungi analysis of complementary strand finished.
Sat Apr 25 04:39:37 2020 : Analysing results of HMM-scan (this might take quite some time)...
Sat Apr 25 04:49:42 2020 : Extraction finished!
-----------------------------------------------------------------
Thank you for using ITSx!
Please report bugs or unsupported lineages to itsx@microbiology.se

2020-04-25 04:49:45 ... done
2020-04-25 04:49:45 Counting ITS sequences (dereplicated)
2020-04-25 04:49:46 ... number of ITS sequences (dereplicated): 326274
2020-04-25 04:49:46 Sorting by ID
[INFO] read sequences ...
[INFO] 326274 sequences loaded
[INFO] sorting ...
[INFO] output ...
2020-04-25 04:49:51 ... done
2020-04-25 04:49:51 Removing short sequences below < 100bp
vsearch v2.14.2_linux_x86_64, 7.8GB RAM, 2 cores
https://github.com/torognes/vsearch

Reading input file 100%
325804 sequences kept (of which 0 truncated), 470 sequences discarded.
2020-04-25 04:49:52 ... done
2020-04-25 04:49:52 Counting length-filtered sequences (dereplicated)
2020-04-25 04:49:53 ... number of length-filtered sequences (dereplicated): 325804
2020-04-25 04:49:53 Re-inflating sequences
2020-04-25 04:50:08 ... done
2020-04-25 04:50:08 Counting sequences after re-inflation
2020-04-25 04:50:08 ... number of sequences with ITS subregion: 704602
2020-04-25 04:50:08 Done - pipits_funits ended successfully. (Your ITS sequences are "out_funits/ITS.fasta")
2020-04-25 04:50:08 Next step: pipits_process [ Example: pipits_process -i out_funits/ITS.fasta -o pipits_process ]
(pipits_env) matt@line:~/Documents/FWS/2019_black_spruce_inventory/data/soil_fungi$ ls
2020-04-24-0635_io.txt  out_funits  out_seqprep  raw_fastq  readpairslist.txt
(pipits_env) matt@line:~/Documents/FWS/2019_black_spruce_inventory/data/soil_fungi$ vsearch --cluster_unoise out_funits/intermediate/derep.ITS2.sizefiltered.fasta --sizein --sizeout --minsize 3 --centroids out_funits/ITS_denoised.fasta
vsearch v2.14.2_linux_x86_64, 7.8GB RAM, 2 cores
https://github.com/torognes/vsearch

Reading file out_funits/intermediate/derep.ITS2.sizefiltered.fasta 100%  
3733528 nt in 21662 seqs, min 102, max 263, avg 172
minsize 3: 304142 sequences discarded.
Masking 100% 
Sorting by abundance 100%
Counting k-mers 100% 
Clustering 100%  
Sorting clusters 100%
Writing clusters 100% 
Clusters: 917 Size min 3, max 65239, avg 23.6
Singletons: 0, 0.0% of seqs, 0.0% of clusters
(pipits_env) matt@line:~/Documents/FWS/2019_black_spruce_inventory/data/soil_fungi$ pipits_rereplicate -i out_funits/ITS_denoised.fasta -o out_funits/ITS_denoised_rerep.fasta --uc out_funits/intermediate/derep.uc
Error: Size values in headers added by USEARCH != number of sequences in .uc
65239 3136
(pipits_env) matt@line:~/Documents/FWS/2019_black_spruce_inventory/data/soil_fungi$ 
```
