
## Friday, March 20

To do:

* *Refuge Notebook* catch-up.
* Early adopter data management homework.
* Blackfish diet analysis.

I set up a project on Arctos for the blakfish diet work at <http://arctos.database.museum/project/10003367>.

I was able to get the SCVUC pipeline going under linux mint, but there was a problem with the reverse primer. I did overcome this later by reverting to the original reverse primer in the ml-jg script.

I kept running into trouble with running out of memory for the RDP Classifier [@wang_naive_2007]. The system would kill the process after it used all memory. The script asks for 8 Gb of RAM; this computer has 4. I tried modifying the script to ask for only 2 Gb, but then RDP Classifier failed within Java, again running out of memory. I will try repartitioning to add a bunch of swap space on that computer.

### Telecon at 11:00

* Send Mark a list of planned work for next week.
* Put together plans for this field season.

Next week work list:

* Finish blackfish diet analysis to get an article into the upcoming issue of the *Newsletter of the Alaska Entomological Society*.
* Catch up on formatting *Refuge Notebook* articles and posting them to the CMS.
* Work on metadata / next data management steps for the Kenai NWR snowshoe hare dataset as part of work for the regional data management early adopters' team.
* Write article on our finding *Bimastos parvus*, a second native earthworm for Alaska, for the *AKES Newsletter*?
* Get started on Slikok watershed project occupancy analysis?

```
cd /home/matt/SCVUC_COI_metabarcode_pipeline
snakemake --jobs 1 --snakefile snakefile_withPseudogeneFiltering --configfile 2020-03-20-0747_config_ml-jg.yaml

cutadapt: error: Character 'I' in adapter sequence 'TAIACYTCIGGRTGICCRAARAAYCA' is not a valid IUPAC code. Use only characters XACGTURYSWKMBDHVN.

Subbing "D" for "I". See <https://en.wikipedia.org/wiki/Nucleotide>.

snakemake --jobs 1 --snakefile snakefile_withPseudogeneFiltering --configfile 2020-03-20-0747_config_ml-jg.yaml

Fatal error: Unable to read from file (ml-jg/cat.denoised)
Error in job chimera_removal while creating output file ml-jg/cat.denoised.nonchimeras.
RuleException:
CalledProcessError in line 412 of /home/matt/SCVUC_COI_metabarcode_pipeline/snakefile_withPseudogeneFiltering:
Command 'vsearch --uchime3_denovo ml-jg/cat.denoised --sizein --xsize --nonchimeras ml-jg/cat.denoised.nonchimeras --relabel 'Zotu' --log ml-jg/chimeraRemoval.log' returned non-zero exit status 1.
  File "/home/matt/SCVUC_COI_metabarcode_pipeline/snakefile_withPseudogeneFiltering", line 412, in __rule_chimera_removal
  File "/home/matt/miniconda3/envs/SCVUCv4.3/lib/python3.6/concurrent/futures/thread.py", line 56, in run
Removing output files of failed job chimera_removal since they might be corrupted:
ml-jg/cat.denoised.nonchimeras
Will exit after finishing currently running jobs.
Exiting because a job execution failed. Look above for error message

This seems to have had problems with the reverse primer. After trimming the reverse primer the files were quite small.

TGRTTYTTYGGNCAYCCNGARGTNTA ## From original config file
TADACYTCDGGRTGDCCRAARAAYCA ## What I ran last
TAIACYTCIGGRTGICCRAARAAYCA ## From lab
TANACYTCNGGRTGNCCRAARAAYCA ## Going to try this.

stop @ 08:50 start @ 09:10

snakemake --jobs 1 --snakefile snakefile_withPseudogeneFiltering --configfile 2020-03-20-0747_config_ml-jg.yaml

## That did not work, either.
## Switching back to the original.

/bin/bash: line 1:  6375 Killed                  java -Xmx8g -jar /home/matt/SCVUC_COI_metabarcode_pipeline/rdp_classifier_2.12/dist/classifier.jar classify -t /home/matt/SCVUC_COI_metabarcode_pipeline/mydata_trained/rRNAClassifier.properties -o ml-jg/rdp.out.tmp ml-jg/cat.denoised.nonchimeras
Error in job taxonomic_assignment while creating output file ml-jg/rdp.out.tmp.
RuleException:
CalledProcessError in line 445 of /home/matt/SCVUC_COI_metabarcode_pipeline/snakefile_withPseudogeneFiltering:
Command 'java -Xmx8g -jar /home/matt/SCVUC_COI_metabarcode_pipeline/rdp_classifier_2.12/dist/classifier.jar classify -t /home/matt/SCVUC_COI_metabarcode_pipeline/mydata_trained/rRNAClassifier.properties -o ml-jg/rdp.out.tmp ml-jg/cat.denoised.nonchimeras' returned non-zero exit status 137.
  File "/home/matt/SCVUC_COI_metabarcode_pipeline/snakefile_withPseudogeneFiltering", line 445, in __rule_taxonomic_assignment
  File "/home/matt/miniconda3/envs/SCVUCv4.3/lib/python3.6/concurrent/futures/thread.py", line 56, in run
Removing output files of failed job taxonomic_assignment since they might be corrupted:
ml-jg/rdp.out.tmp
Will exit after finishing currently running jobs.
Exiting because a job execution failed. Look above for error message

## Made that jar executable.

## Trying again.
snakemake --jobs 1 --snakefile snakefile_withPseudogeneFiltering --configfile 2020-03-20-0747_config_ml-jg.yaml

Error in job taxonomic_assignment while creating output file ml-jg/rdp.out.tmp.
RuleException:
CalledProcessError in line 445 of /home/matt/SCVUC_COI_metabarcode_pipeline/snakefile_withPseudogeneFiltering:
Command 'java -Xmx8g -jar /home/matt/SCVUC_COI_metabarcode_pipeline/rdp_classifier_2.12/dist/classifier.jar classify -t /home/matt/SCVUC_COI_metabarcode_pipeline/mydata_trained/rRNAClassifier.properties -o ml-jg/rdp.out.tmp ml-jg/cat.denoised.nonchimeras' returned non-zero exit status 137.
  File "/home/matt/SCVUC_COI_metabarcode_pipeline/snakefile_withPseudogeneFiltering", line 445, in __rule_taxonomic_assignment
  File "/home/matt/miniconda3/envs/SCVUCv4.3/lib/python3.6/concurrent/futures/thread.py", line 56, in run
Removing output files of failed job taxonomic_assignment since they might be corrupted:
ml-jg/rdp.out.tmp
Will exit after finishing currently running jobs.
Exiting because a job execution failed. Look above for error message

## Just trying that failed command.
java -Xmx8g -jar /home/matt/SCVUC_COI_metabarcode_pipeline/rdp_classifier_2.12/dist/classifier.jar classify -t /home/matt/SCVUC_COI_metabarcode_pipeline/mydata_trained/rRNAClassifier.properties -o ml-jg/rdp.out.tmp ml-jg/cat.denoised.nonchimeras

## This was killed by the system, I think from using too much memory.

Mar 20 09:59:42 aurochs kernel: [ 8396.151925] Out of memory: Killed process 6840 (java) total-vm:10309908kB, anon-rss:3333148kB, file-rss:0kB, shmem-rss:0kB

## Oh, I see. It is asking for 8 Gb of memory on a 4 Gb memory machine.
-Xmx8g

## Edited snakefile_withPseudogeneFiltering, saved as snakefile_withPseudogeneFiltering_2020-03-20-1141

snakemake --jobs 1 --snakefile snakefile_withPseudogeneFiltering_2020-03-20-1141 --configfile 2020-03-20-0747_config_ml-jg.yaml

That failed, also. This thing is really hungry for memory.

Exception in thread "main" java.lang.OutOfMemoryError: Java heap space
	at java.util.Arrays.copyOf(Arrays.java:3210)
	at java.util.Arrays.copyOf(Arrays.java:3181)
	at java.util.ArrayList.grow(ArrayList.java:261)
	at java.util.ArrayList.ensureExplicitCapacity(ArrayList.java:235)
	at java.util.ArrayList.ensureCapacityInternal(ArrayList.java:227)
	at java.util.ArrayList.add(ArrayList.java:458)
	at edu.msu.cme.rdp.classifier.io.GenusWordProbFileParser.createGenusWordProbList(GenusWordProbFileParser.java:59)
	at edu.msu.cme.rdp.classifier.TrainingInfo.createGenusWordProbList(TrainingInfo.java:127)
	at edu.msu.cme.rdp.classifier.utils.ClassifierFactory.getNonDefaultFactory(ClassifierFactory.java:161)
	at edu.msu.cme.rdp.classifier.utils.ClassifierFactory.getFactory(ClassifierFactory.java:145)
	at edu.msu.cme.rdp.multicompare.MultiClassifier.<init>(MultiClassifier.java:64)
	at edu.msu.cme.rdp.multicompare.MultiClassifier.<init>(MultiClassifier.java:75)
	at edu.msu.cme.rdp.multicompare.Main.main(Main.java:247)
	at edu.msu.cme.rdp.classifier.cli.ClassifierMain.main(ClassifierMain.java:67)

## Ok, I repartitioned the hard drive, making 24 Gb more of swap, so now we have 4 Gb of memory and 26 Gb of swap.

cd SC*
conda activate SCVUCv4.3
snakemake --jobs 1 --snakefile snakefile_withPseudogeneFiltering_2020-03-20-1528 --configfile 2020-03-20-0747_config_ml-jg.yaml

## That worked!
```

