  - [January](#january)
      - [Thursday, January 2](#thursday-january-2)
      - [Friday, January 3](#friday-january-3)
      - [Monday, January 6](#monday-january-6)
      - [Tuesday, January 7](#tuesday-january-7)
      - [Wedesday, January 8](#wedesday-january-8)
      - [Thursday, January 9](#thursday-january-9)
      - [Friday, January 10](#friday-january-10)
      - [Monday, January 13](#monday-january-13)
      - [Tuesday, January 14](#tuesday-january-14)
      - [Wednesday, January 15](#wednesday-january-15)
      - [Thursday, January 16](#thursday-january-16)
      - [Friday, January 17](#friday-january-17)
      - [Monday, January 20](#monday-january-20)
      - [Tuesday, January 21](#tuesday-january-21)
      - [Wednesday, January 22](#wednesday-january-22)
      - [Thursday, January 23](#thursday-january-23)
      - [Friday, January 24](#friday-january-24)
      - [Monday, January 27](#monday-january-27)
      - [Tuesday, January 28](#tuesday-january-28)
      - [Wednesday, January 29](#wednesday-january-29)
      - [Thursday, January 30](#thursday-january-30)
      - [Friday, January 31](#friday-january-31)
  - [February](#february)
      - [Monday, February 3](#monday-february-3)
      - [Tuesday, February 4](#tuesday-february-4)
      - [Wednesday, February 5](#wednesday-february-5)
      - [Thursday, February 6](#thursday-february-6)
      - [Friday, February 7](#friday-february-7)
      - [Saturday, February 8](#saturday-february-8)
      - [Monday, February 10](#monday-february-10)
      - [Tuesday, February 11](#tuesday-february-11)
      - [Wednesday, February 12](#wednesday-february-12)
      - [Thursday, February 13](#thursday-february-13)
      - [Friday, February 14](#friday-february-14)
      - [Monday, February 17](#monday-february-17)
      - [Tuesday, February 18](#tuesday-february-18)
      - [Wednesday, February 19](#wednesday-february-19)
      - [Thursday, February 20](#thursday-february-20)
      - [Friday, February 21](#friday-february-21)
      - [Monday, February 24](#monday-february-24)
      - [Tuesday, February 25](#tuesday-february-25)
      - [Wednesday, February 26](#wednesday-february-26)
      - [Thursday, February 27](#thursday-february-27)
      - [Friday, February 28](#friday-february-28)
  - [March](#march)
      - [Monday, March 2](#monday-march-2)
      - [Tuesday, March 3](#tuesday-march-3)
      - [Wednesday, March 4](#wednesday-march-4)
      - [Thursday, March 5](#thursday-march-5)
      - [Friday, March 6](#friday-march-6)
      - [Sunday, March 8](#sunday-march-8)
      - [Monday, March 9](#monday-march-9)
      - [Tuesday, March 10](#tuesday-march-10)
      - [Thursday, March 12](#thursday-march-12)
      - [Friday, March 13](#friday-march-13)
      - [Sunday, March 15](#sunday-march-15)
      - [Monday, March 16](#monday-march-16)
      - [Tuesday, March 17](#tuesday-march-17)
      - [Wednesday, March 18](#wednesday-march-18)
      - [Thursday, March 19](#thursday-march-19)
      - [Friday, March 20](#friday-march-20)
      - [Saturday, March 21](#saturday-march-21)
      - [Monday, March 23](#monday-march-23)
      - [Tuesday, March 24](#tuesday-march-24)
      - [Wednesday, March 25](#wednesday-march-25)
      - [Thursday, March 26](#thursday-march-26)
      - [Friday, March 27](#friday-march-27)
      - [Saturday, March 28](#saturday-march-28)
      - [Sunday, March 29](#sunday-march-29)
      - [Monday, March 30](#monday-march-30)
      - [Tuesday, March 31](#tuesday-march-31)
  - [April](#april)
      - [Wednesday, April 1](#wednesday-april-1)
      - [Thursday, April 2](#thursday-april-2)
      - [Friday, April 3](#friday-april-3)
      - [Saturday, April 4](#saturday-april-4)
      - [Monday, April 6](#monday-april-6)
      - [Tuesday, April 7](#tuesday-april-7)
      - [Wednesday, April 8](#wednesday-april-8)
      - [Thursday, April 9](#thursday-april-9)
      - [Friday, April 10](#friday-april-10)
      - [Saturday, April 11](#saturday-april-11)
      - [Monday, April 13](#monday-april-13)
      - [Tuesday, April 14](#tuesday-april-14)
      - [Wednesday, April 15](#wednesday-april-15)
      - [Thursday, April 16](#thursday-april-16)
      - [Friday, April 17](#friday-april-17)
      - [Monday, April 20](#monday-april-20)
      - [Tuesday, April 21](#tuesday-april-21)
      - [Wednesday, April 22](#wednesday-april-22)
      - [Thursday, April 23](#thursday-april-23)
      - [Friday, April 24](#friday-april-24)
      - [Saturday, April 25](#saturday-april-25)
      - [Monday, April 27](#monday-april-27)
      - [Tuesday, April 28](#tuesday-april-28)
      - [Wednesday, April 29](#wednesday-april-29)
      - [Thursday, April 30](#thursday-april-30)
  - [May](#may)
      - [Friday, May 1](#friday-may-1)
      - [Monday, May 4](#monday-may-4)
      - [Tuesday, May 5](#tuesday-may-5)
      - [Wednesday, May 6](#wednesday-may-6)
      - [Thursday, May 7](#thursday-may-7)
      - [Friday, May 8](#friday-may-8)
      - [Monday, May 11](#monday-may-11)
      - [Tuesday, May 12](#tuesday-may-12)
      - [Wednesday, May 13](#wednesday-may-13)
      - [Thursday, May 14](#thursday-may-14)
      - [Friday, May 15](#friday-may-15)
      - [Saturday, May 16](#saturday-may-16)
      - [Sunday, May 17](#sunday-may-17)
      - [Monday, May 18](#monday-may-18)
      - [Tuesday, May 19](#tuesday-may-19)
      - [Wednesday, May 20](#wednesday-may-20)
      - [Thursday, May 21](#thursday-may-21)
      - [Tuesday, May 26](#tuesday-may-26)
      - [Wednesday, May 27](#wednesday-may-27)
      - [Thursday, May 28](#thursday-may-28)
      - [Friday, May 29](#friday-may-29)
  - [June](#june)
      - [Monday, June 1](#monday-june-1)
      - [Tuesday, June 2](#tuesday-june-2)
      - [Wednesday, June 3](#wednesday-june-3)
      - [Thursday, June 4](#thursday-june-4)
      - [Friday, June 5](#friday-june-5)
      - [Monday, June 8](#monday-june-8)
      - [Tuesday, June 9](#tuesday-june-9)
      - [Wednesday, June 10](#wednesday-june-10)
      - [Thursday, June 11](#thursday-june-11)
      - [Friday, June 12](#friday-june-12)
      - [Tuesday, June 16](#tuesday-june-16)
      - [Wednesday, June 17](#wednesday-june-17)
      - [Thursday, June 18](#thursday-june-18)
      - [Friday, June 19](#friday-june-19)
      - [Monday, June 22](#monday-june-22)
      - [Tuesday, June 23](#tuesday-june-23)
      - [Wednesday, June 24](#wednesday-june-24)
      - [Thursday, June 25](#thursday-june-25)
      - [Friday, June 26](#friday-june-26)
      - [Monday, June 29](#monday-june-29)
      - [Tuesday, June 30](#tuesday-june-30)
  - [July](#july)
      - [Wednesday, July 1](#wednesday-july-1)
      - [Thursday, July 2](#thursday-july-2)
      - [Monday, July 6](#monday-july-6)
      - [Tuesday, July 7](#tuesday-july-7)
      - [Wednesday, July 8](#wednesday-july-8)
      - [Thursday, July 9](#thursday-july-9)
      - [Friday, July 10](#friday-july-10)
      - [Sunday, July 12](#sunday-july-12)
      - [Monday, July 13](#monday-july-13)
      - [Tuesday, July 14](#tuesday-july-14)
      - [Wednesday, July 15](#wednesday-july-15)
      - [Thursday, July 16](#thursday-july-16)
      - [Monday, July 20](#monday-july-20)
      - [Tuesday, July 21](#tuesday-july-21)
      - [Wednesday, July 22](#wednesday-july-22)
      - [Thursday, July 23](#thursday-july-23)
      - [Friday, July 24](#friday-july-24)
      - [Monday, July 27](#monday-july-27)
      - [Tuesday, July 28](#tuesday-july-28)
      - [Wednesday, July 29](#wednesday-july-29)
  - [August](#august)
      - [Monday, August 3](#monday-august-3)
      - [Tuesday, August 4](#tuesday-august-4)
      - [Wednesday, August 5](#wednesday-august-5)
      - [Thursday, August 6](#thursday-august-6)
      - [Friday, August 7](#friday-august-7)
      - [Monday, August 10](#monday-august-10)
      - [Tuesday, August 11](#tuesday-august-11)
      - [Wednesday, August 12](#wednesday-august-12)
      - [Thursday, August 13](#thursday-august-13)
      - [Friday, August 14](#friday-august-14)
      - [Monday, August 17](#monday-august-17)
      - [Tuesday, August 18](#tuesday-august-18)
      - [Wedneday, August 19](#wedneday-august-19)
      - [Thursday, August 20](#thursday-august-20)
      - [Monday, August 24](#monday-august-24)
      - [Tuesday, August 25](#tuesday-august-25)
      - [Wednesday, August 26](#wednesday-august-26)
      - [Thursday, August 27](#thursday-august-27)
      - [Monday, August 31](#monday-august-31)
  - [September](#september)
      - [Tuesday, September 1](#tuesday-september-1)
      - [Wednesday, September 2](#wednesday-september-2)
      - [Thursday, September 3](#thursday-september-3)
      - [Monday, September 8](#monday-september-8)
      - [Wednesday, September 9](#wednesday-september-9)
      - [Thursday, September 10](#thursday-september-10)
      - [Friday, September 11](#friday-september-11)
      - [Monday, September 14](#monday-september-14)
      - [Tuesday, September 15](#tuesday-september-15)
      - [Wednesday, September 16](#wednesday-september-16)
      - [Thursday, September 17](#thursday-september-17)
      - [Friday, September 18](#friday-september-18)
      - [Monday, September 21](#monday-september-21)
      - [Tuesday, September 22](#tuesday-september-22)
      - [Wednesday, September 23](#wednesday-september-23)
  - [Appendixes](#appendixes)
      - [Occurrence data in Arctos bulkloader
        format](#occurrence-data-in-arctos-bulkloader-format)
      - [LifeScanner vials with missing
        data](#lifescanner-vials-with-missing-data)
      - [Data for fungal specimens to be
        sequenced](#data-for-fungal-specimens-to-be-sequenced)
  - [Bibliography](#bibliography)

# January

## Thursday, January 2

To do

  - *Refuge Notebook* catch-up.
  - ~~Add journal 2019 to Arctos~~.
  - Elodea RFP.
  - Visit site SK03?
  - Pesticide applicator CEUs.
  - Publish LTEMP bird data.
  - Finish sorting 2019 black spruce sweep net samples.

I posted my 2019 work journal to Arctos
(<http://arctos.database.museum/publication/10009014>,
<https://doi.org/10.7299/X7RB74XZ>). I also bulkloaded all [83 Arctos
specimen
citations](http://arctos.database.museum/SpecimenResults.cfm?publication_id=10009014).

I initialized a [journal2020](https://github.com/mlbowser/journal2020)
repository.

I sorted sweep net sample
[KNWR:Ento:11356](http://arctos.database.museum/guid/KNWR:Ento:11356).

Contents:

| identification | count |
| :------------- | ----: |
| Diptera        |    85 |
| Araneae        |    16 |
| Empidoidea     |     9 |
| Psyllidae      |     7 |
| Chironomidae   |     4 |
| Insecta        |     4 |
| Ichneumonoidea |     3 |
| Phoridae       |     3 |
| Culicidae      |     2 |
| Sciaroidea     |     2 |
| Tipulidae      |     2 |
| Psocodea       |     1 |
| Stratiomyidae  |     1 |
| Thysanoptera   |     1 |

I also sorted sweep net sample
[KNWR:Ento:11357](http://arctos.database.museum/guid/KNWR:Ento:11357).

Contents:

| identification        | count |
| :-------------------- | ----: |
| Diptera               |    31 |
| Araneae               |     8 |
| Symphypleona          |     7 |
| Hymenoptera           |     6 |
| Psyllidae             |     6 |
| Hemiptera             |     5 |
| Ichneumonoidea        |     5 |
| Aphididae             |     4 |
| Cicadellidae          |     4 |
| Symphyta              |     4 |
| Arachnida             |     3 |
| Lepidoptera           |     3 |
| *Cixius meridionalis* |     2 |
| Culicidae             |     2 |
| Insecta               |     2 |
| Sciaroidea            |     2 |
| Delphacidae           |     1 |
| Empidoidea            |     1 |
| Phoridae              |     1 |
| Simuliidae            |     1 |

## Friday, January 3

To do

  - *Refuge Notebook* catch-up.
  - Elodea RFP.
  - Visit site SK03?
  - Pesticide applicator CEUs.
  - Publish LTEMP bird data.
  - ~~Finish sorting 2019 black spruce sweep net samples.~~

I sorted sweep net sample
[KNWR:Ento:11358](http://arctos.database.museum/guid/KNWR:Ento:11358).

Contents:

| identification | count |
| :------------- | ----: |
| Aphididae      |    26 |
| Ichneumonoidea |    14 |
| Diptera        |     8 |
| Araneae        |     7 |
| Hybotidae      |     5 |
| Culicidae      |     3 |
| Lepidoptera    |     3 |
| Psocodea       |     3 |
| Sciaroidea     |     3 |
| Arachnida      |     2 |
| Hemiptera      |     2 |
| Hymenoptera    |     2 |
| Insecta        |     2 |
| Miridae        |     2 |
| Simuliidae     |     2 |
| Tabanidae      |     2 |
| Cicadellidae   |     1 |
| Coleoptera     |     1 |
| Elateridae     |     1 |
| Empidoidea     |     1 |
| Entomobryidae  |     1 |
| Symphyta       |     1 |
| Torymidae      |     1 |

I sorted sweep net sample
[KNWR:Ento:11359](http://arctos.database.museum/guid/KNWR:Ento:11359).

Contents:

| identification | count |
| :------------- | ----: |
| Aphididae      |    11 |
| Araneae        |     7 |
| Psocodea       |     6 |
| Diptera        |     4 |
| Culicidae      |     3 |
| Symphyta       |     3 |
| Empidoidea     |     2 |
| Sciaroidea     |     2 |
| Arachnida      |     1 |
| Elateridae     |     1 |
| Hemiptera      |     1 |
| Hybotidae      |     1 |
| Hymenoptera    |     1 |
| Ichneumonoidea |     1 |
| Miridae        |     1 |
| Simuliidae     |     1 |
| *Symphoromyia* |     1 |
| Tabanidae      |     1 |

I sorted sweep net sample
[KNWR:Ento:11360](http://arctos.database.museum/guid/KNWR:Ento:11360).

Contents:

| identification | count |
| :------------- | ----: |
| Araneae        |    11 |
| Psocodea       |     6 |
| Aphididae      |     5 |
| Ichneumonoidea |     5 |
| Diptera        |     3 |
| Hemiptera      |     2 |
| Miridae        |     2 |
| Arachnida      |     1 |
| Cicadellidae   |     1 |
| Hymenoptera    |     1 |
| Thysanoptera   |     1 |
| Torymidae      |     1 |

I sorted sweep net sample
[KNWR:Ento:11361](http://arctos.database.museum/guid/KNWR:Ento:11361),
the last of these sweep net samples\!

Contents:

| identification | count |
| :------------- | ----: |
| Culicidae      |     8 |
| Tabanidae      |     2 |
| Araneae        |     1 |
| Diptera        |     1 |
| Ichneumonoidea |     1 |
| Psocodea       |     1 |

Now I need to fill out RTL’s submission form.

``` r
## Downloaded data from Arctos (<http://arctos.database.museum/SpecimenResults.cfm?project_id=10003271&collection_id=46>).
data1 <- read.csv("ArctosData_BB6BD7A74E.csv", stringsAsFactors=FALSE)

## Now get barcodes.
partdetails <- strsplit(data1$PARTDETAIL, "] ")
barcode <- sapply(partdetails, "[", 2)
barcode <- strsplit(barcode, " \\(")
barcode <- sapply(barcode, "[", 1)
barcode
length(unique(barcode))
[1] 26 ## Good.

## Assemble identifiers into data frame.
data1$barcode <- barcode
identifiers <- data1[,c("GUID", "barcode")]
write.csv(identifiers, "2020-01-03-1341_identifiers.csv", row.names=FALSE)
```

I selected the mlCOIlintF/jgHCO2198
(GGWACWGGWTGAACWGTWTAYCCYCC/TAIACYTCIGGRTGICCRAARAAYCA) primer set based
on the analysis of Hajibabaei et
al. ([2019](#ref-hajibabaei_coi_2019)[a](#ref-hajibabaei_coi_2019)).

## Monday, January 6

To do

  - ~~Ship out 2019 black spruce sweep net samples.~~
  - Get updated prescription for Sandpiper Lake.
  - Find *Refuge Notebook* author for this week.
  - Review Sandpiper Lake MRDG.
  - *Refuge Notebook* catch-up.
  - Sandpiper Elodea RFP.
  - Visit site SK03?
  - Pesticide applicator CEUs.
  - Publish LTEMP bird data.
  - Review Programmatic Environmental Assessment of USFWS Management
    Strategy for Elodea and Other Submersed Aquatic Invasive Plants in
    the Alaska Region

I packaged sweep net samples and drove them to Kenai to ship them out to
RTL Genomics (FedEx tracking number: 813677694168).

Contents:

| GUID                                                                  | barcode      |
| :-------------------------------------------------------------------- | :----------- |
| [KNWR:Ento:11332](http://arctos.database.museum/guid/KNWR:Ento:11332) | UAM100185909 |
| [KNWR:Ento:11333](http://arctos.database.museum/guid/KNWR:Ento:11333) | UAM100185910 |
| [KNWR:Ento:11334](http://arctos.database.museum/guid/KNWR:Ento:11334) | UAM100185911 |
| [KNWR:Ento:11335](http://arctos.database.museum/guid/KNWR:Ento:11335) | UAM100185912 |
| [KNWR:Ento:11336](http://arctos.database.museum/guid/KNWR:Ento:11336) | UAM100185913 |
| [KNWR:Ento:11337](http://arctos.database.museum/guid/KNWR:Ento:11337) | UAM100185914 |
| [KNWR:Ento:11342](http://arctos.database.museum/guid/KNWR:Ento:11342) | UAM100185918 |
| [KNWR:Ento:11343](http://arctos.database.museum/guid/KNWR:Ento:11343) | UAM100185919 |
| [KNWR:Ento:11344](http://arctos.database.museum/guid/KNWR:Ento:11344) | UAM100185920 |
| [KNWR:Ento:11345](http://arctos.database.museum/guid/KNWR:Ento:11345) | UAM100185921 |
| [KNWR:Ento:11346](http://arctos.database.museum/guid/KNWR:Ento:11346) | UAM100185922 |
| [KNWR:Ento:11347](http://arctos.database.museum/guid/KNWR:Ento:11347) | UAM100185923 |
| [KNWR:Ento:11348](http://arctos.database.museum/guid/KNWR:Ento:11348) | UAM100185924 |
| [KNWR:Ento:11349](http://arctos.database.museum/guid/KNWR:Ento:11349) | UAM100185925 |
| [KNWR:Ento:11350](http://arctos.database.museum/guid/KNWR:Ento:11350) | UAM100185926 |
| [KNWR:Ento:11351](http://arctos.database.museum/guid/KNWR:Ento:11351) | UAM100185927 |
| [KNWR:Ento:11352](http://arctos.database.museum/guid/KNWR:Ento:11352) | UAM100185928 |
| [KNWR:Ento:11353](http://arctos.database.museum/guid/KNWR:Ento:11353) | UAM100185929 |
| [KNWR:Ento:11354](http://arctos.database.museum/guid/KNWR:Ento:11354) | UAM100185930 |
| [KNWR:Ento:11355](http://arctos.database.museum/guid/KNWR:Ento:11355) | UAM100185931 |
| [KNWR:Ento:11356](http://arctos.database.museum/guid/KNWR:Ento:11356) | UAM100185932 |
| [KNWR:Ento:11357](http://arctos.database.museum/guid/KNWR:Ento:11357) | UAM100185933 |
| [KNWR:Ento:11358](http://arctos.database.museum/guid/KNWR:Ento:11358) | UAM100185934 |
| [KNWR:Ento:11359](http://arctos.database.museum/guid/KNWR:Ento:11359) | UAM100185935 |
| [KNWR:Ento:11360](http://arctos.database.museum/guid/KNWR:Ento:11360) | UAM100185936 |
| [KNWR:Ento:11361](http://arctos.database.museum/guid/KNWR:Ento:11361) | UAM100185937 |

I chose the *mlCOIlintF*/*jgHCO2198*
(GGWACWGGWTGAACWGTWTAYCCYCC/TAIACYTCIGGRTGICCRAARAAYCA) primer set on
the submission form.

These specimens were added to loan number KNWR-2020.01-RTLGenomics-Ento.

I started communicating and planning for the spring herbicide
application to Sandpiper Lake.

I sorted most of Berlese sample
[KNWR:Ento:11362](http://arctos.database.museum/guid/KNWR:Ento:11362),
but this was difficult. I had a hard time transferring tiny mites from
the sample into a vial without also transferring debris. I did not
finish.

## Tuesday, January 7

To do

  - Get updated prescription for Sandpiper Lake.
  - Write *Refuge Notebook* article for this week.
  - Review Sandpiper Lake MRDG.
  - *Refuge Notebook* catch-up.
  - Sandpiper Elodea RFP.
  - Visit site SK03?
  - Pesticide applicator CEUs.
  - Publish LTEMP bird data.
  - Review Programmatic Environmental Assessment of USFWS Management
    Strategy for Elodea and Other Submersed Aquatic Invasive Plants in
    the Alaska Region

I received the technical review evaluation for the Slikok manuscript.
There are a number of requests for changes to the supplementary files.

I started a *Refuge Notebook* article on *Entomobrya nivalis*. I went on
a short walk looking for *E. nivalis* under bark of birches,
cottonwoods, and spruces. I found other arthropods, but no springtails.

I restarted writing a *Refuge Notebook* article, this time Walter Evans’
1897-1898 agricultural exploration of southern Alaska.

## Wedesday, January 8

To do

  - Submit *Refuge Notebook* article for this week.

I worked on my *Refuge Notebook* article, getting it submitted.

I entered data for *Hypnoidus bicolor* specimen
[KNWR:Ento:11363](http://arctos.database.museum/guid/KNWR:Ento:11363).
There was not a previusly entered georeferenced record from the Refuge.

## Thursday, January 9

Derek had entered the AWQMS (Ambient Water Quality Monitoring System)
dataset into UAMObs:Ento. I think the search URL below pulls these
records.

<https://arctos.database.museum/SpecimenResults.cfm?&entered_by=ffdss&beg_entered_date=2020-01-08&end_entered_date=2020-01-08>

I should check these for new Kenai NWR records.

I started addressing the issues with the supplementary files listed in
the technical review of the Slikok manuscript.

## Friday, January 10

I did some communicating regarding elodea eradication work at Sandpiper
Lake this spring.

I continued addressing issues with the supplementary files listed in the
technical review of the Slikok manuscript.

``` r
## Working on the raw occurrence data.
wd <- "D:/projects/Slikok_watershed/data/work_space/2020-01-10_supplementary_files"
setwd(wd)

od1 <- read.csv("oo_358153.csv", stringsAsFactors=FALSE)

summary(od1$COORDINATEUNCERTAINTYINMETERS)

## Looking at records with the smallest values.
od1[od1$COORDINATEUNCERTAINTYINMETERS==3,]
## Those are earthworms in the 0.25 m2 frames, so that is ok.

## Looking at records with the smallest values.
od1[od1$COORDINATEUNCERTAINTYINMETERS==810,]
## The LOCALITY_REMARKS field actually documents this well.

## Still, I will the number of significant figures to conform to the technical review.
od1$COORDINATEUNCERTAINTYINMETERS <- signif(od1$COORDINATEUNCERTAINTYINMETERS, 2)

levels(as.factor(od1$PARTS))
## Why do some have two values, e.g. "whole organism (fluid-preserved); whole organism (fluid-preserved)"?

od1[od1$PARTS=="whole organism (fluid-preserved); whole organism (fluid-preserved)",]
## That actually looked good. These are specimen lots where some individuals were sent in for sequencing and others were retained. I will just clean this up as requested, though.
od1$PARTS[!od1$PARTS=="media"] <- "whole organism"

## Rounding coordinates to 5 decimal places.
od1$DEC_LAT <- round(od1$DEC_LAT, 5)
od1$DEC_LONG <- round(od1$DEC_LONG, 5)

## Saving here.
write.csv(od1, "2020-01-10-1113_occurrences.csv", row.names=FALSE)

## Now for the analysis dataset.
od1 <- read.csv("oo_359083.csv", stringsAsFactors=FALSE)

od1$COORDINATEUNCERTAINTYINMETERS <- signif(od1$COORDINATEUNCERTAINTYINMETERS, 2)

od1$PARTS[!od1$PARTS=="media"] <- "whole organism"

## Rounding coordinates to 5 decimal places.
od1$DEC_LAT <- round(od1$DEC_LAT, 5)
od1$DEC_LONG <- round(od1$DEC_LONG, 5)

## Saving here.
write.csv(od1, "2020-01-10-1122_analysis_dataset.csv", row.names=FALSE)

## Now for the community dataset.
cd1 <- read.csv("oo_365062.csv", stringsAsFactors=FALSE)

cd1$frequency <- round(cd1$frequency, 2)

write.csv(cd1, "2020-01-10-1154_community_dataset.csv", row.names=FALSE)
```

I also worked on revising the text of the Slikok project manuscript
after receiving many comments on it from one reviewer.

## Monday, January 13

To do:

  - Worm information to ADF\&G.
  - ~~Finish revising and resubmit Slikok manuscript.~~

I finished revising and resubmitted the Slikok manuscript for technical
review.

I worked on putting together information for the ADF\&G comments on a
proposed regulation on *Lumbricus* species worms.

I generated maps of *Lumbricus* species distribution data using
SimpleMappr (Shorthouse [2010](#ref-shorthouse_simplemappr_2010)) and
data from GBIF (GBIF.org
[2020](#ref-gbiforg_gbif_2020-1)[f](#ref-gbiforg_gbif_2020-1),
[2020](#ref-gbiforg_gbif_2020)[g](#ref-gbiforg_gbif_2020)).

![Map of *Lumbricus terrestris* distribution in
Alaska.](2020-01-13-1013_Lumbricus_terrestris_in_Alaska.jpg)  
Map of *Lumbricus terrestris* distribution in Alaska. Locations were
obtained from GBIF (GBIF.org
[2020](#ref-gbiforg_gbif_2020-1)[f](#ref-gbiforg_gbif_2020-1)).

![Map of *Lumbricus rubellus* distribution in
Alaska.](2020-01-13-1014_Lumbricus_rubellus_in_Alaska.jpg)  
Map of *Lumbricus rubellus* distribution in Alaska.

I learned that *Lumbricus castaneus* (Savigny, 1826) is now *Enterion
castaneum Savigny*, 1826 according to GBIF (GBIF Secretariat
[2019](#ref-gbif_secretariat_enterion_2019)).

## Tuesday, January 14

To do:

  - ~~Worm information to ADF\&G.~~
  - ~~Submit Slikok manuscript.~~
  - Review this week’s *Refuge Notebook* article.

The Slikok manuscript made it through the technical review. I submitted
it to the journal.

Tyler sent me more coordinates of *Lumbricus* locations in Alaska, from
which I made updated maps.

I worked on putting together information for the ADF\&G comments on a
proposed regulation on *Lumbricus* species worms, getting a draft sent
at the end of the day.

## Wednesday, January 15

To do:

  - ~~Review and submit this week’s *Refuge Notebook* article.~~
  - ~~Review Sandpiper Lake MRDG.~~

I received some more earthworm references through ILL. I also was asked
for more information on native worms. I revised my comments to ADF\&G
and sent these out.

I edited and submitted this week’s *Refuge Notebook* article.

I reviewed the Sandpiper Lake elodea treatment Minimum Requirements
Decision Guide Workbook.

## Thursday, January 16

I picked up some historical documents from the fisheries office, scanned
them, and filed them. There is a record of elodea from Vogel Lake from
1964. I think this might have been a misidentification of *Hippuris*,
but it is hard to know. A thorough survey was done on Sandpiper Lake,
but no elodea was documented there. Later I returned these documents.

## Friday, January 17

We had a biology staff meeting today, our first since John retired.

Tasks for me:

  - Dawn and I need to plan for post-fire burn severity and maybe
    vegetation sampling within the Swan Lake Fire

  - I need to look into elodea genetics to determine the origin of the
    Sandpiper Lake elodea infestation.

  - In the short term I need to get the elodea funding application done
    and write the *Refuge Notebook* article for next week.

Dawn and I determined that some of the most interesting questions with
the Swan Lake fire have to do with burning of alpine tundra and
reburning of some locations by multiple fires.

I mapped LTEMP sites within the fire perimeter, looking for alpine
tundra sites for which we have pre-fire vegetation data.

  - 3161 - tundra north of Mysery Creek. Kenai NWR did veg work.

  - 3175 - tundra north side of Mystery Hills. Kenai NWR did veg work.

  - 3176 - no photo. Was it ever surveyed? It was sampled by FIA for
    veg. This looks like subalpine hemlock forest. Plants documented
    here by FIA: *Pleurozium schreberi*, *Cornus canadensis* *Empetrum
    nigrum*, *Gymnocarpium dryopteris*, *Lycopodium annotinum*,
    *Lycopodium clavatum*, *Menziesia ferruginea*, *Rubus pedatus*,
    *Sorbus sitchensis*, *Trientalis europaea*, *Tsuga mertensiana*,
    *Vaccinium caespitosum*, and *Vaccinium vitis-idaea*. I checked on
    Google Earth. This was a hemlock stand on the north slope north of
    Fuller Lake.

There were really only those two tundra sites within the fire. How about
the Paez sites (Paez [1991](#ref-paez_alpine_1991))?

I entered the coordinates from the Mystery Hills sites:

| plot | utm\_e |  utm\_n |
| ---: | -----: | ------: |
|   62 | 655073 | 6713896 |
|   63 | 655073 | 6714016 |
|   64 | 655039 | 6713748 |
|   79 | 654989 | 6713820 |
|   80 | 655089 | 6713779 |
|   81 | 655003 | 6714155 |
|  102 | 656300 | 6713517 |
|  103 | 656290 | 6713599 |
|  138 | 656294 | 6713683 |
|  139 | 656303 | 6713772 |
|  140 | 656033 | 6713516 |
|  141 | 656108 | 6713513 |
|  142 | 655901 | 6713514 |

I convereted these to lat/lon.

``` r
wd <- "D:/projects/Fire/2019_Swan_Lake_Fire/work_space/2020-01-17_plot_coordinates"
setwd(wd)

library(maptools)

wgs84 <- "+proj=longlat +ellps=WGS84 +datum=WGS84 +no_defs"
utmz5n <- "+proj=utm +zone=5 +ellps=WGS84 +datum=WGS84 +units=m +no_defs"

data1 <- read.csv("2020-01-17-1447_Paez_Mystery_Hills_locaclities.csv")
coordinates(data1) <- c("utm_e", "utm_n")
proj4string(data1) <- CRS(utmz5n)
data2 <- spTransform(data1, CRS(wgs84))

write.csv(data2, "2020-01-17-1506_Paez_Mystery_Hills_locaclities_WGS84.csv", row.names=FALSE)
```

| plot | lat         | lon           |
| ---: | :---------- | :------------ |
|   62 | 60.53113795 | \-150.1737958 |
|   63 | 60.53221407 | \-150.1737019 |
|   64 | 60.52982385 | \-150.1745303 |
|   79 | 60.53048878 | \-150.1753839 |
|   80 | 60.53008257 | \-150.1735961 |
|   81 | 60.53348754 | \-150.1748672 |
|  102 | 60.52726451 | \-150.1517647 |
|  103 | 60.52800372 | \-150.1518821 |
|  138 | 60.52875543 | \-150.1517431 |
|  139 | 60.52955004 | \-150.1515091 |
|  140 | 60.52735916 | \-150.156624  |
|  141 | 60.52730317 | \-150.1552616 |
|  142 | 60.52739238 | \-150.1590275 |

![Map of the Swan Lake Fire and Carlos Paez’s 1988-1989 vegetation plots
in the Mystery Hills.](2020-01-17-1515_Swan_Lake_Fire_Paez.jpg)  
Map of the Swan Lake Fire and Carlos Paez’s 1988-1989 vegetation plots
in the Mystery Hills.

## Monday, January 20

I started researching for a *Refuge Notebook* article on wildfire in
tundra and the Swan Lake Fire.

## Tuesday, January 21

To do:

  - *Refuge Notebook* article
  - Sandpiper Elodea RFP
  - earthworm correspondence
  - ~~blackfish permit reporting~~

I summarized the work done on blackfish in 2019 and got the required
project summary and data sent in.

Blackfish collected in 2019:

| lat       | lon          | date       | age      | number |
| :-------- | :----------- | :--------- | :------- | -----: |
| 60.56993  | \-151.19294  | 2019-06-27 | juvenile |      1 |
| 60.559862 | \-151.187067 | 2019-08-22 | juvenile |      4 |
| 60.559265 | \-151.187372 | 2019-08-23 | juvenile |      1 |
| 60.568077 | \-151.19013  | 2019-08-23 | adult    |      5 |

For the *Refuge Notebook* article on fire in tundra I checked Langille’s
report (Langille [1904](#ref-langille_proposed_1904)). Although he
talked about fire in Canyon Creek and other places, no mention was made
of fire making it up into the alpine.

I found the mention of fire in tundra in the 2014 Funny River Fire
(Morton [2014](#ref-morton_ghosts_2014)).

The 1991 Pothole Lake Fire burned into the alpine to 1012 m judging from
the fire polygon and digital elevation model. I found the KNWR documents
from the Pothole Lake Fire. This fired did burn into the alpine to 3000
ft. in multiple places.

LTEMP plot 3220 was in the Pothole Lake fire. So was plot 3204. Plot
3204 was burned forest in 2004. Plot 3220 was lush alder,
*Calamagrostis*, and cow parsnip in 2004. Neither was in tundra.

The Funny River Fire made it to about 2100 ft.

The 1950 Big Indian Creek fire burned to about 3000 ft. in the alpine
based on the fire history GIS layer.

Later at home I continued work on this *Refuge Notebook* article.

## Wednesday, January 22

To do:

  - ~~*Refuge Notebook* article.~~
  - Sandpiper Elodea RFP.
  - earthworm correspondence.
  - Take A-110 class.

I looked up the 1950 Big Indian Creek fire in the 1950 annual narrative.
There the fire was reported to be 3-5 acres, much smaller than the
polygon in the GIS layer.

I finished a draft of my *Refuge Notebook* article this morning and got
it sent in to the *Clarion*.

I resumed work on the elodea RFP.

I attended Todd’s brown bag lunch talk on hummingbird banding.

I spent the rest of the day on the Sandpiper Lake elodea RFP.

## Thursday, January 23

To do:

  - ~~Get draft Sandpiper Elodea RFP to Steve.~~
  - earthworm correspondence.
  - Take A-110 class.

I completed a draft Sandpiper Lake elodea RFP.

I found photos of Elizabeth Bella, Donna Handley, and Nate Olson
surveying Vogel Lake using throw rakes to look for *Elodea* on August
14, 2013. None was found.

There has been some population genetics work done on *Elodea* (Kadono et
al. [1997](#ref-kadono_genetic_1997); Gross et al.
[2003](#ref-gross_unterscheidung_2003); Huotari et al.
[2011](#ref-huotari_population_2011); Huotari and Korpelainen
[2013](#ref-huotari_comparative_2013)).

I submitted to Alaska DNR a Permitting Quarantined Species
Transport/Import Request for collecting Elodea specimens from Sandpiper
Lake.

Lisa Saperstein sent me a copy of Lipkin
([2007](#ref-lipkin_investigation_2007)) relating to fire in alpine
tundra.

## Friday, January 24

To do:

  - ~~Help with Sandpiper Lake *Elodea* RFP, if needed.~~
  - Start Sandpiper Lake *Elodea* EA.
  - ~~earthworm correspondence.~~
  - Take A-110 class.
  - *Refuge Notebook* catch-up.

I did a little bit of checklisting work on
[KenaiNWRspecies](https://github.com/mlbowser/KenaiNWRspecies).

I attended the a meeting at the ADF\&G office on the Miller Creek
drainage pike earadication project. Rob Massengill gave me some live
elodea that he had collected in Sandpiper Lake and kept alive in a
container in his office. I learned that Sandpiper Lake is a popular lake
for touch-and-go practice for aircraft from Anchorage.

We submitted the Sandpiper Lake *Elodea* RFP.

Todd and I met with Kristi about Swan Lake Fire suppression repair work
beginning next week. I will be going on on Monday and Wednesday.

I entered data for the *Elodea* specimen from Sandpiper Lake in Arctos
([KNWR:Herb:12443](http://arctos.database.museum/guid/KNWR:Herb:12443)).
I keyed this using the key of Bowmer
([1995](#ref-bowmer_identification_1995)).

![*Elodea* from Sandpiper Lake, specimen
KNWR:Herb:12443.](2020-01-24-1546_Elodea_from_Sandpiper.jpg)  
*Elodea* from Sandpiper Lake, specimen
[KNWR:Herb:12443](http://arctos.database.museum/guid/KNWR:Herb:12443).

## Monday, January 27

To do:

  - ~~Swan Lake Fire Suppression Repair scouting.~~

Kyle McNally and I met at the Division of Forestry at 10:00. From there
we drove out Skilak Lake Road to the Marsh Lake area dozer line. We
walked this line for some distance, sizing up the repair work to be
done. Woody debris was stacked on the west side of the line and it
looked like it should work well to pull it back over the line with the
current snow conditions. We walked only a portion of the line.

There were almost no birches this area. I saw no birch seeds in the
snow.

![Marsh Lake area dozer
line.](2020-01-27-1154_Marsh_Lake_area_dozer_line.jpg)  
Marsh Lake area dozer line. Woody material is stacked in a windrow on
the left (west) side of the line. I have no coordinates for this image
because the camera’s GPS had not acquired a position.

We also hiked the Kenai River to Skilak Road line. The seepage area that
had been a difficulty before is still flowing, but Kyle and I thought
that equipment should be able to get across this without much damage to
the ground because the ground around the seeps appears to be frozen
solidly.

![Seepage area near Kenai River.](2020-01-27-1310_seepage_area.jpg)  
Seepage area near Kenai River at 60.459053 °N, 150.178570 °W.

Repair work does need to be completed beyond the seepage area.

![Dozer line in need of repair near Kenai
River.](2020-01-27-1323_Kenai_river_dozer_line.jpg)  
Dozer line in need of repair near Kenai River at 60.457028 °N,
150.175910 °W.

I enjoyed this day of stomping through the woods.

## Tuesday, January 28

To do:

  - ~~File pictures, etc. from yesterday.~~
  - Sandpiper Lake *Elodea* EA.
  - BDJ Slikok manuscript revision.
  - ~~*Elodea* genetics correspondence.~~
  - ~~Website updates.~~
  - Edit this week’s *Refuge Notebook* article.

I learned this morning that our Slikok manuscript has been accepted into
*Biodiversity Data Journal*. Revisions will need to be completed within
one week.

I made some website updates requested by Mark.

I received ITS sequences from Alaska and GenBank *Elodea* samples. I
quickly submitted these to NGPhylogeny.fr just to see what the phylogeny
would look like. I used the “NGPhylogeny Analyse - FastME/OneClick”
analysis (Criscuolo and Gribaldo [2010](#ref-Criscuolo_2010); Desper and
Gascuel [2002](#ref-Desper_2002); Junier and Zdobnov
[2010](#ref-Junier_2010); Katoh and Standley [2013](#ref-Katoh_2013);
Lefort et al. [2015](#ref-Lefort_2015); Lemoine et al.
[2018](#ref-Lemoine_2018)). This analysis is (temporarily?) available at
<https://ngphylogeny.fr/workspace/history/deb556bc487a96e5>.

![Quick phylogram of *Elodea*
sequences.](2020-01-28-1252_Elodea_phylo.png)  
Quick phylogram of *Elodea* sequences made using NGPhylogeny.fr and
visualized using iTOL (Letunic and Bork
[2019](#ref-Letunic_et_al_2019)).

I worked on editing this week’s *Refuge Notebook* article.

To avoid going crazy from too much looking at a computer, I examined
Anisopodidae specimen
[KNWR:Ento:8334](http://arctos.database.museum/guid/KNWR:Ento:8334).
Keying using McAlpine et al. ([1983](#ref-mcalpine_manual_1983)), p. 311
1 → 2 → *Sylvicola*. Keying using Pratt and Pratt
([1980](#ref-pratt_notes_1980)), p. 87 → 1 → 2 *Sylvicola fuscatus*
(Fabricius, 1775). This is a female.

## Wednesday, January 29

To do:

  - ~~Swan Lake suppression repair meeting.~~
  - Sandpiper Lake *Elodea* EA.
  - BDJ Slikok manuscript revision.
  - ~~Finish editing and submit this week’s *Refuge Notebook* article.~~

I worked on revising this week’s *Refuge Notebook* article, getting it
submitted in the afternoon.

Todd, Kristi, and I drove to the Division of Forestry, where we met with
DOF folks to plan the Swan Lake Fire suppression repair work for the
next few weeks.

I worked on revising the BDJ Slikok manuscript.

## Thursday, January 30

To do:

  - BDJ Slikok manuscript revision.
  - Sandpiper Lake *Elodea* EA.

I worked on the BDJ Slikok manuscript revision. I created a flow chart
as requested by one of the reviewers.

``` r
## Trying DiagrammeR package for making graphs for the Slikok manuscript.
wd <- "D:/projects/Slikok_watershed/data/work_space/2020-01-30_flow_charts"
setwd(wd)

library(DiagrammeR)

mermaid("
graph LR

subgraph Legend
AA(field sampling):::field-->BB(laboratory steps):::knwr
CC(commercial services):::service-->DD(published data):::pub
end

A(vascular plant, bryophyte,<br/>and lichen observations):::field -->C(Arctos<br/>occurrence records):::pub

D(vascular plant<br/>specimens):::field -->E(morphological<br/>identification):::knwr
E-->C

F(bryophyte and lichen<br/>specimens):::field -->G(morphological identification<br/>by Enlichened Consulting):::service
G-->C

H(bird call observations):::field -->C

C-->J(GBIF<br/>occurrence records):::pub

K(worms collected by<br/>hot mustard extraction):::field -- adult Lumbricidae -->M(morphological<br/>identification):::knwr
M-->C

K-- Enchytraeidae and<br/>immature Lumbricidae -->O(COI sequencing<br/>by LifeScanner):::service
O-->C
O-->P(BOLD<br/>occurrence records):::pub
P-->J

Q(sweep net samples of<br/>terrestrial arthropods):::field -->S(High-Throughput Sequencing<br/>by RTL Genomics):::service
S-- raw sequence data -->T(metagenomic analysis):::knwr
T-->C
%% S-- raw sequence data -->U(GenBank<br/>Sequence Read Archive):::pub
%% S-- raw sequence data -->V(Zenodo):::pub

classDef field fill:#9DE093;
classDef knwr fill:#ffffff;
classDef service fill:#FFDB00;
classDef pub fill:#FFD3F8;

style Legend fill:#ffffff,stroke:#000000

linkStyle 0 stroke:#fff,stroke-width:0px;
linkStyle 1 stroke:#fff,stroke-width:0px;

")
```

![Slikok project flowchart for BDJ
manuscript](2020-01-30-1117_flowchart.png)  
Slikok project flowchart for BDJ manuscript.

I pulled blackfish specimens out of the freezer so that Apphia and I can
dissect out stomach contents.

In the afternoon, Apphia and I dissected stomach contents from the
larger blackfish that I had collected on August 23, 2019 (field id:
2019-08-23\_MLB02) (Bowser
[2019](#ref-bowser_work_2019)[a](#ref-bowser_work_2019)). Apphia
dissected out the digestive tracts, determined the sex of each fish, and
helped with the photographs. I transfered the contents of the stomachs
and intestines into vials of propylene glycol by squeezing the contents
out using forceps.

| temporary ID      | total length (mm) | sex    |
| :---------------- | ----------------: | :----- |
| 2020-01-30\_MLB01 |                69 | male   |
| 2020-01-30\_MLB02 |                95 | female |
| 2020-01-30\_MLB03 |                96 | female |
| 2020-01-30\_MLB04 |                75 | male   |
| 2020-01-30\_MLB05 |                88 | female |

## Friday, January 31

To do:

  - ~~BDJ Slikok manuscript revision.~~
  - Enter data from blackfish specimens.
  - Sandpiper Lake *Elodea* EA.

I revised the BDJ manuscript and submitted a draft to the journal.

At 10:30 my supervisors and I met to discuss goals and plans for 2020
and beyond.

Priority projects or me:

  - Perform an occupancy analysis of the Slikok project invertebrate
    data and publish this.
  - Sandpiper Lake *Elodea* eradication.
  - Swan Lake Fire burn severity/vegetation plots.

One of my supervisors wants to see an assessment of the 2004–2006 Long
Term Ecological Monitoring Program, wanting to know what we learned from
it and when—if ever—such an effort should be repeated. I expressed that
we are at least planning on publishing datasets from this project.

My supervisors did think it would be appropriate to collaborate with
ADF\&G to sample invertebrates in the Miller Creek drainage ahead of
expected pesticide applications.

Later at home I added records from State of Alaska Department of
Environmental Conservation ([2005](#ref-state_of_alaska_dec_2005)) to
the Kenai National Wildlife Refuge’s checklist. There were only two
species not already on the Refuge’s checklist: *Caenis youngi* and
*Glyphopsyche irrorata*.

# February

## Monday, February 3

To do:

  - ~~Ship out *Elodea* specimens.~~
  - ~~Ship out blackfish stomach contents.~~
  - Sandpiper Lake *Elodea* EA.

I packaged live Sandpiper Lake *Elodea* specimens to go out this morning
(FedEx tracking \#: 813677694043).

I packaged the five samples of blackfish stomach contents from January
30 and packaged these to go out to RTL Genomics this morning (FedEx
tracking \#: 813677694032).

I worked a little on the the [Kenai National Wildlife Refuge
checklist](https://github.com/mlbowser/KenaiNWRspecies), moving three
species (*Corticaria ferruginea*, *Enicmus mimus*, and *Xylita
laevigata*) to FWSpecies and requested the addition of *Corticarina
minuta*, *Cortinarius lucorum*, and *Stephostethus armatulus* to
FWSpecies’ taxonomy.

I found a data sheet from a 2013 vegetation survey of Vogel Lake. I
intend to enter these data into Arctos, but first I wanted to search for
what had been documented previously in the watershed, so I converted the
HUC12 190203021903 “Miller Creek-Frontal Cook Inlet” watershed. I
simplified this using ArcMap’s Simplify Polygon tool, with a
simplification tolerance of 500 m. I converted this to WKT so I could
run a GBIF search (GBIF.org
[2020](#ref-gbiforg_gbif_2020-2)[h](#ref-gbiforg_gbif_2020-2)).

I finished sorting Berlese sample
[KNWR:Ento:11362](http://arctos.database.museum/guid/KNWR:Ento:11362)
that I had started sorting on January 6.

Contents:

| identification | count |
| :------------- | ----: |
| Lepidoptera    |     1 |
| Collembola     |    23 |
| Coleoptera     |     1 |
| Arachnida      |   260 |
| Geophilomorpha |     2 |
| Lithobiomorpha |     1 |
| Myrmicinae     |     1 |
| Nematoda       |     1 |
| *Cixius*       |     1 |

There still were some minute mites in this sample. I had spent about 3–4
hours on this sample. I may need to use some kind of time limit, maybe
2–3 hours per sample.

I uploaded the 2013 data sheet from Vogel Lake onto Arctos (Bella
[2013](#ref-bella_Vogel_2013)), entered observation data from this data
sheet, and made an [Arctos
project](http://arctos.database.museum/project/10003326) for these data.

## Tuesday, February 4

To do:

  - Sandpiper Lake *Elodea* EA.

I worked a little on the the [Kenai National Wildlife Refuge
checklist](https://github.com/mlbowser/KenaiNWRspecies), moving three
species (*Agoliinus congregatus*, *Dendroides ephemeroides*, and
*Rhizophagus dimidiatus*) to FWSpecies.

The Vogel Lake plant records from 2013 are now records
[KNWRObs:Herb:831](http://arctos.database.museum/guid/KNWRObs:Herb:831)–[KNWRObs:Herb:867](http://arctos.database.museum/guid/KNWRObs:Herb:867).
I updated some of the identifications.

I got started on the Sandpiper Lake *Elodea* Environmental Assessment. I
requested Marchand ([1985](#ref-marchand_oxygen_1985)) through
[ARLIS](https://www.arlis.org/).

## Wednesday, February 5

To do:

  - ~~Edit and submit today’s *Refuge Notebook*.~~
  - Sandpiper Lake *Elodea* EA.

I worked some on the *Elodea* Environmental Assessment.

I attended the monthly Alaska Invasive Species Partnership
teleconference at 09:00 to talk about earthworms.

I attended a meeting about preparing for morel hunters in the Swan Lake
Fire this spring.

I edited and submitted this week’s *Refuge Notebook* article.

Inspired by flipping through Hedlin ([1974](#ref-hedlin_cone_1974)), I
took a short walk on the Keen Eye Trail to collect spruce cones. I found
some under a singe tree at 60.46457 °N, 151.07216 °W.

In one cone there was an orange (Diptera?) larva in the stem in a white,
silken cocoon. This I photographed (frames 659–665) and collected
(LifeScanner vial BOLD-268).

![Larva in pith of spruce cone (vial
BOLD-268).](2020-02-05_larva_BOLD-268.jpg)  
Larva in pith of spruce cone (vial BOLD-268).

In another cone I found smaller, silken cocoons containing orange
immatures (Diptera?). The first I photographed (frames 666–670) and
collected into vial BOLD-8W0.

![Larva in scales of spruce cone (vial
BOLD-8W0).](2020-02-05_larva_BOLD-8W0.jpg)  
Larva in scales of spruce cone (vial BOLD-8W0).

In the same cone I photographed another cocoon (frames 671–676) and
collected it into vial BOLD-QH4.

Also from the same cone I photographed another cocoon (frames 677–681)
and collected it into vial BOLD-BS8.

In most of the cones I examined I found no insects.

## Thursday, February 6

To do:

  - Sandpiper Lake *Elodea* SEA.
  - ~~Help Mark with ROV testing.~~

I worked on the Sandpiper Lake *Elodea* SEA.

In the afternoon, Mark and I drove to Arc Lake, where we cut a hole in
the ice and tested the QYSEA ROV. The ROV worked well, but the lake was
quite dark under the ice so that we could not see much unless the ROV
was on the bottom or just under the surface ice.

![Deploying ROV in Arc Lake.](2020-02-06_ROV.jpg)  
Deploying ROV in Arc Lake.

## Friday, February 7

To do:

  - ~~Sandpiper Lake *Elodea* SEA.~~

I worked on the Sandpiper Lake *Elodea* SEA, getting a draft sent out in
the afternoon\!

It snowed heavily this morning, but it was warm. Todd and I saw a number
of live cluster flies (*Pollenia vagabunda*) on the new snow and beside
the visitor center and headquarters buildings.

At the end of the day I examined specimen
[KNWR:Ento:7705](http://arctos.database.museum/guid/KNWR:Ento:7705), a
colletid bee. The forewing has two submarginal cells, so it should be
*Hylaeus*. The only other genus reported from Alaska is *Colletes*,
which has three submarginal cells (Packer et al.
[2007](#ref-packer_bee_genera_2007)).

## Saturday, February 8

I received information about kokanee in Sandpiper Lake, so I
incorporated this into the draft SEA.

I worked on the [Kenai National Wildlife Refuge’s
checklist](https://github.com/mlbowser/KenaiNWRspecies), editing the
script to make a satifactory DwC-A file.

## Monday, February 10

I worked on the [Kenai National Wildlife Refuge’s
checklist](https://github.com/mlbowser/KenaiNWRspecies), adding a
minimal checklist export.

I sorted Berlese sample
[KNWR:Ento:11364](http://arctos.database.museum/guid/KNWR:Ento:11364). I
cut off sorting at a 2 hour limit. Most material had been sorted. A
small number of Collembola and mites remained and were discarded.

| identification | count |
| :------------- | ----: |
| Myrmicinae     |     9 |
| Lithobiomorpha |     1 |
| Coleoptera     |     2 |
| Araneae        |    12 |
| Collembola     |   110 |
| Insecta        |     2 |
| Arachnida      |   204 |
| Chilopoda      |     2 |
| Hemiptera      |     1 |

I learned today that we had purchased herbicide last fall. I inventoried
our aquatic herbicides. We have 41 pails of SonarOne, (20 lbs. each), a
total of 820 lbs. We have 20 boxes of Littora (5 gal. each), a total of
100 gal. of Littora.

I revised the *Elodea* RFP based on these numbers.

## Tuesday, February 11

To do:

  - ~~*Elodea* meeting.~~
  - ~~Review/edit *Refuge Notebook* article.~~

I sorted Berlese sample
[KNWR:Ento:11365](http://arctos.database.museum/guid/KNWR:Ento:11365),
again cutting off sorting at 2 hr.

| identification | count |
| :------------- | ----: |
| Staphylinidae  |     1 |
| Ichneumonoidea |     1 |
| Insecta        |    11 |
| Arachnida      |   288 |
| Araneae        |     5 |
| Hymenoptera    |     2 |
| Collembola     |    45 |
| Hemiptera      |     9 |
| Thysanoptera   |     1 |

In the middle of the day I met with Katherine, Maura, Colin, and Steve
about Sandpiper *Elodea* permitting, logistics, etc.

I edited this week’s *Refuge Notebook* article.

I started going through the lists of plants from Sandpiper Lake
documented by Jakubas and Firman ([1984](#ref-jakubas_sandpiper_1984))
to add to the Sandpiper Lake SEA.

Species documented in and around Sandpiper Lake by Jakubas and Firman
([1984](#ref-jakubas_sandpiper_1984)) with the species codes that they
used on their map:

| identification                             | code |
| :----------------------------------------- | :--- |
| *Alnus* Mill.                              | A    |
| *Betula neoalaskana* Sarg.                 | MB   |
| *Comarum palustre* L.                      | MFF  |
| *Eleocharis* R.Br.                         | SR   |
| *Equisetum fluviatile* L.                  | HTF  |
| *Iris setosa* Pall. ex Link                | WF   |
| *Myrica gale* L.                           | SG   |
| *Myriophyllum* L.                          | MR   |
| *Nuphar polysepalum* Engelm.               | YPL  |
| *Nymphaea tetragona* Georgi                | DWL  |
| *Picea glauca* (Moench) Voss               | MS   |
| *Picea mariana* Britton, Sterns & Poggenb. | BS   |
| *Potamogeton alpinus* Balb.                | PA   |
| *Potamogeton berchtoldii* Fieb.            | PBr  |
| *Potamogeton foliosus* Raf.                | Pfo  |
| *Potamogeton gramineus* L.                 | PGr  |
| *Potamogeton natans* L.                    | PN   |
| *Potamogeton robbinsii* Oakes              | PRb  |
| *Potamogeton zosteriformis* Fernald        | PZ   |
| *Ranunculus aquatilis* L.                  | WWC  |
| *Schoenoplectus* (Rchb.) Palla             | Blr  |
| *Sparganium angustifolium* Michx.          | BR   |
| *Sparganium minimum* (L.) Fr.              | BRm  |

## Wednesday, February 12

To do:

  - ~~Send off *Refuge Notebook* article.~~
  - Board of Fish meeting arrangements.

Identifications

I sorted Berlese sample
[KNWR:Ento:11366](http://arctos.database.museum/guid/KNWR:Ento:11366).
This was a sparser sample that took only about one hour to process.

| identification | count |
| :------------- | ----: |
| Arachnida      |   166 |
| Araneae        |     6 |
| Collembola     |    22 |

I worked a little on the Integrated Pest Management Plan for Eradicating
Elodea from the Kenai Peninsula, Working Draft 2020.

New molecular identifications of some KNWR specimens were made on BOLD.
I checked these using BOLD’s ID engine.

| GUID                                                                | identification            |
| :------------------------------------------------------------------ | :------------------------ |
| [KNWR:Ento:4382](http://arctos.database.museum/guid/KNWR:Ento:4382) | *Dorylomorpha spinosa*    |
| [KNWR:Ento:5264](http://arctos.database.museum/guid/KNWR:Ento:5264) | *Tomosvaryella sylvatica* |
| [KNWR:Ento:5442](http://arctos.database.museum/guid/KNWR:Ento:5442) | *Dorylomorpha albitarsis* |
| [KNWR:Ento:6768](http://arctos.database.museum/guid/KNWR:Ento:6768) | *Pipunculus hertzogi*     |
| [KNWR:Ento:6769](http://arctos.database.museum/guid/KNWR:Ento:6769) | *Eudorylas slovacus*      |

The identification of specimen
[KNWR:Ento:5264](http://arctos.database.museum/guid/KNWR:Ento:5264) was
changed from *Tomosvaryella kuthyi* to *Tomosvaryella sylvatica*. I
searched on Arctos for additional records. Four records from the Slikok
project were identified as *Tomosvaryella kuthyi*. These were all
identifications from ASV SlikokOtu119, which I ran through BOLD’s ID
Engine. This is *Tomosvaryella sylvatica*. Affected records:
[UAMObs:Ento:239975](http://arctos.database.museum/guid/UAMObs:Ento:239975),
[UAMObs:Ento:240012](http://arctos.database.museum/guid/UAMObs:Ento:240012),
[UAMObs:Ento:240458](http://arctos.database.museum/guid/UAMObs:Ento:240458),
and
[UAMObs:Ento:240561](http://arctos.database.museum/guid/UAMObs:Ento:240561).

Now for *Eudorylas* records. BIN
[BOLD:ACZ4721](http://boldsystems.org/index.php/Public_BarcodeCluster?clusteruri=BOLD:ACZ4721)
is now mostly identified as *Eudorylas slovacus*.

SlikokOtu114 is *Eudorylas slovacus*. This applies to record
[UAMObs:Ento:240642](http://arctos.database.museum/guid/UAMObs:Ento:240642).

SlikokOtu130 is also *Eudorylas slovacus*. This applies to records
[UAMObs:Ento:240440](http://arctos.database.museum/guid/UAMObs:Ento:240440),
[UAMObs:Ento:240007](http://arctos.database.museum/guid/UAMObs:Ento:240007),
[UAMObs:Ento:239943](http://arctos.database.museum/guid/UAMObs:Ento:239943).

SlikokOtu775 is also *Eudorylas slovacus*. This applies to record
[UAMObs:Ento:239323](http://arctos.database.museum/guid/UAMObs:Ento:239323).

SlikokOtu1573 is also *Eudorylas slovacus*. This applies to record
[UAMObs:Ento:238974](http://arctos.database.museum/guid/UAMObs:Ento:238974).

SlikokOtu84 is also *Eudorylas slovacus*, affecting record
[UAMObs:Ento:238742](http://arctos.database.museum/guid/UAMObs:Ento:238742).

SlikokOtu62 is *Eudorylas sabroskyi*. This affects records
[UAMObs:Ento:239100](http://arctos.database.museum/guid/UAMObs:Ento:239100),
[UAMObs:Ento:239116](http://arctos.database.museum/guid/UAMObs:Ento:239116),
and
[UAMObs:Ento:239424](http://arctos.database.museum/guid/UAMObs:Ento:239424).

I added *Tomosvaryella sylvatica* and *Pipunculus hertzogi* to the Kenai
National Wildlife Refuge’s checklist in FWSpecies; the other names are
not currently in FWSpecies’ taxonomy.

I did get today’s *Refuge Notebook* article sent to the
[*Clarion*](https://www.peninsulaclarion.com/).

## Thursday, February 13

To do:

  - Board of Fish meeting arrangements.
  - ~~Call with Tracy~~
  - Botany 2020
  - ~~Draft Sandpiper SEA to DNR~~

I received LifeScanner identifications (but not sequences) for a number
of submitted vials today.

I sorted Berlese sample
[KNWR:Ento:11367](http://arctos.database.museum/guid/KNWR:Ento:11367).
This was the sparsest sample yet and took less than one hour to process.

| identification | count |
| :------------- | ----: |
| Araneae        |     3 |
| Arachnida      |    93 |
| Insecta        |     1 |
| Collembola     |     8 |
| Nematoda       |     1 |

I worked some on revising the Draft Sandpiper SEA, getting it sent out\!

## Friday, February 14

Allie and I drove out to the Marsh Lake dozer line to help with finding
the Marsh Lake Trail where the recent dozer line crosses it. We ended up
just discussing this the State Forestry folks and leaving them a GPS
loaded with the Marsh Lake Trail as a track. We decided that it would be
ok to rerout the Marsh Lake Trail for the short distance where the
original trail was hard to find, aligning it with the current dozer line
for that section.

## Monday, February 17

I received sequences from LifeScanner and from the 2019 black spruce
sweep net samples. I started downloading the sequence data from RTL
Genomics.

Two worm specimens collected by Adrian and Tyler from Pilgrim Hot
Springs are either *Aporrectodea longa* or *Aporrectodea trapezoides*.
Both specimens closely matched multiple sequences identified as both of
these species, so this was a little confusing.

I also received a notification on the Slikok project manuscript. It is
currently being edited.

## Tuesday, February 18

To do:

  - ~~Edit and submit this week’s *Refuge Notebook* article.~~
  - ~~Revise and submit Slikok article.~~

I stopped by the office for a short time. I had received a new package
of worms from Tyler and Kyungsoo.

I edited and sent out this week’s *Refuge Notebook* article.

I revised and re-submitted the Slikok manuscript.

## Wednesday, February 19

I sorted Berlese sample
[KNWR:Ento:11368](http://arctos.database.museum/guid/KNWR:Ento:11368).

Contents:

| identification | count |
| :------------- | ----: |
| Formicidae     |     1 |
| Araneae        |     1 |
| Arachnida      |   187 |
| Collembola     |     4 |
| Staphylindae   |     1 |
| Chilopoda      |     1 |

## Thursday, February 20

To do:

  - Sandpiper Lake snowmachine decision/logistics.
  - Pesticide applicator training.
  - Board of Fish meeting travel arrangements.

I sorted Berlese sample
[KNWR:Ento:11369](http://arctos.database.museum/guid/KNWR:Ento:11369).

Contents:

| identification | count |
| :------------- | ----: |
| Arachnida      |   114 |
| Insecta        |     3 |
| Formicinae     |    11 |
| Animalia       |     1 |
| *Cixius*       |     1 |
| Sciaroidea     |     1 |
| Arthropoda     |     2 |
| Thysanoptera   |     1 |
| Collembola     |     1 |

In this sample was something pale and difficult to identify, labeled as
Animalia in the table above. It might have been a contracted tardigrade,
a flatworm, plant material, or something else. There were also a couple
of pale, elongate arthropods that might have been Collembola.

## Friday, February 21

To do:

  - ~~Board of fish comments?~~
  - ~~Sandpiper Lake snowmachine decision/logistics.~~
  - Board of Fish meeting travel arrangements.

I learned today that Juniki Yukawa and others published a paper
including Dominqiue Collet’s work on *Rabdophaga* willow rosette gall
makers (Sato et al. [2020](#ref-Sato_et_al_Rabdophaga_2020)). I am happy
to see this published.

I composed and submitted comments regarding the use of *Lumbricus*
earthworms as live bait for the upcoming Board of Fisheries meeting.

I sorted Berlese sample
[KNWR:Ento:11370](http://arctos.database.museum/guid/KNWR:Ento:11370).

Contents:

| identification | count |
| :------------- | ----: |
| Collembola     |    29 |
| Araneae        |     8 |
| Hemiptera      |    14 |
| Nematoda       |     2 |
| Arachnida      |   359 |
| Staphylinidae  |     1 |
| Insecta        |     8 |
| Oligochaeta    |     1 |

## Monday, February 24

To do:

  - Board of Fish meeting travel arrangements.
  - ~~Elodea SEA call at 11:00.~~

I revised the Slikok manuscript based on the copy editor’s edits and
comments.

I sorted Berlese sample
[KNWR:Ento:11371](http://arctos.database.museum/guid/KNWR:Ento:11371).

Contents:

| identification | count |
| :------------- | ----: |
| Coleoptera     |     1 |
| Collembola     |    14 |
| Arachnida      |   329 |
| Thysanoptera   |     7 |
| Insecta        |     2 |
| Sciaroidea     |     1 |

I participated in an *Elodea* teleconference regariding permitting, etc.

I sorted Berlese sample
[KNWR:Ento:11372](http://arctos.database.museum/guid/KNWR:Ento:11372).

Contents:

| identification | count |
| :------------- | ----: |
| Arachnida      |   375 |
| Collembola     |    10 |
| Nematoda       |     1 |
| Araneae        |     1 |
| Myrmicinae     |     1 |
| Lepidoptera    |     1 |

Attempting to key a tingid bug in Berlese sample
[KNWR:Ento:11370](http://arctos.database.museum/guid/KNWR:Ento:11370)
just by looking through the vial and using the key of Drake and Lattin
([1963](#ref-drake_american_1963)), p. 334. 1 → 4 → 5 → 6 → *Acalypta
nyctalis* Drake, 1928. This species is known from our area (Drake and
Lattin [1963](#ref-drake_american_1963); Froeschner
[1976](#ref-froeschner_zoogeographic_1976)). I made a new record for
this specimen
([UAMObs:Ento:244398](http://arctos.database.museum/guid/UAMObs:Ento:244398)).

## Tuesday, February 25

To do:

  - Board of Fish meeting travel arrangements.

I started making travel arrangements for the Board of Fisheries meeting.

I sorted Berlese sample
[KNWR:Ento:11373](http://arctos.database.museum/guid/KNWR:Ento:11373).

Contents:

| identification | count |
| :------------- | ----: |
| Arachnida      |    76 |
| Thysanoptera   |     1 |
| Aphididae      |     3 |
| Coleoptera     |     1 |
| Insecta        |     2 |
| Hemiptera      |     2 |
| Collembola     |     7 |
| Araneae        |     1 |

The [KNWR:Env](https://arctos.database.museum/knwr_env) collection was
created today to accomodate environmental samples beginning with soil
samples from the black spruce inventory project.

I formatted and posted the *Refuge Notebook* [article from
December 13, 2019](https://www.fws.gov/uploadedFiles/Region_7/NWRS/Zone_2/Kenai/Sections/What_We_Do/In_The_Community/Refuge_Notebooks/2019_Articles/Refuge_Notebook_v21_n44.pdf).

I edited this week’s *Refuge Notebook* article.

## Wednesday, February 26

To do:

  - ~~Submit *Refuge Notebook* article.~~
  - ~~Board of Fish meeting travel arrangements.~~
  - ~~Swan Lake Fire burn severity/vegetation survey planning.~~

I made final edits to this week’s Refuge Notebook\* article and got it
sent off.

Mark asked me to put together data/maps for planning work within the
2019 Swan Lake Fire, so I did.

``` r
nm <- read.csv("site_names.csv", stringsAsFactors=FALSE)
loc <- read.csv("2018-06-06-1309_points_wgs84.csv", stringsAsFactors=FALSE)
loc2 <- loc[loc$site_name %in% nm$site_name,]
write.csv(loc2, "2020-02-26-1122_Bowser_sites.csv", row.names=FALSE)
```

I started entering plant observation records from Jakubas and Firman
([1984](#ref-jakubas_sandpiper_1984)) (records
[KNWRObs:Herb:868](http://arctos.database.museum/guid/KNWRObs:Herb:868)–[KNWRObs:Herb:875](http://arctos.database.museum/guid/KNWRObs:Herb:875)).

## Thursday, February 27

The Slikok project manuscript was published today (Bowser et al.
[2020](#ref-bowser_et_al_2020)).

To do:

  - ~~Add Arctos specimen citations for Bowser et
    al. ([2020](#ref-bowser_et_al_2020)).~~

<!-- end list -->

``` r
## Preparing a file of specimen citations to upload to Arctos.
## Starting with Suppl. material 6: Raw occurrence data from the Slikok watershed biotic inventory (<https://doi.org/10.3897/BDJ.8.e50124.suppl6>).

setwd("D:/projects/Slikok_watershed/manuscript/2020-02-27_Arctos_citations")
d1 <- read.csv("oo_371432.csv", stringsAsFactors=FALSE)

dim(d1)
[1] 4764   28

levels(as.factor(d1$SCIENTIFIC_NAME))

## Starting citation upload data frame.
d2 <- d1[,c("GUID", "SCIENTIFIC_NAME")]
 
d2$publication_id <- 10009135
d2$made_date <- "2020-02-27"
d2$nature_of_id <- "unknown"
d2$type_status <- "voucher"
d2$USE_EXISTING_ACCEPTED_ID <- 0
d2$accepted_id_fg <- 0
d2$use_pub_authors <- 1

## The tough part is that I need to convert some of the identifications to the "A {string}" formula used by Arctos.

## First the names containing "sp."
slsp <- grepl(" sp. ", d2$SCIENTIFIC_NAME)
nsp <- d2$SCIENTIFIC_NAME[slsp]
nsp0 <- strsplit(nsp, " sp. ")
nsp1 <- sapply(nsp0, "[", 1)
nsp2 <- sapply(nsp0, "[", 2)
nsp3 <- paste0(nsp1, " {", nsp, "}")
## There were some double spaces in there.
nsp3 <- gsub("  ", " ", nsp3)
d2$SCIENTIFIC_NAME[slsp] <- nsp3

## Now I think I saw one cf. without a sp.
slcf <- grepl(" cf. ", d2$SCIENTIFIC_NAME)
## I want to exclude those I have already dealt with.
sum(slcf)
[1] 16
slcf[slsp] <- FALSE
sum(slcf)
[1] 9
levels(as.factor(d2$SCIENTIFIC_NAME[slcf]))
## That looked good.

ncf <- d2$SCIENTIFIC_NAME[slcf]
ncf0 <- strsplit(ncf, " cf. ")
ncf1 <- sapply(ncf0, "[", 1)
ncf2 <- sapply(ncf0, "[", 2)
ncf3 <- paste0(ncf1, " {", ncf, "}")
d2$SCIENTIFIC_NAME[slcf] <- ncf3

## Ok, making a test file to upload.
write.csv(d2[1,], file="slikok_citations_0001-0001.csv", row.names=FALSE)

## That worked. Trying some more.
write.csv(d2[2:10,], file="slikok_citations_0002-0010.csv", row.names=FALSE)

## That worked. Trying some more.
write.csv(d2[11:100,], file="slikok_citations_0011-0100.csv", row.names=FALSE)

## Those A {string} identifications were not handled exaclty as I had expected.
## Record 121, KNWR:Herb:12254, Peltigera conspersa nom. prov., should be a good test case.
write.csv(d2[101:150,], file="slikok_citations_0101-0150.csv", row.names=FALSE)
## That one did not work. Got error: taxon 1 not found.

## Fixing this.
slnp <- grepl("nom. prov.", d2$SCIENTIFIC_NAME)
d2$SCIENTIFIC_NAME[slnp]
## Just one name like this.
d2$SCIENTIFIC_NAME[slnp] <- "Peltigera {Peltigera conspersa nom. prov.}"

## Trying again.
write.csv(d2[101:150,], file="slikok_citations_0101-0150.csv", row.names=FALSE)
# That worked.

write.csv(d2[151:500,], file="slikok_citations_0151-0500.csv", row.names=FALSE)
## That did not load. Had problem with Simulium arcticum complex.

d2$SCIENTIFIC_NAME[d2$SCIENTIFIC_NAME=="Simulium arcticum complex"] <- "Simulium {Simulium arcticum complex}"

## Trying again.
write.csv(d2[151:500,], file="slikok_citations_0151-0500.csv", row.names=FALSE)
## That worked.

write.csv(d2[501:1000,], file="slikok_citations_0501-1000.csv", row.names=FALSE)

write.csv(d2[1001:2000,], file="slikok_citations_1001-2000.csv", row.names=FALSE)

write.csv(d2[2001:3000,], file="slikok_citations_2001-3000.csv", row.names=FALSE)
## That gave me an error: value too large for column "UAM"."CF_TEMP_CITATION"."SCIENTIFIC_NAME" (actual: 67, maximum: 60)

summary(nchar(d2$SCIENTIFIC_NAME)) 
   Min. 1st Qu.  Median    Mean 3rd Qu.    Max. 
    6.0    17.0    20.0    23.8    30.0    67.0 

## What was this?   
d2$SCIENTIFIC_NAME[nchar(d2$SCIENTIFIC_NAME)==67]
[1] "Pachynematus albipennis {Pachynematus albipennis sp. SlikokOtu1704}"

d2[nchar(d2$SCIENTIFIC_NAME)==67,]
## That is UAMObs:Ento:239052.

## Bumping that back to a genus resolution ID.
d2$SCIENTIFIC_NAME[nchar(d2$SCIENTIFIC_NAME)==67] <- "Pachynematus {Pachynematus albipennis sp. SlikokOtu1704}"

write.csv(d2[2001:3000,], file="slikok_citations_2001-3000.csv", row.names=FALSE)
## Got an error with the Simulium venustum complex.

d2$SCIENTIFIC_NAME[d2$SCIENTIFIC_NAME=="Simulium venustum complex"] <- "Simulium {Simulium venustum complex}"

write.csv(d2[2001:3000,], file="slikok_citations_2001-3000.csv", row.names=FALSE)

write.csv(d2[3001:4000,], file="slikok_citations_3001-4000.csv", row.names=FALSE)
## Did not like this one:   Arthonia edgewoodensis ined..

d2$SCIENTIFIC_NAME[d2$SCIENTIFIC_NAME=="Arthonia edgewoodensis ined."] <- "Arthonia {Arthonia edgewoodensis ined.}"

write.csv(d2[3001:4000,], file="slikok_citations_3001-4000.csv", row.names=FALSE)
## Did not like Lecanora allophana forma soralifera.

d2$SCIENTIFIC_NAME[d2$SCIENTIFIC_NAME=="Lecanora allophana forma soralifera"] <- "Lecanora allophana {Lecanora allophana forma soralifera}"

write.csv(d2[3001:4000,], file="slikok_citations_3001-4000.csv", row.names=FALSE)

write.csv(d2[4001:4764,], file="slikok_citations_4001-4764.csv", row.names=FALSE)
## Did not like Lasiomma nr. consobrinum.

d2$SCIENTIFIC_NAME[d2$SCIENTIFIC_NAME=="Lasiomma nr. consobrinum"] <- "Lasiomma {Lasiomma nr. consobrinum}"

write.csv(d2[4001:4764,], file="slikok_citations_4001-4764.csv", row.names=FALSE)

## All done.

## Saving the whole data frame.
write.csv(d2, file="slikok_citations.csv", row.names=FALSE)
```

I sorted Berlese sample
[KNWR:Ento:11374](http://arctos.database.museum/guid/KNWR:Ento:11374).

Contents:

| identification | count |
| :------------- | ----: |
| Arachnida      |   220 |
| Insecta        |     1 |
| Formicidae     |     1 |
| Araneae        |     1 |
| Collembola     |     7 |

I sorted Berlese sample
[KNWR:Ento:11375](http://arctos.database.museum/guid/KNWR:Ento:11375).

Contents:

| identification | count |
| :------------- | ----: |
| Araneae        |     5 |
| Collembola     |    11 |
| Arachnida      |    64 |

This was a sparser sample.

I entered more observation records from Jakubas and Firman
([1984](#ref-jakubas_sandpiper_1984)) (records
[KNWRObs:Herb:876](http://arctos.database.museum/guid/KNWRObs:Herb:876)–[KNWRObs:Herb:882](http://arctos.database.museum/guid/KNWRObs:Herb:882)).

I started reading through Alaska Region Data Stewardship Team
([2019](#ref-ak_region_data_stewardship_team_alaska_2020)[a](#ref-ak_region_data_stewardship_team_alaska_2020),
[2019](#ref-alaska_region_data_stewardship_team_alaska_2020)[b](#ref-alaska_region_data_stewardship_team_alaska_2020)),
required reading for tomorrow’s meeting.

## Friday, February 28

To do:

  - ~~Data Management Early Adopter Discussion and
    [mdEditor](https://www.mdeditor.org/) Training at 10:00~~

Potential projects to use as a test case:

  - 2019 black spruce bioinventory
  - 2019 Alaska blackfish diet study
  - 2020 Sandpiper Lake *Elodea* eradication
  - 2020 Swan Lake Fire burn severity and vegetation work

I sorted Berlese sample
[KNWR:Ento:11376](http://arctos.database.museum/guid/KNWR:Ento:11376).

Contents:

| identification | count |
| :------------- | ----: |
| Coleoptera     |     2 |
| Arachnida      |   211 |
| Araneae        |     5 |
| Collembola     |    24 |
| Cicadellidae   |     1 |
| Hemiptera      |    13 |
| Insecta        |     1 |

Most of the Hemiptera look like *Arctorthezia*.

Data Management talk/telecon notes/questions:

  - Why a new standard for metadata? What already exists? How does it
    relate to existing standards, e.g. Darwin Core, Dublin Core, RDF
    (Resource Description Framework)

  - Where are these metadata files going?

  - Where are the nicely structured project archives going?

  - Which laws for sharing data?
    
      - 1967 FOIA
      - Open Governemnt Data Act
      - Also AK regional strategic intent

  - Where is the form for the Regional Data Repository?

  - McCrea is my point of contact.

  - What is the ISO metadata standard that is being referring to? See
    <https://ak-region-dst.gitbook.io/alaska-region-mdeditor-interim-user-guide/glossary-of-terms-lcc-metadata-manual#iso>.
    ISO 19110, ISO 19115-1, ISO 19115-2? Answer: It is based off of ISO
    19115-2 or ISO 19115-1, geospatial standards.

![A slide from the Data Management Early Adopter
Discussion.](2020-02-28_data_manamgement_slide.jpg)  
A slide from the Data Management Early Adopter Discussion.

I entered more observation records from Jakubas and Firman
([1984](#ref-jakubas_sandpiper_1984)) (records
[KNWRObs:Herb:883](http://arctos.database.museum/guid/KNWRObs:Herb:883)–[KNWRObs:Herb:951](http://arctos.database.museum/guid/KNWRObs:Herb:951)).

I talked with McCrea about which project to use for the data management
work. We decided to use the snowshoe hare pellet count survey for this.
I am not excited about this because it is a dataset that is not in the
best shape and it is not one that I have worked with.

# March

## Monday, March 2

To do:

  - Write this week’s *Refuge Notebook* article.
  - ~~Finish arrangements for next week’s Anchorage trip.~~
  - Data Management Early Adopter follow-up. Start preparing hare survey
    data.
  - ~~1:00 pm Swan Lake Fire studies planning meeting.~~
  - Take tissue samples of worm specimens from Tyler.

I sorted Berlese sample
[KNWR:Ento:11377](http://arctos.database.museum/guid/KNWR:Ento:11377).

Contents:

| identification | count |
| :------------- | ----: |
| Arachnida      |   260 |
| Collembola     |    30 |
| Insecta        |     1 |
| Araneae        |     2 |
| Lithobiomorpha |     1 |
| Chilopoda      |     1 |
| Myrmicinae     |     3 |
| Hemiptera      |     3 |

I uploaded citations for Jakubas and Firman
([1984](#ref-jakubas_sandpiper_1984)). When I looked at the map I saw a
couple of missing records, so I added these
([KNWRObs:Herb:952](http://arctos.database.museum/guid/KNWRObs:Herb:952)–[KNWRObs:Herb:953](http://arctos.database.museum/guid/KNWRObs:Herb:953)).

I started work on this week’s *Refuge Notebook* article.

I participated in the planning meeting for burn severity mapping and
invasive species work for the Swan Lake Fire burn area.

``` r
## How many species were documented at site MG28?
## Downloaded analysis dataset.
wd <- "C:/Users/mattbowser/Documents/Refuge_Notebook/2020/submissions/2020-03-06_Refuge_Notebook"
setwd(wd)

d1 <- read.csv("oo_371428.csv", stringsAsFactors=FALSE)

sl <- grepl("site MG28", d1$SPEC_LOCALITY)
d1[sl,]
levels(as.factor(d1$SPECIES[sl]))
## 80 species or BIN resolution identifications were obtained at this site.

```

## Tuesday, March 3

To do:

  - Write this week’s *Refuge Notebook* article.
  - Data managmenet follow-up.
  - Prepare information/comment for next week’s BOF meeting.

I worked some on my *Refuge Notebook* article.

I sorted Berlese sample
[KNWR:Ento:11378](http://arctos.database.museum/guid/KNWR:Ento:11378).

Contents:

| identification | count |
| :------------- | ----: |
| Araneae        |     5 |
| Collembola     |   179 |
| Arachnida      |   333 |
| Insecta        |     3 |
| Oligochaeta    |     7 |
| Nematoda       |     2 |
| Lithobiomorpha |     2 |
| Hemiptera      |     4 |
| Coleoptera     |     2 |
| Hymenoptera    |     1 |
| Staphylinidae  |     1 |

I submitted a Record of General Use to Dan and Katherine for fluridone
applied to Hilda-Seppu Lakes in 2019.

I worked more on my *Refuge Notebook* article.

## Wednesday, March 4

To do:

  - ~~Submit this week’s *Refuge Notebook* article.~~
  - ~~Data managmenet follow-up.~~
  - Prepare information/comment for next week’s BOF meeting.

I finished my *Refuge Notebook* article and got it sent in.

I created an entry for the Kenai National Wildlife Refuge Snowshoe Hare
Grid Surveys project using [mdEditor](https://www.mdeditor.org/) and
sent this to the data management folks.

## Thursday, March 5

To do:

  - Prepare information/comment/images for next week’s BOF meeting.

I sorted Berlese sample
[KNWR:Ento:11379](http://arctos.database.museum/guid/KNWR:Ento:11379).

Contents:

| identification | count |
| :------------- | ----: |
| Hymenoptera    |     2 |
| Araneae        |    25 |
| Insecta        |     4 |
| Collembola     |   158 |
| Arachnida      |   310 |
| Hemiptera      |     3 |
| Phoridae       |     1 |

I worked on my statements for next week’s meeting. I learned that
non-native earthworms can increase nitrogen flux into streams (Pinder
[2013](#ref-pinder_ecology_2013); Costello and Lamberti
[2008](#ref-costello_non-native_2008),
[2009](#ref-costello_biological_2009)).

## Friday, March 6

I prepared for my Anchorage trip this week, including jump-starting and
fueling up the SUV that I am to use.

I added *Cortinarius lucorum*, *Corticarina minuta*, and *Stephostethus
armatulus* to the Kenai National Wildlife Refuge’s checklist on
FWSpecies. I worked a little on trying to fill in missing values in the
taxonomy exported from FWSpecies.

## Sunday, March 8

I rose early, drove to work, switched vehicles, and drove to Anchorage
for the Board of Fisheries meeting. Late in the afternoon I gave my 3
minute testimony on proposal 281 regarding the use of *Lumbricus* worms
as bait in fresh waters.

I gave my three-minute testimony and answered questions from the Board
of Fisheries.

## Monday, March 9

I attended the Board of Fisheries meeting, today the committee of the
whole.

While sitting through discussions of other proposals, I revisited
examining sequences from the *Aporrectodea* specimens from Pilgrim Hot
Springs. These are *Aporrectodea longa* or *Aporrectodea trapezoides*.

Our sequences agreed with sequences from BOLD BIN
[BOLD:AAA6243](http://boldsystems.org/index.php/Public_BarcodeCluster?clusteruri=BOLD:AAA6243),
which is mostly *Aporrectodea trapezoides* (100% to 99.83% similarity to
multiple sequences). There are three sequences of *Aporrectodea longa*
in this BIN. All three of these were collected from Ontario and
identified by Benoit Richard.

The *Aporrectodea longa* sequences from Porco et
al. ([2018](#ref-porco_reference_2018)) are in BIN
[BOLD:AAA2181](http://boldsystems.org/index.php/Public_BarcodeCluster?clusteruri=BOLD:AAA2181).

Our Pilgrim Hot springs sequences aligned with sequences on BOLD from
lineage *Aporrectodea trapezoides* L2 of Martinsson et
al. ([2015](#ref-martinsson_dna-barcoding_2015)) that are on BOLD.
Their *Aporrectodea trapezoides* L1 lineage is in BOLD BIN
[BOLD:AAC5530](http://boldsystems.org/index.php/Public_BarcodeCluster?clusteruri=BOLD:AAC5530).

Ok, our second Pilgrim Hot Springs samples is 100% similar to
*Aporrectodea trapezoides* lineage II, clade H, clone 1 *sensu*
Fernández et al. ([2011](#ref-fernandez_vagrant_2011)). This is the
widespread, peregrine clone.

The meeting ended early at around 15:00, so I checked out of the hotel
and drove home.

## Tuesday, March 10

I attended the Certified Pesticide Applicator workshop at the
Cooperative Extension Service office in Soldotna from 07:30 to 17:00.

I learned that the earthworm proposal did not pass due to enforceability
issues.

## Thursday, March 12

To do

  - ~~Time sheet.~~
  - ~~Get *Refuge Notebook* submitted.~~
  - ~~Travel voucher, etc.~~
  - ~~Data management telecon.~~
  - ~~Send pesticide applicator card info to Casey.~~
  - \~\~*Refuge Notebook* scheduling/correspondence.
  - Deal with Tyler’s earthworms.\~\~
  - Start 2020 *AKES Newsletter.*

I entered data for the Pilgrim Hot Springs worms
([KNWR:Inv:45](http://arctos.database.museum/guid/KNWR:Inv:45) and
[KNWR:Inv:46](http://arctos.database.museum/guid/KNWR:Inv:46)).

``` r
wd <- "D:/projects/earthworms/2020-03-12_Pilgrim_Hot_Springs_worms"
setwd(wd)
## Selecting all sequences from Fernández et al. (2011).
acc <- paste0("JF", 918579:918719)
write(acc, "2020-03-12-1032_Fernandez_accns.txt")
```

I submitted this to [Batch
Entrez](https://www.ncbi.nlm.nih.gov/sites/batchentrez%3E)…

``` r
wd <- "D:/projects/earthworms/2020-03-12_Pilgrim_Hot_Springs_worms"
setwd(wd)
## Selecting all sequences from Martinsson et al. (2015).
acc <- 101:129
acc <- as.character(acc)
acc <- substr(acc, 2, 3)
acc <- paste0("LUSWA0", acc, "-15")
write(acc, "2020-03-12-1057_Martinsson_seqs.txt")
```

### Data management meeting, 11:00

Question: What kinds of contacts should be added in metadata for a
project?

Answer: Anyone involved in the project at any time should be included.
The point of contact should always be current.

Action items:

  - Add more contacts to the snowshoe hare metadata.
  - Organize data into accepted file directory structure.
  - Enter three product records, one of which is a raw data file.

<!-- end list -->

``` r
## I need to clean up those identifiers.
wd <- "D:/projects/earthworms/2020-03-12_Pilgrim_Hot_Springs_worms"
setwd(wd)
library(Biostrings)

fas1 <- readDNAStringSet("sequence_data/Fernandez_COI.fas")
lab <- names(fas1)
labs <- strsplit(lab, " ")
lab <- paste(
 sapply(labs, "[", 1),
 sapply(labs, "[", 2),
 sapply(labs, "[", 3),
 sapply(labs, "[", 4),
 sapply(labs, "[", 5)
 )
lab <- gsub("\\.1", "", lab)

## Saving.
names(fas1) <- lab
writeXStringSet(fas1, "sequence_data/2020-03-12-1248_Fernandez_COI.fas", format="fasta", width=1000)

## Now for the Martinsson sequences.
fas2 <- readDNAStringSet("sequence_data/Martinsson_COI.fas")
lab <- names(fas2)
labs <- strsplit(lab, "\\|")
lab <- paste(
 sapply(labs, "[", 1),
 sapply(labs, "[", 2)
 )
## Saving.
names(fas2) <- lab
writeXStringSet(fas2, "sequence_data/2020-03-12-1253_Martinsson_COI.fas", format="fasta", width=1000)
```

I put all these sequences together into one FASTA file and I submitted
them to NGPhylogeny.fr. I used the “NGPhylogeny Analyse -
FastME/OneClick” analysis. This analysis is (temporarily?) available at
<https://ngphylogeny.fr/workspace/history/4f38834bbcb8bb64>. I rerooted
the tree on iTOL and the tree is available at
<https://itol.embl.de/tree/1641591522291071584068995#>.

![Phylogenetic tree including sequences from the Pilgrim Hot Springs
worms.](2020-03-12_Pilgrim_Hot_Springs_worm_tree.png)  
Phylogenetic tree including COI DNA barcode sequences from Fernández et
al. ([2011](#ref-fernandez_vagrant_2011)), Martinsson et
al. ([2015](#ref-martinsson_dna-barcoding_2015)), and the Pilgrim Hot
Springs worms.

It looks like
[KNWR:Inv:45](http://arctos.database.museum/guid/KNWR:Inv:45) (2-1-1) is
a new clone within *Aporrectodea trapezoides* lineage L2 near clone 1
and [KNWR:Inv:46](http://arctos.database.museum/guid/KNWR:Inv:46)
(1-1-1) is *Aporrectodea trapezoides* clone 1.

I am identifying
[KNWR:Inv:45](http://arctos.database.museum/guid/KNWR:Inv:45) as
*Aporrectodea trapezoides* lineage II, clade H, affinis clone 1. That is
a long name.

I am identifying
[KNWR:Inv:46](http://arctos.database.museum/guid/KNWR:Inv:46) as
*Aporrectodea trapezoides* lineage II, clade H, clone 1.

Both of these are in *Aporrectodea trapezoides* lineage L2 *sensu*
Martinsson et al. ([2015](#ref-martinsson_dna-barcoding_2015)).

## Friday, March 13

I finished data entry for Pilgrim Hot Springs worm specimens I worked
with yesterday.

I sorted Berlese sample
[KNWR:Ento:11380](http://arctos.database.museum/guid/KNWR:Ento:11380).

Contents:

| identification | count |
| :------------- | ----: |
| Chilopoda      |     3 |
| Scarabaeidae   |     2 |
| Arachnida      |   494 |
| Collembola     |    80 |
| Insecta        |     6 |
| Thysanoptera   |     2 |
| Hemiptera      |     2 |
| Araneae        |     2 |

I sorted Berlese sample
[KNWR:Ento:11381](http://arctos.database.museum/guid/KNWR:Ento:11381).

Contents:

| identification | count |
| :------------- | ----: |
| Hemiptera      |     1 |
| Collembola     |    14 |
| Arachnida      |   213 |
| Staphylinidae  |     1 |
| Coleoptera     |     1 |
| Thysanoptera   |     3 |
| Insecta        |     3 |
| Araneae        |     4 |
| Hymenoptera    |     2 |
| Formicidae     |     1 |

## Sunday, March 15

I modified the file structure of the *AKES Newsletter* and a made an
*[AKES Newsletter](https://github.com/mlbowser/AKES_Newsletter)*
repository on GitHub.

## Monday, March 16

To do:

  - *Refuge Notebook* catch-up.
  - ~~Format *AKES Newsletter* meeting article.~~
  - Deal with worm specimens from Tyler.
  - Early adopter data management homework.
  - Slikok occupancy.

I sorted Berlese sample
[KNWR:Ento:11382](http://arctos.database.museum/guid/KNWR:Ento:11382).

Contents:

| identification | count |
| :------------- | ----: |
| Oligochaeta    |    10 |
| Collembola     |   164 |
| Carabidae      |     2 |
| Arachnida      |   332 |
| Araneae        |     4 |
| Thysanoptera   |     4 |
| Staphylinidae  |     1 |
| Insecta        |     4 |

We just received word that we are strongly advised to telework as much
as possible, so I will not be getting any more lab work done for a
while.

I entered data for the earthworm specimens sent to me by Tyler
([KNWR:Inv:47](http://arctos.database.museum/guid/KNWR:Inv:47)–[KNWR:Inv:52](http://arctos.database.museum/guid/KNWR:Inv:52)).

I worked at home in the afternoon. I formatted an initial draft version
of the [meeting
article](https://github.com/mlbowser/AKES_Newsletter/blob/master/2020/01/meeting.tex)
for the *AKES Newsletter*.

This evening I received from RTL Genomics the metagenomic sequence data
from the blackfish gut contents that Apphia and I had sent in on January
30. I set up a project on [mBRAVE](http://www.mbrave.net/) with guesses
at analysis parameters and then I uploaded the first pair of compressed
FASTQ files.

## Tuesday, March 17

To do:

  - *Refuge Notebook* catch-up.
  - Early adopter data management homework.
  - Blackfish diet analysis.
  - Sandpiper Lake *Elodea* project PUPs.

I fiddled around with the first pair of FASTQ files I had uploaded to
mBRAVE. I failed to figure out how to merge the paired FASTQ files.
Maybe this functionality is not fully available yet on mBRAVE. I saw
that Braukmann et al. ([2019](#ref-braukmann_metabarcoding_2019)) used
QIIME to pair all of their sequences before uploading them to mBRAVE.
Stolz ([2019](#ref-stolz_nestling_2019)) used mBRAVE, but with
single-end reads. I could not find any other references where methods
using mBRAVE were detailed. There appears to be no help, guide, or
manual on mBRAVE right now.

I fiddled around with parameters for a while. Selecting “MERGED PAIRED
END READS” turned paired end merging on. I am not sure about the
trimming and masking. I turned primer masking on and selected trimming
the first 26 bp, both of which might remove the primers. Run parameters
are included below.

1.  Trimming  
    Trim Front: 26 bp  
    Trim End: 0 bp  
    Trim Length: 500 bp  
    Primer Masking: On
2.  Filtering  
    Min QV: 10 qv  
    Min Length: 150 bp  
    Max Bases with Low QV ( \<20): 4.0%  
    Max Bases with Ultra Low QV ( \<10): 1.0%
3.  Other Parameters  
    Pre-Clustering Threshold: None  
    ID Distance Threshold: 3.0%  
    Exclude From OTU Threshold: 3.0%  
    Minimum OTU Size: 1OTU Threshold: 0.0%
4.  Paired End (optional - Illumina instruments only)  
    Paired End Merging: On  
    Assembler Min Overlap: 25 bp  
    Assembler Max Substitution: 5 bp

That failed to run on the first filtering steps. I turned primer masking
off. This failed, also.

If I cannot get this to work, then I may be trying the [SCVUC
metabarcode
pipeline](https://github.com/Hajibabaei-Lab/SCVUC_COI_metabarcode_pipeline)
used by Hajibabaei et
al. ([2019](#ref-hajibabaei_coi_2019b)[b](#ref-hajibabaei_coi_2019b)).
This will require some setup to make this happen on Yeti.

For the data managmenet early adopters’ homework, I downloaded the
directory structure template from McCrea, which generally follows the
structure described by the Alaska Region Data Stewardship Team
([2019](#ref-ak_region_data_stewardship_team_alaska_2020)[a](#ref-ak_region_data_stewardship_team_alaska_2020)).

I worked on organizing files and authoring metadata for the snowshoe
hare project, part of the data management homework.

``` r
## Assemble plot location data.
data_dir <- "../data/raw_data/geodata"
file_list <- dir(data_dir)

## Select only .txt files.
file_list <- file_list[grepl("\\.txt", file_list)]

plot_data <- read.csv(paste0(data_dir, "/", file_list[1]), stringsAsFactors=FALSE)

dim(plot_data)
[1] 50  3
## Whoa, there should be 49 rows.
dim(unique(plot_data))
[1] 50  3
## All rows are unique.
length(levels(as.factor(plot_data$Ident)))
[1] 49 ## The number of unique labels is right.

plot_data <- plot_data[order(plot_data$Ident),]
plot_data

## There are two locations for plot CF-43.

## Plotting.
plot(plot_data$Longitude, plot_data$Latitude)
text(plot_data$Longitude, 
 plot_data$Latitude,
 labels=plot_data$Ident
 )
## One of the CF-43's is over by CF-49 far to the west of where it should be.

## Removing the western CF-43.
sl <- (plot_data$Ident=="CF-43") & (plot_data$Longitude < -150.603)
plot_data <- plot_data[!sl,]

## Plotting.
plot(plot_data$Longitude, plot_data$Latitude)
text(plot_data$Longitude, 
 plot_data$Latitude,
 labels=plot_data$Ident
 )

plot_data$grid_name <- gsub("\\.txt", "", gsub("_", " ", file_list[1]))

## Now loop through the rest of the files.
for (this_grid in 2:length(file_list))
 {
 plot_data_temp <- read.csv(paste0(data_dir, "/", file_list[this_grid]),
  stringsAsFactors=FALSE
  )
 plot_data_temp <- plot_data_temp[order(plot_data_temp$Ident),]
 plot_data_temp$grid_name <- gsub("\\.txt", "", gsub("_", " ", file_list[this_grid]))
 plot_data <- rbind(plot_data, plot_data_temp)
 }
 
names(plot_data)[1:3] <- c("plot_name", "longitude", "latitude")
plot_data <- plot_data[,c(4,1,3,2)]

## Now it is time to start reorganizing the observation data.

## Starting with the first sheet, Swanson River.
obs <- read.csv("../data/raw_data/observations/2020-03-17-2151_Swanson_River.csv", stringsAsFactors=FALSE)
obs$plot_name <- paste0("SR-", obs$plot_name)
obs$habitat <- tolower(obs$habitat)

## I will join this to the plot data later.
plot_hab <- obs[,c(1:2)]

obs <- obs[,c(1,3:38)]

library(reshape2)
obs <- melt(obs, id.vars="plot_name")
names(obs)[2:3] <- c("year", "pellet_count")
obs$year <- as.character(obs$year)
obs$year <- gsub("X", "", obs$year)
obs$year <- as.integer(obs$year)

obs_all <- obs

## Oil field.
obs <- read.csv("../data/raw_data/observations/2020-03-17-2218_Oil_field.csv", stringsAsFactors=FALSE)
obs$plot_name <- paste0("OF-", obs$plot_name)
obs$habitat <- tolower(obs$habitat)

## For joining to the plot data later.
plot_hab <- rbind(plot_hab, obs[,c(1:2)])

obs <- obs[,c(1,3:ncol(obs))]

obs <- melt(obs, id.vars="plot_name")
names(obs)[2:3] <- c("year", "pellet_count")
obs$year <- as.character(obs$year)
obs$year <- gsub("X", "", obs$year)
obs$year <- as.integer(obs$year)

obs_all <- rbind(obs_all, obs)

## Campfire.
obs <- read.csv("../data/raw_data/observations/2020-03-17-2229_Campfire.csv", stringsAsFactors=FALSE)
obs$plot_name <- paste0("CF-", obs$plot_name)
obs$habitat <- tolower(obs$habitat)

## For joining to the plot data later.
plot_hab <- rbind(plot_hab, obs[,c(1:2)])

obs <- obs[,c(1,3:ncol(obs))]

obs <- melt(obs, id.vars="plot_name")
names(obs)[2:3] <- c("year", "pellet_count")
obs$year <- as.character(obs$year)
obs$year <- gsub("X", "", obs$year)
obs$year <- as.integer(obs$year)

obs_all <- rbind(obs_all, obs)

## Skilak.
obs <- read.csv("../data/raw_data/observations/2020-03-18-0729_Skilak.csv", stringsAsFactors=FALSE)
obs$plot_name <- paste0("SK-", obs$plot_name)
obs$habitat <- tolower(obs$habitat)

## For joining to the plot data later.
plot_hab <- rbind(plot_hab, obs[,c(1:2)])

obs <- obs[,c(1,3:ncol(obs))]

obs <- melt(obs, id.vars="plot_name")
names(obs)[2:3] <- c("year", "pellet_count")
obs$year <- as.character(obs$year)
obs$year <- gsub("X", "", obs$year)
obs$year <- as.integer(obs$year)

obs_all <- rbind(obs_all, obs)

## Funny River.
obs <- read.csv("../data/raw_data/observations/2020-03-18-0734_Funny_River.csv", stringsAsFactors=FALSE)
obs$plot_name <- paste0("FR-", obs$plot_name)
obs$habitat <- tolower(obs$habitat)

## For joining to the plot data later.
plot_hab <- rbind(plot_hab, obs[,c(1:2)])

obs <- obs[,c(1,3:ncol(obs))]

obs <- melt(obs, id.vars="plot_name")
names(obs)[2:3] <- c("year", "pellet_count")
obs$year <- as.character(obs$year)
obs$year <- gsub("X", "", obs$year)
obs$year <- as.integer(obs$year)

obs_all <- rbind(obs_all, obs)

## Funny River New.
obs <- read.csv("../data/raw_data/observations/2020-03-18-0738_Funny_River_New.csv", stringsAsFactors=FALSE)
obs$plot_name <- paste0("FRN-", obs$plot_name)
obs$habitat <- tolower(obs$habitat)

plot_hab <- rbind(plot_hab, obs[,c(1:2)])

obs <- obs[,c(1,3:ncol(obs))]

obs <- melt(obs, id.vars="plot_name")
names(obs)[2:3] <- c("year", "pellet_count")
obs$year <- as.character(obs$year)
obs$year <- gsub("X", "", obs$year)
obs$year <- as.integer(obs$year)

obs_all <- rbind(obs_all, obs)

## Skilak New.
obs <- read.csv("../data/raw_data/observations/2020-03-18-0742_Skilak_New.csv", stringsAsFactors=FALSE)
obs$plot_name <- paste0("SKN-", obs$plot_name)
obs$habitat <- tolower(obs$habitat)

plot_hab <- rbind(plot_hab, obs[,c(1:2)])

obs <- obs[,c(1,3:ncol(obs))]

obs <- melt(obs, id.vars="plot_name")
names(obs)[2:3] <- c("year", "pellet_count")
obs$year <- as.character(obs$year)
obs$year <- gsub("X", "", obs$year)
obs$year <- as.integer(obs$year)

obs_all <- rbind(obs_all, obs)

## Checking before joining...
levels(as.factor(plot_hab$habitat))
[1] "black spruce" "mixed mature" "reburn mixed"
## Wow, that is really clean!

## Now conforming some of the plot names.
plot_data$plot_name <- gsub("FRB ", "FRN-", plot_data$plot_name)
plot_data$plot_name <- gsub("FRN-0", "FRN-", plot_data$plot_name)

plot_data_2 <- merge(x=plot_data, y=plot_hab, all.x=TRUE)

## Saving here.
write.csv(plot_data_2, "../data/final_data/geodata/2020-03-18_snowshoe_hare_plot_data.csv", row.names=FALSE)

write.csv(obs_all, "../data/final_data/observations/2020-03-18_snowshoe_hare_pellet_counts.csv", row.names=FALSE)
```

## Wednesday, March 18

To do:

  - ~~Submit this week’s *Refuge Notebook* article.~~
  - *Refuge Notebook* catch-up.
  - Early adopter data management homework.
  - Blackfish diet analysis.
  - ~~Sandpiper Lake *Elodea* project PUPs~~.

I finished processing snowshoe hare data that I had started yesterday.

That updated run on mBRAVE that I initiated yesterday never got
anywhere. It gives the message, “Please Wait: Initial Processing of This
Run is Underway.” This is how it has looked since the middle of the day
yesterday. I think there must be a problem with the run parameters.

I received and worked on formatting this week’s *Refuge Notebook*
article, getting it submitted to the
*[Clarion](https://www.peninsulaclarion.com/)*. I worked some on
formatting older articles, but the connection to the CMS was so slow
that I could not get this work posted.

I started working on the PUPs for herbicides to be used in *Elodea*
eradication work, spending most of the afternoon on PUPs.

We had a biology staff meeting via conference call this afternoon. We
are going to be teleworking for a while.

## Thursday, March 19

To do:

  - *Refuge Notebook* catch-up.
  - Early adopter data management homework.
  - Blackfish diet analysis.

I received an e-mail from Todd Steinlage regarding the Sandpiper Lake
*Elodea* sample. Based on a qPCR test and an ITS sequence, this one
appears to be *Elodea canadensis* (specimen record:
[KNWR:Herb:12443](http://arctos.database.museum/guid/KNWR:Herb:12443)).
This is consistent with our guess that this population was started by a
floatplane bringing it from Anchorage because Anchorage populations were
identified as *Elodea canadensis* (Morton et al.
[2019](#ref-morton_responding_2019)).

I entered data for the five *Dallia pectoralis* specimens from which
Apphia and I had dissected out gut contents
([KNWRObs:Fish:12](http://arctos.database.museum/guid/KNWRObs:Fish:12)–[KNWRObs:Fish:16](http://arctos.database.museum/guid/KNWRObs:Fish:16)).

Now I would like to set up the SCVUC pipeline on Yeti for analyzing the
blackfish gut contents. Using the instructions from
<https://github.com/Hajibabaei-Lab/SCVUC_COI_metabarcode_pipeline#prepare-your-environment-to-run-the-pipeline>.

I spent quite a while trying to get the SCVUC pipeline installed and
working. I think I need to give up on ORFfinder for now.

``` 
 846  2020-03-19 11:03:36  ls
  847  2020-03-19 11:03:46  cd /lustre
  848  2020-03-19 11:03:47  ls
  849  2020-03-19 11:03:55  cd projects
  850  2020-03-19 11:03:57  ls
  851  2020-03-19 11:04:00  cd fws
  852  2020-03-19 11:04:03  ls
  853  2020-03-19 11:04:10  cd fws_bio
  854  2020-03-19 11:04:13  ls
  855  2020-03-19 11:04:19  cd mattbowser
  856  2020-03-19 11:04:31  cwd
  857  2020-03-19 11:04:37  pwd
  858  2020-03-19 11:07:01  cd /lustre/projects/fws/fws_bio/mattbowser
  859  2020-03-19 11:10:45  ls
  860  2020-03-19 11:10:52  cd projects
  861  2020-03-19 11:10:54  ls
  862  2020-03-19 11:11:28  cd /lustre/projects/fws/fws_bio/mattbowser/projects
  863  2020-03-19 11:12:23  mkdir COI_metabarcoding
  864  2020-03-19 11:12:35  cd COI_metabarcoding
  865  2020-03-19 11:32:54  git clone https://github.com/Hajibabaei-Lab/SCVUC_COI_metabarcode_pipeline.git
  866  2020-03-19 11:33:03  ls
  867  2020-03-19 11:33:15  cd SCV*
  868  2020-03-19 11:34:08  ls
  869  2020-03-19 11:35:11  module avail
  870  2020-03-19 11:36:31  module load python/anaconda3
  871  2020-03-19 11:40:46  conda env create -f environment.yml
  872  2020-03-19 12:06:37  conda activate SCVUCv4.3
  873  2020-03-19 12:09:03  source activate SCVUCv4.3
  874  2020-03-19 12:09:55  wget ftp://ftp.ncbi.nlm.nih.gov/genomes/TOOLS/ORFfinder/linux-i64/ORFfinder.gz
  875  2020-03-19 12:10:23  gunzip ORFfinder.gz
  876  2020-03-19 12:10:35  chmod 755 ORFfinder
  877  2020-03-19 12:10:58  mv ORFfinder ~/bin/.
  878  2020-03-19 12:11:41  mkdir /home/mattbowser/bin
  879  2020-03-19 12:11:47  mv ORFfinder ~/bin/.
  880  2020-03-19 12:12:04  ORFfinder
  881  2020-03-19 12:15:02  cd ~/miniconda3/envs/SCVUCv4.3/etc/conda/activate.d/
  882  2020-03-19 12:15:08  cd ~/
  883  2020-03-19 12:15:11  ls
  884  2020-03-19 12:28:43  cd /lustre/projects/fws/fws_bio/mattbowser/projects/COI_metabarcoding/SCVUC_COI_metabarcode_pipeline
  885  2020-03-19 12:28:46  ls
  886  2020-03-19 12:32:05  ls -al
  887  2020-03-19 12:32:34  cd ~/
  888  2020-03-19 12:32:38  ls -al
  889  2020-03-19 12:32:46  cd .conda
  890  2020-03-19 12:32:47  ls
  891  2020-03-19 12:32:51  cd envs
  892  2020-03-19 12:32:53  ls
  893  2020-03-19 12:33:27  cd SCV*
  894  2020-03-19 12:33:28  ls
  895  2020-03-19 12:33:35  cd etc
  896  2020-03-19 12:33:36  ls
  897  2020-03-19 12:33:52  cd conda
  898  2020-03-19 12:33:54  ls
  899  2020-03-19 12:34:00  cd activate.d
  900  2020-03-19 12:34:01  ls
  901  2020-03-19 12:34:52  vi LD_PATH.sh
  902  2020-03-19 12:37:05  ls
  903  2020-03-19 12:37:36  cd ..
  904  2020-03-19 12:37:54  cd deactivate.d
  905  2020-03-19 12:37:55  ls
  906  2020-03-19 12:38:14  vi LD_PATH.sh
  907  2020-03-19 12:39:03  ls
  908  2020-03-19 12:39:40  cd ..
  909  2020-03-19 12:39:51  ls
  910  2020-03-19 12:40:16  cd lib64
  911  2020-03-19 12:40:34  ls
  912  2020-03-19 12:40:53  ln -s ../glibc-2.14/lib/libc.so.6 libc.so.6
  913  2020-03-19 12:40:57  ls
  914  2020-03-19 12:43:10  source deactivate SCVUCv4.3
  915  2020-03-19 12:43:35  source activate SCVUCv4.3
  916  2020-03-19 12:44:26  ORFfinder
  917  2020-03-19 12:56:21  ls
  918  2020-03-19 12:59:06  cd ..
  919  2020-03-19 12:59:09  ls
  920  2020-03-19 13:02:57  cd lib
  921  2020-03-19 13:02:59  ls
  922  2020-03-19 13:05:50  cd glib
  923  2020-03-19 13:06:21  cd glib*
  924  2020-03-19 13:06:22  ls
  925  2020-03-19 13:06:29  cd ..
  926  2020-03-19 13:06:33  cd lib
  927  2020-03-19 13:06:34  ls
  928  2020-03-19 13:44:57  cd ~/.conda/env/SCVUCv4.3
  929  2020-03-19 13:45:03  ls
  930  2020-03-19 13:45:07  ls -al
  931  2020-03-19 13:45:11  cd .conda
  932  2020-03-19 13:45:12  ls
  933  2020-03-19 13:45:18  cd envs
  934  2020-03-19 13:45:33  cd ~/.conda/envs/SCVUCv4.3
  935  2020-03-19 13:45:36  ls
  936  2020-03-19 13:45:51  cd include
  937  2020-03-19 13:45:56  ls
  938  2020-03-19 13:46:08  cd ..
  939  2020-03-19 13:46:09  ls
  940  2020-03-19 13:46:22  cd lib
  941  2020-03-19 13:46:25  ls
  942  2020-03-19 13:46:46  cd glib*
  943  2020-03-19 13:46:48  ls
  944  2020-03-19 13:46:52  cd include
  945  2020-03-19 13:46:53  ls
  946  2020-03-19 13:46:56  cd ..
  947  2020-03-19 13:50:15  find ~/.conda/envs/SCVUCv4.3 -name "*glibc*"
  948  2020-03-19 13:51:42  module avail
  949  2020-03-19 14:27:01  cd /lustre/projects/fws/fws_bio/mattbowser/projects/COI_metabarcoding/SCVUC_COI_metabarcode_pipeline
  950  2020-03-19 14:27:03  ls
  951  2020-03-19 14:41:08  cd ~/
  952  2020-03-19 14:41:32  wget http://ftp.gnu.org/gnu/glibc/glibc-2.14.tar.gz
  953  2020-03-19 14:44:28  tar -xzvf glibc-2.14.tar.gz
  954  2020-03-19 14:53:57  ls
  955  2020-03-19 14:54:29  rm -R glibc-2.14
  956  2020-03-19 14:57:24  ls
  957  2020-03-19 14:57:32  ls -al
  958  2020-03-19 14:57:42  cd glibc*
  959  2020-03-19 14:57:44  ls
  960  2020-03-19 14:57:59  rm -R *
  961  2020-03-19 14:59:13  ls
  962  2020-03-19 14:59:31  rm -R *
  963  2020-03-19 15:01:34  ls
  964  2020-03-19 15:01:49  rm -R *
  965  2020-03-19 15:08:51  module avail
  966  2020-03-19 15:15:57  module list
  967  2020-03-19 15:16:15  module load python/miniconda3-gcc6.1.0
  968  2020-03-19 15:16:57  source activate SCVUCv4.3
  969  2020-03-19 15:17:21  cd ..
  970  2020-03-19 15:17:24  ls
  971  2020-03-19 15:17:52  source deactivate SCVUCv4.3
  972  2020-03-19 15:18:00  module purge
  973  2020-03-19 15:19:05  rmdir glibc-2.14
  974  2020-03-19 15:19:07  ls
  975  2020-03-19 15:19:34  tar -xzvf glibc-2.14.tar.gz
  976  2020-03-19 15:21:50  ls
  977  2020-03-19 15:22:04  cd glibc-2.14
  978  2020-03-19 15:22:06  ls
  979  2020-03-19 15:24:06  uname -r
  980  2020-03-19 15:25:51  cat /etc/os-release
  981  2020-03-19 15:26:39  ./configure
  982  2020-03-19 15:27:06  mkdir build
  983  2020-03-19 15:27:38  dmesg | head -1
  984  2020-03-19 15:28:11  cat /proc/version
  985  2020-03-19 15:32:32  cd /lib
  986  2020-03-19 15:32:34  ls
  987  2020-03-19 15:34:56  ldd --version
  988  2020-03-19 15:36:13  /lib64/libc.so.6 --version
  989  2020-03-19 15:37:01  ## So Yeti has glib version glibc version 2.12.
  990  2020-03-19 15:39:59  cd ~/
  991  2020-03-19 15:40:00  ls
  992  2020-03-19 15:40:07  cd glibc*
  993  2020-03-19 15:40:08  ls
  994  2020-03-19 15:40:17  cd build
  995  2020-03-19 15:40:26  ../configure
  996  2020-03-19 15:41:39  ls
  997  2020-03-19 15:41:54  make
  998  2020-03-19 15:44:38  ../configure --prefix /home/mattbowser/.conda/envs/SCVUCv4/glibc-2.14
  999  2020-03-19 15:44:50  ls
 1000  2020-03-19 15:44:56  make
 1001  2020-03-19 15:56:17  wget https://ayera.dl.sourceforge.net/project/rdp-classifier/rdp-classifier/rdp_classifier_2.12.zip
 1002  2020-03-19 16:04:04  ls
 1003  2020-03-19 16:04:11  cd glib*
 1004  2020-03-19 16:04:14  cd build
 1005  2020-03-19 16:04:15  ls
 1006  2020-03-19 16:05:05  make
 1007  2020-03-19 16:26:00  ls
 1008  2020-03-19 16:29:28  history
 1009  2020-03-19 16:30:39  pwd
 1010  2020-03-19 16:32:33  cd /conda/envs/
 1011  2020-03-19 16:32:47  cd ~/.conda/envs
 1012  2020-03-19 16:32:48  ls
 1013  2020-03-19 16:32:56  cd SCVUC*
 1014  2020-03-19 16:32:58  ls
 1015  2020-03-19 16:33:42  mkdir pwd
 1016  2020-03-19 16:33:53  rmdir pwd
 1017  2020-03-19 16:33:56  pwd
 1018  2020-03-19 16:34:49  mkdir glibc-2.14
 1019  2020-03-19 16:34:52  ls
 1020  2020-03-19 16:35:08  cp /home/mattbowser/glibc-2.14/build/* /home/mattbowser/.conda/envs/SCVUCv4.3
 1021  2020-03-19 16:35:29  /glibc-2.14/ls
 1022  2020-03-19 16:35:32  ls
 1023  2020-03-19 16:35:53  cd /home/mattbowser/.conda/envs/SCVUCv4.3
 1024  2020-03-19 16:36:17  /glibc-2.14/ls
 1025  2020-03-19 16:36:20  ls
 1026  2020-03-19 16:36:48  ls -al
 1027  2020-03-19 16:39:10  find . -type f -cmin -10
 1028  2020-03-19 16:39:33  find . -type f -cmin -10 -delete
 1029  2020-03-19 16:39:38  ls
 1030  2020-03-19 16:39:59  cd glibc-2.14
 1031  2020-03-19 16:40:00  ls
 1032  2020-03-19 16:40:16  cp -R /home/mattbowser/glibc-2.14/build/* /home/mattbowser/.conda/envs/SCVUCv4.3/glibc-2.14/
 1033  2020-03-19 16:44:43  ls
 1034  2020-03-19 16:45:15  cd ..
 1035  2020-03-19 16:45:16  ls
 1036  2020-03-19 16:45:23  cd lib64
 1037  2020-03-19 16:45:24  ls
 1038  2020-03-19 16:46:14  ln -s ../glibc-2.14/libc.so.6 libc.so.6
 1039  2020-03-19 16:46:24  rm libc.so.6
 1040  2020-03-19 16:46:28  ln -s ../glibc-2.14/libc.so.6 libc.so.6
 1041  2020-03-19 16:46:30  ls
 1042  2020-03-19 16:47:23  history
 1043  2020-03-19 16:48:27  module load python/anaconda3
 1044  2020-03-19 16:48:49  source activate SCVUCv4.3
 1045  2020-03-19 16:49:16  ORFfinder
 1046  2020-03-19 16:51:57  ## Wow, now we need version 2.17.
 1047  2020-03-19 16:52:21  history
 1048  2020-03-19 16:52:38  history | grep wget
 1049  2020-03-19 16:54:24  cd ~/
 1050  2020-03-19 16:54:26  ls
 1051  2020-03-19 16:54:33  wget http://ftp.gnu.org/gnu/glibc/glibc-2.17.tar.gz
 1052  2020-03-19 16:54:44  ls
 1053  2020-03-19 16:55:13  tar -zxvf glibc-2.17.tar.gz
 1054  2020-03-19 16:56:45  ls
 1055  2020-03-19 16:56:54  cd glibc-2.17
 1056  2020-03-19 16:56:55  ls
 1057  2020-03-19 16:57:00  mkdir build
 1058  2020-03-19 16:57:02  cd build
 1059  2020-03-19 16:57:03  ls
 1060  2020-03-19 16:57:37  history | grep configure
 1061  2020-03-19 16:57:56  ../configure --prefix /home/mattbowser/.conda/envs/SCVUCv4/glibc-2.14
 1062  2020-03-19 16:59:11  ## Maybe I need to give up on this. the autoconf version is not campatible.
 1063  2020-03-19 16:59:22  history
 1000  2020-03-19 17:02:07  exit
 1001  2020-03-19 19:05:32  ls
 1002  2020-03-19 19:05:46  gunzip rdp_class*
 1003  2020-03-19 19:08:06  unzip rdp_class*
 1004  2020-03-19 19:08:24  ls
 1005  2020-03-19 19:10:19  mv ~/rdp_classifier_2.12 /lustre/projects/fws/fws_bio/mattbowser/projects/COI_metabarcoding/SCVUC_COI_metabarcode_pipeline/rdp_classifier_2.12
 1006  2020-03-19 19:10:52  cd /lustre/projects/fws/fws_bio/mattbowser/projects/COI_metabarcoding/SCVUC_COI_metabarcode_pipeline/rdp_classifier_2.12
 1007  2020-03-19 19:10:55  ls
 1008  2020-03-19 19:14:31  cd ..
 1009  2020-03-19 19:14:33  ls
 1010  2020-03-19 19:14:57  wget https://github.com/terrimporter/CO1Classifier/releases/download/v4/CO1v4_trained.tar.gz
 1011  2020-03-19 19:18:12  ls
 1012  2020-03-19 19:18:31  tar -zxvf CO1v4_trained.tar.gz
 1013  2020-03-19 19:20:13  ls
 1014  2020-03-19 19:20:29  rm CO1v4_trained.tar.gz
 1015  2020-03-19 19:20:31  ls
 1016  2020-03-19 19:20:36  cd rdp*
 1017  2020-03-19 19:20:37  ls
 1018  2020-03-19 19:22:56  cd ..
 1019  2020-03-19 19:22:57  ls
 1020  2020-03-19 19:23:01  cd myd*
 1021  2020-03-19 19:23:03  ls
 1022  2020-03-19 19:27:34  pwd
 1023  2020-03-19 19:31:19  cd ..
 1024  2020-03-19 19:31:20  ls
 1025  2020-03-19 19:31:27  mkdir data
 1026  2020-03-19 19:31:28  ls
 1027  2020-03-19 19:31:31  cd data
 1028  2020-03-19 19:32:40  wget https://rtlgenomics.s3.amazonaws.com/7SLpgTzp2k/Bowser_6382Raw03132020.zip?AWSAccessKeyId=AKIAI4T5MZRBWMKZM5NA&Expires=1592265873&Signature=fCzfVZlfw5aQ6sFjUJrUb3466sY%3D
 1029  2020-03-19 20:02:16  history
```

mattbowser@IFW7KEN-730134 MINGW64 \~ $ ssh
mattbowser@fws.gov@yeti.cr.usgs.gov

—— WARNING —– WARNING —– WARNING —– WARNING —– WARNING ——-

This computer system, including all related equipment, networks, and
network devices (including Internet access), is provided by the
Department of the Interior (DOI) in accordance with the agency policy
for official use and limited personal use.

All agency computer systems may be monitored for all lawful purposes,
including but not limited to, ensuring that use is authorized, for
management of the system, to facilitate protection against unauthorized
access, and to verify security procedures, survivability and operational
security. Any information on this computer system may be examined,
recorded, copied and used for authorized purposes at any time. All
information, including personal information, placed or sent over this
system may be monitored, and users of this system are reminded that such
monitoring does occur. Therefore, there should be no expectation of
privacy with respect to use of this system.

By logging into this agency computer system, you acknowledge and consent
to the monitoring of this system. Evidence of your use, authorized or
unauthorized, collected during monitoring may be used for civil,
criminal, administrative, or other adverse action. Unauthorized or
illegal use may subject you to prosecution.

—— WARNING —– WARNING —– WARNING —– WARNING —– WARNING ——-

mattbowser@fws.gov@yeti.cr.usgs.gov’s password: Last login: Wed Mar 18
11:46:24 2020 from 10.209.40.204

-----

Yeti Cluster Info
\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_
/ Your account codes are:  
| | | Account User | | ——————– ———- | | bio mattbowser | | | | /lustre
is online | | /cxfs/projects is online | | | | New Docs:
https://hpcportal.cr.usgs.gov/hpc-user-docs/ |   / ——————————————————–  
  
*|/* ( 0 0 ) *—/ v=v -–* ( ) | |  Y E T I /| | ) | | ))\_(( | | ( / | |
/   | |  
^^^^ |  / | ^^^<sup> </sup>^^^^ ^^^^^

-----

/cxfs/projects is low on space and we are working to resolve the space
issue. If you have files you can delete please do so. If you have files
you can move to BlackPearl please do so or contact hpc@usgs.gov for
help. \[mattbowser@yeti-login20 \~\] ls 2018\_AWCC\_soil\_fungi
2018\_AWCC\_soil\_fungi\_try\_02 perl5 pipits R
\[mattbowser@yeti-login20 \~\] cd /lustre \[mattbowser@yeti-login20
lustre\] ls checkpoints IOR2 IOR-20180416 move\_to\_blackpearl projects
reference\_data scratch tmp \[mattbowser@yeti-login20 lustre\] cd
projects \[mattbowser@yeti-login20 projects\] ls blm climate ecosystems
fws jkreitler magpie software training bor css energyminerals hazards
landfire regions spcmsc water \[mattbowser@yeti-login20 projects\] cd
fws \[mattbowser@yeti-login20 fws\] ls fws\_bio fws\_sciapps imb nc-es
pifwo \[mattbowser@yeti-login20 fws\] cd fws\_bio
\[mattbowser@yeti-login20 fws\_bio\] ls mattbowser
\[mattbowser@yeti-login20 fws\_bio\] cd mattbowser
\[mattbowser@yeti-login20 mattbowser\] cwd -bash: cwd: command not found
\[mattbowser@yeti-login20 mattbowser\] pwd
/lustre/projects/fws/fws\_bio/mattbowser \[mattbowser@yeti-login20
mattbowser\] cd /lustre/projects/fws/fws\_bio/mattbowser
\[mattbowser@yeti-login20 mattbowser\] ^C \[mattbowser@yeti-login20
mattbowser\] ls lib projects \[mattbowser@yeti-login20 mattbowser\] cd
projects \[mattbowser@yeti-login20 projects\] ls
\[mattbowser@yeti-login20 projects\] cd
/lustre/projects/fws/fws\_bio/mattbowser/projects
\[mattbowser@yeti-login20 projects\] mkdir COI\_metabarcoding
\[mattbowser@yeti-login20 projects\] cd COI\_metabarcoding
\[mattbowser@yeti-login20 COI\_metabarcoding\] git clone
https://github.com/Hajibabaei-Lab/SCVUC\_COI\_metabarcode\_pipeline.git
Cloning into ‘SCVUC\_COI\_metabarcode\_pipeline’… remote: Enumerating
objects: 123, done. remote: Counting objects: 100% (123/123), done.
remote: Compressing objects: 100% (100/100), done. remote: Total 314
(delta 69), reused 49 (delta 23), pack-reused 191 Receiving objects:
100% (314/314), 193.80 KiB | 5.70 MiB/s, done. Resolving deltas: 100%
(177/177), done. \[mattbowser@yeti-login20 COI\_metabarcoding\] ls
SCVUC\_COI\_metabarcode\_pipeline \[mattbowser@yeti-login20
COI\_metabarcoding\] cd SCV\* \[mattbowser@yeti-login20
SCVUC\_COI\_metabarcode\_pipeline\] ls bold.hmm config\_BF1.yaml
config\_ml-jg.yaml snakefile\_withoutPseudogeneFiltering bold.hmm.h3f
config\_BF2.yaml environment.yml snakefile\_withPseudogeneFiltering
bold.hmm.h3i config\_BR5.yaml LICENSE.txt bold.hmm.h3m
config\_F230R.yaml perl\_scripts bold.hmm.h3p config\_fwh1.yaml
README.md \[mattbowser@yeti-login20 SCVUC\_COI\_metabarcode\_pipeline\]
module avail

————————————– /usr/share/Modules/modulefiles ————————————– dot
module-git module-info modules null use.own

——————————————— /etc/modulefiles ——————————————— allinea/6.0.6
mpi/mpich-3.2-gcc allinea/7.0.3 mpi/mpich-3.2-intel arm-forge/18.2
mpi/mpich-x86\_64 arm-forge/19.0 mpi/openmpi-1.4.5 autoconf/2.69-gcc
mpi/openmpi-1.4.5-gcc autoconf/2.69-intel mpi/openmpi-1.5.3
automake/1.15-gcc mpi/openmpi-1.5.4 automake/1.15-intel
mpi/openmpi-1.5.5-gcc bayesass/3.0.4-gnu mpi/openmpi-1.6.5-gcc
bayesass/snps/1.1-gnu mpi/openmpi-1.7.4-gcc
beopest/beopest-13.6-gcc-openmpi-1.10.2 mpi/openmpi-1.8.1-gcc
beopest/beopest-13.6-intel mpi/openmpi-1.8.5-gcc
blackpearl-ds3-gui/5.0.7 mpi/openmpi-1.8.5-intel
blackpearl-ds3-java-cli/5.0.1 netcdf/4.4.1.1-gcc6.1.0 boost/1.52.0-gcc
netcdf/4.4.1-intel boost/1.59.0-gcc netcdf-parallel/4.4.1.1-gcc6.1.0
bowtie2/2.2.5-gcc netcdf-parallel/4.4.1-intel bowtie2/2.3.5.1
netcdf-serial/4.4.1.1-hdf51.10.1-gcc6.1.0 bzip2/1.0.5-gcc-4.4.7
netcdf-serial/4.4.1.1-hdf51.8.18-gcc6.1.0 cdo/1.7.1-gcc octave/4.2.0-gcc
cdo/1.9.3-gcc OpenBUGS/3.2.3-gcc cig/relax-1.0.7-intel opencv/3.1.0-gcc
cig-relax/1.0.7-intel openmpi/1.10.1-gcc cmake/2.8.12.1
openmpi/1.10.2-gcc cmake/3.10.1 openmpi/1.10.2-gcc6.1.0 cmake/3.14.2
openmpi/1.10.2-gcc7.1.0 coretran/0.0.0 openmpi/1.10.2-intel cuda/7.5.18
openmpi/1.6.5-gcc cuda/8.0.44 openmpi/1.6.5-intel cuda/9.0.176
openmpi/1.8.5-gcc curl/7.53.1-gcc4.9.3 openmpi/1.8.5-intel dislin/10.6
openmpi/1.8.8-gcc dmtcp/dmtcp-2.2.1 openmpi/1.8.8-gcc6.1.0
ExaML/3.0.18-gcc openmpi/1.8.8-intel ExaML/3.0.18-intel
openmpi/2.1.2-gcc7.1.0 fastStructure/1.0.4 openmpi/2.1.2-pgi2017
gatk/4.0.2.1 openmpi/3.0.0-gcc7.1.0 gcc/4.9.3 openmpi-x86\_64 gcc/5.3
openssl/1.0.2k-gcc4.9.3 gcc/6.1 parallel/20160422 gcc/7.1.0
parallel-netcdf/1.6.1-gcc gcc/8.3.0 parallel-netcdf/1.6.1-intel
gdal/1.11.4-gcc paraview/4.3.1-50 gdal/2.0.2-gcc paraview/5.2.0
gdal/2.1.3-gcc paraview/5.6.0 gdal/2.2.0-gcc pblat/1.6-gcc
gdal/2.2.0-gcc4.9 pcre/8.39-gcc-4.4.7 gdal/2.2.2-gcc perl/perl-5.24.1
gdal/2.3.1-gcc71 perl/perl-bio-perl-1.007001 gdal/2.4.1-gcc71
perl/perl-xml-simple-2.24 genetics/admixture-1.3.0 petsc/3.11.1-gcc
genetics/BayeScan2.1 petsc/3.11.1-intel genetics/beagle-lib-2.1.2
petsc/3.11.1-intel\_openmpi genetics/beast-1.8.2 petsc/3.7.6-intel
genetics/beast-2.2.1 pgi/16.10 genetics/beast-2.4.7 pgi/17.10
genetics/beast-2.5.0 photoscan/photoscan-pro-1.1.6 genetics/blast-2.2.31
photoscan/photoscan-pro-1.2.3 genetics/blast-plus-2.6.0
photoscan/photoscan-pro-1.2.4 genetics/blast-plus-2.9.0
photoscan/photoscan-pro-1.3.2 genetics/blat-35
photoscan/photoscan-pro-1.3.4 genetics/BRATNextGen
photoscan/photoscan-pro-1.4.2 genetics/cd-hit-4.6.7
photoscan/photoscan-pro-1.4.3 genetics/G-PhoCS-1.2.2
photoscan/photoscan-pro-1.4.4 genetics/im-12.17.2009 plavaflow/0.0.0
genetics/ima2-8.27.12 proj/4.9.2-gcc4.4.7 genetics/jmodeltest-2.1.7
proj/4.9.2-gcc6.1.0 genetics/kSNP-3.021 proj/5.0.1-gcc-7.1.0
genetics/ls-bsr proj/5.2.0-gcc-7.1.0 genetics/mrbayes-3.2.5\_openmpi
python/2.7.8-gcc genetics/pagit-1.6 python/anaconda genetics/parsnp-1.2
python/anaconda2 genetics/prodigal-2.6.3 python/anaconda3
genetics/prokka-1.12 python/epd-7.3-2-x86\_64 genetics/quast-4.5
python/geobipy genetics/raxml-8.2.10 python/miniconda3-gcc6.1.0
genetics/samtools-1.5 python/modules/dadi-1.6.3\_epd
genetics/spades-3.10.1 python/pGeobipy\_cxfs genetics/vsearch-2.4.3
python/pGeobipy\_lustre gis/gdal-1.9.2 python/pPython3
gis/gdal-1.9.2-geos python/python-2.7 gis/geos-3.5.0 qiime/1.9.2
gis/grass-7.0.0-svn qt/4.8.5 gis/grass-7.4-spack qt/5.10.0-gcc-7.1.0
gis/grass-7.6-spack qt/5.5.1 gis/liblas-1.7.0 R/3.4.4-gcc7.1.0
gis/TauDEM-5.3.8-gcc-mpich R/3.5.0-gcc7.1.0 gis/TauDEM-openmpi-1.6.5
R/3.5.1-gcc7.1.0 hdf/4.2.11-gcc R/3.6.1-gcc7.1.0 hdf5/1.10.0-gcc
R/3.6.1-intel2018 hdf5/1.10.1-gcc6.1.0 rstudio/0.98.1103
hdf5/1.10.5-gcc7.1.0 samtools/1.9-gcc hdf5/1.8.15-patch1-intel
singularity/3.2.1 hdf5/1.8.18-gcc6.1.0 spack hdf5-parallel/1.10.0-gcc
sqlite/3.23.1-gcc-7.1.0 hdf5-parallel/1.10.1-gcc6.1.0 stacks/1.35
hdf5-parallel/1.10.5-gcc7.1.0 stacks/1.46
hdf5-parallel/1.8.15-patch1-intel stacks/2.0.b8
hdf5-parallel/1.8.18-gcc6.1.0 stacks/2.1 hdf5-serial/1.10.1-gcc6.1.0
stacks/2.2 hdf5-serial/1.10.5-gcc7.1.0 stacks/2.3e
hdf5-serial/1.8.18-gcc6.1.0 stacks/2.4 hdfeos/2.19-gcc stacks/2.41
ima2/8.27.12 stacks/2.5 ima2p/9.29.16 standard-RAxML/8.2.9-gcc
intel/psxe-2015 stereopipeline/2.4.2 intel/psxe-2017u2
stereopipeline/2.5.0 intel/psxe-2018u1 stereopipeline/2.5.1
intel/psxe-2018u3 stereopipeline/2.5.2 ioapi/3.2-gnu
stereopipeline/2.6.1 jags/3.4-gnu structure/2.3.4-gcc jags/3.4-intel
structure/2.3.4-intel jags/4.0.0-gnu subversion/1.9.3-gcc
jags/4.0.0-intel tools/beopest-11.13-gnu jags/4.1.0-gcc
tools/beopest-11.13-intel jags/4.1.0-intel tools/beopest-12.2-gnu
jags/4.3.0-gcc tools/beopest-12.2-intel jags/4.3.0-intel
tools/cdo-1.6.4-gnu java/jdk-11.0.1 tools/fftw-3.2.2 java/jdk-1.8.0\_162
tools/fftw-3.3.4 java/jdk-1.8.0\_45 tools/gifsicle-1.88 jbigkit/2.1-gcc
tools/hdf-4.2.10-gnu jupyterhub/1.0.0 tools/hdf5-1.8.13-gnu
landslide\_triggers/0.0.0 tools/hdf5-1.8.15-intel
libszip/2.1.1-gcc-4.4.7 tools/mct-2.8-intel libtiff/4.0.6-gcc4.4.7
tools/mono-3.4.0 libtool/2.4.6-gcc tools/ncl\_ncarg-6.2.1
libtool/2.4.6-intel tools/nco-4.4.4-gnu libzip/1.2.0-gcc-7.1.0
tools/nco-4.5.2-intel loadbalance/2014.09.05 tools/nco-4.7.8-gnu
m4/1.4.17-gcc tools/netcdf-4.3.2-gnu matlab/mcr-R2013a-v81
tools/netcdf-4.3.2-intel matlab/R2014b tools/netcdf-4.3.3.1-intel
matlab/R2015b tools/netcdf-c-4.3.2-intel matlab/R2018b
tools/netcdf-c-4.6.2-gnu mct/2.8-intel tools/netcdf-c-4.6.2-intel
mct/2.9-gcc tools/netcdf-f-4.4.2-intel mega/7.0.20 tools/netcdf-java-4.3
metis/5.1.0-gcc tools/pest-13.4-gnu metis/5.1.0-intel
tools/pest-13.4-intel migrate-n/3.6.8-gcc-openmpi tools/pest-13.5-gnu
migrate-n/3.6.8-intel tools/pest-13.5-intel
migrate-n/4.4.4-gcc7-openmpi3.1 tools/petsc-3.5.1-impi
migrate-n/4.4.4-intel tools/petsc-3.5.1-openmpi-1.8.1
modflow/6.0.2-gcc-6.1 tools/pnetcdf-1.6.1-gnu modflow/6.0.2-intel-2018
tools/pnetcdf-1.6.1-intel mono/3.4.0 tools/scrip-intel mono/4.2.1-gcc
tools/szip-2.1-gnu mono/4.8.0-gcc tools/udunits-2.2.16-gnu
mono/5.2.0-gcc tools/vtk-6.0.0 mothur/1.36.1-gcc tools/vtk-6.1.0
mothur/1.37.2-gcc treemix/1.13-gcc mothur/1.38.1.1-gcc uvstats/1.1.0+114
mothur/1.39.5-gcc viz/paraview-4.3.1-50 mothur/1.40.5-gcc wine/1.8
mothur/1.41.2-gcc wrf/3.8-intel mothur/1.42.1-gcc xz/5.2.3-gcc4.4.7
mothur/1.43.0-gcc zlib/1.2.11-gcc mpi/impi-4.1-x86\_64
\[mattbowser@yeti-login20 SCVUC\_COI\_metabarcode\_pipeline\] module
load python/anaconda3 \[mattbowser@yeti-login20
SCVUC\_COI\_metabarcode\_pipeline\] conda env create -f environment.yml
Collecting package metadata (repodata.json): done Solving environment:
done

\==\> WARNING: A newer version of conda exists. \<== current version:
4.8.1 latest version: 4.8.3

Please update conda by running

    $ conda update -n base -c defaults conda

Downloading and Extracting Packages aioeasywebdav-2.4.0 | 18 KB |
\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#
| 100% gsl-2.4 | 2.6 MB |
\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#
| 100% perl-tie-ixhash-1.23 | 11 KB |
\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#
| 100% perl-math-spline-0.0 | 8 KB |
\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#
| 100% graphite2-1.3.13 | 101 KB |
\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#
| 100% perl-libxml-perl-0.0 | 38 KB |
\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#
| 100% perl-file-slurp-tiny | 8 KB |
\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#
| 100% intel-openmp-2019.4 | 876 KB |
\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#
| 100% perl-test-1.26 | 14 KB |
\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#
| 100% perl-net-http-6.19 | 19 KB |
\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#
| 100% gcc\_impl\_linux-64-7. | 73.2 MB |
\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#
| 100% perl-parent-0.236 | 5 KB |
\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#
| 100% perl-xml-parser-2.44 | 165 KB |
\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#
| 100% perl-lwp-protocol-ht | 9 KB |
\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#
| 100% libxcb-1.13 | 502 KB |
\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#
| 100% paramiko-1.18.4 | 227 KB |
\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#
| 100% perl-pod-usage-1.69 | 13 KB |
\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#
| 100% perl-html-treebuilde | 12 KB |
\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#
| 100% pango-1.42.4 | 528 KB |
\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#
| 100% perl-exception-class | 19 KB |
\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#
| 100% perl-font-ttf-1.06 | 200 KB |
\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#
| 100% bwidget-1.9.11 | 111 KB |
\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#
| 100% perl-ole-storage\_lit | 18 KB |
\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#
| 100% perl-dist-checkconfl | 10 KB |
\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#
| 100% perl-uri-1.76 | 55 KB |
\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#
| 100% gfortran\_impl\_linux- | 9.0 MB |
\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#
| 100% perl-io-stringy-2.11 | 35 KB |
\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#
| 100% perl-mime-types-2.17 | 56 KB |
\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#
| 100% ncurses-6.1 | 958 KB |
\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#
| 100% ecdsa-0.13 | 79 KB |
\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#
| 100% perl-class-inspector | 14 KB |
\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#
| 100% perl-ipc-run-2018052 | 74 KB |
\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#
| 100% perl-graphviz-2.24 | 36 KB |
\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#
| 100% parallel-20190722 | 1.6 MB |
\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#
| 100% libgfortran-ng-7.3.0 | 1.3 MB |
\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#
| 100% bz2file-0.98 | 13 KB |
\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#
| 100% docutils-0.15.2 | 736 KB |
\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#
| 100% gfortran\_linux-64-7. | 148 KB |
\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#
| 100% cryptography-2.8 | 620 KB |
\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#
| 100% perl-www-robotrules- | 12 KB |
\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#
| 100% perl-mime-tools-5.50 | 113 KB |
\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#
| 100% pycparser-2.19 | 174 KB |
\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#
| 100% perl-clone-0.42 | 13 KB |
\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#
| 100% perl-crypt-rc4-2.02 | 8 KB |
\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#
| 100% perl-sub-exporter-0. | 30 KB |
\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#
| 100% gcc\_linux-64-7.3.0 | 149 KB |
\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#
| 100% perl-xml-libxslt-1.9 | 40 KB |
\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#
| 100% perl-postscript-0.06 | 22 KB |
\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#
| 100% perl-sub-quote-2.006 | 18 KB |
\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#
| 100% perl-dbi-1.642 | 549 KB |
\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#
| 100% gxx\_linux-64-7.3.0 | 148 KB |
\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#
| 100% perl-xml-sax-base-1. | 19 KB |
\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#
| 100% perl-io-tty-1.12 | 30 KB |
\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#
| 100% readline-7.0 | 392 KB |
\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#
| 100% perl-bio-featureio-1 | 39 KB |
\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#
| 100% perl-io-string-1.08 | 10 KB |
\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#
| 100% fasttree-2.1.10 | 333 KB |
\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#
| 100% libtiff-4.0.10 | 604 KB |
\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#
| 100% java-jdk-7.0.91 | 35.7 MB |
\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#
| 100% perl-storable-3.15 | 63 KB |
\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#
| 100% perl-yaml-1.29 | 41 KB |
\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#
| 100% perl-xml-libxml-2.01 | 260 KB |
\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#
| 100% attrs-19.3.0 | 35 KB |
\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#
| 100% \_r-mutex-1.0.0 | 3 KB |
\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#
| 100% perl-xml-regexp-0.04 | 8 KB |
\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#
| 100% perl-encode-locale-1 | 11 KB |
\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#
| 100% perl-db-file-1.852 | 52 KB |
\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#
| 100% perl-eval-closure-0. | 11 KB |
\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#
| 100% pysocks-1.7.1 | 27 KB |
\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#
| 100% perl-bio-tools-phylo | 56 KB |
\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#
| 100% perl-xml-namespacesu | 11 KB |
\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#
| 100% perl-bioperl-run-1.0 | 432 KB |
\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#
| 100% libgcc-7.2.0 | 304 KB |
\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#
| 100% perl-bio-tools-run-a | 23 KB |
\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#
| 100% perl-http-daemon-6.0 | 14 KB |
\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#
| 100% hmmer-3.3 | 4.1 MB |
\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#
| 100% mkl\_fft-1.0.15 | 173 KB |
\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#
| 100% perl-set-scalar-1.29 | 19 KB |
\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#
| 100% perl-svg-graph-0.02 | 26 KB |
\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#
| 100% perl-bio-coordinate- | 34 KB |
\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#
| 100% cyrus-sasl-2.1.26 | 279 KB |
\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#
| 100% perl-class-data-inhe | 7 KB |
\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#
| 100% seqprep-1.3.2 | 38 KB |
\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#
| 100% perl-svg-2.84 | 37 KB |
\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#
| 100% ftputil-3.2 | 85 KB |
\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#
| 100% perl-unicode-map-0.1 | 254 KB |
\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#
| 100% zstd-1.3.7 | 887 KB |
\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#
| 100% perl-soap-lite-1.19 | 117 KB |
\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#
| 100% perl-bioperl-core-1. | 3.2 MB |
\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#
| 100% perl-aceperl-1.92 | 149 KB |
\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#
| 100% emboss-6.6.0 | 94.5 MB |
\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#
| 100% perl-compress-raw-zl | 72 KB |
\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#
| 100% qt-5.6.3 | 45.7 MB |
\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#
| 100% perl-bio-asn1-entrez | 23 KB |
\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#
| 100% libwebp-1.0.1 | 913 KB |
\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#
| 100% typing\_extensions-3. | 38 KB |
\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#
| 100% perl-json-xs-2.34 | 63 KB |
\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#
| 100% appdirs-1.4.3 | 16 KB |
\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#
| 100% libpng-1.6.37 | 364 KB |
\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#
| 100% perl-array-compare-3 | 12 KB |
\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#
| 100% perl-types-serialise | 10 KB |
\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#
| 100% perl-bioperl-1.7.2 | 13 KB |
\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#
| 100% perl-xml-twig-3.52 | 144 KB |
\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#
| 100% pandoc-2.2.3.2 | 20.8 MB |
\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#
| 100% binutils\_linux-64-2. | 148 KB |
\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#
| 100% font-ttf-ubuntu-0.83 | 1.8 MB |
\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#
| 100% mkl-service-2.3.0 | 208 KB |
\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#
| 100% perl-dynaloader-1.25 | 3 KB |
\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#
| 100% perl-text-csv-2.00 | 66 KB |
\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#
| 100% libgcc-ng-9.1.0 | 8.1 MB |
\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#
| 100% perl-mailtools-2.21 | 48 KB |
\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#
| 100% dnaio-0.3 | 130 KB |
\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#
| 100% fribidi-1.0.5 | 112 KB |
\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#
| 100% perl-http-cookies-6. | 19 KB |
\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#
| 100% perl-template-toolki | 319 KB |
\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#
| 100% perl-test-simple-1.3 | 132 KB |
\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#
| 100% perl-role-tiny-2.000 | 15 KB |
\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#
| 100% pynacl-0.3.0 | 237 KB |
\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#
| 100% perl-mime-base64-3.1 | 15 KB |
\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#
| 100% perl-xml-dom-xpath-0 | 9 KB |
\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#
| 100% clustalw-2.1 | 308 KB |
\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#
| 100% perl-io-html-1.001 | 13 KB |
\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#
| 100% perl-xml-simple-2.25 | 48 KB |
\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#
| 100% mkl\_random-1.1.0 | 369 KB |
\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#
| 100% perl-math-derivative | 9 KB |
\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#
| 100% cairo-1.14.12 | 1.3 MB |
\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#
| 100% curl-7.65.3 | 141 KB |
\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#
| 100% xz-5.2.4 | 366 KB |
\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#
| 100% perl-http-message-6. | 51 KB |
\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#
| 100% perl-business-isbn-d | 15 KB |
\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#
| 100% binutils\_impl\_linux- | 16.5 MB |
\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#
| 100% perl-capture-tiny-0. | 17 KB |
\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#
| 100% perl-error-0.17027 | 20 KB |
\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#
| 100% perl-ipc-sharelite-0 | 24 KB |
\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#
| 100% perl-bio-phylo-0.58 | 301 KB |
\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#
| 100% perl-sub-install-0.9 | 10 KB |
\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#
| 100% perl-package-stash-0 | 66 KB |
\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#
| 100% libopenblas-0.3.6 | 7.7 MB |
\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#
| 100% perl-xml-xpath-1.44 | 42 KB |
\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#
| 100% perl-package-depreca | 10 KB |
\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#
| 100% perl-math-round-0.07 | 9 KB |
\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#
| 100% perl-module-runtime- | 7 KB |
\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#
| 100% numpy-base-1.17.4 | 5.3 MB |
\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#
| 100% perl-digest-perl-md5 | 11 KB |
\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#
| 100% libgd-2.2.5 | 290 KB |
\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#
| 100% libsodium-1.0.16 | 302 KB |
\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#
| 100% psutil-5.6.7 | 328 KB |
\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#
| 100% perl-xml-sax-writer- | 16 KB |
\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#
| 100% perl-try-tiny-0.30 | 17 KB |
\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#
| 100% pytz-2019.3 | 237 KB |
\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#
| 100% perl-mldbm-2.05 | 15 KB |
\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#
| 100% perl-type-tiny-1.004 | 119 KB |
\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#
| 100% perl-cache-cache-1.0 | 31 KB |
\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#
| 100% perl-tree-dag\_node-1 | 40 KB |
\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#
| 100% dbus-1.13.12 | 611 KB |
\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#
| 100% perl-test-harness-3. | 82 KB |
\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#
| 100% perl-libwww-perl-6.3 | 94 KB |
\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#
| 100% perl-file-listing-6. | 9 KB |
\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#
| 100% perl-devel-stacktrac | 16 KB |
\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#
| 100% perl-pathtools-3.75 | 41 KB |
\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#
| 100% perl-autoloader-5.74 | 5 KB |
\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#
| 100% perl-graph-0.9704 | 73 KB |
\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#
| 100% unixodbc-2.3.7 | 319 KB |
\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#
| 100% pytools-2019.1.1 | 52 KB |
\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#
| 100% fontconfig-2.13.0 | 291 KB |
\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#
| 100% snakemake-4.0.0 | 210 KB |
\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#
| 100% perl-module-implemen | 9 KB |
\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#
| 100% pysftp-0.2.9 | 31 KB |
\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#
| 100% mafft-7.455 | 3.1 MB |
\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#
| 100% perl-io-socket-ssl-2 | 151 KB |
\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#
| 100% perl-bio-tools-run-a | 24 KB |
\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#
| 100% perl-lib-0.63 | 6 KB |
\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#
| 100% perl-spreadsheet-wri | 259 KB |
\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#
| 100% perl-data-optlist-0. | 10 KB |
\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#
| 100% perl-xml-sax-expat-0 | 10 KB |
\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#
| 100% perl-file-temp-0.230 | 26 KB |
\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#
| 100% perl-mime-lite-3.030 | 51 KB |
\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#
| 100% perl-mro-compat-0.13 | 10 KB |
\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#
| 100% perl-extutils-makema | 153 KB |
\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#
| 100% perl-algorithm-diff- | 34 KB |
\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#
| 100% expat-2.2.6 | 187 KB |
\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#
| 100% perl-statistics-desc | 32 KB |
\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#
| 100% perl-bio-samtools-1. | 172 KB |
\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#
| 100% perl-spreadsheet-par | 63 KB |
\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#
| 100% cffi-1.13.2 | 224 KB |
\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#
| 100% perl-mozilla-ca-2018 | 140 KB |
\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#
| 100% perl-file-path-2.16 | 17 KB |
\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#
| 100% python-dateutil-2.8. | 220 KB |
\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#
| 100% pixman-0.38.0 | 618 KB |
\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#
| 100% perl-xml-writer-0.62 | 18 KB |
\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#
| 100% perl-base-2.23 | 10 KB |
\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#
| 100% \_libgcc\_mutex-0.1 | 2 KB |
\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#
| 100% perl-json-4.02 | 52 KB |
\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#
| 100% perl-sub-name-0.21 | 13 KB |
\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#
| 100% paml-4.9 | 1.5 MB |
\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#
| 100% font-ttf-source-code | 922 KB |
\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#
| 100% libdb-6.1.26 | 21.5 MB |
\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#
| 100% perl-dbd-sqlite-1.64 | 2.3 MB |
\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#
| 100% idna-2.8 | 133 KB |
\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#
| 100% graphviz-2.40.1 | 6.9 MB |
\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#
| 100% perl-test-exception- | 12 KB |
\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#
| 100% urllib3-1.24.2 | 148 KB |
\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#
| 100% perl-image-size-3.30 | 26 KB |
\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#
| 100% make-4.2.1 | 429 KB |
\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#
| 100% idna\_ssl-1.1.0 | 6 KB |
\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#
| 100% perl-getopt-long-2.5 | 27 KB |
\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#
| 100% perl-parse-recdescen | 71 KB |
\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#
| 100% mkl-2019.4 | 204.1 MB |
\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#
| 100% perl-scalar-list-uti | 38 KB |
\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#
| 100% pyyaml-5.1.2 | 189 KB |
\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#
| 100% perl-package-stash-x | 21 KB |
\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#
| 100% gst-plugins-base-1.1 | 6.3 MB |
\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#
| 100% perl-sub-uplevel-0.2 | 13 KB |
\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#
| 100% perl-digest-hmac-1.0 | 9 KB |
\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#
| 100% decorator-4.4.0 | 18 KB |
\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#
| 100% perl-net-ssleay-1.88 | 289 KB |
\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#
| 100% libssh2-1.8.2 | 250 KB |
\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#
| 100% perl-statistics-lite | 9 KB |
\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#
| 100% gstreamer-1.14.0 | 3.8 MB |
\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#
| 100% tk-8.6.8 | 3.1 MB |
\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#
| 100% gblocks-0.91b | 32 KB |
\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#
| 100% sqlite-3.31.1 | 2.0 MB |
\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#
| 100% certifi-2019.11.28 | 149 KB |
\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#
| 100% libxslt-1.1.33 | 577 KB |
\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#
| 100% perl-html-tagset-3.2 | 12 KB |
\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#
| 100% tktable-2.10 | 88 KB |
\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#
| 100% aiohttp-3.6.2 | 621 KB |
\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#
| 100% perl-image-info-1.38 | 41 KB |
\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#
| 100% pandas-0.25.3 | 11.5 MB |
\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#
| 100% pycrypto-2.6.1 | 465 KB |
\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#
| 100% t\_coffee-11.0.8 | 33.2 MB |
\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#
| 100% perl-moose-2.2011 | 444 KB |
\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#
| 100% pcre-8.43 | 260 KB |
\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#
| 100% libntlm-1.5 | 35 KB |
\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#
| 100% perl-bio-eutilities- | 67 KB |
\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#
| 100% perl-text-diff-1.45 | 20 KB |
\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#
| 100% perl-test-most-0.35 | 18 KB |
\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#
| 100% perl-time-hires-1.97 | 25 KB |
\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#
| 100% perl-lwp-mediatypes- | 22 KB |
\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#
| 100% perl-http-date-6.02 | 10 KB |
\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#
| 100% perl-devel-overloadi | 8 KB |
\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#
| 100% gxx\_impl\_linux-64-7. | 18.7 MB |
\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#
| 100% perl-pod-escapes-1.0 | 10 KB |
\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#
| 100% fonts-anaconda-1 | 2 KB |
\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#
| 100% dropbox-5.2.1 | 269 KB |
\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#
| 100% perl-params-util-1.0 | 14 KB |
\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#
| 100% perl-convert-binary- | 252 KB |
\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#
| 100% sed-4.7 | 245 KB |
\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#
| 100% perl-math-cdf-0.1 | 69 KB |
\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#
| 100% perl-xml-dom-1.46 | 56 KB |
\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#
| 100% perl-sub-identify-0. | 12 KB |
\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#
| 100% perl-class-method-mo | 13 KB |
\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#
| 100% perl-module-runtime- | 15 KB |
\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#
| 100% setuptools-42.0.2 | 648 KB |
\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#
| 100% python-3.6.9 | 34.4 MB |
\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#
| 100% perl-appconfig-1.71 | 42 KB |
\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#
| 100% perl-gd-2.71 | 114 KB |
\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#
| 100% glib-2.56.2 | 5.0 MB |
\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#
| 100% perl-html-parser-3.7 | 66 KB |
\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#
| 100% perl-xml-xpathengine | 26 KB |
\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#
| 100% perl-digest-md5-2.55 | 18 KB |
\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#
| 100% six-1.13.0 | 22 KB |
\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#
| 100% pip-20.0.2 | 1.9 MB |
\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#
| 100% perl-sub-exporter-pr | 8 KB |
\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#
| 100% ca-certificates-2020 | 132 KB |
\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#
| 100% libstdcxx-ng-9.1.0 | 4.0 MB |
\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#
| 100% perl-html-tableextra | 23 KB |
\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#
| 100% perl-jcode-2.07 | 24 KB |
\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#
| 100% perl-cgi-4.44 | 116 KB |
\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#
| 100% openjdk-8.0.152 | 128.6 MB |
\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#
| 100% perl-common-sense-3. | 11 KB |
\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#
| 100% perl-http-negotiate- | 13 KB |
\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#
| 100% giflib-5.1.4 | 78 KB |
\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#
| 100% libcurl-7.65.3 | 588 KB |
\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#
| 100% perl-socket-2.027 | 31 KB |
\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#
| 100% perl-test-requiresin | 7 KB |
\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#
| 100% multidict-4.5.2 | 142 KB |
\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#
| 100% gzip-1.7 | 181 KB |
\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#
| 100% perl-moo-2.003004 | 38 KB |
\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#
| 100% perl-io-sessiondata- | 8 KB |
\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#
| 100% openssl-1.1.1d | 3.7 MB |
\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#
| 100% filechunkio-1.6 | 7 KB |
\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#
| 100% perl-html-tidy-1.60 | 20 KB |
\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#
| 100% libuuid-1.0.3 | 16 KB |
\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#
| 100% perl-pdf-api2-2.035 | 2.3 MB |
\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#
| 100% perl-5.26.2 | 15.9 MB |
\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#
| 100% perl-xml-sax-1.02 | 36 KB |
\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#
| 100% cutadapt-2.6 | 173 KB |
\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#
| 100% perl-carp-1.38 | 15 KB |
\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#
| 100% font-ttf-dejavu-sans | 386 KB |
\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#
| 100% bzip2-1.0.8 | 105 KB |
\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#
| 100% perl-timedate-2.30 | 29 KB |
\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#
| 100% perl-encode-2.88 | 2.1 MB |
\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#
| 100% tidyp-1.04 | 288 KB |
\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#
| 100% perl-test-difference | 15 KB |
\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#
| 100% rename-1.601 | 12 KB |
\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#
| 100% perl-ntlm-1.09 | 15 KB |
\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#
| 100% perl-lwp-simple-6.15 | 348 KB |
\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#
| 100% libxml2-2.9.9 | 2.0 MB |
\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#
| 100% perl-html-entities-n | 11 KB |
\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#
| 100% libedit-3.1.20181209 | 188 KB |
\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#
| 100% perl-business-isbn-3 | 15 KB |
\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#
| 100% perl-data-dumper-2.1 | 35 KB |
\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#
| 100% perl-sort-naturally- | 14 KB |
\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#
| 100% perl-task-weaken-1.0 | 8 KB |
\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#
| 100% perl-data-stag-0.14 | 78 KB |
\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#
| 100% perl-html-tree-5.07 | 129 KB |
\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#
| 100% pyopenssl-19.1.0 | 83 KB |
\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#
| 100% freetype-2.9.1 | 822 KB |
\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#
| 100% perl-html-formatter- | 30 KB |
\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#
| 100% perl-email-date-form | 8 KB |
\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#
| 100% async-timeout-3.0.1 | 12 KB |
\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#
| 100% asn1crypto-1.2.0 | 158 KB |
\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#
| 100% perl-convert-binhex- | 25 KB |
\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#
| 100% numpy-1.17.4 | 4 KB |
\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#
| 100% perl-digest-sha1-2.1 | 20 KB |
\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#
| 100% itsx-1.1b | 4.1 MB |
\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#
| 100% xopen-0.7.3 | 11 KB |
\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#
| 100% perl-html-element-ex | 24 KB |
\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#
| 100% vsearch-2.14.1 | 1.2 MB |
\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#
| 100% yarl-1.3.0 | 133 KB |
\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#
| 100% zlib-1.2.11 | 120 KB |
\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#
| 100% perl-font-afm-1.20 | 14 KB |
\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#
| 100% perl-algorithm-munkr | 11 KB |
\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#
| 100% perl-devel-globaldes | 7 KB |
\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#
| 100% perl-xml-filter-buff | 7 KB |
\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#
| 100% pigz-2.4 | 79 KB |
\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#
| 100% perl-class-load-xs-0 | 13 KB |
\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#
| 100% bcrypt-3.1.7 | 44 KB |
\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#
| 100% perl-math-random-0.7 | 55 KB |
\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#
| 100% perl-app-cpanminus-1 | 234 KB |
\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#
| 100% perl-time-local-1.28 | 10 KB |
\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#
| 100% biopython-1.76 | 2.6 MB |
\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#
| 100% perl-file-sort-1.01 | 18 KB |
\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#
| 100% harfbuzz-1.8.8 | 863 KB |
\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#
| 100% perl-class-load-0.25 | 12 KB |
\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#
| 100% wheel-0.33.6 | 35 KB |
\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#
| 100% perl-test-deep-1.128 | 36 KB |
\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#
| 100% font-ttf-inconsolata | 95 KB |
\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#
| 100% requests-2.22.0 | 84 KB |
\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#
| 100% perl-file-which-1.23 | 12 KB |
\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#
| 100% wrapt-1.11.2 | 48 KB |
\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#
| 100% perl-constant-1.33 | 10 KB |
\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#
| 100% perl-test-leaktrace- | 23 KB |
\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#
| 100% perl-apache-test-1.4 | 115 KB |
\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#
| 100% perl-test-warn-0.36 | 13 KB |
\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#
| 100% perl-date-format-2.3 | 29 KB |
\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#
| 100% Preparing transaction: done Verifying transaction: done Executing
transaction: done \# \# To activate this environment, use \# \# $ conda
activate SCVUCv4.3 \# \# To deactivate an active environment, use \# \#
$ conda deactivate

\[mattbowser@yeti-login20 SCVUC\_COI\_metabarcode\_pipeline\] conda
activate SCVUCv4.3

CommandNotFoundError: Your shell has not been properly configured to use
‘conda activate’. To initialize your shell, run

    $ conda init <SHELL_NAME>

Currently supported shells are: - bash - fish - tcsh - xonsh - zsh -
powershell

See ‘conda init –help’ for more information and options.

IMPORTANT: You may need to close and restart your shell after running
‘conda init’.

\[mattbowser@yeti-login20 SCVUC\_COI\_metabarcode\_pipeline\] source
activate SCVUCv4.3 (SCVUCv4.3) \[mattbowser@yeti-login20
SCVUC\_COI\_metabarcode\_pipeline\] wget
ftp://ftp.ncbi.nlm.nih.gov/genomes/TOOLS/ORFfinder/linux-i64/ORFfinder.gz
–2020-03-19 12:09:55–
ftp://ftp.ncbi.nlm.nih.gov/genomes/TOOLS/ORFfinder/linux-i64/ORFfinder.gz
=\> “ORFfinder.gz” Resolving ftp.ncbi.nlm.nih.gov… 130.14.250.10,
2607:f220:41e:250::13 Connecting to
ftp.ncbi.nlm.nih.gov|130.14.250.10|:21… connected. Logging in as
anonymous … Logged in\! ==\> SYST … done. ==\> PWD … done. ==\> TYPE I …
done. ==\> CWD (1) /genomes/TOOLS/ORFfinder/linux-i64 … done. ==\> SIZE
ORFfinder.gz … 7932592 ==\> PASV … done. ==\> RETR ORFfinder.gz … done.
Length: 7932592 (7.6M) (unauthoritative)

100%\[==================================================================================================================================\>\]
7,932,592 1.50M/s in 5.0s

2020-03-19 12:10:03 (1.50 MB/s) - “ORFfinder.gz” saved \[7932592\]

(SCVUCv4.3) \[mattbowser@yeti-login20
SCVUC\_COI\_metabarcode\_pipeline\] gunzip ORFfinder.gz (SCVUCv4.3)
\[mattbowser@yeti-login20 SCVUC\_COI\_metabarcode\_pipeline\] chmod 755
ORFfinder (SCVUCv4.3) \[mattbowser@yeti-login20
SCVUC\_COI\_metabarcode\_pipeline\] mv ORFfinder \~/bin/. mv: cannot
move `ORFfinder' to`/home/mattbowser/bin/.’: No such file or directory
(SCVUCv4.3) \[mattbowser@yeti-login20
SCVUC\_COI\_metabarcode\_pipeline\] mkdir /home/mattbowser/bin
(SCVUCv4.3) \[mattbowser@yeti-login20
SCVUC\_COI\_metabarcode\_pipeline\] mv ORFfinder \~/bin/. (SCVUCv4.3)
\[mattbowser@yeti-login20 SCVUC\_COI\_metabarcode\_pipeline\] ORFfinder
ORFfinder: /lib64/libc.so.6: version `GLIBC_2.17' not found (required by
ORFfinder) ORFfinder: /lib64/libc.so.6: version`GLIBC\_2.14’ not found
(required by ORFfinder) (SCVUCv4.3) \[mattbowser@yeti-login20
SCVUC\_COI\_metabarcode\_pipeline\] cd
\~/miniconda3/envs/SCVUCv4.3/etc/conda/activate.d/ -bash: cd:
/home/mattbowser/miniconda3/envs/SCVUCv4.3/etc/conda/activate.d/: No
such file or directory (SCVUCv4.3) \[mattbowser@yeti-login20
SCVUC\_COI\_metabarcode\_pipeline\] cd \~/ (SCVUCv4.3)
\[mattbowser@yeti-login20 \~\] ls 2018\_AWCC\_soil\_fungi
2018\_AWCC\_soil\_fungi\_try\_02 bin perl5 pipits R (SCVUCv4.3)
\[mattbowser@yeti-login20 \~\] cd /

## Friday, March 20

To do:

  - *Refuge Notebook* catch-up.
  - Early adopter data management homework.
  - Blackfish diet analysis.

I set up a project on Arctos for the blakfish diet work at
<http://arctos.database.museum/project/10003367>.

I was able to get the SCVUC pipeline going under linux mint, but there
was a problem with the reverse primer. I did overcome this later by
reverting to the original reverse primer in the ml-jg script.

I kept running into trouble with running out of memory for the RDP
Classifier (Wang et al. [2007](#ref-wang_naive_2007)). The system would
kill the process after it used all memory. The script asks for 8 Gb of
RAM; this computer has 4. I tried modifying the script to ask for only 2
Gb, but then RDP Classifier failed within Java, again running out of
memory. I will try repartitioning to add a bunch of swap space on that
computer.

### Telecon at 11:00

  - Send Mark a list of planned work for next week.
  - Put together plans for this field season.

Next week work list:

  - Finish blackfish diet analysis to get an article into the upcoming
    issue of the *Newsletter of the Alaska Entomological Society*.
  - Catch up on formatting *Refuge Notebook* articles and posting them
    to the CMS.
  - Work on metadata / next data management steps for the Kenai NWR
    snowshoe hare dataset as part of work for the regional data
    management early adopters’ team.
  - Write article on our finding *Bimastos parvus*, a second native
    earthworm for Alaska, for the *AKES Newsletter*?
  - Get started on Slikok watershed project occupancy analysis?

<!-- end list -->

    cd /home/matt/SCVUC_COI_metabarcode_pipeline
    snakemake --jobs 1 --snakefile snakefile_withPseudogeneFiltering --configfile 2020-03-20-0747_config_ml-jg.yaml
    
    cutadapt: error: Character 'I' in adapter sequence 'TAIACYTCIGGRTGICCRAARAAYCA' is not a valid IUPAC code. Use only characters XACGTURYSWKMBDHVN.
    
    Subbing "D" for "I". See <https://en.wikipedia.org/wiki/Nucleotide>.
    
    snakemake --jobs 1 --snakefile snakefile_withPseudogeneFiltering --configfile 2020-03-20-0747_config_ml-jg.yaml
    
    Fatal error: Unable to read from file (ml-jg/cat.denoised)
    Error in job chimera_removal while creating output file ml-jg/cat.denoised.nonchimeras.
    RuleException:
    CalledProcessError in line 412 of /home/matt/SCVUC_COI_metabarcode_pipeline/snakefile_withPseudogeneFiltering:
    Command 'vsearch --uchime3_denovo ml-jg/cat.denoised --sizein --xsize --nonchimeras ml-jg/cat.denoised.nonchimeras --relabel 'Zotu' --log ml-jg/chimeraRemoval.log' returned non-zero exit status 1.
      File "/home/matt/SCVUC_COI_metabarcode_pipeline/snakefile_withPseudogeneFiltering", line 412, in __rule_chimera_removal
      File "/home/matt/miniconda3/envs/SCVUCv4.3/lib/python3.6/concurrent/futures/thread.py", line 56, in run
    Removing output files of failed job chimera_removal since they might be corrupted:
    ml-jg/cat.denoised.nonchimeras
    Will exit after finishing currently running jobs.
    Exiting because a job execution failed. Look above for error message
    
    This seems to have had problems with the reverse primer. After trimming the reverse primer the files were quite small.
    
    TGRTTYTTYGGNCAYCCNGARGTNTA ## From original config file
    TADACYTCDGGRTGDCCRAARAAYCA ## What I ran last
    TAIACYTCIGGRTGICCRAARAAYCA ## From lab
    TANACYTCNGGRTGNCCRAARAAYCA ## Going to try this.
    
    stop @ 08:50 start @ 09:10
    
    snakemake --jobs 1 --snakefile snakefile_withPseudogeneFiltering --configfile 2020-03-20-0747_config_ml-jg.yaml
    
    ## That did not work, either.
    ## Switching back to the original.
    
    /bin/bash: line 1:  6375 Killed                  java -Xmx8g -jar /home/matt/SCVUC_COI_metabarcode_pipeline/rdp_classifier_2.12/dist/classifier.jar classify -t /home/matt/SCVUC_COI_metabarcode_pipeline/mydata_trained/rRNAClassifier.properties -o ml-jg/rdp.out.tmp ml-jg/cat.denoised.nonchimeras
    Error in job taxonomic_assignment while creating output file ml-jg/rdp.out.tmp.
    RuleException:
    CalledProcessError in line 445 of /home/matt/SCVUC_COI_metabarcode_pipeline/snakefile_withPseudogeneFiltering:
    Command 'java -Xmx8g -jar /home/matt/SCVUC_COI_metabarcode_pipeline/rdp_classifier_2.12/dist/classifier.jar classify -t /home/matt/SCVUC_COI_metabarcode_pipeline/mydata_trained/rRNAClassifier.properties -o ml-jg/rdp.out.tmp ml-jg/cat.denoised.nonchimeras' returned non-zero exit status 137.
      File "/home/matt/SCVUC_COI_metabarcode_pipeline/snakefile_withPseudogeneFiltering", line 445, in __rule_taxonomic_assignment
      File "/home/matt/miniconda3/envs/SCVUCv4.3/lib/python3.6/concurrent/futures/thread.py", line 56, in run
    Removing output files of failed job taxonomic_assignment since they might be corrupted:
    ml-jg/rdp.out.tmp
    Will exit after finishing currently running jobs.
    Exiting because a job execution failed. Look above for error message
    
    ## Made that jar executable.
    
    ## Trying again.
    snakemake --jobs 1 --snakefile snakefile_withPseudogeneFiltering --configfile 2020-03-20-0747_config_ml-jg.yaml
    
    Error in job taxonomic_assignment while creating output file ml-jg/rdp.out.tmp.
    RuleException:
    CalledProcessError in line 445 of /home/matt/SCVUC_COI_metabarcode_pipeline/snakefile_withPseudogeneFiltering:
    Command 'java -Xmx8g -jar /home/matt/SCVUC_COI_metabarcode_pipeline/rdp_classifier_2.12/dist/classifier.jar classify -t /home/matt/SCVUC_COI_metabarcode_pipeline/mydata_trained/rRNAClassifier.properties -o ml-jg/rdp.out.tmp ml-jg/cat.denoised.nonchimeras' returned non-zero exit status 137.
      File "/home/matt/SCVUC_COI_metabarcode_pipeline/snakefile_withPseudogeneFiltering", line 445, in __rule_taxonomic_assignment
      File "/home/matt/miniconda3/envs/SCVUCv4.3/lib/python3.6/concurrent/futures/thread.py", line 56, in run
    Removing output files of failed job taxonomic_assignment since they might be corrupted:
    ml-jg/rdp.out.tmp
    Will exit after finishing currently running jobs.
    Exiting because a job execution failed. Look above for error message
    
    ## Just trying that failed command.
    java -Xmx8g -jar /home/matt/SCVUC_COI_metabarcode_pipeline/rdp_classifier_2.12/dist/classifier.jar classify -t /home/matt/SCVUC_COI_metabarcode_pipeline/mydata_trained/rRNAClassifier.properties -o ml-jg/rdp.out.tmp ml-jg/cat.denoised.nonchimeras
    
    ## This was killed by the system, I think from using too much memory.
    
    Mar 20 09:59:42 aurochs kernel: [ 8396.151925] Out of memory: Killed process 6840 (java) total-vm:10309908kB, anon-rss:3333148kB, file-rss:0kB, shmem-rss:0kB
    
    ## Oh, I see. It is asking for 8 Gb of memory on a 4 Gb memory machine.
    -Xmx8g
    
    ## Edited snakefile_withPseudogeneFiltering, saved as snakefile_withPseudogeneFiltering_2020-03-20-1141
    
    snakemake --jobs 1 --snakefile snakefile_withPseudogeneFiltering_2020-03-20-1141 --configfile 2020-03-20-0747_config_ml-jg.yaml
    
    That failed, also. This thing is really hungry for memory.
    
    Exception in thread "main" java.lang.OutOfMemoryError: Java heap space
        at java.util.Arrays.copyOf(Arrays.java:3210)
        at java.util.Arrays.copyOf(Arrays.java:3181)
        at java.util.ArrayList.grow(ArrayList.java:261)
        at java.util.ArrayList.ensureExplicitCapacity(ArrayList.java:235)
        at java.util.ArrayList.ensureCapacityInternal(ArrayList.java:227)
        at java.util.ArrayList.add(ArrayList.java:458)
        at edu.msu.cme.rdp.classifier.io.GenusWordProbFileParser.createGenusWordProbList(GenusWordProbFileParser.java:59)
        at edu.msu.cme.rdp.classifier.TrainingInfo.createGenusWordProbList(TrainingInfo.java:127)
        at edu.msu.cme.rdp.classifier.utils.ClassifierFactory.getNonDefaultFactory(ClassifierFactory.java:161)
        at edu.msu.cme.rdp.classifier.utils.ClassifierFactory.getFactory(ClassifierFactory.java:145)
        at edu.msu.cme.rdp.multicompare.MultiClassifier.<init>(MultiClassifier.java:64)
        at edu.msu.cme.rdp.multicompare.MultiClassifier.<init>(MultiClassifier.java:75)
        at edu.msu.cme.rdp.multicompare.Main.main(Main.java:247)
        at edu.msu.cme.rdp.classifier.cli.ClassifierMain.main(ClassifierMain.java:67)
    
    ## Ok, I repartitioned the hard drive, making 24 Gb more of swap, so now we have 4 Gb of memory and 26 Gb of swap.
    
    cd SC*
    conda activate SCVUCv4.3
    snakemake --jobs 1 --snakefile snakefile_withPseudogeneFiltering_2020-03-20-1528 --configfile 2020-03-20-0747_config_ml-jg.yaml
    
    ## That worked!

## Saturday, March 21

I looked at results from yesterday and worked on adding identifications.

    ## Downloaded ASV Sequences from Bowser et al. (2020)
    
    Bowser, M. L. 2020. Supplementary material 5. ASV sequences. doi.org/10.3897/BDJ.8.e50124.suppl5
    
    oo_371419.fas
    
    cd SC*/ml-jg
    source activate SCVUCv4.3
    
    ## Assign taxonomy from Bowser (2020).
    vsearch --usearch_global chimera.denoised.nonchimeras.taxon \
    --db 2020-03-21-0600_oo_371419.fas \
    --id .90 \
    --iddef 0 \
    --blast6out 2020-03-21-0626_vsearch_Bowser_2020_matches.txt
    
    ## Assign taxonomy based on DS-BOWSER BOLD project.
    vsearch --usearch_global chimera.denoised.nonchimeras.taxon \
    --db 2020-03-21-0611_DS-BOWSER.fas \
    --id .90 \
    --iddef 0 \
    --blast6out 2020-03-21-0629_vsearch_DS-BOWSER_matches.txt
    
    ## Assign taxonomy based on AKTerrInvCOILib.
    vsearch --usearch_global chimera.denoised.nonchimeras.taxon \
    --db 2019-03-25-0832_clusters.fas \
    --id .90 \
    --iddef 0 \
    --blast6out 2020-03-21-0639_vsearch_AKTerrInvCOILib_matches.txt

I started looking at these identifications. I think it is interesting
that *Dallia pectoralis* did not make it through the filtering. This
makes me think that the filtering is too stringent because there were
*D. pectoralis* reads in the original FASTQ files.

## Monday, March 23

    ## Downloaded ASV Sequences from Bowser et al. (2020)
    
    Bowser, M. L. 2020. Supplementary material 5. ASV sequences. doi.org/10.3897/BDJ.8.e50124.suppl5
    
    oo_371419.fas
    
    cd SC*/ml-jg
    source activate SCVUCv4.3
    
    ## Assign taxonomy from Bowser (2020).
    vsearch --usearch_global chimera.denoised.nonchimeras.taxon \
    --db 2020-03-21-0600_oo_371419.fas \
    --id .90 \
    --iddef 0 \
    --blast6out 2020-03-21-0626_vsearch_Bowser_2020_matches.txt
    
    ## Assign taxonomy based on DS-BOWSER BOLD project.
    vsearch --usearch_global chimera.denoised.nonchimeras.taxon \
    --db 2020-03-21-0611_DS-BOWSER.fas \
    --id .90 \
    --iddef 0 \
    --blast6out 2020-03-21-0629_vsearch_DS-BOWSER_matches.txt
    
    ## Assign taxonomy based on AKTerrInvCOILib.
    vsearch --usearch_global chimera.denoised.nonchimeras.taxon \
    --db 2019-03-25-0832_clusters.fas \
    --id .90 \
    --iddef 0 \
    --blast6out 2020-03-21-0639_vsearch_AKTerrInvCOILib_matches.txt

``` r
setwd("D:/projects/blackfish/2020_blackfish_diet/2020-03_SCVUC/ml-jg")

results1 <- read.csv("results.csv", stringsAsFactors=FALSE)

length(levels(as.factor(results1$ml.jg_GlobalESV)))
[1] 212

aklib <- read.delim("2020-03-23-0659_vsearch_AKTerrInvCOILib_matches.txt", header=FALSE, stringsAsFactors=FALSE)

names(results1)[1] <- "ESV_name"
results1$ESV_name <- gsub("ml-jg_", "", results1$ESV_name)

## column names from http://www.metagenomics.wiki/tools/blast/blastn-output-format-6
names(aklib)[1:3] <- c("ESV_name", "AKlibprocessid", "pident")
```

I accidentally overwrote most of the rest of that R script.

I attended the 11:00 biology check-in telecon.

I prepared data for Rob Massengill, but I accidentally overwrote the R
script. I had downloaded occurrence data from the Vogel Lake vicinity
from GBIF (GBIF.org
[2020](#ref-gbiforg_gbif_2020-3)[i](#ref-gbiforg_gbif_2020-3)), Arctos,
and iNaturalist, then processed these. I had entered data for specimen
records
[KNWR:Ento:11383](http://arctos.database.museum/guid/KNWR:Ento:11383)
(*Gammarus lacustris*),
[KNWR:Ento:11384](http://arctos.database.museum/guid/KNWR:Ento:11384)
(*Caenis youngi*),
[KNWR:Inv:53](http://arctos.database.museum/guid/KNWR:Inv:53)
(*Sphaerium* sp. BOLD:AAG0345), and
[KNWR:Inv:54](http://arctos.database.museum/guid/KNWR:Inv:54)
(Glossiphoniidae).

## Tuesday, March 24

Picking up where I left off yesterday, preparing data for Rob.

``` r
## I did some editing by hand, adding lake_name and removing some terrestrial records.

setwd("D:/projects/Elodea/Sandpiper_Lake/2020-03-23_occurrence_data_for_Rob")

d4 <- read.csv("2020-03-24-0607_Vogel_Lake_vicinity_occurrence_data.csv", stringsAsFactors=FALSE)

library(maptools)
library(rgdal)

wgs84 <- CRS("+proj=longlat +ellps=WGS84 +datum=WGS84")

coordinates(d4) <- c("longitude", "latitude")
proj4string(d4) <- wgs84

writeOGR(
 d4, 
 dsn="2020-03-24-0607_Vogel_Lake_vicinity_occurrence_data.kml", 
 layer="occurrences", 
 driver="KML", 
 dataset_options=c("NameField=scientific_name"), 
 overwrite_layer=TRUE
 )

## I did some editing by hand.
d4 <- read.csv("2020-03-24-0654_Vogel_Lake_vicinity_occurrence_data.csv", stringsAsFactors=FALSE)

coordinates(d4) <- c("longitude", "latitude")
proj4string(d4) <- wgs84

writeOGR(
 d4, 
 dsn="2020-03-24-0654_Vogel_Lake_vicinity_occurrence_data.kml", 
 layer="occurrences", 
 driver="KML", 
 dataset_options=c("NameField=scientific_name"), 
 overwrite_layer=TRUE
 )

## Now for some summaries.
library(reshape2)

d4 <- read.csv("2020-03-24-0654_Vogel_Lake_vicinity_occurrence_data.csv", stringsAsFactors=FALSE)

d5 <- d4[,1:9]
d5$presence <- 1

d6 <- melt(d5, measure.vars="presence")

d7 <- dcast(d6, formula = ... ~ lake_name, fun.aggregate=length)
d7 <- d7[,c(1:8, 10:12)]
to10 <- function(x)
 {
 if (x > 0){1}
 else {0}
 }
d7[,9:11] <- apply(d7[,9:11], c(1,2), to10)
write.csv(d7, "2020-03-24-0728_Vogel_Lake_vicinity_occurrence_table.csv", row.names=FALSE)

## Now for an aggregate table.
d8 <- dcast(d6, formula = ... ~ ., fun.aggregate=length)
d8 <- d8[,c(9,1:8)]
write.csv(d8, "2020-03-24-0737_Vogel_Lake_vicinity_occrrence_aggregate.csv", row.names=FALSE)
```

I added the BOLD results to the results data frame from the blackfish
diet work.

``` r
setwd("D:/projects/blackfish/2020_blackfish_diet/2020-03_SCVUC/ml-jg")

r2 <- read.csv("results_2020-03-23-1201.csv", stringsAsFactors=FALSE)

brt <- read.csv("bold_id_engine_results_0001-0399.csv", stringsAsFactors=FALSE)

br <- brt

brt <- read.csv("bold_id_engine_results_0400-0799.csv", stringsAsFactors=FALSE)
br <- rbind(br, brt)

brt <- read.csv("bold_id_engine_results_0800-1079.csv", stringsAsFactors=FALSE)
names(brt) <- names(br)
br <- rbind(br, brt)

write.csv(br, "2020-03-24-0916_bold_results.csv", row.names=FALSE)

names(br)[c(1,2,4)] <- c("ESV_name", "bold_id", "bold_sim")

r3 <- merge(r2, br[,c(1,2,4)], all.x=TRUE)

write.csv(r3, "results_2020-03-23-1201.csv", row.names=FALSE)

## Now making a dataframe for working with identifications.
ids1 <- unique(r3[,c(1,6:41)])
dim(ids1)
[1] 212  37

write.csv(ids1, "2020-03-24-0925_ids.csv", row.names=FALSE)

## Now manually editing.
```

I worked on improving identifications from the blackfish diet work,
combing through BLAST results (temporary results URL:
<https://blast.ncbi.nlm.nih.gov/Blast.cgi?CMD=Get&RID=7MHZ12H101R>),
BOLD ID Engine results, and my own vsearch searches. I got through ESV
Zotu140.

## Wednesday, March 25

To do:

  - ~~Format and this week’s *Refuge Notebook* article to the
    *Clarion*.~~
  - *Refuge Notebook* catch-up.
  - Finish blackfish diet analysis.
  - Publish blackfish diet raw data.
  - Finish metadata for data management team meeting tomorrow.

I edited this week’s *Refuge Notebook* article.

I worked on identifications from the blackfish diet analysis, getting
the first round of IDs done. BOLD’s ID Engine apparently went down,
though, so I could not finish re-examining the rotifer sequences.

I attended the biology telecon at 11:00.

I finished editing this week’s *Refuge Notebook* article and submitted
it to the paper.

I finished scrutinizing identifications and examined the results using
R.

``` r
setwd("D:/projects/blackfish/2020_blackfish_diet/2020-03_SCVUC/ml-jg")

library(Biostrings)

ids1 <- read.csv("2020-03-25-1313_ids.csv", stringsAsFactors=FALSE)

r3 <- read.csv("results_2020-03-23-1201.csv", stringsAsFactors=FALSE)

dim(ids1)
[1] 212  40
dim(r3)
[1] 275  41

## How many ESVs have we accepted?
sum(ids1$accept==1)
[1] 131

## What is the set of identifications that were accepted?
accid <- levels(as.factor(ids1$scientific_name[ids1$accept==1]))

## How many unique identifications are there?
length(accid)
[1] 103

## I want to see what cutting out all reads except the nominal read length of 313 bp does.
fas1 <- readDNAStringSet("cat.denoised.nonchimeras")
ESV_name <- names(fas1)
fasd <- as.data.frame(ESV_name)
fasd$width <- width(fas1)

ids2 <- merge(ids1, fasd, all.x=TRUE)

## How many unique identifications are accepted and of read length 313?
accid313 <- levels(as.factor(ids2$scientific_name[ids2$accept==1 & ids2$width==313]))
length(accid313)
[1] 79 ## Wow, (103-79)/103 = 23% are dropped. What did we loose?

setdiff(accid, accid313)
 [1] "Aeshna juncea"                             "Aeshna sp. Zotu171"                       
 [3] "Bdelloidea sp. Zotu158"                    "Cyclopidae sp. Zotu83"                    
 [5] "Gyraulus sp. clade B TS-2018 BOLD:AEB5660" "Pleuroxus aduncus"                        
 [7] "Ploima sp. Zotu147"                        "Ploima sp. Zotu151"                       
 [9] "Ploima sp. Zotu153"                        "Ploima sp. Zotu187"                       
[11] "Ploima sp. Zotu188"                        "Procladius sp. BOLD:AAP3007"              
[13] "Rhamphomyia sp. BOLD:ACG3627"              "Rotaria sp. Zotu115"                      
[15] "Rotaria sp. Zotu215"                       "Rotifera sp. Zotu128"                     
[17] "Rotifera sp. Zotu179"                      "Rotifera sp. Zotu30"                      
[19] "Rotifera sp. Zotu40"                       "Rotifera sp. Zotu67"                      
[21] "Rotifera sp. Zotu72"                       "Rotifera sp. Zotu90"                      
[23] "Trichocerca sp. Zotu166"                   "Trichocerca sp. Zotu68"   
## Mostly rotifers are lost here.

## Just having a look at these records.
ids2[ids2$accept==1 & !ids2$width==313,]

## I looked at other files, also. I am going to keep those. I cannot loose that Zotu2, Aeshna juncea, one of the largest Zotus.

## Now I want to prepare a FASTA file to look at the tree.

keeperZotus <- ids1$ESV_name[ids1$accept==1]
keeperlab <- paste0(keeperZotus, " | ", ids1$scientific_name[ids1$accept==1])
keeperlab <- keeperlab[order(keeperlab)]

fask <- fas1[names(fas1) %in% keeperZotus]
fask <- fask[order(names(fask))]

names(fask) <- keeperlab

## Saving.
writeXStringSet(fask, "2020-03-25-1431_accepted_sequences.fas", format="fasta", width=1000)

## Meanwhile, I need to join data together.
r4 <- merge(r3[,1:3], ids2, all.x=TRUE)

## Now I want to remove occurrences with frequencies < 0.05%.

et <- read.delim("ESV.table")
names(et)[1] <- "ESV_name"
et$total <- apply(et[,2:6], 1, sum)
et$min_threshold <- et$total * 0.05/100

et_drop <- et[,1:6]
et_drop[,2:6] <- et[,2:6] <= et$min_threshold
et_drop[,2:6] <- et_drop[,2:6] + 0
et_drop[,2:6] <- et_drop[,2:6] * !et[,2:6]==0

## How many records would be dropped?
sum(et_drop[,2:6])
[1] 2

library(reshape2)
et2 <- melt(et_drop, id.vars="ESV_name")
et2 <- et2[et2$value==1,]
et2
     ESV_name variable value
760     Zotu2    MLB04     1
1015    Zotu4    MLB05     1

r5 <- r4
for (this_record in 1:nrow(et2))
 {
 slesv <- r5$ESV_name == et2$ESV_name[this_record]
 sls <- r5$SampleName == et2$variable[this_record]
 sl <- !(slesv & sls)
 print(sum(sl))
 r5 <- r5[sl,]
 }
 
## Saving here.
write.csv(r5, "results_2020-03-25-1552.csv", row.names=FALSE) 

## How many accepted sample * ESV records do we have?
sum(r5$accept==1)
[1] 167
```

I submitted the FASTA file of accepted sequences to NGPhylogeny.fr using
the “NGPhylogeny Analyse - FastME/OneClick” option. Results should be
available at
<https://ngphylogeny.fr/workspace/history/47db2b18949f0c42>.

## Thursday, March 26

To do:

  - Blackfish analysis.
  - Publish blackfish sequence data.
  - Early Adopters’ data management meeting at 11:00

The phylogenetic analysis from yesterday ran just fine.

For the purpose of keeping all ESV labels distinct for searchin on
Arctos or combining FASTA datasets, I want to relabel at ESVs. In the
output tree I relabeled “Zotu” to “bfdZotu”, with “bfd” standing for
“blackfish diet”. Resulting output tree on iTOL:
<https://itol.embl.de/tree/16415961276811585237767#>.

I worked on examining results of the blackfish work.

``` r
## Today I want to look at results.

setwd("D:/projects/blackfish/2020_blackfish_diet/2020-03_SCVUC/ml-jg")

r5 <- read.csv("results_2020-03-25-1552.csv", stringsAsFactors=FALSE)

guid <- read.csv("2020-03-26-0819_guids.csv", stringsAsFactors=FALSE)

r6 <- merge(r5, guid, all.x=TRUE)

## Renaming ESVs.
r6$ESV_name <- gsub("Zotu", "bfdZotu", r6$ESV_name)
r6$scientific_name <- gsub("Zotu", "bfdZotu", r6$scientific_name)

## Saving here.
write.csv(r6, "results_2020-03-26-0844.csv", row.names=FALSE)

## Now making a summary data frame for making an overall pie chart.
library(reshape2)

## In this one I will look at all ESVs.
sm1 <- melt(r6[r6$accept==1,c("scientific_name", "ESVsize")], measure.vars="ESVsize")
sm1 <- sm1[order(-sm1$value),]

pie(sm1$value, labels = sm1$scientific_name, main="ESV abundances, summed over all samples")

## Now I want to add percentages.
sm1$percentage <- round(sm1$value*100/sum(sm1$value))

pie(sm1$value, labels = paste0(sm1$scientific_name, " ", sm1$percentage, "%"), main="ESV abundances, summed over all samples")

png(filename="2020-03-26-0921_ESV_abundances.png",
 width=1000,
 height=600
 )
 pie(sm1$value, 
  labels = paste0(sm1$scientific_name, " ", sm1$percentage, "%"), 
  main="ESV abundances, summed over all samples"
  )
dev.off()

## Now for each fish.
for (this_fish in 1:5)
 {
 fish <- guid$guid[this_fish]
 smf <- melt(r6[(r6$accept==1 & r6$guid==fish),c("scientific_name", "ESVsize")], measure.vars="ESVsize")
 smf <- smf[order(-smf$value),]
 smf$percentage <- round(smf$value*100/sum(smf$value))
 png(filename=paste0("2020-03-26-0929_ESV_abundances_", this_fish, ".png"),
  width=1000,
  height=600
  )
  pie(smf$value, 
   labels = paste0(smf$scientific_name, " ", smf$percentage, "%"), 
   main=paste("ESV abundances, blackfish", fish)
   )
 dev.off()
 }
## It would be nice if the colors were consistent.

## Downloaded color table from https://en.wikipedia.org/wiki/List_of_colors:_A%E2%80%93F
cdf <- read.csv("2020-03-26-1055_cols_a-f.csv", stringsAsFactors=FALSE)

set.seed(42) 
r6$col <- sample(x=cdf$hex, size=nrow(r6), replace = FALSE)

## Remaking some of those pie charts with new, consistent colors.
sm1 <- melt(r6[r6$accept==1,c("scientific_name", "col", "ESVsize")], measure.vars="ESVsize")
sm1 <- sm1[order(-sm1$value),]
sm1$percentage <- round(sm1$value*100/sum(sm1$value))

png(filename="2020-03-26-1103_ESV_abundances.png",
 width=1000,
 height=600
 )
 pie(sm1$value, 
  labels = paste0(sm1$scientific_name, " ", sm1$percentage, "%"), 
  main="ESV abundances, summed over all samples",
  col=sm1$col
  )
dev.off() 

## Remaking pie charts for each fish.
for (this_fish in 1:5)
 {
 fish <- guid$guid[this_fish]
 smf <- melt(r6[(r6$accept==1 & r6$guid==fish),c("scientific_name", "col", "ESVsize")], measure.vars="ESVsize")
 smf <- smf[order(-smf$value),]
 smf$percentage <- round(smf$value*100/sum(smf$value))
 png(filename=paste0("2020-03-26-1109_ESV_abundances_", this_fish, ".png"),
  width=1000,
  height=600
  )
  pie(smf$value, 
   labels = paste0(smf$scientific_name, " ", smf$percentage, "%"), 
   main=paste("ESV abundances, blackfish", fish),
   col=smf$col
   )
 dev.off()
 }

## Now trying to make a bigger plot for the individual fish.
png(filename="2020-03-26-1131_ESV_abundances.png",
 width=1000*2,
 height=600*3
 ) 
 par(mfrow=c(3,2))
 par(mar=c(0,0,1,0))
 plot.new()
for (this_fish in 1:5)
 {
 fish <- guid$guid[this_fish]
 smf <- melt(r6[(r6$accept==1 & r6$guid==fish),c("scientific_name", "col", "ESVsize")], measure.vars="ESVsize")
 smf <- smf[order(-smf$value),]
 smf$percentage <- round(smf$value*100/sum(smf$value))
  pie(smf$value, 
   labels = paste0(smf$scientific_name, " ", smf$percentage, "%"), 
   main=paste("ESV abundances, blackfish", fish),
   col=smf$col
   )
 }
dev.off() 

## Putting everything on one plot.
pdf(file="2020-03-26-1141_ESV_abundances.pdf",
 width=10*2,
 height=6*3,
 pointsize=12
 ) 
 par(mfrow=c(3,2))
 par(mar=c(0,0,1,0))
 pie(sm1$value, 
  labels = paste0(sm1$scientific_name, " ", sm1$percentage, "%"), 
  main="ESV abundances, summed over all samples",
  col=sm1$col,
  radius=0.8
  )
for (this_fish in 1:5)
 {
 fish <- guid$guid[this_fish]
 smf <- melt(r6[(r6$accept==1 & r6$guid==fish),c("scientific_name", "col", "ESVsize")], measure.vars="ESVsize")
 smf <- smf[order(-smf$value),]
 smf$percentage <- round(smf$value*100/sum(smf$value))
  pie(smf$value, 
   labels = paste0(smf$scientific_name, " ", smf$percentage, "%"), 
   main=paste("ESV abundances, blackfish", fish),
   col=smf$col
   )
 }
dev.off() 

## I might try to make the areas proportional to the number of reads.
rtdf <- melt(r6[(r6$accept==1),c("guid", "ESVsize")], measure.vars="ESVsize")
rtdf <- dcast(rtdf, guid ~ ., sum)

tr <- sum(r6$ESVsize[r6$accept==1])
ta <- pi*0.8^2 ## Area of the summed circle.

rtdf$a <- rtdf[,2]*ta/tr
rtdf$r <- (rtdf$a/pi)^0.5

## Putting everything on one plot.
pdf(file="2020-03-1432_ESV_abundances.pdf",
 width=10*2,
 height=6*3,
 pointsize=12
 ) 
 par(mfrow=c(3,2))
 par(mar=c(0,0,2,0))
 pie(sm1$value, 
  labels = paste0(sm1$scientific_name, " ", sm1$percentage, "%"), 
  main="ESV abundances, summed over all samples",
  col=sm1$col,
  radius=0.8
  )
for (this_fish in 1:5)
 {
 fish <- guid$guid[this_fish]
 smf <- melt(r6[(r6$accept==1 & r6$guid==fish),c("scientific_name", "col", "ESVsize")], measure.vars="ESVsize")
 smf <- smf[order(-smf$value),]
 smf$percentage <- round(smf$value*100/sum(smf$value))
  pie(smf$value, 
   labels = paste0(smf$scientific_name, " ", smf$percentage, "%"), 
   main=paste("ESV abundances, blackfish", fish),
   col=smf$col,
   radius=rtdf$r[this_fish]
   )
 }
dev.off() 

write.csv(r6, "results_2020-03-26-1854.csv", row.names=FALSE)
save.image("2020-03-26-1854_workspace.RData")
```

### 1:00 pm early adopters’ data management meeting

  - Question: How can the metadata refer to a number of files? Answer:
    Under the distribution tab there will be a way to associated files /
    URLs, etc.

  - Hilmar commented on mdEditor files. Should not use the “product”
    resource type; use something more specific. Raw data means RAW data,
    e.g. data sheets, that have not been processed at all.

  - Collections. Question: I have thousands of files. Do I need to make
    a metadata entry for each file? Answer: No, there is a collection
    resource type for this kind of situation where there are many files
    in a standard format.

  - Data dictionaries. This tells the reader what each field is in your
    data file. Data dictionaries must be linked to resources.

  - Next steps are to complete three product medata records and to make
    a data dictionary.

I started writing an article on blackfish diet for the *AKES
Newsletter*.

## Friday, March 27

To do:

  - Blackfish analysis.
  - Publish blackfish sequence data.
  - Format Derek’s black spruce article.
  - Early adopters’ homework.

I worked on uploading images to the blackfish records on Arctos.

I uploaded pecimen data downloaded from Arctos and DNA extraction
methods, DNA sequencing methods, resulting raw metagenomic data, and
related files from RTL Genomics to Zenodo (Bowser and Bowser
[2020](#ref-bowser_raw_2020)).

I attended the biology check-in telecon at 11:00.

I wrote the introduction and methods sections for the blackfish diet
article.

### Saftey Committee Meeting at 14:00

  - Task for me for next week: We will need Job Hazard Analyses for
    herbicide applications and other activities under COVID-19 concerns.

## Saturday, March 28

I worked on editing Derek’s brown recluse article for the *AKES
Newsletter*, getting a formatted draft sent back to him.

## Sunday, March 29

I applied requested edits to Derek’s recluse article for the *AKES
Newsletter.*

I worked on preparing to upload blackfish diet records to Arctos.

``` r
## Today I want to prepare an Arctos upload for the diet observation records.
setwd("D:/projects/blackfish/2020_blackfish_diet/2020-03_SCVUC/ml-jg")

load("2020-03-26-1854_workspace.RData")

ad <- read.csv("ArctosData_02FB79E490.csv")

## Joining these and renaming so that I can recycle code from 2019-09-06.
names(ad)[2] <- "guid"
d1 <- merge(r6[r6$accept==1,], ad[c("guid", "COLLECTING_EVENT_ID")], all.x=TRUE)
names(d1)[1] <- "Other_Id_Num_1"
d1$Other_Id_Num_1 <- gsub("KNWRObs:Fish:", "", d1$Other_Id_Num_1)
names(d1)[11] <- "TAXON_NAME"
names(d1)[4] <- "n_reads"
names(d1)[3] <- "otu_id"
names(d1)[13] <- "Identification_Remarks"

## Recycling code from 2019-09-06...
## Sorting.
d1 <- d1[order(d1$Other_Id_Num_1, d1$TAXON_NAME, -d1$n_reads, d1$otu_id),]

d1$Part_Remark_1 <- paste(d1$otu_id, ": ", d1$n_reads, " reads", sep="")

d2 <- aggregate(d1$Part_Remark_1, by=list(d1$TAXON_NAME, d1$Other_Id_Num_1), FUN=paste, sep=" ")

names(d2) <- c("TAXON_NAME", "Other_Id_Num_1", "Part_Remark_1")

ag2 <- aggregate(d1$Identification_Remarks, by=list(d1$TAXON_NAME, d1$Other_Id_Num_1), FUN=paste0)
ag2$x <- as.character(ag2$x)
levels(as.factor(ag2$x))
ag2$x <- gsub("\"", "", ag2$x)
ag2$x <- gsub("c\\(", "", ag2$x)
ag2$x <- gsub("\\)", "", ag2$x)
ag2$x <- gsub(", ", "", ag2$x)
levels(as.factor(ag2$x))

d2$Identification_Remarks <- ag2$x

## Just checking...
sl <- !(d2$Identification_Remarks == "")
d2[sl,]  ## looked pretty good.  There were a couple of places where commas or paranthesis were dropped, but this is ok.

## Saving here for now.
save.image("2020-03-29-0733_workspace.RData")
```

## Monday, March 30

I worked on preparing to upload blackfish diet records to Arctos,
getting all the records from the blackfish dataset uploaded to Arctos
([UAMObs:Ento:244406](http://arctos.database.museum/guid/UAMObs:Ento:244406)–[UAMObs:Ento:244542](http://arctos.database.museum/guid/UAMObs:Ento:244542)).

``` r
## Continuing to prepare an Arctos upload.
setwd("D:/projects/blackfish/2020_blackfish_diet/2020-03_SCVUC/ml-jg")

load("2020-03-29-0733_workspace.RData")

ad <- read.csv("ArctosData_02FB79E490.csv")

## Joining these and renaming so that I can recycle code from 2019-09-06.
names(ad)[2] <- "guid"
d1 <- merge(r6[r6$accept==1,], ad[c("guid", "COLLECTING_EVENT_ID")], all.x=TRUE)
names(d1)[1] <- "Other_Id_Num_1"
d1$Other_Id_Num_1 <- gsub("KNWRObs:Fish:", "", d1$Other_Id_Num_1)
names(d1)[11] <- "TAXON_NAME"
names(d1)[4] <- "n_reads"
names(d1)[3] <- "otu_id"
names(d1)[13] <- "Identification_Remarks"

## Recycling code from 2019-09-06...
## Sorting.
d1 <- d1[order(d1$Other_Id_Num_1, d1$TAXON_NAME, -d1$n_reads, d1$otu_id),]

d1$Part_Remark_1 <- paste(d1$otu_id, ": ", d1$n_reads, " reads", sep="")

d2 <- aggregate(d1$Part_Remark_1, by=list(d1$TAXON_NAME, d1$Other_Id_Num_1), FUN=paste, sep=" ")

names(d2) <- c("TAXON_NAME", "Other_Id_Num_1", "Part_Remark_1")

d2$Part_Remark_1 <- as.character(d2$Part_Remark_1)
d2$Part_Remark_1 <- as.character(d2$Part_Remark_1)
d2$Part_Remark_1 <- gsub("\"", "", d2$Part_Remark_1)
d2$Part_Remark_1 <- gsub("c\\(", "", d2$Part_Remark_1)
d2$Part_Remark_1 <- gsub("\\)", "", d2$Part_Remark_1)
d2$Part_Remark_1 <- gsub(", ", "", d2$Part_Remark_1)


## Just checking...
sl <- !(d2$Identification_Remarks == "")
d2[sl,]  ## looked pretty good.  There were a couple of places where commas or paranthesis were dropped, but this is ok.

## Adding other field values.
d2$COLLECTION_OBJECT_ID <- 1:nrow(d2)
d2$ENTEREDBY <- "mbowser"
d2$Collector_Agent_1 <- "Matt Bowser"
d2$VERBATIM_DATE <- "August 23, 2019"
d2$BEGAN_DATE <- "2019-08-23"
d2$ENDED_DATE <- "2019-08-23"
d2$COLLECTING_EVENT_ID <- 11449049
d2$COLLECTING_SOURCE <- "wild caught"
d2$ACCN <- "UAM-2020.02-Kenai_blackfish-Ento"
d2$Verbatim_Locality <- "USA: Alaska, Kenai, pond at Candelight Drive and Kenai Spur Highway"
d2$Collector_Role_1 <- "collector"
d2$NATURE_OF_ID <- "molecular"
d2$ID_MADE_BY_AGENT <- "Matt Bowser"

## I remembered to stop before I get too far here and downloaded the current Alaska checklists from Arctos, linked to form http://www.akentsoc.org/the_fauna/checklist. This will enable me to check for new distribution records.

d2$MADE_DATE <- "2020-03-25"
d2$PART_NAME_1 <- "media"
d2$PART_CONDITION_1 <- "not applicable"
d2$PART_LOT_COUNT_1 <- 1
d2$PART_DISPOSITION_1 <- "not applicable"
d2$Other_Id_Num_Type_1 <- "KNWRObs:Fish"
d2$Other_Id_References_1 <- "eaten by"
d2$GUID_PREFIX <- "UAMObs:Ento"
d2$EVENT_ASSIGNED_BY_AGENT <- "Matt Bowser"
d2$EVENT_ASSIGNED_DATE <- "2020-03-30"
d2$SPECIMEN_EVENT_TYPE <- "collection"
d2$COLLECTING_METHOD <- "COI metabarcoding of fish gut contents"

## I need to reformat many of these as Arctos' "A string" format, e.g. "Peltigera {Peltigera conspersa nom. prov.}".
spt <- strsplit(d2$TAXON_NAME, " sp. ")
A <- sapply(spt, "[", 1)
string <- sapply(spt, "[", 2)
sl <- !is.na(string)
d2$TAXON_NAME[sl] <- paste(A[sl], " {", d2$TAXON_NAME[sl], "}", sep="")
levels(as.factor(d2$TAXON_NAME)) ## That looked ok.

## Ok, now trying a record.
write.csv(d2[1,], "2020-03-30-0741_blackfish_Arctos_upload_records_001-001.csv", row.names=FALSE)
## That worked.

## Trying some more.
write.csv(d2[2:10,], "2020-03-30-0949_blackfish_Arctos_upload_records_002-010.csv", row.names=FALSE) ## There were some problems with that one, which I notice before uploading it.

## Fixing.
d2$Part_Remark_1 <- gsub("readsbfdZotu", "reads, bfdZotu", d2$Part_Remark_1)
write.csv(d2[2:10,], "2020-03-30-0949_blackfish_Arctos_upload_records_002-010.csv", row.names=FALSE)
## One record will not load: ; TAXON_NAME (Chironomus bifurcatus) not found.
## This is a relatively newly described species.

## Added it to Arctos (http://arctos.database.museum/name/Chironomus%20bifurcatus) and I requested WUELKER et al. (2009) (https://doi.org/10.11646/zootaxa.2023.1.2) via ILL.

## Trying uploading that last file again.
## That worked.
## Preparing the next file...
write.csv(d2[11:50,], "2020-03-30-1044_blackfish_Arctos_upload_records_011-050.csv", row.names=FALSE)

err <- read.csv("2020-03-30-1049_bulkloader_errors.csv", stringsAsFactors=FALSE)
levels(as.factor(err$LOADED))
[1] ""                                                     
[2] "; TAXON_NAME (Cladopelma bicarinatum) not found"      
[3] "; TAXON_NAME (Hydra utahensis strain AK12b) not found"
[4] "; TAXON_NAME (Lumbriculis) not found"                 
[5] "; TAXON_NAME (Psectrocladius conjungens) not found"   

## One is an easy fix.
d2$TAXON_NAME[d2$TAXON_NAME=="Hydra utahensis strain AK12b"] <- "Hydra utahensis"

d2$TAXON_NAME[d2$TAXON_NAME=="Cladopelma bicarinatum"] <- "Cladopelma bicarinata"

d2$TAXON_NAME <- gsub("Lumbriculis", "Lumbriculus", d2$TAXON_NAME)

## Psectrocladius conjungens is not in Arctos, but a junior synonym, Chaetocladius conjugens, is.

d2$TAXON_NAME <- gsub("Psectrocladius conjungens", "Chaetocladius conjugens", d2$TAXON_NAME)

## Trying again now.
write.csv(d2[11:50,], "2020-03-30-1108_blackfish_Arctos_upload_records_011-050.csv", row.names=FALSE)

## That loaded. I changed the Hydra value in the bulkloader to "Hydra utahensis {Hydra utahensis strain AK12b}".

## Now for the next set.
write.csv(d2[51:100,], "2020-03-30-1113_blackfish_Arctos_upload_records_051-100.csv", row.names=FALSE)
## Now the bulkloader is busy, so that will have to wait.

## Saving here.
save.image("2020-03-30-1116_workspace.RData")

## I can work on graphics/summaries while I wait.

## Summarizing by phyla.
sl <- r7$accept==1
agp <- aggregate(r7$ESVsize[sl], by=list(r7$Phylum[sl]), sum)
agp
     Group.1     x
1   Annelida   384
2 Arthropoda 62166
3   Cnidaria    31
4   Mollusca   160
5   Rotifera   431

names(agp) <- c("phylum", "n_reads")
agp$percent <- agp$n_reads*100/sum(agp$n_reads)
agp <- agp[order(-agp$n_reads),]
agp
      phylum n_reads     percent
2 Arthropoda   62166 98.40752232
5   Rotifera     431  0.68226429
1   Annelida     384  0.60786424
4   Mollusca     160  0.25327677
3   Cnidaria      31  0.04907237

## Wow, 98% arthropods. Now I want to break it into groups at various levels that seem to make sense.

r7$group <- r7$Phylum
sla <- sl & (r7$Phylum=="Arthropoda")
## Summarizing arthrods.
aga <- aga <- aggregate(r7$ESVsize[sla], by=list(r7$Class[sla]), sum)
aga
       Group.1     x
1 Branchiopoda 12934
2  Hexanauplia  4378
3      Insecta 34959
4    Ostracoda  9895

r7$group[(r7$Phylum=="Arthropoda") & (r7$Class=="Ostracoda")] <- "Ostracoda"

## the bulkloader was available, so I got that last set loaded. They loaded just fine.
## Now the rest.
write.csv(d2[101:137,], "2020-03-30-1135_blackfish_Arctos_upload_records_101-137.csv", row.names=FALSE)
## That loaded.

## Back to working toward summaries.
r7$group[(r7$Phylum=="Arthropoda") & (r7$Class=="Insecta")] <- "Insecta"

## Looking at branchiopods.
sl <- r7$Class=="Branchiopoda"
agb <- aggregate(r7$ESVsize[sl], by=list(r7$Order[sl]), sum)
agb
      Group.1     x
1 Diplostraca 13130

## Looking at these. 
sl <- r7$Order=="Diplostraca"
agd <- aggregate(r7$ESVsize[sl], by=list(r7$Family[sl]), sum)
agd
       Group.1     x
1                196
2   Chydoridae   491
3   Daphniidae   118
4 Eurycercidae 12325

## Eurycercidae is a Cladoceran. So is Chydoridae. So is Daphniidae. Calling these all Cladocera.
r7$group[sl] <- "Cladocera"

sl <- r7$Class=="Hexanauplia"
agh <- aggregate(r7$ESVsize[sl], by=list(r7$Order[sl]), sum)
agh
     Group.1    x
1               3
2 Cyclopoida 7099
## We will call these copepods.
r7$group[sl] <- "Copepoda"

sl <- r7$Class=="Insecta"
agh <- aggregate(r7$ESVsize[sl], by=list(r7$Order[sl]), sum)
agh
      Group.1     x
1  Coleoptera    20
2     Diptera 15065
3 Hymenoptera    28
4 Lepidoptera    57
5     Odonata 19967
6 Trichoptera    22

r7$group[r7$Order=="Diptera"] <- "Diptera"
r7$group[r7$Order=="Odonata"] <- "Odonata"
r7$group[r7$group=="Insecta"] <- "orther insects"

## Now looking at the summary.

## Summarizing by phyla.
sl <- r7$accept==1
agg <- aggregate(r7$ESVsize[sl], by=list(r7$group[sl]), sum)
agg

names(agg) <- c("group", "n_reads")
agg$percent <- agg$n_reads*100/sum(agg$n_reads)
agg <- agg[order(-agg$n_reads),]
agg 
            group n_reads     percent
7         Odonata   19967 31.60735769
5         Diptera   14970 23.69720762
2       Cladocera   12934 20.47426075
9       Ostracoda    9895 15.66358513
4        Copepoda    4378  6.93028557
10       Rotifera     431  0.68226429
1        Annelida     384  0.60786424
6        Mollusca     160  0.25327677
3        Cnidaria      31  0.04907237
8  orther insects      22  0.03482556

## now consolidating the rare stuff.
ag2 <- agg[1:6,]
ag2$group[6] <- "other"
ag2$n_reads[6] <- sum(agg$n_reads[6:10])
ag2
       group n_reads    percent
7    Odonata   19967 31.6073577
5    Diptera   14970 23.6972076
2  Cladocera   12934 20.4742607
9  Ostracoda    9895 15.6635851
4   Copepoda    4378  6.9302856
10     other    1028  0.6822643

## A little accounting.
sum(ag2$n_reads)
[1] 63172
sum(r7$ESVsize[r7$accept==1])
[1] 63172
## Good.

## Adding colors.
set.seed(42)
ag2$col <- sample(x=cdf$hex, size=nrow(ag2), replace = FALSE)

pdf(file="2020-03-30-1308_ESV_abundances_by_group.pdf",
 width=5.5,
 height=4
 )
 par(mar=c(0,0,1,0))
 pie(ag2$n_reads, 
  labels = paste0(ag2$group, " ", round(ag2$percent), "%"), 
  main="ESV abundances by group",
  col=ag2$col,
  radius=0.8
  )
dev.off()

pdf(file="2020-03-30-1320_ESV_abundances_by_group.pdf",
 width=4.1,
 height=2.6
 )
 par(mar=c(0,0,1,0))
 pie(ag2$n_reads, 
  labels = paste0(ag2$group, " ", round(ag2$percent), "%"), 
  main="ESV abundances by group",
  col=ag2$col,
  radius=0.8
  )
dev.off()
 
## Checking to see what might have been added to the checklist.
lb <- read.csv("2020-03-30-0700_Arctos_AK_checklist_species_only.csv", stringsAsFactors=FALSE)
la <- read.csv("2020-03-30-1335_Arctos_AK_checklist_species_only.csv", stringsAsFactors=FALSE)
lb <- lb$SCIENTIFIC_NAME
la <- la$SCIENTIFIC_NAME

adds <- setdiff(la, lb)
adds
 [1] "Eurycercus longirostris" 
 [2] "Pleuroxus aduncus"       
 [3] "Simocephalus exspinosus" 
 [4] "Chaetocladius conjugens" 
 [5] "Chironomus bifurcatus"   
 [6] "Cladopelma bicarinata"   
 [7] "Cricotopus trifasciatus" 
 [8] "Dicrotendipes tritomus"  
 [9] "Orthocladius smolandicus"
[10] "Procladius nigriventris" 
[11] "Psectrocladius obvius"   
[12] "Angarotipula illustris"  
[13] "Candona candida"         
[14] "Cypridopsis vidua"    

## Saving this.
write.csv(adds, "2020-03-30-1344_additions.csv", row.names=FALSE)

## Saving here.
save.image("2020-03-30-1345_workspace.RData")
```

![Pie chart of read numbers of major groups eaten by
blackfish.](2020-03-30_pie_chart.png)  
Pie chart of read numbers of major groups eaten by blackfish.

I worked on reviewing these potential new records.

## Tuesday, March 31

To do:

  - ~~Create reciprocal relationship links on Arctos from blackfish
    records to food items.~~
  - ~~Finish reviewing the potential new distribution records.~~

I did finish reviewing potential new distribution records from blackfish
gut contents.

### Bio staff telecon at 11:00

  - Mark and and I need to work on Job Hazard Analyses.

<!-- end list -->

``` r
## Continuing to prepare an Arctos upload.
setwd("D:/projects/blackfish/2020_blackfish_diet/2020-03_SCVUC/ml-jg")

ad <- read.csv("2020-03-31-1137_Arctos_download.csv", stringsAsFactors=FALSE)

ad$guid_prefix <- "KNWRObs:Fish"

ad$existing_other_id_type <- "catalog number"

ad$existing_other_id_number <- gsub("\\(eaten by\\) KNWRObs:Fish http://arctos.database.museum/guid/KNWRObs:Fish:", "", ad$RELATEDCATALOGEDITEMS)

ad$new_other_id_number <- gsub("UAMObs:Ento:", "", ad$GUID)

ad$new_other_id_type <- "UAMObs:Ento"

ad$new_other_id_references <- "ate"

rl <- ad[,14:19]

## Testing one record.
write.csv(rl[1,], "2020-03-31-1405_relationsips_upload_001-001.csv", row.names=FALSE)

## That looked good. Going for the rest.
write.csv(rl[2:137,], "2020-03-31-1405_relationsips_upload_002-137.csv", row.names=FALSE)

## Also need catalog numbers for adding media relationships.
rl$space <- " "
write.csv(rl[,c("new_other_id_number", "space")], "2020-03-31-1432_catnums.csv", row.names=FALSE)

Uploaded the FASTA file to Arctos (http://dx.doi.org/10.7299/X7V1253J).

## Now I want to add colors to the phylogenetic tree.
load("2020-03-30-1345_workspace.RData")

ag2
```

I edited the tree on iTOL, making it prettier, uploaded this to Arctos,
and associated it with the occurrence records
(<http://arctos.database.museum/media/10608330>).

# April

## Wednesday, April 1

To do:

  - ~~Edit, format, and submit today’s *Refuge Notebook* article.~~
  - *Refuge Notebook* catch-up.
  - ~~Format USFS *AKES Newsletter* article.~~
  - JHAs
  - Finish blackfish article.

I finished editing of this week’s *Refuge Notebook* article, getting it
sent to the *Clarion*.

I worked on editing the USFS article for the *AKES Newsletter*, getting
an initial draft finished.

I composed a recommendation letter that was asked of me.

I have the following JHAs to complete.

  - ~~snowmachines~~
  - pesticide application
  - field data collection

I completed a JHA for snowmachine operations.

## Thursday, April 2

To do:

  - *Refuge Notebook* catch-up.
  - JHAs
  - Finish blackfish article.
  - Images/article review to John.

I have the following JHAs to complete.

  - pesticide application
  - field data collection

I worked on the blackfish diet article.

### 12:30 safety committee meeting

  - Need to review and comment on completed JHAs.

<!-- end list -->

``` r
## Continuing to prepare an Arctos upload.
setwd("D:/projects/blackfish/2020_blackfish_diet/2020-03_SCVUC/ml-jg")

load("2020-03-30-1345_workspace.RData")

## Trying to make a bunch of separate plots for read abundances by ESV.
## Putting everything on one plot.
pdf(file="2020-04-02-1618_ESV_abundances.pdf",
 width=8.2,
 height=4,
 pointsize=8
 ) 
 par(mar=c(0,0,1,0))
 pie(sm1$value, 
  labels = paste0(sm1$scientific_name, " ", sm1$percentage, "%"), 
  main="ESV abundances, summed over all samples",
  col=sm1$col,
  radius=0.8
  )
dev.off()

library(reshape2)
  
for (this_fish in 1:5)
 {
 file_name <- paste0("2020-04-02-1618_", this_fish, ".pdf")
 pdf(file=file_name,
  width=8.2, 
  height=2,
  pointsize=8
  )
 par(mar=c(0,0,1,0))
 fish <- guid$guid[this_fish]
 smf <- melt(r6[(r6$accept==1 & r6$guid==fish),c("scientific_name", "col", "ESVsize")], measure.vars="ESVsize")
 smf <- smf[order(-smf$value),]
 smf$percentage <- round(smf$value*100/sum(smf$value))
  pie(smf$value, 
   labels = paste0(smf$scientific_name, " ", smf$percentage, "%"), 
   main=paste("ESV abundances, blackfish", fish),
   col=smf$col,
   radius=rtdf$r[this_fish]
   )
 dev.off()   
 }
## Those were not great.

## Putting everything on one plot.
pdf(file="2020-04-02-1637_ESV_abundances.pdf",
 width=7.9,
 height=5.8,
 pointsize=8
 ) 
 par(mfrow=c(3,2))
 par(mar=c(0,0,2,0))
 pie(sm1$value, 
  labels = paste0(sm1$scientific_name, " ", sm1$percentage, "%"), 
  main="ESV abundances, summed over all samples",
  col=sm1$col,
  radius=0.8
  )
for (this_fish in 1:5)
 {
 fish <- guid$guid[this_fish]
 smf <- melt(r6[(r6$accept==1 & r6$guid==fish),c("scientific_name", "col", "ESVsize")], measure.vars="ESVsize")
 smf <- smf[order(-smf$value),]
 smf$percentage <- round(smf$value*100/sum(smf$value))
  pie(smf$value, 
   labels = paste0(smf$scientific_name, " ", smf$percentage, "%"), 
   main=paste("ESV abundances, blackfish", fish),
   col=smf$col,
   radius=rtdf$r[this_fish]
   )
 }
dev.off() 
```

## Friday, April 3

To do:

  - *Refuge Notebook* catch-up.
  - JHAs
  - Finish blackfish article.
  - Get herbicie quote.

I have the following JHAs to complete.

  - pesticide application
  - field data collection

<!-- end list -->

``` r
## Continuing to prepare an Arctos upload.
setwd("D:/projects/blackfish/2020_blackfish_diet/2020-03_SCVUC/ml-jg")

load("2020-03-30-1345_workspace.RData")

library(reshape2)

## Putting everything on one plot.
pdf(file="2020-04-03-0649_ESV_abundances.pdf",
 width=8.2,
 height=5.8,
 pointsize=8
 ) 
 par(mfrow=c(3,2))
 par(mar=c(0,0,2,0))
 pie(sm1$value, 
  labels = paste0(sm1$scientific_name, " ", sm1$percentage, "%"), 
  main="ESV abundances, summed over all samples",
  col=sm1$col,
  radius=0.8
  )
for (this_fish in 1:5)
 {
 fish <- guid$guid[this_fish]
 smf <- melt(r6[(r6$accept==1 & r6$guid==fish),c("scientific_name", "col", "ESVsize")], measure.vars="ESVsize")
 smf <- smf[order(-smf$value),]
 smf$percentage <- round(smf$value*100/sum(smf$value))
  pie(smf$value, 
   labels = paste0(smf$scientific_name, " ", smf$percentage, "%"), 
   main=paste("ESV abundances, blackfish", fish),
   col=smf$col,
   radius=rtdf$r[this_fish]
   )
 }
dev.off() 

## How many unique identifications?
length(levels(as.factor(r7$scientific_name[r7$accept==1])))
[1] 103

## How many occurrences?
sum(r7$accept)
[1] 167

## How many ESVs?
length(levels(as.factor(r7$ESV_name[r7$accept==1])))
[1] 131

## Considering a different kind of plot.
## How many identifications represent >= 1%?
sum(sm1$percentage >= 1)
[1] 20
sm1[sm1$percentage >= 1,]
                           scientific_name     col variable value percentage
69                           Aeshna juncea #FFD300  ESVsize 12261         19
99                 Eurycercus longirostris #00BFFF  ESVsize 11956         19
121                         Aeshna eremita #E03C31  ESVsize  7069         11
35            Ceratopogonidae sp. bfdZotu7 #1034A6  ESVsize  4808          8
159            Procladius sp. BOLD:AAP3007 #006A4E  ESVsize  4709          7
151                Cyprididae sp. bfdZotu3 #AF6E4D  ESVsize  4531          7
106               Podocopida sp. bfdZotu12 #996666  ESVsize  1613          3
132          Macrocyclops sp. BOLD:AAG9780 #B2FFFF  ESVsize  1576          2
43           Macrocyclops sp. BOLD:AAG9778 #FFBCD9  ESVsize  1554          2
30             Chironomus sp. BOLD:ACR4689 #1B4D3E  ESVsize  1249          2
23                 Cyprididae sp. bfdZotu3 #2F4F4F  ESVsize  1056          2
57                 Cyprididae sp. bfdZotu3 #568203  ESVsize   959          2
162          Macrocyclops sp. BOLD:AAG9778 #246BCE  ESVsize   955          2
134                 Cladopelma bicarinatum #8F9779  ESVsize   677          1
28  Procladius sp. cf. fuscus BOLD:AAQ0606 #FE6F5E  ESVsize   637          1
136                        Candona candida #77B5FE  ESVsize   473          1
115                      Dasyhelea modesta #DE6FA1  ESVsize   471          1
14                  Cladopelma bicarinatum #15F4EE  ESVsize   467          1
131               Chydoridae sp. bfdZotu22 #967117  ESVsize   402          1
59                    Leucorrhinia proxima #B87333  ESVsize   380          1

## Recalculating without rounding.
sm1$percentage <- sm1$value*100/sum(sm1$value)

sum(sm1$percentage >= 1)
[1] 15
sm1[sm1$percentage >= 1,]
                           scientific_name     col variable value percentage
69                           Aeshna juncea #FFD300  ESVsize 12261  19.408915
99                 Eurycercus longirostris #00BFFF  ESVsize 11956  18.926107
121                         Aeshna eremita #E03C31  ESVsize  7069  11.190084
35            Ceratopogonidae sp. bfdZotu7 #1034A6  ESVsize  4808   7.610967
159            Procladius sp. BOLD:AAP3007 #006A4E  ESVsize  4709   7.454252
151                Cyprididae sp. bfdZotu3 #AF6E4D  ESVsize  4531   7.172481
106               Podocopida sp. bfdZotu12 #996666  ESVsize  1613   2.553346
132          Macrocyclops sp. BOLD:AAG9780 #B2FFFF  ESVsize  1576   2.494776
43           Macrocyclops sp. BOLD:AAG9778 #FFBCD9  ESVsize  1554   2.459951
30             Chironomus sp. BOLD:ACR4689 #1B4D3E  ESVsize  1249   1.977142
23                 Cyprididae sp. bfdZotu3 #2F4F4F  ESVsize  1056   1.671627
57                 Cyprididae sp. bfdZotu3 #568203  ESVsize   959   1.518078
162          Macrocyclops sp. BOLD:AAG9778 #246BCE  ESVsize   955   1.511746
134                 Cladopelma bicarinatum #8F9779  ESVsize   677   1.071677
28  Procladius sp. cf. fuscus BOLD:AAQ0606 #FE6F5E  ESVsize   637   1.008358

abspec <- unique(sm1$scientific_name[sm1$percentage >= 1])
abspec ## Whoa, there are only 12 of those. Remaking summary stuff.

agr <- aggregate(r7$ESVsize[r7$accept==1], by=list(r7$scientific_name[r7$accept==1]), sum)
dim(agr)
[1] 103   2
names(agr) <- c("scientific_name", "n_reads")
agr$percentage <- agr$n_reads*100/sum(agr$n_reads)
abspec <- unique(agr$scientific_name[agr$percentage >= 1])
abspec
agr <- agr[agr$percentage >= 1,]
agr <- agr[order(-agr$percentage),]
agr
                          scientific_name n_reads percentage
3                           Aeshna juncea   12261  19.408915
41                Eurycercus longirostris   12248  19.388337
2                          Aeshna eremita    7069  11.190084
35                Cyprididae sp. bfdZotu3    6839  10.825999
17           Ceratopogonidae sp. bfdZotu7    4808   7.610967
80            Procladius sp. BOLD:AAP3007    4718   7.468499
62          Macrocyclops sp. BOLD:AAG9778    2624   4.153739
76               Podocopida sp. bfdZotu12    1692   2.678402
63          Macrocyclops sp. BOLD:AAG9780    1691   2.676819
22            Chironomus sp. BOLD:ACR4689    1249   1.977142
27                 Cladopelma bicarinatum    1144   1.810929
11                        Candona candida     750   1.187235
83 Procladius sp. cf. fuscus BOLD:AAQ0606     637   1.008358

## Now for a data frame for graphing.
sl <- (r7$accept == 1) & (r7$scientific_name %in% abspec)

ag3 <- aggregate(r7$ESVsize[sl], by=list(r7$scientific_name[sl], r7$guid[sl]), sum)
dim(ag3)
[1] 27  3

names(ag3) <- c("scientific_name", "guid", "n_reads")
ag3$x <- as.numeric(substr(ag3$guid, 14, 15))
ag3$r <- (ag3$n_reads/300*pi)^0.5

set.seed(43) 
ndf <- as.data.frame(abspec)
ndf$col <- sample(x=cdf$hex, size=nrow(ndf), replace = FALSE)

## Now trying a primitive plot
pdf(file="2020-04-03-1006_abundance_plot.pdf",
 width=7,
 height=7
 )
par(mar=c(0.5,0.5,8.5,17))
plot(x=c(12,16),
 y=c(1,13),
 xlim=c(11.7,16.3),
 ylim=c(0.9,13.8),
 yaxt="n",
 type="n",
 ylab="",
 xlab="",
 xaxt="n"
 )
for (this_thing in 1:13)
 {
 sn <- agr$scientific_name[this_thing]
 tdf <- ag3[ag3$scientific_name==sn,]
 print(tdf)
 points(tdf$x,
  rep(14-this_thing, nrow(tdf)),
  cex=tdf$r,
  pch=21,
  col="#00000060",
  bg=paste0(ndf$col[this_thing], "80"),
  lwd=1.5
  )
 text(tdf$x,
  rep((14-this_thing)+0.27, nrow(tdf)),
  labels=tdf$n_reads
  )
 }
axis(side=4,
 at=13:1,
 labels=abspec,
 las=2
 )
axis(side=3,
 at=12:16,
 labels=unique(ag3$guid),
 las=2
 )
dev.off()
```

![Read abundances of identified food items from each of five blackfish
specimens.](2020-04-03_abundance_plot.png)  
Read abundances of identified food items from each of five blackfish
specimens. Only food items that represented \>= 1% of the total number
of reads were included. The area of each circle is proportional to read
abundances.

### Biology check-in at 11:00

### USFWS R7 Town Hall telecon at 11:30

I worked on the blackfish article.

## Saturday, April 4

The kids and I skied, snowshoed, or kicksledded down to a seep by the
Kasilof River at 60.365425N, 151.285797W where we collected a worm, some
fly larvae, and one caddisfly larva.

![Collecting invertebrates from a seep near the Kasiof
River.](2020-04-04_seep.jpg)  
Collecting invertebrates from a seep near the Kasiof River.

## Monday, April 6

To do:

  - *Refuge Notebook* catch-up.
  - JHAs
  - Finish blackfish article.
  - pesticide applicator training.
  - ~~Write Lifescanner team and RTL Genomics to find out whether or not
    they are accepting specimens.~~
  - Data management early adopters’ homework.
  - ~~Upload iNaturalist/Lifescanner records.~~

I have the following JHAs to complete.

  - pesticide application
  - field data collection

I uploaded records to inaturalist records for a sciaroid fly
([iNaturalist:41528440](https://www.inaturalist.org/observations/41528440)),
a segmented worm
([iNaturalist:41529724](https://www.inaturalist.org/observations/41529724)),
a fly larva
([iNaturalist:41531388](https://www.inaturalist.org/observations/41531388)),
a chironomid midge
([iNaturalist:41532738](https://www.inaturalist.org/observations/41532738)),
and a caddisfly
([iNaturalist:41534382](https://www.inaturalist.org/observations/41534382))
and I made corresponding Lifescanner records.

I worked on the blackfish diet article.

### Biology staff meeting at 11:00

  - Going to pick up microscope, etc. from lab.
  - Need to obtain pesticide applicator CEUs.

## Tuesday, April 7

To do:

  - *Refuge Notebook* catch-up.
  - JHAs
  - ~~Finish blackfish article draft.~~
  - pesticide applicator training.
  - ~~Data management early adopters’ homework.~~
  - ~~Pick up microscope, etc. from the office.~~

I have the following JHAs to complete.

  - pesticide application
  - field data collection

I finished a draft of the blackfish diet article.

At our last early adopter data managmeent meeting, I had recorded that
next steps are to complete three product medata records and to make a
data dictionary. I made a data dictionary im mdEditor and added metadata
records for all data/products that I have in the directory right now.

### Vogel Lake/Miller Creek Field Work meeting at 13:00

  - No field work can be done until May 1 at the earliest.
  - I learned that a plume of low salinity freshwater from the Susitna
    River comes straight across to Point Possession, making a viable
    possibility for pike to get to the northern Kenai across the salt
    water.

In the evening I drove to the office, where I picked up a microscope and
other equipment so that I can process Berlese soil samples at home.

## Wednesday, April 8

To do:

  - ~~Edit and submit this week’s *Refuge Notebook*.~~
  - *Refuge Notebook* catch-up.
  - JHAs
  - pesticide applicator training.
  - ~~Summarize snowshoe hare data?~~

I have the following JHAs to complete.

  - pesticide application
  - field data collection

I did a little editing of my blackfish article.

I summarized snowshoe hare data.

``` r
################################################################################
# Summarize snowshoe hare pellet count data.                                   #
#                                                                              #
# Author: First and last name <matt_bowser@fws.gov>                            #
# Date created: 2020-04-08                                                     #
# Date last edited: 2020-04-08                                                 #
################################################################################


# Setup ------------------------------------------------------------------------

# Load packages
library(reshape2)

# Source external scripts
#source("directory_path/script_foo.R")


# 1 Load data -----------------------------------------------------------------
## ---- load_data  

pellet_data <- read.csv("../data/final_data/observations/2020-03-18_snowshoe_hare_pellet_counts.csv",
 stringsAsFactors=FALSE
 )
 
plot_data <- read.csv("../data/final_data/geodata/2020-03-18_snowshoe_hare_plot_data.csv",
 stringsAsFactors=FALSE
 )


# 2 Prepare data ---------------------------------------------------------------
## ---- prep_data  

## Join the data to get the grid_name values.
pellet_data_joined <- merge(
 plot_data[,c("plot_name", "grid_name")],
 pellet_data
 )

## Convert the plot names to plot numbers.
plot_number <- strsplit(pellet_data_joined$plot_name,
 "-"
 )
plot_number <- sapply(plot_number,
 "[",
 2
 ) 
pellet_data_joined$plot_name <- as.numeric(plot_number)
 
## Reshape the data into a grid_name X plot number X year array.
pellet_data_molten <- melt(pellet_data_joined,
 measure.vars="pellet_count"
 )

pellet_data_array <- acast(
 pellet_data_molten,
 grid_name ~ plot_name ~ year,
 fun.aggregate=sum,
 fill=9999 ## Temporary fill value.
 )
pellet_data_array[pellet_data_array==9999] <- NA
 
n_grids <- dim(pellet_data_array)[1]
n_plots <- dim(pellet_data_array)[2]
n_years <- dim(pellet_data_array)[3]

## Set to NA any plot for which the previous year's count was NA.
for (this_grid in 1:n_grids)
 {
 for (this_plot in 1:n_plots)
  {
  for (this_year in n_years:2)
   if(is.na(pellet_data_array[this_grid,this_plot,this_year-1]))
    {
    pellet_data_array[this_grid,this_plot,this_year] <- NA
    }
  }
 }
 
## Also set all year 1 observations to NA.
pellet_data_array[,,1] <- NA

## Now get averages.
pellet_count_means <- as.data.frame(apply(pellet_data_array,
 c(1,3),
 mean,
 na.rm=TRUE
 ))

## Plot.
plot_title <- paste0("Kenai NWR snowshoe hare pellet counts, ",
 names(pellet_count_means)[2],
 "–",
 names(pellet_count_means)[ncol(pellet_count_means)]
 )

years <- as.numeric(names(pellet_count_means)[2]):as.numeric(names(pellet_count_means)[ncol(pellet_count_means)]) 

## Colors.
grid_colors <- c("#EE34D2",
 "#B33B24",
 "#FFDB00",
 "#5E8C31",
 "#9C51B6",
 "#5DADEC",
 "#353839"
 )

file_name <- paste0("../documents/",
 as.Date(Sys.time()),
 "_pellet_count_means_over_time.pdf"
 )

pdf(file=file_name,
 width=7,
 height=6
 )
plot(x=c(min(years),
 max(years)),
 y=c(0,max(pellet_count_means, na.rm=TRUE)),
 type="n",
 main=plot_title,
 xlab="year",
 ylab="mean pellets/m2"
 )
for (this_grid in 1:n_grids)
 {
 lines(years,
  pellet_count_means[this_grid,2:n_years],
  col=grid_colors[this_grid],
  lwd=2
  )
 }
legend("topright",
 legend = rownames(pellet_count_means),
 col = grid_colors,
 lty=1,
 lwd=2
 )
dev.off()

## Save means summaries.
file_name <- paste0("../documents/",
 as.Date(Sys.time()),
 "_pellet_count_means_over_time.csv"
 )
write.csv(pellet_count_means,
 file_name
 ) 
```

![Kenai National Wildlife Refuge snoeshoe hare pellet counts, means for
each hare grid and year.](2020-04-08_pellet_count_means_over_time.png)  
Kenai National Wildlife Refuge snoeshoe hare pellet counts, means for
each hare grid and year.

I started editing this week’s *Refuge Notebook* article.

### Biology staff check-in at 11:00

I finished editing this week’s *Refuge Notebook* article and got it sent
to the *Clarion*.

### Safety meeting at 14:30

I worked a little on the Kenai NWR checklist.

## Thursday, April 9

To do:

  - *Refuge Notebook* catch-up.
  - JHAs
  - pesticide applicator training.

I have the following JHAs to complete.

  - pesticide application
  - field data collection

I worked on formatting Liz’s article for the AKES Newsletter.

I helped make some face masks.

I worked some on the Kenai National Wildlife Refuge checklist.

### Early adopters’ data management meeting at 13:00

  - No new homework this time?
  - Could add spatial extent.
  - Next big step will be to actually archive the dataset.

In the afternoon, Apphia and I walked down to Coal Creek where it
crosses under Kalifornsky Beach Road. Where there was a little open
water we collected two caddisflies (BOLD-UB0 and BOLD-7L9) and one worm
(BOLD-OC2).

![Collecting invertebrates at Coal Creek.](2020-04-09_Coal_Creek.jpg)  
Collecting invertebrates at Coal Creek.

## Friday, April 10

To do:

  - *Refuge Notebook* catch-up.
  - JHAs
  - pesticide applicator training.
  - mid-year performance plans.
  - credit card reconciliation.
  - chainsaw refresher.

I entered data from specimens collected yesterday into lifescanner.net.

I sorted Berlese sample
[KNWR:Ento:11385](http://arctos.database.museum/guid/KNWR:Ento:11385).

Contents:

| identification | count |
| :------------- | ----: |
| Araneae        |     1 |
| Arachnida      |   384 |
| Collembola     |    84 |
| Hemiptera      |     3 |
| Coleoptera     |     1 |
| Insecta        |     2 |
| Thysanoptera   |     3 |

### Refuge staff meeting at 11:00

## Saturday, April 11

I sorted a Berlese sample from site SK22, soil sample 2. I put this into
a vial with barcode UAM100185961.

Contents:

| identification | count |
| :------------- | ----: |
| Araneae        |    32 |
| Collembola     |   108 |
| Arachnida      |   226 |
| Insecta        |     2 |
| Hemiptera      |     4 |

One of the Collembola looked like *Megalothorax*.

## Monday, April 13

I sorted a Berlese sample from site 3092, soil sample 2. I put this into
a vial with barcode UAM100185962. Arctos is down for maintenance, so I
do not have a GUID or URL for this record yet.

Contents:

| identification | count |
| :------------- | ----: |
| Coleoptera     |     2 |
| Geophilomorpha |     1 |
| Arachnida      |   354 |
| Collembola     |   172 |
| Thysanoptera   |     8 |
| Insecta        |     6 |
| Araneae        |     4 |
| Annelida       |     1 |

### Biology check-in meeting at 11:00

  - There is a lot of trail use in the Skilak area and Skyline, even
    though trails are closed.
  - Next bio meeting is Thursday at 11:00.

### Alaska Native Plant Material Meeting at 13:00

## Tuesday, April 14

I sorted a Berlese soil sample from site 3069, soil sample 1. Vial
barcode: UAM100185963.

Contents:

| identification | count |
| :------------- | ----: |
| Arachnida      |   255 |
| Araneae        |     2 |
| Collembola     |    13 |
| Thysanoptera   |     5 |
| Hymenoptera    |     1 |
| Hemiptera      |     2 |

As has been typical in these samples, there were really more springtails
and mites that were added in clumps at the end.

## Wednesday, April 15

I edited this week’s *Refuge Notebook* article and got it sent to the
*Clarion*.

I worked on formatting Derek’s *Scathophaga* misidentification article
and Adam’s hare article for the *AKES Newsletter*.

## Thursday, April 16

I finished some editing for the upcomeing *AKES Newsletter*.

I formatted the *Refuge Notebook* article form December 27, 2019 and
started with 2020 articles.

### Biology check-in at 11:00

I attempted to attend a 13:30 meeting about the Mystery Creek gravel pit
rehabilitation, but the connection was too poor.

### AKEPIC meeting at 14:00

I formatted and posted *Refuge Notebook* articles through January 17.

## Friday, April 17

I sorted the Berlese sample from site MG12, soil sample 1 (barcode:
UAM100185964).

Contents:

| identification | count |
| :------------- | ----: |
| Lumbricidae    |     1 |
| Araneae        |     4 |
| Arachnida      |   300 |
| Insecta        |     2 |
| Collembola     |    42 |
| Hemiptera      |     5 |
| Coleoptera     |     4 |
| Annelida       |     1 |
| Hymenoptera    |     1 |

I worked on formatting *Refuge Notebook* articles. I also did some
editing of my blackfish article.

## Monday, April 20

To do:

  - ~~Finish sorting Berlese samples.~~
  - ~~Get other Berlese samples and LifeScanner samples from work.~~
  - Compile data from sweep and Berlese samples.
  - Slikok occupancy?
  - Edit this week’s *Refuge Notebook* article.
  - Proposals for 2019 to Mark?
  - Contract for fluridone.
  - Review Rob’s pike treatment plan.

This mornig I sorted a Berlese sample from site MG39, soil sample 1,
barcode UAM100185965.

| identification | count |
| :------------- | ----: |
| Thysanoptera   |     6 |
| Araneae        |     6 |
| Coleoptera     |     4 |
| Arachnida      |   341 |
| Collembola     |    23 |
| Insecta        |     2 |

Arctos is still down, so I still cannot enter data for these samples. I
assigned catalog numbers manually.

| barcode      | GUID                                                                  |
| :----------- | :-------------------------------------------------------------------- |
| UAM100185961 | [KNWR:Ento:11386](http://arctos.database.museum/guid/KNWR:Ento:11386) |
| UAM100185962 | [KNWR:Ento:11387](http://arctos.database.museum/guid/KNWR:Ento:11387) |
| UAM100185963 | [KNWR:Ento:11388](http://arctos.database.museum/guid/KNWR:Ento:11388) |
| UAM100185964 | [KNWR:Ento:11389](http://arctos.database.museum/guid/KNWR:Ento:11389) |
| UAM100185965 | [KNWR:Ento:11390](http://arctos.database.museum/guid/KNWR:Ento:11390) |

I worked on preparing documents for a contract for acquiring pelleted
fluridone.

I started to edits this week’s Refuge Notebook article.

I did some reading, evaluating whether or not we should start doing
metagenomic work in house with a minION sequencer.

In the evening I drove to the Refuge, where I picked up the rest of the
Berlese samples to ship out to RTL Genomics.

While there, I also attempted to collect some *Badonnelia* from the
floor drains. I did collect several minute arthropods, at least one of
which was a pale psocid. There were a number of springtails, also.

## Tuesday, April 21

To do:

  - ~~Prepare samples to ship out tomorrow morning.~~
  - Compile data from sweep and Berlese samples.
  - Slikok occupancy?
  - ~~Edit this week’s *Refuge Notebook* article.~~
  - ~~Proposals for 2019 to Mark?~~
  - Review Rob’s pike treatment plan.
  - ~~Review FIA stuff ahead of this afternoon’s meeting.~~

I did get a proposal to Mark for building capacity to do some DNA work
in house.

### Safety committee meeting at 09:00

  - We need to be checking in on the radio when we go out in the field.

I started entering tally data from the 2019 black spruce sweep net
samples.

I finished editing this week’s *Refuge Notebook* article.

I posted the [January 24 *Refuge Notebook*
article](https://www.fws.gov/uploadedFiles/Region_7/NWRS/Zone_2/Kenai/Sections/What_We_Do/In_The_Community/Refuge_Notebooks/2020_Articles/Refuge_Notebook_v22_n04.pdf).

I inventoried and packaged the Berlese samples from the 2019 black
spruce work.

I attempted multiple times to attend the Biologists’ teleconference this
afternoon, but I failed to get through.

## Wednesday, April 22

To do:

  - ~~Ship out Berlese samples.~~
  - ~~Compile data from sweep and Berlese samples.~~
  - Slikok occupancy?
  - ~~Submit this week’s *Refuge Notebook* article.~~
  - Review Rob’s pike treatment plan.

I did some administrative work on the akentsoc.org website.

I did some more editing of this week’s *Refuge Notebook* article and got
it sent out.

I drove the Berlese samples to FedEx Kenai and got them shipped out
(FedeX tracking number: 813677693827).

I finished entering data from 2019 black spruce project Berlese and soil
samples.

``` r

## Looking at arthropod sample tally data.

tally_data <- read.csv("../../data/arthropod_samples/sorting_tallies/arthropod_sample_tallies.csv",
 stringsAsFactors=FALSE
 )

dim(tally_data)
[1] 595   4

length(levels(as.factor(tally_data$guid)))
[1] 52

sample_data <- unique(tally_data[,c("guid", "sample_type")])
sample_data

write.csv(sample_data,
 "2020-04-22-1313_sample_data.csv",
 row.names=FALSE
 )


aggregate(sample_data$guid, by=list(sample_data$sample_type), length)
    Group.1  x
1   Berlese 26
2 sweep net 26
## Looks good.

## How many specimens (an understimate due to inability to count clumps of springtails, mites, etc. in soil samples)?
sum(tally_data$count)
[1] 10636

## How many for each method?
aggregate(tally_data$count,
 by=list(tally_data$sample_type),
 sum
 )
    Group.1    x
1   Berlese 8568
2 sweep net 2068

## per sample averages.
aggregate(tally_data$count,
 by=list(tally_data$sample_type),
 sum
 )$x/26
[1] 329.53846  79.53846
 
## Now we need higher taxonomy for summarizing.
identifications <- levels(as.factor(tally_data$identification))
write.csv(identifications,
 "2020-04-22-1318_identifications.csv",
 row.names=FALSE
 )
 
## I edited that in MS Excel.
classification <- read.csv("2020-04-22-1335_classifications.csv",
 stringsAsFactors=FALSE
 ) 

## Putting this together. 
tally_data_wclass <- merge(tally_data,
 classification,
 all.x=TRUE
 )
 
## Saving this.
write.csv(tally_data_wclass, "../../data/arthropod_samples/sorting_tallies/arthropod_sample_tallies.csv",
 row.names=FALSE
 ) 
 
ag_sample_type_order <- aggregate(tally_data_wclass$count,
 by=list(tally_data_wclass$ord,
  tally_data_wclass$sample_type
  ),
 sum
 )

ag_sample_type_order
names(ag_sample_type_order) <- c("ord", "sample_type", "count")

ag_sample_type_order <- ag_sample_type_order[order(ag_sample_type_order$sample_type,
 -ag_sample_type_order$count,
 ag_sample_type_order$ord
 ),]
ag_sample_type_order <- ag_sample_type_order[nrow(ag_sample_type_order):1,]
 
## Now plotting.

pdf(file="2020-04-22-1457_barplot.pdf",
 width=7,
 height=6
 ) 
par(las=2) 
par(mar=c(5,9,1,1)) 
bp <- barplot(ag_sample_type_order$count,
 horiz=TRUE,
 names.arg=ag_sample_type_order$ord,
 cex.names=0.8,
 xlab="count of specimens",
 xlim=c(0,7500),
 main="Counts of specimens by sample type and order"
 ) 
text(ag_sample_type_order$count + 300,
 bp,
 ag_sample_type_order$count
 )
text(rep(4000,2),
 c(mean(c(bp[34], bp[20])), mean(c(bp[19], bp[1]))),
 c("Berlese samples", "sweep net samples")
 )
abline(h=18.2,
 lty="dotted"
 ) 
dev.off()

save.image("2020-04-22-1523_workspace.RData")
```

![Counts of specimens by sample type and
order.](2020-04-22-1457_barplot.png)  
Counts of specimens by sample type and order.

## Thursday, April 23

Last night I downloaded and installed Miniconda 3 and PIPITS 2.5.

To do:

  - Slikok occupancy?
  - Review Rob’s pike treatment plan.

I spent much of the morning setting up and testing a PIPITS 2.5
environment (Gweon et al. [2015](#ref-gweon_pipits_2015))
(<https://github.com/hsgweon/pipits>) and working on some tweaks to run
it using exact sequence variant methods instead of the OTU clustering
method that is the default.

Testing, following directions at <https://github.com/hsgweon/pipits>.

``` shell
cd ~/Documents/FWS

wget https://sourceforge.net/projects/pipits/files/PIPITS_TESTDATA/pipits_test.tar.gz -O pipits_test.tar.gz
tar xvfz pipits_test.tar.gz

cd pipits_test
source activate pipits_env
pispino_createreadpairslist -i rawdata -o readpairslist.txt
pispino_seqprep -i rawdata -o out_seqprep -l readpairslist.txt
pipits_funits -i out_seqprep/prepped.fasta -o out_funits -x ITS2 -v -r
pipits_process -i out_funits/ITS.fasta -o out_process -v -r
```

That worked.

Now I would like to use an exact sequence variant filtering method and
avoid the OTU clustering.

I looked at how this was done in the SCVUC pipeline shell: “vsearch
–cluster\_unoise {input} –sizein –sizeout –minsize
{config\[VSEARCH\_DENOISE\]\[minsize\]} –centroids {output} –log {log}”
VSEARCH\_DENOISE: minsize: 3

``` shell
vsearch --cluster_unoise out_funits/ITS.fasta --sizein --sizeout --minsize 3 --centroids out_funits/ITS_denoised.fasta
```

That got rid of all the sequences. Trying `--minsize 2`.

``` shell
rm out_funits/ITS_denoised.fasta
vsearch --cluster_unoise out_funits/ITS.fasta --sizein --sizeout --minsize 2 --centroids out_funits/ITS_denoised.fasta
```

That also discarded all sequences.

I edited the input FASTA, manually making more copies of one sequence.

``` shell
rm out_funits/ITS_denoised.fasta
vsearch --cluster_unoise out_funits/ITS_edited.fasta --sizein --sizeout --minsize 3 --centroids out_funits/ITS_denoised.fasta
```

Oh, I think I see. The `-sizein` and `--sizeout` options are for
dereplicated input and output.

``` shell
rm out_funits/ITS_denoised.fasta
vsearch --cluster_unoise out_funits/ITS.fasta --minsize 3 --centroids out_funits/ITS_denoised.fasta
```

That still filtered out all sequences. Trying again to make sure I did
not mess anything up. Deleted products of ITS and processing steps.

``` shell
pipits_funits -i out_seqprep/prepped.fasta -o out_funits -x ITS2 -v -r
```

Ok, I see that the dereplicated file after ITS sequencing is
out\_funits/intermediate/derep.ITS2.sizefiltered.fasta

``` shell
vsearch --cluster_unoise out_funits/intermediate/derep.ITS2.sizefiltered.fasta --sizein --sizeout --minsize 3 --centroids out_funits/ITS_denoised.fasta
```

Need to rereplicate now.

``` shell
pipits_rereplicate -i out_funits/ITS_denoised.fasta -o out_funits/ITS.fasta --uc out_funits/intermediate/derep.uc
```

Now trying the rest with no OTU clustering.

``` shell
pipits_process -i out_funits/ITS.fasta -o out_process -v -r -d 1.00
```

That worked\! Output looked good.

Later in the day, trying this out on 2019 black spruce soil fungal
samples.

``` shell
cd ~/Documents/FWS/2019_black_spruce_inventory/data/soil_fungi
source activate pipits_env
## renaming files.
cd raw_fastq
rename -n 's/6202-soifun//' *
rename 's/6202-soifun//' *
rename -n 's/-ITS3mix-ITS4ngs//' *
rename 's/-ITS3mix-ITS4ngs//' *
cd ..
pispino_createreadpairslist -i raw_fastq -o readpairslist.txt
pispino_seqprep -i raw_fastq -o out_seqprep -l readpairslist.txt
pipits_funits -i out_seqprep/prepped.fasta -o out_funits -x ITS2 -v -r
```

I worked some on adding geographic extents to the metadata of the
snowshow hare data in preparation for the data management early
adopters’ meeting.

### Data management early adopters’ meeting at 13:00

  - Still no specific homework.

Transferring 2019 black spruce soil fungi raw metagenomic data from the
server over VPN to my laptop took a long time, but it worked in the end.
I initiated a PIPITS analysis of this dataset.

Pretty complete I/O from today:

``` shell
(pipits_env) matt@line:~/Documents/FWS/2019_black_spruce_inventory/data/soil_fungi$ cd raw_fastq
(pipits_env) matt@line:~/Documents/FWS/2019_black_spruce_inventory/data/soil_fungi/raw_fastq$ rename -n 's/6202-soifun//' *
rename(6202-soifun3069-1-ITS3mix-ITS4ngs_R1.fastq, 3069-1-ITS3mix-ITS4ngs_R1.fastq)
rename(6202-soifun3069-1-ITS3mix-ITS4ngs_R2.fastq, 3069-1-ITS3mix-ITS4ngs_R2.fastq)
rename(6202-soifun3069-2-ITS3mix-ITS4ngs_R1.fastq, 3069-2-ITS3mix-ITS4ngs_R1.fastq)
rename(6202-soifun3069-2-ITS3mix-ITS4ngs_R2.fastq, 3069-2-ITS3mix-ITS4ngs_R2.fastq)
rename(6202-soifun3072-1-ITS3mix-ITS4ngs_R1.fastq, 3072-1-ITS3mix-ITS4ngs_R1.fastq)
rename(6202-soifun3072-1-ITS3mix-ITS4ngs_R2.fastq, 3072-1-ITS3mix-ITS4ngs_R2.fastq)
rename(6202-soifun3072-2-ITS3mix-ITS4ngs_R1.fastq, 3072-2-ITS3mix-ITS4ngs_R1.fastq)
rename(6202-soifun3072-2-ITS3mix-ITS4ngs_R2.fastq, 3072-2-ITS3mix-ITS4ngs_R2.fastq)
rename(6202-soifun3074-1-ITS3mix-ITS4ngs_R1.fastq, 3074-1-ITS3mix-ITS4ngs_R1.fastq)
rename(6202-soifun3074-1-ITS3mix-ITS4ngs_R2.fastq, 3074-1-ITS3mix-ITS4ngs_R2.fastq)
rename(6202-soifun3074-2-ITS3mix-ITS4ngs_R1.fastq, 3074-2-ITS3mix-ITS4ngs_R1.fastq)
rename(6202-soifun3074-2-ITS3mix-ITS4ngs_R2.fastq, 3074-2-ITS3mix-ITS4ngs_R2.fastq)
rename(6202-soifun3092-1-ITS3mix-ITS4ngs_R1.fastq, 3092-1-ITS3mix-ITS4ngs_R1.fastq)
rename(6202-soifun3092-1-ITS3mix-ITS4ngs_R2.fastq, 3092-1-ITS3mix-ITS4ngs_R2.fastq)
rename(6202-soifun3092-2-ITS3mix-ITS4ngs_R1.fastq, 3092-2-ITS3mix-ITS4ngs_R1.fastq)
rename(6202-soifun3092-2-ITS3mix-ITS4ngs_R2.fastq, 3092-2-ITS3mix-ITS4ngs_R2.fastq)
rename(6202-soifun3140-1-ITS3mix-ITS4ngs_R1.fastq, 3140-1-ITS3mix-ITS4ngs_R1.fastq)
rename(6202-soifun3140-1-ITS3mix-ITS4ngs_R2.fastq, 3140-1-ITS3mix-ITS4ngs_R2.fastq)
rename(6202-soifun3140-2-ITS3mix-ITS4ngs_R1.fastq, 3140-2-ITS3mix-ITS4ngs_R1.fastq)
rename(6202-soifun3140-2-ITS3mix-ITS4ngs_R2.fastq, 3140-2-ITS3mix-ITS4ngs_R2.fastq)
rename(6202-soifun3202-1-ITS3mix-ITS4ngs_R1.fastq, 3202-1-ITS3mix-ITS4ngs_R1.fastq)
rename(6202-soifun3202-1-ITS3mix-ITS4ngs_R2.fastq, 3202-1-ITS3mix-ITS4ngs_R2.fastq)
rename(6202-soifun3202-2-ITS3mix-ITS4ngs_R1.fastq, 3202-2-ITS3mix-ITS4ngs_R1.fastq)
rename(6202-soifun3202-2-ITS3mix-ITS4ngs_R2.fastq, 3202-2-ITS3mix-ITS4ngs_R2.fastq)
rename(6202-soifun3288-1-ITS3mix-ITS4ngs_R1.fastq, 3288-1-ITS3mix-ITS4ngs_R1.fastq)
rename(6202-soifun3288-1-ITS3mix-ITS4ngs_R2.fastq, 3288-1-ITS3mix-ITS4ngs_R2.fastq)
rename(6202-soifun3288-2-ITS3mix-ITS4ngs_R1.fastq, 3288-2-ITS3mix-ITS4ngs_R1.fastq)
rename(6202-soifun3288-2-ITS3mix-ITS4ngs_R2.fastq, 3288-2-ITS3mix-ITS4ngs_R2.fastq)
rename(6202-soifun5555-1-ITS3mix-ITS4ngs_R1.fastq, 5555-1-ITS3mix-ITS4ngs_R1.fastq)
rename(6202-soifun5555-1-ITS3mix-ITS4ngs_R2.fastq, 5555-1-ITS3mix-ITS4ngs_R2.fastq)
rename(6202-soifun5555-2-ITS3mix-ITS4ngs_R1.fastq, 5555-2-ITS3mix-ITS4ngs_R1.fastq)
rename(6202-soifun5555-2-ITS3mix-ITS4ngs_R2.fastq, 5555-2-ITS3mix-ITS4ngs_R2.fastq)
rename(6202-soifunMG07-1-ITS3mix-ITS4ngs_R1.fastq, MG07-1-ITS3mix-ITS4ngs_R1.fastq)
rename(6202-soifunMG07-1-ITS3mix-ITS4ngs_R2.fastq, MG07-1-ITS3mix-ITS4ngs_R2.fastq)
rename(6202-soifunMG07-2-ITS3mix-ITS4ngs_R1.fastq, MG07-2-ITS3mix-ITS4ngs_R1.fastq)
rename(6202-soifunMG07-2-ITS3mix-ITS4ngs_R2.fastq, MG07-2-ITS3mix-ITS4ngs_R2.fastq)
rename(6202-soifunMG19-1-ITS3mix-ITS4ngs_R1.fastq, MG19-1-ITS3mix-ITS4ngs_R1.fastq)
rename(6202-soifunMG19-1-ITS3mix-ITS4ngs_R2.fastq, MG19-1-ITS3mix-ITS4ngs_R2.fastq)
rename(6202-soifunMG19-2-ITS3mix-ITS4ngs_R1.fastq, MG19-2-ITS3mix-ITS4ngs_R1.fastq)
rename(6202-soifunMG19-2-ITS3mix-ITS4ngs_R2.fastq, MG19-2-ITS3mix-ITS4ngs_R2.fastq)
rename(6202-soifunMG24-1-ITS3mix-ITS4ngs_R1.fastq, MG24-1-ITS3mix-ITS4ngs_R1.fastq)
rename(6202-soifunMG24-1-ITS3mix-ITS4ngs_R2.fastq, MG24-1-ITS3mix-ITS4ngs_R2.fastq)
rename(6202-soifunMG24-2-ITS3mix-ITS4ngs_R1.fastq, MG24-2-ITS3mix-ITS4ngs_R1.fastq)
rename(6202-soifunMG24-2-ITS3mix-ITS4ngs_R2.fastq, MG24-2-ITS3mix-ITS4ngs_R2.fastq)
rename(6202-soifunMG39-1-ITS3mix-ITS4ngs_R1.fastq, MG39-1-ITS3mix-ITS4ngs_R1.fastq)
rename(6202-soifunMG39-1-ITS3mix-ITS4ngs_R2.fastq, MG39-1-ITS3mix-ITS4ngs_R2.fastq)
rename(6202-soifunMG39-2-ITS3mix-ITS4ngs_R1.fastq, MG39-2-ITS3mix-ITS4ngs_R1.fastq)
rename(6202-soifunMG39-2-ITS3mix-ITS4ngs_R2.fastq, MG39-2-ITS3mix-ITS4ngs_R2.fastq)
rename(6202-soifunSK22-1-ITS3mix-ITS4ngs_R1.fastq, SK22-1-ITS3mix-ITS4ngs_R1.fastq)
rename(6202-soifunSK22-1-ITS3mix-ITS4ngs_R2.fastq, SK22-1-ITS3mix-ITS4ngs_R2.fastq)
rename(6202-soifunSK22-2-ITS3mix-ITS4ngs_R1.fastq, SK22-2-ITS3mix-ITS4ngs_R1.fastq)
rename(6202-soifunSK22-2-ITS3mix-ITS4ngs_R2.fastq, SK22-2-ITS3mix-ITS4ngs_R2.fastq)
(pipits_env) matt@line:~/Documents/FWS/2019_black_spruce_inventory/data/soil_fungi/raw_fastq$ rename 's/6202-soifun//' *
(pipits_env) matt@line:~/Documents/FWS/2019_black_spruce_inventory/data/soil_fungi/raw_fastq$ ls
3069-1-ITS3mix-ITS4ngs_R1.fastq  3140-2-ITS3mix-ITS4ngs_R1.fastq  MG19-1-ITS3mix-ITS4ngs_R1.fastq
3069-1-ITS3mix-ITS4ngs_R2.fastq  3140-2-ITS3mix-ITS4ngs_R2.fastq  MG19-1-ITS3mix-ITS4ngs_R2.fastq
3069-2-ITS3mix-ITS4ngs_R1.fastq  3202-1-ITS3mix-ITS4ngs_R1.fastq  MG19-2-ITS3mix-ITS4ngs_R1.fastq
3069-2-ITS3mix-ITS4ngs_R2.fastq  3202-1-ITS3mix-ITS4ngs_R2.fastq  MG19-2-ITS3mix-ITS4ngs_R2.fastq
3072-1-ITS3mix-ITS4ngs_R1.fastq  3202-2-ITS3mix-ITS4ngs_R1.fastq  MG24-1-ITS3mix-ITS4ngs_R1.fastq
3072-1-ITS3mix-ITS4ngs_R2.fastq  3202-2-ITS3mix-ITS4ngs_R2.fastq  MG24-1-ITS3mix-ITS4ngs_R2.fastq
3072-2-ITS3mix-ITS4ngs_R1.fastq  3288-1-ITS3mix-ITS4ngs_R1.fastq  MG24-2-ITS3mix-ITS4ngs_R1.fastq
3072-2-ITS3mix-ITS4ngs_R2.fastq  3288-1-ITS3mix-ITS4ngs_R2.fastq  MG24-2-ITS3mix-ITS4ngs_R2.fastq
3074-1-ITS3mix-ITS4ngs_R1.fastq  3288-2-ITS3mix-ITS4ngs_R1.fastq  MG39-1-ITS3mix-ITS4ngs_R1.fastq
3074-1-ITS3mix-ITS4ngs_R2.fastq  3288-2-ITS3mix-ITS4ngs_R2.fastq  MG39-1-ITS3mix-ITS4ngs_R2.fastq
3074-2-ITS3mix-ITS4ngs_R1.fastq  5555-1-ITS3mix-ITS4ngs_R1.fastq  MG39-2-ITS3mix-ITS4ngs_R1.fastq
3074-2-ITS3mix-ITS4ngs_R2.fastq  5555-1-ITS3mix-ITS4ngs_R2.fastq  MG39-2-ITS3mix-ITS4ngs_R2.fastq
3092-1-ITS3mix-ITS4ngs_R1.fastq  5555-2-ITS3mix-ITS4ngs_R1.fastq  SK22-1-ITS3mix-ITS4ngs_R1.fastq
3092-1-ITS3mix-ITS4ngs_R2.fastq  5555-2-ITS3mix-ITS4ngs_R2.fastq  SK22-1-ITS3mix-ITS4ngs_R2.fastq
3092-2-ITS3mix-ITS4ngs_R1.fastq  MG07-1-ITS3mix-ITS4ngs_R1.fastq  SK22-2-ITS3mix-ITS4ngs_R1.fastq
3092-2-ITS3mix-ITS4ngs_R2.fastq  MG07-1-ITS3mix-ITS4ngs_R2.fastq  SK22-2-ITS3mix-ITS4ngs_R2.fastq
3140-1-ITS3mix-ITS4ngs_R1.fastq  MG07-2-ITS3mix-ITS4ngs_R1.fastq
3140-1-ITS3mix-ITS4ngs_R2.fastq  MG07-2-ITS3mix-ITS4ngs_R2.fastq
(pipits_env) matt@line:~/Documents/FWS/2019_black_spruce_inventory/data/soil_fungi/raw_fastq$ rename -n 's/-ITS3mix-ITS4ngs//' *
rename(3069-1-ITS3mix-ITS4ngs_R1.fastq, 3069-1_R1.fastq)
rename(3069-1-ITS3mix-ITS4ngs_R2.fastq, 3069-1_R2.fastq)
rename(3069-2-ITS3mix-ITS4ngs_R1.fastq, 3069-2_R1.fastq)
rename(3069-2-ITS3mix-ITS4ngs_R2.fastq, 3069-2_R2.fastq)
rename(3072-1-ITS3mix-ITS4ngs_R1.fastq, 3072-1_R1.fastq)
rename(3072-1-ITS3mix-ITS4ngs_R2.fastq, 3072-1_R2.fastq)
rename(3072-2-ITS3mix-ITS4ngs_R1.fastq, 3072-2_R1.fastq)
rename(3072-2-ITS3mix-ITS4ngs_R2.fastq, 3072-2_R2.fastq)
rename(3074-1-ITS3mix-ITS4ngs_R1.fastq, 3074-1_R1.fastq)
rename(3074-1-ITS3mix-ITS4ngs_R2.fastq, 3074-1_R2.fastq)
rename(3074-2-ITS3mix-ITS4ngs_R1.fastq, 3074-2_R1.fastq)
rename(3074-2-ITS3mix-ITS4ngs_R2.fastq, 3074-2_R2.fastq)
rename(3092-1-ITS3mix-ITS4ngs_R1.fastq, 3092-1_R1.fastq)
rename(3092-1-ITS3mix-ITS4ngs_R2.fastq, 3092-1_R2.fastq)
rename(3092-2-ITS3mix-ITS4ngs_R1.fastq, 3092-2_R1.fastq)
rename(3092-2-ITS3mix-ITS4ngs_R2.fastq, 3092-2_R2.fastq)
rename(3140-1-ITS3mix-ITS4ngs_R1.fastq, 3140-1_R1.fastq)
rename(3140-1-ITS3mix-ITS4ngs_R2.fastq, 3140-1_R2.fastq)
rename(3140-2-ITS3mix-ITS4ngs_R1.fastq, 3140-2_R1.fastq)
rename(3140-2-ITS3mix-ITS4ngs_R2.fastq, 3140-2_R2.fastq)
rename(3202-1-ITS3mix-ITS4ngs_R1.fastq, 3202-1_R1.fastq)
rename(3202-1-ITS3mix-ITS4ngs_R2.fastq, 3202-1_R2.fastq)
rename(3202-2-ITS3mix-ITS4ngs_R1.fastq, 3202-2_R1.fastq)
rename(3202-2-ITS3mix-ITS4ngs_R2.fastq, 3202-2_R2.fastq)
rename(3288-1-ITS3mix-ITS4ngs_R1.fastq, 3288-1_R1.fastq)
rename(3288-1-ITS3mix-ITS4ngs_R2.fastq, 3288-1_R2.fastq)
rename(3288-2-ITS3mix-ITS4ngs_R1.fastq, 3288-2_R1.fastq)
rename(3288-2-ITS3mix-ITS4ngs_R2.fastq, 3288-2_R2.fastq)
rename(5555-1-ITS3mix-ITS4ngs_R1.fastq, 5555-1_R1.fastq)
rename(5555-1-ITS3mix-ITS4ngs_R2.fastq, 5555-1_R2.fastq)
rename(5555-2-ITS3mix-ITS4ngs_R1.fastq, 5555-2_R1.fastq)
rename(5555-2-ITS3mix-ITS4ngs_R2.fastq, 5555-2_R2.fastq)
rename(MG07-1-ITS3mix-ITS4ngs_R1.fastq, MG07-1_R1.fastq)
rename(MG07-1-ITS3mix-ITS4ngs_R2.fastq, MG07-1_R2.fastq)
rename(MG07-2-ITS3mix-ITS4ngs_R1.fastq, MG07-2_R1.fastq)
rename(MG07-2-ITS3mix-ITS4ngs_R2.fastq, MG07-2_R2.fastq)
rename(MG19-1-ITS3mix-ITS4ngs_R1.fastq, MG19-1_R1.fastq)
rename(MG19-1-ITS3mix-ITS4ngs_R2.fastq, MG19-1_R2.fastq)
rename(MG19-2-ITS3mix-ITS4ngs_R1.fastq, MG19-2_R1.fastq)
rename(MG19-2-ITS3mix-ITS4ngs_R2.fastq, MG19-2_R2.fastq)
rename(MG24-1-ITS3mix-ITS4ngs_R1.fastq, MG24-1_R1.fastq)
rename(MG24-1-ITS3mix-ITS4ngs_R2.fastq, MG24-1_R2.fastq)
rename(MG24-2-ITS3mix-ITS4ngs_R1.fastq, MG24-2_R1.fastq)
rename(MG24-2-ITS3mix-ITS4ngs_R2.fastq, MG24-2_R2.fastq)
rename(MG39-1-ITS3mix-ITS4ngs_R1.fastq, MG39-1_R1.fastq)
rename(MG39-1-ITS3mix-ITS4ngs_R2.fastq, MG39-1_R2.fastq)
rename(MG39-2-ITS3mix-ITS4ngs_R1.fastq, MG39-2_R1.fastq)
rename(MG39-2-ITS3mix-ITS4ngs_R2.fastq, MG39-2_R2.fastq)
rename(SK22-1-ITS3mix-ITS4ngs_R1.fastq, SK22-1_R1.fastq)
rename(SK22-1-ITS3mix-ITS4ngs_R2.fastq, SK22-1_R2.fastq)
rename(SK22-2-ITS3mix-ITS4ngs_R1.fastq, SK22-2_R1.fastq)
rename(SK22-2-ITS3mix-ITS4ngs_R2.fastq, SK22-2_R2.fastq)
(pipits_env) matt@line:~/Documents/FWS/2019_black_spruce_inventory/data/soil_fungi/raw_fastq$ rename 's/-ITS3mix-ITS4ngs//' *
(pipits_env) matt@line:~/Documents/FWS/2019_black_spruce_inventory/data/soil_fungi/raw_fastq$ ls
3069-1_R1.fastq  3074-1_R2.fastq  3140-2_R1.fastq  3288-2_R2.fastq  MG19-1_R1.fastq  MG39-1_R2.fastq
3069-1_R2.fastq  3074-2_R1.fastq  3140-2_R2.fastq  5555-1_R1.fastq  MG19-1_R2.fastq  MG39-2_R1.fastq
3069-2_R1.fastq  3074-2_R2.fastq  3202-1_R1.fastq  5555-1_R2.fastq  MG19-2_R1.fastq  MG39-2_R2.fastq
3069-2_R2.fastq  3092-1_R1.fastq  3202-1_R2.fastq  5555-2_R1.fastq  MG19-2_R2.fastq  SK22-1_R1.fastq
3072-1_R1.fastq  3092-1_R2.fastq  3202-2_R1.fastq  5555-2_R2.fastq  MG24-1_R1.fastq  SK22-1_R2.fastq
3072-1_R2.fastq  3092-2_R1.fastq  3202-2_R2.fastq  MG07-1_R1.fastq  MG24-1_R2.fastq  SK22-2_R1.fastq
3072-2_R1.fastq  3092-2_R2.fastq  3288-1_R1.fastq  MG07-1_R2.fastq  MG24-2_R1.fastq  SK22-2_R2.fastq
3072-2_R2.fastq  3140-1_R1.fastq  3288-1_R2.fastq  MG07-2_R1.fastq  MG24-2_R2.fastq
3074-1_R1.fastq  3140-1_R2.fastq  3288-2_R1.fastq  MG07-2_R2.fastq  MG39-1_R1.fastq
(pipits_env) matt@line:~/Documents/FWS/2019_black_spruce_inventory/data/soil_fungi/raw_fastq$ pispino_createreadpairslist -i raw_fastq -o readpairslist.txt
Generating a read-pair list file from the input directory...
Traceback (most recent call last):
  File "/home/matt/miniconda3/envs/pipits_env/bin/pispino_createreadpairslist", line 164, in <module>
    make_read_pairs_list(options)
  File "/home/matt/miniconda3/envs/pipits_env/bin/pispino_createreadpairslist", line 37, in make_read_pairs_list
    for file in os.listdir(options.inputdir):
FileNotFoundError: [Errno 2] No such file or directory: 'raw_fastq'
(pipits_env) matt@line:~/Documents/FWS/2019_black_spruce_inventory/data/soil_fungi/raw_fastq$ cd ..
(pipits_env) matt@line:~/Documents/FWS/2019_black_spruce_inventory/data/soil_fungi$ pispino_createreadpairslist -i raw_fastq -o readpairslist.txt
Generating a read-pair list file from the input directory...
Done - "readpairslist.txt" created
(pipits_env) matt@line:~/Documents/FWS/2019_black_spruce_inventory/data/soil_fungi$ pispino_seqprep -i rawdata -o out_seqprep -l readpairslist.txt
2020-04-23 22:22:54 pispino_seqprep started
2020-04-23 22:22:54 ERROR: Cannot find "rawdata" directory. Ensure you have the correct name of the input directory.
(pipits_env) matt@line:~/Documents/FWS/2019_black_spruce_inventory/data/soil_fungi$ pispino_seqprep -i raw_fastq -o out_seqprep -l readpairslist.txt
2020-04-23 22:23:16 pispino_seqprep started
2020-04-23 22:23:16 Checking listfile
2020-04-23 22:23:16 ... done
2020-04-23 22:23:16 Counting sequences in rawdata
2020-04-23 22:23:17 ... number of reads: 748853
2020-04-23 22:23:17 Reindexing forward reads
2020-04-23 22:23:25 ... done
2020-04-23 22:23:25 Reindexing reverse reads
2020-04-23 22:23:32 ... done
2020-04-23 22:23:32 Joining paired-end reads [VSEARCH]
2020-04-23 22:24:09 ... number of joined reads: 710194
2020-04-23 22:24:09 Quality filtering [FASTX]
2020-04-23 22:31:43 ... number of quality filtered reads: 708943
2020-04-23 22:31:43 Converting FASTQ to FASTA [FASTX] (also removing reads with "N" nucleotide if specified with "--FASTX-n")
2020-04-23 22:35:34 ... number of prepped sequences: 708943
2020-04-23 22:35:34 Merging into a single file
2020-04-23 22:35:37 ... done
2020-04-23 22:35:37 Cleaning temporary directory
2020-04-23 22:35:38 ... done
2020-04-23 22:35:38 Done - pispino_seqprep completed (Resulting file: out_seqprep/prepped.fasta)
(pipits_env) matt@line:~/Documents/FWS/2019_black_spruce_inventory/data/soil_fungi$ pipits_funits -i out_seqprep/prepped.fasta -o out_funits -x ITS2 -v -r
pipits_funits 2.5, the PIPITS Project
https://github.com/hsgweon/pipits
---------------------------------

2020-04-23 22:37:33 pipits_funits started
2020-04-23 22:37:33 Checking input FASTA for illegal characters
2020-04-23 22:37:34 ... done
2020-04-23 22:37:34 Counting input sequences
2020-04-23 22:37:35 ... number of input sequences: 708943
2020-04-23 22:37:35 Dereplicating sequences for efficiency
vsearch v2.14.2_linux_x86_64, 7.8GB RAM, 2 cores
https://github.com/torognes/vsearch

Dereplicating file out_seqprep/prepped.fasta 100%
257686201 nt in 708943 seqs, min 112, max 530, avg 363
Sorting 100%
328449 unique sequences, avg cluster 2.2, median 1, max 3136
Writing output file 100%
Writing uc file, first part 100%
Writing uc file, second part 100%
2020-04-23 22:37:41 ... done
2020-04-23 22:37:41 Counting dereplicated sequences
2020-04-23 22:37:41 ... number of dereplicated sequences: 328449
2020-04-23 22:37:41 Extracting ITS2 from sequences [ITSx]
ITSx -- Identifies ITS sequences and extracts the ITS region
by Johan Bengtsson-Palme et al., University of Gothenburg
Version: 1.1b1
-----------------------------------------------------------------
Thu Apr 23 22:37:41 2020 : Preparing HMM database (should be quick)...
Thu Apr 23 22:37:42 2020 : Checking and handling input sequence data (should not take long)...
Thu Apr 23 22:37:50 2020 : Comparing sequences to HMM database (this may take a long while)...
```

## Friday, April 24

As of this morning that PIPITS analysis is still on the ITSx step.

Yesterday I saw that we have some *Rabdophaga* willow rosette galls on
*Salix pulchra* and *Salix fuscescens* in the swamp behind our house. I
had been looking for rosette galls on *S. fuscescens* last year because
it looked like there might possibly be a difference between COI
sequences of *Rabdophaga* gall makers on these two hosts based on
previous work (Bowser et al. [2018](#ref-bowser_dna_2018)), but this
included only two sequences from *Rabdophaga* on *S. pulchra* and one
sequence from *Rabdophaga* on *S. fuscescens*. Sato et
al. ([2020](#ref-Sato_et_al_Rabdophaga_2020)) did not include any
sequences from *Rabdophaga* on *S. fuscescens*.

I downloaded *Rabdophaga* sequences from GenBank with the search string
“Sato Rabdophaga” and all *Rabdophaga* sequences from BOLD with the
search string “Rabdophaga”.

I put these sequences together into one FASTA file and I submitted them
to NGPhylogeny.fr. I used the “NGPhylogeny Analyse - FastME/OneClick”
analysis. This analysis is available at
<https://ngphylogeny.fr/workspace/history/5d1c74c0f1258998>.

### Biology check in meeting at 10:00.

  - Acquisition of a MinION may be funded.
  - Should try using NGS to look at sheep diet.

That did not include the new sequences from Sato et
al. ([2020](#ref-Sato_et_al_Rabdophaga_2020)). Sequence
[AB244581](https://www.ncbi.nlm.nih.gov/nuccore/AB244581) was from the
*Rabdophaga rosaria* group, though. It was identified as *Rabdophaga
strobilina* by Sato and Yukawa ([2006](#ref-sato_resurrection_2006)).
This is also in the BOLD dataset.

I tried searching GenBank with the string “Genetic variation in
intraspecific populations of Rabdophaga rosaria (Diptera: Cecidomyiidae)
indicating possible diversification scenarios into sibling species along
with host range expansion on willows (Salicaceae: Salix)”. This yielded
44 sequences.

I combined the BOLD and GenBank sequences and submitted them to
NGPhylogeny.fr, using the “NGPhylogeny Analyse - FastME/OneClick”
analysis. This analysis is available at
<https://ngphylogeny.fr/workspace/history/9bcb72e8e1d20fe1>.

It looks like all of our *Rabdophaga* sequences from *S. pulchra* and
*S. fuscescens* are *Rabdophaga rosaria* subclade 2B *sensu* Sato et
al. ([2020](#ref-Sato_et_al_Rabdophaga_2020)).

![Phylogenetic tree of *Rabdophaga* COI
sequences.](2020-04-24-1325_Rabdophaga_tree.png)  
Phylogenetic tree of *Rabdophaga* COI sequences.

In the afternoon Ethan and I collected rosette galls on *Salix
fuscescens* from the swamp in our back yard to determine whether the
observed genetic distance between *Rabdophaga rosaria* on *Salix
pulchra* and *Salix fuscescens* is real or an artifact of small sample
size and errors. We have a total of four COI sequences from *Rabdophaga
rosaria* on *Salix pulchra* (2 from Sato et
al. ([2020](#ref-Sato_et_al_Rabdophaga_2020)) + 2 of ours) and one from
*Rabdophaga rosaria* on *Salix fuscescens* (ours). I expect that in the
end we will find that there is one *Rabdophaga* on both willow hosts,
but we will check.

The PIPITS `ITSx` step is still running. Complete I/O from today:

``` shell
    Fri Apr 24 15:47:53 2020 : Fungi analysis of main strand finished.
```

## Saturday, April 25

``` shell
## Denoising.
vsearch --cluster_unoise out_funits/intermediate/derep.ITS2.sizefiltered.fasta --sizein --sizeout --minsize 3 --centroids out_funits/ITS_denoised.fasta
## rereplicate.
pipits_rereplicate -i out_funits/ITS_denoised.fasta -o out_funits/ITS_denoised_rerep.fasta --uc out_funits/intermediate/derep.uc
## That did not work. Trying cluster_unoise on the ITS.fasta file.
vsearch --cluster_unoise out_funits/ITS.fasta --minsize 3 --centroids out_funits/ITS_denoised.fasta
## That did not work.
```

Complete I/O from today:

``` shell
    Sat Apr 25 04:39:37 2020 : Fungi analysis of complementary strand finished.
Sat Apr 25 04:39:37 2020 : Analysing results of HMM-scan (this might take quite some time)...
Sat Apr 25 04:49:42 2020 : Extraction finished!
-----------------------------------------------------------------
Thank you for using ITSx!
Please report bugs or unsupported lineages to itsx@microbiology.se

2020-04-25 04:49:45 ... done
2020-04-25 04:49:45 Counting ITS sequences (dereplicated)
2020-04-25 04:49:46 ... number of ITS sequences (dereplicated): 326274
2020-04-25 04:49:46 Sorting by ID
[INFO] read sequences ...
[INFO] 326274 sequences loaded
[INFO] sorting ...
[INFO] output ...
2020-04-25 04:49:51 ... done
2020-04-25 04:49:51 Removing short sequences below < 100bp
vsearch v2.14.2_linux_x86_64, 7.8GB RAM, 2 cores
https://github.com/torognes/vsearch

Reading input file 100%
325804 sequences kept (of which 0 truncated), 470 sequences discarded.
2020-04-25 04:49:52 ... done
2020-04-25 04:49:52 Counting length-filtered sequences (dereplicated)
2020-04-25 04:49:53 ... number of length-filtered sequences (dereplicated): 325804
2020-04-25 04:49:53 Re-inflating sequences
2020-04-25 04:50:08 ... done
2020-04-25 04:50:08 Counting sequences after re-inflation
2020-04-25 04:50:08 ... number of sequences with ITS subregion: 704602
2020-04-25 04:50:08 Done - pipits_funits ended successfully. (Your ITS sequences are "out_funits/ITS.fasta")
2020-04-25 04:50:08 Next step: pipits_process [ Example: pipits_process -i out_funits/ITS.fasta -o pipits_process ]
(pipits_env) matt@line:~/Documents/FWS/2019_black_spruce_inventory/data/soil_fungi$ ls
2020-04-24-0635_io.txt  out_funits  out_seqprep  raw_fastq  readpairslist.txt
(pipits_env) matt@line:~/Documents/FWS/2019_black_spruce_inventory/data/soil_fungi$ vsearch --cluster_unoise out_funits/intermediate/derep.ITS2.sizefiltered.fasta --sizein --sizeout --minsize 3 --centroids out_funits/ITS_denoised.fasta
vsearch v2.14.2_linux_x86_64, 7.8GB RAM, 2 cores
https://github.com/torognes/vsearch

Reading file out_funits/intermediate/derep.ITS2.sizefiltered.fasta 100%  
3733528 nt in 21662 seqs, min 102, max 263, avg 172
minsize 3: 304142 sequences discarded.
Masking 100% 
Sorting by abundance 100%
Counting k-mers 100% 
Clustering 100%  
Sorting clusters 100%
Writing clusters 100% 
Clusters: 917 Size min 3, max 65239, avg 23.6
Singletons: 0, 0.0% of seqs, 0.0% of clusters
(pipits_env) matt@line:~/Documents/FWS/2019_black_spruce_inventory/data/soil_fungi$ pipits_rereplicate -i out_funits/ITS_denoised.fasta -o out_funits/ITS_denoised_rerep.fasta --uc out_funits/intermediate/derep.uc
Error: Size values in headers added by USEARCH != number of sequences in .uc
65239 3136
```

## Monday, April 27

To do:

  - ~~Finish PIPITS soil fungal analysis.~~
  - ~~Edit next week’s *Refuge Notebook* article.~~

<!-- end list -->

``` shell
## Trying to figure out why denoising didn't work.
## The denoise command I issued the other day yielded an output file with 0 lines.
## Command: vsearch --cluster_unoise out_funits/intermediate/derep.ITS2.sizefiltered.fasta --sizein --sizeout --minsize 3 --centroids out_funits/ITS_denoised.fasta

source activate pipits_env
cd ~/Documents/FWS/2019_black_spruce_inventory/data/soil_fungi
vsearch --cluster_unoise out_funits/intermediate/derep.ITS2.sizefiltered.fasta --sizein --sizeout --minsize 3 --centroids out_funits/ITS_denoised.fasta
vsearch v2.14.2_linux_x86_64, 7.8GB RAM, 2 cores
https://github.com/torognes/vsearch

Reading file out_funits/intermediate/derep.ITS2.sizefiltered.fasta 100% 
3733528 nt in 21662 seqs, min 102, max 263, avg 172
minsize 3: 304142 sequences discarded.
Masking 100% 
Sorting by abundance 100%
Counting k-mers 100% 
Clustering 100% 
Sorting clusters 100%
Writing clusters 100% 
Clusters: 917 Size min 3, max 65239, avg 23.6
Singletons: 0, 0.0% of seqs, 0.0% of clusters
## That seemed to work.

## Trying to rereplicate
pipits_rereplicate -i out_funits/ITS_denoised.fasta -o out_funits/ITS_denoised_rerep.fasta --uc out_funits/intermediate/derep.uc
Error: Size values in headers added by USEARCH != number of sequences in .uc
65239 3136

## Ok, I see what happened. unoise clustered the sequences, and now the clusters are bigger than in the --uc file. ## How can I deal with this?
```

I am editing the `pipits_process` script, renaming it
`pipits_process_ESV`. I am replacing with the OTU clustering section
below.

``` python
    ##################
    # OTU clustering #
    ##################

    logger(ENDC + "Picking OTUs [VSEARCH]" + ENDC, logging_file, display = True)
    cmd = " ".join([VSEARCH, 
                    "--cluster_fast", tmpDir + "/input_nr.fasta", 
                    "--id", options.VSEARCH_id,
                    "--centroids", tmpDir + "/input_nr_otus.fasta",
                    "--uc", tmpDir + "/input_nr_otus.uc",
                    "--threads", options.threads])
    run_cmd(cmd, logging_file, options.verbose)
```

Replacement:

``` python
    ##################
    # Denoising      #
    ##################

    logger(ENDC + "Denoising [VSEARCH]" + ENDC, logging_file, display = True)
    cmd = " ".join([VSEARCH, 
                    "--cluster_unoise", tmpDir + "/input_nr.fasta", 
                    "--minsize 3",
                    "--centroids", tmpDir + "/input_nr_otus.fasta",
                    "--uc", tmpDir + "/input_nr_otus.uc",
                    "--threads", options.threads])
    run_cmd(cmd, logging_file, options.verbose)
```

Trying it out.

``` shell
pipits_process_ESV -i out_funits/ITS.fasta -o out_process -v -r

## That worked!!
```

Complete relevant I/O from today:

```` shell
(base) matt@line:~/Documents/FWS/2019_black_spruce_inventory/data/soil_fungi$ source activate pipits_env
(pipits_env) matt@line:~/Documents/FWS/2019_black_spruce_inventory/data/soil_fungi$ cd ~/Documents/FWS/2019_black_spruce_inventory/data/soil_fungi
(pipits_env) matt@line:~/Documents/FWS/2019_black_spruce_inventory/data/soil_fungi$ ls
2020-04-24-0635_io.txt  out_funits  out_seqprep  raw_fastq  readpairslist.txt
(pipits_env) matt@line:~/Documents/FWS/2019_black_spruce_inventory/data/soil_fungi$ vsearch --cluster_unoise out_funits/intermediate/derep.ITS2.sizefiltered.fasta --sizein --sizeout --minsize 3 --centroids out_funits/ITS_denoised.fasta
vsearch v2.14.2_linux_x86_64, 7.8GB RAM, 2 cores
https://github.com/torognes/vsearch

Reading file out_funits/intermediate/derep.ITS2.sizefiltered.fasta 100%  
3733528 nt in 21662 seqs, min 102, max 263, avg 172
minsize 3: 304142 sequences discarded.
Masking 100% 
Sorting by abundance 100%
Counting k-mers 100% 
Clustering 100%  
Sorting clusters 100%
Writing clusters 100% 
Clusters: 917 Size min 3, max 65239, avg 23.6
Singletons: 0, 0.0% of seqs, 0.0% of clusters
(pipits_env) matt@line:~/Documents/FWS/2019_black_spruce_inventory/data/soil_fungi$ ls -al out_funits
total 131980
drwxrwxr-x 3 matt matt      4096 Apr 25 07:36 .
drwxrwxr-x 5 matt matt      4096 Apr 25 10:52 ..
drwxrwxr-x 2 matt matt      4096 Apr 25 04:49 intermediate
-rw-rw-r-- 1 matt matt    178365 Apr 27 06:03 ITS_denoised.fasta
-rw-rw-r-- 1 matt matt         0 Apr 25 07:36 ITS_denoised_rerep.fasta
-rw-rw-r-- 1 matt matt 134937545 Apr 25 04:50 ITS.fasta
-rw-rw-r-- 1 matt matt      3490 Apr 25 04:50 output.log
-rw-rw-r-- 1 matt matt       228 Apr 25 04:50 summary.log
-rw-rw-r-- 1 matt matt      7068 Apr 25 04:50 versions.log
(pipits_env) matt@line:~/Documents/FWS/2019_black_spruce_inventory/data/soil_fungi$ head ITS_denoised.fasta
head: cannot open 'ITS_denoised.fasta' for reading: No such file or directory
(pipits_env) matt@line:~/Documents/FWS/2019_black_spruce_inventory/data/soil_fungi$ head out_funits/ITS_denoised.fasta
>3069-1_320;size=65239
CTGCCCTCAAGCACGGCTTGTGTGTTGGGCTCCGTCCCCCCGGGGACGGGTCCGAAAGGCAGCGGCGGCACCGAGTCCGG
TCCTCGAGCGTATGGGGCTTTGTCACCCGCTCTGTAGGCCCGGCCGGCGCCAGCCGACAACCAATCATCCTTTTCA
>3074-1_997;size=22437
AACTCTCACCCTCTAGCTTTCTTAATCGTGGCTAGCGGCGTGGACGTGAGCGCTGCTGCTTTGTTGCGGCTCGCTCGAAA
TGCATTAGCAGACCCTTTTCGTAATCGGTTCCACTCAACGTGATAAGTATTTCGTTGAGGACAGTTGCAGCAATGCGGCT
GGCCGGGATAAGAAAGGCATAGTTGTCAGCTTCTAATCGCCCTTGGGCAATTTTTTATG
>3288-1_526;size=21465
AACTCTCAACCTATAGATTTTTGTTAATCTTGTAGGCTTGGACTTGGAAGTCTTGCTGGCTGAATAAGCCGGCTCTTCTG
AAATGTATTAGCAGATCCTCTTTGTAATTGGTTCCTACTCAACGTGATAAGTATTTCGTTGAGGACATTTCCTTTATTGG
(pipits_env) matt@line:~/Documents/FWS/2019_black_spruce_inventory/data/soil_fungi$ pipits_rereplicate -i out_funits/ITS_denoised.fasta -o out_funits/ITS_denoised_rerep.fasta --uc out_funits/intermediate/derep.uc
Error: Size values in headers added by USEARCH != number of sequences in .uc
65239 3136
(pipits_env) matt@line:~/Documents/FWS/2019_black_spruce_inventory/data/soil_fungi$ ls
2020-04-24-0635_io.txt  out_funits  out_seqprep  pipits_process_ESV  raw_fastq  readpairslist.txt
(pipits_env) matt@line:~/Documents/FWS/2019_black_spruce_inventory/data/soil_fungi$ Trying it out.
Trying: command not found
(pipits_env) matt@line:~/Documents/FWS/2019_black_spruce_inventory/data/soil_fungi$ 
(pipits_env) matt@line:~/Documents/FWS/2019_black_spruce_inventory/data/soil_fungi$ ```shell
> ./pipits_process_ESV -i out_funits/ITS.fasta -o out_process -v -r
> ```

Command 'shell' not found, did you mean:

  command 'spell' from deb spell
  command 'qshell' from deb qtile
  command 'shelr' from deb shelr
  command 'jshell' from deb openjdk-11-jdk-headless
  command 'bshell' from deb avahi-ui-utils
  command 'pshell' from deb python-pyramid

Try: sudo apt install <deb name>

usage: PIPITS_PROCESS: Sequences to OTU Table [-h] -i <FILE> [-o <DIR>]
                                              [-d <FLOAT>] [-c <FLOAT>]
                                              [-l <TXT>] [--includeuniqueseqs]
                                              [-r] [-v] [-t <INT>]
                                              [--Xms <INT>] [--Xmx <INT>]
                                              [--warcup]
                                              [--unite {04.02.2020,02.02.2019,01.12.2017,28.06.2017}]
PIPITS_PROCESS: Sequences to OTU Table: error: unrecognized arguments: 2.5, the PIPITS Project https://github.com/hsgweon/pipits --------------------------------- 2020-04-27 07:08:27 pipits_process started 2020-04-27 07:08:27 Generating a sample list from the input sequences python: can't open file '/home/matt/Documents/FWS/2019_black_spruce_inventory/data/soil_fungi/pipits_getsamplelistfromfasta': [Errno 2] No such file or directory 2020-04-27 07:08:27 Error: None zero returncode: python /home/matt/Documents/FWS/2019_black_spruce_inventory/data/soil_fungi/pipits_getsamplelistfromfasta
(pipits_env) matt@line:~/Documents/FWS/2019_black_spruce_inventory/data/soil_fungi$ 
(pipits_env) matt@line:~/Documents/FWS/2019_black_spruce_inventory/data/soil_fungi$ ./pipits_process_ESV -i out_funits/ITS.fasta -o out_process -v -r
pipits_process 2.5, the PIPITS Project
https://github.com/hsgweon/pipits
---------------------------------

2020-04-27 07:10:02 pipits_process started
2020-04-27 07:10:02 Generating a sample list from the input sequences
python: can't open file '/home/matt/Documents/FWS/2019_black_spruce_inventory/data/soil_fungi/pipits_getsamplelistfromfasta': [Errno 2] No such file or directory
2020-04-27 07:10:02 Error: None zero returncode: python /home/matt/Documents/FWS/2019_black_spruce_inventory/data/soil_fungi/pipits_getsamplelistfromfasta -i out_funits/ITS.fasta -o out_process/sampleIDs.txt
(pipits_env) matt@line:~/Documents/FWS/2019_black_spruce_inventory/data/soil_fungi$ pipits_process_ESV -i out_funits/ITS.fasta -o out_process -v -r
pipits_process 2.5, the PIPITS Project
https://github.com/hsgweon/pipits
---------------------------------

2020-04-27 07:11:12 pipits_process started
2020-04-27 07:11:12 Generating a sample list from the input sequences
2020-04-27 07:11:15 Downloading UNITE trained database, version: 04.02.2020
[###################################################################################################################]100% | 275.1 KiB/s | 131112800 of 131112800 | Time:  0:07:45
File size: 125.04 MB
2020-04-27 07:19:05 ... Unpacking
2020-04-27 07:19:15 ... done
2020-04-27 07:19:15 Downloading WARCUP trained database: 
[#####################################################################################################################]100% | 292.9 KiB/s | 17880783 of 17880783 | Time:  0:00:59
File size: 17.05 MB
2020-04-27 07:20:19 ... Unpacking
2020-04-27 07:20:20 ... done
2020-04-27 07:20:20 Downloading UCHIME database for chimera filtering: 
[#######################################################################################################################]100% | 200.0 KiB/s | 4578703 of 4578703 | Time:  0:00:22
File size: 4.37 MB
2020-04-27 07:20:46 ... Unpacking
2020-04-27 07:20:47 ... done
2020-04-27 07:20:47 Dereplicating and removing unique sequences prior to picking OTUs
vsearch v2.14.2_linux_x86_64, 7.8GB RAM, 2 cores
https://github.com/torognes/vsearch

Dereplicating file out_funits/ITS.fasta 100%
124639323 nt in 704602 seqs, min 101, max 332, avg 177
Sorting 100%
75623 unique sequences, avg cluster 9.3, median 1, max 72255
Writing output file 100%
19852 uniques written, 55771 clusters discarded (73.7%)
2020-04-27 07:20:49 Denoising [VSEARCH]
vsearch v2.14.2_linux_x86_64, 7.8GB RAM, 2 cores
https://github.com/torognes/vsearch

Reading file out_process/intermediate/input_nr.fasta 100%
2138936 nt in 11839 seqs, min 101, max 277, avg 181
minsize 3: 8013 sequences discarded.
Masking 100%
Sorting by abundance 100%
Counting k-mers 100%
Clustering 100%
Sorting clusters 100%
Writing clusters 100%
Clusters: 3413 Size min 1, max 449, avg 3.5
Singletons: 2868, 24.2% of seqs, 84.0% of clusters
2020-04-27 07:20:57 Removing chimeras [VSEARCH]
vsearch v2.14.2_linux_x86_64, 7.8GB RAM, 2 cores
https://github.com/torognes/vsearch

Reading file pipits_db/uchime_reference_dataset_28.06.2017/uchime_reference_dataset_28.06.2017.fasta 100%
16786547 nt in 30555 seqs, min 146, max 2570, avg 549
Masking 100%
Counting k-mers 100%
Creating k-mer index 100%
Detecting chimeras 100%
Found 1069 (31.3%) chimeras, 2327 (68.2%) non-chimeras,
and 17 (0.5%) borderline sequences in 3413 unique sequences.
Taking abundance information into account, this corresponds to
12349 (2.3%) chimeras, 526932 (97.4%) non-chimeras,
and 1902 (0.4%) borderline sequences in 541183 total sequences.
2020-04-27 07:21:19 Renaming OTUs
2020-04-27 07:21:19 Mapping reads onto centroids [VSEARCH]
vsearch v2.14.2_linux_x86_64, 7.8GB RAM, 2 cores
https://github.com/torognes/vsearch

Reading file out_process/intermediate/input_nr_otus_nonchimeras_relabelled.fasta 100%
397037 nt in 2327 seqs, min 102, max 277, avg 171
Masking 100%
Counting k-mers 100%
Creating k-mer index 100%
Searching 100%
Matching unique query sequences: 682959 of 704602 (96.93%)
2020-04-27 07:27:06 Making OTU table
2020-04-27 07:27:11 Converting classic tabular OTU into a BIOM format [BIOM]
2020-04-27 07:27:15 Assigning taxonomy with UNITE [RDP Classifier]
2020-04-27 07:38:25 Reformatting RDP_Classifier output
2020-04-27 07:38:25 Adding assignment to OTU table [BIOM]
2020-04-27 07:38:30 Converting OTU table with taxa assignment into a BIOM format [BIOM]
2020-04-27 07:38:32 Phylotyping OTU table
2020-04-27 07:38:39     Number of reads used to generate OTU table: 682959
2020-04-27 07:38:39     Number of OTUs:                             2327
2020-04-27 07:38:39     Number of phylotypes:                       784
2020-04-27 07:38:39     Number of samples:                          26
2020-04-27 07:38:39 Done - Resulting files are in "out_process" directory
2020-04-27 07:38:39 pipits_process ended successfully.
````

I worked on editing a *Refuge Notebook* article on biotech life for next
week, getting a revised draft back to the author.

Arctos is still down.

Update: Arctos is up as of 12:39\!

I updated content for the Kenai NWR content in update to the [Kenai
Peninsula Cooperative Weed Management Area](http://www.kenaiweeds.org/)
strategic plan.

I installed a qiime2 environment in preparation for processing the
bacterial 16S data from the 2019 black spruce project.

## Tuesday, April 28

Arctos is down again. I was going to work on uploading LTEMP bird and
veg data.

I formatted and posted the *Refuge Notebook* articles from January 31
and February 7.

## Wednesday, April 29

To do:

  - ~~Submit a *Refuge Notebook* article for this week.~~

Arctos is up now, but with a restored data backup from the night of
March 15. Media are still uploaded from mid April, but links to these
media will need to be remade if possible.

I checked:

Specimen record
[KNWR:Ento:11382](http://arctos.database.museum/guid/KNWR:Ento:11382)
exists in Arctos; Specimen record
[KNWR:Ento:11383](http://arctos.database.museum/guid/KNWR:Ento:11383)
does not. I had entered data through
[KNWR:Ento:11385](http://arctos.database.museum/guid/KNWR:Ento:11385).
KNWR Invertebrate Collecion records
[KNWR:Inv:46](http://arctos.database.museum/guid/KNWR:Inv:46)–[KNWR:Inv:53](http://arctos.database.museum/guid/KNWR:Inv:53)
also no longer exist. This is frustrating.

I worked on getting Jake set up for updating the Refuge’s checklist on
FWSpecies.

I edited this week’s *Refuge Notebook* article and sent it to the
*Clarion*.

Thankfully, it looks like Arctos had uploaded data to GBIF on March 23.

| Arctos\_GUID                                                          |                                         GBIF\_occurrence |
| :-------------------------------------------------------------------- | -------------------------------------------------------: |
| [KNWR:Ento:11383](http://arctos.database.museum/guid/KNWR:Ento:11383) | [2592190565](https://www.gbif.org/occurrence/2592190565) |
| [KNWR:Ento:11384](http://arctos.database.museum/guid/KNWR:Ento:11384) | [2592190569](https://www.gbif.org/occurrence/2592190569) |
| [KNWR:Inv:46](http://arctos.database.museum/guid/KNWR:Inv:46)         | [2592190546](https://www.gbif.org/occurrence/2592190546) |
| [KNWR:Inv:47](http://arctos.database.museum/guid/KNWR:Inv:47)         | [2592190550](https://www.gbif.org/occurrence/2592190550) |
| [KNWR:Inv:48](http://arctos.database.museum/guid/KNWR:Inv:48)         | [2592190549](https://www.gbif.org/occurrence/2592190549) |
| [KNWR:Inv:49](http://arctos.database.museum/guid/KNWR:Inv:49)         | [2592190554](https://www.gbif.org/occurrence/2592190554) |
| [KNWR:Inv:50](http://arctos.database.museum/guid/KNWR:Inv:50)         | [2592190553](https://www.gbif.org/occurrence/2592190553) |
| [KNWR:Inv:51](http://arctos.database.museum/guid/KNWR:Inv:51)         | [2592190548](https://www.gbif.org/occurrence/2592190548) |
| [KNWR:Inv:52](http://arctos.database.museum/guid/KNWR:Inv:52)         | [2592190552](https://www.gbif.org/occurrence/2592190552) |

There appears to be no record for
[KNWR:Ento:11385](http://arctos.database.museum/guid/KNWR:Ento:11385) on
GBIF. I should be able to figure out which sample this was by the
process of elimination.

I met with Jake over the phone about going through our Refuge’s
checklist to move species from our old checklist to FWSpecies.

I received a request to make my old bristletail key available, so I
posted it as a preprint via ResearchGate (Bowser
[2012](#ref-bowser_key_2012)).

I was able to make a search to download the GBIF version of the lost
data (GBIF.org
[2020](#ref-gbiforg_gbif_2020-4)[a](#ref-gbiforg_gbif_2020-4)).

## Thursday, April 30

To do:

  - Intra-Service Section 7 Biological Evaluation Form

I worked on the Intra-Service Section 7 Biological Evaluation Form for
the Sandpiper Lake *Elodea* erradication project.

### I\&M call at 11:30

  - Discussed possibility adding to FIA’s Interior inventories.
  - Status of Kenai NWR Resources of Concern process: not much happening
    right now.

# May

## Friday, May 1

To do:

  - Intra-Service Section 7 Biological Evaluation Form

Responding to comments on iNaturalist
(<https://www.inaturalist.org/observations/3815139>), I looked into
*Dilophus femoratus* Meigen, 1804. From Skartveit
([2002](#ref-skartveit_larvae_2002)):

> Eurytopical species, most common on flower-rich meadows. Common up to
> the subalpine zone. Full-grown larvae found in March to June.

Interestingly, the Alaskan material identified as *D. femoratus* is
different than European *D. femoratus*. The two BINs for these are not
nearest neighbors based on BOLD’s BIN algorithm. This deserves looking
into.

BIN including Alaskan material:  
<http://bins.boldsystems.org/index.php/Public_BarcodeCluster?clusteruri=BOLD:AAP6925>

BIN including Palearctic D. femoratus:  
<http://bins.boldsystems.org/index.php/Public_BarcodeCluster?clusteruri=BOLD:AAG6648>

I found old correspondence and updated identifications of all biobionid
specimens identified by John Skartveit in 2006.

I downloaded all *Dilophus* COI sequences and one *Bibio* sequence
(outgroup) and submitted these to NGPhylogeny.fr
(<https://ngphylogeny.fr/workspace/history/30d01ce00572c3b7>).

I worked on the Intra-Service Section 7 Biological Evaluation Form for
the Sandpiper Lake *Elodea* eradication project.

## Monday, May 4

To do:

  - Intra-Service Section 7 Biological Evaluation Form

I downloaded Kittlitz’s murrelet (GBIF.org
[2020](#ref-gbiforg_gbif_2020-5)[b](#ref-gbiforg_gbif_2020-5)) and
beluga (GBIF.org
[2020](#ref-gbiforg_gbif_2020-6)[c](#ref-gbiforg_gbif_2020-6))
occurrence records from GBIF and worked on mapping these and related
data for the Intra-Service Section 7 Biological Evaluation Form.

Sandpiper Lake is 59 km from the nearest occurrence of Kittlitz’s
murrelet, 19 miles from the nearest potential nesting habitat for
Kittlitz’s murrelet, 9.4 km from the nearest occurrence of belugas, and
1.4 miles from beluga critical habitat.

![Map of Cook Inlet Beluga occurrences and Cook Inlet Beluga Critical
Habitat in relation to Sandpiper
Lake.](2020-05-04-1550_beluga_map.jpg)  
Map of Cook Inlet Beluga occurrences and Cook Inlet Beluga Critical
Habitat in relation to Sandpiper Lake. Occurrence data are from GBIF.org
([2020](#ref-gbiforg_gbif_2020-6)[c](#ref-gbiforg_gbif_2020-6)) and
Critical Habitat areas are from NOAA Fisheries
([2019](#ref-NOAA_beluga_habitat)).

![Map of Kittlitz’s Murrelet occurrences and Kittlitz’s Murrelet
potential nesting habitat in relation to Sandpiper
Lake.](2020-05-04-1550_murrelet_map.jpg)  
Map of Kittlitz’s Murrelet occurrences and Kittlitz’s Murrelet potential
nesting habitat in relation to Sandpiper Lake. Occurrence data are from
GBIF.org ([2020](#ref-gbiforg_gbif_2020-5)[b](#ref-gbiforg_gbif_2020-5))
and potential nesting areas are from Felis et
al. ([2016](#ref-Felis_et_al_2016)).

## Tuesday, May 5

To do:

  - Intra-Service Section 7 Biological Evaluation Form

I worked on the Intra-Service Section 7 Biological Evaluation Form for
the Sandpiper Lake project. I downloaded occurrences from Sandpiper Lake
(GBIF.org
[2020](#ref-gbiforg_gbif_2020-7_Sandpiper)[d](#ref-gbiforg_gbif_2020-7_Sandpiper)).

### Biology staff meeting at 11:00

I happened to notice that the locations seem to be wrong on specimens
[KNWR:Herb:1480](http://arctos.database.museum/guid/KNWR:Herb:1480)-[KNWR:Herb:1482](http://arctos.database.museum/guid/KNWR:Herb:1482).
I need to check the labels on these specimens.

### Reed canary grass meeting at 11:30

Getting a species list from Sandpiper Lake.

``` r
sp1 <- read.delim("0056501-200221144449610/0056501-200221144449610.csv")

plot(sp1$decimalLongitude, sp1$decimalLatitude)

su1 <- unique(sp1[,c("kingdom", "phylum", "class", "order", "family", "genus", "species", "scientificName")])

write.csv(su1, "2020-05-05-1337_Sandpiper_Lake_species.csv", row.names=FALSE)
```

I finished the Intra-Service Section 7 Biological Evaluation Form for
the Sandpiper Lake elodea eradication project and sent it off.

## Wednesday, May 6

### Safety committee meeting at 09:00

  - We are having trouble acquiring hand sanitizer. I may make some in
    the lab.
  - I will be responsible for delivering a blood borne pathogens talk
    via zoom.

I walked Jake through the process of moving species from our spreadsheet
onto FWSpecies and providing evidence.

## Thursday, May 7

To do:

  - Review Sandpiper Lake FONSI.
  - Finish *AKES Newsletter*.

I took care of correspondence, worked with Jake some on getting records
moved to FWSpecies, got this week’s *Refuge Notebook* submitted, etc.

## Friday, May 8

To do:

  - Review Sandpiper Lake FONSI.
  - Finish *AKES Newsletter*.

## Monday, May 11

To do:

  - ~~Review Sandpiper Lake FONSI.~~
  - ~~Take pesticide training.~~
  - Finish *AKES Newsletter*.
  - Improve Kenai NWR checklist processing/formatting.
  - ~~Examine *Rabodophaga* specimens~~.
  - Slikok occupancy.
  - Finish 2019 black spruce project.
  - Clean up stuff on Yeti.

I reviewed the Sandpiper Lake *Elodea* eradication FONSI document.

I formatted a UAM specimen count article for the *AKES Newsletter*. This
one took a little while.

I took the pesticide applicator training that I need.

Examining rosette galls from *Salix fuscescens* that I collected
yesterday.

2020-05-10\_MLB01: This contained a single hymenopteran larva.

2020-05-10\_MLB02: This contained the gooey remains of some insect
larva.

Examining rosette galls I had collected from *Salix fuscescens* from our
swamp on April 24.

2020-04-24\_MLB01: This gall contained a hymenopteran pupa.

2020-04-24\_MLB02: Two sciaroid flies had emerged. These I put into
vials BOLD-1H8 and BOLD-LB2.

## Tuesday, May 12

To do:

  - Finish *AKES Newsletter*.
  - ~~Take pesticide applicator training test.~~
  - Edit this week’s *Refuge Notebook* article.
  - Prepare blood borne pathogens presentation.
  - First aid / CPR training.
  - Improve Kenai NWR checklist processing/formatting.
  - Enter data for *Rabdophaga* specimens.
  - Slikok occupancy.
  - Finish 2019 black spruce project.
  - Clean up stuff on Yeti.

I took the final exam for the UAF CES class I was taking for pesticide
applicator CEUs.

### 10:00 Swan Lake Fire invasives/burn severity meeting.

  - I need to get GIS files (trailheads, etc. to Justin) and locations
    of exotic species to Justin.

  - I need to get Kenai NWR contacts to Justin.

  - We need to come up with a sampling design for burn severity work for
    this summer.

I applied edits to the specimen count article for the *AKES Newsletter*.

### 14:00 meeting with Hillcorp regarding non-native plant management.

  - They have been making good progress reducing white sweetclover, reed
    canary grass, and hawkweeds.

I started formatting the *AKES Newsletter* article on *Bombus*.

## Wednesday, May 13

To do:

  - Finish *AKES Newsletter*.
  - ~~Edit and submit this week’s *Refuge Notebook* article.~~
  - Prepare blood borne pathogens presentation.
  - First aid / CPR training.
  - Improve Kenai NWR checklist processing/formatting.
  - ~~Get list of new Kenai NWR species records from Slikok project to
    Jake?~~
  - Enter data for *Rabdophaga* specimens.
  - Slikok occupancy.
  - Finish 2019 black spruce project.
  - Clean up stuff on Yeti.

I edited and submitted this weeks’ *Refuge Notebook* article.

I worked on updating the [Kenai National Wildlife Refuge species
list](https://github.com/mlbowser/KenaiNWRspecies) based on Jake’s
recent updates and preparing the list of additional records from Bowser
et al. ([2020](#ref-bowser_et_al_2020)).

``` r
## Get new records from Slikok project dataset.

## First load data.
data1 <- assemble_csvs(directory="../data/FWSpecies")
fields_crosswalk <- read.csv("../data/field_name_crosswalk.csv", colClasses="character")
establishmentMeans_crosswalk <- read.csv("../data/establishmentMeans_crosswalk.csv", colClasses="character")
fillin <- read.csv("../data/taxonomy_fill-in.csv", colClasses="character")

## Renaming fields.
for (this_field in 1:nrow(fields_crosswalk))
 {
 sl <- which(names(data1) == fields_crosswalk$FWSpecies_field[this_field])
 names(data1)[sl] <- fields_crosswalk$DwC_field[this_field]
 }
 
## Filling in some missing values.
data1$scientificName[data1$scientificName == ""] <- data1$SciName[data1$scientificName == ""]
 
## Now load Slikok project records.
slikok1 <- read.csv("../data/2020-05-13-0800_Slikok_project_download.csv", colClasses="character")

## Get the list of names from the Slikok project that are not yet in FWSpecies.
fwspecies_names <- levels(as.factor(data1$SciName))

slikok_names <- levels(as.factor(slikok1$SCIENTIFIC_NAME))

new_names <- setdiff(slikok_names, fwspecies_names)

## How many new names?
length(new_names)
[1] 974 ## A lot of these are provisional names, though.

## How many were already known?
length(intersect(slikok_names, fwspecies_names))
[1] 339

## Now pull representative records from the Slikok project.
slikok_new <- slikok1[slikok1$SCIENTIFIC_NAME %in% new_names,]

## Now select representative records.
slikok_selected <- slikok_new[rep(1, length(new_names)),]
for (this_name in 1:length(new_names))
 {
 this_sp_records <- which(slikok_new$SCIENTIFIC_NAME == new_names[this_name])
 print(this_sp_records)
 if(length(this_sp_records)==1)
  {
  selected_record <- this_sp_records
  }
 else
  {
  selected_record <- sample(this_sp_records, 1)
  }
 print(selected_record)
 if (this_name == 1)
  {
  slikok_selected <- slikok_new[selected_record,]
  }
 else
  {
  slikok_selected <- rbind(slikok_selected, slikok_new[selected_record,])
  }
 }

## Now load the file of records to add to FWSpecies so that we can append these new ones.
additions1 <- read.csv("../data/non_FWSpecies_records.csv")
dim(additions1)
[1] 560  34

## Modify column names.
names(additions1) <- gsub("\\.", "_", names(additions1))
names(additions1) <- gsub("__", "_", names(additions1))

## Check to see that none of these "new" names were not already among the additions.

add_names <- levels(as.factor(additions1$Scientific_Name))

already_listed_to_add <- intersect(add_names, new_names)

## How many were already on the list?
length(already_listed_to_add)
[1] 19

## Removing these.
slikok_selected <- slikok_selected[!(slikok_selected$SCIENTIFIC_NAME %in% already_listed_to_add),]
dim(slikok_selected)
[1] 955  12

## Adding URLs.
slikok_selected$evidence_URL <- paste0("http://arctos.database.museum/guid/", slikok_selected$GUID)

## appending.
additions_new <- additions1[rep(1,nrow(slikok_selected)),]
additions_new[,] <- ""
additions_new$Scientific_Name <- slikok_selected$SCIENTIFIC_NAME
additions_new$evidence_URL <- slikok_selected$evidence_URL
additions_new$Valid_in_Gbif_refuge_YY_NN_ <- "?Y"
additions1$evidence_URL <- NA
additions2 <- rbind(additions1, additions_new)
dim(additions2)
[1] 1515   34

## Now dropping records that have already been transferred to FWSpecies.
levels(as.factor(additions2$transferred_to_FWSpecies_date))
to_drop <- grepl("2020", additions2$transferred_to_FWSpecies_date)
sum(to_drop)
[1] 190 ## So Jake got 190 records transferred (actually 189. I had transferred one.). He has done some good work.

additions2 <- additions2[!to_drop,]
dim(additions2)
[1] 1325   34

## Ok, overwriting the old addtions file.
write.csv(additions2, "../data/non_FWSpecies_records.csv", row.names=FALSE)
```

I continued working on formatting the *Bombus* article for the *AKES
Newsletter*.

## Thursday, May 14

To do:

  - Finish *AKES Newsletter*.
  - Prepare blood borne pathogens presentation.
  - First aid / CPR training.
  - Improve Kenai NWR checklist processing/formatting.
  - Enter data for *Rabdophaga* specimens.
  - Slikok occupancy.
  - Finish 2019 black spruce project.
  - Clean up stuff on Yeti.

I continued working on formatting the *Bombus* article for the *AKES
Newsletter*, finishing this at the end of the day. I sent a draft of
this article to the authors for review and I sent the whole issue out
the editorial committee for review.

I added *Deschampsia beringensis* to the Refuge’s species list based on
UAM Herbarium specimens.

## Friday, May 15

To do:

  - Finish *AKES Newsletter*.
  - ~~Prepare blood borne pathogens presentation.~~
  - First aid / CPR training.
  - Improve Kenai NWR checklist processing/formatting.
  - Enter data for *Rabdophaga* specimens.
  - Slikok occupancy.
  - Finish 2019 black spruce project.
  - Clean up stuff on Yeti.

I drove to the office this morning to take care of an administrative
issue.

I made corrections to the *Bombus* article for the *AKES Newsletter*.

I worked on my bloodborne pathogens presentation for next week, getting
it revised and submitted.

## Saturday, May 16

Ethan, Esther, and I collected morels for DNA sequencing to determine
what species these are.

| field ID          |                                                iNaturalist ID |
| :---------------- | ------------------------------------------------------------: |
| 2020-05-16\_MLB01 | [46129997](https://www.inaturalist.org/observations/46129997) |
| 2020-05-16\_MLB02 | [46132276](https://www.inaturalist.org/observations/46132276) |

Both of these were collected under cottonwoods.

![*Morchella* specimen 2020-05-16\_MLB01 (iNaturalist:
[46129997](https://www.inaturalist.org/observations/46129997)) from
Kasilof.](2020-05-16_MLB01.jpg)  
*Morchella* specimen 2020-05-16\_MLB01 (iNaturalist:
[46129997](https://www.inaturalist.org/observations/46129997)) from
Kasilof.

## Sunday, May 17

Ethan, Esther, Apphia, and I collected morels for DNA sequencing from
near Kenai National Wildlife Refuge headquarters.

| field ID          |                                                iNaturalist ID |
| :---------------- | ------------------------------------------------------------: |
| 2020-05-17\_MLB01 | [46292862](https://www.inaturalist.org/observations/46292862) |
| 2020-05-17\_MLB02 | [46293272](https://www.inaturalist.org/observations/46293272) |

Both of these were collected under cottonwoods.

![*Morchella* specimen 2020-05-17\_MLB01 (iNaturalist:
[46292862](https://www.inaturalist.org/observations/46292862)) from near
Kenai National Wildlife Refuge headquarters.](2020-05-17_MLB01.jpg)  
*Morchella* specimen 2020-05-17\_MLB01 (iNaturalist:
[46292862](https://www.inaturalist.org/observations/46292862)) from near
Kenai National Wildlife Refuge headquarters.

## Monday, May 18

To do:

  - Finish *AKES Newsletter*.
  - First aid / CPR training.
  - Improve Kenai NWR checklist processing/formatting.
  - ~~Enter data for *Rabdophaga* specimens.~~
  - Slikok occupancy.
  - Finish 2019 black spruce project.
  - Clean up stuff on Yeti.

I added *Geopyxis carbonaria* to the Kenai National Wildlife Refuge’s
checklist based on iNaturalist observation
[46393003](https://www.inaturalist.org/observations/46393003).

I have been corresponding quite a bit regarding grasses to be used for
revegation along a power line right of way.

I entered data from a couple of recent LifeScanner specimens into
lifescanner.net.

I collected a rosette gall on *Salix fuscescens* (2020-05-18\_MLB01,
iNatualist:
[46407797](https://www.inaturalist.org/observations/46407797)) and
dissected it. Inside the gall were a hymenopteran larva and an egg on
the outside of the larva. I attempted to collect the egg, but I dropped
it and could not find it. The larva I placed in LifeScanner vial
BOLD-JO2.

![Hymenoptera larva and egg inside rosette gall on *Salix fuscescens*
(2020-05-18\_MLB01, iNatualist:
[46407797](https://www.inaturalist.org/observations/46407797))](2020-05-18_MLB01_Hymenoptera.jpg)  
Hymenoptera larva and egg inside rosette gall on *Salix fuscescens*
(2020-05-18\_MLB01, iNatualist:
[46407797](https://www.inaturalist.org/observations/46407797))

A second rosette gall on *Salix fuscescens* (2020-05-18\_MLB02,
iNatualist:
[46413661](https://www.inaturalist.org/observations/46413661)), had a
minute larva next to an old Lepidoptera head and thorax. This was among
the outer leaves of the rosette. I collected this larva and placed it
into vial BOLD-I87. In the middle of the rosette was a hymenopteran
larva, which I did not retain.

I collected two more rosette galls. One contained a hymenopteran larva
and the other was empty.

As I am collecting more I should put results into a table.

| contents                                      | count |
| :-------------------------------------------- | ----: |
| Hymenoptera larva                             |     3 |
| Hymenoptera pupa                              |     3 |
| Hymenoptera larva and egg                     |     1 |
| empty                                         |     3 |
| 2 Hymenoptera larvae, one large and one small |     1 |

One of these included a Hymenoptera larva and what might be a
hyperparasitoid (2020-05-18\_MLB03, iNaturalist:
[46433634](https://www.inaturalist.org/observations/46433634)). The
smaller larva that might be a hyperparasitoid I put into LifeScanner
vial BOLD-G57. The larger larva I put into LifeScanner vial BOLD-UJ8.

I might be too late. The *Rabdophaga rosaria* group midges may already
be on the wing. This is frustrating because the first gall on *Salix
barclayi* that I looked at still had a *Rabdophaga* larva in it.

## Tuesday, May 19

To do:

  - ~~Edit this week’s *Refuge Notebook* article.~~
  - ~~Check out new Sruvey123 form.~~
  - Finish *AKES Newsletter*.
  - First aid / CPR training.
  - Improve Kenai NWR checklist processing/formatting.
  - Slikok occupancy.
  - Finish 2019 black spruce project.
  - Clean up stuff on Yeti.

I edited and submitted this week’s *Refuge Notebook* article.

I downloaded, used, and reviewed a Survey123 form designed by Amanda for
documenting non-native plant occurrence, recommending a number of
changes. Most significantly, I think indicated how I think we should be
integrating pesticide use reporting with the form.

I started the American Heart Association First Aid CPR AED Online
training, but I did not finish it.

## Wednesday, May 20

To do:

  - Finish *AKES Newsletter*.
  - Get non-native plant data to AKNHP folks.
  - Get burn severity plot data to Lisa.
  - ~~First aid / CPR training.~~
  - Test new version of Survey123 form.
  - Improve Kenai NWR checklist processing/formatting.
  - Slikok occupancy.
  - Finish 2019 black spruce project.
  - Clean up stuff on Yeti.

I completed the on-line first aid training.

I drove to the Refuge for the 1:00 pm hand-on first aid training.

## Thursday, May 21

To do:

  - ~~Presentation~~
  - ~~mid-year appraisal~~
  - terrestrial IPM plan to Ben
  - ~~non-native species list to Amanda.~~
  - Get non-native plant data to AKNHP folks.
  - Get burn severity plot data to Lisa.
  - ~~Test new version of Survey123 form.~~
  - Improve Kenai NWR checklist processing/formatting.
  - Slikok occupancy.
  - Finish 2019 black spruce project.
  - Clean up stuff on Yeti.

I gave my presentation at the seasonal employee training zoom meeting.

I filled out my part of my mid-year appraisal in DOI Talent.

I downloaded non-native plant records from a polygon covering the Kenai
Peninsula.

``` r
## Getting a species list for our Survey123 form...
data1 <- read.csv("akepic.f8a5bcdd-1937-4de1-aae3-d7f21e7f2cac.csv", stringsAsFactors=FALSE)

species_list <- levels(as.factor(data1$scientific_name))

length(species_list)
[1] 125
 
## There were some wierd special characters.
species_list <- gsub("Â ", " ", species_list)

write.csv(species_list,
 "2020-05-21-1540_AKEPIC_KP_species_list.csv",
 row.names=FALSE
 )

```

## Tuesday, May 26

To do:

  - Write this week’s *Refuge Notebook* article.
  - Revise aquatic herbicide PUPs.
  - Get shortened list of non-native plant species to Amanda.
  - Get non-native plant data to AKNHP folks.
  - Get burn severity plot data to Lisa.
  - Improve Kenai NWR checklist processing/formatting.
  - Slikok occupancy.
  - Finish 2019 black spruce project.
  - Clean up stuff on Yeti.

I started a *Refuge Notebook* article on morels.

Article notes:

There is apparently only a single species of morel identified from the
Kenai Peninsula: *Morchella tomentosa* (gray morel) from the Crescent
Lake Trail viscinity (iNaturalist:
[17941556](https://www.inaturalist.org/observations/17941556)).

I looked through the identifications of fungi from the 2019 black spruce
inventory. None of these were *Morchella*.

The unidentified *Morchella* from Interior Alaska in Stefani et
al. ([2010](#ref-stefani_morchella_2010)) (Herbarium ID: QFB 8581;
GenBank accessions:
[GQ304982](https://www.ncbi.nlm.nih.gov/nuccore/GQ304982)-[GQ304986](https://www.ncbi.nlm.nih.gov/nuccore/GQ304986))
appears to be *Morchella exuberans* based on NCBI BLAST searches. Like
*Morchella tomentosa*, this is another post-fire species (Richard et al.
[2015](#ref-richard_true_2015)).

I was unable to revise the aquatic herbicide PUPs. They appear to be
locked.

I examined the morels that Colin had collected on May 19. Three had dark
ridges that were not tomentose; a fourth appeared to be *Morchella
tomentosa*.

![Detail of tomentose ridges of a morel specimen with iNaturalist
observation ID 47411794, apparently *Morchella
tomentosa*.](2020-05-26-1402_iNat-47411794.jpg)  
Detail of tomentose ridges of a morel specimen with iNaturalist
observation ID
[47411794](https://www.inaturalist.org/observations/47411794),
apparently *Morchella tomentosa*.

Ethan, Apphia, Esther, and I drove to the pull-off across from Katmai
Avenue to look for morels. Here we found two morels (2020-05-26\_MLB01
and 2020-05-26\_MLB02) at the edge of the gravel pad under small spruce,
birches, short cottonwoods, and grass.

## Wednesday, May 27

To do:

  - Write this week’s *Refuge Notebook* article.
  - Revise aquatic herbicide PUPs.
  - ~~Get shortened list of non-native plant species to Amanda.~~
  - Get non-native plant data to AKNHP folks.
  - Get burn severity plot data to Lisa.
  - Improve Kenai NWR checklist processing/formatting.
  - Slikok occupancy.
  - Finish 2019 black spruce project.
  - Clean up stuff on Yeti.

I drove in and worked at the office this morning just to check in.

I got a list of non-native plant species to Amanda.

Steve Miller gave me two specimens of morels he had collected
(iNaturalist observation IDs:
[47524671](https://www.inaturalist.org/observations/47524671) and
[47524963](https://www.inaturalist.org/observations/47524963)).

I worked on the *Refuge Notebook* article due today.

I added *Morchella tomentosa* to the Refuge’s checklist.

## Thursday, May 28

To do:

  - ~~Submit this week’s *Refuge Notebook* article.~~
  - Revise aquatic herbicide PUPs.
  - ~~Get non-native plant data to AKNHP folks.~~
  - ~~Get burn severity plot data to Lisa.~~
  - Improve Kenai NWR checklist processing/formatting.
  - Slikok occupancy.
  - Finish 2019 black spruce project.
  - Clean up stuff on Yeti.

I finished revising and submitted my *Refuge Notebook* article on morels
this morning.

I mapped FIA points within the Swan Lake fire and near roads for Lisa.

![Map of FIA points that are within the Swan Lake Fire perimeter and
that are within 1 mile from
roads.](2020-05-28-1132_Swan_Lake_Fire_LTEMP.jpg)  
Map of FIA points that are within the Swan Lake Fire perimeter and that
are within 1 mile from roads.

For extracting non-native plant data I obtained the extent of the Swan
Lake Fire perimeter in WGS84 coordinates:

I downloaded data from GBIF (GBIF.org
[2020](#ref-gbiforg_gbif_2020-8_Swan_Lake_Fire_plants)[e](#ref-gbiforg_gbif_2020-8_Swan_Lake_Fire_plants))
and from iNaturalist.org
(<https://www.inaturalist.org/observations?nelat=60.728891&nelng=-149.854865&place_id=any&swlat=60.430024&swlng=-150.529237&taxon_id=47126>),
extracted records within the fire perimeter, dropped the iNaturalist
records from the GBIF dataset, and send these data to Justin.

``` r
## Now I am working on improving checklist output.

source("make_DWC-A.R")

sl <- dwc1[,3]=="" | 
 dwc1[,4]=="" | 
 dwc1[,5]=="" | 
 dwc1[,6]=="" | 
 dwc1[,7]==""  

## Saving this to manually edit. 
write.csv(dwc1[sl,], "../data/taxonomy_fill-ins.csv", row.names=FALSE)
```

I updated the
[make\_DWC-A.R](https://github.com/mlbowser/KenaiNWRspecies/blob/master/scripts/make_DwC-A.R)
script to incorporate the edited data. The
[checklist](https://github.com/mlbowser/KenaiNWRspecies/blob/master/text/checklist.md)
now looks much better. It now includes 1,868 species.

## Friday, May 29

To do:

  - Revise aquatic herbicide PUPs.
  - Improve Kenai NWR checklist processing/formatting.
  - Slikok occupancy.
  - Finish 2019 black spruce project.
  - Clean up stuff on Yeti.

I am checking through the recent Kenai NWR checklist draft DwC-A file
using GBIF’s validator (<https://www.gbif.org/tools/data-validator>).
For some reason 5% of the taxa are missing genus names. Also, 189 out of
1868 records were flagged as “Distribution invalid”, but I could not see
what was different about these records.

``` r
## Filling in missing genus values.

source("make_DWC-A.R")

sl <- dwc1$genus=="" 
spt <- strsplit(dwc1$scientificName[sl], " ")
genus <- sapply(spt, "[", 1)
dwc1_add <- dwc1[sl,]
dwc1_add$genus <- genus
fillin <- rbind(fillin, dwc1_add)
write.csv(fillin, "../data/taxonomy_fill-ins.csv", row.names=FALSE)
```

This still had the 189 distribution invalid records. I do not know what
is going on here.

I worked on improving output formatting of the checlist.

Later, Ethan, Apphia, and I drove to Cohoe Beach to look for morels. I
had read that there are some special morel species that grow on dunes
and in grasslands, so I wanted to check the best place I know locally
for these kinds of situations. We looked in the dunes and around spruces
that are growing out in the dunes, but we found no morels. The dunes
have recovered quite a bit since the fence was but in and vehicles have
been kept off of the dunes.

# June

## Monday, June 1

To do:

  - ~~Edit this week’s *Refuge Notebook* article.~~
  - ~~Get the *AKES Newsletter* out.~~
  - *Refuge Notebook* catch-up.
  - Revise aquatic herbicide PUPs.
  - Improve Kenai NWR checklist processing/formatting.
  - Slikok occupancy.
  - Finish 2019 black spruce project.
  - Clean up stuff on Yeti.

I edited a draft of this week’s *Refuge Notebook* article.

I worked on applying edits to articles in the *AKES Newsletter*

AKES\_newsletter\_2020\_n1\_a01.pdf

I spent some time getting [this issue of the *AKES
Newsletter*](http://www.akentsoc.org/newsletter-v13-n1) posted, getting
all of the DOIs set up, etc.

I worked on merging Jake’s work in FWSpecies and generating new DwC-A
and text files, getting a draft done.

## Tuesday, June 2

To do:

  - Fill out a USFWS Alaska Region 2020 field project go/no go screening
    matrix and individual project safety plan for the Swan Lake Fire
    burn severity and vegetation response project.
  - Submit this week’s *Refuge Notebook* article.
  - *Refuge Notebook* catch-up.
  - Revise aquatic herbicide PUPs.
  - Improve Kenai NWR checklist processing/formatting.
  - Slikok occupancy.
  - Finish 2019 black spruce project.

I started completing a field project plan for the Swan Lake Fire burn
severity and vegetation response project that is now required under the
current COVID-19 situation.

``` r
Filling in taxonomy again.

source("make_DWC-A.R")

sl <- dwc1[,3]=="" | 
 dwc1[,4]=="" | 
 dwc1[,5]=="" | 
 dwc1[,6]=="" | 
 dwc1[,7]=="" |
 dwc1[,8]==""

## Filling in genera.
slg <- dwc1$genus == ""
spt <- strsplit(dwc1$scientificName[slg], " ")
genus <- sapply(spt, "[", 1)
dwc1$genus[slg] <- genus
 
## Saving this to manually edit. 
write.csv(dwc1[sl,], "../data/taxonomy_fill-ins_new.csv", row.names=FALSE)
```

I see that some of the distribution information is missing for some
species.

``` r
## Generate a distribution file.
dist1 <- data1[,c(
 "ID",
 "locality",
 "occurrenceStatus",
 "Origin"
 )]

 head(dist1)
## The species with ID 654758 has a missing Origin value of "".

levels(as.factor(dist1$Origin))
[1] ""           "Native"     "Non-native"

missing_origin <- data1[data1$Origin=="" & data1$taxonRank=="species",]
nrow(missing_origin)
[1] 458 ## That is a lot.

## Saving this to deal with it. 
write.csv(missing_origin, "../data/missing_origin.csv", row.names=FALSE)
```

I worked on generating pdf output of the checklist, getting a first
iteration done.

Later I received two independent reports of thousands of caterpillars
off of the Marsh Lake Trail. I looked through caterpillar images. I
think they are noctuids, maybe Hadeninae.

## Wednesday, June 3

To do:

  - Fill out a USFWS Alaska Region 2020 field project go/no go screening
    matrix and individual project safety plan for the Swan Lake Fire
    burn severity and vegetation response project.
  - ~~Submit this week’s *Refuge Notebook* article.~~
  - *Refuge Notebook* catch-up.
  - Revise aquatic herbicide PUPs.
  - Improve Kenai NWR checklist processing/formatting.
  - Slikok occupancy.
  - Finish 2019 black spruce project.

I made final edits on this week’s *Refuge Notebook* article and got it
sent in.

I worked on adding origin values for species in the checklist for which
this value was missing, starting at the bottom of the list with species
starting with Z, but then FWSpecies’ editing interface apparently went
down when I was trying to edit *Stereocaulon alpinum*.

In the afternoon I drove in to the office, then to the Snowshoe Gun Club
for bear firearms training, where I qualified.

I received some of the caterpillars collected from the Marsh Lake Trail.

## Thursday, June 4

To do:

  - Fill out a USFWS Alaska Region 2020 field project go/no go screening
    matrix and individual project safety plan for the Swan Lake Fire
    burn severity and vegetation response project.
  - *Refuge Notebook* catch-up.
  - Revise aquatic herbicide PUPs.
  - Improve Kenai NWR checklist processing/formatting.
  - Slikok occupancy.
  - Finish 2019 black spruce project.

### 09:00 KPWCMA meeting

During the meeting I documented some reed canary grass on state land
(iNaturalist:
[48456585](https://www.inaturalist.org/observations/48456585)).

I photographed the Marsh Lake Trail caterpillars and posted photos (
BugGuide.Net: [1826667](https://bugguide.net/node/view/1826667);
iNaturalist:
[48469943](https://www.inaturalist.org/observations/48469943)).

![Noctuid larva from Marsh Lake Trail.](2020-06-04_noctuid.jpg)  
Noctuid larva from Marsh Lake Trail.

I reviewed and edited the Miller Creek Drainage Restoration Northern
Pike Eradication Treatment Plan and got it back to Rob.

I started a USFWS Alaska Region 2020 Field Project Go/No Go Screening
Matrix and Individual Project Safety Plan for the Sandpiper Lake elodea
eradication project.

## Friday, June 5

To do:

  - Fill out a USFWS Alaska Region 2020 field project go/no go screening
    matrix and individual project safety plan for the Sandpiper Lake
    elodea eradication project.
  - *Refuge Notebook* catch-up.
  - Revise aquatic herbicide PUPs.
  - Improve Kenai NWR checklist processing/formatting.
  - Slikok occupancy.
  - Finish 2019 black spruce project.

I worked on the USFWS Alaska Region 2020 Field Project Go/No Go
Screening Matrix and Individual Project Safety Plan for the Sandpiper
Lake elodea eradication project.

### 10:00 meeting regarding the Swan Lake Fire burn severity project.

I finished and sent off a draft USFWS Alaska Region 2020 field project
go/no go screening matrix and individual project safety plan for the
Swan Lake Fire burn severity and vegetation response project.

I worked on the same kind of document for the elodea project and I also
started on the elodea project aviation safety plan.

## Monday, June 8

To do:

  - Fill out a USFWS Alaska Region 2020 field project go/no go screening
    matrix and individual project safety plan for the Sandpiper Lake
    elodea eradication project.
  - *Refuge Notebook* catch-up.
  - Revise aquatic herbicide PUPs.
  - Improve Kenai NWR checklist processing/formatting.
  - Publish LTEMP bird and plant data.
  - Slikok occupancy.
  - Finish 2019 black spruce project.

I made a map for the Sandpiper Lake Project Aviation Safety Plan.

![Sandpiper Lake elodea eradication project aviation
map.](2020-06-08_Sandpiper_Lake_aviation_map.jpg)  
Sandpiper Lake elodea eradication project aviation map.

### 11:00 Biology staff meeting

Steve Swenson tentatively identified the Marsh Lake Trail caterpillars
as *Sunira verberata* and sent me photos from the Richardson Highway on
cottonwoods in 2017. There were outbreaks of this species in Alaska in
the 2000s and in 2017 (USDA Forest Service, Alaska Region
[2005](#ref-usda_forest_service_alaska_region_forest_2005),
[2006](#ref-usda_forest_service_alaska_region_forest_2006),
[2007](#ref-usda_forest_service_alaska_region_forest_2007),
[2008](#ref-usda_forest_service_alaska_region_forest_2008); FS-R10-FHP
[2018](#ref-fs-r10-fhp_forest_2018)). I had collected this species on
June 10, 2017 on the Fuller Lakes Trail (iNaturalist:
[6586033](https://www.inaturalist.org/observations/6586033), BOLD:
[MOBIL6501-18](http://boldsystems.org/index.php/Public_RecordView?processid=MOBIL6501-18)),
where I had noted that these were numerous.

## Tuesday, June 9

To do:

  - Fill out a USFWS Alaska Region 2020 field project go/no go screening
    matrix and individual project safety plan for the Sandpiper Lake
    elodea eradication project.
  - Project aviation safety plan for the Sandpiper Lake elodea
    eradication project.
  - Take A-110 Aviation Transportation of Hazardous Materials course.
  - Edit this week’s *Refuge Notebook* article.
  - *Refuge Notebook* catch-up.
  - Revise aquatic herbicide PUPs.

I added the vernacularName field to Kenai NWR’s checklist DwC-A output.

### 12:00 MRA discussion

I have had a terrible time with internet connectivity today, so I worked
on improving checklist format, which I can do locally. I added indexing
for kingdoms through species.

## Wednesday, June 10

To do:

  - Fill out a USFWS Alaska Region 2020 field project go/no go screening
    matrix and individual project safety plan for the Sandpiper Lake
    elodea eradication project.
  - Project aviation safety plan for the Sandpiper Lake elodea
    eradication project.
  - Take A-110 Aviation Transportation of Hazardous Materials course.
  - ~~Edit and submit this week’s *Refuge Notebook* article.~~
  - *Refuge Notebook* catch-up.
  - Revise aquatic herbicide PUPs.

I received this week’s *Refuge Notebook* article today. I revised this
and got it sent to the *Clarion*.

In the evening I entered data for LifeScanner specimens: an immature
earthworm from the large island in West Forest Lake (vial BOLD-XH4,
iNaturalist:
[49066582](https://www.inaturalist.org/observations/49066582)) and a
tiny caterpillar from a volunteering *Brassica* sp. in our high tunnels
(vial BOLD-0S1). I was not able to enter data from a leaf mining Diptera
larva in a leaf of bok choy collected this evening (vial BOLD-BS8)
because the LifeScanner interface would not accept this vial label.

## Thursday, June 11

To do:

  - Fill out a USFWS Alaska Region 2020 field project go/no go screening
    matrix and individual project safety plan for the Sandpiper Lake
    elodea eradication project.
  - Project aviation safety plan for the Sandpiper Lake elodea
    eradication project.
  - Take A-110 Aviation Transportation of Hazardous Materials course.
  - *Refuge Notebook* catch-up.
  - Revise aquatic herbicide PUPs.

Jake and I worked on the checklist some today, fixing records for which
origin was lacking. I also worked on improving the formatting of the
checklist.

I worked some on the Sandpiper Lake elodea eradication project aviation
safety plan.

I stopped by the office in the evening to get my field gear together for
field work tomorrow. I also walked quickly down to Headquarters Lake.
There does not appear to be a caterpillar outbreak going on in these
woods. There were some geometrid caterpillars, but I saw no noctuids.
The mosquitoes were nasty. I enjoyed just seeing some of these woods
again.

## Friday, June 12

I arrived early to get things together for today.

Mark, Amanda, Jake, and I hauled a UTV out to Adkins Road, then took the
UTV down the pipeline, continuing with weed survey work that they had
begun earlier.

Our first stop was at a gated valve on the pipeline at 60.57667 °N,
150.471202 °W. Here we just saw *Taraxacum officinale*, *Trifolium*, and
other common weeds of trails and roads.

Our next stop was at DP-100 at 60.5848 °N, 150.444657 °W. This was quite
clean. The only non-native plant we saw here was *Taraxacum officinale*.

The vegetation response has looked quite nice along this pipeline. Most
of the woods have burned with moderate severity and already aspens,
willows, and fireweed are resprouting. It looks like these will be
mostly aspen woods soon.

![Burned woods at DP-100.](2020-06-12_woods_at_DP-100.jpg)  
Burned woods at DP-100.

Here I collected aphids from *Salix bebbiana* → vial BOLD-PB6.

We stopped next at H-35 at 60.588818 °N, 150.430852 °W. Here were just
*Taraxacum officinale*, *Plantago major*, and *Matricaria discoidea*.

Our next stop was DP-110 at 60.597067 °N, 150.395643 °W. There was what
I believe is *Leucanthemum vulgare* here, but it was immature. Near here
I collected one morel with root-like structures (2020-06-12-MLB01,
iNaturalist:
[50778241](https://www.inaturalist.org/observations/50778241)).

As we made water crossings the belt-driven transmission began to slip
due to water getting into the CVT. We drained the transmission case, but
the problem continued so that we would begin to stall as we tried to
drive through water crossings.

Nate was already working off of Mystery Creek Road, so he drove his UTV
down the pipeline and towed us through remaining water crossings. Our
UTV was able to make it down Myster Creek Road.

![Our UTV being towed through water.](2020-06-12_UTVs.jpg)  
Our UTV being towed through water.

Nate drove Amanda, Jake, and me to Adkins Road. Jake and Amanda returned
to the office; I drove back to Mystery Creek Road to pick up Mark and
the UTV.

Observations from the day are available at
<https://www.inaturalist.org/calendar/mbowser/2020/6/12>.

## Tuesday, June 16

I edited this week’s *Refuge Notebook* article and sent comments back to
the author.

I also quickly responded with an explanation of the DNA sequencing “Big
Ideas” project funding request.

## Wednesday, June 17

I worked on making edits to this week’s *Refuge Notebook* article,
getting it sent in.

### Biology staff meeting at 09:00

## Thursday, June 18

I uploaded photo observations from June 12 to iNaturliast.

I tried to download data from FWSpecies to generate a list of species
lacking evidence for Jake to work on, but I was unable to download these
due to connectivity issues.

``` r
## Get a list of species lacking evidence.

## Load functions and libraries.
source("functions.R")

## First load data.
data1 <- assemble_csvs(directory="../data/FWSpecies")
fields_crosswalk <- read.csv("../data/field_name_crosswalk.csv", colClasses="character")
establishmentMeans_crosswalk <- read.csv("../data/establishmentMeans_crosswalk.csv", colClasses="character")
fillin <- read.csv("../data/taxonomy_fill-ins.csv", colClasses="character")

## Renaming fields.
for (this_field in 1:nrow(fields_crosswalk))
 {
 sl <- which(names(data1) == fields_crosswalk$FWSpecies_field[this_field])
 names(data1)[sl] <- fields_crosswalk$DwC_field[this_field]
 }
 
## Filling in some missing values.
data1$scientificName[data1$scientificName == ""] <- data1$SciName[data1$scientificName == ""]
 
data1$taxonRank <- tolower(data1$taxonRank)

data1 <- data1[data1$taxonRank == "species",] ## Limiting the list to species only for now.

sl <- data1$source == ""

nosource <- data1[sl,c("SciName", "SpeciesNote")]
nosource <- nosource[order(nosource$SciName),]

write.csv(nosource,
 "../data/species_lacking_evidence.csv",
 row.names=FALSE
 )
```

I got this list of 813 species on the checklist but lacking evidence to
Jake for him to work on.

## Friday, June 19

I added *Gnathacmaeops pratensis* to the Refuge’s checklist based on
iNaturalist observations. I started going through FS-R10-FHP
([2020](#ref-fs-r10-fhp_forest_2020)) for Kenai National Wildlife Refuge
records. I need to request that *Lirula macrospora* (Hartig) Darker be
added to FWSpecies’ taxonomy.

## Monday, June 22

I updated the map for the Sandpiper Lake Aviation Safety Plan.

![Sandpiper Lake project aviation safety plan
map.](2020-06-22_Sandpiper_Lake_aviation_map.jpg)  
Sandpiper Lake project aviation safety plan map.

### 11:00 Biology staff check-in

  - We may be able to begin some field work tomorrow.

I edited this week’s *Refuge Notebook* article.

I worked on the Sandpiper Lake Aviation Safety Plan.

## Tuesday, June 23

I took the Cybersecurity - Federal Information Systems Security
Awareness (FISSA) training due June 30.

I received word of a new paper on birches (Rowe et al.
[2020](#ref-rowe_how_2020)), which I pored over. This work does confirm
what we thought about the relative distributions of *Betula neoalaskana*
(*Betula pendula* ssp. *mandshurica*) and *Betula papyrifera* in Alaska.
They did not find any obvious pattern for something that would match
what we expect for *Betula kenaica*.

Their samples 1031, 8576, 8577, 8578 are all from the Russian River area
from Along the Sterling Highway up to Lower Russian Lake and are all
identified as *Betula neoalaskana*.

I am most curious about the different genotypes they obtained in the
Anchorage area, Alaska’s biggest city. It is peculiar that these were
not picked up in other areas. These different genotypes were both on
Joint Base Elmendorf-Richardson several miles north of the main part of
the city. I do not know if ornamental birches were planted anywhere on
the base, but a variety of ornamental birches including *Betula
papyrifera*, *Betula papyrifera* ‘Multistem’ (maybe the same as *Betula
papyrifera* ‘Clump’), *Betula pendula* ‘Dalecarlica’, and *Betula
pendula* ‘Youngii’ are sold and planted in the Anchorage area (one
example: <https://faltzland.com/projects/deciduous-trees/>).

I started looking into mapping the two birch genomes we received using
bwa (<https://github.com/lh3/bwa>) to map against the *Betula pendula*
genome of Salojärvi et al. ([2017](#ref-salojarvi_genome_2017))
(<https://genomevolution.org/CoGe/GenomeInfo.pl?gid=35079>,<https://genomevolution.org/CoGe/GenomeInfo.pl?gid=35080>).

## Wednesday, June 24

I finished editing and sent off this week’s *Refuge Notebook* article.

I prepared answers for the KBBI interview about pollinators this
evening.

### KBBI pollinator interview at 18:00

## Thursday, June 25

To do:

  - Contact DEC about pesticide applicator license.
  - Finish Sandpiper Lake PASP.

I worked on obtaining emergency contacts for the Sandpiper Lake Project
Aviation Safety Plan.

I tried multiple ways of contacting AK DEC to see about getting my
pesticide applicator license renewed.

I worked on entering specimen data into Arctos for which data were loss
in the Arctos data loss event this spring.

## Friday, June 26

I worked on entering specimen data into Arctos for which data were loss
in the Arctos data loss event this spring.

I created records for
[KNWR:Inv:54](https://arctos.database.museum/guid/KNWR:Inv:54) and
[KNWR:Ento:11385](https://arctos.database.museum/guid/KNWR:Ento:11385)-[KNWR:Ento:11390](https://arctos.database.museum/guid/KNWR:Ento:11390).

[KNWR:Ento:11389](https://arctos.database.museum/guid/KNWR:Ento:11389)
was problematic. I had written on [April 17](#friday-april-17) when I
sorted this one that this was from site MG12, soil sample 1 (barcode:
UAM100185964), but we did not visit site MG 12 in 2019.

That left two uncertain associations. These were Long Term Ecological
Monitoring Program site 3074, soil sample 2 and Slikok watershed project
site MG24, soil sample 1.
[KNWR:Ento:11389](https://arctos.database.museum/guid/KNWR:Ento:11389)
must have been from MG24, soil sample 1, so this is how I assigned it.

[KNWR:Ento:11385](https://arctos.database.museum/guid/KNWR:Ento:11385)
must have been from Long Term Ecological Monitoring Program site 3074,
soil sample 2.

I made small edits on the Sandpiper Lake PASP.

I need to make Arctos records for the fungal and bacterial soil sample
vials.

I started entering data for these samples.

| RTL sample ID | Arctos GUID                                                  |
| :------------ | :----------------------------------------------------------- |
| soifun3069-1  | [KNWR:Env:1](https://arctos.database.museum/guid/KNWR:Env:1) |
| soifun3069-2  | [KNWR:Env:2](https://arctos.database.museum/guid/KNWR:Env:2) |

## Monday, June 29

Entering data for soil samples:

| RTL sample ID | Arctos GUID                                                    |
| :------------ | :------------------------------------------------------------- |
| soifun3072-1  | [KNWR:Env:3](https://arctos.database.museum/guid/KNWR:Env:3)   |
| soifun3072-2  | [KNWR:Env:4](https://arctos.database.museum/guid/KNWR:Env:4)   |
| soifun3074-1  | [KNWR:Env:5](https://arctos.database.museum/guid/KNWR:Env:5)   |
| soifun3074-2  | [KNWR:Env:6](https://arctos.database.museum/guid/KNWR:Env:6)   |
| soifun3092-1  | [KNWR:Env:7](https://arctos.database.museum/guid/KNWR:Env:7)   |
| soifun3092-2  | [KNWR:Env:8](https://arctos.database.museum/guid/KNWR:Env:8)   |
| soifun3140-1  | [KNWR:Env:9](https://arctos.database.museum/guid/KNWR:Env:9)   |
| soifun3140-2  | [KNWR:Env:10](https://arctos.database.museum/guid/KNWR:Env:10) |
| soifun3202-1  | [KNWR:Env:11](https://arctos.database.museum/guid/KNWR:Env:11) |
| soifun3202-2  | [KNWR:Env:12](https://arctos.database.museum/guid/KNWR:Env:12) |
| soifun3288-1  | [KNWR:Env:13](https://arctos.database.museum/guid/KNWR:Env:13) |
| soifun3288-2  | [KNWR:Env:14](https://arctos.database.museum/guid/KNWR:Env:14) |
| soifun5555-1  | [KNWR:Env:15](https://arctos.database.museum/guid/KNWR:Env:15) |
| soifun5555-2  | [KNWR:Env:16](https://arctos.database.museum/guid/KNWR:Env:16) |
| soifunMG07-1  | [KNWR:Env:17](https://arctos.database.museum/guid/KNWR:Env:17) |
| soifunMG07-2  | [KNWR:Env:18](https://arctos.database.museum/guid/KNWR:Env:18) |

### Biology staff meeting at 11:00

  - I will be beginning field work this week on Composite Burn Index
    (CBI) training with Lisa.

I finished drafts of the Project Aviation Safety Plan and the USFWS
Alaska Region 2020 Field Project Go/No Go Screening Matrix and
Individual Project Safety Plan for the Sandpiper Lake elodea eradication
project. I also got my new pesticide applicator license. I asked to have
the two PUPs disapproved so that I could change the lead applicator to
my name.

Just looking at the map, it looks like LTEMP site 3202 might have burned
in the Swan Lake Fire after we had sampled it in early July 2019. I
checked: it is in the fire perimeter.

## Tuesday, June 30

I added *Lycogala epidendrum* to the Refuge’s checklist based on
iNaturalist observation
[51006945](https://www.inaturalist.org/observations/51006945).

I edited and submitted this week’s *Refuge Notebook* article.

### 10:00 interview

I learned that *Myosotis alpestris* ssp. *asiatica* Vesterg is the name
currently used by the [Alaska Center for Conservation
Science](https://accs.uaa.alaska.edu/) for our native, alpine
*Myosotis*.

I also learned that *Myosotis scorpioides*, a non-native species, is
present near the Refuge
(<https://www.inaturalist.org/observations/15957166>,
<https://nas.er.usgs.gov/queries/SpecimenViewer.aspx?SpecimenID=1628591>).

Entering data for soil samples:

| RTL sample ID | Arctos GUID                                                    |
| :------------ | :------------------------------------------------------------- |
| soifunMG19-1  | [KNWR:Env:19](https://arctos.database.museum/guid/KNWR:Env:19) |
| soifunMG19-2  | [KNWR:Env:20](https://arctos.database.museum/guid/KNWR:Env:20) |
| soifunMG24-1  | [KNWR:Env:21](https://arctos.database.museum/guid/KNWR:Env:21) |
| soifunMG24-2  | [KNWR:Env:22](https://arctos.database.museum/guid/KNWR:Env:22) |
| soifunMG39-1  | [KNWR:Env:23](https://arctos.database.museum/guid/KNWR:Env:23) |
| soifunMG39-2  | [KNWR:Env:24](https://arctos.database.museum/guid/KNWR:Env:24) |
| soifunSK22-1  | [KNWR:Env:25](https://arctos.database.museum/guid/KNWR:Env:25) |
| soifunSK22-2  | [KNWR:Env:26](https://arctos.database.museum/guid/KNWR:Env:26) |

I loaded maps, etc. onto my iPad for field work tomorrow.

# July

## Wednesday, July 1

<!-- 07:00-17:30 -->

Mark, Dawn, Lisa Saperstein, and I drove out to the west entrance of
Skilak Lake Road so that Lisa could train us on Composite Burn Index
(CBI) and related plot methods. We followed the protocol and datasheets
of Hrobak and Barnes ([2020](#ref-hrobak_standard_2020)).

Our first plot, KE-SWAN-01, was near the Marsh Lake Trail on the north
side of the road. This was in an aspen stand with some black spruce. It
has been burned quite severely so that no aspen suckers were coming up.
*Marchantia polymorpha*, *Ceratodon purpureus*, and fireweed covered
much of the mineral soil. Most of the trees had fallen down due to
burned-out roots. We saw one black spruce seedling, but no other tree or
shrub recruitment.

![Burn severity sampling site KE-SWAN-01, looking north from plot
center.](2020-07-01_KE-SWAN-01_N.jpg)  
Burn severity sampling site KE-SWAN-01, looking north from plot center.

Our second plot, KE-SWAN-05, was off of the road to Lower Skilak
Campground. This was in a wet black spruce stand with some birch. This
had been burned pretty severely, removing much of the organic soil and
burning out almost all tree roots, but the organic soil here was deep
and damp so that the lower layers remained. Some willows were coming
back here.

![Burn severity sampling site KE-SWAN-05, looking north from plot
center.](2020-07-01_KE-SWAN-05_N.jpg)  
Burn severity sampling site KE-SWAN-05, looking north from plot center.

## Thursday, July 2

<!-- 06:45-17:00 -->

I came in early to take care of notes, charge batteries, etc.

I spent the day off of the Sterling Highway and Mystery Creek with Lisa,
Dawn, and Amanda. We worked on more combined burn index plot training.

Our first plot was FE-SWAN-18 northeast of Egumen Lake. This ended up
being unburned black spruce swamp.

![Burn severity sampling site KE-SWAN-18, looking north from plot
center.](2020-07-02_KE-SWAN-18_N.jpg)  
Burn severity sampling site KE-SWAN-18, looking north from plot center.

On the way back to the truck we found some orchids that were unfamiliar
to me. Back at the office I identified these as *Galearis rotundifolia*
(iNaturalist:
[51766406](https://www.inaturalist.org/observations/51766406)). This
looks like a new record for the Kenai Peninsula.

We had lunch at Picnic Lake. There were lots of bryozoans under the
rocks in the water.

Our next plot was KE-SWAN-21 east of Picnic Lake. This was black spruce
forest that was a bit more moderately burned than what we had looked at
yesterday. More duff remained and more plants were resprouting.

![Burn severity sampling site KE-SWAN-21, looking north from plot
center.](2020-07-02_KE-SWAN-21_N.jpg)  
Burn severity sampling site KE-SWAN-21, looking north from plot center.

Here I collected what looked like *Morchella tomentosa* based on the
large, underground root-like structures, the radiscisclerotium of
Stefani ([2010](#ref-stefani_morchella_2010)). There were abundant,
immature aradid bugs apparently feeding on the bases of all of the
morels I collected. I collected one of these for sequencing (vial
BOLD-094).

Our last plot for the day was KE-SWAN-19 near the south end of Mystery
Creek Road. This was again in well-drained black spruce forest, but this
had been even more moderatly burned, with *Salix bebbiana*, some aspen,
fireweed, a little *Calamagrostis canadensis*, and many *Betula
glandulosa* resprouting. We did not have time to complete the CBI
protocols here; we just looked at it before leaving the field.

![Burn severity sampling site KE-SWAN-19, looking north from plot
center.](2020-07-02_KE-SWAN-19_N.jpg)  
Burn severity sampling site KE-SWAN-19, looking more or less north from
near plot center.

## Monday, July 6

<!-- 08:00-16:40 -->

I printed off datasheets and otherwise prepared for field work today on
the Hakala plots (Berg [2000](#ref-berg_post-fire_2000)).

Ed, Sarah, Jake, and I visited Hakala plot HAK 1 south of the radio
tower at the west border of the Refuge off of the Sterling Highway. At
the plot we first took photographs from just outside of the plot looking
into the plot from the north and south. Next, we set up the first
milacre plot and tallied all plant species. For dense species like
*Vaccinium vitis-idaea* we made quick visual tallies. Grasses were
counted as clumps. We also took diameter at breast height measurements
of all trees.

![Ed tallying *Vaccinium vitis-idaea* on Hakala plot 1, milacre
1.](2020-07-06_HAK1-1.jpg)  
Ed tallying *Vaccinium vitis-idaea* on Hakala plot 1, milacre 1.

Jake and I left after finishing the seventh milacre plot.

## Tuesday, July 7

<!-- 07:15-16:45 -->

I saw two snowshoe hares along the Sterling Highway east of Sterling
near the east entrance of Skilak Lake Road. They are definitely becoming
more abundant.

I joined Ed and Sarah at plot HAK2. They were on milacre 2 when I
arrived. This plot has extensive layering of black spruce, which may
become separate trees. Ed explained that he keeps a separate tally of
layering to avoid anomalies in the tree numbers.

![Hakala plot 2, milacre 2.](2020-07-07_HAK2.jpg)  
Hakala plot 2, milacre 2.

Ed pointed out a lichen, *Leptogium saturninum*, growing on aspen
(iNaturalist:
[52283404](https://www.inaturalist.org/observations/52283404)). This was
a species that I did not know.

After completing HAK2 we drove east down the road to HAK3, an aspen
stand which had been lightly burned in the 2015 Card Street Fire. Now
the aspens have fallen in all directions in this area, making getting
around quite difficult. We only located a subset of the stakes. We
decided to return to this one later. I obtained a GPS averaged location
for the south end of the plot. This was at 60.49992 °N, 150.49989 °W
with a GPS averaged accuracy of 1 m.

![Part of Hakala plot 3.](2020-07-07_HAK3.jpg)  
Part of Hakala plot 3.

On the way back to the truck I looked for noctuid pupae in the leaf
litter. I found one living, but apparently dehyrdrated larva that looked
like *Sunira verberata* (iNaturalist:
[52380873](https://www.inaturalist.org/observations/52380873)). I found
no pupae. I collected this (vial BOLD-QK7).

We drove further east to HAK4, which had been burned in the Swan Lake
Fire. This was severely burned, whith downed black spruce and aspen all
over and *Marchantia* and fireweed populating it. Dawn filled out a CBI
datasheet here. I left just as we were getting started on this plot.

![Hakala plot 4 photographed looking south over the north end of the
plot.](2020-07-07_HAK4-1.jpg)  
Hakala plot 4 photographed looking south over the north end of the plot.

I photographed and collected a gray morel, *Morchella tomentosa* in the
vicinity of the plot (iNaturalist:
[52382438](https://www.inaturalist.org/observations/52382438)). A spruce
seedling was growing by the morel. I had noticed no seedlings in the
plot.

## Wednesday, July 8

<!-- 07:30-10:00, 16:00-16:45 -->

I put together my notes and images from the last couple of days.

I edited and submitted this week’s *Refuge Notebook* article.

## Thursday, July 9

<!-- 08:00-17:30 -->

I met Ed and Sarah at Hakala plot HAK5, which had burned in the Swan
Lake Fire. There were too many downed trees for us to safely work here,
so we proceeded to HAK6.

![Vicinity of plot HAK5. Photo by Ed
Berg.](2020-07-09_HAK5_vicinity.jpg)  
Vicinity of plot HAK5. Photo by Ed Berg.

Dawn joined us at HAK6. This was a beautiful, lush site under birches
and *Alnus incana*.

![Surveying part of Hakala plot HAK6.](2020-07-09_HAK6.jpg)  
Surveying part of Hakala plot HAK6.

The alders were extensively damaged by green sawfly larvae that must be
<i>Monsoma pulveratum</i> (iNaturalist:
[52964721](https://www.inaturalist.org/observations/52964721)).

![*Monsoma pulveratum* larva on *Alnus
incana*.](2020-07-09_Monsoma.jpg)  
*Monsoma pulveratum* larva on *Alnus incana*.

## Friday, July 10

<!-- 07:10-16:45 -->

Ed, Sarah, Dawn, and I surveyed Hakala plot HAK7 in a black spruce
muskeg. Difficulties here illustrated problems with trying to use
individual counts of some of these species. Especially with species like
grasses and *Ledum*, it was difficult to recognize individuals. For
*Vaccinium vitis-idaea* we used basically a stem count. For *Ledum
palustre* and grasses we used clumps of individuals, yielding a much
smaller number than the stem count. I much prefer point intercept counts
for quantifying abundance.

![Surveying part of plot HAK7.](2020-07-10_HAK7.jpg)  
Surveying part of plot HAK7.

Between this plot and the highway we came across what looked like debris
from one of the orignal camps from the crews that built the highway in
the 1940s.

In the afternoon we proceeded to plot HAK8, a much easier plot to deal
with because the understory was extremely sparse. Mark joined us here.

![Surveying part of plot HAK8.](2020-07-10_HAK8.jpg)  
Surveying part of plot HAK8.

While others worked on HAK8, Dawn and I drove over to HAK9 to see how it
had been affected by the fire. We found that it had not been burned even
though the fire had reached to within less than 10 meters from the plot
in places. This was a rich, moist, lush plot.

![View of plot HAK9 looking north from the south end of the
plot.](2020-07-10_HAK9.jpg)  
View of plot HAK9 looking north from the south end of the plot.

## Sunday, July 12

An adult moth had emerged in each of the two jars of caterpillars
collected from Marsh Lake Trail on June 3. Both appeared to be *Sunira
verberata*.

![*Sunira verberata* adult reared from larvae collected by Dan Thompson
near Marsh Lake Trail on June
3.](2020-07-12-Sunira_verberata_Thompson.jpg)  
*Sunira verberata* adult reared from larvae collected by Dan Thompson
near Marsh Lake Trail on June 3.

## Monday, July 13

Ed sent me his notes and photos from the Hakala plots. He and Sarah did
survey plot HAK9 on Saturday.

I started on a *Refuge Notebook* article on *Sunira verberata* for this
week.

I learned that *Sunira verberata* was documented defoliating *Viburnum
edule* and consuming cultivated crops in 1976 in Alaska (Interagency
Arctic Research Coordinating Committee
[1977](#ref-interagency_arctic_research_coordinating_committee_department_1977)),
but the specific locality was not given. The authors reported that
disease and parasitism killed some of the larvae.

### 10:00 Biology meeting

  - Finish last two Hakala plots.

  - LTEMP/alpine CBI plots.

While searching through emails I found a thread entitled “Caterpillars
run amok, take Mat Valley by storm\!” in the IPM-L list. Initial post:

> I’ve several calls on these guys lately, anyone know what this one is?
> They seem to be munching on a wide variety of plants (inc rhubarb),
> covering the sides of homes and keeping at least one puppy fed.

![Presumed *Sunira verberata* caterpillar from the Mat-Su Valley in June
2017.](2020-07-13_2017_MatSu_Sunira.jpg)  
Presumed *Sunira verberata* caterpillar from the Mat-Su Valley in June
2017.

Later in the thread:

> I collected what I believe are the same thing on Friday in Anchorage.
> …they were found on a variety of hosts (aspen, elderberry, alder,
> and devil’s club) as well as houses, sidewalks and other man-made
> structures.

### Telecon with McCrea at 15:00

  - It looks like I should store observer names in a separate table.

  - I should try using R Studio and Roxygen for comments on R scripts.

  - R scripts will need mdEditor metadata files.

  - For this Thursday’s meeting I will need to explain what I have done
    so far and list remaining steps.

  - We can soon prepare to deposit this dataset in the regional
    repository.

## Tuesday, July 14

Annaleese and I drove out to Hakala plot HAK3 to survery it. We had to
begin by locating as much of the original rebar as we could, then pound
in enough replacement rebar where needed to make it so that we could
survey it well.

HAK3 had been burned in the 2015 Card Street fire, killing all standing
trees, but not burning deeply into the soil. It was now full of aspen
suckers, downed trees, and grasses. This was a difficult place to work,
so surveying this plot took the day.

![Hakala plot HAK3.](2020-07-14_HAK3.jpg)  
Hakala plot HAK3.

## Wednesday, July 15

I worked on my *Refuge Notebook* article on *Sunira verberata*, getting
it finished at the end of the day.

HOSTS (Robinson et al. [2020](#ref-robinson_hosts_2020)) lists *Cornus
sericea*, *Salix*, and *Symphoricarpos* as hosts for *Sunira verberata*
in Canada.

## Thursday, July 16

To do

  - *Refuge Notebook* catch-up
  - Update Biology News.
  - Add *Myosotis scorpioides* to the Refuge’s checklist.
  - Add metadata for snowshoe hare R scripts.
  - Revise data structure and processing of snowshoe hare data to
    include multiple observers per survey.
  - ~~Review new CBI Survey123 form.~~

I posted the *Refuge Notebook* articles from February 14 and 21.

### Data management early adopters’ meeting at 13:00

  - We will work toward depositing data in the regional repository by
    the end of the fiscal year (September).

## Monday, July 20

<!-- 07:00-16:30 -->

Mark and I drove out to plot HAK5, the last Hakala plot to be surveyed.
This was a burned plot within the Swan Lake Fire. It had burned pretty
severely, but some aspens were resprouting. Sparse birch and black
spruce seedlings were also present. I had difficulty identifying an
immature plant that I think is *Chamerion latifolium*. Lisa and Hunter
joined us here and did a CBI plot. I did the 5.64 m radius plant survey
and I took a soil sample of the top 10 cm of soil for soil fungal DNA
extraction. These were at 15 m east and west of plot center
corresponding to flagging for t he CBI plot.

![Hakala plot HAK5.](2020-07-20_HAK5.jpg)  
Hakala plot HAK5.

There were morels here, two of which I collected. I had dug the soil pit
for the western sample before I realized that there was a morel
immediately beside the soil pit. I hope that morel DNA is picked up in
the soil sample.

After that I stopped by plot HAK4 and did the same LTEMP veg protocol
and soil samples. I saw no resprouting plants here. I saw what looked
like immature *Chamerion latifolium* here as well.

Vegetation data are entered as an
[appendix](#occurrence-data-in-arctos-bulkloader-format).

## Tuesday, July 21

<!-- 06:30- -->

I put together some of my notes from yesterday.

I pulled FMH datasheets and scanned them for Lisa and Hunter.

I edited this week’s *Refuge Notebook* article.

I spent the day with Lynnda Kahn examining recent work done by HEA on
the power line right of way between Three Johns Road and Jean Lake. Some
poles had recently been removed cleanly with minimal damage. In
contrast, the access road that HEA put in around Jean Lake and Jean
Creek was poorly seeded, eroding, and weedy in places.

![Eroded access road between Jean Lake and the east end of Skilak Lake
Road.](2020-07-21_eroded_slope.jpg)  
Eroded access road between Jean Lake and the east end of Skilak Lake
Road.

## Wednesday, July 22

to do

  - ~~drop off vehicle 806 to be checked out.~~
  - LTEMP/FIA plot study plan
  - Swan Lake Fire BAER invasive plant survey planning.
  - Hakala plots follow-up
  - ~~Send out this week’s *Refuge Notebook*.~~
  - Provide feedback on NPS *Refuge Notebook* article

I worked in the office on cleaning up and catching up.

I did some checking of recent identifications. A *Cerastium* I had seen
the other day (iNaturalsit:
[52986938](https://www.inaturalist.org/observations/52986938)) was
*Cerastium fontanum* ssp. *vulgare*.

I looked at what data are available from my 2018 work in the Mystery
Hills (Bowser [2018](#ref-bowser_lab_notebook_2018)) and from Paez’s
work in 1988–1989 (Paez [1991](#ref-paez_alpine_1991)).

## Thursday, July 23

Ethan and I hiked up the Skyline Trail to do BAER invasive plant surveys
and to look at burn severity in the alpine and subalpine.

We documented non-native plant occurrences along the trail using
Survey123, occasionally just snapping photos. Along the trail we saw the
non-native species *Matricaria discoidea* (iNaturalist:
[54180986](https://www.inaturalist.org/observations/54180986)), *Phleum
pratense* (iNaturalist:
[54181213](https://www.inaturalist.org/observations/54181213)),
*Trifolium repens* (iNaturalist:
[54181736](https://www.inaturalist.org/observations/54181736)),
*Capsella bursa-pastoris* (iNaturalist:
[54181941](https://www.inaturalist.org/observations/54181941)),
*Cerastium fontanum* ssp. *vulgare* (inaturalist:
[54182218](https://www.inaturalist.org/observations/54182218)),
*Plantago major*, and *Poa annua*. These were all restricted to the
immediate vicinity of the trail.

The hemlock stand at the saddle at the top of the Skyline Trail had
burned, leaving few trees that will survive. Most of the trees in the
stand had yellowing needles that were dropping. The trees themselves had
been left standing, but the soil had burned beneath them. Now it was
soft ash under fresh, yellow needles.

![Dying hemlocks at the saddle.](2020-07-23_burned_hemlocks.jpg)  
Dying hemlocks at the saddle.

We looked for the sites from Paez’s work in 1988–1989 (Paez
[1991](#ref-paez_alpine_1991)) that were just above this hemlock stand
to the west and up the hill.

We first looked for Paez’s site 80. We navigated to the coordinates, but
failed to find any rebar or tags. We did search around the area. We also
navigated to and failed to find Paez’s sites 64 and 79, but we failed to
find these, also.

This area of what was probably low shrubs and tundra was severely
burned. Now only rock, soil, ash, and some coarse woody debris remained,
with very few surviving plants.

![Soil erosion in the alpine above the Skyline Trail near some of the
Paez plots.](2020-07-23_erosion.jpg)  
Soil erosion in the alpine above the Skyline Trail near some of the Paez
plots.

![Woody debris, ash, rock, and soil, all washing
downslope.](2020-07-23_woody_debris.jpg)  
Woody debris, ash, rock, and soil, all washing downslope.

At the center point of the coordinates we had or Paez’s site 79 we did
the CBI survey via Survey123. This was at 60.53048, -150.17537, GPS
averaged to \<2 m GPS error. We also inventoried all vascular plants
within a 5.64 m radius (only *Chamerion angustifolium*, *Achillea
borealis*, and *Ceratodon purpureus*) and collected two soil samples
from the first 10 cm of soil for soil fungal DNA work.

![5.64 m radius circular plot at Paez site 79
vicinity.](2020-07-23_Paez_site_79_vicinity_564m_radius_plot.jpg)  
5.64 m radius circular plot at Paez site 79 vicinity.

At the Skyline Trail’s underpass under the Sterling Highway we saw a
mustard with dissected leaves that was unfamiliar to me (iNaturalist:
[54176380](https://www.inaturalist.org/observations/54176380)). I
collected it.

## Friday, July 24

to do

  - FIA/LTEMP research plan/proposal
  - *Refuge Notebook* article comments
  - Inquire about remaining NGS samples/contract
  - Make Avenza map for Swan Lake Fire BAER invasive plant survey work
  - Review non-native species list
  - ~~Check projections, etc. for Paez site data.~~
  - ~~Send Ed Hakala plot data.~~

I examined the mustard from yesterday (iNaturalist:
[54176380](https://www.inaturalist.org/observations/54176380)). It
apears to be a *Descurainia*, but I was not able to identify it because
the fruits are too immature. I pressed the specimen.

I looked at the coordintaes from Paez’s ([1991](#ref-paez_alpine_1991))
vegetation plots, assuming multiple datums. The NAD27 coordinates are \>
200 m away from the NAD83/WGS84 coordinates, a big difference in the
alpine.

I had some trouble with ArcMap and converting between datums. I ended up
with three possibilities for where each point would be. Trying this in
R.

``` r
wd <- "D:/projects/Fire/2019_Swan_Lake_Fire/work_space/2020-07-24_Paez_coordinates"
setwd(wd)

require(maptools)
require(rgdal)

wgs84 <- "+proj=longlat +ellps=WGS84 +datum=WGS84 +no_defs"
nad27UTMz5N <- "+proj=utm +zone=5 +datum=NAD27 +units=m +no_defs" # From https://epsg.io/26705

paeznad27 <- read.csv("2020-01-17-1447_Paez_Mystery_Hills_locaclities.csv")
coordinates(paeznad27) <- c("utm_e", "utm_n")

proj4string(paeznad27) <- CRS(nad27UTMz5N)
paezwgs84 <- spTransform(paeznad27, CRS(wgs84))

write.csv(paezwgs84, "2020-07-24-1208_Paez_from_NAD27_UTM_to_lat_lon.csv", row.names=FALSE)
```

![Mystery Hills vegetation plot coordinates from Paez’s vegetation
plots, assuming multiple
datums.](2020-07-24-1215_Paez_coordinates.jpg)  
Mystery Hills vegetation plot coordinates from Paez’s vegetation plots,
assuming multiple datums.

Both the R script above and ArcMap placed the coordinates to the north
and west fo where I went, agreeing withthe pink hexagons in the map
above. This set of reprojected coordinates also seems to be most
consistent with alpine tundra habitat. I loaded this map (Avenza pdf) to
my iPad and coordinates into my GPS.

I noticed that Paez’s Mystery Hill sites were resampled in 1998 (Paez
[2009](#ref-paez_dynamics_2009)). I wonder if these data can be dug up
from ADF\&G’s files. I found no current contact information for Carlos
Paez.

After examining more examples over the last few days, I am pretty sure
that we were seeing small *Chamerion latifolium* plants on Hakala plot
HAK5 on [July 20](#monday-july-20).

## Monday, July 27

<!-- 07:30-17:45 -->

I saw an abandoned or feral cat at the headquarters building this
morning (iNaturalist:
[54647478](https://www.inaturalist.org/observations/54647478)).

Annaleese and I hiked Skyline, this time with a goal looking at how the
fire burned in tundra and completing CBI plots at sites where we had
historical vegetation data.

We first hiked to the corrected coordinates (assuming NAD27 datum) for
Paez’s vegetation plots 142 and 140, but we were unable to locate the
plots. From here we walked over the ridge looking into the next valley.
Much of the tundra had burned here on the northeast side of the Mystery
Hills.

![Edge of unburned patch of tundra showing thickness of burned
duff.](2020-07-27_burn_edge.jpg)  
Edge of unburned patch of tundra showing thickness of burned duff.

We next walked down the southwest-facing slope to site Skyline354-5 that
I had put in in 2018 (Bowser [2018](#ref-bowser_lab_notebook_2018),
[2019](#ref-bowser_field_2019)[b](#ref-bowser_field_2019)). Most of this
plot had burned severely so that it was now mostly ash, soil, and rock,
but there were unburned patches. A couple of hemlock seedlings had
already gotten started. Here we did a CBI plot, collected soil samples
for detecting soil fungi, recorded presence of plant species within a
5.64 m radius circular plot, and repeated the two (N and S) point
intercept transects as was done in 2018. These were at every 0.5 m from
0.5 m to 10 m on the right side of the tape starting at plot center.

![Plot Skyline354-5, viewed from the
south.](2020-07-27_plot_354-5.jpg)  
Plot 354-5, viewed from the south.

![Plot Skyline354-5, viewed from the south on June 20,
2018.](2018-06-20_plot_354-5.jpg)  
Plot Skyline354-5, viewed from the south on June 20, 2018.

It was remarkable how severly the tundra and low shrubs had burned.
There were unburned areas, but where the fire had come through it was
generally burned to mineral soil, greatly transforming the landscape.

We stopped by plot Skyline354-2 on the way back. This had burned, also.

![Plot 354-2, viewed from the south.](2020-07-27_plot_354-2.jpg)  
Plot 354-2, viewed from the south. The burned fiberglass rod marking
plot center is visible in the foreground.

![Plot 354-2, viewed from the south on June 11,
2018.](2018-06-11_plot_354-2.jpg)  
Plot 354-2, viewed from the south on June 11, 2018.

On the way down we observed one occurrence of a non-native plant that I
had missed on Thursday, *Alopecurus pratensis* (iNaturalist:
[54664464](https://www.inaturalist.org/observations/54664464)).

## Tuesday, July 28

to do

  - ~~Notes from yesterday~~
  - ~~Edit *Refuge Notebook* article~~
  - ~~Review non-native species list.~~
  - Get snowshoe hare project metadata ready for Thursday meeting.

I edited this week’s *Refuge Notebook* article, getting it sent off.

I started an Arctos project for this year’s vegetation work following
the Swan Lake Fire (<https://arctos.database.museum/project/10003388>).

Now I need to review the list of non-native terrestrial invertebrates of
Alaska, updated data from the earlier list (Simpson et al.
[2019](#ref-simpson_list_2019)).

``` r
## Comparing the new list to the previous version.
wd <- "D:/projects/non-native_species/2020_comprehensive_list_review"
setwd(wd)

old <- read.csv("Non-native_Master_V2-0_20190912.csv", colClasses="character")
new <- read.csv("NNSL_AK-HI-L48_V3_2020-06-10_ForReview.csv", colClasses="character")

select_old <- (old$Region == "AK") &
 (old$Kingdom == "Animalia")
old_animals <- old[select_old, 1:33]

dim(old_animals)
[1] 142  33

select_new <- (new$Region == "AK") &
 (new$Kingdom == "Animalia")
new_animals <- new[select_new, 1:33]

dim(new_animals)
[1] 142  33

setdiff(old_animals$scientificName, new_animals$scientificName)
character(0)

setdiff(new_animals$scientificName, old_animals$scientificName)
character(0)
## It appears that there are no differences in the lists of species.

write.csv(new_animals, "2020-07-28-1456_AK_animals.csv", row.names=FALSE)
```

I edited the record for *Eisenia fetida*, added *Eisenia andrei*, and
got these edits to Annie.

## Wednesday, July 29

to do

  - ~~Respond to Lisa~~
  - ~~Addition of *Adelges piceae* to the checklist of non-native
    species~~
  - Prepare to send out morel specimens?
  - Inquire about NGS contract extension.
  - snowshoe hare processing / metadata

I edited the list of non-native animal species of Alaska and got these
edits sent out.

I looked over next’s weeks’ *Refuge Notebook* article draft and provided
edits and comments.

I looked into sequencing morel specimens. The North American Mycoflora
project is currently transitioning to using BOLD for its sequencing
service and the sequencing service is temporarily unavailable.

I worked on reorganizing the snowshoe hare pellet count dataset.

``` r
## I need to reorganize data some in order to associate observers and observations, keeping these in separate tables.

## Load data.

pellet_data <- read.csv("../data/final_data/observations/snowshoe_hare_pellet_counts.csv",
 stringsAsFactors=FALSE
 )
 
                                       
plot_data <- read.csv("../data/final_data/geodata/2020-03-18_snowshoe_hare_plot_data.csv",
 stringsAsFactors=FALSE
 )
 
## Join the data to get the grid_name values.
pellet_data_joined <- merge(
 plot_data[,c("plot_name", "grid_name")],
 pellet_data
 )

## Using eventID sensu http://rs.tdwg.org/dwc/terms/eventID

event <- unique(pellet_data_joined[,c("grid_name", "date")])
event <- event[order(event$date, event$grid_name),]
event$eventID <- 1:nrow(event)
  
## Now joining these back to the original data.
pellet_data_new <- merge(pellet_data_joined, event[,c("grid_name", "date", "eventID")])
## That looked good.

## Now making the observer table.
## Using recordedBy sensu http://rs.tdwg.org/dwc/terms/recordedBy
## This will be easier to enter manually...

## Saving the observation table.
pellet_data_new <- pellet_data_new[order(pellet_data_new$date, pellet_data_new$plot_name),]
write.csv(pellet_data_new[,c("plot_name", "date", "eventID", "pellet_count")],
  "../data/final_data/observations/snowshoe_hare_pellet_counts.csv",
  row.names=FALSE
  )
```

I also updated the script for generating graphic summaries.

![Snowshoe hare pellet counts over
time.](2020-07-29_pellet_count_means_over_time.png)  
Snowshoe hare pellet counts over time.

# August

## Monday, August 3

To do

  - ~~Edit this week’s *Refuge Notebook*.~~
  - ~~weed survey mapping/planning.~~

### Biology staff meeting at 08:00

  - I will be working on weed survey and Skyline field work this week.
  - The Sandpiper Lake elodea eradication project may go forward.

I receieved an ITS sequences from mushroom specimens I had submitted.

Specimen iNaturliast:
[34708025](https://www.inaturalist.org/observations/34708025)) I had
thought was a *Cortinarius*, but the sequence is identified as *Galerina
badipes* using UNITE and blastn searches. My photos do not look like
*Galerina badipes*. I wonder if this is a case of contamination.

Specimen iNaturliast:
[34709612](https://www.inaturalist.org/observations/34709612)) is
*Hygrophorus* sp. SH2252126.08FU
(<https://unite.ut.ee/sh/SH2252126.08FU>) based on its ITS sequence.

Specimen iNaturliast:
[34710090](https://www.inaturalist.org/observations/34710090)) appears
to be *Cystodermella granulosa*. It is 99.55% similar to a sequence in
species hypothesis
[SH2460475.08FU](https://unite.ut.ee/sh/SH2460475.08FU) (GenBank
accession
[GU234151.1](https://www.ncbi.nlm.nih.gov/nucleotide/GU234151.1)).

I mapped our current invasive species records in the Swan Lake Fire from
the Survey123 form.

![Map of current non-native species data in the Swan Lake Fire area from
our Survey123 form.](2020-08-03-1111_invasives_mapping.jpg)  
Map of current non-native species data in the Swan Lake Fire area from
our Survey123 form.

Mike sent me a link to a useful pdf map at
<https://ftp.nifc.gov/public/incident_specific_data/alaska/2019/2019_SwanLake/Maps/20190930/Ops_arch_e_land_20190930_SwanLake_AKKKS903181.pdf>.

Looking at the maps, it looks like we should visit sites near Fuller
Lakes Trail and off of Mystery Creek Road.

I edited this week’s *Refuge Notebook* article.

## Tuesday, August 4

I received the Berlese sample data from RTL Genomics this morning. I
perused these quickly. What I saw looked good.

Annaleese and I first drove to the Marsh Lake Trail trailhead, where we
looked for *Bromus tectorum*, which had been reported from this site.
There were many grasses here, but we failed to find *B. tectorum*.
*Agrostis scabra* and *Deschampsia cespitosa* were the dominant grasses.

Next we hiked the beginning of the Fuller Lakes Trail surveying for
weeds using Survey123. The trail had been used as a fire break during
the Swan Lake Fire. On the trail were typical trailside weeds:
*Taraxacum officinale*, *Poa annua*, *Plantago major*, and *Phleum
pratense*.

We walked as far as the power lines, then we crossed the creek,
following the power lines west from the Fuller Lakes Trail. The power
lines were quite weedy. We saw a plant that I could not identify. I did
not collect it because there were no inflorescenses. I thought that I
would not be able to identify it. I may try to return to this spot to
identify this plant.

![An unfamiliar plant from the powerlines (iNaturalist:
[55658833](https://www.inaturalist.org/observations/55658833)).](2020-08-04_msytery_plant.jpg)  
An unfamiliar plant from the powerlines (iNaturalist:
[55658833](https://www.inaturalist.org/observations/55658833)).

## Wednesday, August 5

I spent the day on Skyline, surveying two more of the shrub plots I had
put in in 2018 with Mark. The rest of Mark’s family accompanied us as
far as the saddle. From there they hiked northwest up the small peak to
the northwest looking for the Carlos Paez Caribou range study sites
using the updated coordinates. They failed to find any of these sites.

Mark and I surveyed shrub plot 354-2, which had burned severely. We also
surveyed plot 354-1, which had burned, but much less severely. Much of
the plot had burned to mineral soil, but in other places only the top of
the litter/duff had burned lightly so that vegetation was essentially
unchanged. At these two sites we completed the CBI protocols, two 10 m,
20 point vegetation transects, and 5.64 m radius, circular plot
vegetation inventories.

![Plot Skyline354-2, looking north over plot
center.](2020-08-05_Skyline354-2_north.jpg)  
Plot Skyline354-2, looking north over plot center.

![Plot Skyline354-1, looking north over plot
center.](2020-08-05_Skyline354-1_north.jpg)  
Plot Skyline354-1, looking north over plot center.

![Plot Skyline354-1, looking north from plot center on June 11,
2018.](2018-06-11_Skyline354-1_north.jpg)  
Plot Skyline354-1, looking north from plot center on June 11, 2018.

Tonight I sent in this week’s *Refuge Notebook* article, almost
forgetting to do so.

## Thursday, August 6

I am trying to identify a *Rorippa* that I collected from the power line
off of Fuller Lakes Trail on [August 4](#tuesday-august-4) (iNaturalist:
[55618332](https://www.inaturalist.org/observations/55618332)).

Keying using the key to *Rorippa* in the efloras Flora of North America
treatment
(<http://www.efloras.org/florataxon.aspx?flora_id=1&taxon_id=128745>).

1 → 2 → 4 → *Rorippa barbareifolia*. The fruits are consistently 4
valved.

The *Polygonum* I collected from the same area on August 4 looks like
*Polygonum aviculare*.

A grass I collected on August 4 is *Lolium multiflorum* (iNaturalist:
[55627973](https://www.inaturalist.org/observations/55627973)).

I worked on bulkloading Hakala plot 5.64 m radius plot vegetation data
into Arctos.

``` r
## This time generating UUIDs for these specimen records.
library(uuid)

uuids <- UUIDgenerate(n=17)

write.csv(uuids, "uuids.csv")
```

I uploaded these records
([KNWRObs:Herb:954](https://arctos.database.museum/guid/KNWRObs:Herb:954)–[KNWRObs:Herb:970](https://arctos.database.museum/guid/KNWRObs:Herb:970)).

## Friday, August 7

I did some data entry, catch-up on my notes, etc.

We received the go-ahead to pursue the Sandpiper Lake elodea eradication
project.

I manned the radio for a while.

I met Ben today, our new invasives EDRR lead.

I spent most of the afternoon going over equipment, getting dimensions,
putting things together, and trying out the equipment in an Alumacraft
T14V like we will have at Vogel Lake.

The boat’s front hollow is 72 cm long by 114 cm wide at its narrowest
end toward the bow.

The spreader base is 68 cm long by 60 cm wide. overall dimensions are
101 cm tall by 90 cm long by 60 cm wide, but I think the backpack blower
could be removed, making the length about 70 cm. I tried it out and it
fits well in the boat. It is ready to go.

![Spreader in the boat.](2020-08-07_spreader_in_boat.jpg)  
Spreader in the boat.

The 55 gallon liquid tank is 60 cm wide and 92 cm long without valves
attached. It fits, but it would need to be raised on something to keep
the valve fitting from having weight on it.

![Liquid application tank in the
Alumacraft.](2020-08-07_tank_in_boat.jpg)  
Liquid application tank in the Alumacraft.

I tried assembling the apparatuses for pelleted and liquid applications.
The pellet spreader is ready to go. We appear to be missing one of the
nozzles from the liquid application setup. I wrote our staff and some
others asking about this.

## Monday, August 10

I worked on communicating, planning, and getting equipmet together for
the Sandpiper Lake elodea application.

Jake and I transferred SonarONE pelleted fluridone, Littora liquid
diquat, the outboard motor, the pellet spreader, and the liquid 55
gallon tank to the hangar. We also weighed these items.

Looking at the Alamacraft, it can hold 650 lbs. The liquid tank filled
to 50 gallons would be 418 pounds, leaving only 232 pounds, not enough
for people and gear. The 24 gallon, tank intended for the ATV would be
200 pounds.

I must make calculations for applying liquid Littora. This should be
applied at a maximum of 2 gallons per surface acre. In the past we have
applied it at 1 gallon per surface acre.

## Tuesday, August 11

Jake and I spent the day preparing for the Sandpiper Lake elodea
eradication. We spent much of the afternoon adapting the ATV spreader
for use on a boat and calibrating it.

## Wednesday, August 12

Jake, Mark, and I met Nate and Jim at Soldotna Airport at 09:00. We
loaded both 206’s with herbicide. Nate and Jim flew this out to
Sandpiper Lake, where they worked on caching it on the south shore of
Sandpiper Lake.

![Loading herbicide.](2020-08-12_loading_herbicide.jpg)  
Loading herbicide.

While they were away, Jake and I filled the gas tank of the outboard,
connected it, and tested the outboard.

We met Nate and Jim for a second load. We inspected their aircraft for
*Elodea*. We only found one fragment of *Potamogeton*. this time taking
out almost all of the application equipment, including the outboard.
This was also cached at Sandpiper Lake. After dropping this off, Jim and
Nate returned to Anchorage.

We spent the rest of the day getting equipment together. I printed off
all of the permits, the prescription, signs, etc.

![Pellet spreader and outboard in the
206.](2020-08-12_spreader_in_206.jpg)  
Pellet spreader and outboard in the 206.

## Thursday, August 13

Jake, Colin, and I brought remaining equipment down to the floatplane
dock at Headquarters Lake. Shay flew out the two canoes one at a time.
Nate first took Jake and almost all of our things out to Vogel Lake on
the first load. He took Colin and me on the second load.

Once at Vogel Lake cabin, we arranged our things, then loaded up
herbicide application gear and took the two canoes and the Alamacraft
over to Sandpiper Lake. Getting those boats through the wetlands took a
while.

Once at Sandpiper Lake, we set up the outboard on the boat and then Jake
and I surveyed the lake for *Elodea*. We detected *Elodea* at 3 out of
22 sites, all detections in the northernmost part of the lake.

![Healthy *Elodea* at Sandpiper Lake (iNaturalist:
[56759042](https://www.inaturalist.org/observations/56759042)).](2020-08-13_elodea_detail.jpg)  
Healthy *Elodea* at Sandpiper Lake (iNaturalist:
[56759042](https://www.inaturalist.org/observations/56759042)).

![A dense bed of *Elodea*.](2020-08-13_elodea_bed.jpg)  
A dense bed of *Elodea*.

![A rake full of long, luxuriant
*Elodea*.](2020-08-13_elodea_rake.jpg)  
A rake full of long, luxuriant *Elodea*.

![A dense patch of *Elodea* under *Potamogeton
natans*.](2020-08-13_elodea_Potamogeton.jpg)  
A dense patch of *Elodea* under *Potamogeton natans*.

### *Elodea* rake survey data from Sandpiper Lake on August 13.

“Shoreward” and “lakeward” indicate rake throws away from and towards
shore. Values range from a missing value for non-detections, 1 for 1–2
strands of *Elodea* on the rake, 2 for 3 strands to \< 1/2 of the rake
tines having picked up *Elodea*, and 3 indicates \> 1/2 of the rake
tines having picked up *Elodea*.

| waypoint | site\_name  | Latitude  | Longitude    | shoreward | lakeward |
| -------: | :---------- | :-------- | :----------- | --------: | -------: |
|      161 | Sandpiper01 | 61.000824 | \-150.410436 |           |          |
|      162 | Sandpiper02 | 61.000637 | \-150.40904  |           |          |
|      163 | Sandpiper03 | 61.001086 | \-150.408184 |           |          |
|      164 | Sandpiper04 | 61.00169  | \-150.407146 |           |          |
|      165 | Sandpiper05 | 61.002258 | \-150.406141 |           |          |
|      166 | Sandpiper06 | 61.00275  | \-150.404943 |           |          |
|      167 | Sandpiper07 | 61.003281 | \-150.403278 |           |          |
|      168 | Sandpiper08 | 61.004301 | \-150.403085 |           |          |
|      169 | Sandpiper09 | 61.005236 | \-150.403548 |           |          |
|      170 | Sandpiper10 | 61.005938 | \-150.403293 |           |          |
|      171 | Sandpiper11 | 61.006677 | \-150.403633 |           |          |
|      172 | Sandpiper12 | 61.007141 | \-150.403428 |           |        3 |
|      173 | Sandpiper13 | 61.00764  | \-150.404848 |           |          |
|      174 | Sandpiper14 | 61.007455 | \-150.405791 |         3 |          |
|      175 | Sandpiper15 | 61.007046 | \-150.406244 |         3 |          |
|      176 | Sandpiper16 | 61.005954 | \-150.406746 |           |          |
|      177 | Sandpiper17 | 61.005253 | \-150.408325 |           |          |
|      178 | Sandpiper18 | 61.004642 | \-150.410032 |           |          |
|      179 | Sandpiper19 | 61.003777 | \-150.411578 |           |          |
|      180 | Sandpiper20 | 61.003107 | \-150.412391 |           |          |
|      181 | Sandpiper21 | 61.002255 | \-150.412013 |           |          |
|      182 | Sandpiper22 | 61.001149 | \-150.411302 |           |          |

Afer surveying, Colin began treating Sandpiper Lake.

It was late when we made it back to the cabin in the canoes.

## Friday, August 14

![Vogel Lake cabin and our tents. We did not all stay in the cabin due
to COVID-19 concerns.](2020-08-14_tents_and_cabin.jpg)  
Vogel Lake cabin and our tents. We did not all stay in the cabin due to
COVID-19 concerns.

Colin, Jake, and I took the canoes back to Sandpiper Lake in the
morning. Colin and I completed the application at Sandpiper Lake: 5
gallons of Littora at 2 gallons/acre and 840 lbs. of SonarONE.

The bigger task for the day was getting all of our remaining equipment
back to Vogel Lake cabin for pick-up. We loaded all of the equipment and
the unused Littora into the Alumacraft and canoes, which we had to drag
much of the way through the little streams and around the pike barriers.
We cleaned and inspected our boat and equipment to make sure that we
brought neither *Elodea* nor residual herbicide through the other lakes.

![Dragging the boat around a fish
barrier.](2020-08-14_dragging_boat.jpg)  
Dragging the boat around a fish barrier.

I was surprised to see one individual of *Scorzoneroides autumnalis* in
front of the Vogel Lake public use cabin (iNaturalist:
[56765869](https://www.inaturalist.org/observations/56765869)). I pulled
it.

Later Shay took us and all of our equipment to Headquarters Lake in
three loads in the 185.

## Monday, August 17

To do:

  - ~~Record notes from Sandpiper Lake treatment.~~
  - ~~Finish putting away equipment.~~
  - ~~Review spruce genetics document.~~
  - Non-native plant photos to Lynnda.

I made a quick map of the distribution of *Elodea* at Sandpiper Lake
based on a rake throw survey.

![Map of *Elodea* detections at Sandpiper
Lake.](2020-08-17_Sandpiper_elodea_map.jpg)  
Map of *Elodea* detections at Sandpiper Lake. Red dots indicate
detections; white dots indicate non-detections. The two red balloons
marked “E” mark the approximate limits of the infestation.

Jake and I put away all of the equipment from the Sandpiper Lake
treatment.

## Tuesday, August 18

  - ~~Photos/weed records to Lynnda.~~
  - meetings
  - Edit *Refuge Notebook* article.

### Biology staff meeting at 08:00

  - I will be doing non-native plant surveys off of Mystery Creek Road.

I sent Lynnda non-native plant observations from powerlines from [July
21](#tuesday-july-21) and [August 4](#tuesday-august-4).

### AKISP Elodea Committee Meeting at 14:00

  - Toby asked me to get a cost per acre for treating remote lakes.

## Wedneday, August 19

To do

  - ~~Submit *Refuge Notebook* article.~~
  - ~~Process spruce genetics data.~~

<!-- end list -->

``` r
## Comparing genetics of trees and seedlings.

wd <- "I:/BIOLOGY/Data/ProjectData/Grasslands/2015_grassland_project/spruce_genetics/work_space/2020-08-19_age_classes"
setwd(wd)

data1 <- read.csv("individuals.csv")

## Extract age classes from sample names.
data1$age_class <- NA
data1$age_class[grepl("TREE", data1$Sample)] <- "tree"
data1$age_class[grepl("SEED", data1$Sample)] <- "seedling"
data1$age_class[grepl("SAPL", data1$Sample)] <- "sapling"
data1$age_class[grepl("SAMPLING", data1$Sample)] <- "sapling"
data1$age_class[grepl("SAPLING", data1$Sample)] <- "sapling"

## Extract some of the site names.
sample1 <- strsplit(data1$Sample, "X")
sample1.1 <- sapply(sample1, "[", 1)
site1 <- strsplit(sample1.1, "-")
site1.2 <- sapply(site1, "[", 2)
data1$site <- paste0(site1.2, "X")

for (this_string in c("FSEEDX", "FTREEX", "FSAPLX", "FSAPX", "FSAPLINGX"))
 {
 data1$site[grepl(this_string, data1$site)] <- gsub(this_string, "F", data1$site[grepl(this_string, data1$site)])
 }

data1 <- data1[,c(2,15,14,1,3:13)]
data1 <- data1[order(data1$site, data1$age_class, data1$Sample),]

## Much of the site names will need to be cleaned up manually.
write.csv(data1, "2020-08-19-0906_individuals.csv", row.names=FALSE)

data2 <- read.csv("2020-08-19-0908_individuals.csv", stringsAsFactors=FALSE)

## Now starting the process of associating coordinates...
sites1 <- read.csv("site_data.csv", stringsAsFactors=FALSE)

setdiff(data2$site, sites1$locality)

setdiff(sites1$locality, data2$site)

## I think one of these was a transcription error:
data2$site[data2$site=="169X"] <- "69X"

## Making another site name more consistent:
data2$site[data2$site=="Lost Lake"] <- "Lost Lake Trailhead"

data2$site[data2$site=="Fox Creek Fire 1"] <- "100F"

## Loading edited site data.
sites2 <- read.csv("2020-08-19-0959_site_data.csv", stringsAsFactors=FALSE)

setdiff(data2$site, sites2$locality)

setdiff(sites2$locality, data2$site)
## That looked great!

names(sites2)[1] <- "site"

dim(data2)
[1] 446  15

dim(sites2)
[1] 60  7

data3 <- merge(data2, sites2, all.x=TRUE)

## Saving this.
write.csv(data3, "2020-08-19-1053_individuals.csv", row.names=FALSE)

## Now it is time to look at these data.

## From Diana's graphics, the salmon color for Sitka spruce is f19b78.

## For the purpose of this exercise I am going to lump saplings and seedlings.
data4 <- data3
data4$age_class[data4$age_class=="sapling"] <- "seedling"

## Cutting out missing data.
data4 <- data4[!is.na(data4$age_class),]
data4 <- data4[!is.na(data4$Sitka),]
dim(data4)
[1] 272  21

## Now for summaries...
ag1 <- aggregate(data4$Sample, by=list(data4$site), FUN=length)
names(ag1) <- c("site", "n_samples")

ag1$n_tree <- aggregate(data4$age_class=="tree", by=list(data4$site), FUN=sum, na.rm=FALSE)$x
ag1$n_seedling <- aggregate(data4$age_class=="seedling", by=list(data4$site), FUN=sum, na.rm=FALSE)$x

## Checking:
ag1$n_tree + ag1$n_seedling == ag1$n_samples
## Looked good.

ag1$n_tree_white <- aggregate(data4$age_class=="tree" & data4$Species=="w", by=list(data4$site), FUN=sum, na.rm=FALSE)$x
ag1$n_tree_sitka <- aggregate(data4$age_class=="tree" & data4$Species=="s", by=list(data4$site), FUN=sum, na.rm=FALSE)$x
ag1$n_tree_lutz <- aggregate(data4$age_class=="tree" & data4$Species=="HYBRID", by=list(data4$site), FUN=sum, na.rm=FALSE)$x
ag1$n_tree_black <- aggregate(data4$age_class=="tree" & data4$Species=="b", by=list(data4$site), FUN=sum, na.rm=FALSE)$x

ag1$n_seedling_white <- aggregate(data4$age_class=="seedling" & data4$Species=="w", by=list(data4$site), FUN=sum, na.rm=FALSE)$x
ag1$n_seedling_sitka <- aggregate(data4$age_class=="seedling" & data4$Species=="s", by=list(data4$site), FUN=sum, na.rm=FALSE)$x
ag1$n_seedling_lutz <- aggregate(data4$age_class=="seedling" & data4$Species=="HYBRID", by=list(data4$site), FUN=sum, na.rm=FALSE)$x
ag1$n_seedling_black <- aggregate(data4$age_class=="seedling" & data4$Species=="b", by=list(data4$site), FUN=sum, na.rm=FALSE)$x

## Saving this aggregate here.
write.csv(ag1, "2020-08-19-1223_counts.csv", row.names=FALSE)

## Now graphing.
pdf(file="2020-08-19-1251_barplots.pdf", width=8.5, height=11)
cols <- c("#ffffff", "#006600", "#4dff4d", "#000000")
labs <- c("P. glauca", "P. sitchensis", "P. x lutzii", "P. mariana")
ymax <- max(ag1[,5:12])
headers <- c("trees", "seedlings")
ylabs <- c("number of individuals", "")
this_site <- 0
par(mfrow=c(5,2) )
for (this_site in 1:55)
  {
  header <- ""
  ylab <- ""
  print_header <- is.element(this_site, (0:10)*5+1)
  print_ylab <- is.element(this_site, (0:10)*5+3)
  for (this_age in 1:2)
   {
   if (print_header==TRUE)
    {
    header <- headers[this_age]
    }
   if (print_ylab==TRUE)
    {
    ylab <- ylabs[this_age]
    } 
   barplot(as.vector(t(ag1[this_site,(this_age*4+1):(this_age*4+4)])), ylim=c(0,ymax), col=cols, names.arg=labs)
   mtext(header, side=3)
   mtext(ylab, side=2, line=3)
   if (this_age==2)
    {
    mtext(ag1$site[this_site], side=4)
    }
   } 
  }
dev.off()

## Now I just want to look at proportions of white:Sitka
## Dropping the non-numeric sites.

sums1 <- apply(ag1[1:52,2:12], 2, sum, na.rm=FALSE)

## Ratio of P. x lutzii to P. glauca in trees:
sums1[6]/sums1[4]
n_tree_lutz 
  0.1869159 

## Ratio of P. x lutzii to P. glauca in seedlings/saplings:
sums1[10]/sums1[8]
n_seedling_lutz 
  0.09708738

write.csv(sums1, "2020-08-19-1353_sums.csv")

## Quick map.
pdf("2020-08-19-1412_map.pdf", width=8.5, height=11, pointsize=6)
plot(sites2$lon, sites2$lat, cex=0.5, col="red")
text(sites2$lon, sites2$lat, sites2$site)
dev.off()
```

I sent out this week’s *Refuge Notebook* article.

## Thursday, August 20

Annaleese and I drove out to Mystery Creek Road to survey for non-native
plants in the Swan Lake Fire. We first followed an old dozer line to
60.58368, -150.31352. This line had apparently not been used by recent
fire operations and was free of non-native plants.

We walked down the other direction down the line to 60.58134,-150.31758
and again we saw no non-native plants. We saw only *Phleum pratense* and
*Plantago major* on the road. We used Survey123 to document non-native
plant occurrences.

![An old dozer line at 60.58135, -150.31763 that was quite
weed-free.](2020-08-20_dozer_line.jpg)  
An old dozer line at 60.58135, -150.31763 that was quite weed-free.

We stopped at a small gravel pit parking lot that was quite weedy. A
fuel break west of this was weed-free. We walked as far as
60.58923,-150.31615.

![A weedy gravel pit at 60.58744,
-150.31427.](2020-08-20_gravel_pit.jpg)  
A weedy gravel pit at 60.58744, -150.31427.

We walked east to a helispot at 60.58739,-150.30463. It was quite clean
in that we saw no non-native plants.

We next drove and walked to the RAWS station at 60.59851,-150.30811. We
saw no weeds here.

Back at the road we checked puddles for *Psilocarphus*, but we found
none. We did some surveying for weeds here.

## Monday, August 24

I took care of correspondence regarding *Elodea* surveys and
invoicing/billing for recent sequencing work.

I examined a plant collected at the gravel pit on Mystery Creek Road on
[Thursday](#thursday-august-20) (iNaturalist:
[57466121](https://www.inaturalist.org/observations/57466121)). It
appears to be *Veronica peregrina*, a species that I was unfamiliar
with.

Ben and I took a canoe out to Canoe Lake to look at the reed canary
grass infestation out there. We found much of it along the stream
between Canoe Lake \#1 and Canoe Lake \#2, with the biggest patch on
Canoe Lake \#2 at the portage to Canoe Lake \#3. We cut inflorescences
from most of what we saw, but we had to stop so that we would not be out
too late.

![Cutting inflorescences of *Phalaris arundinacea* along the stream
between Canoe Lake \#1 and Canoe Lake
\#2.](2020-08-24_cutting_Phalaris_at_Canoe_Lake.jpg)  
Cutting inflorescences of *Phalaris arundinacea* along the stream
between Canoe Lake \#1 and Canoe Lake \#2.

## Tuesday, August 25

To do:

  - ~~Pesticide use reporting.~~
  - ~~*Refuge Notebook*.~~
  - ~~Contact RTL regarding missing invoices.~~
  - Catch-up from yesterday.
  - Prepare for Sandpiper Lake Friday.
  - Prepare for work with HEA on Thursday.
  - Deal with plant specimens.
  - ~~Get list of DNA sequencing equipment acquisitions to Mark.~~
  - Inquire about sending Dominique’s material to UAM.
  - Data back-up.

I looked at the best imagery we have of Sandpiper Lake, aerial imagery
captured in 2016. The patches of *Elodea* could not be seen in this
imagery. I did this in response to a request in the last statewide
*Elodea* meeting to find out if *Elodea* could be seen in aerial
imagery.

![Image of Sandpiper Lake captured in
2016.](2020-08-25_Sandpiper_Lake_2016_aerial.jpg)  
Image of Sandpiper Lake captured in 2016.

I edited and sent out this week’s *Refuge Notebook* article.

``` r
## Adding catalog numbers to the bulkloader file.

loaded <- read.csv("ArctosData_35BF9F98FD.csv", stringsAsFactors=FALSE)

data1 <- read.table("a01_Arctos_bulkload.csv", sep=",", quote="\"", header=TRUE)

loaded$uuid <- substr(loaded$OTHERCATALOGNUMBERS, 6, 41)

loaded$cat_num <- sapply(strsplit(loaded$GUID, ":"), "[", 3)

loaded <- loaded[,c("uuid", "cat_num")]

data2 <- merge(data1, loaded, all.x=TRUE)

write.csv(data2, "a01_Arctos_bulkload_v2020-08-25-1249.csv", row.names=FALSE, na="")
```

An aquatic plant I collected yesterday at Canoe Lake (iNaturalist:
[57584513](https://www.inaturalist.org/observations/57584513)) is
*Caltha natans*, a species that we did not have on the Kenai National
Wildlife Refuge’s checklist. I processed this specimen to go into the
collection
([KNWR:Herb:12445](https://arctos.database.museum/guid/KNWR:Herb:12445)).

## Wednesday, August 26

To do:

  - ~~Prepare for Sandpiper Lake Friday.~~
  - ~~Prepare for work with HEA on Thursday.~~
  - ~~Inquire about sending Dominique’s material to UAM.~~
  - *Refuge Notebook* catch-up.

I selected sites for fluridone concentration monitoring, spreading the
fours sites along the lake and over a range of depths. I pre-labeled
vials, printed a map, and loaded coordinates onto a GPS for Mark and
Jake to go out on Friday.

![FasTEST sites for fluridone concentration monitoring in Sandpiper
Lake.](2020-08-26_Sandpiper_Lake_FasTEST_sites.jpg)  
FasTEST sites for fluridone concentration monitoring in Sandpiper Lake.

### Sandpiper Lake FasTEST sites

| site\_name | latitude    | longitude     |
| :--------- | :---------- | :------------ |
| FasSA1     | 61.00720005 | \-150.404666  |
| FasSA2     | 61.00473547 | \-150.405292  |
| FasSA3     | 61.00243803 | \-150.4081228 |
| FasSA4     | 61.00165954 | \-150.411796  |

I walked over to the shop and reserved ATVs for tomorrow. Later I
learned that we will not be going out tomorrow.

I saw in a report on a 1984 survey of Canoe Lake (Friedersdorff
[1986](#ref-friedersdorff_remote_1986)) that *Sagittaria cuneata* was
not observed. It was abundant and present over much of the lake among
the *Nuphar* when I visited on [Monday](#monday-august-24), so I believe
that it has been introduced relatively recently. I do not think that it
would have been missed in 1984 if it looked then like it does in 2020.

## Thursday, August 27

I worked at home today writing a *Refuge Notebook* article on Kenai
birch. I found a great new article on birch phylogeny (Wang et al.
[2020](#ref-wang_resolving_2020)).

## Monday, August 31

To do:

  - ~~Finish *Refuge Notebook* article.~~
  - Prepare fungi specimens for DNA sequencing submission.

I finished the *Refuge Notebook* article on Kenai birch.

I got together the data, notes, etc. from the whole genome sequencing
data from birches that we had done last fall and uploaded the dataset to
Zenodo (Bowser [2020](#ref-bowser_2020_4009469)).

Jake stopped by and returned the camera from his trip with Mark out to
Sandpiper Lake on Friday, August 28. The *Elodea* that had been treated
with diquat was now yellowing and brittle.

![Dense bed of *Elodea* in Sandpiper Lake on August 28 after treatment
with diquat.](2020-08-28_Sandpiper_Lake_elodea_in_water.jpg)  
Dense bed of *Elodea* in Sandpiper Lake on August 28 after treatment
with diquat.

![Brittle, yellowed *Elodea* from Sandpiper Lake on August
28.](2020-08-28_Sandpiper_Lake_elodea.jpg)  
Brittle, yellowed *Elodea* from Sandpiper Lake on August 28.

![Other aquatic plant species appeared to be affected less by the diquat
than *Elodea* was.](2020-08-28_Sandpiper_Lake_vegetation.jpg)  
Other aquatic plant species appeared to be affected less by the diquat
than *Elodea* was.

At home I downloaded the *Betula pendula* genome in FASTA format from
<https://ftp.ncbi.nlm.nih.gov/genomes/all/GCA/900/184/695/GCA_900184695.1_Bpev01/GCA_900184695.1_Bpev01_genomic.fna.gz>.

I tried running bowtie2 to build an index.

``` sh
bowtie2-build GCA_900184695.1_Bpev01_genomic.fna GCA_900184695.1_Bpev01_genomic
```

I later ran bowtie2, but this crashed later. I also think I might not be
using the correct index for *Betula pendula*.

# September

## Tuesday, September 1

I packaged fluridone water samples from Sandpiper Lake and drove them to
FedEx Kenai to go out today (FedEx tracking \#: 396397274746).

### 12:30 Canoe Lake reed canary grass discussion

I re-entered data for the earthworm specimens sent to me by Tyler
([KNWR:Inv:47](http://arctos.database.museum/guid/KNWR:Inv:47)–[KNWR:Inv:52](http://arctos.database.museum/guid/KNWR:Inv:52)),
this time using Arctos’ bulkloader so that I will not lose my work. I
photographed these specimens, cut tissue samples from their posterior
ends, and placed tissue in LifeScanner vials, and worked on entering
data into Arctos and [lifescanner.net](http://lifescanner.net).

## Wednesday, September 2

I gathered LifeScanner samples and started preparing them for shipping.

I revised and submitted this week’s *Refuge Notebook* article.

I accompanied Lynnda on a walk down the HEA powerline right of way north
of the Sterling Highway and west of Fuller Lakes Trail. She needed to
document the work that had been done here by HEA.

![Documenting the changes to the power line
access.](2020-09-02_powerline.jpg)  
Documenting the changes to the power line access.

We revisited the unfamiliar plant that Annaleese and I had seen on
[August 4](#tuesday-august-4) (iNaturalist:
[55658833](https://www.inaturalist.org/observations/55658833)). Some of
these now had inflorescences that were not mature. I photographed and
collected these. They appear to be an *Amaranthus*. We pulled all eight
or so of these plants.

I also collected a muscoid fly from the power line just above the ravine
at Fuller Lakes Trail (BOLD-KO7).

## Thursday, September 3

I think the *Amaranthus* from [yesterday](#wednesday-september-2) is
*Amaranthus retroflexus* because the appearance looks right and there
are two previous Alaskan records
([UAM:Herb:16434](https://arctos.database.museum/guid/UAM:Herb:16434),
[UAM:Herb:46957](https://arctos.database.museum/guid/UAM:Herb:46957)).
There are also five records on AKEPIC.

I prepared 29 LifeScanner specimens to go out (USPS tracking \#:
LZ991546597US, Arctos container KNWRC4610). [Some
vials](#lifescanner-vials-with-missing-data) were missing data, which I
will need to track down and send next time.

In preparation for sequencing of fungal specimens I started an
iNaturalist project
(<https://www.inaturalist.org/projects/2020-kenai-peninsula-fungal-sequencing>)
and exported the [data](#data-for-fungal-specimens-to-be-sequenced). I
took tissue samples from the 20 specimens, using lighter to sterilize
the forceps between each specimen.

Over lunch I tried looking for an ITS sequence in one of our birch
genomes. I tried VSEARCH:

Slurm script:

``` sh
#!/bin/bash
#SBATCH --job-name=vsbirch
#SBATCH -n 1 # number of nodes
#SBATCH -n 1 # number of tasks
#SBATCH -p long # parition
#SBATCH --account=bio # account code
#SBATCH --time=1-04:00:00 # requested job time D-HH:MM:SS
#SBATCH --mail-type=ALL
#SBATCH --mail-user=matt_bowser@fws.gov
#SBATCH -o 2020-09-03-1235_vsearch_matches-%j.out

module load legacy
module load genetics/vsearch-2.4.3

cd /lustre/projects/fws/fws_bio/mattbowser/projects/birch/data

vsearch --usearch_global KT309005.fas \
--db MB_BK01_refv1.fasta \
--id .90 \
--iddef 0 \
--blast6out 2020-09-03-1235_vsearch_matches.txt

module purge
```

Output:

``` sh
vsearch v2.4.3_linux_x86_64, 125.8GB RAM, 20 cores
https://github.com/torognes/vsearch

Reading file MB_BK01_refv1.fasta 100%
37304819 nt in 2281 seqs, min 467, max 49940, avg 16355
WARNING: 2172 sequences longer than 50000 nucleotides discarded.
Masking 100%
Counting unique k-mers 100%
Creating index of unique k-mers 100%
Searching 100%
Matching query sequences: 0 of 1 (0.00%)
```

So apparently VSEARCH does not search against these long contigs.

## Monday, September 8

To do:

  - Spruce data/analysis to John.

Today I received fluridone concentrations results from samples collected
at Sandpiper Lake on August 28.

### Fluridone concentrations from Sandpiper Lake in water samples collected on August 28, 2020.

| site\_name | latitude    | longitude     | date       | fluridone\_concentration\_ppb |
| :--------- | :---------- | :------------ | :--------- | ----------------------------: |
| FasSA1     | 61.00720005 | \-150.404666  | 2020-08-28 |                           8.2 |
| FasSA2     | 61.00473547 | \-150.405292  | 2020-08-28 |                           3.7 |
| FasSA3     | 61.00243803 | \-150.4081228 | 2020-08-28 |                           3.8 |
| FasSA4     | 61.00165954 | \-150.411796  | 2020-08-28 |                           3.5 |

``` r
## Get the average, etc.
fl <- c(8.2, 3.7, 3.8, 3.5)
summary(fl)
   Min. 1st Qu.  Median    Mean 3rd Qu.    Max. 
   3.50    3.65    3.75    4.80    4.90    8.20 
```

I did a little working adding records to FWSpecies.

Mark had posted video footage from the ROV in Sandpiper Lake on August
28. It showed brittle, yellow, *Elodea* that broke apart when the ROV
passed through it. Interestingly, the native *Potamogeton* species in
the images did not appear to be as heavily damaged as the *Elodea*.

![ROV footage from the northernmost bay of Sandpiper Lake on August
28.](2020-08-28-130259_Sandpiper_Lake_elodea.jpg)  
ROV footage from the northernmost bay of Sandpiper Lake on August 28.

I filled out the online submission form for fungal samples via the
[Fungal Diversity Survey](https://fundis.org/), packaged the specimens,
and drove them to FedEx Kenai (tracking \#: 815820190845).

I participated in a USFWS Vogel Lake/Miller Creek NEPA discussion call.
I plan to supply species occurrence data for this.

I imported the remaining 2020 snowshoe hare pellet count data. I also
worked on organizing the data and revising the metadata.

![Snowshoe hare pellet counts,
1984–2020.](2020-09-08_pellet_count_means_over_time.jpg)  
Snowshoe hare pellet counts, 1984–2020.

## Wednesday, September 9

To do:

  - ~~Edit and submit this week’s *Refuge Notebook* article.~~

I entered data of dates and observers from 2014 snowshoe hare work.

I edited and sent off this week’s *Refuge Notebook* article.

### KPCWMA meeting @ 10:00

I provided an update via chat: We found three species records that are,
to the best of my knowledge, new records for the Kenai Peninsula:

Sagittaria cuneata This species is native to Interior Alaska, but we
believe it is not native to Southcentral Alaska and that it was likely
introduced by users of the Swan Lake Canoe Trail.
https://www.inaturalist.org/observations/33451572
https://arctos.database.museum/guid/KNWR:Herb:12446

Amaranthus sp., I think Amaranthus retroflexus
https://www.inaturalist.org/observations/55658833
https://www.inaturalist.org/observations/55433248

Veronica peregrina https://www.inaturalist.org/observations/57466121
https://arctos.database.museum/guid/KNWR:Herb:12444

We should be submitting all of these except the Sagittaria to AKEPIC
this winter with our annual data submission.

I started to look at snowshoe hare density data in preparation for a
*Refuge Notebook* on this article for next week.

Berge and Gese ([2010](#ref-berg_relationship_2010)) found a
relationship of hare density (hares/ha) = 0.0921 X (pellets/m2yr) +
0.0524 in Wyoming.

I entered more snowshoe hare pellet count data and scanned some data
sheets.

Colin and I mixed up some herbicide for treating reed canarygrass at
Canoe Lake tomorrow.

## Thursday, September 10

<!-- 7:00- -->

I loaded up the canoe and equipment to go to Canoe Lake today.

I scanned some more snowshoe hare pellet data sheets.

Ben and I drove out to Canoe Lake, but then we reconsidered that we need
further review before applying herbicide in wilderness, so we did not
not unload anything.

Since I was already out there, I stopped by Sucker Creek to look for
freshwater mussels. Years ago someone told me that he believed that
there was another species of mussel in the stream. The stream was quite
high today. I saw no mussels.

Here I did collect a caddisfly (vial with human-readable lable BOLD-9O1,
QR code BOLD-3NLV2Y9O1) and a small fly (BOLD-8D9, BOLD-3NLWK08D9), a
duckweed, and a streamside grass.

### Data management team meeting at 13:15.

  - I asked about what to copy over to the regional repository in the
    “incoming” directory. The answer was that I could copy a snapshot
    of the whole repository at its current state.

A duckweed I collected at Sucker Creek is *Spirodela polyrhiza*
(iNaturalist:
[59163202](https://www.inaturalist.org/observations/59163202)). This had
been observed in 2016 and 2018 in Nikiski (iNaturalist:
[16015862](https://www.inaturalist.org/observations/16015862),
[4123994](https://www.inaturalist.org/observations/4123994)).

A specimen identified as *Lemna minor* from Fish Lake
([UAM:Herb:20823](https://arctos.database.museum/guid/UAM:Herb:20823))
appears to have mostly *Spirodela polyrhiza* (L.) Schleid. on it with
some *Lemna trisulca* L. There could be Lemna minor L., but I cannot
tell from the image. I annotated the specimen record on Arctos.

A grass from Sucker Creek I tentatively identified as *Torreyochloa
pallida* (iNaturalist:
[59171269](https://www.inaturalist.org/observations/59171269)), a
species not previously recorded from the Refuge.

## Friday, September 11

I worked at home this morning on entering data from the snowshare hare
pellet survey. While looking up some of the observers, I found that some
of the *Refuge Notebook* articles from 2012 are inaccessible on our
website due to broken links. I must fix this.

I worked on improving the calculations and graphics from snowshoe hare
data.

``` r
################################################################################
# Summarize snowshoe hare pellet count data.                                   #
#                                                                              #
# Author: First and last name <matt_bowser@fws.gov>                            #
# Date created: 2020-04-08                                                     #
# Date last edited: 2020-09-11                                                 #
################################################################################


# Setup ------------------------------------------------------------------------

# Load packages
library(reshape2)

# Source external scripts
#source("directory_path/script_foo.R")


# 1 Load data -----------------------------------------------------------------
## ---- load_data  

pellet_data <- read.csv("../data/final_data/observations/snowshoe_hare_pellet_counts.csv",
 stringsAsFactors=FALSE
 )
 
plot_data <- read.csv("../data/final_data/geodata/snowshoe_hare_plot_data.csv",
 stringsAsFactors=FALSE
 )

# 2 Prepare data ---------------------------------------------------------------
## ---- prep_data  

## Adding a year field for now, which should work.
pellet_data$year <- as.numeric(substr(pellet_data$date, 1, 4))

## Join the data to get the grid_name values.
pellet_data_joined <- merge(
 plot_data[,c("plot_name", "grid_name")],
 pellet_data
 )
 
## Now it would be good to fill in missing dates. Because each grid was supposed to been surveyed at about the same time each year, we should be able to use the known dates to infer rough dates for when these were usually done.

## First, for each grid we should get the Julian dates when they were usually surveyed.
has_date <- nchar(pellet_data_joined$date) == 10
grids_dates <- unique(pellet_data_joined[has_date, c("grid_name", "date", "year")])
grids_dates$origin <- as.Date(paste0(grids_dates$year, "-01-01"))
grids_dates$julian <- NA
for (this_row in 1:nrow(grids_dates))
 {
 grids_dates$julian[this_row] <- julian(as.Date(grids_dates$date[this_row]), origin=grids_dates$origin[this_row]) + 1
 }
grids_mean_dates <- aggregate(grids_dates$julian, by=list(grids_dates$grid_name), mean)
names(grids_mean_dates) <- c("grid_name", "julian")
grids_mean_dates$julian <- round(grids_mean_dates$julian)

## Now we should fill in missing dates with the average dates.
pellet_data_joined <- merge(pellet_data_joined, grids_mean_dates, by="grid_name", all.x=TRUE)
pellet_data_joined$date_est <- as.Date(paste0(pellet_data_joined$year, pellet_data_joined$julian), format="%Y%j")
no_date <- nchar(pellet_data_joined$date) == 4
pellet_data_joined$date[no_date] <- as.character(pellet_data_joined$date_est[no_date])
pellet_data_joined <- pellet_data_joined[,1:7] 

## Convert the plot names to plot numbers.
plot_number <- strsplit(pellet_data_joined$plot_name,
 "-"
 )
plot_number <- sapply(plot_number,
 "[",
 2
 ) 
pellet_data_joined$plot_name <- as.numeric(plot_number)
 
## Reshape the data into a grid_name X plot number X year array.
pellet_data_molten <- melt(pellet_data_joined,
 measure.vars="pellet_count"
 )

pellet_data_array <- acast(
 pellet_data_molten,
 grid_name ~ plot_name ~ year,
 fun.aggregate=sum,
 fill=9999 ## Temporary fill value.
 )
pellet_data_array[pellet_data_array==9999] <- NA
 
n_grids <- dim(pellet_data_array)[1]
n_plots <- dim(pellet_data_array)[2]
n_years <- dim(pellet_data_array)[3]

## Set to NA any plot for which the previous year's count was NA.
for (this_grid in 1:n_grids)
 {
 for (this_plot in 1:n_plots)
  {
  for (this_year in n_years:2)
   if(is.na(pellet_data_array[this_grid,this_plot,this_year-1]))
    {
    pellet_data_array[this_grid,this_plot,this_year] <- NA
    }
  }
 }
 
## Also set all year 1 observations to NA.
pellet_data_array[,,1] <- NA

## Now get averages.
pellet_count_means <- as.data.frame(apply(pellet_data_array,
 c(1,3),
 mean,
 na.rm=TRUE
 ))

## Reformatting so that start and end dates can be incorporated. 
pellet_count_means$grid_name <- row.names(pellet_count_means)
pellet_count_means_molten <- melt(pellet_count_means, id.vars="grid_name")
names(pellet_count_means_molten)[2] <- "year" 
pellet_count_means_molten$variable <- "mean_pellets_m2"

## We need start and end dates for each grid and year.
pellet_data_joined$end_date <- pellet_data_joined$date
## Getting the start dates will take a little more work.
pellet_data_dates <- unique(pellet_data_joined[c("grid_name", "year", "end_date")])
pellet_data_dates$year <- pellet_data_dates$year + 1
names(pellet_data_dates)[3] <- "start_date"
pellet_data_joined <- merge(pellet_data_joined, pellet_data_dates, by=c("grid_name", "year"), all.x=TRUE)
pellet_data_dates <- unique(pellet_data_joined[c("grid_name", "year", "start_date", "end_date")])

pellet_count_annual_data <- merge(pellet_count_means_molten,
 pellet_data_dates,
 by=c("grid_name", "year"),
 all.x=TRUE
 )
## Dropping the no data years.
#pellet_count_annual_data <- pellet_count_annual_data[!is.nan(pellet_count_annual_data$value),]

## Now calculate the time differences between start and end dates.
pellet_count_annual_data$time_dif_days <- as.numeric(as.Date(pellet_count_annual_data$end_date) - as.Date(pellet_count_annual_data$start_date))
pellet_count_annual_data$time_dif_years <- pellet_count_annual_data$time_dif_days/365.2422

## Calculate pellets per m2 per year.
pellet_count_annual_data$pellets_m2yr <- pellet_count_annual_data$value/pellet_count_annual_data$time_dif_years

## Cleaning up this data frame a bit for use later.
pellet_count_annual_data$year <-as.numeric(as.character(pellet_count_annual_data$year))
pellet_count_annual_data$start_date <- as.Date(pellet_count_annual_data$start_date)
pellet_count_annual_data$end_date <- as.Date(pellet_count_annual_data$end_date)
names(pellet_count_annual_data)[3] <- "pellets_m2"
pellet_count_annual_data <- pellet_count_annual_data[,c(1:3,5:9)]

## It might be useful to calculate a grand mean for clarity on the graph.
data_agg <- aggregate(pellet_count_annual_data$pellets_m2yr,
 by=list(pellet_count_annual_data$year),
 mean,
 na.rm=TRUE
 )
names(data_agg) <-c("year", "pellets_m2yr")
data_agg$start_date <- aggregate(pellet_count_annual_data$start_date,
 by=list(pellet_count_annual_data$year),
 mean,
 na.rm=TRUE
 )$x
data_agg$end_date <- aggregate(pellet_count_annual_data$end_date,
 by=list(pellet_count_annual_data$year),
 mean,
 na.rm=TRUE
 )$x 
data_agg <- data_agg[2:nrow(data_agg),]
 
## Plot.

plot_title <- paste0("Kenai NWR snowshoe hare pellet counts, ",
 min(pellet_count_annual_data$year),
 "–",
 max(pellet_count_annual_data$year)
 )
 
file_name <- paste0("../documents/",
 as.Date(Sys.time()),
 "_pellet_count_over_time.pdf"
 )
 
lcol <- "#0000001F"
lwd=2
#lty=c("11",
# "31",
# "12",
# "44",
# "5111",
# "62",
# "117111"
# )
lty=rep(1, 7)

grid_name <- levels(as.factor(pellet_count_annual_data$grid_name))
this_grid <- 1

sl <- pellet_count_annual_data$grid_name == grid_name[this_grid]
sln <- pellet_count_annual_data$grid_name == grid_name[this_grid] & !is.nan(pellet_count_annual_data$pellets_m2yr)

pdf(file=file_name,
 width=6,
 height=5
 )
par(mar=c(5,5,3,1))
plot(pellet_count_annual_data$start_date[sl],
 pellet_count_annual_data$pellets_m2yr[sl],
 type="n",
 ylim=c(0, max(pellet_count_annual_data$pellets_m2yr, na.rm=TRUE)),
 main=plot_title,
 xlab="year",
 ylab=expression(paste("hare pellets per m"^"2", " per year"))
 )

## Draw segments showing pellet densities. 
segments(pellet_count_annual_data$start_date[sln],
 pellet_count_annual_data$pellets_m2yr[sln],
 pellet_count_annual_data$end_date[sln],
 pellet_count_annual_data$pellets_m2yr[sln],
 col=lcol,
 lwd=lwd,
 lty=lty[this_grid]
 )
## connect non-missing segments.
for (this_row in 2:nrow(pellet_count_annual_data))
 {
 if (sln[this_row-1] & sln[this_row])
  {
  segments(pellet_count_annual_data$start_date[this_row],
   pellet_count_annual_data$pellets_m2yr[this_row-1],
   pellet_count_annual_data$start_date[this_row],
   pellet_count_annual_data$pellets_m2yr[this_row],
   col=lcol,
   lwd=lwd,
   lty=lty[this_grid]
   )
  }
 }
 
for (this_grid in 2:n_grids)
 {
 sln <- pellet_count_annual_data$grid_name == grid_name[this_grid] & !is.nan(pellet_count_annual_data$pellets_m2yr)
## Draw segments showing pellet densities. 
segments(pellet_count_annual_data$start_date[sln],
 pellet_count_annual_data$pellets_m2yr[sln],
 pellet_count_annual_data$end_date[sln],
 pellet_count_annual_data$pellets_m2yr[sln],
 col=lcol,
 lwd=lwd,
 lty=lty[this_grid]
 )
## connect non-missing segments.
for (this_row in 2:nrow(pellet_count_annual_data))
 {
 if (sln[this_row-1] & sln[this_row])
  {
  segments(pellet_count_annual_data$start_date[this_row],
   pellet_count_annual_data$pellets_m2yr[this_row-1],
   pellet_count_annual_data$start_date[this_row],
   pellet_count_annual_data$pellets_m2yr[this_row],
   col=lcol,
   lwd=lwd,
   lty=lty[this_grid]
   )
  }
 } 
 }
 ## Now plot the grand mean lines.
 segments(data_agg$start_date,
  data_agg$pellets_m2yr,
  data_agg$end_date,
  data_agg$pellets_m2yr,
  col="#000000",
  lwd=4,
  lty=1
  )
 for (this_row in 2:nrow(data_agg))
 {
  segments(data_agg$start_date[this_row],
   data_agg$pellets_m2yr[this_row-1],
   data_agg$start_date[this_row],
   data_agg$pellets_m2yr[this_row],
   col="#000000",
   lwd=4,
   lty=1
   )
 } 
dev.off() 

## Save means summaries.
file_name <- paste0("../documents/",
 as.Date(Sys.time()),
 "_pellet_count_means_over_time.csv"
 )
write.csv(pellet_count_annual_data,
 file_name,
 row.names=FALSE
 ) 

## Save the grand mean summary, also.
 file_name <- paste0("../documents/",
 as.Date(Sys.time()),
 "_grand_means_over_time.csv"
 )
write.csv(data_agg,
 file_name,
 row.names=FALSE
 ) 
```

![Kenai National Wildlife Refuge snowshoe hare pellet counts,
1983–2020.](2020-09-11_pellet_count_over_time.jpg)  
Kenai National Wildlife Refuge snowshoe hare pellet counts, 1983–2020.

## Monday, September 14

I worked on calculating hare densities from Krebs et
al. ([1987](#ref-krebs_estimation_1987),
[2001](#ref-krebs_estimating_2001)), Murray et
al. ([2002](#ref-murray_estimating_2002)), Berg and Gese
([2010](#ref-berg_relationship_2010)), and Mills
([2005](#ref-mills_pellet_2005)).

``` r
################################################################################
# Summarize snowshoe hare pellet count data.                                   #
#                                                                              #
# Author: First and last name <matt_bowser@fws.gov>                            #
# Date created: 2020-04-08                                                     #
# Date last edited: 2020-09-14                                                 #
################################################################################


# Setup ------------------------------------------------------------------------

# Load packages
library(reshape2)

# Source external scripts
#source("directory_path/script_foo.R")


# 1 Load data -----------------------------------------------------------------
## ---- load_data  

pellet_data <- read.csv("../data/final_data/observations/snowshoe_hare_pellet_counts.csv",
 stringsAsFactors=FALSE
 )
 
plot_data <- read.csv("../data/final_data/geodata/snowshoe_hare_plot_data.csv",
 stringsAsFactors=FALSE
 )
 
parameter_data <- read.csv("../data/final_data/parameters/hare_density_parameters.csv") 

# 2 Prepare data ---------------------------------------------------------------
## ---- prep_data  

## Adding a year field for now, which should work.
pellet_data$year <- as.numeric(substr(pellet_data$date, 1, 4))

## Join the data to get the grid_name values.
pellet_data_joined <- merge(
 plot_data[,c("plot_name", "grid_name")],
 pellet_data
 )
 
## Now it would be good to fill in missing dates. Because each grid was supposed to been surveyed at about the same time each year, we should be able to use the known dates to infer rough dates for when these were usually done.

## First, for each grid we should get the Julian dates when they were usually surveyed.
has_date <- nchar(pellet_data_joined$date) == 10
grids_dates <- unique(pellet_data_joined[has_date, c("grid_name", "date", "year")])
grids_dates$origin <- as.Date(paste0(grids_dates$year, "-01-01"))
grids_dates$julian <- NA
for (this_row in 1:nrow(grids_dates))
 {
 grids_dates$julian[this_row] <- julian(as.Date(grids_dates$date[this_row]), origin=grids_dates$origin[this_row]) + 1
 }
grids_mean_dates <- aggregate(grids_dates$julian, by=list(grids_dates$grid_name), mean)
names(grids_mean_dates) <- c("grid_name", "julian")
grids_mean_dates$julian <- round(grids_mean_dates$julian)

## Now we should fill in missing dates with the average dates.
pellet_data_joined <- merge(pellet_data_joined, grids_mean_dates, by="grid_name", all.x=TRUE)
pellet_data_joined$date_est <- as.Date(paste0(pellet_data_joined$year, pellet_data_joined$julian), format="%Y%j")
no_date <- nchar(pellet_data_joined$date) == 4
pellet_data_joined$date[no_date] <- as.character(pellet_data_joined$date_est[no_date])
pellet_data_joined <- pellet_data_joined[,1:7] 

## Convert the plot names to plot numbers.
plot_number <- strsplit(pellet_data_joined$plot_name,
 "-"
 )
plot_number <- sapply(plot_number,
 "[",
 2
 ) 
pellet_data_joined$plot_name <- as.numeric(plot_number)
 
## Reshape the data into a grid_name X plot number X year array.
pellet_data_molten <- melt(pellet_data_joined,
 measure.vars="pellet_count"
 )

pellet_data_array <- acast(
 pellet_data_molten,
 grid_name ~ plot_name ~ year,
 fun.aggregate=sum,
 fill=9999 ## Temporary fill value.
 )
pellet_data_array[pellet_data_array==9999] <- NA
 
n_grids <- dim(pellet_data_array)[1]
n_plots <- dim(pellet_data_array)[2]
n_years <- dim(pellet_data_array)[3]

## Set to NA any plot for which the previous year's count was NA.
for (this_grid in 1:n_grids)
 {
 for (this_plot in 1:n_plots)
  {
  for (this_year in n_years:2)
   if(is.na(pellet_data_array[this_grid,this_plot,this_year-1]))
    {
    pellet_data_array[this_grid,this_plot,this_year] <- NA
    }
  }
 }
 
## Also set all year 1 observations to NA.
pellet_data_array[,,1] <- NA

## Now get averages.
pellet_count_means <- as.data.frame(apply(pellet_data_array,
 c(1,3),
 mean,
 na.rm=TRUE
 ))

## Reformatting so that start and end dates can be incorporated. 
pellet_count_means$grid_name <- row.names(pellet_count_means)
pellet_count_means_molten <- melt(pellet_count_means, id.vars="grid_name")
names(pellet_count_means_molten)[2] <- "year" 
pellet_count_means_molten$variable <- "mean_pellets_m2"

## We need start and end dates for each grid and year.
pellet_data_joined$end_date <- pellet_data_joined$date
## Getting the start dates will take a little more work.
pellet_data_dates <- unique(pellet_data_joined[c("grid_name", "year", "end_date")])
pellet_data_dates$year <- pellet_data_dates$year + 1
names(pellet_data_dates)[3] <- "start_date"
pellet_data_joined <- merge(pellet_data_joined, pellet_data_dates, by=c("grid_name", "year"), all.x=TRUE)
pellet_data_dates <- unique(pellet_data_joined[c("grid_name", "year", "start_date", "end_date")])

pellet_count_annual_data <- merge(pellet_count_means_molten,
 pellet_data_dates,
 by=c("grid_name", "year"),
 all.x=TRUE
 )
## Dropping the no data years.
#pellet_count_annual_data <- pellet_count_annual_data[!is.nan(pellet_count_annual_data$value),]

## Now calculate the time differences between start and end dates.
pellet_count_annual_data$time_dif_days <- as.numeric(as.Date(pellet_count_annual_data$end_date) - as.Date(pellet_count_annual_data$start_date))
pellet_count_annual_data$time_dif_years <- pellet_count_annual_data$time_dif_days/365.2422

## Calculate pellets per m2 per year.
pellet_count_annual_data$pellets_m2yr <- pellet_count_annual_data$value/pellet_count_annual_data$time_dif_years

## Cleaning up this data frame a bit for use later.
pellet_count_annual_data$year <-as.numeric(as.character(pellet_count_annual_data$year))
pellet_count_annual_data$start_date <- as.Date(pellet_count_annual_data$start_date)
pellet_count_annual_data$end_date <- as.Date(pellet_count_annual_data$end_date)
names(pellet_count_annual_data)[3] <- "pellets_m2"
pellet_count_annual_data <- pellet_count_annual_data[,c(1:3,5:9)]

## It might be useful to calculate a grand mean for clarity on the graph.
data_agg <- aggregate(pellet_count_annual_data$pellets_m2yr,
 by=list(pellet_count_annual_data$year),
 mean,
 na.rm=TRUE
 )
names(data_agg) <-c("year", "pellets_m2yr")
data_agg$start_date <- aggregate(pellet_count_annual_data$start_date,
 by=list(pellet_count_annual_data$year),
 mean,
 na.rm=TRUE
 )$x
data_agg$end_date <- aggregate(pellet_count_annual_data$end_date,
 by=list(pellet_count_annual_data$year),
 mean,
 na.rm=TRUE
 )$x 
data_agg <- data_agg[2:nrow(data_agg),]

## Now add density estimates.

## Functions from papers.
Krebs_et_al_1987 <- function(pellets_m2yr)
 {
 ## First convert pellets_m2yr to pellets per 0.155 m2 per year.
 pellets_0155 <- pellets_m2yr*0.155
 
 hares_ha <- 0.27*pellets_0155 + 0.42
 hares_ha
 }
Krebs_et_al_1987(data_agg$pellets_m2yr)

Krebs_et_al_2001 <- function(pellets_m2yr)
 {
 ## First convert pellets_m2yr to pellets per 0.155 m2 per year.
 pellets_0155 <- pellets_m2yr*0.155
 
 ## Apply regression.
 hares_ha <- exp(-1.203 + 0.889*log(pellets_0155))
 
 ## Apply correction factor.
 hares_ha <- 1.567*hares_ha
 hares_ha
 }
Krebs_et_al_2001(data_agg$pellets_m2yr)

Murray_et_al_2002 <- function(pellets_m2yr)
 {
 pellets_m2yr <- log(pellets_m2yr + 1/6)
 
 hares_ha <- 1.423 * (2.077 - exp(-.662*pellets_m2yr))
 hares_ha
 
 hares_ha <- exp(hares_ha) - 1/6
 hares_ha
 }
Murray_et_al_2002(data_agg$pellets_m2yr)

Mills_et_al_2005_Seeley <- function(pellets_m2yr)
 {
 ## First convert pellets_m2yr to pellets per 0.155 m2 per year.
 pellets_0155 <- pellets_m2yr*0.155
 
 ## Apply regression.
 hares_ha <- exp(-1.67 + 0.77*log(pellets_0155))
 
 ## Apply correction factor.
 hares_ha <- 1.567*hares_ha
 hares_ha
 }
Mills_et_al_2005_Seeley(data_agg$pellets_m2yr)

Mills_et_al_2005_Tally <- function(pellets_m2yr)
 {
 ## First convert pellets_m2yr to pellets per 0.155 m2 per year.
 pellets_0155 <- pellets_m2yr*0.155
 
 ## Apply regression.
 hares_ha <- exp(-1.14 + 0.63*log(pellets_0155))
 
 ## Apply correction factor.
 hares_ha <- 1.567*hares_ha
 hares_ha
 }
Mills_et_al_2005_Tally(data_agg$pellets_m2yr)

Berge_and_Gese_2010 <- function(pellets_m2yr)
 {
 hares_ha <- 0.0921*pellets_m2yr + 0.0524
 hares_ha
 }
Berge_and_Gese_2010(data_agg$pellets_m2yr)
 



## Plot.

plot_title <- paste0("Kenai NWR snowshoe hare pellet counts, ",
 min(pellet_count_annual_data$year),
 "–",
 max(pellet_count_annual_data$year)
 )
 
file_name <- paste0("../documents/",
 as.Date(Sys.time()),
 "_pellet_count_over_time.pdf"
 )
 
lcol <- "#0000001F"
lwd=2
#lty=c("11",
# "31",
# "12",
# "44",
# "5111",
# "62",
# "117111"
# )
lty=rep(1, 7)

grid_name <- levels(as.factor(pellet_count_annual_data$grid_name))
this_grid <- 1

sl <- pellet_count_annual_data$grid_name == grid_name[this_grid]
sln <- pellet_count_annual_data$grid_name == grid_name[this_grid] & !is.nan(pellet_count_annual_data$pellets_m2yr)

pdf(file=file_name,
 width=6,
 height=5
 )
par(mar=c(5,5,3,1))
plot(pellet_count_annual_data$start_date[sl],
 pellet_count_annual_data$pellets_m2yr[sl],
 type="n",
 ylim=c(0, max(pellet_count_annual_data$pellets_m2yr, na.rm=TRUE)),
 main=plot_title,
 xlab="year",
 ylab=expression(paste("hare pellets per m"^"2", " per year"))
 )

## Draw segments showing pellet densities. 
segments(pellet_count_annual_data$start_date[sln],
 pellet_count_annual_data$pellets_m2yr[sln],
 pellet_count_annual_data$end_date[sln],
 pellet_count_annual_data$pellets_m2yr[sln],
 col=lcol,
 lwd=lwd,
 lty=lty[this_grid]
 )
## connect non-missing segments.
for (this_row in 2:nrow(pellet_count_annual_data))
 {
 if (sln[this_row-1] & sln[this_row])
  {
  segments(pellet_count_annual_data$start_date[this_row],
   pellet_count_annual_data$pellets_m2yr[this_row-1],
   pellet_count_annual_data$start_date[this_row],
   pellet_count_annual_data$pellets_m2yr[this_row],
   col=lcol,
   lwd=lwd,
   lty=lty[this_grid]
   )
  }
 }
 
for (this_grid in 2:n_grids)
 {
 sln <- pellet_count_annual_data$grid_name == grid_name[this_grid] & !is.nan(pellet_count_annual_data$pellets_m2yr)
## Draw segments showing pellet densities. 
segments(pellet_count_annual_data$start_date[sln],
 pellet_count_annual_data$pellets_m2yr[sln],
 pellet_count_annual_data$end_date[sln],
 pellet_count_annual_data$pellets_m2yr[sln],
 col=lcol,
 lwd=lwd,
 lty=lty[this_grid]
 )
## connect non-missing segments.
for (this_row in 2:nrow(pellet_count_annual_data))
 {
 if (sln[this_row-1] & sln[this_row])
  {
  segments(pellet_count_annual_data$start_date[this_row],
   pellet_count_annual_data$pellets_m2yr[this_row-1],
   pellet_count_annual_data$start_date[this_row],
   pellet_count_annual_data$pellets_m2yr[this_row],
   col=lcol,
   lwd=lwd,
   lty=lty[this_grid]
   )
  }
 } 
 }
 ## Now plot the grand mean lines.
 segments(data_agg$start_date,
  data_agg$pellets_m2yr,
  data_agg$end_date,
  data_agg$pellets_m2yr,
  col="#000000",
  lwd=4,
  lty=1
  )
 for (this_row in 2:nrow(data_agg))
 {
  segments(data_agg$start_date[this_row],
   data_agg$pellets_m2yr[this_row-1],
   data_agg$start_date[this_row],
   data_agg$pellets_m2yr[this_row],
   col="#000000",
   lwd=4,
   lty=1
   )
 } 
dev.off() 

## Save means summaries.
file_name <- paste0("../documents/",
 as.Date(Sys.time()),
 "_pellet_count_means_over_time.csv"
 )
write.csv(pellet_count_annual_data,
 file_name,
 row.names=FALSE
 ) 

## Save the grand mean summary, also.
 file_name <- paste0("../documents/",
 as.Date(Sys.time()),
 "_grand_means_over_time.csv"
 )
write.csv(data_agg,
 file_name,
 row.names=FALSE
 ) 
```

## Tuesday, September 15

I repackaged LifeScanner specimens to go out, filled out the customs
form, etc. (USPS tracking \#: LZ991668352US).

I continued with the hare data, also adding the linear regression of
McCann et al ([2008](#ref-McCann_using_2008))

I found and scanned a file summarizing snowshoe hare data and presenting
density estimates from program CAPTURE (Otis et al.
[1978](#ref-otis_statistical_1978)). Of the models in the literature,
the closest match for our data appears to be that of Krebes et
al. ([2001](#ref-krebs_estimating_2001)), which produced estimates
quite close to what was estimated by program CAPTURE.

``` r
################################################################################
# Summarize snowshoe hare pellet count data.                                   #
#                                                                              #
# Author: First and last name <matt_bowser@fws.gov>                            #
# Date created: 2020-04-08                                                     #
# Date last edited: 2020-09-15                                                 #
################################################################################


# Setup ------------------------------------------------------------------------

# Load packages
library(reshape2)

# Source external scripts
#source("directory_path/script_foo.R")


# 1 Load data -----------------------------------------------------------------
## ---- load_data  

pellet_data <- read.csv("../data/final_data/observations/snowshoe_hare_pellet_counts.csv",
 stringsAsFactors=FALSE
 )
 
plot_data <- read.csv("../data/final_data/geodata/snowshoe_hare_plot_data.csv",
 stringsAsFactors=FALSE
 )

# 2 Prepare data ---------------------------------------------------------------
## ---- prep_data  

## Adding a year field for now, which should work.
pellet_data$year <- as.numeric(substr(pellet_data$date, 1, 4))

## Join the data to get the grid_name values.
pellet_data_joined <- merge(
 plot_data[,c("plot_name", "grid_name")],
 pellet_data
 )
 
## Now it would be good to fill in missing dates. Because each grid was supposed to been surveyed at about the same time each year, we should be able to use the known dates to infer rough dates for when these were usually done.

## First, for each grid we should get the Julian dates when they were usually surveyed.
has_date <- nchar(pellet_data_joined$date) == 10
grids_dates <- unique(pellet_data_joined[has_date, c("grid_name", "date", "year")])
grids_dates$origin <- as.Date(paste0(grids_dates$year, "-01-01"))
grids_dates$julian <- NA
for (this_row in 1:nrow(grids_dates))
 {
 grids_dates$julian[this_row] <- julian(as.Date(grids_dates$date[this_row]), origin=grids_dates$origin[this_row]) + 1
 }
grids_mean_dates <- aggregate(grids_dates$julian, by=list(grids_dates$grid_name), mean)
names(grids_mean_dates) <- c("grid_name", "julian")
grids_mean_dates$julian <- round(grids_mean_dates$julian)

## Now we should fill in missing dates with the average dates.
pellet_data_joined <- merge(pellet_data_joined, grids_mean_dates, by="grid_name", all.x=TRUE)
pellet_data_joined$date_est <- as.Date(paste0(pellet_data_joined$year, pellet_data_joined$julian), format="%Y%j")
no_date <- nchar(pellet_data_joined$date) == 4
pellet_data_joined$date[no_date] <- as.character(pellet_data_joined$date_est[no_date])
pellet_data_joined <- pellet_data_joined[,1:7] 

## Convert the plot names to plot numbers.
plot_number <- strsplit(pellet_data_joined$plot_name,
 "-"
 )
plot_number <- sapply(plot_number,
 "[",
 2
 ) 
pellet_data_joined$plot_name <- as.numeric(plot_number)
 
## Reshape the data into a grid_name X plot number X year array.
pellet_data_molten <- melt(pellet_data_joined,
 measure.vars="pellet_count"
 )

pellet_data_array <- acast(
 pellet_data_molten,
 grid_name ~ plot_name ~ year,
 fun.aggregate=sum,
 fill=9999 ## Temporary fill value.
 )
pellet_data_array[pellet_data_array==9999] <- NA
 
n_grids <- dim(pellet_data_array)[1]
n_plots <- dim(pellet_data_array)[2]
n_years <- dim(pellet_data_array)[3]

## Set to NA any plot for which the previous year's count was NA.
for (this_grid in 1:n_grids)
 {
 for (this_plot in 1:n_plots)
  {
  for (this_year in n_years:2)
   if(is.na(pellet_data_array[this_grid,this_plot,this_year-1]))
    {
    pellet_data_array[this_grid,this_plot,this_year] <- NA
    }
  }
 }
 
## Also set all year 1 observations to NA.
pellet_data_array[,,1] <- NA

## Now get averages.
pellet_count_means <- as.data.frame(apply(pellet_data_array,
 c(1,3),
 mean,
 na.rm=TRUE
 ))

## Reformatting so that start and end dates can be incorporated. 
pellet_count_means$grid_name <- row.names(pellet_count_means)
pellet_count_means_molten <- melt(pellet_count_means, id.vars="grid_name")
names(pellet_count_means_molten)[2] <- "year" 
pellet_count_means_molten$variable <- "mean_pellets_m2"

## We need start and end dates for each grid and year.
pellet_data_joined$end_date <- pellet_data_joined$date
## Getting the start dates will take a little more work.
pellet_data_dates <- unique(pellet_data_joined[c("grid_name", "year", "end_date")])
pellet_data_dates$year <- pellet_data_dates$year + 1
names(pellet_data_dates)[3] <- "start_date"
pellet_data_joined <- merge(pellet_data_joined, pellet_data_dates, by=c("grid_name", "year"), all.x=TRUE)
pellet_data_dates <- unique(pellet_data_joined[c("grid_name", "year", "start_date", "end_date")])

pellet_count_annual_data <- merge(pellet_count_means_molten,
 pellet_data_dates,
 by=c("grid_name", "year"),
 all.x=TRUE
 )
## Dropping the no data years.
#pellet_count_annual_data <- pellet_count_annual_data[!is.nan(pellet_count_annual_data$value),]

## Now calculate the time differences between start and end dates.
pellet_count_annual_data$time_dif_days <- as.numeric(as.Date(pellet_count_annual_data$end_date) - as.Date(pellet_count_annual_data$start_date))
pellet_count_annual_data$time_dif_years <- pellet_count_annual_data$time_dif_days/365.2422

## Calculate pellets per m2 per year.
pellet_count_annual_data$pellets_m2yr <- pellet_count_annual_data$value/pellet_count_annual_data$time_dif_years

## Cleaning up this data frame a bit for use later.
pellet_count_annual_data$year <-as.numeric(as.character(pellet_count_annual_data$year))
pellet_count_annual_data$start_date <- as.Date(pellet_count_annual_data$start_date)
pellet_count_annual_data$end_date <- as.Date(pellet_count_annual_data$end_date)
names(pellet_count_annual_data)[3] <- "pellets_m2"
pellet_count_annual_data <- pellet_count_annual_data[,c(1:3,5:9)]

## It might be useful to calculate a grand mean for clarity on the graph.
data_agg <- aggregate(pellet_count_annual_data$pellets_m2yr,
 by=list(pellet_count_annual_data$year),
 mean,
 na.rm=TRUE
 )
names(data_agg) <-c("year", "pellets_m2yr")
data_agg$start_date <- aggregate(pellet_count_annual_data$start_date,
 by=list(pellet_count_annual_data$year),
 mean,
 na.rm=TRUE
 )$x
data_agg$end_date <- aggregate(pellet_count_annual_data$end_date,
 by=list(pellet_count_annual_data$year),
 mean,
 na.rm=TRUE
 )$x 
data_agg <- data_agg[2:nrow(data_agg),]

## Now add density estimates.

## Functions from papers.
Krebs_et_al_1987 <- function(pellets_m2yr)
 {
 ## First convert pellets_m2yr to pellets per 0.155 m2 per year.
 pellets_0155 <- pellets_m2yr*0.155
 
 hares_ha <- 0.27*pellets_0155 + 0.42
 hares_ha
 }
data_agg$Krebs_et_al_1987 <- Krebs_et_al_1987(data_agg$pellets_m2yr)

Krebs_et_al_2001 <- function(pellets_m2yr)
 {
 ## First convert pellets_m2yr to pellets per 0.155 m2 per year.
 pellets_0155 <- pellets_m2yr*0.155
 
 ## Apply regression.
 hares_ha <- exp(-1.203 + 0.889*log(pellets_0155))
 
 ## Apply correction factor.
 hares_ha <- 1.567*hares_ha
 hares_ha
 }
data_agg$Krebs_et_al_2001 <- Krebs_et_al_2001(data_agg$pellets_m2yr)

Murray_et_al_2002 <- function(pellets_m2yr)
 {
 pellets_m2yr <- log(pellets_m2yr + 1/6)
 
 hares_ha <- 1.423 * (2.077 - exp(-.662*pellets_m2yr))
 hares_ha
 
 hares_ha <- exp(hares_ha) - 1/6
 hares_ha
 }
#data_agg$Murray_et_al_2002 <- Murray_et_al_2002(data_agg$pellets_m2yr)
## This estimate is an order of magnitude higher than the other estimates, so I am dropping it. I may be getting the formula wrong.

Mills_et_al_2005_Seeley <- function(pellets_m2yr)
 {
 ## First convert pellets_m2yr to pellets per 0.155 m2 per year.
 pellets_0155 <- pellets_m2yr*0.155
 
 ## Apply regression.
 hares_ha <- exp(-1.67 + 0.77*log(pellets_0155))
 
 ## Apply correction factor.
 hares_ha <- 1.567*hares_ha
 hares_ha
 }
data_agg$Mills_et_al_2005_Seeley <- Mills_et_al_2005_Seeley(data_agg$pellets_m2yr)

Mills_et_al_2005_Tally <- function(pellets_m2yr)
 {
 ## First convert pellets_m2yr to pellets per 0.155 m2 per year.
 pellets_0155 <- pellets_m2yr*0.155
 
 ## Apply regression.
 hares_ha <- exp(-1.14 + 0.63*log(pellets_0155))
 
 ## Apply correction factor.
 hares_ha <- 1.567*hares_ha
 hares_ha
 }
data_agg$Mills_et_al_2005_Tally <- Mills_et_al_2005_Tally(data_agg$pellets_m2yr)

McCann_et_al_2008 <- function(pellets_m2yr)
 {
 hares_ha <- 0.060*pellets_m2yr + 0.398
 hares_ha
 }
data_agg$McCann_et_al_2008 <- McCann_et_al_2008(data_agg$pellets_m2yr) 

Berge_and_Gese_2010 <- function(pellets_m2yr)
 {
 hares_ha <- 0.0921*pellets_m2yr + 0.0524
 hares_ha
 }
data_agg$Berge_and_Gese_2010 <- Berge_and_Gese_2010(data_agg$pellets_m2yr)
 

## Plot.

## Flag to use metric or standard units.
#units <- "metric"
units <- "standard"

plot_title <- paste0("Kenai NWR snowshoe hare pellet counts, ",
 min(pellet_count_annual_data$year),
 "–",
 max(pellet_count_annual_data$year)
 )

ylab <- expression(paste("hare pellets per m"^"2", " per year"))
dlab <- "estimated hare density (hares per hectare)"

density_ticks <- 0:10

if (units == "standard")
 {
 ylab <- expression(paste("hare pellets per ft"^"2", " per year"))
 pellet_count_annual_data$pellets_m2yr <- pellet_count_annual_data$pellets_m2yr/10.7639
 data_agg$pellets_m2yr <- data_agg$pellets_m2yr/10.7639
 dlab <- "estimated hare density (hares per acre)"
 data_agg$Krebs_et_al_2001 <- data_agg$Krebs_et_al_2001/2.47105
 density_ticks <- (0:10)/2
 }
 
file_name <- paste0("../documents/",
 as.Date(Sys.time()),
 "_pellet_count_over_time.pdf"
 )
 
lcol <- "#0000001F"
lwd=1.5
lty=rep(1, 7)

grid_name <- levels(as.factor(pellet_count_annual_data$grid_name))
this_grid <- 1

sl <- pellet_count_annual_data$grid_name == grid_name[this_grid]
sln <- pellet_count_annual_data$grid_name == grid_name[this_grid] & !is.nan(pellet_count_annual_data$pellets_m2yr)

ylim <- c(0, max(pellet_count_annual_data$pellets_m2yr, na.rm=TRUE))

pdf(file=file_name,
 width=6,
 height=5
 )
par(mar=c(4,4,2,4))
plot(pellet_count_annual_data$start_date[sl],
 pellet_count_annual_data$pellets_m2yr[sl],
 type="n",
 ylim=ylim,
 main=plot_title,
 xlab="",
 ylab=""
 )
mtext(text="year",
 side=1,
 line=2
 )
mtext(text=ylab,
 side=2,
 line=2
 )
mtext(text=dlab,
  side=4,
  line=2
  )
## Draw segments showing pellet densities. 
segments(pellet_count_annual_data$start_date[sln],
 pellet_count_annual_data$pellets_m2yr[sln],
 pellet_count_annual_data$end_date[sln],
 pellet_count_annual_data$pellets_m2yr[sln],
 col=lcol,
 lwd=lwd,
 lty=lty[this_grid]
 )
## connect non-missing segments.
for (this_row in 2:nrow(pellet_count_annual_data))
 {
 if (sln[this_row-1] & sln[this_row])
  {
  segments(pellet_count_annual_data$start_date[this_row],
   pellet_count_annual_data$pellets_m2yr[this_row-1],
   pellet_count_annual_data$start_date[this_row],
   pellet_count_annual_data$pellets_m2yr[this_row],
   col=lcol,
   lwd=lwd,
   lty=lty[this_grid]
   )
  }
 }
 
for (this_grid in 2:n_grids)
 {
 sln <- pellet_count_annual_data$grid_name == grid_name[this_grid] & !is.nan(pellet_count_annual_data$pellets_m2yr)
## Draw segments showing pellet densities. 
segments(pellet_count_annual_data$start_date[sln],
 pellet_count_annual_data$pellets_m2yr[sln],
 pellet_count_annual_data$end_date[sln],
 pellet_count_annual_data$pellets_m2yr[sln],
 col=lcol,
 lwd=lwd,
 lty=lty[this_grid]
 )
## connect non-missing segments.
for (this_row in 2:nrow(pellet_count_annual_data))
 {
 if (sln[this_row-1] & sln[this_row])
  {
  segments(pellet_count_annual_data$start_date[this_row],
   pellet_count_annual_data$pellets_m2yr[this_row-1],
   pellet_count_annual_data$start_date[this_row],
   pellet_count_annual_data$pellets_m2yr[this_row],
   col=lcol,
   lwd=lwd,
   lty=lty[this_grid]
   )
  }
 } 
 }
 ## Now plot the grand mean lines.
 segments(data_agg$start_date,
  data_agg$pellets_m2yr,
  data_agg$end_date,
  data_agg$pellets_m2yr,
  col="#777777",
  lwd=2,
  lty=1
  )
 for (this_row in 2:nrow(data_agg))
 {
  segments(data_agg$start_date[this_row],
   data_agg$pellets_m2yr[this_row-1],
   data_agg$start_date[this_row],
   data_agg$pellets_m2yr[this_row],
   col="#777777",
   lwd=2,
   lty=1
   )
 }
 ## Now plot the estimated hare densities.
 adjustment_factor <- max(pellet_count_annual_data$pellets_m2yr, na.rm=TRUE)/max(data_agg$Krebs_et_al_2001)*.98
 data_agg$Krebs_et_al_2001_adj <- data_agg$Krebs_et_al_2001 * adjustment_factor
  segments(data_agg$start_date,
  data_agg$Krebs_et_al_2001_adj,
  data_agg$end_date,
  data_agg$Krebs_et_al_2001_adj,
  col="#000000",
  lwd=2,
  lty="11"
  )
 for (this_row in 2:nrow(data_agg))
 {
  segments(data_agg$start_date[this_row],
   data_agg$Krebs_et_al_2001_adj[this_row-1],
   data_agg$start_date[this_row],
   data_agg$Krebs_et_al_2001_adj[this_row],
   col="#000000",
   lwd=2,
   lty="11"
   )
 }
 axis(side=4,
  at=density_ticks*adjustment_factor,
  labels=density_ticks
  )

text(x=as.Date("2000-01-01"),
 y=ylim[2] * c(1,0.95,0.9),
 labels= c("pellet counts, individual grids", "pellet counts, averaged", "estimated hare density"),
 pos=4
 )
segments(x0 = as.Date("1998-01-01"),
 y0 = ylim[2] * c(1,0.95),
 x1 = as.Date("2000-01-01"),
 y1 = ylim[2] * c(1,0.95),
 lwd = c(1.5, 2),
 lty = c(1,1),
 col = c("#0000001F", "#777777")
 ) 
segments(x0 = as.Date("1998-01-01"),
 y0 = ylim[2] * 0.9,
 x1 = as.Date("2000-01-01"),
 y1 = ylim[2] * 0.9,
 lwd = 2,
 lty = "11",
 col=("#000000")
 ) 
dev.off() 

## Save means summaries.
file_name <- paste0("../documents/",
 as.Date(Sys.time()),
 "_pellet_count_means_over_time.csv"
 )
write.csv(pellet_count_annual_data,
 file_name,
 row.names=FALSE
 ) 

## Save the grand mean summary, also.
 file_name <- paste0("../documents/",
 as.Date(Sys.time()),
 "_grand_means_over_time.csv"
 )
write.csv(data_agg,
 file_name,
 row.names=FALSE
 ) 
```

![Snowshoe hare pellet counts over time and estimated snowshoe hare
densities.](2020-09-15_pellet_count_over_time.jpg)  
Snowshoe hare pellet counts over time and estimated snowshoe hare
densities.

I took a walk down to Headquarters Lake and Nordic Lake in the
afternoon. I checked both lakes for any duckweeds, looking especially
for *Spirodela*. I saw no duckweeds. I collected some larvae from
mushrooms and soil, which I placed in LifeScanner vials.

In looking into the ecology of hare cycles, I found Boutin et
al. ([1995](#ref-boutin_population_1995)) and Krebs et
al. ([2018](#ref-krebs_using_2018)) to be helpful. Interestingly, both
Buckley ([1954](#ref-buckley_animal_1954)) and Boutin et
al. ([1995](#ref-boutin_population_1995)) documented the same
relationship of grouse peaking just before hares.

## Wednesday, September 16

I worked on my *Refuge Notebook* article about snowshoe hares due today,
getting it sent in near the end of the day.

Looking at the literature, there have been peak hare abundances on the
Kenai Peninsula in 1913–1915 (Buckley [1954](#ref-buckley_animal_1954))
and 1924–1925 (Buckley [1954](#ref-buckley_animal_1954)).

## Thursday, September 17

To do:

  - Review ADOT\&PF Veg Management Plan.
  - Finish snowshoe hare data management work.
  - Get Funny River Fire weed data to Lisa.
  - *Refuge Notebook* catch-up.

I did some LifeScanner data entry, looking for data for vials missing
data.

I entered some additional data into our KNWR invasive plant surveillence
survey Survey123 form, then exported these data into a new project
folder.

I scanned some more snowshoe hare pellet data sheets.

``` r

## R script to summarize and reformat data from non-native plant surveys collected using Survey123.

## By Matt Bowser, 17.September2020

library(maptools)
library(rgdal)
library(raster)


## Load data.
aggressiveness <- read.csv("../data/raw_data/code_tables/aggressiveness.csv")
commonName <- read.csv("../data/raw_data/code_tables/commonName.csv")
controlAction <- read.csv("../data/raw_data/code_tables/controlAction.csv")
disturbanceType.csv <- read.csv("../data/raw_data/code_tables/disturbanceType.csv")
observers <- read.csv("../data/raw_data/code_tables/observers.csv")
scientificName <- read.csv("../data/raw_data/code_tables/scientificName.csv")
vegetationCommunity <- read.csv("../data/raw_data/code_tables/vegetationCommunity.csv")

wgs84 <- CRS("+proj=longlat +ellps=WGS84 +datum=WGS84")
exported_data <- readShapeSpatial("../data/raw_data/observations/2020-09-17-0927_survey123_export.shp", 
 proj4string=wgs84
 )

exported_data <- readOGR(dsn="../data/raw_data/observations/2020-09-17-0927_survey123_export.shp",
 stringsAsFactors=FALSE 
 ) 

## Pull the shapefile's table to manipulate it. 
data01 <- exported_data@data

## First I will do work needed for reporting in the short term, but I will use field names for import into the AKEPIC data entry form  at the URL below 
## https://accs.uaa.alaska.edu/wp-content/uploads/AKEPIC_DataEntry.zip

data01$gps_locati
## This is a mess. I think it needs to be edited in a text editor or spreadsheet.
write.csv(data01$gps_locati, "../data/raw_data/observations/2020-09-17-1459_coordinates.csv", row.names=FALSE)
coords1 <- read.csv("../data/raw_data/observations/2020-09-17-1507_coordinates.csv")
coords1$latitude_m[is.na(coords1$latitude_m)] <- 0
coords1$longitude_m[is.na(coords1$longitude_m)] <- 0
coords1$longitude_dd <- -abs(coords1$longitude_dd)
coords1$latitude_dd <- coords1$latitude_dd + coords1$latitude_m/60
coords1$longitude_dd <- coords1$longitude_dd - coords1$longitude_m/60
data01$latitude <- coords1$latitude_dd
data01$longitude <- coords1$longitude_dd
## Just checking to make sure these end up in about the same places.
plot(exported_data)
points(data01$longitude, data01$latitude)
## That looked pretty good.

## Now we must get the plant species.
scientific_split <- strsplit(data01$scientific, ",")
n_species_obs <- sapply(scientific_split, length)
total_species_obs <- sum(sapply(scientific_split, length))
globalid <- rep(NA, total_species_obs)
scientificName_id <- rep(NA, total_species_obs)
scinamedf <- as.data.frame(cbind(globalid, scientificName_id))
this_record <- 0
for (this_event in 1:nrow(data01))
 {
 n_species <- n_species_obs[this_event]
 for (this_species in 1:n_species)
  {
  this_record <- this_record + 1
  scinamedf$globalid[this_record] <- data01$globalid[this_event]
  scinamedf$scientificName_id[this_record] <- sapply(scientific_split[this_event], "[", this_species)
  }
 }
scinamedf$scientificName_id <- as.numeric(scinamedf$scientificName_id)
scinamedf2 <- merge(x=scinamedf,
 y=scientificName,
 all.x=TRUE
 )
data02 <- merge(data01, 
 scinamedf2,
 by="globalid"
 )
## Now fill in those species that were not in th list.
sl <- data02$scientificName == "other species not on list"
data02$scientificName[sl] <- data02$other_name[sl]
levels(as.factor(data02$scientificName)) 

## Now for control actions. 
levels(as.factor(data02$control_ac))
## That is a strange mix of codes and text, maybe due to updates of the form.
## I will just fix this manually.
data02$control_ac[data02$control_ac=="1"] <- "Manual (pulling/digging)"
data02$control_ac[data02$control_ac=="6"] <- "Other"
data02$control_ac[data02$control_ac=="7"] <- "None"

## Now I need to get the areas surveyed and treated.
exported_data@data$area_m2 <- area(exported_data)

## I need to look at this spreadsheet.
write.csv(data01, "../data/raw_data/observations/2020-09-17-1633_survey123_export.csv", row.names=FALSE)

data01$area_treat
data01$area_tre_1
data01$area_treated_acres <- data01$area_tre_1
data01$area_treated_acres[is.na(data01$area_treated_acres)] <- data01$area_treat[is.na(data01$area_treated_acres)]
data01$area_treated_acres[data01$area_treated_acres=="1/2 acre"] <- "0.5"
data01$area_treated_acres <- as.numeric(data01$area_treated_acres)

## This has to be all for today.
save.image("2020-09-17-1643_workspace.RData")
```

## Friday, September 18

``` r

## R script to summarize and reformat data from non-native plant surveys collected using Survey123.

## By Matt Bowser, 18.September2020

library(maptools)
library(rgdal)
library(raster)

load("2020-09-17-1643_workspace.RData")

## Load Swan Lake Fire perimeter.
Swan_Lake_Fire <- readOGR(dsn="../source_data/geodata/2019_Swan_Lake_Fire/181_Swan_Lake_20190917_1100.shp",
 stringsAsFactors=FALSE 
 ) 

exported_data01 <- exported_data
exported_data01@data <- data01

Swan_Lake_Fire_wgs84 <- spTransform(Swan_Lake_Fire, CRS(proj4string(exported_data01)))

in_fire <- over(exported_data01, Swan_Lake_Fire_wgs84)

exported_data01@data$in_fire <- in_fire$NAME

## Do any of the treated areas in the fire lack a polygon?
exported_data01@data[!is.na(exported_data01@data$area_treated_acres),c("in_fire", "area_treated_acres", "Shape__Are")]
## Only four of the treated areas are in the fire and they all have non-zero shape area values.

exported_data_in_fire <- exported_data01[!is.na(exported_data01@data$in_fire),]

## Summarize for Lisa the number of acres treated in the fire:
sum(exported_data_in_fire@data$area_treated_acres, na.rm=TRUE)
[1] 2.85

## Now exporting this for Lisa.
writeOGR(obj=exported_data_in_fire, 
 dsn="../data/final_data/observations/2020-09-18-0852_treatments_in_fire.shp",
 layer="exported_data_in_fire",
 driver="ESRI Shapefile"
 )

## Now I need to get non-native plant species records to Lisa.
## There are still some missing values.
scinamemiss <- is.na(data02$scientificName)
data02[scinamemiss,]
## One has a common name only.
data02$scientificName[scinamemiss & data02$common_nam==27] <- "Taraxacum officinale F.H. Wigg."
scinamemiss <- is.na(data02$scientificName)
data02[scinamemiss,]
data02$scientificName[(scinamemiss & data02$other_na_1=="Reed canary grass")] <- "Phalaris arundinacea"
scinamemiss <- is.na(data02$scientificName)
data02[scinamemiss,]
data02$scientificName[(scinamemiss & data02$other_na_1=="Water Forget Me Not")] <- "Myosotis scorpioides"
scinamemiss <- is.na(data02$scientificName)
data02[scinamemiss,]
data02$scientificName[(scinamemiss & data02$other_na_1=="Water Forget-Me-Not")] <- "Myosotis scorpioides"
scinamemiss <- is.na(data02$scientificName)
data02[scinamemiss,]
data02$scientificName[(scinamemiss & data02$other_na_1=="Orange hawkweed")] <- "Hieracium aurantiacum"
scinamemiss <- is.na(data02$scientificName)
data02[scinamemiss,]
data02$scientificName[(scinamemiss & data02$other_na_1=="Reed canary grass and meadow forget me not present")] <- "Phalaris arundinacea"
## Duplicating these records.
addrows <- data02[(scinamemiss & data02$other_na_1=="Reed canary grass and meadow forget me not present"),]
addrows$scientificName <- "Myosotis scorpioides"
data02 <- rbind(data02, addrows)
scinamemiss <- is.na(data02$scientificName)
data02[scinamemiss,]

## Dropping those Juncus bufonius records, a native plant.
data03 <- data02[!(data02$scientificName=="Juncus bufonis"),]
## Saving this.
write.csv(data03, 
 "../data/final_data/observations/2020-09-18-1007_observations.csv",
 row.names=FALSE
 )

## Now selecting records from within the fire.
coordinates(data03) <- c("longitude", "latitude")
proj4string(data03) <- proj4string(exported_data01) 
in_fire <- over(data03, Swan_Lake_Fire_wgs84)
data03$in_fire <- in_fire$NAME

## Just checking.
plot(Swan_Lake_Fire_wgs84, col="red")
points(data03[!is.na(data03$in_fire),])
## That looked good.

weed_obs_in_fire <- data03[!is.na(data03$in_fire),]
dim(weed_obs_in_fire)
[1] 82 48
## So 82 observations in the fire.

## How many species in the fire?
length(levels(as.factor(weed_obs_in_fire$scientificName)))
[1] 20

## Save these.
write.table(levels(as.factor(weed_obs_in_fire$scientificName)), 
 "../data/final_data/observations/2020-09-18-1015_weed_species_in_fire.csv",
 row.names=FALSE,
 col.names="scientificName"
 )
 
## How many acres were infested?
sum(weed_obs_in_fire$infested_a)
## No!!! These are a mess, too. Why do these have to be text and not numeric?
weed_obs_in_fire$infested_a <- gsub(" acre", "", weed_obs_in_fire$infested_a) 
weed_obs_in_fire$infested_a <- gsub("s", "", weed_obs_in_fire$infested_a)
## Assuming 0.001 acres for those where nothing was given.
weed_obs_in_fire$infested_a[is.na(weed_obs_in_fire$infested_a)] <- "0.001"
weed_obs_in_fire$infested_a <- as.numeric(weed_obs_in_fire$infested_a)
sum(weed_obs_in_fire$infested_a)
[1] 57.983

## Finally, save the weed observations in the fire as a shapefile.
writeOGR(obj=weed_obs_in_fire, 
 dsn="../data/final_data/observations/2020-09-18-1018_non-native_plant_obs_in_fire.shp",
 layer="weed_obs_in_fire",
 driver="ESRI Shapefile"
 ) 
```

I sent these data to Lisa.

I drove out Swanson River Road looking for *Spirodela*.

At Weed Lake I saw no duck weeds, but I did collect a geophilomorph
centipede floating on the water, apparently flooded out (iNaturalist:
[60324623](https://www.inaturalist.org/observations/60324623);
BOLD-H26).

I also saw no duck weeds at Dolly Varden Lake.

At Fish Lake I also found no duck weeds despite the wind blowing from
the south. The changes due to earthworms here are amazing\! There is
basically no litter here despite the overstory of birch and cottonwoods
and alders and *Viburnum edule*. The affected area is also large. At the
shore I collected a tissue sample from a pale slug (iNaturalist:
[60326105](https://www.inaturalist.org/observations/60326105);
BOLD-ZA3).

![Worm-worked soil under alders and *Viburnum edule* at Fish Lake along
the trail between the campsite and the
lake.](2020-09-18_soil_surface_at_Fish_Lake.jpg)  
Worm-worked soil under alders and *Viburnum edule* at Fish Lake along
the trail between the campsite and the lake.

On the north side of the road across from the Fish Lake camp site, the
nightcrawlers have gotten some distance into a stand of aspens and
cottonwoods. The change here has been dramatic, with roots exposed and
leaf litter removed. I had not seen effects of nightcrawlers in aspen
stands. This is the biggest nightcrawler infestation that I have seen.

![Roots of aspen exposed by earthworm
activity.](2020-09-18_exposed_roots.jpg)  
Roots of aspen exposed by earthworm activity.

At Sucker Creek I collected more *Spirodela* (iNaturalist:
[60326892](https://www.inaturalist.org/observations/60326892)). I saw
that there were multiple stands of reed canarygrass upstream of the
culvert (iNaturalist
[60327151](https://www.inaturalist.org/observations/60327151)), so it
has already escaped out of Canoe Lake.

![Reed canarygrass at the beaver dam in Sucker Creek below Sucker
Lake.](2020-09-18_reed_canrygrass_at_Sucker_Creek.jpg)  
Reed canarygrass at the beaver dam in Sucker Creek below Sucker Lake.

Below the culvert I saw some silver salmon (iNaturalist:
[60327513](https://www.inaturalist.org/observations/60327513)) and
collected a rams horn snail (iNaturalist:
[60327761](https://www.inaturalist.org/observations/60327761);
BOLD-XA4).

![Salmonids in Sucker Creek just below the
culvert.](2020-09-18_salmonids_in_Sucker_Creek.jpg)  
Salmonids in Sucker Creek just below the culvert.

From a tuft of plants on a rock at water line I collected a fly larva
(iNaturalist:
[60329563](https://www.inaturalist.org/observations/60329563); BOLD-8M6)
and a leech (iNaturalist:
[60329992](https://www.inaturalist.org/observations/60329992);
BOLD-YK0). I found no reed canarygrass downstream, but I did not go very
far.

At Merganser Lake I saw no duck weeds. The soils here were *Lumbricus*
worked. I saw one immature *Lumbricus*.

I saw no duck weeds at Portage Lake or at a little pond along the way at
60.732333 N, 150.619277 W.

On the way back I stopped at another lake at 60.727215 N, 150.558883 W,
where the only floating duckweed I saw was *Lemna minor* (iNaturalist:
[60331267](https://www.inaturalist.org/observations/60331267)).

## Monday, September 21

To do:

  - ~~Take care of notes and data entry from Friday.~~
  - Finish snowshoe hare data work.

I searched through paper files, found as many of the snowhshoe hare
pellet count data sheets as I could, and scanned these.

At the back door by the lab I collected a fly trapped by spider silk
(BOLD-M90).

On the shore of Headquarters Lake I collected a midge from a swarm among
*Ledum*.

Some tipulids were apparently just emerging from the lake. These were
bouncing about, climbing vegetation frantically, and repeatedly falling,
right on the lake shore. Their wings had not yet expanded (BOLD-OJ9).

From deliquescent fungi (I think *Pholiota*) on a birch I collected a
larva (BOLD-F59).

## Tuesday, September 22

I worked on flattening the snowshoe hare data, conforming the data
definitions to to Darwin Core standards, and updating the data
dictionary.

``` r
## Today I am reformatting the data, totally flattening the dataset so that it is as transparent and clear as possible.

library(uuid)

## Load data. 

pellet_data <- read.csv("../data/final_data/observations/snowshoe_hare_pellet_counts.csv",
 stringsAsFactors=FALSE
 )
 
plot_data <- read.csv("../data/final_data/geodata/snowshoe_hare_plot_data.csv",
 stringsAsFactors=FALSE
 )
 
observer_data <- read.csv("../data/final_data/observations/event_recordedBy.csv",
 stringsAsFactors=FALSE)

## Join plot data and observation data. 
data2 <- merge(x=pellet_data,
 y=plot_data,
 all.x=TRUE
 ) 
 
 


## Now concatenate the observers
## First sort them.
observer_data <- observer_data[order(observer_data$eventID, observer_data$primary_observer, observer_data$recordedBy),]

## Start a data frame for the concatenated observer data.
eventID <- observer_data$eventID
recordedBy1 <- as.data.frame(eventID)
recordedBy1 <- unique(recordedBy1)
recordedBy1$recordedBy <- NA

## Now popupulate it.
for (this_event in 1:nrow(recordedBy1))
 {
 sl <- observer_data$eventID == recordedBy1$eventID[this_event]
 observers <- observer_data$recordedBy[sl]
 observersc <- paste(observers, collapse=" | ")
 recordedBy1$recordedBy[this_event] <- observersc
 }

## Now join these data together.
data3 <- merge(x=data2,
 y=recordedBy1,
 all.x=TRUE
 ) 
 
## Generate occurrenceID values.
uuid <- UUIDgenerate(n=nrow(data3))
## Conform these to the RDF standard (https://dwc.tdwg.org/rdf/)
data3$occurrenceID <- paste0("urn:uuid:", uuid)

## Conforming some of the field names to DwC terms.
names(data3)[which(names(data3)=="latitude")] <- "decimalLatitude"
names(data3)[which(names(data3)=="longitude")] <- "decimalLongitude"

## Conforming the actual counts with DwC terms.
data3$measurementType <- "pellet count"
names(data3)[which(names(data3)=="pellet_count")] <- "measurementValue"

## Trying to conform the locations to DwC while maintaining the ability to aggregate.

names(data3)[which(names(data3)=="plot_name")] <- "locationID"

grid_name <- unique(data3$grid_name)
location <- as.data.frame(grid_name)
location$Location <- c(
 "Campfire Lake",
 "Funny River",
 "Swanson River",
 "Oil field",
 "Skilak Lake",
 "Funny River New",
 "Skilak Lake New"
 )
location$Location <- paste(location$Location, "snowshoe hare grid")

data4 <- merge(x=data3,
 y=location,
 all.x=TRUE
 ) 

## Reorganizing. 
data5 <- data4[,c("occurrenceID",
 "eventID",
 "Location",
 "locationID",
 "decimalLatitude",
 "decimalLongitude",
 "habitat",
 "eventDate",
 "measurementType",
 "measurementValue",
 "recordedBy",
 "eventRemarks"
 )]
 
data5 <- data5[order(data5$eventID,
 data5$locationID
 ),]
 
## Saving here.
write.csv(data5, "../data/final_data/observations/Kenai_NWR_snowshoe_hare_pellet_count_data.csv", row.names=FALSE)
```

I need to extract an older version of the original data file I had
worked with.

``` sh
git log

git archive 416dea846fe6bbd88e285fc6e413b48c9854756a | gzip >../416dea846fe6bbd88e285fc6e413b48c9854756a.tar.gz
```

I entered field notebook data for 2008 to 2007.

## Wednesday, September 23

To do:

  - ~~Edit and submit this week’s *Refuge Notebook* article.~~
  - invasive species person nomination

I did some final editing on the snowshoe hare dataset and submitted it
to the regional repository.

I edited and sent out this weeks’ *Refuge Notebook* article.

I was given some earthworms collected a few days ago from Lower Copper
Lake.

I formatted and posted the *Refuge Notebook* article from January 28.

Examining worm specimens from Lower Copper Lake given to me today.
Pigmented red, setae widely spaced, clitellum on segments 25–31 or
26–31, tuburcula pubertatis on 28–30. This is *Bimastos rubidus*. I
collected a tissue sample (BOLD-OI8).

Clif had asked me to inspect the roof of the visitor center for
non-native plants, so I did so. The only non-native plant I saw up there
were a few individuals of *Taraxacum officinale*.

![Plant community on the roof of the Kenai National Wildlife Refuge’s
visitor center.](2020-09-23_visitor_center_roof.jpg)  
Plant community on the roof of the Kenai National Wildlife Refuge’s
visitor center.

One of the grasses from the roof I collected and identified as *Poa
alpina*. In the spikelets were many amber-colored mites, some of which I
collected (BOLD-6M6).

# Appendixes

## Occurrence data in Arctos bulkloader format

| collection\_object\_id | uuid                                 | guid\_prefix | cat\_num | Other\_Id\_Num\_Type\_1 | Other\_Id\_Num\_1 | enteredby | accn                          | taxon\_name              | nature\_of\_id | id\_made\_by\_agent | made\_date | identification\_remarks                                                                                                                                                                                    | verbatim\_date       | began\_date | ended\_date | higher\_geog                                                  | spec\_locality                                                                                       | verbatim\_locality                                                                                    | orig\_lat\_long\_units | dec\_lat    | dec\_long     | datum                      | georeference\_source            | max\_error\_distance | max\_error\_units | georeference\_protocol          | Locality\_Remarks                               | collecting\_source | habitat                                               | specimen\_event\_type | collecting\_method                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                 | event\_assigned\_by\_agent | event\_assigned\_date | verificationstatus | collector\_agent\_1 | collector\_role\_1 | collector\_agent\_2  | collector\_role\_2 | part\_name\_1  | part\_condition\_1     | part\_disposition\_1 | part\_lot\_count\_1 | Part\_Barcode\_1 |
| :--------------------- | :----------------------------------- | :----------- | :------- | :---------------------- | :---------------- | :-------- | :---------------------------- | :----------------------- | :------------- | :------------------ | :--------- | :--------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------- | :------------------- | :---------- | :---------- | :------------------------------------------------------------ | :--------------------------------------------------------------------------------------------------- | :---------------------------------------------------------------------------------------------------- | :--------------------- | :---------- | :------------ | :------------------------- | :------------------------------ | :------------------- | :---------------- | :------------------------------ | :---------------------------------------------- | :----------------- | :---------------------------------------------------- | :-------------------- | :------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------- | :------------------------- | :-------------------- | :----------------- | :------------------ | :----------------- | :------------------- | :----------------- | :------------- | :--------------------- | :------------------- | :------------------ | :--------------- |
| 1                      | 3f2b99b3-60f7-46be-8898-615f9cbb78eb | KNWRObs:Herb | 956      |                         |                   | mbowser   | KNWR-2020.02-SwanLake-ObsHerb | Betula neoalaskana       | features       | Matt Bowser         | 2020-07-20 | Seedling                                                                                                                                                                                                   | 2020-07-20           | 2020-07-20  | 2020-07-20  | North America, United States, Alaska, Kenai Peninsula Borough | South side of Skilak Lake Road \~ 2 km east of road to Lower Skilak Campground, Hakala plot 4 (HAK4) | South side of Skilak Lake Road \~ 2 km east of road to Lower Skilak Campground, Hakala plot 4 (HAK4)  | decimal degrees        | 60.4728     | \-150.415314  | World Geodetic System 1984 | Legacy plot coordinates         | 30                   | m                 | MaNIS georeferencing guidelines |                                                 | wild caught        | recently burned black spruce forest                   | observation           | 5.64 m radius, circular plot presence-only vegetation survey using the methods of Morton et al. (2009). Reference: Morton, J. M., Bowser, M. L., Berg, E., Magness, D., & Eskelin, T. (2009). Long Term Ecological Monitoring Program on the Kenai National Wildlife Refuge, Alaska: An FIA adjunct inventory. In W. McWilliams, G. Moisen, & R. Czaplewski (Eds.), 2008 Forest Inventory and Analysis (FIA) Symposium; October 21-23, 2008; Park City, UT (pp. 1-17). U.S. Department of Agriculture, Forest Service, Rocky Mountain Research Station. http://www.treesearch.fs.fed.us/pubs/33332 | Matt Bowser                | 2020-08-06            | unverified         | Matt Bowser         | collector          |                      |                    | media          | not applicable         | not applicable       | 1                   |                  |
| 2                      | 9b8a76e8-d6ae-4ec9-a3c4-2e715b4fcb65 | KNWRObs:Herb | 957      |                         |                   | mbowser   | KNWR-2020.02-SwanLake-ObsHerb | Ceratodon purpureus      | features       | Matt Bowser         | 2020-07-20 |                                                                                                                                                                                                            | 2020-07-20           | 2020-07-20  | 2020-07-20  | North America, United States, Alaska, Kenai Peninsula Borough | South side of Skilak Lake Road \~ 2 km east of road to Lower Skilak Campground, Hakala plot 4 (HAK4) | South side of Skilak Lake Road \~ 2 km east of road to Lower Skilak Campground, Hakala plot 4 (HAK4)  | decimal degrees        | 60.4728     | \-150.415314  | World Geodetic System 1984 | Legacy plot coordinates         | 30                   | m                 | MaNIS georeferencing guidelines |                                                 | wild caught        | recently burned black spruce forest                   | observation           | 5.64 m radius, circular plot presence-only vegetation survey using the methods of Morton et al. (2009). Reference: Morton, J. M., Bowser, M. L., Berg, E., Magness, D., & Eskelin, T. (2009). Long Term Ecological Monitoring Program on the Kenai National Wildlife Refuge, Alaska: An FIA adjunct inventory. In W. McWilliams, G. Moisen, & R. Czaplewski (Eds.), 2008 Forest Inventory and Analysis (FIA) Symposium; October 21-23, 2008; Park City, UT (pp. 1-17). U.S. Department of Agriculture, Forest Service, Rocky Mountain Research Station. http://www.treesearch.fs.fed.us/pubs/33332 | Matt Bowser                | 2020-08-06            | unverified         | Matt Bowser         | collector          |                      |                    | media          | not applicable         | not applicable       | 1                   |                  |
| 3                      | 8000f3f2-855e-4247-ba52-b96983f199e9 | KNWRObs:Herb | 958      |                         |                   | mbowser   | KNWR-2020.02-SwanLake-ObsHerb | Chamerion angustifolium  | features       | Matt Bowser         | 2020-07-20 |                                                                                                                                                                                                            | 2020-07-20           | 2020-07-20  | 2020-07-20  | North America, United States, Alaska, Kenai Peninsula Borough | South side of Skilak Lake Road \~ 2 km east of road to Lower Skilak Campground, Hakala plot 4 (HAK4) | South side of Skilak Lake Road \~ 2 km east of road to Lower Skilak Campground, Hakala plot 4 (HAK4)  | decimal degrees        | 60.4728     | \-150.415314  | World Geodetic System 1984 | Legacy plot coordinates         | 30                   | m                 | MaNIS georeferencing guidelines |                                                 | wild caught        | recently burned black spruce forest                   | observation           | 5.64 m radius, circular plot presence-only vegetation survey using the methods of Morton et al. (2009). Reference: Morton, J. M., Bowser, M. L., Berg, E., Magness, D., & Eskelin, T. (2009). Long Term Ecological Monitoring Program on the Kenai National Wildlife Refuge, Alaska: An FIA adjunct inventory. In W. McWilliams, G. Moisen, & R. Czaplewski (Eds.), 2008 Forest Inventory and Analysis (FIA) Symposium; October 21-23, 2008; Park City, UT (pp. 1-17). U.S. Department of Agriculture, Forest Service, Rocky Mountain Research Station. http://www.treesearch.fs.fed.us/pubs/33332 | Matt Bowser                | 2020-08-06            | unverified         | Matt Bowser         | collector          |                      |                    | media          | not applicable         | not applicable       | 1                   |                  |
| 4                      | 7685316b-f32e-4ff0-8c11-4da807915665 | KNWRObs:Herb | 959      |                         |                   | mbowser   | KNWR-2020.02-SwanLake-ObsHerb | Chamerion latifolium     | features       | Matt Bowser         | 2020-07-20 | Small immature.                                                                                                                                                                                            | 2020-07-20           | 2020-07-20  | 2020-07-20  | North America, United States, Alaska, Kenai Peninsula Borough | South side of Skilak Lake Road \~ 2 km east of road to Lower Skilak Campground, Hakala plot 4 (HAK4) | South side of Skilak Lake Road \~ 2 km east of road to Lower Skilak Campground, Hakala plot 4 (HAK4)  | decimal degrees        | 60.4728     | \-150.415314  | World Geodetic System 1984 | Legacy plot coordinates         | 30                   | m                 | MaNIS georeferencing guidelines |                                                 | wild caught        | recently burned black spruce forest                   | observation           | 5.64 m radius, circular plot presence-only vegetation survey using the methods of Morton et al. (2009). Reference: Morton, J. M., Bowser, M. L., Berg, E., Magness, D., & Eskelin, T. (2009). Long Term Ecological Monitoring Program on the Kenai National Wildlife Refuge, Alaska: An FIA adjunct inventory. In W. McWilliams, G. Moisen, & R. Czaplewski (Eds.), 2008 Forest Inventory and Analysis (FIA) Symposium; October 21-23, 2008; Park City, UT (pp. 1-17). U.S. Department of Agriculture, Forest Service, Rocky Mountain Research Station. http://www.treesearch.fs.fed.us/pubs/33332 | Matt Bowser                | 2020-08-06            | unverified         | Matt Bowser         | collector          |                      |                    | media          | not applicable         | not applicable       | 1                   |                  |
| 5                      | 3a4e4a82-cd3c-4d91-b144-1aefd16598c5 | KNWRObs:Herb | 960      |                         |                   | mbowser   | KNWR-2020.02-SwanLake-ObsHerb | Epilobium ciliatum       | features       | Matt Bowser         | 2020-07-20 |                                                                                                                                                                                                            | 2020-07-20           | 2020-07-20  | 2020-07-20  | North America, United States, Alaska, Kenai Peninsula Borough | South side of Skilak Lake Road \~ 2 km east of road to Lower Skilak Campground, Hakala plot 4 (HAK4) | South side of Skilak Lake Road \~ 2 km east of road to Lower Skilak Campground, Hakala plot 4 (HAK4)  | decimal degrees        | 60.4728     | \-150.415314  | World Geodetic System 1984 | Legacy plot coordinates         | 30                   | m                 | MaNIS georeferencing guidelines |                                                 | wild caught        | recently burned black spruce forest                   | observation           | 5.64 m radius, circular plot presence-only vegetation survey using the methods of Morton et al. (2009). Reference: Morton, J. M., Bowser, M. L., Berg, E., Magness, D., & Eskelin, T. (2009). Long Term Ecological Monitoring Program on the Kenai National Wildlife Refuge, Alaska: An FIA adjunct inventory. In W. McWilliams, G. Moisen, & R. Czaplewski (Eds.), 2008 Forest Inventory and Analysis (FIA) Symposium; October 21-23, 2008; Park City, UT (pp. 1-17). U.S. Department of Agriculture, Forest Service, Rocky Mountain Research Station. http://www.treesearch.fs.fed.us/pubs/33332 | Matt Bowser                | 2020-08-06            | unverified         | Matt Bowser         | collector          |                      |                    | media          | not applicable         | not applicable       | 1                   |                  |
| 6                      | 079ccca8-34ec-4c32-b7e7-5fac9a0ee993 | KNWRObs:Herb | 961      |                         |                   | mbowser   | KNWR-2020.02-SwanLake-ObsHerb | Marchantia polymorpha    | features       | Matt Bowser         | 2020-07-20 |                                                                                                                                                                                                            | 2020-07-20           | 2020-07-20  | 2020-07-20  | North America, United States, Alaska, Kenai Peninsula Borough | South side of Skilak Lake Road \~ 2 km east of road to Lower Skilak Campground, Hakala plot 4 (HAK4) | South side of Skilak Lake Road \~ 2 km east of road to Lower Skilak Campground, Hakala plot 4 (HAK4)  | decimal degrees        | 60.4728     | \-150.415314  | World Geodetic System 1984 | Legacy plot coordinates         | 30                   | m                 | MaNIS georeferencing guidelines |                                                 | wild caught        | recently burned black spruce forest                   | observation           | 5.64 m radius, circular plot presence-only vegetation survey using the methods of Morton et al. (2009). Reference: Morton, J. M., Bowser, M. L., Berg, E., Magness, D., & Eskelin, T. (2009). Long Term Ecological Monitoring Program on the Kenai National Wildlife Refuge, Alaska: An FIA adjunct inventory. In W. McWilliams, G. Moisen, & R. Czaplewski (Eds.), 2008 Forest Inventory and Analysis (FIA) Symposium; October 21-23, 2008; Park City, UT (pp. 1-17). U.S. Department of Agriculture, Forest Service, Rocky Mountain Research Station. http://www.treesearch.fs.fed.us/pubs/33332 | Matt Bowser                | 2020-08-06            | unverified         | Matt Bowser         | collector          |                      |                    | media          | not applicable         | not applicable       | 1                   |                  |
| 7                      | 6fa0f042-f296-4445-8947-40a13eed694e | KNWRObs:Herb | 962      |                         |                   | mbowser   | KNWR-2020.02-SwanLake-ObsHerb | Picea mariana            | features       | Matt Bowser         | 2020-07-20 | Seedling                                                                                                                                                                                                   | 2020-07-20           | 2020-07-20  | 2020-07-20  | North America, United States, Alaska, Kenai Peninsula Borough | South side of Skilak Lake Road \~ 2 km east of road to Lower Skilak Campground, Hakala plot 4 (HAK4) | South side of Skilak Lake Road \~ 2 km east of road to Lower Skilak Campground, Hakala plot 4 (HAK4)  | decimal degrees        | 60.4728     | \-150.415314  | World Geodetic System 1984 | Legacy plot coordinates         | 30                   | m                 | MaNIS georeferencing guidelines |                                                 | wild caught        | recently burned black spruce forest                   | observation           | 5.64 m radius, circular plot presence-only vegetation survey using the methods of Morton et al. (2009). Reference: Morton, J. M., Bowser, M. L., Berg, E., Magness, D., & Eskelin, T. (2009). Long Term Ecological Monitoring Program on the Kenai National Wildlife Refuge, Alaska: An FIA adjunct inventory. In W. McWilliams, G. Moisen, & R. Czaplewski (Eds.), 2008 Forest Inventory and Analysis (FIA) Symposium; October 21-23, 2008; Park City, UT (pp. 1-17). U.S. Department of Agriculture, Forest Service, Rocky Mountain Research Station. http://www.treesearch.fs.fed.us/pubs/33332 | Matt Bowser                | 2020-08-06            | unverified         | Matt Bowser         | collector          |                      |                    | media          | not applicable         | not applicable       | 1                   |                  |
| 8                      | 66a14818-ccaa-46f6-87ab-4a11f09c77c8 | KNWRObs:Herb | 963      |                         |                   | mbowser   | KNWR-2020.02-SwanLake-ObsHerb | Populus tremuloides      | features       | Matt Bowser         | 2020-07-20 | Fallen and dying but not dead yet.                                                                                                                                                                         | 2020-07-20           | 2020-07-20  | 2020-07-20  | North America, United States, Alaska, Kenai Peninsula Borough | South side of Skilak Lake Road \~ 2 km east of road to Lower Skilak Campground, Hakala plot 4 (HAK4) | South side of Skilak Lake Road \~ 2 km east of road to Lower Skilak Campground, Hakala plot 4 (HAK4)  | decimal degrees        | 60.4728     | \-150.415314  | World Geodetic System 1984 | Legacy plot coordinates         | 30                   | m                 | MaNIS georeferencing guidelines |                                                 | wild caught        | recently burned black spruce forest                   | observation           | 5.64 m radius, circular plot presence-only vegetation survey using the methods of Morton et al. (2009). Reference: Morton, J. M., Bowser, M. L., Berg, E., Magness, D., & Eskelin, T. (2009). Long Term Ecological Monitoring Program on the Kenai National Wildlife Refuge, Alaska: An FIA adjunct inventory. In W. McWilliams, G. Moisen, & R. Czaplewski (Eds.), 2008 Forest Inventory and Analysis (FIA) Symposium; October 21-23, 2008; Park City, UT (pp. 1-17). U.S. Department of Agriculture, Forest Service, Rocky Mountain Research Station. http://www.treesearch.fs.fed.us/pubs/33332 | Matt Bowser                | 2020-08-06            | unverified         | Matt Bowser         | collector          |                      |                    | media          | not applicable         | not applicable       | 1                   |                  |
| 9                      | 10ea38a2-34a5-4dc2-a7b5-20dcb1049449 | KNWRObs:Herb | 964      |                         |                   | mbowser   | KNWR-2020.02-SwanLake-ObsHerb | Betula neoalaskana       | features       | Matt Bowser         | 2020-07-20 | Seedlings                                                                                                                                                                                                  | 2020-07-20           | 2020-07-20  | 2020-07-20  | North America, United States, Alaska, Kenai Peninsula Borough | North side of Skilak Lake Road \~ 1.4 km west of road to Engineer Lake, Hakala plot 5 (HAK5)         | North side of Skilak Lake Road \~ 1.4 km west of road to Engineer Lake, Hakala plot 5 (HAK5)          | decimal degrees        | 60.469875   | \-150.358209  | World Geodetic System 1984 | Legacy plot coordinates         | 30                   | m                 | MaNIS georeferencing guidelines |                                                 | wild caught        | recently burned black spruce forest                   | observation           | 5.64 m radius, circular plot presence-only vegetation survey using the methods of Morton et al. (2009). Reference: Morton, J. M., Bowser, M. L., Berg, E., Magness, D., & Eskelin, T. (2009). Long Term Ecological Monitoring Program on the Kenai National Wildlife Refuge, Alaska: An FIA adjunct inventory. In W. McWilliams, G. Moisen, & R. Czaplewski (Eds.), 2008 Forest Inventory and Analysis (FIA) Symposium; October 21-23, 2008; Park City, UT (pp. 1-17). U.S. Department of Agriculture, Forest Service, Rocky Mountain Research Station. http://www.treesearch.fs.fed.us/pubs/33332 | Matt Bowser                | 2020-08-06            | unverified         | Matt Bowser         | collector          |                      |                    | media          | not applicable         | not applicable       | 1                   |                  |
| 10                     | 273f6cd2-be2d-420d-a5ff-00abe9e6364b | KNWRObs:Herb | 965      |                         |                   | mbowser   | KNWR-2020.02-SwanLake-ObsHerb | Ceratodon purpureus      | features       | Matt Bowser         | 2020-07-20 |                                                                                                                                                                                                            | 2020-07-20           | 2020-07-20  | 2020-07-20  | North America, United States, Alaska, Kenai Peninsula Borough | North side of Skilak Lake Road \~ 1.4 km west of road to Engineer Lake, Hakala plot 5 (HAK5)         | North side of Skilak Lake Road \~ 1.4 km west of road to Engineer Lake, Hakala plot 5 (HAK5)          | decimal degrees        | 60.469875   | \-150.358209  | World Geodetic System 1984 | Legacy plot coordinates         | 30                   | m                 | MaNIS georeferencing guidelines |                                                 | wild caught        | recently burned black spruce forest                   | observation           | 5.64 m radius, circular plot presence-only vegetation survey using the methods of Morton et al. (2009). Reference: Morton, J. M., Bowser, M. L., Berg, E., Magness, D., & Eskelin, T. (2009). Long Term Ecological Monitoring Program on the Kenai National Wildlife Refuge, Alaska: An FIA adjunct inventory. In W. McWilliams, G. Moisen, & R. Czaplewski (Eds.), 2008 Forest Inventory and Analysis (FIA) Symposium; October 21-23, 2008; Park City, UT (pp. 1-17). U.S. Department of Agriculture, Forest Service, Rocky Mountain Research Station. http://www.treesearch.fs.fed.us/pubs/33332 | Matt Bowser                | 2020-08-06            | unverified         | Matt Bowser         | collector          |                      |                    | media          | not applicable         | not applicable       | 1                   |                  |
| 11                     | 7af78837-fe32-4688-8158-e5ae08b229e1 | KNWRObs:Herb | 966      |                         |                   | mbowser   | KNWR-2020.02-SwanLake-ObsHerb | Chamerion angustifolium  | features       | Matt Bowser         | 2020-07-20 |                                                                                                                                                                                                            | 2020-07-20           | 2020-07-20  | 2020-07-20  | North America, United States, Alaska, Kenai Peninsula Borough | North side of Skilak Lake Road \~ 1.4 km west of road to Engineer Lake, Hakala plot 5 (HAK5)         | North side of Skilak Lake Road \~ 1.4 km west of road to Engineer Lake, Hakala plot 5 (HAK5)          | decimal degrees        | 60.469875   | \-150.358209  | World Geodetic System 1984 | Legacy plot coordinates         | 30                   | m                 | MaNIS georeferencing guidelines |                                                 | wild caught        | recently burned black spruce forest                   | observation           | 5.64 m radius, circular plot presence-only vegetation survey using the methods of Morton et al. (2009). Reference: Morton, J. M., Bowser, M. L., Berg, E., Magness, D., & Eskelin, T. (2009). Long Term Ecological Monitoring Program on the Kenai National Wildlife Refuge, Alaska: An FIA adjunct inventory. In W. McWilliams, G. Moisen, & R. Czaplewski (Eds.), 2008 Forest Inventory and Analysis (FIA) Symposium; October 21-23, 2008; Park City, UT (pp. 1-17). U.S. Department of Agriculture, Forest Service, Rocky Mountain Research Station. http://www.treesearch.fs.fed.us/pubs/33332 | Matt Bowser                | 2020-08-06            | unverified         | Matt Bowser         | collector          |                      |                    | media          | not applicable         | not applicable       | 1                   |                  |
| 12                     | e79ca27d-85ec-4481-8ad6-0936ade07bf3 | KNWRObs:Herb | 967      |                         |                   | mbowser   | KNWR-2020.02-SwanLake-ObsHerb | Chamerion latifolium     | features       | Matt Bowser         | 2020-07-20 | Small immature.                                                                                                                                                                                            | 2020-07-20           | 2020-07-20  | 2020-07-20  | North America, United States, Alaska, Kenai Peninsula Borough | North side of Skilak Lake Road \~ 1.4 km west of road to Engineer Lake, Hakala plot 5 (HAK5)         | North side of Skilak Lake Road \~ 1.4 km west of road to Engineer Lake, Hakala plot 5 (HAK5)          | decimal degrees        | 60.469875   | \-150.358209  | World Geodetic System 1984 | Legacy plot coordinates         | 30                   | m                 | MaNIS georeferencing guidelines |                                                 | wild caught        | recently burned black spruce forest                   | observation           | 5.64 m radius, circular plot presence-only vegetation survey using the methods of Morton et al. (2009). Reference: Morton, J. M., Bowser, M. L., Berg, E., Magness, D., & Eskelin, T. (2009). Long Term Ecological Monitoring Program on the Kenai National Wildlife Refuge, Alaska: An FIA adjunct inventory. In W. McWilliams, G. Moisen, & R. Czaplewski (Eds.), 2008 Forest Inventory and Analysis (FIA) Symposium; October 21-23, 2008; Park City, UT (pp. 1-17). U.S. Department of Agriculture, Forest Service, Rocky Mountain Research Station. http://www.treesearch.fs.fed.us/pubs/33332 | Matt Bowser                | 2020-08-06            | unverified         | Matt Bowser         | collector          |                      |                    | media          | not applicable         | not applicable       | 1                   |                  |
| 13                     | 6e6ee267-de53-4e85-955a-fe740285cbd7 | KNWRObs:Herb | 968      |                         |                   | mbowser   | KNWR-2020.02-SwanLake-ObsHerb | Lupinus nootkatensis     | features       | Matt Bowser         | 2020-07-20 |                                                                                                                                                                                                            | 2020-07-20           | 2020-07-20  | 2020-07-20  | North America, United States, Alaska, Kenai Peninsula Borough | North side of Skilak Lake Road \~ 1.4 km west of road to Engineer Lake, Hakala plot 5 (HAK5)         | North side of Skilak Lake Road \~ 1.4 km west of road to Engineer Lake, Hakala plot 5 (HAK5)          | decimal degrees        | 60.469875   | \-150.358209  | World Geodetic System 1984 | Legacy plot coordinates         | 30                   | m                 | MaNIS georeferencing guidelines |                                                 | wild caught        | recently burned black spruce forest                   | observation           | 5.64 m radius, circular plot presence-only vegetation survey using the methods of Morton et al. (2009). Reference: Morton, J. M., Bowser, M. L., Berg, E., Magness, D., & Eskelin, T. (2009). Long Term Ecological Monitoring Program on the Kenai National Wildlife Refuge, Alaska: An FIA adjunct inventory. In W. McWilliams, G. Moisen, & R. Czaplewski (Eds.), 2008 Forest Inventory and Analysis (FIA) Symposium; October 21-23, 2008; Park City, UT (pp. 1-17). U.S. Department of Agriculture, Forest Service, Rocky Mountain Research Station. http://www.treesearch.fs.fed.us/pubs/33332 | Matt Bowser                | 2020-08-06            | unverified         | Matt Bowser         | collector          |                      |                    | media          | not applicable         | not applicable       | 1                   |                  |
| 14                     | 3177ab21-3720-41d3-bb91-07a4cf2a8dea | KNWRObs:Herb | 969      |                         |                   | mbowser   | KNWR-2020.02-SwanLake-ObsHerb | Marchantia polymorpha    | features       | Matt Bowser         | 2020-07-20 |                                                                                                                                                                                                            | 2020-07-20           | 2020-07-20  | 2020-07-20  | North America, United States, Alaska, Kenai Peninsula Borough | North side of Skilak Lake Road \~ 1.4 km west of road to Engineer Lake, Hakala plot 5 (HAK5)         | North side of Skilak Lake Road \~ 1.4 km west of road to Engineer Lake, Hakala plot 5 (HAK5)          | decimal degrees        | 60.469875   | \-150.358209  | World Geodetic System 1984 | Legacy plot coordinates         | 30                   | m                 | MaNIS georeferencing guidelines |                                                 | wild caught        | recently burned black spruce forest                   | observation           | 5.64 m radius, circular plot presence-only vegetation survey using the methods of Morton et al. (2009). Reference: Morton, J. M., Bowser, M. L., Berg, E., Magness, D., & Eskelin, T. (2009). Long Term Ecological Monitoring Program on the Kenai National Wildlife Refuge, Alaska: An FIA adjunct inventory. In W. McWilliams, G. Moisen, & R. Czaplewski (Eds.), 2008 Forest Inventory and Analysis (FIA) Symposium; October 21-23, 2008; Park City, UT (pp. 1-17). U.S. Department of Agriculture, Forest Service, Rocky Mountain Research Station. http://www.treesearch.fs.fed.us/pubs/33332 | Matt Bowser                | 2020-08-06            | unverified         | Matt Bowser         | collector          |                      |                    | media          | not applicable         | not applicable       | 1                   |                  |
| 15                     | d7e0ef40-0db2-43f8-a32a-00a29d1481db | KNWRObs:Herb | 970      |                         |                   | mbowser   | KNWR-2020.02-SwanLake-ObsHerb | Picea mariana            | features       | Matt Bowser         | 2020-07-20 |                                                                                                                                                                                                            | 2020-07-20           | 2020-07-20  | 2020-07-20  | North America, United States, Alaska, Kenai Peninsula Borough | North side of Skilak Lake Road \~ 1.4 km west of road to Engineer Lake, Hakala plot 5 (HAK5)         | North side of Skilak Lake Road \~ 1.4 km west of road to Engineer Lake, Hakala plot 5 (HAK5)          | decimal degrees        | 60.469875   | \-150.358209  | World Geodetic System 1984 | Legacy plot coordinates         | 30                   | m                 | MaNIS georeferencing guidelines |                                                 | wild caught        | recently burned black spruce forest                   | observation           | 5.64 m radius, circular plot presence-only vegetation survey using the methods of Morton et al. (2009). Reference: Morton, J. M., Bowser, M. L., Berg, E., Magness, D., & Eskelin, T. (2009). Long Term Ecological Monitoring Program on the Kenai National Wildlife Refuge, Alaska: An FIA adjunct inventory. In W. McWilliams, G. Moisen, & R. Czaplewski (Eds.), 2008 Forest Inventory and Analysis (FIA) Symposium; October 21-23, 2008; Park City, UT (pp. 1-17). U.S. Department of Agriculture, Forest Service, Rocky Mountain Research Station. http://www.treesearch.fs.fed.us/pubs/33332 | Matt Bowser                | 2020-08-06            | unverified         | Matt Bowser         | collector          |                      |                    | media          | not applicable         | not applicable       | 1                   |                  |
| 16                     | e1a07637-a5db-4fd0-8d5d-7ef368b906de | KNWRObs:Herb | 954      |                         |                   | mbowser   | KNWR-2020.02-SwanLake-ObsHerb | Polytrichum              | features       | Matt Bowser         | 2020-07-20 |                                                                                                                                                                                                            | 2020-07-20           | 2020-07-20  | 2020-07-20  | North America, United States, Alaska, Kenai Peninsula Borough | North side of Skilak Lake Road \~ 1.4 km west of road to Engineer Lake, Hakala plot 5 (HAK5)         | North side of Skilak Lake Road \~ 1.4 km west of road to Engineer Lake, Hakala plot 5 (HAK5)          | decimal degrees        | 60.469875   | \-150.358209  | World Geodetic System 1984 | Legacy plot coordinates         | 30                   | m                 | MaNIS georeferencing guidelines |                                                 | wild caught        | recently burned black spruce forest                   | observation           | 5.64 m radius, circular plot presence-only vegetation survey using the methods of Morton et al. (2009). Reference: Morton, J. M., Bowser, M. L., Berg, E., Magness, D., & Eskelin, T. (2009). Long Term Ecological Monitoring Program on the Kenai National Wildlife Refuge, Alaska: An FIA adjunct inventory. In W. McWilliams, G. Moisen, & R. Czaplewski (Eds.), 2008 Forest Inventory and Analysis (FIA) Symposium; October 21-23, 2008; Park City, UT (pp. 1-17). U.S. Department of Agriculture, Forest Service, Rocky Mountain Research Station. http://www.treesearch.fs.fed.us/pubs/33332 | Matt Bowser                | 2020-08-06            | unverified         | Matt Bowser         | collector          |                      |                    | media          | not applicable         | not applicable       | 1                   |                  |
| 17                     | ad159a6b-8b29-4db5-8171-87e6b183f793 | KNWRObs:Herb | 955      |                         |                   | mbowser   | KNWR-2020.02-SwanLake-ObsHerb | Populus tremuloides      | features       | Matt Bowser         | 2020-07-20 | Suckers                                                                                                                                                                                                    | 2020-07-20           | 2020-07-20  | 2020-07-20  | North America, United States, Alaska, Kenai Peninsula Borough | North side of Skilak Lake Road \~ 1.4 km west of road to Engineer Lake, Hakala plot 5 (HAK5)         | North side of Skilak Lake Road \~ 1.4 km west of road to Engineer Lake, Hakala plot 5 (HAK5)          | decimal degrees        | 60.469875   | \-150.358209  | World Geodetic System 1984 | Legacy plot coordinates         | 30                   | m                 | MaNIS georeferencing guidelines |                                                 | wild caught        | recently burned black spruce forest                   | observation           | 5.64 m radius, circular plot presence-only vegetation survey using the methods of Morton et al. (2009). Reference: Morton, J. M., Bowser, M. L., Berg, E., Magness, D., & Eskelin, T. (2009). Long Term Ecological Monitoring Program on the Kenai National Wildlife Refuge, Alaska: An FIA adjunct inventory. In W. McWilliams, G. Moisen, & R. Czaplewski (Eds.), 2008 Forest Inventory and Analysis (FIA) Symposium; October 21-23, 2008; Park City, UT (pp. 1-17). U.S. Department of Agriculture, Forest Service, Rocky Mountain Research Station. http://www.treesearch.fs.fed.us/pubs/33332 | Matt Bowser                | 2020-08-06            | unverified         | Matt Bowser         | collector          |                      |                    | media          | not applicable         | not applicable       | 1                   |                  |
| 18                     | e316b759-cfac-49b2-a741-022e0331fd63 | KNWRObs:Herb | 992      |                         |                   | mbowser   | KNWR-2020.02-SwanLake-ObsHerb | Cornus                   | features       | Matt Bowser         | 2020-07-27 | No floweres were available. This is likely Cornus unalaschkensis based on nearby observations. See https://www.inaturalist.org/observations/13564293 and https://www.inaturalist.org/observations/7028590. | 2020-07-27           | 2020-07-27  | 2020-07-27  | North America, United States, Alaska, Kenai Peninsula Borough | Mystery Hills off of the Skyline Trail, Skyline shrub study site 354-5                               | Mystery Hills off of the Skyline Trail, Skyline shrub study site 354-5                                | decimal degrees        | 60.52521111 | \-150.1637561 | World Geodetic System 1984 | Pre-determined plot coordinates | 30                   | m                 | MaNIS georeferencing guidelines |                                                 | wild caught        | recently burned low shrub tundra                      | observation           | 5.64 m radius, circular plot presence-only vegetation survey using the methods of Morton et al. (2009). Reference: Morton, J. M., Bowser, M. L., Berg, E., Magness, D., & Eskelin, T. (2009). Long Term Ecological Monitoring Program on the Kenai National Wildlife Refuge, Alaska: An FIA adjunct inventory. In W. McWilliams, G. Moisen, & R. Czaplewski (Eds.), 2008 Forest Inventory and Analysis (FIA) Symposium; October 21-23, 2008; Park City, UT (pp. 1-17). U.S. Department of Agriculture, Forest Service, Rocky Mountain Research Station. http://www.treesearch.fs.fed.us/pubs/33332 | Matt Bowser                | 2020-08-07            | unverified         | Matt Bowser         | collector          | Annaleese T. Rasanen | collector          | media          | not applicable         | not applicable       | 1                   |                  |
| 19                     | 9fdcd914-14aa-4a25-9452-cb73dee98870 | KNWRObs:Herb | 972      |                         |                   | mbowser   | KNWR-2020.02-SwanLake-ObsHerb | Angelica lucida          | features       | Matt Bowser         | 2020-07-27 |                                                                                                                                                                                                            | 2020-07-27           | 2020-07-27  | 2020-07-27  | North America, United States, Alaska, Kenai Peninsula Borough | Mystery Hills off of the Skyline Trail, Skyline shrub study site 354-5                               | Mystery Hills off of the Skyline Trail, Skyline shrub study site 354-5                                | decimal degrees        | 60.52521111 | \-150.1637561 | World Geodetic System 1984 | Pre-determined plot coordinates | 30                   | m                 | MaNIS georeferencing guidelines |                                                 | wild caught        | recently burned low shrub tundra                      | observation           | 5.64 m radius, circular plot presence-only vegetation survey using the methods of Morton et al. (2009). Reference: Morton, J. M., Bowser, M. L., Berg, E., Magness, D., & Eskelin, T. (2009). Long Term Ecological Monitoring Program on the Kenai National Wildlife Refuge, Alaska: An FIA adjunct inventory. In W. McWilliams, G. Moisen, & R. Czaplewski (Eds.), 2008 Forest Inventory and Analysis (FIA) Symposium; October 21-23, 2008; Park City, UT (pp. 1-17). U.S. Department of Agriculture, Forest Service, Rocky Mountain Research Station. http://www.treesearch.fs.fed.us/pubs/33332 | Matt Bowser                | 2020-08-07            | unverified         | Matt Bowser         | collector          | Annaleese T. Rasanen | collector          | media          | not applicable         | not applicable       | 1                   |                  |
| 20                     | 536c4af9-78bd-4a8f-b786-d0bf6df4c5d8 | KNWRObs:Herb | 974      |                         |                   | mbowser   | KNWR-2020.02-SwanLake-ObsHerb | Chamerion angustifolium  | features       | Matt Bowser         | 2020-07-27 |                                                                                                                                                                                                            | 2020-07-27           | 2020-07-27  | 2020-07-27  | North America, United States, Alaska, Kenai Peninsula Borough | Mystery Hills off of the Skyline Trail, Skyline shrub study site 354-5                               | Mystery Hills off of the Skyline Trail, Skyline shrub study site 354-5                                | decimal degrees        | 60.52521111 | \-150.1637561 | World Geodetic System 1984 | Pre-determined plot coordinates | 30                   | m                 | MaNIS georeferencing guidelines |                                                 | wild caught        | recently burned low shrub tundra                      | observation           | 5.64 m radius, circular plot presence-only vegetation survey using the methods of Morton et al. (2009). Reference: Morton, J. M., Bowser, M. L., Berg, E., Magness, D., & Eskelin, T. (2009). Long Term Ecological Monitoring Program on the Kenai National Wildlife Refuge, Alaska: An FIA adjunct inventory. In W. McWilliams, G. Moisen, & R. Czaplewski (Eds.), 2008 Forest Inventory and Analysis (FIA) Symposium; October 21-23, 2008; Park City, UT (pp. 1-17). U.S. Department of Agriculture, Forest Service, Rocky Mountain Research Station. http://www.treesearch.fs.fed.us/pubs/33332 | Matt Bowser                | 2020-08-07            | unverified         | Matt Bowser         | collector          | Annaleese T. Rasanen | collector          | media          | not applicable         | not applicable       | 1                   |                  |
| 21                     | 71d1021f-4f3d-4342-b985-b8be2a6c2b19 | KNWRObs:Herb | 973      |                         |                   | mbowser   | KNWR-2020.02-SwanLake-ObsHerb | Empetrum nigrum          | features       | Matt Bowser         | 2020-07-27 |                                                                                                                                                                                                            | 2020-07-27           | 2020-07-27  | 2020-07-27  | North America, United States, Alaska, Kenai Peninsula Borough | Mystery Hills off of the Skyline Trail, Skyline shrub study site 354-5                               | Mystery Hills off of the Skyline Trail, Skyline shrub study site 354-5                                | decimal degrees        | 60.52521111 | \-150.1637561 | World Geodetic System 1984 | Pre-determined plot coordinates | 30                   | m                 | MaNIS georeferencing guidelines |                                                 | wild caught        | recently burned low shrub tundra                      | observation           | 5.64 m radius, circular plot presence-only vegetation survey using the methods of Morton et al. (2009). Reference: Morton, J. M., Bowser, M. L., Berg, E., Magness, D., & Eskelin, T. (2009). Long Term Ecological Monitoring Program on the Kenai National Wildlife Refuge, Alaska: An FIA adjunct inventory. In W. McWilliams, G. Moisen, & R. Czaplewski (Eds.), 2008 Forest Inventory and Analysis (FIA) Symposium; October 21-23, 2008; Park City, UT (pp. 1-17). U.S. Department of Agriculture, Forest Service, Rocky Mountain Research Station. http://www.treesearch.fs.fed.us/pubs/33332 | Matt Bowser                | 2020-08-07            | unverified         | Matt Bowser         | collector          | Annaleese T. Rasanen | collector          | media          | not applicable         | not applicable       | 1                   |                  |
| 22                     | ed446cd9-f55a-4b1f-9ac1-3cff226b8f21 | KNWRObs:Herb | 975      |                         |                   | mbowser   | KNWR-2020.02-SwanLake-ObsHerb | Linnaea borealis         | features       | Matt Bowser         | 2020-07-27 |                                                                                                                                                                                                            | 2020-07-27           | 2020-07-27  | 2020-07-27  | North America, United States, Alaska, Kenai Peninsula Borough | Mystery Hills off of the Skyline Trail, Skyline shrub study site 354-5                               | Mystery Hills off of the Skyline Trail, Skyline shrub study site 354-5                                | decimal degrees        | 60.52521111 | \-150.1637561 | World Geodetic System 1984 | Pre-determined plot coordinates | 30                   | m                 | MaNIS georeferencing guidelines |                                                 | wild caught        | recently burned low shrub tundra                      | observation           | 5.64 m radius, circular plot presence-only vegetation survey using the methods of Morton et al. (2009). Reference: Morton, J. M., Bowser, M. L., Berg, E., Magness, D., & Eskelin, T. (2009). Long Term Ecological Monitoring Program on the Kenai National Wildlife Refuge, Alaska: An FIA adjunct inventory. In W. McWilliams, G. Moisen, & R. Czaplewski (Eds.), 2008 Forest Inventory and Analysis (FIA) Symposium; October 21-23, 2008; Park City, UT (pp. 1-17). U.S. Department of Agriculture, Forest Service, Rocky Mountain Research Station. http://www.treesearch.fs.fed.us/pubs/33332 | Matt Bowser                | 2020-08-07            | unverified         | Matt Bowser         | collector          | Annaleese T. Rasanen | collector          | media          | not applicable         | not applicable       | 1                   |                  |
| 23                     | be7bf093-56f4-4515-b541-1f97bdd81379 | KNWRObs:Herb | 976      |                         |                   | mbowser   | KNWR-2020.02-SwanLake-ObsHerb | Artemisia                | features       | Matt Bowser         | 2020-07-27 | Only basal leaves were present.                                                                                                                                                                            | 2020-07-27           | 2020-07-27  | 2020-07-27  | North America, United States, Alaska, Kenai Peninsula Borough | Mystery Hills off of the Skyline Trail, Skyline shrub study site 354-5                               | Mystery Hills off of the Skyline Trail, Skyline shrub study site 354-5                                | decimal degrees        | 60.52521111 | \-150.1637561 | World Geodetic System 1984 | Pre-determined plot coordinates | 30                   | m                 | MaNIS georeferencing guidelines |                                                 | wild caught        | recently burned low shrub tundra                      | observation           | 5.64 m radius, circular plot presence-only vegetation survey using the methods of Morton et al. (2009). Reference: Morton, J. M., Bowser, M. L., Berg, E., Magness, D., & Eskelin, T. (2009). Long Term Ecological Monitoring Program on the Kenai National Wildlife Refuge, Alaska: An FIA adjunct inventory. In W. McWilliams, G. Moisen, & R. Czaplewski (Eds.), 2008 Forest Inventory and Analysis (FIA) Symposium; October 21-23, 2008; Park City, UT (pp. 1-17). U.S. Department of Agriculture, Forest Service, Rocky Mountain Research Station. http://www.treesearch.fs.fed.us/pubs/33332 | Matt Bowser                | 2020-08-07            | unverified         | Matt Bowser         | collector          | Annaleese T. Rasanen | collector          | media          | not applicable         | not applicable       | 1                   |                  |
| 24                     | e177f0b4-cccd-4e0b-a59c-1f24f1db6df8 | KNWRObs:Herb | 977      |                         |                   | mbowser   | KNWR-2020.02-SwanLake-ObsHerb | Poaceae                  | features       | Matt Bowser         | 2020-07-27 | No inflorescences were present.                                                                                                                                                                            | 2020-07-27           | 2020-07-27  | 2020-07-27  | North America, United States, Alaska, Kenai Peninsula Borough | Mystery Hills off of the Skyline Trail, Skyline shrub study site 354-5                               | Mystery Hills off of the Skyline Trail, Skyline shrub study site 354-5                                | decimal degrees        | 60.52521111 | \-150.1637561 | World Geodetic System 1984 | Pre-determined plot coordinates | 30                   | m                 | MaNIS georeferencing guidelines |                                                 | wild caught        | recently burned low shrub tundra                      | observation           | 5.64 m radius, circular plot presence-only vegetation survey using the methods of Morton et al. (2009). Reference: Morton, J. M., Bowser, M. L., Berg, E., Magness, D., & Eskelin, T. (2009). Long Term Ecological Monitoring Program on the Kenai National Wildlife Refuge, Alaska: An FIA adjunct inventory. In W. McWilliams, G. Moisen, & R. Czaplewski (Eds.), 2008 Forest Inventory and Analysis (FIA) Symposium; October 21-23, 2008; Park City, UT (pp. 1-17). U.S. Department of Agriculture, Forest Service, Rocky Mountain Research Station. http://www.treesearch.fs.fed.us/pubs/33332 | Matt Bowser                | 2020-08-07            | unverified         | Matt Bowser         | collector          | Annaleese T. Rasanen | collector          | media          | not applicable         | not applicable       | 1                   |                  |
| 25                     | 383bb335-3a9d-44a8-9734-c46ee4bcefce | KNWRObs:Herb | 978      |                         |                   | mbowser   | KNWR-2020.02-SwanLake-ObsHerb | Tsuga mertensiana        | features       | Matt Bowser         | 2020-07-27 | Seedlings                                                                                                                                                                                                  | 2020-07-27           | 2020-07-27  | 2020-07-27  | North America, United States, Alaska, Kenai Peninsula Borough | Mystery Hills off of the Skyline Trail, Skyline shrub study site 354-5                               | Mystery Hills off of the Skyline Trail, Skyline shrub study site 354-5                                | decimal degrees        | 60.52521111 | \-150.1637561 | World Geodetic System 1984 | Pre-determined plot coordinates | 30                   | m                 | MaNIS georeferencing guidelines |                                                 | wild caught        | recently burned low shrub tundra                      | observation           | 5.64 m radius, circular plot presence-only vegetation survey using the methods of Morton et al. (2009). Reference: Morton, J. M., Bowser, M. L., Berg, E., Magness, D., & Eskelin, T. (2009). Long Term Ecological Monitoring Program on the Kenai National Wildlife Refuge, Alaska: An FIA adjunct inventory. In W. McWilliams, G. Moisen, & R. Czaplewski (Eds.), 2008 Forest Inventory and Analysis (FIA) Symposium; October 21-23, 2008; Park City, UT (pp. 1-17). U.S. Department of Agriculture, Forest Service, Rocky Mountain Research Station. http://www.treesearch.fs.fed.us/pubs/33332 | Matt Bowser                | 2020-08-07            | unverified         | Matt Bowser         | collector          | Annaleese T. Rasanen | collector          | media          | not applicable         | not applicable       | 1                   |                  |
| 26                     | 5664b609-d2ce-42ad-92b2-70321294e8d1 | KNWRObs:Herb | 979      |                         |                   | mbowser   | KNWR-2020.02-SwanLake-ObsHerb | Gymnocarpium dryopteris  | features       | Matt Bowser         | 2020-07-27 |                                                                                                                                                                                                            | 2020-07-27           | 2020-07-27  | 2020-07-27  | North America, United States, Alaska, Kenai Peninsula Borough | Mystery Hills off of the Skyline Trail, Skyline shrub study site 354-5                               | Mystery Hills off of the Skyline Trail, Skyline shrub study site 354-5                                | decimal degrees        | 60.52521111 | \-150.1637561 | World Geodetic System 1984 | Pre-determined plot coordinates | 30                   | m                 | MaNIS georeferencing guidelines |                                                 | wild caught        | recently burned low shrub tundra                      | observation           | 5.64 m radius, circular plot presence-only vegetation survey using the methods of Morton et al. (2009). Reference: Morton, J. M., Bowser, M. L., Berg, E., Magness, D., & Eskelin, T. (2009). Long Term Ecological Monitoring Program on the Kenai National Wildlife Refuge, Alaska: An FIA adjunct inventory. In W. McWilliams, G. Moisen, & R. Czaplewski (Eds.), 2008 Forest Inventory and Analysis (FIA) Symposium; October 21-23, 2008; Park City, UT (pp. 1-17). U.S. Department of Agriculture, Forest Service, Rocky Mountain Research Station. http://www.treesearch.fs.fed.us/pubs/33332 | Matt Bowser                | 2020-08-07            | unverified         | Matt Bowser         | collector          | Annaleese T. Rasanen | collector          | media          | not applicable         | not applicable       | 1                   |                  |
| 27                     | 15f45b14-d088-439d-972c-b25f3118256a | KNWRObs:Herb | 980      |                         |                   | mbowser   | KNWR-2020.02-SwanLake-ObsHerb | Vaccinium uliginosum     | features       | Matt Bowser         | 2020-07-27 |                                                                                                                                                                                                            | 2020-07-27           | 2020-07-27  | 2020-07-27  | North America, United States, Alaska, Kenai Peninsula Borough | Mystery Hills off of the Skyline Trail, Skyline shrub study site 354-5                               | Mystery Hills off of the Skyline Trail, Skyline shrub study site 354-5                                | decimal degrees        | 60.52521111 | \-150.1637561 | World Geodetic System 1984 | Pre-determined plot coordinates | 30                   | m                 | MaNIS georeferencing guidelines |                                                 | wild caught        | recently burned low shrub tundra                      | observation           | 5.64 m radius, circular plot presence-only vegetation survey using the methods of Morton et al. (2009). Reference: Morton, J. M., Bowser, M. L., Berg, E., Magness, D., & Eskelin, T. (2009). Long Term Ecological Monitoring Program on the Kenai National Wildlife Refuge, Alaska: An FIA adjunct inventory. In W. McWilliams, G. Moisen, & R. Czaplewski (Eds.), 2008 Forest Inventory and Analysis (FIA) Symposium; October 21-23, 2008; Park City, UT (pp. 1-17). U.S. Department of Agriculture, Forest Service, Rocky Mountain Research Station. http://www.treesearch.fs.fed.us/pubs/33332 | Matt Bowser                | 2020-08-07            | unverified         | Matt Bowser         | collector          | Annaleese T. Rasanen | collector          | media          | not applicable         | not applicable       | 1                   |                  |
| 28                     | 33c6ab78-24ca-49e6-80a6-b1aa10f2185b | KNWRObs:Herb | 971      |                         |                   | mbowser   | KNWR-2020.02-SwanLake-ObsHerb | Spiraea stevenii         | features       | Matt Bowser         | 2020-07-27 |                                                                                                                                                                                                            | 2020-07-27           | 2020-07-27  | 2020-07-27  | North America, United States, Alaska, Kenai Peninsula Borough | Mystery Hills off of the Skyline Trail, Skyline shrub study site 354-5                               | Mystery Hills off of the Skyline Trail, Skyline shrub study site 354-5                                | decimal degrees        | 60.52521111 | \-150.1637561 | World Geodetic System 1984 | Pre-determined plot coordinates | 30                   | m                 | MaNIS georeferencing guidelines |                                                 | wild caught        | recently burned low shrub tundra                      | observation           | 5.64 m radius, circular plot presence-only vegetation survey using the methods of Morton et al. (2009). Reference: Morton, J. M., Bowser, M. L., Berg, E., Magness, D., & Eskelin, T. (2009). Long Term Ecological Monitoring Program on the Kenai National Wildlife Refuge, Alaska: An FIA adjunct inventory. In W. McWilliams, G. Moisen, & R. Czaplewski (Eds.), 2008 Forest Inventory and Analysis (FIA) Symposium; October 21-23, 2008; Park City, UT (pp. 1-17). U.S. Department of Agriculture, Forest Service, Rocky Mountain Research Station. http://www.treesearch.fs.fed.us/pubs/33332 | Matt Bowser                | 2020-08-07            | unverified         | Matt Bowser         | collector          | Annaleese T. Rasanen | collector          | media          | not applicable         | not applicable       | 1                   |                  |
| 29                     | 0cc009ba-8199-4d84-a18d-f11fa0970bcf | KNWRObs:Herb | 981      |                         |                   | mbowser   | KNWR-2020.02-SwanLake-ObsHerb | Carex                    | features       | Matt Bowser         | 2020-07-27 | No inflorescences were present.                                                                                                                                                                            | 2020-07-27           | 2020-07-27  | 2020-07-27  | North America, United States, Alaska, Kenai Peninsula Borough | Mystery Hills off of the Skyline Trail, Skyline shrub study site 354-5                               | Mystery Hills off of the Skyline Trail, Skyline shrub study site 354-5                                | decimal degrees        | 60.52521111 | \-150.1637561 | World Geodetic System 1984 | Pre-determined plot coordinates | 30                   | m                 | MaNIS georeferencing guidelines |                                                 | wild caught        | recently burned low shrub tundra                      | observation           | 5.64 m radius, circular plot presence-only vegetation survey using the methods of Morton et al. (2009). Reference: Morton, J. M., Bowser, M. L., Berg, E., Magness, D., & Eskelin, T. (2009). Long Term Ecological Monitoring Program on the Kenai National Wildlife Refuge, Alaska: An FIA adjunct inventory. In W. McWilliams, G. Moisen, & R. Czaplewski (Eds.), 2008 Forest Inventory and Analysis (FIA) Symposium; October 21-23, 2008; Park City, UT (pp. 1-17). U.S. Department of Agriculture, Forest Service, Rocky Mountain Research Station. http://www.treesearch.fs.fed.us/pubs/33332 | Matt Bowser                | 2020-08-07            | unverified         | Matt Bowser         | collector          | Annaleese T. Rasanen | collector          | media          | not applicable         | not applicable       | 1                   |                  |
| 30                     | 0e0815df-bda0-48ef-a4de-1c85a3d5f2f0 | KNWRObs:Herb | 982      |                         |                   | mbowser   | KNWR-2020.02-SwanLake-ObsHerb | Sorbus scopulina         | features       | Matt Bowser         | 2020-07-27 |                                                                                                                                                                                                            | 2020-07-27           | 2020-07-27  | 2020-07-27  | North America, United States, Alaska, Kenai Peninsula Borough | Mystery Hills off of the Skyline Trail, Skyline shrub study site 354-5                               | Mystery Hills off of the Skyline Trail, Skyline shrub study site 354-5                                | decimal degrees        | 60.52521111 | \-150.1637561 | World Geodetic System 1984 | Pre-determined plot coordinates | 30                   | m                 | MaNIS georeferencing guidelines |                                                 | wild caught        | recently burned low shrub tundra                      | observation           | 5.64 m radius, circular plot presence-only vegetation survey using the methods of Morton et al. (2009). Reference: Morton, J. M., Bowser, M. L., Berg, E., Magness, D., & Eskelin, T. (2009). Long Term Ecological Monitoring Program on the Kenai National Wildlife Refuge, Alaska: An FIA adjunct inventory. In W. McWilliams, G. Moisen, & R. Czaplewski (Eds.), 2008 Forest Inventory and Analysis (FIA) Symposium; October 21-23, 2008; Park City, UT (pp. 1-17). U.S. Department of Agriculture, Forest Service, Rocky Mountain Research Station. http://www.treesearch.fs.fed.us/pubs/33332 | Matt Bowser                | 2020-08-07            | unverified         | Matt Bowser         | collector          | Annaleese T. Rasanen | collector          | media          | not applicable         | not applicable       | 1                   |                  |
| 31                     | 7d81a623-6964-4462-878e-b853cca0ebc3 | KNWRObs:Herb | 983      |                         |                   | mbowser   | KNWR-2020.02-SwanLake-ObsHerb | Phleum alpinum           | features       | Matt Bowser         | 2020-07-27 |                                                                                                                                                                                                            | 2020-07-27           | 2020-07-27  | 2020-07-27  | North America, United States, Alaska, Kenai Peninsula Borough | Mystery Hills off of the Skyline Trail, Skyline shrub study site 354-5                               | Mystery Hills off of the Skyline Trail, Skyline shrub study site 354-5                                | decimal degrees        | 60.52521111 | \-150.1637561 | World Geodetic System 1984 | Pre-determined plot coordinates | 30                   | m                 | MaNIS georeferencing guidelines |                                                 | wild caught        | recently burned low shrub tundra                      | observation           | 5.64 m radius, circular plot presence-only vegetation survey using the methods of Morton et al. (2009). Reference: Morton, J. M., Bowser, M. L., Berg, E., Magness, D., & Eskelin, T. (2009). Long Term Ecological Monitoring Program on the Kenai National Wildlife Refuge, Alaska: An FIA adjunct inventory. In W. McWilliams, G. Moisen, & R. Czaplewski (Eds.), 2008 Forest Inventory and Analysis (FIA) Symposium; October 21-23, 2008; Park City, UT (pp. 1-17). U.S. Department of Agriculture, Forest Service, Rocky Mountain Research Station. http://www.treesearch.fs.fed.us/pubs/33332 | Matt Bowser                | 2020-08-07            | unverified         | Matt Bowser         | collector          | Annaleese T. Rasanen | collector          | media          | not applicable         | not applicable       | 1                   |                  |
| 32                     | c8631c00-be15-40e3-859b-d4abeddd07eb | KNWRObs:Herb | 984      |                         |                   | mbowser   | KNWR-2020.02-SwanLake-ObsHerb | Calamagrostis canadensis | features       | Matt Bowser         | 2020-08-05 |                                                                                                                                                                                                            | 2020-08-05           | 2020-08-05  | 2020-08-05  | North America, United States, Alaska, Kenai Peninsula Borough | Mystery Hills off of the Skyline Trail, Skyline shrub study site 354-2                               | Mystery Hills off of the Skyline Trail, Skyline shrub study site 354-2                                | decimal degrees        | 60.52639136 | \-150.1659054 | World Geodetic System 1984 | Pre-determined plot coordinates | 30                   | m                 | MaNIS georeferencing guidelines |                                                 | wild caught        | former alder thicket defoliated by moths, then burned | observation           | 5.64 m radius, circular plot presence-only vegetation survey using the methods of Morton et al. (2009). Reference: Morton, J. M., Bowser, M. L., Berg, E., Magness, D., & Eskelin, T. (2009). Long Term Ecological Monitoring Program on the Kenai National Wildlife Refuge, Alaska: An FIA adjunct inventory. In W. McWilliams, G. Moisen, & R. Czaplewski (Eds.), 2008 Forest Inventory and Analysis (FIA) Symposium; October 21-23, 2008; Park City, UT (pp. 1-17). U.S. Department of Agriculture, Forest Service, Rocky Mountain Research Station. http://www.treesearch.fs.fed.us/pubs/33332 | Matt Bowser                | 2020-08-07            | unverified         | Matt Bowser         | collector          | Mark Laker           | collector          | media          | not applicable         | not applicable       | 1                   |                  |
| 33                     | 1aac2032-dedb-4788-aee7-65e2b4c252ab | KNWRObs:Herb | 985      |                         |                   | mbowser   | KNWR-2020.02-SwanLake-ObsHerb | Chamerion angustifolium  | features       | Matt Bowser         | 2020-08-05 |                                                                                                                                                                                                            | 2020-08-05           | 2020-08-05  | 2020-08-05  | North America, United States, Alaska, Kenai Peninsula Borough | Mystery Hills off of the Skyline Trail, Skyline shrub study site 354-2                               | Mystery Hills off of the Skyline Trail, Skyline shrub study site 354-2                                | decimal degrees        | 60.52639136 | \-150.1659054 | World Geodetic System 1984 | Pre-determined plot coordinates | 30                   | m                 | MaNIS georeferencing guidelines |                                                 | wild caught        | former alder thicket defoliated by moths, then burned | observation           | 5.64 m radius, circular plot presence-only vegetation survey using the methods of Morton et al. (2009). Reference: Morton, J. M., Bowser, M. L., Berg, E., Magness, D., & Eskelin, T. (2009). Long Term Ecological Monitoring Program on the Kenai National Wildlife Refuge, Alaska: An FIA adjunct inventory. In W. McWilliams, G. Moisen, & R. Czaplewski (Eds.), 2008 Forest Inventory and Analysis (FIA) Symposium; October 21-23, 2008; Park City, UT (pp. 1-17). U.S. Department of Agriculture, Forest Service, Rocky Mountain Research Station. http://www.treesearch.fs.fed.us/pubs/33332 | Matt Bowser                | 2020-08-07            | unverified         | Matt Bowser         | collector          | Mark Laker           | collector          | media          | not applicable         | not applicable       | 1                   |                  |
| 34                     | 56d49705-97e6-489b-b38b-e9d023e5b89a | KNWRObs:Herb | 986      |                         |                   | mbowser   | KNWR-2020.02-SwanLake-ObsHerb | Ceratodon purpureus      | features       | Matt Bowser         | 2020-08-05 |                                                                                                                                                                                                            | 2020-08-05           | 2020-08-05  | 2020-08-05  | North America, United States, Alaska, Kenai Peninsula Borough | Mystery Hills off of the Skyline Trail, Skyline shrub study site 354-2                               | Mystery Hills off of the Skyline Trail, Skyline shrub study site 354-2                                | decimal degrees        | 60.52639136 | \-150.1659054 | World Geodetic System 1984 | Pre-determined plot coordinates | 30                   | m                 | MaNIS georeferencing guidelines |                                                 | wild caught        | former alder thicket defoliated by moths, then burned | observation           | 5.64 m radius, circular plot presence-only vegetation survey using the methods of Morton et al. (2009). Reference: Morton, J. M., Bowser, M. L., Berg, E., Magness, D., & Eskelin, T. (2009). Long Term Ecological Monitoring Program on the Kenai National Wildlife Refuge, Alaska: An FIA adjunct inventory. In W. McWilliams, G. Moisen, & R. Czaplewski (Eds.), 2008 Forest Inventory and Analysis (FIA) Symposium; October 21-23, 2008; Park City, UT (pp. 1-17). U.S. Department of Agriculture, Forest Service, Rocky Mountain Research Station. http://www.treesearch.fs.fed.us/pubs/33332 | Matt Bowser                | 2020-08-07            | unverified         | Matt Bowser         | collector          | Mark Laker           | collector          | media          | not applicable         | not applicable       | 1                   |                  |
| 35                     | 3e477282-24c5-4ea2-abb4-3a30dbfd0164 | KNWRObs:Herb | 987      |                         |                   | mbowser   | KNWR-2020.02-SwanLake-ObsHerb | Heracleum maximum        | features       | Matt Bowser         | 2020-08-05 |                                                                                                                                                                                                            | 2020-08-05           | 2020-08-05  | 2020-08-05  | North America, United States, Alaska, Kenai Peninsula Borough | Mystery Hills off of the Skyline Trail, Skyline shrub study site 354-2                               | Mystery Hills off of the Skyline Trail, Skyline shrub study site 354-2                                | decimal degrees        | 60.52639136 | \-150.1659054 | World Geodetic System 1984 | Pre-determined plot coordinates | 30                   | m                 | MaNIS georeferencing guidelines |                                                 | wild caught        | former alder thicket defoliated by moths, then burned | observation           | 5.64 m radius, circular plot presence-only vegetation survey using the methods of Morton et al. (2009). Reference: Morton, J. M., Bowser, M. L., Berg, E., Magness, D., & Eskelin, T. (2009). Long Term Ecological Monitoring Program on the Kenai National Wildlife Refuge, Alaska: An FIA adjunct inventory. In W. McWilliams, G. Moisen, & R. Czaplewski (Eds.), 2008 Forest Inventory and Analysis (FIA) Symposium; October 21-23, 2008; Park City, UT (pp. 1-17). U.S. Department of Agriculture, Forest Service, Rocky Mountain Research Station. http://www.treesearch.fs.fed.us/pubs/33332 | Matt Bowser                | 2020-08-07            | unverified         | Matt Bowser         | collector          | Mark Laker           | collector          | media          | not applicable         | not applicable       | 1                   |                  |
| 36                     | c81ebe95-0c98-452f-be81-e6029f099eb1 | KNWRObs:Herb | 988      |                         |                   | mbowser   | KNWR-2020.02-SwanLake-ObsHerb | Sambucus racemosa        | features       | Matt Bowser         | 2020-08-05 |                                                                                                                                                                                                            | 2020-08-05           | 2020-08-05  | 2020-08-05  | North America, United States, Alaska, Kenai Peninsula Borough | Mystery Hills off of the Skyline Trail, Skyline shrub study site 354-2                               | Mystery Hills off of the Skyline Trail, Skyline shrub study site 354-2                                | decimal degrees        | 60.52639136 | \-150.1659054 | World Geodetic System 1984 | Pre-determined plot coordinates | 30                   | m                 | MaNIS georeferencing guidelines |                                                 | wild caught        | former alder thicket defoliated by moths, then burned | observation           | 5.64 m radius, circular plot presence-only vegetation survey using the methods of Morton et al. (2009). Reference: Morton, J. M., Bowser, M. L., Berg, E., Magness, D., & Eskelin, T. (2009). Long Term Ecological Monitoring Program on the Kenai National Wildlife Refuge, Alaska: An FIA adjunct inventory. In W. McWilliams, G. Moisen, & R. Czaplewski (Eds.), 2008 Forest Inventory and Analysis (FIA) Symposium; October 21-23, 2008; Park City, UT (pp. 1-17). U.S. Department of Agriculture, Forest Service, Rocky Mountain Research Station. http://www.treesearch.fs.fed.us/pubs/33332 | Matt Bowser                | 2020-08-07            | unverified         | Matt Bowser         | collector          | Mark Laker           | collector          | media          | not applicable         | not applicable       | 1                   |                  |
| 37                     | 19ac833b-e8a6-4322-a74f-4939b5a49e08 | KNWRObs:Herb | 989      |                         |                   | mbowser   | KNWR-2020.02-SwanLake-ObsHerb | Veratrum viride          | features       | Matt Bowser         | 2020-08-05 |                                                                                                                                                                                                            | 2020-08-05           | 2020-08-05  | 2020-08-05  | North America, United States, Alaska, Kenai Peninsula Borough | Mystery Hills off of the Skyline Trail, Skyline shrub study site 354-2                               | Mystery Hills off of the Skyline Trail, Skyline shrub study site 354-2                                | decimal degrees        | 60.52639136 | \-150.1659054 | World Geodetic System 1984 | Pre-determined plot coordinates | 30                   | m                 | MaNIS georeferencing guidelines |                                                 | wild caught        | former alder thicket defoliated by moths, then burned | observation           | 5.64 m radius, circular plot presence-only vegetation survey using the methods of Morton et al. (2009). Reference: Morton, J. M., Bowser, M. L., Berg, E., Magness, D., & Eskelin, T. (2009). Long Term Ecological Monitoring Program on the Kenai National Wildlife Refuge, Alaska: An FIA adjunct inventory. In W. McWilliams, G. Moisen, & R. Czaplewski (Eds.), 2008 Forest Inventory and Analysis (FIA) Symposium; October 21-23, 2008; Park City, UT (pp. 1-17). U.S. Department of Agriculture, Forest Service, Rocky Mountain Research Station. http://www.treesearch.fs.fed.us/pubs/33332 | Matt Bowser                | 2020-08-07            | unverified         | Matt Bowser         | collector          | Mark Laker           | collector          | media          | not applicable         | not applicable       | 1                   |                  |
| 38                     | 7f092b3c-80e4-4725-9da4-d24cb2332ec0 | KNWRObs:Herb | 990      |                         |                   | mbowser   | KNWR-2020.02-SwanLake-ObsHerb | Alnus viridis            | features       | Matt Bowser         | 2020-08-05 |                                                                                                                                                                                                            | 2020-08-05           | 2020-08-05  | 2020-08-05  | North America, United States, Alaska, Kenai Peninsula Borough | Mystery Hills off of the Skyline Trail, Skyline shrub study site 354-2                               | Mystery Hills off of the Skyline Trail, Skyline shrub study site 354-2                                | decimal degrees        | 60.52639136 | \-150.1659054 | World Geodetic System 1984 | Pre-determined plot coordinates | 30                   | m                 | MaNIS georeferencing guidelines |                                                 | wild caught        | former alder thicket defoliated by moths, then burned | observation           | 5.64 m radius, circular plot presence-only vegetation survey using the methods of Morton et al. (2009). Reference: Morton, J. M., Bowser, M. L., Berg, E., Magness, D., & Eskelin, T. (2009). Long Term Ecological Monitoring Program on the Kenai National Wildlife Refuge, Alaska: An FIA adjunct inventory. In W. McWilliams, G. Moisen, & R. Czaplewski (Eds.), 2008 Forest Inventory and Analysis (FIA) Symposium; October 21-23, 2008; Park City, UT (pp. 1-17). U.S. Department of Agriculture, Forest Service, Rocky Mountain Research Station. http://www.treesearch.fs.fed.us/pubs/33332 | Matt Bowser                | 2020-08-07            | unverified         | Matt Bowser         | collector          | Mark Laker           | collector          | media          | not applicable         | not applicable       | 1                   |                  |
| 39                     | 33f401b2-aa6b-4290-89d1-a19cdca3e667 | KNWRObs:Herb | 991      |                         |                   | mbowser   | KNWR-2020.02-SwanLake-ObsHerb | Populus tremuloides      | features       | Matt Bowser         | 2020-08-05 | Seedlings                                                                                                                                                                                                  | 2020-08-05           | 2020-08-05  | 2020-08-05  | North America, United States, Alaska, Kenai Peninsula Borough | Mystery Hills off of the Skyline Trail, Skyline shrub study site 354-2                               | Mystery Hills off of the Skyline Trail, Skyline shrub study site 354-2                                | decimal degrees        | 60.52639136 | \-150.1659054 | World Geodetic System 1984 | Pre-determined plot coordinates | 30                   | m                 | MaNIS georeferencing guidelines |                                                 | wild caught        | former alder thicket defoliated by moths, then burned | observation           | 5.64 m radius, circular plot presence-only vegetation survey using the methods of Morton et al. (2009). Reference: Morton, J. M., Bowser, M. L., Berg, E., Magness, D., & Eskelin, T. (2009). Long Term Ecological Monitoring Program on the Kenai National Wildlife Refuge, Alaska: An FIA adjunct inventory. In W. McWilliams, G. Moisen, & R. Czaplewski (Eds.), 2008 Forest Inventory and Analysis (FIA) Symposium; October 21-23, 2008; Park City, UT (pp. 1-17). U.S. Department of Agriculture, Forest Service, Rocky Mountain Research Station. http://www.treesearch.fs.fed.us/pubs/33332 | Matt Bowser                | 2020-08-07            | unverified         | Matt Bowser         | collector          | Mark Laker           | collector          | media          | not applicable         | not applicable       | 1                   |                  |
| 40                     | 4ea03d89-3933-4b0c-9259-bd19054defae | KNWR:Herb    | 12444    | iNaturalist.org         | 57466121          | mbowser   | KNWR-2020.02-misc-Herb        | Veronica peregrina       | features       | Matt Bowser         | 2020-08-24 |                                                                                                                                                                                                            | 20.August.2020       | 2020-08-20  | 2020-08-20  | North America, United States, Alaska, Kenai Peninsula Borough | Mystery Creek Road. Gravel pit at 60.58768 N, 150.313995 W.                                          | USA: Alaska, Kenai Peninsula, Mystery Creek Road. Gravel pit at 60.58768 N, 150.313995 W.             | decimal degrees        | 60.58768    | \-150.313995  | World Geodetic System 1984 | Olympus TG-6                    | 15                   | m                 | MaNIS georeferencing guidelines |                                                 | wild caught        | roadside gravel pit                                   | collection            | non-native plant survey                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                            | Matt Bowser                | 2020-08-24            | unverified         | Matt Bowser         | collector          | Annaleese T. Rasanen | collector          | whole organism | good                   | being processed      | 1                   | KNWRC4488        |
| 41                     | 1ed45d1f-15df-4463-96ac-36ce8d720f23 | KNWR:Herb    | 12445    | iNaturalist.org         | 57584513          | mbowser   | KNWR-2020.02-misc-Herb        | Caltha natans            | features       | Matt Bowser         | 2020-08-25 |                                                                                                                                                                                                            | 24.August.2020       | 2020-08-25  | 2020-08-25  | North America, United States, Alaska, Kenai Peninsula Borough | Canoe Lake 1 at portage between Canoe Lake 1 and Canoe Lake 2                                        | USA: Alaska, Kenai Peninsula, Canoe Lake 1 at portage between Canoe Lake 1 and Canoe Lake 2           | decimal degrees        | 60.7127     | \-150.694075  | World Geodetic System 1984 | Olympus TG-6                    | 15                   | m                 | MaNIS georeferencing guidelines |                                                 | wild caught        | lakeshore at stream inlet                             | collection            |                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                    | Matt Bowser                | 2020-08-25            | unverified         | Matt Bowser         | collector          | Benyamin K. Wishnek  | collector          | whole organism | good                   | being processed      | 1                   | KNWRC4419        |
| 42                     | 99b651da-04bb-4b29-b2a3-80d9dedbaa22 | KNWR:Herb    | 12446    | iNaturalist.org         | 33451572          | mbowser   | KNWR-2019.01-misc-Herb        | Sagittaria cuneata       | features       | Matt Bowser         | 2019-09-26 |                                                                                                                                                                                                            | 24.Sept.2019         | 2019-09-24  | 2019-09-24  | North America, United States, Alaska, Kenai Peninsula Borough | Canoe Lake, Swanson River Road. At put-in at Canoe Lake.                                             | Canoe Lake, Swanson River Road. At put-in at Canoe Lake.                                              | decimal degrees        | 60.721758   | \-150.700083  | World Geodetic System 1984 | Olympus TG-6                    | 15                   | m                 | MaNIS georeferencing guidelines |                                                 | wild caught        | lake shore                                            | collection            |                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                    | Matt Bowser                | 2020-08-26            | unverified         | Matt Bowser         | collector          | Takeyuki Nakamura    | collector          | whole organism | good                   | being processed      | 1                   | KNWRC4454        |
| 43                     | 231ea657-b575-45fc-a14b-fa08193f7098 | KNWR:Inv     | 47       |                         |                   | mbowser   | KNWR-2020.01-Baumann-Inv      | Lumbricidae              | features       | Tyler Baumann       | 2020-02-12 |                                                                                                                                                                                                            | 2019-09-09           | 2019-09-09  | 2019-09-09  | North America, United States, Alaska                          | Wolf Lake 5-1                                                                                        | Wolf Lake 5-1                                                                                         | decimal degrees        | 61.64375    | \-149.27388   | World Geodetic System 1984 | GPS                             | 20                   | m                 | MaNIS georeferencing guidelines | Coordinate accuracy not provided. Assumed 20 m. | wild caught        |                                                       | collection            |                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                    | Matt Bowser                | 2020-09-01            | unverified         | Tyler Baumann       | collector          | Adrian Wackett       | collector          | whole organism | condition not assessed | being processed      | 3                   |                  |
| 44                     | b6fc0d85-52dd-431a-8d1a-96c55432b16f | KNWR:Inv     | 48       |                         |                   | mbowser   | KNWR-2020.01-Baumann-Inv      | Bimastos parvus          | features       | Tyler Baumann       | 2020-02-12 |                                                                                                                                                                                                            | 2019-08-09           | 2019-08-09  | 2019-08-09  | North America, United States, Alaska                          | Chena Hot Springs G3                                                                                 | Chena Hot Springs G3                                                                                  | decimal degrees        | 65.05209    | \-146.05615   | World Geodetic System 1984 | GPS                             | 20                   | m                 | MaNIS georeferencing guidelines | Coordinate accuracy not provided. Assumed 20 m. | wild caught        |                                                       | collection            |                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                    | Matt Bowser                | 2020-09-01            | unverified         | Tyler Baumann       | collector          | Adrian Wackett       | collector          | whole organism | condition not assessed | being processed      | 3                   |                  |
| 45                     | ac35ebf4-fd5f-45e1-9efc-69bceae09287 | KNWR:Inv     | 49       |                         |                   | mbowser   | KNWR-2020.01-Baumann-Inv      | Lumbricidae              | features       | Tyler Baumann       | 2020-02-12 |                                                                                                                                                                                                            | 2019-08-09           | 2019-08-09  | 2019-08-09  | North America, United States, Alaska                          | Chena Hot Springs G4                                                                                 | Chena Hot Springs G4                                                                                  | decimal degrees        | 65.05211    | \-146.05625   | World Geodetic System 1984 | GPS                             | 20                   | m                 | MaNIS georeferencing guidelines | Coordinate accuracy not provided. Assumed 20 m. | wild caught        |                                                       | collection            |                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                    | Matt Bowser                | 2020-09-01            | unverified         | Tyler Baumann       | collector          | Adrian Wackett       | collector          | whole organism | condition not assessed | being processed      | 3                   |                  |
| 46                     | ed2d9c37-999a-4037-85cf-d4192b3c2a99 | KNWR:Inv     | 50       |                         |                   | mbowser   | KNWR-2020.01-Baumann-Inv      | Aporrectodea caliginosa  | features       | Tyler Baumann       | 2020-02-12 |                                                                                                                                                                                                            | 2019-10-04           | 2019-10-04  | 2019-10-04  | North America, United States, Alaska                          | Behnke 2-3                                                                                           | Behnke 2-3                                                                                            | decimal degrees        | 58.25991    | \-134.32252   | World Geodetic System 1984 | GPS                             | 20                   | m                 | MaNIS georeferencing guidelines | Coordinate accuracy not provided. Assumed 20 m. | wild caught        |                                                       | collection            |                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                    | Matt Bowser                | 2020-09-01            | unverified         | Tyler Baumann       | collector          | Adrian Wackett       | collector          | whole organism | condition not assessed | being processed      | 1                   |                  |
| 47                     | d2658e51-db1c-4e05-a222-d25935611406 | KNWR:Inv     | 51       |                         |                   | mbowser   | KNWR-2020.01-Baumann-Inv      | Aporrectodea             | features       | Tyler Baumann       | 2020-02-12 |                                                                                                                                                                                                            | 2019-10-03           | 2019-10-03  | 2019-10-03  | North America, United States, Alaska                          | Behnke 3-1                                                                                           | Behnke 3-1                                                                                            | decimal degrees        | 58.26002    | \-134.32137   | World Geodetic System 1984 | GPS                             | 20                   | m                 | MaNIS georeferencing guidelines | Coordinate accuracy not provided. Assumed 20 m. | wild caught        |                                                       | collection            |                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                    | Matt Bowser                | 2020-09-01            | unverified         | Tyler Baumann       | collector          | Adrian Wackett       | collector          | whole organism | condition not assessed | being processed      | 1                   |                  |
| 48                     | b9d029cc-c607-4ce3-9718-75bf7d558468 | KNWR:Inv     | 52       |                         |                   | mbowser   | KNWR-2020.01-Baumann-Inv      | Aporrectodea limicola    | features       | Tyler Baumann       | 2020-02-12 |                                                                                                                                                                                                            | 2019-10-04           | 2019-10-04  | 2019-10-04  | North America, United States, Alaska                          | Behnke 3-3                                                                                           | Behnke 3-3                                                                                            | decimal degrees        | 58.26007    | \-134.32147   | World Geodetic System 1984 | GPS                             | 20                   | m                 | MaNIS georeferencing guidelines | Coordinate accuracy not provided. Assumed 20 m. | wild caught        |                                                       | collection            |                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                    | Matt Bowser                | 2020-09-01            | unverified         | Tyler Baumann       | collector          | Adrian Wackett       | collector          | whole organism | condition not assessed | being processed      | 3                   |                  |
| 49                     | c5e894b6-8038-4105-a11e-ab65f095fbab | KNWR:Herb    | 12447    | iNaturalist.org         | 59171269          | mbowser   | KNWR-2020.02-misc-Herb        | Torreyochloa pallida     | features       | Matt Bowser         | 2020-09-10 |                                                                                                                                                                                                            | 10.September.2020    | 2020-09-10  | 2020-09-10  | North America, United States, Alaska, Kenai Peninsula Borough | Sucker Creek immediately downstream of the culvert under Swan Lake Road                              | USA: Alaska, Kenai Peninsula, Sucker Creek immediately downstream of the culvert under Swan Lake Road | decimal degrees        | 60.72511    | \-150.676132  | World Geodetic System 1984 | Olympus TG-6                    | 20                   | m                 | MaNIS georeferencing guidelines |                                                 | wild caught        | streamside                                            | collection            |                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                    | Matt Bowser                | 2020-09-10            | unverified         | Matt Bowser         | collector          |                      |                    | whole organism | good                   | being processed      | 1                   |                  |
| 50                     | 5b66999a-687f-4bf9-a194-676b5e00b8e9 | KNWR:Herb    | 12448    | iNaturalist.org         | 59163202          | mbowser   | KNWR-2020.02-misc-Herb        | Spirodela polyrhiza      | features       | Matt Bowser         | 2020-09-10 |                                                                                                                                                                                                            | 10.September.2020    | 2020-09-10  | 2020-09-10  | North America, United States, Alaska, Kenai Peninsula Borough | Sucker Creek immediately upstream of the culvert under Swan Lake Road                                | USA: Alaska, Kenai Peninsula, Sucker Creek immediately upstream of the culvert under Swan Lake Road   | decimal degrees        | 60.724743   | \-150.678206  | World Geodetic System 1984 | 2020 Google Earth               | 30                   | m                 | MaNIS georeferencing guidelines |                                                 | wild caught        | surface of quiet part of stream                       | collection            |                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                    | Matt Bowser                | 2020-09-10            | unverified         | Matt Bowser         | collector          |                      |                    | whole organism | good                   | being processed      | 1                   |                  |
| 51                     | 2dbbaa01-0070-40cd-a43f-718f78f42502 | KNWR:Inv     | 55       |                         |                   | mbowser   | KNWR-2020.02-misc-Inv         | Bimastos rubidus         | features       | Matt Bowser         | 2020-09-23 |                                                                                                                                                                                                            | 16-20.September.2020 | 2020-09-16  | 2020-09-20  | North America, United States, Alaska                          | Lower Copper Lake                                                                                    | USA: Alaska, Lower Copper Lake                                                                        | decimal degrees        | 59.5620165  | \-154.2179026 | World Geodetic System 1984 | 2020 Google Earth               | 1.3                  | km                | MaNIS georeferencing guidelines |                                                 | wild caught        |                                                       | collection            |                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                    | Matt Bowser                | 2020-09-23            | unverified         | Mark Ervice         | collector          |                      |                    | whole organism | good                   | being processed      | 8                   | UAM100185988     |

## LifeScanner vials with missing data

| human\_readable\_label | qr\_code       |
| :--------------------- | :------------- |
| BOLD-EB0               | BOLD-3NJSCKEB0 |

## Data for fungal specimens to be sequenced

Data exported from the [2020 Kenai Peninsula fungal
sequencing](https://www.inaturalist.org/projects/2020-kenai-peninsula-fungal-sequencing)
iNaturalist project following the [guidelines of the Fungal Diversity
Survey](https://fundis.org/sequence/sequence/submit-tissue).

| id       | observed\_on | user\_login | url                                               | image\_url                                                            | description                                                                        | place\_guess                       | latitude      | longitude        | place\_county\_name | place\_state\_name | place\_country\_name | scientific\_name    | taxon\_phylum\_name | taxon\_class\_name | taxon\_order\_name | taxon\_family\_name | taxon\_genus\_name | taxon\_species\_name |
| :------- | :----------- | :---------- | :------------------------------------------------ | :-------------------------------------------------------------------- | :--------------------------------------------------------------------------------- | :--------------------------------- | :------------ | :--------------- | :------------------ | :----------------- | :------------------- | :------------------ | :------------------ | :----------------- | :----------------- | :------------------ | :----------------- | :------------------- |
| 46129997 | 2020-05-16   | mbowser     | https://www.inaturalist.org/observations/46129997 | https://static.inaturalist.org/photos/73163328/medium.jpeg?1589657840 | 2020-05-16\_MLB01, under cottonwoods and spruce in mixed forest                    | Kalifornsky, AK, USA               | 60.3637540556 | \-151.2697553889 | Kenai Peninsula     | Alaska             | United States        | Morchella           | Ascomycota          | Pezizomycetes      | Pezizales          | Morchellaceae       | Morchella          |                      |
| 46132276 | 2020-05-16   | mbowser     | https://www.inaturalist.org/observations/46132276 | https://static.inaturalist.org/photos/73166513/medium.jpeg?1589658711 | 2020-05-16\_MLB02, under spruce and birch                                          | Kalifornsky, AK, USA               | 60.36399351   | \-151.26916317   | Kenai Peninsula     | Alaska             | United States        | Morchella           | Ascomycota          | Pezizomycetes      | Pezizales          | Morchellaceae       | Morchella          |                      |
| 46292862 | 2020-05-17   | mbowser     | https://www.inaturalist.org/observations/46292862 | https://static.inaturalist.org/photos/73418758/medium.jpeg?1589748649 | 2020-05-17\_MLB01, under cottonwoods                                               | Soldotna, AK 99669, USA            | 60.4632655833 | \-151.0784531667 | Kenai Peninsula     | Alaska             | United States        | Morchella           | Ascomycota          | Pezizomycetes      | Pezizales          | Morchellaceae       | Morchella          |                      |
| 46293272 | 2020-05-17   | mbowser     | https://www.inaturalist.org/observations/46293272 | https://static.inaturalist.org/photos/73419346/medium.jpeg?1589748803 | 2020-05-17\_MLB02, under cottonwoods                                               | Soldotna, AK 99669, USA            | 60.4633523889 | \-151.0786019722 | Kenai Peninsula     | Alaska             | United States        | Morchella           | Ascomycota          | Pezizomycetes      | Pezizales          | Morchellaceae       | Morchella          |                      |
| 47411794 | 2020-05-19   | colcan927   | https://www.inaturalist.org/observations/47411794 | https://static.inaturalist.org/photos/75195597/medium.jpg?1590509826  |                                                                                    | Mystery Creek Rd, Sterling, AK, US | 60.54915333   | \-150.27118      | Kenai Peninsula     | Alaska             | United States        | Morchella tomentosa | Ascomycota          | Pezizomycetes      | Pezizales          | Morchellaceae       | Morchella          | Morchella tomentosa  |
| 47411855 | 2020-05-19   | colcan927   | https://www.inaturalist.org/observations/47411855 | https://static.inaturalist.org/photos/75195705/medium.jpg?1590509866  |                                                                                    | Mystery Creek Rd, Sterling, AK, US | 60.5491       | \-150.27108833   | Kenai Peninsula     | Alaska             | United States        | Morchella           | Ascomycota          | Pezizomycetes      | Pezizales          | Morchellaceae       | Morchella          |                      |
| 47524671 | 2020-05-22   | cahaba      | https://www.inaturalist.org/observations/47524671 | https://static.inaturalist.org/photos/75377675/medium.jpg?1590596949  | vert\_angle\_deg=-71.7 / horiz\_angle\_deg=8.0                                     | Kenai Peninsula Borough, AK, USA   | 60.52905      | \-150.2225972222 | Kenai Peninsula     | Alaska             | United States        | Morchella           | Ascomycota          | Pezizomycetes      | Pezizales          | Morchellaceae       | Morchella          |                      |
| 47524963 | 2020-05-22   | cahaba      | https://www.inaturalist.org/observations/47524963 | https://static.inaturalist.org/photos/75378086/medium.jpg?1590597064  | vert\_angle\_deg=-46.1 / horiz\_angle\_deg=-1.3                                    | Kenai Peninsula Borough, AK, USA   | 60.5306472222 | \-150.2244194444 | Kenai Peninsula     | Alaska             | United States        | Morchella           | Ascomycota          | Pezizomycetes      | Pezizales          | Morchellaceae       | Morchella          |                      |
| 47738982 | 2020-05-26   | mbowser     | https://www.inaturalist.org/observations/47738982 | https://static.inaturalist.org/photos/75726878/medium.jpeg?1590765903 | 2020-05-26\_MLB01, collected near spruce, birch, stunted cottonwoods, and grasses. | Kalifornsky, AK, USA               | 60.4484561274 | \-151.2819276111 | Kenai Peninsula     | Alaska             | United States        | Morchella           | Ascomycota          | Pezizomycetes      | Pezizales          | Morchellaceae       | Morchella          |                      |
| 47739517 | 2020-05-26   | mbowser     | https://www.inaturalist.org/observations/47739517 | https://static.inaturalist.org/photos/75727906/medium.jpeg?1590766290 | 2020-05-26\_MLB02, near spruce, birch, stunted cottonwoods, and grass.             | Kalifornsky, AK, USA               | 60.4484733333 | \-151.28173      | Kenai Peninsula     | Alaska             | United States        | Morchella           | Ascomycota          | Pezizomycetes      | Pezizales          | Morchellaceae       | Morchella          |                      |
| 48409941 | 2020-06-03   | mbowser     | https://www.inaturalist.org/observations/48409941 | https://static.inaturalist.org/photos/76812602/medium.jpeg?1591245301 | Under cottonwoods                                                                  | Kalifornsky, AK, USA               | 60.3975791944 | \-151.2938145833 | Kenai Peninsula     | Alaska             | United States        | Morchella           | Ascomycota          | Pezizomycetes      | Pezizales          | Morchellaceae       | Morchella          |                      |
| 50778241 | 2020-06-12   | mbowser     | https://www.inaturalist.org/observations/50778241 | https://static.inaturalist.org/photos/80652362/medium.jpeg?1593011658 |                                                                                    | Kenai Peninsula Borough, AK, USA   | 60.5974233333 | \-150.3945166667 | Kenai Peninsula     | Alaska             | United States        | Morchella           | Ascomycota          | Pezizomycetes      | Pezizales          | Morchellaceae       | Morchella          |                      |
| 50807719 | 2020-06-12   | mbowser     | https://www.inaturalist.org/observations/50807719 | https://static.inaturalist.org/photos/80698469/medium.jpeg?1593027640 |                                                                                    | Kenai Peninsula Borough, AK, USA   | 60.624565     | \-150.3236083333 | Kenai Peninsula     | Alaska             | United States        | Gyromitra           | Ascomycota          | Pezizomycetes      | Pezizales          | Discinaceae         | Gyromitra          |                      |
| 51704078 | 2020-07-01   | mbowser     | https://www.inaturalist.org/observations/51704078 | https://static.inaturalist.org/photos/82164530/medium.jpeg?1593706879 | Collected.                                                                         | Kenai Peninsula Borough, AK, USA   | 60.4875933333 | \-150.4623133333 | Kenai Peninsula     | Alaska             | United States        | Leccinum            | Basidiomycota       | Agaricomycetes     | Boletales          | Boletaceae          | Leccinum           |                      |
| 52167014 | 2020-07-02   | mbowser     | https://www.inaturalist.org/observations/52167014 | https://static.inaturalist.org/photos/82911543/medium.jpeg?1594056152 |                                                                                    | Kenai Peninsula Borough, AK, USA   | 60.54855      | \-150.2666983333 | Kenai Peninsula     | Alaska             | United States        | Morchella           | Ascomycota          | Pezizomycetes      | Pezizales          | Morchellaceae       | Morchella          |                      |
| 52167410 | 2020-07-02   | mbowser     | https://www.inaturalist.org/observations/52167410 | https://static.inaturalist.org/photos/82912084/medium.jpeg?1594056330 |                                                                                    | Kenai Peninsula Borough, AK, USA   | 60.5485566667 | \-150.2666483333 | Kenai Peninsula     | Alaska             | United States        | Morchella tomentosa | Ascomycota          | Pezizomycetes      | Pezizales          | Morchellaceae       | Morchella          | Morchella tomentosa  |
| 52382438 | 2020-07-07   | mbowser     | https://www.inaturalist.org/observations/52382438 | https://static.inaturalist.org/photos/83266253/medium.jpeg?1594227231 |                                                                                    | Kenai Peninsula Borough, AK, USA   | 60.4727516667 | \-150.415445     | Kenai Peninsula     | Alaska             | United States        | Morchella tomentosa | Ascomycota          | Pezizomycetes      | Pezizales          | Morchellaceae       | Morchella          | Morchella tomentosa  |
| 57686894 | 2020-08-26   | mbowser     | https://www.inaturalist.org/observations/57686894 | https://static.inaturalist.org/photos/92038251/medium.jpeg?1598483892 |                                                                                    | Kenai Peninsula Borough, AK, USA   | 60.4649783333 | \-151.0730483333 | Kenai Peninsula     | Alaska             | United States        | Agaricus            | Basidiomycota       | Agaricomycetes     | Agaricales         | Agaricaceae         | Agaricus           |                      |
| 58435560 | 2020-07-20   | mbowser     | https://www.inaturalist.org/observations/58435560 | https://static.inaturalist.org/photos/93294510/medium.jpeg?1599156890 | 2020-07-20-MLB01                                                                   | Kenai Peninsula Borough, AK, USA   | 60.469955     | \-150.358425     | Kenai Peninsula     | Alaska             | United States        | Morchella           | Ascomycota          | Pezizomycetes      | Pezizales          | Morchellaceae       | Morchella          |                      |
| 58435927 | 2020-07-20   | mbowser     | https://www.inaturalist.org/observations/58435927 | https://static.inaturalist.org/photos/93295093/medium.jpeg?1599157091 |                                                                                    | Kenai Peninsula Borough, AK, USA   | 60.46987      | \-150.3586616667 | Kenai Peninsula     | Alaska             | United States        | Morchella           | Ascomycota          | Pezizomycetes      | Pezizales          | Morchellaceae       | Morchella          |                      |

# Bibliography

<div id="refs" class="references">

<div id="ref-ak_region_data_stewardship_team_alaska_2020">

Alaska Region Data Stewardship Team. 2019a. Alaska Region Interim Data
Management User Guide (v.3).
doi:[10.7944/P9JF0RT6](https://doi.org/10.7944/P9JF0RT6).

</div>

<div id="ref-alaska_region_data_stewardship_team_alaska_2020">

Alaska Region Data Stewardship Team. 2019b. Alaska Region Interim
Metadata Guide. Available from
<https://ak-region-dst.gitbook.io/alaska-region-mdeditor-interim-user-guide/>
\[accessed 27 February 2020\].

</div>

<div id="ref-bella_Vogel_2013">

Bella, E. 2013. Kenai national wildlife refuge invasive plant survey
2013. Vogel lake.
doi:[10.7299/X7V40VHG](https://doi.org/10.7299/X7V40VHG).

</div>

<div id="ref-berg_post-fire_2000">

Berg, E.E. 2000. Post-fire forest regeneration on the western Kenai
Peninsula, Alaska; a forty-eight year record from nine permanent plots.
USFWS Kenai National Wildlife Refuge, Soldotna, Alaska. Available from
<https://www.fws.gov/uploadedFiles/Berg_EE_2000b.pdf>.

</div>

<div id="ref-berg_relationship_2010">

Berg, N.D., and Gese, E.M. 2010. Relationship between fecal pellet
counts and snowshoe hare density in western Wyoming. The Journal of
Wildlife Management **74**(8): 1745–1751.
doi:[10.2193/2009-354](https://doi.org/10.2193/2009-354).

</div>

<div id="ref-boutin_population_1995">

Boutin, S., Krebs, C.J., Boonstra, R., Dale, M.R.T., Hannon, S.J.,
Martin, K., Sinclair, A.R.E., Smith, J.N.M., Turkington, R., Blower, M.,
Byrom, A., Doyle, F.I., Doyle, C., Hik, D., Hofer, L., Hubbs, A.,
Karels, T., Murray, D.L., Nams, V., O’Donoghue, M., Rohner, C., and
Schweiger, S. 1995. Population changes of the vertebrate community
during a snowshoe hare cycle in Canada’s boreal forest. Oikos **74**(1):
69–80. doi:[10.2307/3545676](https://doi.org/10.2307/3545676).

</div>

<div id="ref-bowmer_identification_1995">

Bowmer, K.H., Jacobs, S.W.L., and Sainty, G.R. 1995. Identification,
biology and management of *Elodea canadensis*, Hydrocharitaceae. Journal
of Aquatic Plant Management **33**: 13–19. Available from
<http://www.apms.org/wp/wp-content/uploads/2012/10/v33p13.pdf>.

</div>

<div id="ref-bowser_key_2012">

Bowser, M.L. 2012. Key and checklist of the bristletails (Microcoryphia)
of America north of Mexico. Soldotna, Alaska.
doi:[10.13140/RG.2.2.19142.22087](https://doi.org/10.13140/RG.2.2.19142.22087).

</div>

<div id="ref-bowser_lab_notebook_2018">

Bowser, M.L. 2018. Lab notebook, 28.Sept.2017 - 30.August.2018. USFWS
Kenai National Wildlife Refuge, Soldotna, Alaska.
doi:[10.7299/X7736R6J](https://doi.org/10.7299/X7736R6J).

</div>

<div id="ref-bowser_work_2019">

Bowser, M.L. 2019a. Work journal 2019. Exported pdf version. U.S. Fish &
Wildlife Service, Kenai National Wildlife Refuge., Soldotna, Alaska.
doi:[10.7299/X7RB74XZ](https://doi.org/10.7299/X7RB74XZ).

</div>

<div id="ref-bowser_field_2019">

Bowser, M.L. 2019b. Field Notebook, June 11, 2018 - May 29, 2019. USFWS
Kenai National Wildlife Refuge, Soldotna, Alaska. Available from
<https://arctos.database.museum/media/10612204>.

</div>

<div id="ref-bowser_2020_4009469">

Bowser, M.L. 2020. Whole Genome Sequencing of birches (Betulaceae:
*Betula*) from the Kenai Peninsula, Alaska. Zenodo.
doi:[10.5281/zenodo.4009469](https://doi.org/10.5281/zenodo.4009469).

</div>

<div id="ref-bowser_raw_2020">

Bowser, M.L., and Bowser, A.M. 2020. Raw metagenomic data from gut
contents of *Dallia pectoralis* specimens collected in Kenai, Alaska, in
2019. Zenodo.
doi:[10.5281/zenodo.3731327](https://doi.org/10.5281/zenodo.3731327).

</div>

<div id="ref-bowser_dna_2018">

Bowser, M.L., Bowser, A.M., Bowser, E.L., Bowser, M.D., Collet, D.M.,
and Melvin, T. 2018. DNA barcoding Alaskan willow rosette gall makers
(Diptera: Cecidomyiidae: *Rabdophaga*). Newsletter of the Alaska
Entomological Society **11**(1): 8–14.
doi:[10.7299/X7833SBX](https://doi.org/10.7299/X7833SBX).

</div>

<div id="ref-bowser_et_al_2020">

Bowser, M.L., Brassfield, R., Dziergowski, A., Eskelin, T., Hester, J.,
Magness, D.R., McInnis, M., Melvin, T., Morton, J.M., and Stone, J.
2020. Towards conserving natural diversity: A biotic inventory by
observations, specimens, dna barcoding and high-throughput sequencing
methods. Biodiversity Data Journal **8**: e50124. Pensoft Publishers.
doi:[10.3897/BDJ.8.e50124](https://doi.org/10.3897/BDJ.8.e50124).

</div>

<div id="ref-braukmann_metabarcoding_2019">

Braukmann, T.W.A., Ivanova, N.V., Prosser, S.W.J., Elbrecht, V.,
Steinke, D., Ratnasingham, S., Waard, J.R. de, Sones, J.E., Zakharov,
E.V., and Hebert, P.D.N. 2019. Metabarcoding a diverse arthropod mock
community. Molecular Ecology Resources **19**(3): 711–727.
doi:[10.1111/1755-0998.13008](https://doi.org/10.1111/1755-0998.13008).

</div>

<div id="ref-buckley_animal_1954">

Buckley, J.L. 1954. Animal population fluctuations in Alaska–a history.
*In* Transactions of the Nineteenth North American Wildlife Conference,
March 8, 9, and 10, 1954. Wildlife Management Institute, Washington,
D.C. pp. 338–357.

</div>

<div id="ref-costello_non-native_2008">

Costello, D.M., and Lamberti, G.A. 2008. Non-native earthworms in
riparian soils increase nitrogen flux into adjacent aquatic ecosystems.
Oecologia **158**(3): 499–510.
doi:[10.1007/s00442-008-1149-0](https://doi.org/10.1007/s00442-008-1149-0).

</div>

<div id="ref-costello_biological_2009">

Costello, D.M., and Lamberti, G.A. 2009. Biological and physical effects
of non-native earthworms on nitrogen cycling in riparian soils. Soil
Biology and Biochemistry **41**(10): 2230–2235.
doi:[10.1016/j.soilbio.2009.08.007](https://doi.org/10.1016/j.soilbio.2009.08.007).

</div>

<div id="ref-Criscuolo_2010">

Criscuolo, A., and Gribaldo, S. 2010. BMGE (block mapping and gathering
with entropy): A new software for selection of phylogenetic informative
regions from multiple sequence alignments. BMC Evolutionary Biology
**10**(1): 210. Springer Nature.
doi:[10.1186/1471-2148-10-210](https://doi.org/10.1186/1471-2148-10-210).

</div>

<div id="ref-Desper_2002">

Desper, R., and Gascuel, O. 2002. Fast and accurate phylogeny
reconstruction algorithms based on the minimum-evolution principle.
Journal of Computational Biology **9**(5): 687–705. Mary Ann Liebert
Inc.
doi:[10.1089/106652702761034136](https://doi.org/10.1089/106652702761034136).

</div>

<div id="ref-drake_american_1963">

Drake, C.J., and Lattin, J.D. 1963. American species of the lacebug
genus *Acalypta* (Hemiptera: Tingidae). Proceedings of the United States
National Museum **115**(3486): 331–345 +15 plates.
doi:[10.5479/si.00963801.115-3486.331](https://doi.org/10.5479/si.00963801.115-3486.331).

</div>

<div id="ref-Felis_et_al_2016">

Felis, J.J., Kissling, M.L., Kaler, R.S.A., Kenney, L.A., and Lawonn,
M.J. and. 2016. Identifying kittlitz’s murrelet nesting habitat in north
america at the landscape scale: U.S. Geological survey data release.
doi:[10.5066/F71V5C2W](https://doi.org/10.5066/F71V5C2W).

</div>

<div id="ref-fernandez_vagrant_2011">

Fernández, R., Almodóvar, A., Novo, M., Gutiérrez, M., and Díaz Cosín,
D.J. 2011. A vagrant clone in a peregrine species: Phylogeography, high
clonal diversity and geographical distribution in the earthworm
*Aporrectodea trapezoides* (Dugès, 1828). Soil Biology and Biochemistry
**43**(10): 2085–2093.
doi:[10.1016/j.soilbio.2011.06.007](https://doi.org/10.1016/j.soilbio.2011.06.007).

</div>

<div id="ref-friedersdorff_remote_1986">

Friedersdorff, J.W. 1986. Remote and roadside lake study - 1984. U.S.
Fish; Wildlife Service, Kenai Fishery Resources Station, Kenai, Alaska.
Available from
<http://www.fws.gov/uploadedFiles/Friedersdorff_JW_1986.pdf>.

</div>

<div id="ref-froeschner_zoogeographic_1976">

Froeschner, R.C. 1976. Zoogeographic notes on the lace bug genus
*Acalypta* Westwood in the Americas with description of a new species
from Mexico (Hemiptera: Tingidae). The American Midland Naturalist
**96**(2): 257–269.
doi:[10.2307/2424067](https://doi.org/10.2307/2424067).

</div>

<div id="ref-fs-r10-fhp_forest_2018">

FS-R10-FHP. 2018. Forest Health Conditions in Alaska - 2017. FHP
Protection Report, U.S. Forest Service, Alaska Region, Anchorage,
Alaska. Available from
<https://www.fs.usda.gov/Internet/FSE_DOCUMENTS/fseprd572286.pdf>.

</div>

<div id="ref-fs-r10-fhp_forest_2020">

FS-R10-FHP. 2020. Forest Health Conditions in Alaska 2019. FHP Report,
U.S. Forest Service, Alaska Region, Anchorage, Alaska. Available from
<https://www.fs.usda.gov/Internet/FSE_DOCUMENTS/fseprd712413.pdf>.

</div>

<div id="ref-gbiforg_gbif_2020-4">

GBIF.org. 2020a. GBIF Occurrence Download. GBIF.org.
doi:[10.15468/dl.ryvkwn](https://doi.org/10.15468/dl.ryvkwn).

</div>

<div id="ref-gbiforg_gbif_2020-5">

GBIF.org. 2020b. GBIF Occurrence Download. GBIF.org.
doi:[10.15468/dl.kxbpnt](https://doi.org/10.15468/dl.kxbpnt).

</div>

<div id="ref-gbiforg_gbif_2020-6">

GBIF.org. 2020c. GBIF Occurrence Download. GBIF.org.
doi:[10.15468/dl.tt25kv](https://doi.org/10.15468/dl.tt25kv).

</div>

<div id="ref-gbiforg_gbif_2020-7_Sandpiper">

GBIF.org. 2020d. GBIF Occurrence Download. GBIF.org.
doi:[10.15468/dl.sc5xmu](https://doi.org/10.15468/dl.sc5xmu).

</div>

<div id="ref-gbiforg_gbif_2020-8_Swan_Lake_Fire_plants">

GBIF.org. 2020e. GBIF Occurrence Download. GBIF.org.
doi:[10.15468/dl.f46pcs](https://doi.org/10.15468/dl.f46pcs).

</div>

<div id="ref-gbiforg_gbif_2020-1">

GBIF.org. 2020f. GBIF Occurrence Download. GBIF.org.
doi:[10.15468/dl.bguubm](https://doi.org/10.15468/dl.bguubm).

</div>

<div id="ref-gbiforg_gbif_2020">

GBIF.org. 2020g. GBIF Occurrence Download. GBIF.org.
doi:[10.15468/dl.3dfs26](https://doi.org/10.15468/dl.3dfs26).

</div>

<div id="ref-gbiforg_gbif_2020-2">

GBIF.org. 2020h. GBIF Occurrence Download. GBIF.org.
doi:[10.15468/dl.deymas](https://doi.org/10.15468/dl.deymas).

</div>

<div id="ref-gbiforg_gbif_2020-3">

GBIF.org. 2020i. GBIF Occurrence Download. GBIF.org.
doi:[10.15468/dl.azbzw2](https://doi.org/10.15468/dl.azbzw2).

</div>

<div id="ref-gbif_secretariat_enterion_2019">

GBIF Secretariat. 2019. *Enterion castaneum* Savigny, 1826. *In* GBIF
Backbone Taxonomy. Checklist dataset. GBIF.org.
doi:[10.15468/39omei](https://doi.org/10.15468/39omei).

</div>

<div id="ref-gross_unterscheidung_2003">

Gross, E.M., Erhard, D., Glaschke, A., and Haid, V. 2003. Unterscheidung
unklarer morphotypen von *Elodea* spp. Durch chemotaxonomische und
molekulartaxonomische methoden. Deutsche Gesellschaft für Limnologie
(DGL) - Tagungsbericht (Köln). Available from
<https://docplayer.org/7100923-Unterscheidung-unklarer-morphotypen-von-elodea-spp-durch-chemotaxonomische-und-molekulartaxonomische-methoden.html>.

</div>

<div id="ref-gweon_pipits_2015">

Gweon, H.S., Oliver, A., Taylor, J., Booth, T., Gibbs, M., Read, D.S.,
Griffiths, R.I., and Schonrogge, K. 2015. PIPITS: An automated pipeline
for analyses of fungal internal transcribed spacer sequences from the
Illumina sequencing platform. Methods in Ecology and Evolution **6**(8):
973–980.
doi:[10.1111/2041-210X.12399](https://doi.org/10.1111/2041-210X.12399).

</div>

<div id="ref-hajibabaei_coi_2019">

Hajibabaei, M., Porter, T.M., Wright, M., and Rudar, J. 2019a. COI
metabarcoding primer choice affects richness and recovery of indicator
taxa in freshwater systems. bioRxiv.
doi:[10.1101/572628](https://doi.org/10.1101/572628).

</div>

<div id="ref-hajibabaei_coi_2019b">

Hajibabaei, M., Porter, T.M., Wright, M., and Rudar, J. 2019b. COI
metabarcoding primer choice affects richness and recovery of indicator
taxa in freshwater systems. PLOS ONE **14**(9): e0220953.
doi:[10.1371/journal.pone.0220953](https://doi.org/10.1371/journal.pone.0220953).

</div>

<div id="ref-hedlin_cone_1974">

Hedlin, A.F. 1974. Cone and Seed Insects of British Columbia.
Environment Canada, Forestry Service, Pacific Forest Research Centre,
Victoria, British Columbia.

</div>

<div id="ref-hrobak_standard_2020">

Hrobak, J.L., and Barnes, J.L. 2020. Standard Operating Procedure (SOP)
2.5: Burn Severity Field Methods, Version 1.0. Fire Monitoring Protocol
for the Arctic Network: Fire Extent, Severity, and Effects. National
Park Service, Fairbanks, Alaska. Available from
<https://irma.nps.gov/DataStore/Reference/Profile/2272205>.

</div>

<div id="ref-huotari_comparative_2013">

Huotari, T., and Korpelainen, H. 2013. Comparative analyses of plastid
sequences between native and introduced populations of aquatic weeds
*Elodea canadensis* and *e*. *Nuttallii*. PLOS ONE **8**(4): e58073.
doi:[10.1371/journal.pone.0058073](https://doi.org/10.1371/journal.pone.0058073).

</div>

<div id="ref-huotari_population_2011">

Huotari, T., Korpelainen, H., Leskinen, E., and Kostamo, K. 2011.
Population genetics of the invasive water weed *Elodea canadensis* in
Finnish waterways. Plant Systematics and Evolution **294**(1): 27.
doi:[10.1007/s00606-011-0442-2](https://doi.org/10.1007/s00606-011-0442-2).

</div>

<div id="ref-interagency_arctic_research_coordinating_committee_department_1977">

Interagency Arctic Research Coordinating Committee. 1977. Terrestrial
biology and ecology: Department of Agriculture. Arctic Bulletin
**2**(11): 237–243. Available from
<https://books.google.com/books?id=yJTlyWTDBKoC>.

</div>

<div id="ref-jakubas_sandpiper_1984">

Jakubas, W.J., and Firman, A.S. 1984. Sandpiper Lake, 7/3/84. U.S. Fish
& Wildlife Service.
doi:[10.7299/X7JM29X1](https://doi.org/10.7299/X7JM29X1).

</div>

<div id="ref-Junier_2010">

Junier, T., and Zdobnov, E.M. 2010. The newick utilities:
High-throughput phylogenetic tree processing in the unix shell.
Bioinformatics **26**(13): 1669–1670. Oxford University Press (OUP).
doi:[10.1093/bioinformatics/btq243](https://doi.org/10.1093/bioinformatics/btq243).

</div>

<div id="ref-kadono_genetic_1997">

Kadono, Y., Nakamura, T., and Suzuki, T. 1997. Genetic uniformity of two
aquatic plants, *Egeria densa* Planch. And *Elodea nuttallii* (Planch.)
St. John, introduced in Japan. Japanese Journal of Limnology
(Rikusuigaku Zasshi) **58**(2): 197–203.
doi:[10.3739/rikusui.58.197](https://doi.org/10.3739/rikusui.58.197).

</div>

<div id="ref-Katoh_2013">

Katoh, K., and Standley, D.M. 2013. MAFFT multiple sequence alignment
software version 7: Improvements in performance and usability. Molecular
Biology and Evolution **30**(4): 772–780. Oxford University Press (OUP).
doi:[10.1093/molbev/mst010](https://doi.org/10.1093/molbev/mst010).

</div>

<div id="ref-krebs_using_2018">

Krebs, C.J., Boonstra, R., and Boutin, S. 2018. Using experimentation to
understand the 10-year snowshoe hare cycle in the boreal forest of North
America. Journal of Animal Ecology **87**(1): 87–100.
doi:[10.1111/1365-2656.12720](https://doi.org/10.1111/1365-2656.12720).

</div>

<div id="ref-krebs_estimating_2001">

Krebs, C.J., Boonstra, R., Nams, V., O’Donoghue, M., Hodges, K.E., and
Boutin, S. 2001. Estimating snowshoe hare population density from pellet
plots: A further evaluation. Canadian Journal of Zoology **79**(1): 1–4.
doi:[10.1139/z00-177](https://doi.org/10.1139/z00-177).

</div>

<div id="ref-krebs_estimation_1987">

Krebs, C.J., Gilbert, B.S., Boutin, S., and Boonstra, R. 1987.
Estimation of snowshoe hare population density from turd transects.
Canadian Journal of Zoology **65**(3): 565–567.
doi:[10.1139/z87-087](https://doi.org/10.1139/z87-087).

</div>

<div id="ref-langille_proposed_1904">

Langille, W.A. 1904. The proposed forest reserve on the Kenai Peninsula,
Alaska. United States Forest Service.
doi:[10.5962/bhl.title.124553](https://doi.org/10.5962/bhl.title.124553).

</div>

<div id="ref-Lefort_2015">

Lefort, V., Desper, R., and Gascuel, O. 2015. FastME 2.0: A
comprehensive, accurate, and fast distance-based phylogeny inference
program. Molecular Biology and Evolution **32**(10): 2798–2800. Oxford
University Press (OUP).
doi:[10.1093/molbev/msv150](https://doi.org/10.1093/molbev/msv150).

</div>

<div id="ref-Lemoine_2018">

Lemoine, F., Domelevo Entfellner, J.-B., Wilkinson, E., Correia, D.,
Dávila Felipe, M., De Oliveira, T., and Gascuel, O. 2018. Renewing
felsenstein’s phylogenetic bootstrap in the era of big data. Nature
**556**(7702): 452–456. Springer Nature.
doi:[10.1038/s41586-018-0043-0](https://doi.org/10.1038/s41586-018-0043-0).

</div>

<div id="ref-Letunic_et_al_2019">

Letunic, I., and Bork, P. 2019. Interactive Tree Of Life (iTOL) v4:
recent updates and new developments. Nucleic Acids Research **47**(W1):
W256–W259. doi:[10.1093/nar/gkz239](https://doi.org/10.1093/nar/gkz239).

</div>

<div id="ref-lipkin_investigation_2007">

Lipkin, R. 2007. Investigation of Potentially Sensitive Plant
Communities in the Old Dummy Burn, Kanuti National Wildlife Refuge,
Alaska, 2006. Alaska Natural Heritage Program, Environment; Natural
Resources Institute, University of Alaska Anchorage, Anchorage, Alaska.
Available from <https://www.arlis.org/docs/vol1/A/313839549.pdf>.

</div>

<div id="ref-marchand_oxygen_1985">

Marchand, P.J. 1985. Oxygen evolution by *Elodea canadensis* under ice
and snow cover: A case for winter photosynthesis in subnivean vascular
plants. Aquilo, Serie Botanica **23**: 57–61.

</div>

<div id="ref-martinsson_dna-barcoding_2015">

Martinsson, S., Cui, Y., Martin, P.J., Pinder, A., Quinlan, K., Wetzel,
M.J., and Erséus, C. 2015. DNA-barcoding of invasive European earthworms
(Clitellata: Lumbricidae) in south-western Australia. Biological
Invasions **17**(9): 2527–2532.
doi:[10.1007/s10530-015-0910-7](https://doi.org/10.1007/s10530-015-0910-7).

</div>

<div id="ref-mcalpine_manual_1983">

McAlpine, J.F., Peterson, B.V., Shewell, G.E., Teskey, H.J., Vockeroth,
J.R., and Wood, D.M. (*Editors*). 1983. Manual of Nearctic Diptera,
volume 1. Research Branch, Agriculture Canada, Ottawa, Ontario, Canada.

</div>

<div id="ref-McCann_using_2008">

McCann, N.P., Moen, R.A., and Niemi, G.J. 2008. Using pellet counts to
estimate snowshoe hare numbers in Minnesota. Journal of Wildlife
Management **72**(4): 955–958.
doi:[10.2193/2007-083](https://doi.org/10.2193/2007-083).

</div>

<div id="ref-mills_pellet_2005">

Mills, L.S., Griffin, P.C., Hodges, K.E., McKelvey, K., Ruggiero, L.,
and Ulizio, T. 2005. Pellet count indeces compared to mark-recapture
estimates for evaluating snowshoe hare density. Journal of Wildlife
Management **69**(3): 1053–1062.
doi:[10.2193/0022-541X(2005)069\[1053:PCICTM\]2.0.CO;2](https://doi.org/10.2193/0022-541X\(2005\)069%5B1053:PCICTM%5D2.0.CO;2).

</div>

<div id="ref-morton_ghosts_2014">

Morton, J. 2014. Ghosts of fires past. *In* Refuge Notebook. USFWS Kenai
National Wildlife Refuge, Soldotna, Alaska. pp. 45–46. Available from
<https://www.fws.gov/uploadedFiles/Region_7/NWRS/Zone_2/Kenai/Sections/What_We_Do/In_The_Community/Refuge_Notebooks/2014_Articles/Refuge_Notebook_v16_n23.pdf>.

</div>

<div id="ref-morton_responding_2019">

Morton, J., Bowser, M., and Eskelin, T. 2019, July. Responding rapidly
to Elodea—the first freshwater invasive plant in Alaska. San Diego,
California. Available from
<https://www.fws.gov/uploadedFiles/Morton_JM_et_al_2019.pdf>.

</div>

<div id="ref-murray_estimating_2002">

Murray, D.L., Roth, J.D., Ellsworth, E., Wirsing, A.J., and Steury, T.D.
2002. Estimating low-density snowshoe hare populations using fecal
pellet counts. Canadian Journal of Zoology **80**(4): 771–781.
doi:[10.1139/z02-027](https://doi.org/10.1139/z02-027).

</div>

<div id="ref-NOAA_beluga_habitat">

NOAA Fisheries. 2019. Beluga whale – cook inlet dps critical habitat.
Available from
<https://www.fisheries.noaa.gov/resource/map/beluga-whale-cook-inlet-dps-critical-habitat>.

</div>

<div id="ref-otis_statistical_1978">

Otis, D.L., Burnham, K.P., White, G.C., and Anderson, D.R. 1978.
Statistical Inference from Capture Data on Closed Animal Populations.
Wildlife Monographs (62): 3–135. Available from
<http://www.jstor.org/stable/3830650> \[accessed 15 September 2020\].

</div>

<div id="ref-packer_bee_genera_2007">

Packer, L., Genaro, J.A., and Sheffield, C.S. 2007. The bee genera of
eastern canada. Canadian Journal of Arthropod Identification **3**.
doi:[10.3752/cjai.2007.03](https://doi.org/10.3752/cjai.2007.03).

</div>

<div id="ref-paez_alpine_1991">

Paez, C.E. 1991. Alpine vegetation of areas utilized by introduced
populations of caribou (*Rangifer tarandus*) on the Kenai Peninsula,
Alaska. Master’s thesis, University of Wisconsin-Madison, Madison,
Wisconsin. Available from
<https://www.fws.gov/uploadedFiles/Paez_CE_1991.pdf>.

</div>

<div id="ref-paez_dynamics_2009">

Paez, C.E. 2009. Dynamics of alpine ranges and areas utilized by
introduced caribou populations on the Kenai Peninsula, Alaska. State of
Alaska, Alaska. Available from
<http://www.arlis.org/docs/vol1/A/325194251.pdf>.

</div>

<div id="ref-pinder_ecology_2013">

Pinder, R.A. 2013. Ecology of earthworms in riparian habitats. PhD
Thesis, University at Albany, State University of New York, Albany, New
York. Available from <https://search.proquest.com/docview/1419999664>.

</div>

<div id="ref-porco_reference_2018">

Porco, D., Chang, C.-H., Dupont, L., James, S., Richard, B., and
Decaëns, T. 2018. A reference library of DNA barcodes for the
earthworms from Upper Normandy: Biodiversity assessment, new records,
potential cases of cryptic diversity and ongoing speciation. Applied
Soil Ecology **124**: 362–371.
doi:[10.1016/j.apsoil.2017.11.001](https://doi.org/10.1016/j.apsoil.2017.11.001).

</div>

<div id="ref-pratt_notes_1980">

Pratt, G.K., and Pratt, H.D. 1980. Notes on Nearctic *Sylvicola*
(Diptera: Anisopodidae). Proceedings of the Entomological Society of
Washington **82**: 86–98.

</div>

<div id="ref-richard_true_2015">

Richard, F., Bellanger, J.-M., Clowez, P., Hansen, K., O’Donnell, K.,
Urban, A., Sauve, M., Courtecuisse, R., and Moreau, P.-A. 2015. True
morels (*Morchella*, Pezizales) of Europe and North America:
Evolutionary relationships inferred from multilocus data and a unified
taxonomy. Mycologia **107**(2): 359–382.
doi:[10.3852/14-166](https://doi.org/10.3852/14-166).

</div>

<div id="ref-robinson_hosts_2020">

Robinson, G.S., Ackery, P.R., Kitching, I.J., Beccaloni, G.W., and
Hernández, L.M. 2020. HOSTS - a database of the world’s lepidopteran
hostplants. Natural History Museum, London. Available from
<https://www.nhm.ac.uk/our-science/data/hostplants/index.html>.

</div>

<div id="ref-rowe_how_2020">

Rowe, C.A., Lichvar, R.W., and Wolf, P.G. 2020. How many tree species of
birch are in Alaska? Implications for wetland designations. Frontiers in
Plant Science **11**: 750.
doi:[10.3389/fpls.2020.00750](https://doi.org/10.3389/fpls.2020.00750).

</div>

<div id="ref-salojarvi_genome_2017">

Salojärvi, J., Smolander, O.-P., Nieminen, K., Rajaraman, S., Safronov,
O., Safdari, P., Lamminmäki, A., Immanen, J., Lan, T., Tanskanen, J.,
Rastas, P., Amiryousefi, A., Jayaprakash, B., Kammonen, J.I., Hagqvist,
R., Eswaran, G., Ahonen, V.H., Serra, J.A., Asiegbu, F.O., Dios
Barajas-Lopez, J. de, Blande, D., Blokhina, O., Blomster, T., Broholm,
S., Brosché, M., Cui, F., Dardick, C., Ehonen, S.E., Elomaa, P.,
Escamez, S., Fagerstedt, K.V., Fujii, H., Gauthier, A., Gollan, P.J.,
Halimaa, P., Heino, P.I., Himanen, K., Hollender, C., Kangasjärvi, S.,
Kauppinen, L., Kelleher, C.T., Kontunen-Soppela, S., Koskinen, J.P.,
Kovalchuk, A., Kärenlampi, S.O., Kärkönen, A.K., Lim, K.-J., Leppälä,
J., Macpherson, L., Mikola, J., Mouhu, K., Mähönen, A.P., Niinemets, Ü.,
Oksanen, E., Overmyer, K., Palva, E.T., Pazouki, L., Pennanen, V.,
Puhakainen, T., Poczai, P., Possen, B.J.H.M., Punkkinen, M., Rahikainen,
M.M., Rousi, M., Ruonala, R., Schoot, C. van der, Shapiguzov, A.,
Sierla, M., Sipilä, T.P., Sutela, S., Teeri, T.H., Tervahauta, A.I.,
Vaattovaara, A., Vahala, J., Vetchinnikova, L., Welling, A., Wrzaczek,
M., Xu, E., Paulin, L.G., Schulman, A.H., Lascoux, M., Albert, V.A.,
Auvinen, P., Helariutta, Y., and Kangasjärvi, J. 2017. Genome sequencing
and population genomic analyses provide insights into the adaptive
landscape of silver birch. Nature Genetics **49**: 904.
doi:[10.1038/ng.3862](https://doi.org/10.1038/ng.3862).

</div>

<div id="ref-Sato_et_al_Rabdophaga_2020">

Sato, S., Harris, K.M., Collet, D.M., Kim, W., and Yukawa, J. 2020.
Genetic variation in intraspecific populations of *Rabdophaga rosaria*
(Diptera: Cecidomyiidae) indicating possible diversification scenarios
into sibling species along with host range expansion on willows
(Salicaceae: *Salix*). Zoological Journal of the Linnean Society.
doi:[10.1093/zoolinnean/zlz179](https://doi.org/10.1093/zoolinnean/zlz179).

</div>

<div id="ref-sato_resurrection_2006">

Sato, S., and Yukawa, J. 2006. Resurrection of *Rabdophaga salicivora*
Shinji (Diptera: Cecidomyiidae), a Japanese gall midge formerly
misidentified as a North American species, *Rabdophaga rigidae* (Osten
Sacken), with observations on the phylogenetic relationships of its
populations in Japan and the Russian Far East. Entomological Science
**9**(4): 423–434.
doi:[10.1111/j.1479-8298.2006.00188.x](https://doi.org/10.1111/j.1479-8298.2006.00188.x).

</div>

<div id="ref-shorthouse_simplemappr_2010">

Shorthouse, D.P. 2010. SimpleMappr, an online tool to produce
publication-quality point maps. Available from
<https://www.simplemappr.net>.

</div>

<div id="ref-simpson_list_2019">

Simpson, A., Eyler, M.C., Sikes, D., Bowser, M., Sellers, E., Guala,
G.F., Cannister, M., Libby, R., and Kozlowski, N. 2019. A comprehensive
list of non-native species established in three major regions of the
united states: Version 2.0.
doi:[10.5066/P9E5K160](https://doi.org/10.5066/P9E5K160).

</div>

<div id="ref-skartveit_larvae_2002">

Skartveit, J. 2002. The larvae of European Bibioninae (Diptera,
Bibionidae). Journal of Natural History **36**(4): 449–485.
doi:[10.1080/00222930010023466](https://doi.org/10.1080/00222930010023466).

</div>

<div id="ref-state_of_alaska_dec_2005">

State of Alaska Department of Environmental Conservation. 2005. Ambient
water quality monitoring system. Available from
<https://awqms2.goldsystems.com/Login.aspx>.

</div>

<div id="ref-stefani_morchella_2010">

Stefani, F.O.P., Sokolski, S., Wurtz, T.L., Piché, Y., Hamelin, R.C.,
Fortin, J.A., and Bérubé, J.A. 2010. *Morchella tomentosa*: A unique
belowground structure and a new clade of morels. Mycologia **102**(5):
1082–1088. doi:[10.3852/09-294](https://doi.org/10.3852/09-294).

</div>

<div id="ref-stolz_nestling_2019">

Stolz, C. 2019. The nestling diet of Svalbard snow buntings identified
by DNA metabarcoding. Master’s thesis, The Arctic University of Norway.
Available from <https://hdl.handle.net/10037/15438>.

</div>

<div id="ref-usda_forest_service_alaska_region_forest_2005">

USDA Forest Service, Alaska Region. 2005. Forest Health Conditions in
Alaska—2004. USDA Forest Service, Alaska Region, Anchorage, Alaska.
Available from
<https://www.fs.usda.gov/Internet/FSE_DOCUMENTS/fsbdev2_038152.pdf>.

</div>

<div id="ref-usda_forest_service_alaska_region_forest_2006">

USDA Forest Service, Alaska Region. 2006. Forest Health Conditions in
Alaska—2005. Protection Report, USDA Forest Service, Alaska Region.
Available from
<https://www.fs.usda.gov/Internet/FSE_DOCUMENTS/fsbdev2_038151.pdf>.

</div>

<div id="ref-usda_forest_service_alaska_region_forest_2007">

USDA Forest Service, Alaska Region. 2007. Forest Health Conditions in
Alaska–2006. Protection Report, USDA Forest Service, Alaska Region.
Available from
<https://www.fs.usda.gov/Internet/FSE_DOCUMENTS/fsbdev2_037574.pdf>.

</div>

<div id="ref-usda_forest_service_alaska_region_forest_2008">

USDA Forest Service, Alaska Region. 2008. Forest Health Conditions in
Alaska—2007. Protection Report, United States Department of Agriculture
Forest Service, Alaska Region. Available from
<https://www.fs.usda.gov/Internet/FSE_DOCUMENTS/fsbdev2_038216.pdf>.

</div>

<div id="ref-wang_resolving_2020">

Wang, N., Kelly, L.J., McAllister, H.A., Zohren, J., and Buggs, R.J.A.
2020. Resolving phylogeny and polyploid parentage using genus-wide
genome-wide sequence data from birch trees. bioRxiv.
doi:[10.1101/2020.07.13.200444](https://doi.org/10.1101/2020.07.13.200444).

</div>

<div id="ref-wang_naive_2007">

Wang, Q., Garrity, G.M., Tiedje, J.M., and Cole, J.R. 2007. Naïve
Bayesian classifier for rapid assignment of rRNA sequences into the new
bacterial taxonomy. Applied and Environmental Microbiology **73**(16):
5261. doi:[10.1128/AEM.00062-07](https://doi.org/10.1128/AEM.00062-07).

</div>

</div>
