  - [January](#january)
      - [Thursday, January 2](#thursday-january-2)
      - [Friday, January 3](#friday-january-3)
      - [Monday, January 6](#monday-january-6)
      - [Tuesday, January 7](#tuesday-january-7)
      - [Wedesday, January 8](#wedesday-january-8)
      - [Thursday, January 9](#thursday-january-9)
      - [Friday, January 10](#friday-january-10)
      - [Monday, January 13](#monday-january-13)
      - [Tuesday, January 14](#tuesday-january-14)
      - [Wednesday, January 15](#wednesday-january-15)
      - [Thursday, January 16](#thursday-january-16)
      - [Friday, January 17](#friday-january-17)
      - [Monday, January 20](#monday-january-20)
      - [Tuesday, January 21](#tuesday-january-21)
      - [Wednesday, January 22](#wednesday-january-22)
      - [Thursday, January 23](#thursday-january-23)
      - [Friday, January 24](#friday-january-24)
      - [Monday, January 27](#monday-january-27)
      - [Tuesday, January 28](#tuesday-january-28)
      - [Wednesday, January 29](#wednesday-january-29)
      - [Thursday, January 30](#thursday-january-30)
      - [Friday, January 31](#friday-january-31)
  - [February](#february)
      - [Monday, February 3](#monday-february-3)
      - [Tuesday, February 4](#tuesday-february-4)
      - [Wednesday, February 5](#wednesday-february-5)
      - [Thursday, February 6](#thursday-february-6)
      - [Friday, February 7](#friday-february-7)
      - [Saturday, February 8](#saturday-february-8)
      - [Monday, February 10](#monday-february-10)
      - [Tuesday, February 11](#tuesday-february-11)
      - [Wednesday, February 12](#wednesday-february-12)
      - [Thursday, February 13](#thursday-february-13)
      - [Friday, February 14](#friday-february-14)
      - [Monday, February 17](#monday-february-17)
      - [Tuesday, February 18](#tuesday-february-18)
      - [Wednesday, February 19](#wednesday-february-19)
      - [Thursday, February 20](#thursday-february-20)
      - [Friday, February 21](#friday-february-21)
      - [Monday, February 24](#monday-february-24)
      - [Tuesday, February 25](#tuesday-february-25)
      - [Wednesday, February 26](#wednesday-february-26)
      - [Thursday, February 27](#thursday-february-27)
      - [Friday, February 28](#friday-february-28)
  - [March](#march)
      - [Monday, March 2](#monday-march-2)
      - [Tuesday, March 3](#tuesday-march-3)
      - [Wednesday, March 4](#wednesday-march-4)
      - [Thursday, March 5](#thursday-march-5)
      - [Friday, March 6](#friday-march-6)
      - [Sunday, March 8](#sunday-march-8)
      - [Monday, March 9](#monday-march-9)
      - [Tuesday, March 10](#tuesday-march-10)
      - [Thursday, March 12](#thursday-march-12)
      - [Friday, March 13](#friday-march-13)
      - [Sunday, March 15](#sunday-march-15)
      - [Monday, March 16](#monday-march-16)
      - [Tuesday, March 17](#tuesday-march-17)
      - [Wednesday, March 18](#wednesday-march-18)
      - [Thursday, March 19](#thursday-march-19)
      - [Friday, March 20](#friday-march-20)
      - [Saturday, March 21](#saturday-march-21)
      - [Monday, March 23](#monday-march-23)
      - [Tuesday, March 24](#tuesday-march-24)
      - [Wednesday, March 25](#wednesday-march-25)
      - [Thursday, March 26](#thursday-march-26)
      - [Friday, March 27](#friday-march-27)
  - [Bibliography](#bibliography)

# January

## Thursday, January 2

To do

  - *Refuge Notebook* catch-up.
  - ~~Add journal 2019 to Arctos~~.
  - Elodea RFP.
  - Visit site SK03?
  - Pesticide applicator CEUs.
  - Publish LTEMP bird data.
  - Finish sorting 2019 black spruce sweep net samples.

I posted my 2019 work journal to Arctos
(<http://arctos.database.museum/publication/10009014>,
<https://doi.org/10.7299/X7RB74XZ>). I also bulkloaded all [83 Arctos
specimen
citations](http://arctos.database.museum/SpecimenResults.cfm?publication_id=10009014).

I initialized a [journal2020](https://github.com/mlbowser/journal2020)
repository.

I sorted sweep net sample
[KNWR:Ento:11356](http://arctos.database.museum/guid/KNWR:Ento:11356).

Contents:

| identification | count |
| :------------- | ----: |
| Diptera        |    85 |
| Araneae        |    16 |
| Empidoidea     |     9 |
| Psyllidae      |     7 |
| Chironomidae   |     4 |
| Insecta        |     4 |
| Ichneumonoidea |     3 |
| Phoridae       |     3 |
| Culicidae      |     2 |
| Sciaroidea     |     2 |
| Tipulidae      |     2 |
| Psocodea       |     1 |
| Stratiomyidae  |     1 |
| Thysanoptera   |     1 |

I also sorted sweep net sample
[KNWR:Ento:11357](http://arctos.database.museum/guid/KNWR:Ento:11357).

Contents:

| identification        | count |
| :-------------------- | ----: |
| Diptera               |    31 |
| Araneae               |     8 |
| Symphypleona          |     7 |
| Hymenoptera           |     6 |
| Psyllidae             |     6 |
| Hemiptera             |     5 |
| Ichneumonoidea        |     5 |
| Aphididae             |     4 |
| Cicadellidae          |     4 |
| Symphyta              |     4 |
| Arachnida             |     3 |
| Lepidoptera           |     3 |
| *Cixius meridionalis* |     2 |
| Culicidae             |     2 |
| Insecta               |     2 |
| Sciaroidea            |     2 |
| Delphacidae           |     1 |
| Empidoidea            |     1 |
| Phoridae              |     1 |
| Simuliidae            |     1 |

## Friday, January 3

To do

  - *Refuge Notebook* catch-up.
  - Elodea RFP.
  - Visit site SK03?
  - Pesticide applicator CEUs.
  - Publish LTEMP bird data.
  - ~~Finish sorting 2019 black spruce sweep net samples.~~

I sorted sweep net sample
[KNWR:Ento:11358](http://arctos.database.museum/guid/KNWR:Ento:11358).

Contents:

| identification | count |
| :------------- | ----: |
| Aphididae      |    26 |
| Ichneumonoidea |    14 |
| Diptera        |     8 |
| Araneae        |     7 |
| Hybotidae      |     5 |
| Culicidae      |     3 |
| Lepidoptera    |     3 |
| Psocodea       |     3 |
| Sciaroidea     |     3 |
| Arachnida      |     2 |
| Hemiptera      |     2 |
| Hymenoptera    |     2 |
| Insecta        |     2 |
| Miridae        |     2 |
| Simuliidae     |     2 |
| Tabanidae      |     2 |
| Cicadellidae   |     1 |
| Coleoptera     |     1 |
| Elateridae     |     1 |
| Empidoidea     |     1 |
| Entomobryidae  |     1 |
| Symphyta       |     1 |
| Torymidae      |     1 |

I sorted sweep net sample
[KNWR:Ento:11359](http://arctos.database.museum/guid/KNWR:Ento:11359).

Contents:

| identification | count |
| :------------- | ----: |
| Aphididae      |    11 |
| Araneae        |     7 |
| Psocodea       |     6 |
| Diptera        |     4 |
| Culicidae      |     3 |
| Symphyta       |     3 |
| Empidoidea     |     2 |
| Sciaroidea     |     2 |
| Arachnida      |     1 |
| Elateridae     |     1 |
| Hemiptera      |     1 |
| Hybotidae      |     1 |
| Hymenoptera    |     1 |
| Ichneumonoidea |     1 |
| Miridae        |     1 |
| Simuliidae     |     1 |
| *Symphoromyia* |     1 |
| Tabanidae      |     1 |

I sorted sweep net sample
[KNWR:Ento:11360](http://arctos.database.museum/guid/KNWR:Ento:11360).

Contents:

| identification | count |
| :------------- | ----: |
| Araneae        |    11 |
| Psocodea       |     6 |
| Aphididae      |     5 |
| Ichneumonoidea |     5 |
| Diptera        |     3 |
| Hemiptera      |     2 |
| Miridae        |     2 |
| Arachnida      |     1 |
| Cicadellidae   |     1 |
| Hymenoptera    |     1 |
| Thysanoptera   |     1 |
| Torymidae      |     1 |

I sorted sweep net sample
[KNWR:Ento:11361](http://arctos.database.museum/guid/KNWR:Ento:11361),
the last of these sweep net samples\!

Contents:

| identification | count |
| :------------- | ----: |
| Culicidae      |     8 |
| Tabanidae      |     2 |
| Araneae        |     1 |
| Diptera        |     1 |
| Ichneumonoidea |     1 |
| Psocodea       |     1 |

Now I need to fill out RTL’s submission form.

``` r
## Downloaded data from Arctos (<http://arctos.database.museum/SpecimenResults.cfm?project_id=10003271&collection_id=46>).
data1 <- read.csv("ArctosData_BB6BD7A74E.csv", stringsAsFactors=FALSE)

## Now get barcodes.
partdetails <- strsplit(data1$PARTDETAIL, "] ")
barcode <- sapply(partdetails, "[", 2)
barcode <- strsplit(barcode, " \\(")
barcode <- sapply(barcode, "[", 1)
barcode
length(unique(barcode))
[1] 26 ## Good.

## Assemble identifiers into data frame.
data1$barcode <- barcode
identifiers <- data1[,c("GUID", "barcode")]
write.csv(identifiers, "2020-01-03-1341_identifiers.csv", row.names=FALSE)
```

I selected the mlCOIlintF/jgHCO2198
(GGWACWGGWTGAACWGTWTAYCCYCC/TAIACYTCIGGRTGICCRAARAAYCA) primer set based
on the analysis of Hajibabaei et
al. ([2019](#ref-hajibabaei_coi_2019)[a](#ref-hajibabaei_coi_2019)).

## Monday, January 6

To do

  - ~~Ship out 2019 black spruce sweep net samples.~~
  - Get updated prescription for Sandpiper Lake.
  - Find *Refuge Notebook* author for this week.
  - Review Sandpiper Lake MRDG.
  - *Refuge Notebook* catch-up.
  - Sandpiper Elodea RFP.
  - Visit site SK03?
  - Pesticide applicator CEUs.
  - Publish LTEMP bird data.
  - Review Programmatic Environmental Assessment of USFWS Management
    Strategy for Elodea and Other Submersed Aquatic Invasive Plants in
    the Alaska Region

I packaged sweep net samples and drove them to Kenai to ship them out to
RTL Genomics (FedEx tracking number: 813677694168).

Contents:

| GUID                                                                  | barcode      |
| :-------------------------------------------------------------------- | :----------- |
| [KNWR:Ento:11332](http://arctos.database.museum/guid/KNWR:Ento:11332) | UAM100185909 |
| [KNWR:Ento:11333](http://arctos.database.museum/guid/KNWR:Ento:11333) | UAM100185910 |
| [KNWR:Ento:11334](http://arctos.database.museum/guid/KNWR:Ento:11334) | UAM100185911 |
| [KNWR:Ento:11335](http://arctos.database.museum/guid/KNWR:Ento:11335) | UAM100185912 |
| [KNWR:Ento:11336](http://arctos.database.museum/guid/KNWR:Ento:11336) | UAM100185913 |
| [KNWR:Ento:11337](http://arctos.database.museum/guid/KNWR:Ento:11337) | UAM100185914 |
| [KNWR:Ento:11342](http://arctos.database.museum/guid/KNWR:Ento:11342) | UAM100185918 |
| [KNWR:Ento:11343](http://arctos.database.museum/guid/KNWR:Ento:11343) | UAM100185919 |
| [KNWR:Ento:11344](http://arctos.database.museum/guid/KNWR:Ento:11344) | UAM100185920 |
| [KNWR:Ento:11345](http://arctos.database.museum/guid/KNWR:Ento:11345) | UAM100185921 |
| [KNWR:Ento:11346](http://arctos.database.museum/guid/KNWR:Ento:11346) | UAM100185922 |
| [KNWR:Ento:11347](http://arctos.database.museum/guid/KNWR:Ento:11347) | UAM100185923 |
| [KNWR:Ento:11348](http://arctos.database.museum/guid/KNWR:Ento:11348) | UAM100185924 |
| [KNWR:Ento:11349](http://arctos.database.museum/guid/KNWR:Ento:11349) | UAM100185925 |
| [KNWR:Ento:11350](http://arctos.database.museum/guid/KNWR:Ento:11350) | UAM100185926 |
| [KNWR:Ento:11351](http://arctos.database.museum/guid/KNWR:Ento:11351) | UAM100185927 |
| [KNWR:Ento:11352](http://arctos.database.museum/guid/KNWR:Ento:11352) | UAM100185928 |
| [KNWR:Ento:11353](http://arctos.database.museum/guid/KNWR:Ento:11353) | UAM100185929 |
| [KNWR:Ento:11354](http://arctos.database.museum/guid/KNWR:Ento:11354) | UAM100185930 |
| [KNWR:Ento:11355](http://arctos.database.museum/guid/KNWR:Ento:11355) | UAM100185931 |
| [KNWR:Ento:11356](http://arctos.database.museum/guid/KNWR:Ento:11356) | UAM100185932 |
| [KNWR:Ento:11357](http://arctos.database.museum/guid/KNWR:Ento:11357) | UAM100185933 |
| [KNWR:Ento:11358](http://arctos.database.museum/guid/KNWR:Ento:11358) | UAM100185934 |
| [KNWR:Ento:11359](http://arctos.database.museum/guid/KNWR:Ento:11359) | UAM100185935 |
| [KNWR:Ento:11360](http://arctos.database.museum/guid/KNWR:Ento:11360) | UAM100185936 |
| [KNWR:Ento:11361](http://arctos.database.museum/guid/KNWR:Ento:11361) | UAM100185937 |

I chose the *mlCOIlintF*/*jgHCO2198*
(GGWACWGGWTGAACWGTWTAYCCYCC/TAIACYTCIGGRTGICCRAARAAYCA) primer set on
the submission form.

These specimens were added to loan number KNWR-2020.01-RTLGenomics-Ento.

I started communicating and planning for the spring herbicide
application to Sandpiper Lake.

I sorted most of Berlese sample
[KNWR:Ento:11362](http://arctos.database.museum/guid/KNWR:Ento:11362),
but this was difficult. I had a hard time transferring tiny mites from
the sample into a vial without also transferring debris. I did not
finish.

## Tuesday, January 7

To do

  - Get updated prescription for Sandpiper Lake.
  - Write *Refuge Notebook* article for this week.
  - Review Sandpiper Lake MRDG.
  - *Refuge Notebook* catch-up.
  - Sandpiper Elodea RFP.
  - Visit site SK03?
  - Pesticide applicator CEUs.
  - Publish LTEMP bird data.
  - Review Programmatic Environmental Assessment of USFWS Management
    Strategy for Elodea and Other Submersed Aquatic Invasive Plants in
    the Alaska Region

I received the technical review evaluation for the Slikok manuscript.
There are a number of requests for changes to the supplementary files.

I started a *Refuge Notebook* article on *Entomobrya nivalis*. I went on
a short walk looking for *E. nivalis* under bark of birches,
cottonwoods, and spruces. I found other arthropods, but no springtails.

I restarted writing a *Refuge Notebook* article, this time Walter Evans’
1897-1898 agricultural exploration of southern Alaska.

## Wedesday, January 8

To do

  - Submit *Refuge Notebook* article for this week.

I worked on my *Refuge Notebook* article, getting it submitted.

I entered data for *Hypnoidus bicolor* specimen
[KNWR:Ento:11363](http://arctos.database.museum/guid/KNWR:Ento:11363).
There was not a previusly entered georeferenced record from the Refuge.

## Thursday, January 9

Derek had entered the AWQMS (Ambient Water Quality Monitoring System)
dataset into UAMObs:Ento. I think the search URL below pulls these
records.

<https://arctos.database.museum/SpecimenResults.cfm?&entered_by=ffdss&beg_entered_date=2020-01-08&end_entered_date=2020-01-08>

I should check these for new Kenai NWR records.

I started addressing the issues with the supplementary files listed in
the technical review of the Slikok manuscript.

## Friday, January 10

I did some communicating regarding elodea eradication work at Sandpiper
Lake this spring.

I continued addressing issues with the supplementary files listed in the
technical review of the Slikok manuscript.

``` r
## Working on the raw occurrence data.
wd <- "D:/projects/Slikok_watershed/data/work_space/2020-01-10_supplementary_files"
setwd(wd)

od1 <- read.csv("oo_358153.csv", stringsAsFactors=FALSE)

summary(od1$COORDINATEUNCERTAINTYINMETERS)

## Looking at records with the smallest values.
od1[od1$COORDINATEUNCERTAINTYINMETERS==3,]
## Those are earthworms in the 0.25 m2 frames, so that is ok.

## Looking at records with the smallest values.
od1[od1$COORDINATEUNCERTAINTYINMETERS==810,]
## The LOCALITY_REMARKS field actually documents this well.

## Still, I will the number of significant figures to conform to the technical review.
od1$COORDINATEUNCERTAINTYINMETERS <- signif(od1$COORDINATEUNCERTAINTYINMETERS, 2)

levels(as.factor(od1$PARTS))
## Why do some have two values, e.g. "whole organism (fluid-preserved); whole organism (fluid-preserved)"?

od1[od1$PARTS=="whole organism (fluid-preserved); whole organism (fluid-preserved)",]
## That actually looked good. These are specimen lots where some individuals were sent in for sequencing and others were retained. I will just clean this up as requested, though.
od1$PARTS[!od1$PARTS=="media"] <- "whole organism"

## Rounding coordinates to 5 decimal places.
od1$DEC_LAT <- round(od1$DEC_LAT, 5)
od1$DEC_LONG <- round(od1$DEC_LONG, 5)

## Saving here.
write.csv(od1, "2020-01-10-1113_occurrences.csv", row.names=FALSE)

## Now for the analysis dataset.
od1 <- read.csv("oo_359083.csv", stringsAsFactors=FALSE)

od1$COORDINATEUNCERTAINTYINMETERS <- signif(od1$COORDINATEUNCERTAINTYINMETERS, 2)

od1$PARTS[!od1$PARTS=="media"] <- "whole organism"

## Rounding coordinates to 5 decimal places.
od1$DEC_LAT <- round(od1$DEC_LAT, 5)
od1$DEC_LONG <- round(od1$DEC_LONG, 5)

## Saving here.
write.csv(od1, "2020-01-10-1122_analysis_dataset.csv", row.names=FALSE)

## Now for the community dataset.
cd1 <- read.csv("oo_365062.csv", stringsAsFactors=FALSE)

cd1$frequency <- round(cd1$frequency, 2)

write.csv(cd1, "2020-01-10-1154_community_dataset.csv", row.names=FALSE)
```

I also worked on revising the text of the Slikok project manuscript
after receiving many comments on it from one reviewer.

## Monday, January 13

To do:

  - Worm information to ADF\&G.
  - ~~Finish revising and resubmit Slikok manuscript.~~

I finished revising and resubmitted the Slikok manuscript for technical
review.

I worked on putting together information for the ADF\&G comments on a
proposed regulation on *Lumbricus* species worms.

I generated maps of *Lumbricus* species distribution data using
SimpleMappr (Shorthouse [2010](#ref-shorthouse_simplemappr_2010)) and
data from GBIF (GBIF.org
[2020](#ref-gbiforg_gbif_2020-1)[a](#ref-gbiforg_gbif_2020-1),
[2020](#ref-gbiforg_gbif_2020)[b](#ref-gbiforg_gbif_2020)).

![Map of *Lumbricus terrestris* distribution in
Alaska.](2020-01-13-1013_Lumbricus_terrestris_in_Alaska.jpg)  
Map of *Lumbricus terrestris* distribution in Alaska. Locations were
obtained from GBIF (GBIF.org
[2020](#ref-gbiforg_gbif_2020-1)[a](#ref-gbiforg_gbif_2020-1)).

![Map of *Lumbricus rubellus* distribution in
Alaska.](2020-01-13-1014_Lumbricus_rubellus_in_Alaska.jpg)  
Map of *Lumbricus rubellus* distribution in Alaska.

I learned that *Lumbricus castaneus* (Savigny, 1826) is now *Enterion
castaneum Savigny*, 1826 according to GBIF (GBIF Secretariat
[2019](#ref-gbif_secretariat_enterion_2019)).

## Tuesday, January 14

To do:

  - ~~Worm information to ADF\&G.~~
  - ~~Submit Slikok manuscript.~~
  - Review this week’s *Refuge Notebook* article.

The Slikok manuscript made it through the technical review. I submitted
it to the journal.

Tyler sent me more coordinates of *Lumbricus* locations in Alaska, from
which I made updated maps.

I worked on putting together information for the ADF\&G comments on a
proposed regulation on *Lumbricus* species worms, getting a draft sent
at the end of the day.

## Wednesday, January 15

To do:

  - ~~Review and submit this week’s *Refuge Notebook* article.~~
  - ~~Review Sandpiper Lake MRDG.~~

I received some more earthworm references through ILL. I also was asked
for more information on native worms. I revised my comments to ADF\&G
and sent these out.

I edited and submitted this week’s *Refuge Notebook* article.

I reviewed the Sandpiper Lake elodea treatment Minimum Requirements
Decision Guide Workbook.

## Thursday, January 16

I picked up some historical documents from the fisheries office, scanned
them, and filed them. There is a record of elodea from Vogel Lake from
1964. I think this might have been a misidentification of *Hippuris*,
but it is hard to know. A thorough survey was done on Sandpiper Lake,
but no elodea was documented there. Later I returned these documents.

## Friday, January 17

We had a biology staff meeting today, our first since John retired.

Tasks for me:

  - Dawn and I need to plan for post-fire burn severity and maybe
    vegetation sampling within the Swan Lake Fire

  - I need to look into elodea genetics to determine the origin of the
    Sandpiper Lake elodea infestation.

  - In the short term I need to get the elodea funding application done
    and write the *Refuge Notebook* article for next week.

Dawn and I determined that some of the most interesting questions with
the Swan Lake fire have to do with burning of alpine tundra and
reburning of some locations by multiple fires.

I mapped LTEMP sites within the fire perimeter, looking for alpine
tundra sites for which we have pre-fire vegetation data.

  - 3161 - tundra north of Mysery Creek. Kenai NWR did veg work.

  - 3175 - tundra north side of Mystery Hills. Kenai NWR did veg work.

  - 3176 - no photo. Was it ever surveyed? It was sampled by FIA for
    veg. This looks like subalpine hemlock forest. Plants documented
    here by FIA: *Pleurozium schreberi*, *Cornus canadensis* *Empetrum
    nigrum*, *Gymnocarpium dryopteris*, *Lycopodium annotinum*,
    *Lycopodium clavatum*, *Menziesia ferruginea*, *Rubus pedatus*,
    *Sorbus sitchensis*, *Trientalis europaea*, *Tsuga mertensiana*,
    *Vaccinium caespitosum*, and *Vaccinium vitis-idaea*. I checked on
    Google Earth. This was a hemlock stand on the north slope north of
    Fuller Lake.

There were really only those two tundra sites within the fire. How about
the Paez sites (Paez [1991](#ref-paez_alpine_1991))?

I entered the coordinates from the Mystery Hills sites:

| plot | utm\_e |  utm\_n |
| ---: | -----: | ------: |
|   62 | 655073 | 6713896 |
|   63 | 655073 | 6714016 |
|   64 | 655039 | 6713748 |
|   79 | 654989 | 6713820 |
|   80 | 655089 | 6713779 |
|   81 | 655003 | 6714155 |
|  102 | 656300 | 6713517 |
|  103 | 656290 | 6713599 |
|  138 | 656294 | 6713683 |
|  139 | 656303 | 6713772 |
|  140 | 656033 | 6713516 |
|  141 | 656108 | 6713513 |
|  142 | 655901 | 6713514 |

I convereted these to lat/lon.

``` r
wd <- "D:/projects/Fire/2019_Swan_Lake_Fire/work_space/2020-01-17_plot_coordinates"
setwd(wd)

library(maptools)

wgs84 <- "+proj=longlat +ellps=WGS84 +datum=WGS84 +no_defs"
utmz5n <- "+proj=utm +zone=5 +ellps=WGS84 +datum=WGS84 +units=m +no_defs"

data1 <- read.csv("2020-01-17-1447_Paez_Mystery_Hills_locaclities.csv")
coordinates(data1) <- c("utm_e", "utm_n")
proj4string(data1) <- CRS(utmz5n)
data2 <- spTransform(data1, CRS(wgs84))

write.csv(data2, "2020-01-17-1506_Paez_Mystery_Hills_locaclities_WGS84.csv", row.names=FALSE)
```

| plot | lat         | lon           |
| ---: | :---------- | :------------ |
|   62 | 60.53113795 | \-150.1737958 |
|   63 | 60.53221407 | \-150.1737019 |
|   64 | 60.52982385 | \-150.1745303 |
|   79 | 60.53048878 | \-150.1753839 |
|   80 | 60.53008257 | \-150.1735961 |
|   81 | 60.53348754 | \-150.1748672 |
|  102 | 60.52726451 | \-150.1517647 |
|  103 | 60.52800372 | \-150.1518821 |
|  138 | 60.52875543 | \-150.1517431 |
|  139 | 60.52955004 | \-150.1515091 |
|  140 | 60.52735916 | \-150.156624  |
|  141 | 60.52730317 | \-150.1552616 |
|  142 | 60.52739238 | \-150.1590275 |

![Map of the Swan Lake Fire and Carlos Paez’s 1988-1989 vegetation plots
in the Mystery Hills.](2020-01-17-1515_Swan_Lake_Fire_Paez.jpg)  
Map of the Swan Lake Fire and Carlos Paez’s 1988-1989 vegetation plots
in the Mystery Hills.

## Monday, January 20

I started researching for a *Refuge Notebook* article on wildfire in
tundra and the Swan Lake Fire.

## Tuesday, January 21

To do:

  - *Refuge Notebook* article
  - Sandpiper Elodea RFP
  - earthworm correspondence
  - ~~blackfish permit reporting~~

I summarized the work done on blackfish in 2019 and got the required
project summary and data sent in.

Blackfish collected in 2019:

| lat       | lon          | date       | age      | number |
| :-------- | :----------- | :--------- | :------- | -----: |
| 60.56993  | \-151.19294  | 2019-06-27 | juvenile |      1 |
| 60.559862 | \-151.187067 | 2019-08-22 | juvenile |      4 |
| 60.559265 | \-151.187372 | 2019-08-23 | juvenile |      1 |
| 60.568077 | \-151.19013  | 2019-08-23 | adult    |      5 |

For the *Refuge Notebook* article on fire in tundra I checked Langille’s
report (Langille [1904](#ref-langille_proposed_1904)). Although he
talked about fire in Canyon Creek and other places, no mention was made
of fire making it up into the alpine.

I found the mention of fire in tundra in the 2014 Funny River Fire
(Morton [2014](#ref-morton_ghosts_2014)).

The 1991 Pothole Lake Fire burned into the alpine to 1012 m judging from
the fire polygon and digital elevation model. I found the KNWR documents
from the Pothole Lake Fire. This fired did burn into the alpine to 3000
ft. in multiple places.

LTEMP plot 3220 was in the Pothole Lake fire. So was plot 3204. Plot
3204 was burned forest in 2004. Plot 3220 was lush alder,
*Calamagrostis*, and cow parsnip in 2004. Neither was in tundra.

The Funny River Fire made it to about 2100 ft.

The 1950 Big Indian Creek fire burned to about 3000 ft. in the alpine
based on the fire history GIS layer.

Later at home I continued work on this *Refuge Notebook* article.

## Wednesday, January 22

To do:

  - ~~*Refuge Notebook* article.~~
  - Sandpiper Elodea RFP.
  - earthworm correspondence.
  - Take A-110 class.

I looked up the 1950 Big Indian Creek fire in the 1950 annual narrative.
There the fire was reported to be 3-5 acres, much smaller than the
polygon in the GIS layer.

I finished a draft of my *Refuge Notebook* article this morning and got
it sent in to the *Clarion*.

I resumed work on the elodea RFP.

I attended Todd’s brown bag lunch talk on hummingbird banding.

I spent the rest of the day on the Sandpiper Lake elodea RFP.

## Thursday, January 23

To do:

  - ~~Get draft Sandpiper Elodea RFP to Steve.~~
  - earthworm correspondence.
  - Take A-110 class.

I completed a draft Sandpiper Lake elodea RFP.

I found photos of Elizabeth Bella, Donna Handley, and Nate Olson
surveying Vogel Lake using throw rakes to look for *Elodea* on August
14, 2013. None was found.

There has been some population genetics work done on *Elodea* (Kadono et
al. [1997](#ref-kadono_genetic_1997); Gross et al.
[2003](#ref-gross_unterscheidung_2003); Huotari et al.
[2011](#ref-huotari_population_2011); Huotari and Korpelainen
[2013](#ref-huotari_comparative_2013)).

I submitted to Alaska DNR a Permitting Quarantined Species
Transport/Import Request for collecting Elodea specimens from Sandpiper
Lake.

Lisa Saperstein sent me a copy of Lipkin
([2007](#ref-lipkin_investigation_2007)) relating to fire in alpine
tundra.

## Friday, January 24

To do:

  - ~~Help with Sandpiper Lake *Elodea* RFP, if needed.~~
  - Start Sandpiper Lake *Elodea* EA.
  - ~~earthworm correspondence.~~
  - Take A-110 class.
  - *Refuge Notebook* catch-up.

I did a little bit of checklisting work on
[KenaiNWRspecies](https://github.com/mlbowser/KenaiNWRspecies).

I attended the a meeting at the ADF\&G office on the Miller Creek
drainage pike earadication project. Rob Massengill gave me some live
elodea that he had collected in Sandpiper Lake and kept alive in a
container in his office. I learned that Sandpiper Lake is a popular lake
for touch-and-go practice for aircraft from Anchorage.

We submitted the Sandpiper Lake *Elodea* RFP.

Todd and I met with Kristi about Swan Lake Fire suppression repair work
beginning next week. I will be going on on Monday and Wednesday.

I entered data for the *Elodea* specimen from Sandpiper Lake in Arctos
([KNWR:Herb:12443](http://arctos.database.museum/guid/KNWR:Herb:12443)).
I keyed this using the key of Bowmer
([1995](#ref-bowmer_identification_1995)).

![*Elodea* from Sandpiper Lake, specimen
KNWR:Herb:12443.](2020-01-24-1546_Elodea_from_Sandpiper.jpg)  
*Elodea* from Sandpiper Lake, specimen
[KNWR:Herb:12443](http://arctos.database.museum/guid/KNWR:Herb:12443).

## Monday, January 27

To do:

  - ~~Swan Lake Fire Suppression Repair scouting.~~

Kyle McNally and I met at the Division of Forestry at 10:00. From there
we drove out Skilak Lake Road to the Marsh Lake area dozer line. We
walked this line for some distance, sizing up the repair work to be
done. Woody debris was stacked on the west side of the line and it
looked like it should work well to pull it back over the line with the
current snow conditions. We walked only a portion of the line.

There were almost no birches this area. I saw no birch seeds in the
snow.

![Marsh Lake area dozer
line.](2020-01-27-1154_Marsh_Lake_area_dozer_line.jpg)  
Marsh Lake area dozer line. Woody material is stacked in a windrow on
the left (west) side of the line. I have no coordinates for this image
because the camera’s GPS had not acquired a position.

We also hiked the Kenai River to Skilak Road line. The seepage area that
had been a difficulty before is still flowing, but Kyle and I thought
that equipment should be able to get across this without much damage to
the ground because the ground around the seeps appears to be frozen
solidly.

![Seepage area near Kenai River.](2020-01-27-1310_seepage_area.jpg)  
Seepage area near Kenai River at 60.459053 °N, 150.178570 °W.

Repair work does need to be completed beyond the seepage area.

![Dozer line in need of repair near Kenai
River.](2020-01-27-1323_Kenai_river_dozer_line.jpg)  
Dozer line in need of repair near Kenai River at 60.457028 °N,
150.175910 °W.

I enjoyed this day of stomping through the woods.

## Tuesday, January 28

To do:

  - ~~File pictures, etc. from yesterday.~~
  - Sandpiper Lake *Elodea* EA.
  - BDJ Slikok manuscript revision.
  - ~~*Elodea* genetics correspondence.~~
  - ~~Website updates.~~
  - Edit this week’s *Refuge Notebook* article.

I learned this morning that our Slikok manuscript has been accepted into
*Biodiversity Data Journal*. Revisions will need to be completed within
one week.

I made some website updates requested by Mark.

I received ITS sequences from Alaska and GenBank *Elodea* samples. I
quickly submitted these to NGPhylogeny.fr just to see what the phylogeny
would look like. I used the “NGPhylogeny Analyse - FastME/OneClick”
analysis (Criscuolo and Gribaldo [2010](#ref-Criscuolo_2010); Desper and
Gascuel [2002](#ref-Desper_2002); Junier and Zdobnov
[2010](#ref-Junier_2010); Katoh and Standley [2013](#ref-Katoh_2013);
Lefort et al. [2015](#ref-Lefort_2015); Lemoine et al.
[2018](#ref-Lemoine_2018)). This analysis is (temporarily?) available at
<https://ngphylogeny.fr/workspace/history/deb556bc487a96e5>.

![Quick phylogram of *Elodea*
sequences.](2020-01-28-1252_Elodea_phylo.png)  
Quick phylogram of *Elodea* sequences made using NGPhylogeny.fr and
visualized using iTOL (Letunic and Bork
[2019](#ref-Letunic_et_al_2019)).

I worked on editing this week’s *Refuge Notebook* article.

To avoid going crazy from too much looking at a computer, I examined
Anisopodidae specimen
[KNWR:Ento:8334](http://arctos.database.museum/guid/KNWR:Ento:8334).
Keying using McAlpine et al. ([1983](#ref-mcalpine_manual_1983)), p. 311
1 → 2 → *Sylvicola*. Keying using Pratt and Pratt
([1980](#ref-pratt_notes_1980)), p. 87 → 1 → 2 *Sylvicola fuscatus*
(Fabricius, 1775). This is a female.

## Wednesday, January 29

To do:

  - ~~Swan Lake suppression repair meeting.~~
  - Sandpiper Lake *Elodea* EA.
  - BDJ Slikok manuscript revision.
  - ~~Finish editing and submit this week’s *Refuge Notebook* article.~~

I worked on revising this week’s *Refuge Notebook* article, getting it
submitted in the afternoon.

Todd, Kristi, and I drove to the Division of Forestry, where we met with
DOF folks to plan the Swan Lake Fire suppression repair work for the
next few weeks.

I worked on revising the BDJ Slikok manuscript.

## Thursday, January 30

To do:

  - BDJ Slikok manuscript revision.
  - Sandpiper Lake *Elodea* EA.

I worked on the BDJ Slikok manuscript revision. I created a flow chart
as requested by one of the reviewers.

``` r
## Trying DiagrammeR package for making graphs for the Slikok manuscript.
wd <- "D:/projects/Slikok_watershed/data/work_space/2020-01-30_flow_charts"
setwd(wd)

library(DiagrammeR)

mermaid("
graph LR

subgraph Legend
AA(field sampling):::field-->BB(laboratory steps):::knwr
CC(commercial services):::service-->DD(published data):::pub
end

A(vascular plant, bryophyte,<br/>and lichen observations):::field -->C(Arctos<br/>occurrence records):::pub

D(vascular plant<br/>specimens):::field -->E(morphological<br/>identification):::knwr
E-->C

F(bryophyte and lichen<br/>specimens):::field -->G(morphological identification<br/>by Enlichened Consulting):::service
G-->C

H(bird call observations):::field -->C

C-->J(GBIF<br/>occurrence records):::pub

K(worms collected by<br/>hot mustard extraction):::field -- adult Lumbricidae -->M(morphological<br/>identification):::knwr
M-->C

K-- Enchytraeidae and<br/>immature Lumbricidae -->O(COI sequencing<br/>by LifeScanner):::service
O-->C
O-->P(BOLD<br/>occurrence records):::pub
P-->J

Q(sweep net samples of<br/>terrestrial arthropods):::field -->S(High-Throughput Sequencing<br/>by RTL Genomics):::service
S-- raw sequence data -->T(metagenomic analysis):::knwr
T-->C
%% S-- raw sequence data -->U(GenBank<br/>Sequence Read Archive):::pub
%% S-- raw sequence data -->V(Zenodo):::pub

classDef field fill:#9DE093;
classDef knwr fill:#ffffff;
classDef service fill:#FFDB00;
classDef pub fill:#FFD3F8;

style Legend fill:#ffffff,stroke:#000000

linkStyle 0 stroke:#fff,stroke-width:0px;
linkStyle 1 stroke:#fff,stroke-width:0px;

")
```

![Slikok project flowchart for BDJ
manuscript](2020-01-30-1117_flowchart.png)  
Slikok project flowchart for BDJ manuscript.

I pulled blackfish specimens out of the freezer so that Apphia and I can
dissect out stomach contents.

In the afternoon, Apphia and I dissected stomach contents from the
larger blackfish that I had collected on August 23, 2019 (field id:
2019-08-23\_MLB02) (Bowser [2019](#ref-bowser_work_2019)). Apphia
dissected out the digestive tracts, determined the sex of each fish, and
helped with the photographs. I transfered the contents of the stomachs
and intestines into vials of propylene glycol by squeezing the contents
out using forceps.

| temporary ID      | total length (mm) | sex    |
| :---------------- | ----------------: | :----- |
| 2020-01-30\_MLB01 |                69 | male   |
| 2020-01-30\_MLB02 |                95 | female |
| 2020-01-30\_MLB03 |                96 | female |
| 2020-01-30\_MLB04 |                75 | male   |
| 2020-01-30\_MLB05 |                88 | female |

## Friday, January 31

To do:

  - ~~BDJ Slikok manuscript revision.~~
  - Enter data from blackfish specimens.
  - Sandpiper Lake *Elodea* EA.

I revised the BDJ manuscript and submitted a draft to the journal.

At 10:30 my supervisors and I met to discuss goals and plans for 2020
and beyond.

Priority projects or me:

  - Perform an occupancy analysis of the Slikok project invertebrate
    data and publish this.
  - Sandpiper Lake *Elodea* eradication.
  - Swan Lake Fire burn severity/vegetation plots.

One of my supervisors wants to see an assessment of the 2004–2006 Long
Term Ecological Monitoring Program, wanting to know what we learned from
it and when—if ever—such an effort should be repeated. I expressed that
we are at least planning on publishing datasets from this project.

My supervisors did think it would be appropriate to collaborate with
ADF\&G to sample invertebrates in the Miller Creek drainage ahead of
expected pesticide applications.

Later at home I added records from State of Alaska Department of
Environmental Conservation ([2005](#ref-state_of_alaska_dec_2005)) to
the Kenai National Wildlife Refuge’s checklist. There were only two
species not already on the Refuge’s checklist: *Caenis youngi* and
*Glyphopsyche irrorata*.

# February

## Monday, February 3

To do:

  - ~~Ship out *Elodea* specimens.~~
  - ~~Ship out blackfish stomach contents.~~
  - Sandpiper Lake *Elodea* EA.

I packaged live Sandpiper Lake *Elodea* specimens to go out this morning
(FedEx tracking \#: 813677694043).

I packaged the five samples of blackfish stomach contents from January
30 and packaged these to go out to RTL Genomics this morning (FedEx
tracking \#: 813677694032).

I worked a little on the the [Kenai National Wildlife Refuge
checklist](https://github.com/mlbowser/KenaiNWRspecies), moving three
species (*Corticaria ferruginea*, *Enicmus mimus*, and *Xylita
laevigata*) to FWSpecies and requested the addition of *Corticarina
minuta*, *Cortinarius lucorum*, and *Stephostethus armatulus* to
FWSpecies’ taxonomy.

I found a data sheet from a 2013 vegetation survey of Vogel Lake. I
intend to enter these data into Arctos, but first I wanted to search for
what had been documented previously in the watershed, so I converted the
HUC12 190203021903 “Miller Creek-Frontal Cook Inlet” watershed. I
simplified this using ArcMap’s Simplify Polygon tool, with a
simplification tolerance of 500 m. I converted this to WKT so I could
run a GBIF search (GBIF.org
[2020](#ref-gbiforg_gbif_2020-2)[c](#ref-gbiforg_gbif_2020-2)).

I finished sorting Berlese sample
[KNWR:Ento:11362](http://arctos.database.museum/guid/KNWR:Ento:11362)
that I had started sorting on January 6.

Contents:

| identification | count |
| :------------- | ----: |
| Lepidoptera    |     1 |
| Collembola     |    23 |
| Coleoptera     |     1 |
| Arachnida      |   260 |
| Geophilomorpha |     2 |
| Lithobiomorpha |     1 |
| Myrmicinae     |     1 |
| Nematoda       |     1 |
| *Cixius*       |     1 |

There still were some minute mites in this sample. I had spent about 3–4
hours on this sample. I may need to use some kind of time limit, maybe
2–3 hours per sample.

I uploaded the 2013 data sheet from Vogel Lake onto Arctos (Bella
[2013](#ref-bella_Vogel_2013)), entered observation data from this data
sheet, and made an [Arctos
project](http://arctos.database.museum/project/10003326) for these data.

## Tuesday, February 4

To do:

  - Sandpiper Lake *Elodea* EA.

I worked a little on the the [Kenai National Wildlife Refuge
checklist](https://github.com/mlbowser/KenaiNWRspecies), moving three
species (*Agoliinus congregatus*, *Dendroides ephemeroides*, and
*Rhizophagus dimidiatus*) to FWSpecies.

The Vogel Lake plant records from 2013 are now records
[KNWRObs:Herb:831](http://arctos.database.museum/guid/KNWRObs:Herb:831)–[KNWRObs:Herb:867](http://arctos.database.museum/guid/KNWRObs:Herb:867).
I updated some of the identifications.

I got started on the Sandpiper Lake *Elodea* Environmental Assessment. I
requested Marchand ([1985](#ref-marchand_oxygen_1985)) through
[ARLIS](https://www.arlis.org/).

## Wednesday, February 5

To do:

  - ~~Edit and submit today’s *Refuge Notebook*.~~
  - Sandpiper Lake *Elodea* EA.

I worked some on the *Elodea* Environmental Assessment.

I attended the monthly Alaska Invasive Species Partnership
teleconference at 09:00 to talk about earthworms.

I attended a meeting about preparing for morel hunters in the Swan Lake
Fire this spring.

I edited and submitted this week’s *Refuge Notebook* article.

Inspired by flipping through Hedlin ([1974](#ref-hedlin_cone_1974)), I
took a short walk on the Keen Eye Trail to collect spruce cones. I found
some under a singe tree at 60.46457 °N, 151.07216 °W.

In one cone there was an orange (Diptera?) larva in the stem in a white,
silken cocoon. This I photographed (frames 659–665) and collected
(LifeScanner vial BOLD-268).

![Larva in pith of spruce cone (vial
BOLD-268).](2020-02-05_larva_BOLD-268.jpg)  
Larva in pith of spruce cone (vial BOLD-268).

In another cone I found smaller, silken cocoons containing orange
immatures (Diptera?). The first I photographed (frames 666–670) and
collected into vial BOLD-8W0.

![Larva in scales of spruce cone (vial
BOLD-8W0).](2020-02-05_larva_BOLD-8W0.jpg)  
Larva in scales of spruce cone (vial BOLD-8W0).

In the same cone I photographed another cocoon (frames 671–676) and
collected it into vial BOLD-QH4.

Also from the same cone I photographed another cocoon (frames 677–681)
and collected it into vial BOLD-BS8.

In most of the cones I examined I found no insects.

## Thursday, February 6

To do:

  - Sandpiper Lake *Elodea* SEA.
  - ~~Help Mark with ROV testing.~~

I worked on the Sandpiper Lake *Elodea* SEA.

In the afternoon, Mark and I drove to Arc Lake, where we cut a hole in
the ice and tested the QYSEA ROV. The ROV worked well, but the lake was
quite dark under the ice so that we could not see much unless the ROV
was on the bottom or just under the surface ice.

![Deploying ROV in Arc Lake.](2020-02-06_ROV.jpg)  
Deploying ROV in Arc Lake.

## Friday, February 7

To do:

  - ~~Sandpiper Lake *Elodea* SEA.~~

I worked on the Sandpiper Lake *Elodea* SEA, getting a draft sent out in
the afternoon\!

It snowed heavily this morning, but it was warm. Todd and I saw a number
of live cluster flies (*Pollenia vagabunda*) on the new snow and beside
the visitor center and headquarters buildings.

At the end of the day I examined specimen
[KNWR:Ento:7705](http://arctos.database.museum/guid/KNWR:Ento:7705), a
colletid bee. The forewing has two submarginal cells, so it should be
*Hylaeus*. The only other genus reported from Alaska is *Colletes*,
which has three submarginal cells (Packer et al.
[2007](#ref-packer_bee_genera_2007)).

## Saturday, February 8

I received information about kokanee in Sandpiper Lake, so I
incorporated this into the draft SEA.

I worked on the [Kenai National Wildlife Refuge’s
checklist](https://github.com/mlbowser/KenaiNWRspecies), editing the
script to make a satifactory DwC-A file.

## Monday, February 10

I worked on the [Kenai National Wildlife Refuge’s
checklist](https://github.com/mlbowser/KenaiNWRspecies), adding a
minimal checklist export.

I sorted Berlese sample
[KNWR:Ento:11364](http://arctos.database.museum/guid/KNWR:Ento:11364). I
cut off sorting at a 2 hour limit. Most material had been sorted. A
small number of Collembola and mites remained and were discarded.

| identification | count |
| :------------- | ----: |
| Myrmicinae     |     9 |
| Lithobiomorpha |     1 |
| Coleoptera     |     2 |
| Araneae        |    12 |
| Collembola     |   110 |
| Insecta        |     2 |
| Arachnida      |   204 |
| Chilopoda      |     2 |
| Hemiptera      |     1 |

I learned today that we had purchased herbicide last fall. I inventoried
our aquatic herbicides. We have 41 pails of SonarOne, (20 lbs. each), a
total of 820 lbs. We have 20 boxes of Littora (5 gal. each), a total of
100 gal. of Littora.

I revised the *Elodea* RFP based on these numbers.

## Tuesday, February 11

To do:

  - ~~*Elodea* meeting.~~
  - ~~Review/edit *Refuge Notebook* article.~~

I sorted Berlese sample
[KNWR:Ento:11365](http://arctos.database.museum/guid/KNWR:Ento:11365),
again cutting off sorting at 2 hr.

| identification | count |
| :------------- | ----: |
| Staphylinidae  |     1 |
| Ichneumonoidea |     1 |
| Insecta        |    11 |
| Arachnida      |   288 |
| Araneae        |     5 |
| Hymenoptera    |     2 |
| Collembola     |    45 |
| Hemiptera      |     9 |
| Thysanoptera   |     1 |

In the middle of the day I met with Katherine, Maura, Colin, and Steve
about Sandpiper *Elodea* permitting, logistics, etc.

I edited this week’s *Refuge Notebook* article.

I started going through the lists of plants from Sandpiper Lake
documented by Jakubas and Firman ([1984](#ref-jakubas_sandpiper_1984))
to add to the Sandpiper Lake SEA.

Species documented in and around Sandpiper Lake by Jakubas and Firman
([1984](#ref-jakubas_sandpiper_1984)) with the species codes that they
used on their map:

| identification                             | code |
| :----------------------------------------- | :--- |
| *Alnus* Mill.                              | A    |
| *Betula neoalaskana* Sarg.                 | MB   |
| *Comarum palustre* L.                      | MFF  |
| *Eleocharis* R.Br.                         | SR   |
| *Equisetum fluviatile* L.                  | HTF  |
| *Iris setosa* Pall. ex Link                | WF   |
| *Myrica gale* L.                           | SG   |
| *Myriophyllum* L.                          | MR   |
| *Nuphar polysepalum* Engelm.               | YPL  |
| *Nymphaea tetragona* Georgi                | DWL  |
| *Picea glauca* (Moench) Voss               | MS   |
| *Picea mariana* Britton, Sterns & Poggenb. | BS   |
| *Potamogeton alpinus* Balb.                | PA   |
| *Potamogeton berchtoldii* Fieb.            | PBr  |
| *Potamogeton foliosus* Raf.                | Pfo  |
| *Potamogeton gramineus* L.                 | PGr  |
| *Potamogeton natans* L.                    | PN   |
| *Potamogeton robbinsii* Oakes              | PRb  |
| *Potamogeton zosteriformis* Fernald        | PZ   |
| *Ranunculus aquatilis* L.                  | WWC  |
| *Schoenoplectus* (Rchb.) Palla             | Blr  |
| *Sparganium angustifolium* Michx.          | BR   |
| *Sparganium minimum* (L.) Fr.              | BRm  |

## Wednesday, February 12

To do:

  - ~~Send off *Refuge Notebook* article.~~
  - Board of Fish meeting arrangements.

Identifications

I sorted Berlese sample
[KNWR:Ento:11366](http://arctos.database.museum/guid/KNWR:Ento:11366).
This was a sparser sample that took only about one hour to process.

| identification | count |
| :------------- | ----: |
| Arachnida      |   166 |
| Araneae        |     6 |
| Collembola     |    22 |

I worked a little on the Integrated Pest Management Plan for Eradicating
Elodea from the Kenai Peninsula, Working Draft 2020.

New molecular identifications of some KNWR specimens were made on BOLD.
I checked these using BOLD’s ID engine.

| GUID                                                                | identification            |
| :------------------------------------------------------------------ | :------------------------ |
| [KNWR:Ento:4382](http://arctos.database.museum/guid/KNWR:Ento:4382) | *Dorylomorpha spinosa*    |
| [KNWR:Ento:5264](http://arctos.database.museum/guid/KNWR:Ento:5264) | *Tomosvaryella sylvatica* |
| [KNWR:Ento:5442](http://arctos.database.museum/guid/KNWR:Ento:5442) | *Dorylomorpha albitarsis* |
| [KNWR:Ento:6768](http://arctos.database.museum/guid/KNWR:Ento:6768) | *Pipunculus hertzogi*     |
| [KNWR:Ento:6769](http://arctos.database.museum/guid/KNWR:Ento:6769) | *Eudorylas slovacus*      |

The identification of specimen
[KNWR:Ento:5264](http://arctos.database.museum/guid/KNWR:Ento:5264) was
changed from *Tomosvaryella kuthyi* to *Tomosvaryella sylvatica*. I
searched on Arctos for additional records. Four records from the Slikok
project were identified as *Tomosvaryella kuthyi*. These were all
identifications from ASV SlikokOtu119, which I ran through BOLD’s ID
Engine. This is *Tomosvaryella sylvatica*. Affected records:
[UAMObs:Ento:239975](http://arctos.database.museum/guid/UAMObs:Ento:239975),
[UAMObs:Ento:240012](http://arctos.database.museum/guid/UAMObs:Ento:240012),
[UAMObs:Ento:240458](http://arctos.database.museum/guid/UAMObs:Ento:240458),
and
[UAMObs:Ento:240561](http://arctos.database.museum/guid/UAMObs:Ento:240561).

Now for *Eudorylas* records. BIN
[BOLD:ACZ4721](http://boldsystems.org/index.php/Public_BarcodeCluster?clusteruri=BOLD:ACZ4721)
is now mostly identified as *Eudorylas slovacus*.

SlikokOtu114 is *Eudorylas slovacus*. This applies to record
[UAMObs:Ento:240642](http://arctos.database.museum/guid/UAMObs:Ento:240642).

SlikokOtu130 is also *Eudorylas slovacus*. This applies to records
[UAMObs:Ento:240440](http://arctos.database.museum/guid/UAMObs:Ento:240440),
[UAMObs:Ento:240007](http://arctos.database.museum/guid/UAMObs:Ento:240007),
[UAMObs:Ento:239943](http://arctos.database.museum/guid/UAMObs:Ento:239943).

SlikokOtu775 is also *Eudorylas slovacus*. This applies to record
[UAMObs:Ento:239323](http://arctos.database.museum/guid/UAMObs:Ento:239323).

SlikokOtu1573 is also *Eudorylas slovacus*. This applies to record
[UAMObs:Ento:238974](http://arctos.database.museum/guid/UAMObs:Ento:238974).

SlikokOtu84 is also *Eudorylas slovacus*, affecting record
[UAMObs:Ento:238742](http://arctos.database.museum/guid/UAMObs:Ento:238742).

SlikokOtu62 is *Eudorylas sabroskyi*. This affects records
[UAMObs:Ento:239100](http://arctos.database.museum/guid/UAMObs:Ento:239100),
[UAMObs:Ento:239116](http://arctos.database.museum/guid/UAMObs:Ento:239116),
and
[UAMObs:Ento:239424](http://arctos.database.museum/guid/UAMObs:Ento:239424).

I added *Tomosvaryella sylvatica* and *Pipunculus hertzogi* to the Kenai
National Wildlife Refuge’s checklist in FWSpecies; the other names are
not currently in FWSpecies’ taxonomy.

I did get today’s *Refuge Notebook* article sent to the
[*Clarion*](https://www.peninsulaclarion.com/).

## Thursday, February 13

To do:

  - Board of Fish meeting arrangements.
  - ~~Call with Tracy~~
  - Botany 2020
  - ~~Draft Sandpiper SEA to DNR~~

I received LifeScanner identifications (but not sequences) for a number
of submitted vials today.

I sorted Berlese sample
[KNWR:Ento:11367](http://arctos.database.museum/guid/KNWR:Ento:11367).
This was the sparsest sample yet and took less than one hour to process.

| identification | count |
| :------------- | ----: |
| Araneae        |     3 |
| Arachnida      |    93 |
| Insecta        |     1 |
| Collembola     |     8 |
| Nematoda       |     1 |

I worked some on revising the Draft Sandpiper SEA, getting it sent out\!

## Friday, February 14

Allie and I drove out to the Marsh Lake dozer line to help with finding
the Marsh Lake Trail where the recent dozer line crosses it. We ended up
just discussing this the State Forestry folks and leaving them a GPS
loaded with the Marsh Lake Trail as a track. We decided that it would be
ok to rerout the Marsh Lake Trail for the short distance where the
original trail was hard to find, aligning it with the current dozer line
for that section.

## Monday, February 17

I received sequences from LifeScanner and from the 2019 black spruce
sweep net samples. I started downloading the sequence data from RTL
Genomics.

Two worm specimens collected by Adrian and Tyler from Pilgrim Hot
Springs are either *Aporrectodea longa* or *Aporrectodea trapezoides*.
Both specimens closely matched multiple sequences identified as both of
these species, so this was a little confusing.

I also received a notification on the Slikok project manuscript. It is
currently being edited.

## Tuesday, February 18

To do:

  - ~~Edit and submit this week’s *Refuge Notebook* article.~~
  - ~~Revise and submit Slikok article.~~

I stopped by the office for a short time. I had received a new package
of worms from Tyler and Kyungsoo.

I edited and sent out this week’s *Refuge Notebook* article.

I revised and re-submitted the Slikok manuscript.

## Wednesday, February 19

I sorted Berlese sample
[KNWR:Ento:11368](http://arctos.database.museum/guid/KNWR:Ento:11368).

Contents:

| identification | count |
| :------------- | ----: |
| Formicidae     |     1 |
| Araneae        |     1 |
| Arachnida      |   187 |
| Collembola     |     4 |
| Staphylindae   |     1 |
| Chilopoda      |     1 |

## Thursday, February 20

To do:

  - Sandpiper Lake snowmachine decision/logistics.
  - Pesticide applicator training.
  - Board of Fish meeting travel arrangements.

I sorted Berlese sample
[KNWR:Ento:11369](http://arctos.database.museum/guid/KNWR:Ento:11369).

Contents:

| identification | count |
| :------------- | ----: |
| Arachnida      |   114 |
| Insecta        |     3 |
| Formicinae     |    11 |
| Animalia       |     1 |
| *Cixius*       |     1 |
| Sciaroidea     |     1 |
| Arthropoda     |     2 |
| Thysanoptera   |     1 |
| Collembola     |     1 |

In this sample was something pale and difficult to identify, labeled as
Animalia in the table above. It might have been a contracted tardigrade,
a flatworm, plant material, or something else. There were also a couple
of pale, elongate arthropods that might have been Collembola.

## Friday, February 21

To do:

  - ~~Board of fish comments?~~
  - ~~Sandpiper Lake snowmachine decision/logistics.~~
  - Board of Fish meeting travel arrangements.

I learned today that Juniki Yukawa and others published a paper
including Dominqiue Collet’s work on *Rabdophaga* willow rosette gall
makers (Sato et al. [2020](#ref-Sato_et_al_Rabdophaga_2020)). I am happy
to see this published.

I composed and submitted comments regarding the use of *Lumbricus*
earthworms as live bait for the upcoming Board of Fisheries meeting.

I sorted Berlese sample
[KNWR:Ento:11370](http://arctos.database.museum/guid/KNWR:Ento:11370).

Contents:

| identification | count |
| :------------- | ----: |
| Collembola     |    29 |
| Araneae        |     8 |
| Hemiptera      |    14 |
| Nematoda       |     2 |
| Arachnida      |   359 |
| Staphylinidae  |     1 |
| Insecta        |     8 |
| Oligochaeta    |     1 |

## Monday, February 24

To do:

  - Board of Fish meeting travel arrangements.
  - ~~Elodea SEA call at 11:00.~~

I revised the Slikok manuscript based on the copy editor’s edits and
comments.

I sorted Berlese sample
[KNWR:Ento:11371](http://arctos.database.museum/guid/KNWR:Ento:11371).

Contents:

| identification | count |
| :------------- | ----: |
| Coleoptera     |     1 |
| Collembola     |    14 |
| Arachnida      |   329 |
| Thysanoptera   |     7 |
| Insecta        |     2 |
| Sciaroidea     |     1 |

I participated in an *Elodea* teleconference regariding permitting, etc.

I sorted Berlese sample
[KNWR:Ento:11372](http://arctos.database.museum/guid/KNWR:Ento:11372).

Contents:

| identification | count |
| :------------- | ----: |
| Arachnida      |   375 |
| Collembola     |    10 |
| Nematoda       |     1 |
| Araneae        |     1 |
| Myrmicinae     |     1 |
| Lepidoptera    |     1 |

Attempting to key a tingid bug in Berlese sample
[KNWR:Ento:11370](http://arctos.database.museum/guid/KNWR:Ento:11370)
just by looking through the vial and using the key of Drake and Lattin
([1963](#ref-drake_american_1963)), p. 334. 1 → 4 → 5 → 6 → *Acalypta
nyctalis* Drake, 1928. This species is known from our area (Drake and
Lattin [1963](#ref-drake_american_1963); Froeschner
[1976](#ref-froeschner_zoogeographic_1976)). I made a new record for
this specimen
([UAMObs:Ento:244398](http://arctos.database.museum/guid/UAMObs:Ento:244398)).

## Tuesday, February 25

To do:

  - Board of Fish meeting travel arrangements.

I started making travel arrangements for the Board of Fisheries meeting.

I sorted Berlese sample
[KNWR:Ento:11373](http://arctos.database.museum/guid/KNWR:Ento:11373).

Contents:

| identification | count |
| :------------- | ----: |
| Arachnida      |    76 |
| Thysanoptera   |     1 |
| Aphididae      |     3 |
| Coleoptera     |     1 |
| Insecta        |     2 |
| Hemiptera      |     2 |
| Collembola     |     7 |
| Araneae        |     1 |

The [KNWR:Env](https://arctos.database.museum/knwr_env) collection was
created today to accomodate environmental samples beginning with soil
samples from the black spruce inventory project.

I formatted and posted the *Refuge Notebook* [article from
December 13, 2019](https://www.fws.gov/uploadedFiles/Region_7/NWRS/Zone_2/Kenai/Sections/What_We_Do/In_The_Community/Refuge_Notebooks/2019_Articles/Refuge_Notebook_v21_n44.pdf).

I edited this week’s *Refuge Notebook* article.

## Wednesday, February 26

To do:

  - ~~Submit *Refuge Notebook* article.~~
  - ~~Board of Fish meeting travel arrangements.~~
  - ~~Swan Lake Fire burn severity/vegetation survey planning.~~

I made final edits to this week’s Refuge Notebook\* article and got it
sent off.

Mark asked me to put together data/maps for planning work within the
2019 Swan Lake Fire, so I did.

``` r
nm <- read.csv("site_names.csv", stringsAsFactors=FALSE)
loc <- read.csv("2018-06-06-1309_points_wgs84.csv", stringsAsFactors=FALSE)
loc2 <- loc[loc$site_name %in% nm$site_name,]
write.csv(loc2, "2020-02-26-1122_Bowser_sites.csv", row.names=FALSE)
```

I started entering plant observation records from Jakubas and Firman
([1984](#ref-jakubas_sandpiper_1984)) (records
[KNWRObs:Herb:868](http://arctos.database.museum/guid/KNWRObs:Herb:868)–[KNWRObs:Herb:875](http://arctos.database.museum/guid/KNWRObs:Herb:875)).

## Thursday, February 27

The Slikok project manuscript was published today (Bowser et al.
[2020](#ref-bowser_et_al_2020)).

To do:

  - ~~Add Arctos specimen citations for Bowser et
    al. ([2020](#ref-bowser_et_al_2020)).~~

<!-- end list -->

``` r
## Preparing a file of specimen citations to upload to Arctos.
## Starting with Suppl. material 6: Raw occurrence data from the Slikok watershed biotic inventory (<https://doi.org/10.3897/BDJ.8.e50124.suppl6>).

setwd("D:/projects/Slikok_watershed/manuscript/2020-02-27_Arctos_citations")
d1 <- read.csv("oo_371432.csv", stringsAsFactors=FALSE)

dim(d1)
[1] 4764   28

levels(as.factor(d1$SCIENTIFIC_NAME))

## Starting citation upload data frame.
d2 <- d1[,c("GUID", "SCIENTIFIC_NAME")]
 
d2$publication_id <- 10009135
d2$made_date <- "2020-02-27"
d2$nature_of_id <- "unknown"
d2$type_status <- "voucher"
d2$USE_EXISTING_ACCEPTED_ID <- 0
d2$accepted_id_fg <- 0
d2$use_pub_authors <- 1

## The tough part is that I need to convert some of the identifications to the "A {string}" formula used by Arctos.

## First the names containing "sp."
slsp <- grepl(" sp. ", d2$SCIENTIFIC_NAME)
nsp <- d2$SCIENTIFIC_NAME[slsp]
nsp0 <- strsplit(nsp, " sp. ")
nsp1 <- sapply(nsp0, "[", 1)
nsp2 <- sapply(nsp0, "[", 2)
nsp3 <- paste0(nsp1, " {", nsp, "}")
## There were some double spaces in there.
nsp3 <- gsub("  ", " ", nsp3)
d2$SCIENTIFIC_NAME[slsp] <- nsp3

## Now I think I saw one cf. without a sp.
slcf <- grepl(" cf. ", d2$SCIENTIFIC_NAME)
## I want to exclude those I have already dealt with.
sum(slcf)
[1] 16
slcf[slsp] <- FALSE
sum(slcf)
[1] 9
levels(as.factor(d2$SCIENTIFIC_NAME[slcf]))
## That looked good.

ncf <- d2$SCIENTIFIC_NAME[slcf]
ncf0 <- strsplit(ncf, " cf. ")
ncf1 <- sapply(ncf0, "[", 1)
ncf2 <- sapply(ncf0, "[", 2)
ncf3 <- paste0(ncf1, " {", ncf, "}")
d2$SCIENTIFIC_NAME[slcf] <- ncf3

## Ok, making a test file to upload.
write.csv(d2[1,], file="slikok_citations_0001-0001.csv", row.names=FALSE)

## That worked. Trying some more.
write.csv(d2[2:10,], file="slikok_citations_0002-0010.csv", row.names=FALSE)

## That worked. Trying some more.
write.csv(d2[11:100,], file="slikok_citations_0011-0100.csv", row.names=FALSE)

## Those A {string} identifications were not handled exaclty as I had expected.
## Record 121, KNWR:Herb:12254, Peltigera conspersa nom. prov., should be a good test case.
write.csv(d2[101:150,], file="slikok_citations_0101-0150.csv", row.names=FALSE)
## That one did not work. Got error: taxon 1 not found.

## Fixing this.
slnp <- grepl("nom. prov.", d2$SCIENTIFIC_NAME)
d2$SCIENTIFIC_NAME[slnp]
## Just one name like this.
d2$SCIENTIFIC_NAME[slnp] <- "Peltigera {Peltigera conspersa nom. prov.}"

## Trying again.
write.csv(d2[101:150,], file="slikok_citations_0101-0150.csv", row.names=FALSE)
# That worked.

write.csv(d2[151:500,], file="slikok_citations_0151-0500.csv", row.names=FALSE)
## That did not load. Had problem with Simulium arcticum complex.

d2$SCIENTIFIC_NAME[d2$SCIENTIFIC_NAME=="Simulium arcticum complex"] <- "Simulium {Simulium arcticum complex}"

## Trying again.
write.csv(d2[151:500,], file="slikok_citations_0151-0500.csv", row.names=FALSE)
## That worked.

write.csv(d2[501:1000,], file="slikok_citations_0501-1000.csv", row.names=FALSE)

write.csv(d2[1001:2000,], file="slikok_citations_1001-2000.csv", row.names=FALSE)

write.csv(d2[2001:3000,], file="slikok_citations_2001-3000.csv", row.names=FALSE)
## That gave me an error: value too large for column "UAM"."CF_TEMP_CITATION"."SCIENTIFIC_NAME" (actual: 67, maximum: 60)

summary(nchar(d2$SCIENTIFIC_NAME)) 
   Min. 1st Qu.  Median    Mean 3rd Qu.    Max. 
    6.0    17.0    20.0    23.8    30.0    67.0 

## What was this?   
d2$SCIENTIFIC_NAME[nchar(d2$SCIENTIFIC_NAME)==67]
[1] "Pachynematus albipennis {Pachynematus albipennis sp. SlikokOtu1704}"

d2[nchar(d2$SCIENTIFIC_NAME)==67,]
## That is UAMObs:Ento:239052.

## Bumping that back to a genus resolution ID.
d2$SCIENTIFIC_NAME[nchar(d2$SCIENTIFIC_NAME)==67] <- "Pachynematus {Pachynematus albipennis sp. SlikokOtu1704}"

write.csv(d2[2001:3000,], file="slikok_citations_2001-3000.csv", row.names=FALSE)
## Got an error with the Simulium venustum complex.

d2$SCIENTIFIC_NAME[d2$SCIENTIFIC_NAME=="Simulium venustum complex"] <- "Simulium {Simulium venustum complex}"

write.csv(d2[2001:3000,], file="slikok_citations_2001-3000.csv", row.names=FALSE)

write.csv(d2[3001:4000,], file="slikok_citations_3001-4000.csv", row.names=FALSE)
## Did not like this one:   Arthonia edgewoodensis ined..

d2$SCIENTIFIC_NAME[d2$SCIENTIFIC_NAME=="Arthonia edgewoodensis ined."] <- "Arthonia {Arthonia edgewoodensis ined.}"

write.csv(d2[3001:4000,], file="slikok_citations_3001-4000.csv", row.names=FALSE)
## Did not like Lecanora allophana forma soralifera.

d2$SCIENTIFIC_NAME[d2$SCIENTIFIC_NAME=="Lecanora allophana forma soralifera"] <- "Lecanora allophana {Lecanora allophana forma soralifera}"

write.csv(d2[3001:4000,], file="slikok_citations_3001-4000.csv", row.names=FALSE)

write.csv(d2[4001:4764,], file="slikok_citations_4001-4764.csv", row.names=FALSE)
## Did not like Lasiomma nr. consobrinum.

d2$SCIENTIFIC_NAME[d2$SCIENTIFIC_NAME=="Lasiomma nr. consobrinum"] <- "Lasiomma {Lasiomma nr. consobrinum}"

write.csv(d2[4001:4764,], file="slikok_citations_4001-4764.csv", row.names=FALSE)

## All done.

## Saving the whole data frame.
write.csv(d2, file="slikok_citations.csv", row.names=FALSE)
```

I sorted Berlese sample
[KNWR:Ento:11374](http://arctos.database.museum/guid/KNWR:Ento:11374).

Contents:

| identification | count |
| :------------- | ----: |
| Arachnida      |   220 |
| Insecta        |     1 |
| Formicidae     |     1 |
| Araneae        |     1 |
| Collembola     |     7 |

I sorted Berlese sample
[KNWR:Ento:11375](http://arctos.database.museum/guid/KNWR:Ento:11375).

Contents:

| identification | count |
| :------------- | ----: |
| Araneae        |     5 |
| Collembola     |    11 |
| Arachnida      |    64 |

This was a sparser sample.

I entered more observation records from Jakubas and Firman
([1984](#ref-jakubas_sandpiper_1984)) (records
[KNWRObs:Herb:876](http://arctos.database.museum/guid/KNWRObs:Herb:876)–[KNWRObs:Herb:882](http://arctos.database.museum/guid/KNWRObs:Herb:882)).

I started reading through Alaska Region Data Stewardship Team
([2019](#ref-ak_region_data_stewardship_team_alaska_2020)[a](#ref-ak_region_data_stewardship_team_alaska_2020),
[2019](#ref-alaska_region_data_stewardship_team_alaska_2020)[b](#ref-alaska_region_data_stewardship_team_alaska_2020)),
required reading for tomorrow’s meeting.

## Friday, February 28

To do:

  - ~~Data Management Early Adopter Discussion and
    [mdEditor](https://www.mdeditor.org/) Training at 10:00~~

Potential projects to use as a test case:

  - 2019 black spruce bioinventory
  - 2019 Alaska blackfish diet study
  - 2020 Sandpiper Lake *Elodea* eradication
  - 2020 Swan Lake Fire burn severity and vegetation work

I sorted Berlese sample
[KNWR:Ento:11376](http://arctos.database.museum/guid/KNWR:Ento:11376).

Contents:

| identification | count |
| :------------- | ----: |
| Coleoptera     |     2 |
| Arachnida      |   211 |
| Araneae        |     5 |
| Collembola     |    24 |
| Cicadellidae   |     1 |
| Hemiptera      |    13 |
| Insecta        |     1 |

Most of the Hemiptera look like *Arctorthezia*.

Data Management talk/telecon notes/questions:

  - Why a new standard for metadata? What already exists? How does it
    relate to existing standards, e.g. Darwin Core, Dublin Core, RDF
    (Resource Description Framework)

  - Where are these metadata files going?

  - Where are the nicely structured project archives going?

  - Which laws for sharing data?
    
      - 1967 FOIA
      - Open Governemnt Data Act
      - Also AK regional strategic intent

  - Where is the form for the Regional Data Repository?

  - McCrea is my point of contact.

  - What is the ISO metadata standard that is being referring to? See
    <https://ak-region-dst.gitbook.io/alaska-region-mdeditor-interim-user-guide/glossary-of-terms-lcc-metadata-manual#iso>.
    ISO 19110, ISO 19115-1, ISO 19115-2? Answer: It is based off of ISO
    19115-2 or ISO 19115-1, geospatial standards.

![A slide from the Data Management Early Adopter
Discussion.](2020-02-28_data_manamgement_slide.jpg)  
A slide from the Data Management Early Adopter Discussion.

I entered more observation records from Jakubas and Firman
([1984](#ref-jakubas_sandpiper_1984)) (records
[KNWRObs:Herb:883](http://arctos.database.museum/guid/KNWRObs:Herb:883)–[KNWRObs:Herb:951](http://arctos.database.museum/guid/KNWRObs:Herb:951)).

I talked with McCrea about which project to use for the data management
work. We decided to use the snowshoe hare pellet count survey for this.
I am not excited about this because it is a dataset that is not in the
best shape and it is not one that I have worked with.

# March

## Monday, March 2

To do:

  - Write this week’s *Refuge Notebook* article.
  - ~~Finish arrangements for next week’s Anchorage trip.~~
  - Data Management Early Adopter follow-up. Start preparing hare survey
    data.
  - ~~1:00 pm Swan Lake Fire studies planning meeting.~~
  - Take tissue samples of worm specimens from Tyler.

I sorted Berlese sample
[KNWR:Ento:11377](http://arctos.database.museum/guid/KNWR:Ento:11377).

Contents:

| identification | count |
| :------------- | ----: |
| Arachnida      |   260 |
| Collembola     |    30 |
| Insecta        |     1 |
| Araneae        |     2 |
| Lithobiomorpha |     1 |
| Chilopoda      |     1 |
| Myrmicinae     |     3 |
| Hemiptera      |     3 |

I uploaded citations for Jakubas and Firman
([1984](#ref-jakubas_sandpiper_1984)). When I looked at the map I saw a
couple of missing records, so I added these
([KNWRObs:Herb:952](http://arctos.database.museum/guid/KNWRObs:Herb:952)–[KNWRObs:Herb:953](http://arctos.database.museum/guid/KNWRObs:Herb:953)).

I started work on this week’s *Refuge Notebook* article.

I participated in the planning meeting for burn severity mapping and
invasive species work for the Swan Lake Fire burn area.

``` r
## How many species were documented at site MG28?
## Downloaded analysis dataset.
wd <- "C:/Users/mattbowser/Documents/Refuge_Notebook/2020/submissions/2020-03-06_Refuge_Notebook"
setwd(wd)

d1 <- read.csv("oo_371428.csv", stringsAsFactors=FALSE)

sl <- grepl("site MG28", d1$SPEC_LOCALITY)
d1[sl,]
levels(as.factor(d1$SPECIES[sl]))
## 80 species or BIN resolution identifications were obtained at this site.

```

## Tuesday, March 3

To do:

  - Write this week’s *Refuge Notebook* article.
  - Data managmenet follow-up.
  - Prepare information/comment for next week’s BOF meeting.

I worked some on my *Refuge Notebook* article.

I sorted Berlese sample
[KNWR:Ento:11378](http://arctos.database.museum/guid/KNWR:Ento:11378).

Contents:

| identification | count |
| :------------- | ----: |
| Araneae        |     5 |
| Collembola     |   179 |
| Arachnida      |   333 |
| Insecta        |     3 |
| Oligochaeta    |     7 |
| Nematoda       |     2 |
| Lithobiomorpha |     2 |
| Hemiptera      |     4 |
| Coleoptera     |     2 |
| Hymenoptera    |     1 |
| Staphylinidae  |     1 |

I submitted a Record of General Use to Dan and Katherine for fluridone
applied to Hilda-Seppu Lakes in 2019.

I worked more on my *Refuge Notebook* article.

## Wednesday, March 4

To do:

  - ~~Submit this week’s *Refuge Notebook* article.~~
  - ~~Data managmenet follow-up.~~
  - Prepare information/comment for next week’s BOF meeting.

I finished my *Refuge Notebook* article and got it sent in.

I created an entry for the Kenai National Wildlife Refuge Snowshoe Hare
Grid Surveys project using [mdEditor](https://www.mdeditor.org/) and
sent this to the data management folks.

## Thursday, March 5

To do:

  - Prepare information/comment/images for next week’s BOF meeting.

I sorted Berlese sample
[KNWR:Ento:11379](http://arctos.database.museum/guid/KNWR:Ento:11379).

Contents:

| identification | count |
| :------------- | ----: |
| Hymenoptera    |     2 |
| Araneae        |    25 |
| Insecta        |     4 |
| Collembola     |   158 |
| Arachnida      |   310 |
| Hemiptera      |     3 |
| Phoridae       |     1 |

I worked on my statements for next week’s meeting. I learned that
non-native earthworms can increase nitrogen flux into streams (Pinder
[2013](#ref-pinder_ecology_2013); Costello and Lamberti
[2008](#ref-costello_non-native_2008),
[2009](#ref-costello_biological_2009)).

## Friday, March 6

I prepared for my Anchorage trip this week, including jump-starting and
fueling up the SUV that I am to use.

I added *Cortinarius lucorum*, *Corticarina minuta*, and *Stephostethus
armatulus* to the Kenai National Wildlife Refuge’s checklist on
FWSpecies. I worked a little on trying to fill in missing values in the
taxonomy exported from FWSpecies.

## Sunday, March 8

I rose early, drove to work, switched vehicles, and drove to Anchorage
for the Board of Fisheries meeting. Late in the afternoon I gave my 3
minute testimony on proposal 281 regarding the use of *Lumbricus* worms
as bait in fresh waters.

I gave my three-minute testimony and answered questions from the Board
of Fisheries.

## Monday, March 9

I attended the Board of Fisheries meeting, today the committee of the
whole.

While sitting through discussions of other proposals, I revisited
examining sequences from the *Aporrectodea* specimens from Pilgrim Hot
Springs. These are *Aporrectodea longa* or *Aporrectodea trapezoides*.

Our sequences agreed with sequences from BOLD BIN
[BOLD:AAA6243](http://boldsystems.org/index.php/Public_BarcodeCluster?clusteruri=BOLD:AAA6243),
which is mostly *Aporrectodea trapezoides* (100% to 99.83% similarity to
multiple sequences). There are three sequences of *Aporrectodea longa*
in this BIN. All three of these were collected from Ontario and
identified by Benoit Richard.

The *Aporrectodea longa* sequences from Porco et
al. ([2018](#ref-porco_reference_2018)) are in BIN
[BOLD:AAA2181](http://boldsystems.org/index.php/Public_BarcodeCluster?clusteruri=BOLD:AAA2181).

Our Pilgrim Hot springs sequences aligned with sequences on BOLD from
lineage *Aporrectodea trapezoides* L2 of Martinsson et
al. ([2015](#ref-martinsson_dna-barcoding_2015)) that are on BOLD.
Their *Aporrectodea trapezoides* L1 lineage is in BOLD BIN
[BOLD:AAC5530](http://boldsystems.org/index.php/Public_BarcodeCluster?clusteruri=BOLD:AAC5530).

Ok, our second Pilgrim Hot Springs samples is 100% similar to
*Aporrectodea trapezoides* lineage II, clade H, clone 1 *sensu*
Fernández et al. ([2011](#ref-fernandez_vagrant_2011)). This is the
widespread, peregrine clone.

The meeting ended early at around 15:00, so I checked out of the hotel
and drove home.

## Tuesday, March 10

I attended the Certified Pesticide Applicator workshop at the
Cooperative Extension Service office in Soldotna from 07:30 to 17:00.

I learned that the earthworm proposal did not pass due to enforceability
issues.

## Thursday, March 12

To do

  - ~~Time sheet.~~
  - ~~Get *Refuge Notebook* submitted.~~
  - ~~Travel voucher, etc.~~
  - ~~Data management telecon.~~
  - ~~Send pesticide applicator card info to Casey.~~
  - \~\~*Refuge Notebook* scheduling/correspondence.
  - Deal with Tyler’s earthworms.\~\~
  - Start 2020 *AKES Newsletter.*

I entered data for the Pilgrim Hot Springs worms
([KNWR:Inv:45](http://arctos.database.museum/guid/KNWR:Inv:45) and
[KNWR:Inv:46](http://arctos.database.museum/guid/KNWR:Inv:46)).

``` r
wd <- "D:/projects/earthworms/2020-03-12_Pilgrim_Hot_Springs_worms"
setwd(wd)
## Selecting all sequences from Fernández et al. (2011).
acc <- paste0("JF", 918579:918719)
write(acc, "2020-03-12-1032_Fernandez_accns.txt")
```

I submitted this to [Batch
Entrez](https://www.ncbi.nlm.nih.gov/sites/batchentrez%3E)…

``` r
wd <- "D:/projects/earthworms/2020-03-12_Pilgrim_Hot_Springs_worms"
setwd(wd)
## Selecting all sequences from Martinsson et al. (2015).
acc <- 101:129
acc <- as.character(acc)
acc <- substr(acc, 2, 3)
acc <- paste0("LUSWA0", acc, "-15")
write(acc, "2020-03-12-1057_Martinsson_seqs.txt")
```

### Data management meeting, 11:00

Question: What kinds of contacts should be added in metadata for a
project?

Answer: Anyone involved in the project at any time should be included.
The point of contact should always be current.

Action items:

  - Add more contacts to the snowshoe hare metadata.
  - Organize data into accepted file directory structure.
  - Enter three product records, one of which is a raw data file.

<!-- end list -->

``` r
## I need to clean up those identifiers.
wd <- "D:/projects/earthworms/2020-03-12_Pilgrim_Hot_Springs_worms"
setwd(wd)
library(Biostrings)

fas1 <- readDNAStringSet("sequence_data/Fernandez_COI.fas")
lab <- names(fas1)
labs <- strsplit(lab, " ")
lab <- paste(
 sapply(labs, "[", 1),
 sapply(labs, "[", 2),
 sapply(labs, "[", 3),
 sapply(labs, "[", 4),
 sapply(labs, "[", 5)
 )
lab <- gsub("\\.1", "", lab)

## Saving.
names(fas1) <- lab
writeXStringSet(fas1, "sequence_data/2020-03-12-1248_Fernandez_COI.fas", format="fasta", width=1000)

## Now for the Martinsson sequences.
fas2 <- readDNAStringSet("sequence_data/Martinsson_COI.fas")
lab <- names(fas2)
labs <- strsplit(lab, "\\|")
lab <- paste(
 sapply(labs, "[", 1),
 sapply(labs, "[", 2)
 )
## Saving.
names(fas2) <- lab
writeXStringSet(fas2, "sequence_data/2020-03-12-1253_Martinsson_COI.fas", format="fasta", width=1000)
```

I put all these sequences together into one FASTA file and I submitted
them to NGPhylogeny.fr. I used the “NGPhylogeny Analyse -
FastME/OneClick” analysis. This analysis is (temporarily?) available at
<https://ngphylogeny.fr/workspace/history/4f38834bbcb8bb64>. I rerooted
the tree on iTOL and the tree is available at
<https://itol.embl.de/tree/1641591522291071584068995#>.

![Phylogenetic tree including sequences from the Pilgrim Hot Springs
worms.](2020-03-12_Pilgrim_Hot_Springs_worm_tree.png)  
Phylogenetic tree including COI DNA barcode sequences from Fernández et
al. ([2011](#ref-fernandez_vagrant_2011)), Martinsson et
al. ([2015](#ref-martinsson_dna-barcoding_2015)), and the Pilgrim Hot
Springs worms.

It looks like
[KNWR:Inv:45](http://arctos.database.museum/guid/KNWR:Inv:45) (2-1-1) is
a new clone within *Aporrectodea trapezoides* lineage L2 near clone 1
and [KNWR:Inv:46](http://arctos.database.museum/guid/KNWR:Inv:46)
(1-1-1) is *Aporrectodea trapezoides* clone 1.

I am identifying
[KNWR:Inv:45](http://arctos.database.museum/guid/KNWR:Inv:45) as
*Aporrectodea trapezoides* lineage II, clade H, affinis clone 1. That is
a long name.

I am identifying
[KNWR:Inv:46](http://arctos.database.museum/guid/KNWR:Inv:46) as
*Aporrectodea trapezoides* lineage II, clade H, clone 1.

Both of these are in *Aporrectodea trapezoides* lineage L2 *sensu*
Martinsson et al. ([2015](#ref-martinsson_dna-barcoding_2015)).

## Friday, March 13

I finished data entry for Pilgrim Hot Springs worm specimens I worked
with yesterday.

I sorted Berlese sample
[KNWR:Ento:11380](http://arctos.database.museum/guid/KNWR:Ento:11380).

Contents:

| identification | count |
| :------------- | ----: |
| Chilopoda      |     3 |
| Scarabaeidae   |     2 |
| Arachnida      |   494 |
| Collembola     |    80 |
| Insecta        |     6 |
| Thysanoptera   |     2 |
| Hemiptera      |     2 |
| Araneae        |     2 |

I sorted Berlese sample
[KNWR:Ento:11381](http://arctos.database.museum/guid/KNWR:Ento:11381).

Contents:

| identification | count |
| :------------- | ----: |
| Hemiptera      |     1 |
| Collembola     |    14 |
| Arachnida      |   213 |
| Staphylinidae  |     1 |
| Coleoptera     |     1 |
| Thysanoptera   |     3 |
| Insecta        |     3 |
| Araneae        |     4 |
| Hymenoptera    |     2 |
| Formicidae     |     1 |

## Sunday, March 15

I modified the file structure of the *AKES Newsletter* and a made an
*[AKES Newsletter](https://github.com/mlbowser/AKES_Newsletter)*
repository on GitHub.

## Monday, March 16

To do:

  - *Refuge Notebook* catch-up.
  - ~~Format *AKES Newsletter* meeting article.~~
  - Deal with worm specimens from Tyler.
  - Early adopter data management homework.
  - Slikok occupancy.

I sorted Berlese sample
[KNWR:Ento:11382](http://arctos.database.museum/guid/KNWR:Ento:11382).

Contents:

| identification | count |
| :------------- | ----: |
| Oligochaeta    |    10 |
| Collembola     |   164 |
| Carabidae      |     2 |
| Arachnida      |   332 |
| Araneae        |     4 |
| Thysanoptera   |     4 |
| Staphylinidae  |     1 |
| Insecta        |     4 |

We just received word that we are strongly advised to telework as much
as possible, so I will not be getting any more lab work done for a
while.

I entered data for the earthworm specimens sent to me by Tyler
([KNWR:Inv:47](http://arctos.database.museum/guid/KNWR:Inv:47)–[KNWR:Inv:52](http://arctos.database.museum/guid/KNWR:Inv:52)).

I worked at home in the afternoon. I formatted an initial draft version
of the [meeting
article](https://github.com/mlbowser/AKES_Newsletter/blob/master/2020/01/meeting.tex)
for the *AKES Newsletter*.

This evening I received from RTL Genomics the metagenomic sequence data
from the blackfish gut contents that Apphia and I had sent in on January
30. I set up a project on [mBRAVE](http://www.mbrave.net/) with guesses
at analysis parameters and then I uploaded the first pair of compressed
FASTQ files.

## Tuesday, March 17

To do:

  - *Refuge Notebook* catch-up.
  - Early adopter data management homework.
  - Blackfish diet analysis.
  - Sandpiper Lake *Elodea* project PUPs.

I fiddled around with the first pair of FASTQ files I had uploaded to
mBRAVE. I failed to figure out how to merge the paired FASTQ files.
Maybe this functionality is not fully available yet on mBRAVE. I saw
that Braukmann et al. ([2019](#ref-braukmann_metabarcoding_2019)) used
QIIME to pair all of their sequences before uploading them to mBRAVE.
Stolz ([2019](#ref-stolz_nestling_2019)) used mBRAVE, but with
single-end reads. I could not find any other references where methods
using mBRAVE were detailed. There appears to be no help, guide, or
manual on mBRAVE right now.

I fiddled around with parameters for a while. Selecting “MERGED PAIRED
END READS” turned paired end merging on. I am not sure about the
trimming and masking. I turned primer masking on and selected trimming
the first 26 bp, both of which might remove the primers. Run parameters
are included below.

1.  Trimming  
    Trim Front: 26 bp  
    Trim End: 0 bp  
    Trim Length: 500 bp  
    Primer Masking: On
2.  Filtering  
    Min QV: 10 qv  
    Min Length: 150 bp  
    Max Bases with Low QV ( \<20): 4.0%  
    Max Bases with Ultra Low QV ( \<10): 1.0%
3.  Other Parameters  
    Pre-Clustering Threshold: None  
    ID Distance Threshold: 3.0%  
    Exclude From OTU Threshold: 3.0%  
    Minimum OTU Size: 1OTU Threshold: 0.0%
4.  Paired End (optional - Illumina instruments only)  
    Paired End Merging: On  
    Assembler Min Overlap: 25 bp  
    Assembler Max Substitution: 5 bp

That failed to run on the first filtering steps. I turned primer masking
off. This failed, also.

If I cannot get this to work, then I may be trying the [SCVUC
metabarcode
pipeline](https://github.com/Hajibabaei-Lab/SCVUC_COI_metabarcode_pipeline)
used by Hajibabaei et
al. ([2019](#ref-hajibabaei_coi_2019b)[b](#ref-hajibabaei_coi_2019b)).
This will require some setup to make this happen on Yeti.

For the data managmenet early adopters’ homework, I downloaded the
directory structure template from McCrea, which generally follows the
structure described by the Alaska Region Data Stewardship Team
([2019](#ref-ak_region_data_stewardship_team_alaska_2020)[a](#ref-ak_region_data_stewardship_team_alaska_2020)).

I worked on organizing files and authoring metadata for the snowshoe
hare project, part of the data management homework.

``` r
## Assemble plot location data.
data_dir <- "../data/raw_data/geodata"
file_list <- dir(data_dir)

## Select only .txt files.
file_list <- file_list[grepl("\\.txt", file_list)]

plot_data <- read.csv(paste0(data_dir, "/", file_list[1]), stringsAsFactors=FALSE)

dim(plot_data)
[1] 50  3
## Whoa, there should be 49 rows.
dim(unique(plot_data))
[1] 50  3
## All rows are unique.
length(levels(as.factor(plot_data$Ident)))
[1] 49 ## The number of unique labels is right.

plot_data <- plot_data[order(plot_data$Ident),]
plot_data

## There are two locations for plot CF-43.

## Plotting.
plot(plot_data$Longitude, plot_data$Latitude)
text(plot_data$Longitude, 
 plot_data$Latitude,
 labels=plot_data$Ident
 )
## One of the CF-43's is over by CF-49 far to the west of where it should be.

## Removing the western CF-43.
sl <- (plot_data$Ident=="CF-43") & (plot_data$Longitude < -150.603)
plot_data <- plot_data[!sl,]

## Plotting.
plot(plot_data$Longitude, plot_data$Latitude)
text(plot_data$Longitude, 
 plot_data$Latitude,
 labels=plot_data$Ident
 )

plot_data$grid_name <- gsub("\\.txt", "", gsub("_", " ", file_list[1]))

## Now loop through the rest of the files.
for (this_grid in 2:length(file_list))
 {
 plot_data_temp <- read.csv(paste0(data_dir, "/", file_list[this_grid]),
  stringsAsFactors=FALSE
  )
 plot_data_temp <- plot_data_temp[order(plot_data_temp$Ident),]
 plot_data_temp$grid_name <- gsub("\\.txt", "", gsub("_", " ", file_list[this_grid]))
 plot_data <- rbind(plot_data, plot_data_temp)
 }
 
names(plot_data)[1:3] <- c("plot_name", "longitude", "latitude")
plot_data <- plot_data[,c(4,1,3,2)]

## Now it is time to start reorganizing the observation data.

## Starting with the first sheet, Swanson River.
obs <- read.csv("../data/raw_data/observations/2020-03-17-2151_Swanson_River.csv", stringsAsFactors=FALSE)
obs$plot_name <- paste0("SR-", obs$plot_name)
obs$habitat <- tolower(obs$habitat)

## I will join this to the plot data later.
plot_hab <- obs[,c(1:2)]

obs <- obs[,c(1,3:38)]

library(reshape2)
obs <- melt(obs, id.vars="plot_name")
names(obs)[2:3] <- c("year", "pellet_count")
obs$year <- as.character(obs$year)
obs$year <- gsub("X", "", obs$year)
obs$year <- as.integer(obs$year)

obs_all <- obs

## Oil field.
obs <- read.csv("../data/raw_data/observations/2020-03-17-2218_Oil_field.csv", stringsAsFactors=FALSE)
obs$plot_name <- paste0("OF-", obs$plot_name)
obs$habitat <- tolower(obs$habitat)

## For joining to the plot data later.
plot_hab <- rbind(plot_hab, obs[,c(1:2)])

obs <- obs[,c(1,3:ncol(obs))]

obs <- melt(obs, id.vars="plot_name")
names(obs)[2:3] <- c("year", "pellet_count")
obs$year <- as.character(obs$year)
obs$year <- gsub("X", "", obs$year)
obs$year <- as.integer(obs$year)

obs_all <- rbind(obs_all, obs)

## Campfire.
obs <- read.csv("../data/raw_data/observations/2020-03-17-2229_Campfire.csv", stringsAsFactors=FALSE)
obs$plot_name <- paste0("CF-", obs$plot_name)
obs$habitat <- tolower(obs$habitat)

## For joining to the plot data later.
plot_hab <- rbind(plot_hab, obs[,c(1:2)])

obs <- obs[,c(1,3:ncol(obs))]

obs <- melt(obs, id.vars="plot_name")
names(obs)[2:3] <- c("year", "pellet_count")
obs$year <- as.character(obs$year)
obs$year <- gsub("X", "", obs$year)
obs$year <- as.integer(obs$year)

obs_all <- rbind(obs_all, obs)

## Skilak.
obs <- read.csv("../data/raw_data/observations/2020-03-18-0729_Skilak.csv", stringsAsFactors=FALSE)
obs$plot_name <- paste0("SK-", obs$plot_name)
obs$habitat <- tolower(obs$habitat)

## For joining to the plot data later.
plot_hab <- rbind(plot_hab, obs[,c(1:2)])

obs <- obs[,c(1,3:ncol(obs))]

obs <- melt(obs, id.vars="plot_name")
names(obs)[2:3] <- c("year", "pellet_count")
obs$year <- as.character(obs$year)
obs$year <- gsub("X", "", obs$year)
obs$year <- as.integer(obs$year)

obs_all <- rbind(obs_all, obs)

## Funny River.
obs <- read.csv("../data/raw_data/observations/2020-03-18-0734_Funny_River.csv", stringsAsFactors=FALSE)
obs$plot_name <- paste0("FR-", obs$plot_name)
obs$habitat <- tolower(obs$habitat)

## For joining to the plot data later.
plot_hab <- rbind(plot_hab, obs[,c(1:2)])

obs <- obs[,c(1,3:ncol(obs))]

obs <- melt(obs, id.vars="plot_name")
names(obs)[2:3] <- c("year", "pellet_count")
obs$year <- as.character(obs$year)
obs$year <- gsub("X", "", obs$year)
obs$year <- as.integer(obs$year)

obs_all <- rbind(obs_all, obs)

## Funny River New.
obs <- read.csv("../data/raw_data/observations/2020-03-18-0738_Funny_River_New.csv", stringsAsFactors=FALSE)
obs$plot_name <- paste0("FRN-", obs$plot_name)
obs$habitat <- tolower(obs$habitat)

plot_hab <- rbind(plot_hab, obs[,c(1:2)])

obs <- obs[,c(1,3:ncol(obs))]

obs <- melt(obs, id.vars="plot_name")
names(obs)[2:3] <- c("year", "pellet_count")
obs$year <- as.character(obs$year)
obs$year <- gsub("X", "", obs$year)
obs$year <- as.integer(obs$year)

obs_all <- rbind(obs_all, obs)

## Skilak New.
obs <- read.csv("../data/raw_data/observations/2020-03-18-0742_Skilak_New.csv", stringsAsFactors=FALSE)
obs$plot_name <- paste0("SKN-", obs$plot_name)
obs$habitat <- tolower(obs$habitat)

plot_hab <- rbind(plot_hab, obs[,c(1:2)])

obs <- obs[,c(1,3:ncol(obs))]

obs <- melt(obs, id.vars="plot_name")
names(obs)[2:3] <- c("year", "pellet_count")
obs$year <- as.character(obs$year)
obs$year <- gsub("X", "", obs$year)
obs$year <- as.integer(obs$year)

obs_all <- rbind(obs_all, obs)

## Checking before joining...
levels(as.factor(plot_hab$habitat))
[1] "black spruce" "mixed mature" "reburn mixed"
## Wow, that is really clean!

## Now conforming some of the plot names.
plot_data$plot_name <- gsub("FRB ", "FRN-", plot_data$plot_name)
plot_data$plot_name <- gsub("FRN-0", "FRN-", plot_data$plot_name)

plot_data_2 <- merge(x=plot_data, y=plot_hab, all.x=TRUE)

## Saving here.
write.csv(plot_data_2, "../data/final_data/geodata/2020-03-18_snowshoe_hare_plot_data.csv", row.names=FALSE)

write.csv(obs_all, "../data/final_data/observations/2020-03-18_snowshoe_hare_pellet_counts.csv", row.names=FALSE)
```

## Wednesday, March 18

To do:

  - ~~Submit this week’s *Refuge Notebook* article.~~
  - *Refuge Notebook* catch-up.
  - Early adopter data management homework.
  - Blackfish diet analysis.
  - ~~Sandpiper Lake *Elodea* project PUPs~~.

I finished processing snowshoe hare data that I had started yesterday.

That updated run on mBRAVE that I initiated yesterday never got
anywhere. It gives the message, “Please Wait: Initial Processing of This
Run is Underway.” This is how it has looked since the middle of the day
yesterday. I think there must be a problem with the run parameters.

I received and worked on formatting this week’s *Refuge Notebook*
article, getting it submitted to the
*[Clarion](https://www.peninsulaclarion.com/)*. I worked some on
formatting older articles, but the connection to the CMS was so slow
that I could not get this work posted.

I started working on the PUPs for herbicides to be used in *Elodea*
eradication work, spending most of the afternoon on PUPs.

We had a biology staff meeting via conference call this afternoon. We
are going to be teleworking for a while.

## Thursday, March 19

To do:

  - *Refuge Notebook* catch-up.
  - Early adopter data management homework.
  - Blackfish diet analysis.

I received an e-mail from Todd Steinlage regarding the Sandpiper Lake
*Elodea* sample. Based on a qPCR test and an ITS sequence, this one
appears to be *Elodea canadensis* (specimen record:
[KNWR:Herb:12443](http://arctos.database.museum/guid/KNWR:Herb:12443)).
This is consistent with our guess that this population was started by a
floatplane bringing it from Anchorage because Anchorage populations were
identified as *Elodea canadensis* (Morton et al.
[2019](#ref-morton_responding_2019)).

I entered data for the five *Dallia pectoralis* specimens from which
Apphia and I had dissected out gut contents
([KNWRObs:Fish:12](http://arctos.database.museum/guid/KNWRObs:Fish:12)–[KNWRObs:Fish:16](http://arctos.database.museum/guid/KNWRObs:Fish:16)).

Now I would like to set up the SCVUC pipeline on Yeti for analyzing the
blackfish gut contents. Using the instructions from
<https://github.com/Hajibabaei-Lab/SCVUC_COI_metabarcode_pipeline#prepare-your-environment-to-run-the-pipeline>.

I spent quite a while trying to get the SCVUC pipeline installed and
working. I think I need to give up on ORFfinder for now.

``` 
 846  2020-03-19 11:03:36  ls
  847  2020-03-19 11:03:46  cd /lustre
  848  2020-03-19 11:03:47  ls
  849  2020-03-19 11:03:55  cd projects
  850  2020-03-19 11:03:57  ls
  851  2020-03-19 11:04:00  cd fws
  852  2020-03-19 11:04:03  ls
  853  2020-03-19 11:04:10  cd fws_bio
  854  2020-03-19 11:04:13  ls
  855  2020-03-19 11:04:19  cd mattbowser
  856  2020-03-19 11:04:31  cwd
  857  2020-03-19 11:04:37  pwd
  858  2020-03-19 11:07:01  cd /lustre/projects/fws/fws_bio/mattbowser
  859  2020-03-19 11:10:45  ls
  860  2020-03-19 11:10:52  cd projects
  861  2020-03-19 11:10:54  ls
  862  2020-03-19 11:11:28  cd /lustre/projects/fws/fws_bio/mattbowser/projects
  863  2020-03-19 11:12:23  mkdir COI_metabarcoding
  864  2020-03-19 11:12:35  cd COI_metabarcoding
  865  2020-03-19 11:32:54  git clone https://github.com/Hajibabaei-Lab/SCVUC_COI_metabarcode_pipeline.git
  866  2020-03-19 11:33:03  ls
  867  2020-03-19 11:33:15  cd SCV*
  868  2020-03-19 11:34:08  ls
  869  2020-03-19 11:35:11  module avail
  870  2020-03-19 11:36:31  module load python/anaconda3
  871  2020-03-19 11:40:46  conda env create -f environment.yml
  872  2020-03-19 12:06:37  conda activate SCVUCv4.3
  873  2020-03-19 12:09:03  source activate SCVUCv4.3
  874  2020-03-19 12:09:55  wget ftp://ftp.ncbi.nlm.nih.gov/genomes/TOOLS/ORFfinder/linux-i64/ORFfinder.gz
  875  2020-03-19 12:10:23  gunzip ORFfinder.gz
  876  2020-03-19 12:10:35  chmod 755 ORFfinder
  877  2020-03-19 12:10:58  mv ORFfinder ~/bin/.
  878  2020-03-19 12:11:41  mkdir /home/mattbowser/bin
  879  2020-03-19 12:11:47  mv ORFfinder ~/bin/.
  880  2020-03-19 12:12:04  ORFfinder
  881  2020-03-19 12:15:02  cd ~/miniconda3/envs/SCVUCv4.3/etc/conda/activate.d/
  882  2020-03-19 12:15:08  cd ~/
  883  2020-03-19 12:15:11  ls
  884  2020-03-19 12:28:43  cd /lustre/projects/fws/fws_bio/mattbowser/projects/COI_metabarcoding/SCVUC_COI_metabarcode_pipeline
  885  2020-03-19 12:28:46  ls
  886  2020-03-19 12:32:05  ls -al
  887  2020-03-19 12:32:34  cd ~/
  888  2020-03-19 12:32:38  ls -al
  889  2020-03-19 12:32:46  cd .conda
  890  2020-03-19 12:32:47  ls
  891  2020-03-19 12:32:51  cd envs
  892  2020-03-19 12:32:53  ls
  893  2020-03-19 12:33:27  cd SCV*
  894  2020-03-19 12:33:28  ls
  895  2020-03-19 12:33:35  cd etc
  896  2020-03-19 12:33:36  ls
  897  2020-03-19 12:33:52  cd conda
  898  2020-03-19 12:33:54  ls
  899  2020-03-19 12:34:00  cd activate.d
  900  2020-03-19 12:34:01  ls
  901  2020-03-19 12:34:52  vi LD_PATH.sh
  902  2020-03-19 12:37:05  ls
  903  2020-03-19 12:37:36  cd ..
  904  2020-03-19 12:37:54  cd deactivate.d
  905  2020-03-19 12:37:55  ls
  906  2020-03-19 12:38:14  vi LD_PATH.sh
  907  2020-03-19 12:39:03  ls
  908  2020-03-19 12:39:40  cd ..
  909  2020-03-19 12:39:51  ls
  910  2020-03-19 12:40:16  cd lib64
  911  2020-03-19 12:40:34  ls
  912  2020-03-19 12:40:53  ln -s ../glibc-2.14/lib/libc.so.6 libc.so.6
  913  2020-03-19 12:40:57  ls
  914  2020-03-19 12:43:10  source deactivate SCVUCv4.3
  915  2020-03-19 12:43:35  source activate SCVUCv4.3
  916  2020-03-19 12:44:26  ORFfinder
  917  2020-03-19 12:56:21  ls
  918  2020-03-19 12:59:06  cd ..
  919  2020-03-19 12:59:09  ls
  920  2020-03-19 13:02:57  cd lib
  921  2020-03-19 13:02:59  ls
  922  2020-03-19 13:05:50  cd glib
  923  2020-03-19 13:06:21  cd glib*
  924  2020-03-19 13:06:22  ls
  925  2020-03-19 13:06:29  cd ..
  926  2020-03-19 13:06:33  cd lib
  927  2020-03-19 13:06:34  ls
  928  2020-03-19 13:44:57  cd ~/.conda/env/SCVUCv4.3
  929  2020-03-19 13:45:03  ls
  930  2020-03-19 13:45:07  ls -al
  931  2020-03-19 13:45:11  cd .conda
  932  2020-03-19 13:45:12  ls
  933  2020-03-19 13:45:18  cd envs
  934  2020-03-19 13:45:33  cd ~/.conda/envs/SCVUCv4.3
  935  2020-03-19 13:45:36  ls
  936  2020-03-19 13:45:51  cd include
  937  2020-03-19 13:45:56  ls
  938  2020-03-19 13:46:08  cd ..
  939  2020-03-19 13:46:09  ls
  940  2020-03-19 13:46:22  cd lib
  941  2020-03-19 13:46:25  ls
  942  2020-03-19 13:46:46  cd glib*
  943  2020-03-19 13:46:48  ls
  944  2020-03-19 13:46:52  cd include
  945  2020-03-19 13:46:53  ls
  946  2020-03-19 13:46:56  cd ..
  947  2020-03-19 13:50:15  find ~/.conda/envs/SCVUCv4.3 -name "*glibc*"
  948  2020-03-19 13:51:42  module avail
  949  2020-03-19 14:27:01  cd /lustre/projects/fws/fws_bio/mattbowser/projects/COI_metabarcoding/SCVUC_COI_metabarcode_pipeline
  950  2020-03-19 14:27:03  ls
  951  2020-03-19 14:41:08  cd ~/
  952  2020-03-19 14:41:32  wget http://ftp.gnu.org/gnu/glibc/glibc-2.14.tar.gz
  953  2020-03-19 14:44:28  tar -xzvf glibc-2.14.tar.gz
  954  2020-03-19 14:53:57  ls
  955  2020-03-19 14:54:29  rm -R glibc-2.14
  956  2020-03-19 14:57:24  ls
  957  2020-03-19 14:57:32  ls -al
  958  2020-03-19 14:57:42  cd glibc*
  959  2020-03-19 14:57:44  ls
  960  2020-03-19 14:57:59  rm -R *
  961  2020-03-19 14:59:13  ls
  962  2020-03-19 14:59:31  rm -R *
  963  2020-03-19 15:01:34  ls
  964  2020-03-19 15:01:49  rm -R *
  965  2020-03-19 15:08:51  module avail
  966  2020-03-19 15:15:57  module list
  967  2020-03-19 15:16:15  module load python/miniconda3-gcc6.1.0
  968  2020-03-19 15:16:57  source activate SCVUCv4.3
  969  2020-03-19 15:17:21  cd ..
  970  2020-03-19 15:17:24  ls
  971  2020-03-19 15:17:52  source deactivate SCVUCv4.3
  972  2020-03-19 15:18:00  module purge
  973  2020-03-19 15:19:05  rmdir glibc-2.14
  974  2020-03-19 15:19:07  ls
  975  2020-03-19 15:19:34  tar -xzvf glibc-2.14.tar.gz
  976  2020-03-19 15:21:50  ls
  977  2020-03-19 15:22:04  cd glibc-2.14
  978  2020-03-19 15:22:06  ls
  979  2020-03-19 15:24:06  uname -r
  980  2020-03-19 15:25:51  cat /etc/os-release
  981  2020-03-19 15:26:39  ./configure
  982  2020-03-19 15:27:06  mkdir build
  983  2020-03-19 15:27:38  dmesg | head -1
  984  2020-03-19 15:28:11  cat /proc/version
  985  2020-03-19 15:32:32  cd /lib
  986  2020-03-19 15:32:34  ls
  987  2020-03-19 15:34:56  ldd --version
  988  2020-03-19 15:36:13  /lib64/libc.so.6 --version
  989  2020-03-19 15:37:01  ## So Yeti has glib version glibc version 2.12.
  990  2020-03-19 15:39:59  cd ~/
  991  2020-03-19 15:40:00  ls
  992  2020-03-19 15:40:07  cd glibc*
  993  2020-03-19 15:40:08  ls
  994  2020-03-19 15:40:17  cd build
  995  2020-03-19 15:40:26  ../configure
  996  2020-03-19 15:41:39  ls
  997  2020-03-19 15:41:54  make
  998  2020-03-19 15:44:38  ../configure --prefix /home/mattbowser/.conda/envs/SCVUCv4/glibc-2.14
  999  2020-03-19 15:44:50  ls
 1000  2020-03-19 15:44:56  make
 1001  2020-03-19 15:56:17  wget https://ayera.dl.sourceforge.net/project/rdp-classifier/rdp-classifier/rdp_classifier_2.12.zip
 1002  2020-03-19 16:04:04  ls
 1003  2020-03-19 16:04:11  cd glib*
 1004  2020-03-19 16:04:14  cd build
 1005  2020-03-19 16:04:15  ls
 1006  2020-03-19 16:05:05  make
 1007  2020-03-19 16:26:00  ls
 1008  2020-03-19 16:29:28  history
 1009  2020-03-19 16:30:39  pwd
 1010  2020-03-19 16:32:33  cd /conda/envs/
 1011  2020-03-19 16:32:47  cd ~/.conda/envs
 1012  2020-03-19 16:32:48  ls
 1013  2020-03-19 16:32:56  cd SCVUC*
 1014  2020-03-19 16:32:58  ls
 1015  2020-03-19 16:33:42  mkdir pwd
 1016  2020-03-19 16:33:53  rmdir pwd
 1017  2020-03-19 16:33:56  pwd
 1018  2020-03-19 16:34:49  mkdir glibc-2.14
 1019  2020-03-19 16:34:52  ls
 1020  2020-03-19 16:35:08  cp /home/mattbowser/glibc-2.14/build/* /home/mattbowser/.conda/envs/SCVUCv4.3
 1021  2020-03-19 16:35:29  /glibc-2.14/ls
 1022  2020-03-19 16:35:32  ls
 1023  2020-03-19 16:35:53  cd /home/mattbowser/.conda/envs/SCVUCv4.3
 1024  2020-03-19 16:36:17  /glibc-2.14/ls
 1025  2020-03-19 16:36:20  ls
 1026  2020-03-19 16:36:48  ls -al
 1027  2020-03-19 16:39:10  find . -type f -cmin -10
 1028  2020-03-19 16:39:33  find . -type f -cmin -10 -delete
 1029  2020-03-19 16:39:38  ls
 1030  2020-03-19 16:39:59  cd glibc-2.14
 1031  2020-03-19 16:40:00  ls
 1032  2020-03-19 16:40:16  cp -R /home/mattbowser/glibc-2.14/build/* /home/mattbowser/.conda/envs/SCVUCv4.3/glibc-2.14/
 1033  2020-03-19 16:44:43  ls
 1034  2020-03-19 16:45:15  cd ..
 1035  2020-03-19 16:45:16  ls
 1036  2020-03-19 16:45:23  cd lib64
 1037  2020-03-19 16:45:24  ls
 1038  2020-03-19 16:46:14  ln -s ../glibc-2.14/libc.so.6 libc.so.6
 1039  2020-03-19 16:46:24  rm libc.so.6
 1040  2020-03-19 16:46:28  ln -s ../glibc-2.14/libc.so.6 libc.so.6
 1041  2020-03-19 16:46:30  ls
 1042  2020-03-19 16:47:23  history
 1043  2020-03-19 16:48:27  module load python/anaconda3
 1044  2020-03-19 16:48:49  source activate SCVUCv4.3
 1045  2020-03-19 16:49:16  ORFfinder
 1046  2020-03-19 16:51:57  ## Wow, now we need version 2.17.
 1047  2020-03-19 16:52:21  history
 1048  2020-03-19 16:52:38  history | grep wget
 1049  2020-03-19 16:54:24  cd ~/
 1050  2020-03-19 16:54:26  ls
 1051  2020-03-19 16:54:33  wget http://ftp.gnu.org/gnu/glibc/glibc-2.17.tar.gz
 1052  2020-03-19 16:54:44  ls
 1053  2020-03-19 16:55:13  tar -zxvf glibc-2.17.tar.gz
 1054  2020-03-19 16:56:45  ls
 1055  2020-03-19 16:56:54  cd glibc-2.17
 1056  2020-03-19 16:56:55  ls
 1057  2020-03-19 16:57:00  mkdir build
 1058  2020-03-19 16:57:02  cd build
 1059  2020-03-19 16:57:03  ls
 1060  2020-03-19 16:57:37  history | grep configure
 1061  2020-03-19 16:57:56  ../configure --prefix /home/mattbowser/.conda/envs/SCVUCv4/glibc-2.14
 1062  2020-03-19 16:59:11  ## Maybe I need to give up on this. the autoconf version is not campatible.
 1063  2020-03-19 16:59:22  history
 1000  2020-03-19 17:02:07  exit
 1001  2020-03-19 19:05:32  ls
 1002  2020-03-19 19:05:46  gunzip rdp_class*
 1003  2020-03-19 19:08:06  unzip rdp_class*
 1004  2020-03-19 19:08:24  ls
 1005  2020-03-19 19:10:19  mv ~/rdp_classifier_2.12 /lustre/projects/fws/fws_bio/mattbowser/projects/COI_metabarcoding/SCVUC_COI_metabarcode_pipeline/rdp_classifier_2.12
 1006  2020-03-19 19:10:52  cd /lustre/projects/fws/fws_bio/mattbowser/projects/COI_metabarcoding/SCVUC_COI_metabarcode_pipeline/rdp_classifier_2.12
 1007  2020-03-19 19:10:55  ls
 1008  2020-03-19 19:14:31  cd ..
 1009  2020-03-19 19:14:33  ls
 1010  2020-03-19 19:14:57  wget https://github.com/terrimporter/CO1Classifier/releases/download/v4/CO1v4_trained.tar.gz
 1011  2020-03-19 19:18:12  ls
 1012  2020-03-19 19:18:31  tar -zxvf CO1v4_trained.tar.gz
 1013  2020-03-19 19:20:13  ls
 1014  2020-03-19 19:20:29  rm CO1v4_trained.tar.gz
 1015  2020-03-19 19:20:31  ls
 1016  2020-03-19 19:20:36  cd rdp*
 1017  2020-03-19 19:20:37  ls
 1018  2020-03-19 19:22:56  cd ..
 1019  2020-03-19 19:22:57  ls
 1020  2020-03-19 19:23:01  cd myd*
 1021  2020-03-19 19:23:03  ls
 1022  2020-03-19 19:27:34  pwd
 1023  2020-03-19 19:31:19  cd ..
 1024  2020-03-19 19:31:20  ls
 1025  2020-03-19 19:31:27  mkdir data
 1026  2020-03-19 19:31:28  ls
 1027  2020-03-19 19:31:31  cd data
 1028  2020-03-19 19:32:40  wget https://rtlgenomics.s3.amazonaws.com/7SLpgTzp2k/Bowser_6382Raw03132020.zip?AWSAccessKeyId=AKIAI4T5MZRBWMKZM5NA&Expires=1592265873&Signature=fCzfVZlfw5aQ6sFjUJrUb3466sY%3D
 1029  2020-03-19 20:02:16  history
```

## Friday, March 20

To do:

  - *Refuge Notebook* catch-up.
  - Early adopter data management homework.
  - Blackfish diet analysis.

I set up a project on Arctos for the blakfish diet work at
<http://arctos.database.museum/project/10003367>.

I was able to get the SCVUC pipeline going under linux mint, but there
was a problem with the reverse primer. I did overcome this later by
reverting to the original reverse primer in the ml-jg script.

I kept running into trouble with running out of memory for the RDP
Classifier (Wang et al. [2007](#ref-wang_naive_2007)). The system would
kill the process after it used all memory. The script asks for 8 Gb of
RAM; this computer has 4. I tried modifying the script to ask for only 2
Gb, but then RDP Classifier failed within Java, again running out of
memory. I will try repartitioning to add a bunch of swap space on that
computer.

### Telecon at 11:00

  - Send Mark a list of planned work for next week.
  - Put together plans for this field season.

Next week work list:

  - Finish blackfish diet analysis to get an article into the upcoming
    issue of the *Newsletter of the Alaska Entomological Society*.
  - Catch up on formatting *Refuge Notebook* articles and posting them
    to the CMS.
  - Work on metadata / next data management steps for the Kenai NWR
    snowshoe hare dataset as part of work for the regional data
    management early adopters’ team.
  - Write article on our finding *Bimastos parvus*, a second native
    earthworm for Alaska, for the *AKES Newsletter*?
  - Get started on Slikok watershed project occupancy analysis?

<!-- end list -->

    cd /home/matt/SCVUC_COI_metabarcode_pipeline
    snakemake --jobs 1 --snakefile snakefile_withPseudogeneFiltering --configfile 2020-03-20-0747_config_ml-jg.yaml
    
    cutadapt: error: Character 'I' in adapter sequence 'TAIACYTCIGGRTGICCRAARAAYCA' is not a valid IUPAC code. Use only characters XACGTURYSWKMBDHVN.
    
    Subbing "D" for "I". See <https://en.wikipedia.org/wiki/Nucleotide>.
    
    snakemake --jobs 1 --snakefile snakefile_withPseudogeneFiltering --configfile 2020-03-20-0747_config_ml-jg.yaml
    
    Fatal error: Unable to read from file (ml-jg/cat.denoised)
    Error in job chimera_removal while creating output file ml-jg/cat.denoised.nonchimeras.
    RuleException:
    CalledProcessError in line 412 of /home/matt/SCVUC_COI_metabarcode_pipeline/snakefile_withPseudogeneFiltering:
    Command 'vsearch --uchime3_denovo ml-jg/cat.denoised --sizein --xsize --nonchimeras ml-jg/cat.denoised.nonchimeras --relabel 'Zotu' --log ml-jg/chimeraRemoval.log' returned non-zero exit status 1.
      File "/home/matt/SCVUC_COI_metabarcode_pipeline/snakefile_withPseudogeneFiltering", line 412, in __rule_chimera_removal
      File "/home/matt/miniconda3/envs/SCVUCv4.3/lib/python3.6/concurrent/futures/thread.py", line 56, in run
    Removing output files of failed job chimera_removal since they might be corrupted:
    ml-jg/cat.denoised.nonchimeras
    Will exit after finishing currently running jobs.
    Exiting because a job execution failed. Look above for error message
    
    This seems to have had problems with the reverse primer. After trimming the reverse primer the files were quite small.
    
    TGRTTYTTYGGNCAYCCNGARGTNTA ## From original config file
    TADACYTCDGGRTGDCCRAARAAYCA ## What I ran last
    TAIACYTCIGGRTGICCRAARAAYCA ## From lab
    TANACYTCNGGRTGNCCRAARAAYCA ## Going to try this.
    
    stop @ 08:50 start @ 09:10
    
    snakemake --jobs 1 --snakefile snakefile_withPseudogeneFiltering --configfile 2020-03-20-0747_config_ml-jg.yaml
    
    ## That did not work, either.
    ## Switching back to the original.
    
    /bin/bash: line 1:  6375 Killed                  java -Xmx8g -jar /home/matt/SCVUC_COI_metabarcode_pipeline/rdp_classifier_2.12/dist/classifier.jar classify -t /home/matt/SCVUC_COI_metabarcode_pipeline/mydata_trained/rRNAClassifier.properties -o ml-jg/rdp.out.tmp ml-jg/cat.denoised.nonchimeras
    Error in job taxonomic_assignment while creating output file ml-jg/rdp.out.tmp.
    RuleException:
    CalledProcessError in line 445 of /home/matt/SCVUC_COI_metabarcode_pipeline/snakefile_withPseudogeneFiltering:
    Command 'java -Xmx8g -jar /home/matt/SCVUC_COI_metabarcode_pipeline/rdp_classifier_2.12/dist/classifier.jar classify -t /home/matt/SCVUC_COI_metabarcode_pipeline/mydata_trained/rRNAClassifier.properties -o ml-jg/rdp.out.tmp ml-jg/cat.denoised.nonchimeras' returned non-zero exit status 137.
      File "/home/matt/SCVUC_COI_metabarcode_pipeline/snakefile_withPseudogeneFiltering", line 445, in __rule_taxonomic_assignment
      File "/home/matt/miniconda3/envs/SCVUCv4.3/lib/python3.6/concurrent/futures/thread.py", line 56, in run
    Removing output files of failed job taxonomic_assignment since they might be corrupted:
    ml-jg/rdp.out.tmp
    Will exit after finishing currently running jobs.
    Exiting because a job execution failed. Look above for error message
    
    ## Made that jar executable.
    
    ## Trying again.
    snakemake --jobs 1 --snakefile snakefile_withPseudogeneFiltering --configfile 2020-03-20-0747_config_ml-jg.yaml
    
    Error in job taxonomic_assignment while creating output file ml-jg/rdp.out.tmp.
    RuleException:
    CalledProcessError in line 445 of /home/matt/SCVUC_COI_metabarcode_pipeline/snakefile_withPseudogeneFiltering:
    Command 'java -Xmx8g -jar /home/matt/SCVUC_COI_metabarcode_pipeline/rdp_classifier_2.12/dist/classifier.jar classify -t /home/matt/SCVUC_COI_metabarcode_pipeline/mydata_trained/rRNAClassifier.properties -o ml-jg/rdp.out.tmp ml-jg/cat.denoised.nonchimeras' returned non-zero exit status 137.
      File "/home/matt/SCVUC_COI_metabarcode_pipeline/snakefile_withPseudogeneFiltering", line 445, in __rule_taxonomic_assignment
      File "/home/matt/miniconda3/envs/SCVUCv4.3/lib/python3.6/concurrent/futures/thread.py", line 56, in run
    Removing output files of failed job taxonomic_assignment since they might be corrupted:
    ml-jg/rdp.out.tmp
    Will exit after finishing currently running jobs.
    Exiting because a job execution failed. Look above for error message
    
    ## Just trying that failed command.
    java -Xmx8g -jar /home/matt/SCVUC_COI_metabarcode_pipeline/rdp_classifier_2.12/dist/classifier.jar classify -t /home/matt/SCVUC_COI_metabarcode_pipeline/mydata_trained/rRNAClassifier.properties -o ml-jg/rdp.out.tmp ml-jg/cat.denoised.nonchimeras
    
    ## This was killed by the system, I think from using too much memory.
    
    Mar 20 09:59:42 aurochs kernel: [ 8396.151925] Out of memory: Killed process 6840 (java) total-vm:10309908kB, anon-rss:3333148kB, file-rss:0kB, shmem-rss:0kB
    
    ## Oh, I see. It is asking for 8 Gb of memory on a 4 Gb memory machine.
    -Xmx8g
    
    ## Edited snakefile_withPseudogeneFiltering, saved as snakefile_withPseudogeneFiltering_2020-03-20-1141
    
    snakemake --jobs 1 --snakefile snakefile_withPseudogeneFiltering_2020-03-20-1141 --configfile 2020-03-20-0747_config_ml-jg.yaml
    
    That failed, also. This thing is really hungry for memory.
    
    Exception in thread "main" java.lang.OutOfMemoryError: Java heap space
        at java.util.Arrays.copyOf(Arrays.java:3210)
        at java.util.Arrays.copyOf(Arrays.java:3181)
        at java.util.ArrayList.grow(ArrayList.java:261)
        at java.util.ArrayList.ensureExplicitCapacity(ArrayList.java:235)
        at java.util.ArrayList.ensureCapacityInternal(ArrayList.java:227)
        at java.util.ArrayList.add(ArrayList.java:458)
        at edu.msu.cme.rdp.classifier.io.GenusWordProbFileParser.createGenusWordProbList(GenusWordProbFileParser.java:59)
        at edu.msu.cme.rdp.classifier.TrainingInfo.createGenusWordProbList(TrainingInfo.java:127)
        at edu.msu.cme.rdp.classifier.utils.ClassifierFactory.getNonDefaultFactory(ClassifierFactory.java:161)
        at edu.msu.cme.rdp.classifier.utils.ClassifierFactory.getFactory(ClassifierFactory.java:145)
        at edu.msu.cme.rdp.multicompare.MultiClassifier.<init>(MultiClassifier.java:64)
        at edu.msu.cme.rdp.multicompare.MultiClassifier.<init>(MultiClassifier.java:75)
        at edu.msu.cme.rdp.multicompare.Main.main(Main.java:247)
        at edu.msu.cme.rdp.classifier.cli.ClassifierMain.main(ClassifierMain.java:67)
    
    ## Ok, I repartitioned the hard drive, making 24 Gb more of swap, so now we have 4 Gb of memory and 26 Gb of swap.
    
    cd SC*
    conda activate SCVUCv4.3
    snakemake --jobs 1 --snakefile snakefile_withPseudogeneFiltering_2020-03-20-1528 --configfile 2020-03-20-0747_config_ml-jg.yaml
    
    ## That worked!

## Saturday, March 21

I looked at results from yesterday and worked on adding identifications.

    ## Downloaded ASV Sequences from Bowser et al. (2020)
    
    Bowser, M. L. 2020. Supplementary material 5. ASV sequences. doi.org/10.3897/BDJ.8.e50124.suppl5
    
    oo_371419.fas
    
    cd SC*/ml-jg
    source activate SCVUCv4.3
    
    ## Assign taxonomy from Bowser (2020).
    vsearch --usearch_global chimera.denoised.nonchimeras.taxon \
    --db 2020-03-21-0600_oo_371419.fas \
    --id .90 \
    --iddef 0 \
    --blast6out 2020-03-21-0626_vsearch_Bowser_2020_matches.txt
    
    ## Assign taxonomy based on DS-BOWSER BOLD project.
    vsearch --usearch_global chimera.denoised.nonchimeras.taxon \
    --db 2020-03-21-0611_DS-BOWSER.fas \
    --id .90 \
    --iddef 0 \
    --blast6out 2020-03-21-0629_vsearch_DS-BOWSER_matches.txt
    
    ## Assign taxonomy based on AKTerrInvCOILib.
    vsearch --usearch_global chimera.denoised.nonchimeras.taxon \
    --db 2019-03-25-0832_clusters.fas \
    --id .90 \
    --iddef 0 \
    --blast6out 2020-03-21-0639_vsearch_AKTerrInvCOILib_matches.txt

I started looking at these identifications. I think it is interesting
that *Dallia pectoralis* did not make it through the filtering. This
makes me think that the filtering is too stringent because there were
*D. pectoralis* reads in the original FASTQ files.

## Monday, March 23

    ## Downloaded ASV Sequences from Bowser et al. (2020)
    
    Bowser, M. L. 2020. Supplementary material 5. ASV sequences. doi.org/10.3897/BDJ.8.e50124.suppl5
    
    oo_371419.fas
    
    cd SC*/ml-jg
    source activate SCVUCv4.3
    
    ## Assign taxonomy from Bowser (2020).
    vsearch --usearch_global chimera.denoised.nonchimeras.taxon \
    --db 2020-03-21-0600_oo_371419.fas \
    --id .90 \
    --iddef 0 \
    --blast6out 2020-03-21-0626_vsearch_Bowser_2020_matches.txt
    
    ## Assign taxonomy based on DS-BOWSER BOLD project.
    vsearch --usearch_global chimera.denoised.nonchimeras.taxon \
    --db 2020-03-21-0611_DS-BOWSER.fas \
    --id .90 \
    --iddef 0 \
    --blast6out 2020-03-21-0629_vsearch_DS-BOWSER_matches.txt
    
    ## Assign taxonomy based on AKTerrInvCOILib.
    vsearch --usearch_global chimera.denoised.nonchimeras.taxon \
    --db 2019-03-25-0832_clusters.fas \
    --id .90 \
    --iddef 0 \
    --blast6out 2020-03-21-0639_vsearch_AKTerrInvCOILib_matches.txt

``` r
setwd("D:/projects/blackfish/2020_blackfish_diet/2020-03_SCVUC/ml-jg")

results1 <- read.csv("results.csv", stringsAsFactors=FALSE)

length(levels(as.factor(results1$ml.jg_GlobalESV)))
[1] 212

aklib <- read.delim("2020-03-23-0659_vsearch_AKTerrInvCOILib_matches.txt", header=FALSE, stringsAsFactors=FALSE)

names(results1)[1] <- "ESV_name"
results1$ESV_name <- gsub("ml-jg_", "", results1$ESV_name)

## column names from http://www.metagenomics.wiki/tools/blast/blastn-output-format-6
names(aklib)[1:3] <- c("ESV_name", "AKlibprocessid", "pident")
```

I accidentally overwrote most of the rest of that R script.

I attended the 11:00 biology check-in telecon.

I prepared data for Rob Massengill, but I accidentally overwrote the R
script. I had downloaded occurrence data from the Vogel Lake vicinity
from GBIF (GBIF.org
[2020](#ref-gbiforg_gbif_2020-3)[d](#ref-gbiforg_gbif_2020-3)), Arctos,
and iNaturalist, then processed these. I had entered data for specimen
records
[KNWR:Ento:11383](http://arctos.database.museum/guid/KNWR:Ento:11383)
(*Gammarus lacustris*),
[KNWR:Ento:11384](http://arctos.database.museum/guid/KNWR:Ento:11384)
(*Caenis youngi*),
[KNWR:Inv:53](http://arctos.database.museum/guid/KNWR:Inv:53)
(*Sphaerium* sp. BOLD:AAG0345), and
KNWR:Inv:54\](http://arctos.database.museum/guid/KNWR:Inv:54)
(Glossiphoniidae).

## Tuesday, March 24

Picking up where I left off yesterday, preparing data for Rob.

``` r
## I did some editing by hand, adding lake_name and removing some terrestrial records.

setwd("D:/projects/Elodea/Sandpiper_Lake/2020-03-23_occurrence_data_for_Rob")

d4 <- read.csv("2020-03-24-0607_Vogel_Lake_vicinity_occurrence_data.csv", stringsAsFactors=FALSE)

library(maptools)
library(rgdal)

wgs84 <- CRS("+proj=longlat +ellps=WGS84 +datum=WGS84")

coordinates(d4) <- c("longitude", "latitude")
proj4string(d4) <- wgs84

writeOGR(
 d4, 
 dsn="2020-03-24-0607_Vogel_Lake_vicinity_occurrence_data.kml", 
 layer="occurrences", 
 driver="KML", 
 dataset_options=c("NameField=scientific_name"), 
 overwrite_layer=TRUE
 )

## I did some editing by hand.
d4 <- read.csv("2020-03-24-0654_Vogel_Lake_vicinity_occurrence_data.csv", stringsAsFactors=FALSE)

coordinates(d4) <- c("longitude", "latitude")
proj4string(d4) <- wgs84

writeOGR(
 d4, 
 dsn="2020-03-24-0654_Vogel_Lake_vicinity_occurrence_data.kml", 
 layer="occurrences", 
 driver="KML", 
 dataset_options=c("NameField=scientific_name"), 
 overwrite_layer=TRUE
 )

## Now for some summaries.
library(reshape2)

d4 <- read.csv("2020-03-24-0654_Vogel_Lake_vicinity_occurrence_data.csv", stringsAsFactors=FALSE)

d5 <- d4[,1:9]
d5$presence <- 1

d6 <- melt(d5, measure.vars="presence")

d7 <- dcast(d6, formula = ... ~ lake_name, fun.aggregate=length)
d7 <- d7[,c(1:8, 10:12)]
to10 <- function(x)
 {
 if (x > 0){1}
 else {0}
 }
d7[,9:11] <- apply(d7[,9:11], c(1,2), to10)
write.csv(d7, "2020-03-24-0728_Vogel_Lake_vicinity_occurrence_table.csv", row.names=FALSE)

## Now for an aggregate table.
d8 <- dcast(d6, formula = ... ~ ., fun.aggregate=length)
d8 <- d8[,c(9,1:8)]
write.csv(d8, "2020-03-24-0737_Vogel_Lake_vicinity_occrrence_aggregate.csv", row.names=FALSE)
```

I added the BOLD results to the results data frame from the blackfish
diet work.

``` r
setwd("D:/projects/blackfish/2020_blackfish_diet/2020-03_SCVUC/ml-jg")

r2 <- read.csv("results_2020-03-23-1201.csv", stringsAsFactors=FALSE)

brt <- read.csv("bold_id_engine_results_0001-0399.csv", stringsAsFactors=FALSE)

br <- brt

brt <- read.csv("bold_id_engine_results_0400-0799.csv", stringsAsFactors=FALSE)
br <- rbind(br, brt)

brt <- read.csv("bold_id_engine_results_0800-1079.csv", stringsAsFactors=FALSE)
names(brt) <- names(br)
br <- rbind(br, brt)

write.csv(br, "2020-03-24-0916_bold_results.csv", row.names=FALSE)

names(br)[c(1,2,4)] <- c("ESV_name", "bold_id", "bold_sim")

r3 <- merge(r2, br[,c(1,2,4)], all.x=TRUE)

write.csv(r3, "results_2020-03-23-1201.csv", row.names=FALSE)

## Now making a dataframe for working with identifications.
ids1 <- unique(r3[,c(1,6:41)])
dim(ids1)
[1] 212  37

write.csv(ids1, "2020-03-24-0925_ids.csv", row.names=FALSE)

## Now manually editing.
```

I worked on improving identifications from the blackfish diet work,
combing through BLAST results (temporary results URL:
<https://blast.ncbi.nlm.nih.gov/Blast.cgi?CMD=Get&RID=7MHZ12H101R>),
BOLD ID Engine results, and my own vsearch searches. I got through ESV
Zotu140.

## Wednesday, March 25

To do:

  - ~~Format and this week’s *Refuge Notebook* article to the
    *Clarion*.~~
  - *Refuge Notebook* catch-up.
  - Finish blackfish diet analysis.
  - Publish blackfish diet raw data.
  - Finish metadata for data management team meeting tomorrow.

I edited this week’s *Refuge Notebook* article.

I worked on identifications from the blackfish diet analysis, getting
the first round of IDs done. BOLD’s ID Engine apparently went down,
though, so I could not finish re-examining the rotifer sequences.

I attended the biology telecon at 11:00.

I finished editing this week’s *Refuge Notebook* article and submitted
it to the paper.

I finished scrutinizing identifications and examined the results using
R.

``` r
setwd("D:/projects/blackfish/2020_blackfish_diet/2020-03_SCVUC/ml-jg")

library(Biostrings)

ids1 <- read.csv("2020-03-25-1313_ids.csv", stringsAsFactors=FALSE)

r3 <- read.csv("results_2020-03-23-1201.csv", stringsAsFactors=FALSE)

dim(ids1)
[1] 212  40
dim(r3)
[1] 275  41

## How many ESVs have we accepted?
sum(ids1$accept==1)
[1] 131

## What is the set of identifications that were accepted?
accid <- levels(as.factor(ids1$scientific_name[ids1$accept==1]))

## How many unique identifications are there?
length(accid)
[1] 103

## I want to see what cutting out all reads except the nominal read length of 313 bp does.
fas1 <- readDNAStringSet("cat.denoised.nonchimeras")
ESV_name <- names(fas1)
fasd <- as.data.frame(ESV_name)
fasd$width <- width(fas1)

ids2 <- merge(ids1, fasd, all.x=TRUE)

## How many unique identifications are accepted and of read length 313?
accid313 <- levels(as.factor(ids2$scientific_name[ids2$accept==1 & ids2$width==313]))
length(accid313)
[1] 79 ## Wow, (103-79)/103 = 23% are dropped. What did we loose?

setdiff(accid, accid313)
 [1] "Aeshna juncea"                             "Aeshna sp. Zotu171"                       
 [3] "Bdelloidea sp. Zotu158"                    "Cyclopidae sp. Zotu83"                    
 [5] "Gyraulus sp. clade B TS-2018 BOLD:AEB5660" "Pleuroxus aduncus"                        
 [7] "Ploima sp. Zotu147"                        "Ploima sp. Zotu151"                       
 [9] "Ploima sp. Zotu153"                        "Ploima sp. Zotu187"                       
[11] "Ploima sp. Zotu188"                        "Procladius sp. BOLD:AAP3007"              
[13] "Rhamphomyia sp. BOLD:ACG3627"              "Rotaria sp. Zotu115"                      
[15] "Rotaria sp. Zotu215"                       "Rotifera sp. Zotu128"                     
[17] "Rotifera sp. Zotu179"                      "Rotifera sp. Zotu30"                      
[19] "Rotifera sp. Zotu40"                       "Rotifera sp. Zotu67"                      
[21] "Rotifera sp. Zotu72"                       "Rotifera sp. Zotu90"                      
[23] "Trichocerca sp. Zotu166"                   "Trichocerca sp. Zotu68"   
## Mostly rotifers are lost here.

## Just having a look at these records.
ids2[ids2$accept==1 & !ids2$width==313,]

## I looked at other files, also. I am going to keep those. I cannot loose that Zotu2, Aeshna juncea, one of the largest Zotus.

## Now I want to prepare a FASTA file to look at the tree.

keeperZotus <- ids1$ESV_name[ids1$accept==1]
keeperlab <- paste0(keeperZotus, " | ", ids1$scientific_name[ids1$accept==1])
keeperlab <- keeperlab[order(keeperlab)]

fask <- fas1[names(fas1) %in% keeperZotus]
fask <- fask[order(names(fask))]

names(fask) <- keeperlab

## Saving.
writeXStringSet(fask, "2020-03-25-1431_accepted_sequences.fas", format="fasta", width=1000)

## Meanwhile, I need to join data together.
r4 <- merge(r3[,1:3], ids2, all.x=TRUE)

## Now I want to remove occurrences with frequencies < 0.05%.

et <- read.delim("ESV.table")
names(et)[1] <- "ESV_name"
et$total <- apply(et[,2:6], 1, sum)
et$min_threshold <- et$total * 0.05/100

et_drop <- et[,1:6]
et_drop[,2:6] <- et[,2:6] <= et$min_threshold
et_drop[,2:6] <- et_drop[,2:6] + 0
et_drop[,2:6] <- et_drop[,2:6] * !et[,2:6]==0

## How many records would be dropped?
sum(et_drop[,2:6])
[1] 2

library(reshape2)
et2 <- melt(et_drop, id.vars="ESV_name")
et2 <- et2[et2$value==1,]
et2
     ESV_name variable value
760     Zotu2    MLB04     1
1015    Zotu4    MLB05     1

r5 <- r4
for (this_record in 1:nrow(et2))
 {
 slesv <- r5$ESV_name == et2$ESV_name[this_record]
 sls <- r5$SampleName == et2$variable[this_record]
 sl <- !(slesv & sls)
 print(sum(sl))
 r5 <- r5[sl,]
 }
 
## Saving here.
write.csv(r5, "results_2020-03-25-1552.csv", row.names=FALSE) 

## How many accepted sample * ESV records do we have?
sum(r5$accept==1)
[1] 167
```

I submitted the FASTA file of accepted sequences to NGPhylogeny.fr using
the “NGPhylogeny Analyse - FastME/OneClick” option. Results should be
available at
<https://ngphylogeny.fr/workspace/history/47db2b18949f0c42>.

## Thursday, March 26

To do:

  - Blackfish analysis.
  - Publish blackfish sequence data.
  - Early Adopters’ data management meeting at 11:00

The phylogenetic analysis from yesterday ran just fine.

For the purpose of keeping all ESV labels distinct for searchin on
Arctos or combining FASTA datasets, I want to relabel at ESVs. In the
output tree I relabeled “Zotu” to “bfdZotu”, with “bfd” standing for
“blackfish diet”. Resulting output tree on iTOL:
<https://itol.embl.de/tree/16415961276811585237767#>.

I worked on examining results of the blackfish work.

``` r
## Today I want to look at results.

setwd("D:/projects/blackfish/2020_blackfish_diet/2020-03_SCVUC/ml-jg")

r5 <- read.csv("results_2020-03-25-1552.csv", stringsAsFactors=FALSE)

guid <- read.csv("2020-03-26-0819_guids.csv", stringsAsFactors=FALSE)

r6 <- merge(r5, guid, all.x=TRUE)

## Renaming ESVs.
r6$ESV_name <- gsub("Zotu", "bfdZotu", r6$ESV_name)
r6$scientific_name <- gsub("Zotu", "bfdZotu", r6$scientific_name)

## Saving here.
write.csv(r6, "results_2020-03-26-0844.csv", row.names=FALSE)

## Now making a summary data frame for making an overall pie chart.
library(reshape2)

## In this one I will look at all ESVs.
sm1 <- melt(r6[r6$accept==1,c("scientific_name", "ESVsize")], measure.vars="ESVsize")
sm1 <- sm1[order(-sm1$value),]

pie(sm1$value, labels = sm1$scientific_name, main="ESV abundances, summed over all samples")

## Now I want to add percentages.
sm1$percentage <- round(sm1$value*100/sum(sm1$value))

pie(sm1$value, labels = paste0(sm1$scientific_name, " ", sm1$percentage, "%"), main="ESV abundances, summed over all samples")

png(filename="2020-03-26-0921_ESV_abundances.png",
 width=1000,
 height=600
 )
 pie(sm1$value, 
  labels = paste0(sm1$scientific_name, " ", sm1$percentage, "%"), 
  main="ESV abundances, summed over all samples"
  )
dev.off()

## Now for each fish.
for (this_fish in 1:5)
 {
 fish <- guid$guid[this_fish]
 smf <- melt(r6[(r6$accept==1 & r6$guid==fish),c("scientific_name", "ESVsize")], measure.vars="ESVsize")
 smf <- smf[order(-smf$value),]
 smf$percentage <- round(smf$value*100/sum(smf$value))
 png(filename=paste0("2020-03-26-0929_ESV_abundances_", this_fish, ".png"),
  width=1000,
  height=600
  )
  pie(smf$value, 
   labels = paste0(smf$scientific_name, " ", smf$percentage, "%"), 
   main=paste("ESV abundances, blackfish", fish)
   )
 dev.off()
 }
## It would be nice if the colors were consistent.

## Downloaded color table from https://en.wikipedia.org/wiki/List_of_colors:_A%E2%80%93F
cdf <- read.csv("2020-03-26-1055_cols_a-f.csv", stringsAsFactors=FALSE)

set.seed(42) 
r6$col <- sample(x=cdf$hex, size=nrow(r6), replace = FALSE)

## Remaking some of those pie charts with new, consistent colors.
sm1 <- melt(r6[r6$accept==1,c("scientific_name", "col", "ESVsize")], measure.vars="ESVsize")
sm1 <- sm1[order(-sm1$value),]
sm1$percentage <- round(sm1$value*100/sum(sm1$value))

png(filename="2020-03-26-1103_ESV_abundances.png",
 width=1000,
 height=600
 )
 pie(sm1$value, 
  labels = paste0(sm1$scientific_name, " ", sm1$percentage, "%"), 
  main="ESV abundances, summed over all samples",
  col=sm1$col
  )
dev.off() 

## Remaking pie charts for each fish.
for (this_fish in 1:5)
 {
 fish <- guid$guid[this_fish]
 smf <- melt(r6[(r6$accept==1 & r6$guid==fish),c("scientific_name", "col", "ESVsize")], measure.vars="ESVsize")
 smf <- smf[order(-smf$value),]
 smf$percentage <- round(smf$value*100/sum(smf$value))
 png(filename=paste0("2020-03-26-1109_ESV_abundances_", this_fish, ".png"),
  width=1000,
  height=600
  )
  pie(smf$value, 
   labels = paste0(smf$scientific_name, " ", smf$percentage, "%"), 
   main=paste("ESV abundances, blackfish", fish),
   col=smf$col
   )
 dev.off()
 }

## Now trying to make a bigger plot for the individual fish.
png(filename="2020-03-26-1131_ESV_abundances.png",
 width=1000*2,
 height=600*3
 ) 
 par(mfrow=c(3,2))
 par(mar=c(0,0,1,0))
 plot.new()
for (this_fish in 1:5)
 {
 fish <- guid$guid[this_fish]
 smf <- melt(r6[(r6$accept==1 & r6$guid==fish),c("scientific_name", "col", "ESVsize")], measure.vars="ESVsize")
 smf <- smf[order(-smf$value),]
 smf$percentage <- round(smf$value*100/sum(smf$value))
  pie(smf$value, 
   labels = paste0(smf$scientific_name, " ", smf$percentage, "%"), 
   main=paste("ESV abundances, blackfish", fish),
   col=smf$col
   )
 }
dev.off() 

## Putting everything on one plot.
pdf(file="2020-03-26-1141_ESV_abundances.pdf",
 width=10*2,
 height=6*3,
 pointsize=12
 ) 
 par(mfrow=c(3,2))
 par(mar=c(0,0,1,0))
 pie(sm1$value, 
  labels = paste0(sm1$scientific_name, " ", sm1$percentage, "%"), 
  main="ESV abundances, summed over all samples",
  col=sm1$col,
  radius=0.8
  )
for (this_fish in 1:5)
 {
 fish <- guid$guid[this_fish]
 smf <- melt(r6[(r6$accept==1 & r6$guid==fish),c("scientific_name", "col", "ESVsize")], measure.vars="ESVsize")
 smf <- smf[order(-smf$value),]
 smf$percentage <- round(smf$value*100/sum(smf$value))
  pie(smf$value, 
   labels = paste0(smf$scientific_name, " ", smf$percentage, "%"), 
   main=paste("ESV abundances, blackfish", fish),
   col=smf$col
   )
 }
dev.off() 

## I might try to make the areas proportional to the number of reads.
rtdf <- melt(r6[(r6$accept==1),c("guid", "ESVsize")], measure.vars="ESVsize")
rtdf <- dcast(rtdf, guid ~ ., sum)

tr <- sum(r6$ESVsize[r6$accept==1])
ta <- pi*0.8^2 ## Area of the summed circle.

rtdf$a <- rtdf[,2]*ta/tr
rtdf$r <- (rtdf$a/pi)^0.5

## Putting everything on one plot.
pdf(file="2020-03-1432_ESV_abundances.pdf",
 width=10*2,
 height=6*3,
 pointsize=12
 ) 
 par(mfrow=c(3,2))
 par(mar=c(0,0,2,0))
 pie(sm1$value, 
  labels = paste0(sm1$scientific_name, " ", sm1$percentage, "%"), 
  main="ESV abundances, summed over all samples",
  col=sm1$col,
  radius=0.8
  )
for (this_fish in 1:5)
 {
 fish <- guid$guid[this_fish]
 smf <- melt(r6[(r6$accept==1 & r6$guid==fish),c("scientific_name", "col", "ESVsize")], measure.vars="ESVsize")
 smf <- smf[order(-smf$value),]
 smf$percentage <- round(smf$value*100/sum(smf$value))
  pie(smf$value, 
   labels = paste0(smf$scientific_name, " ", smf$percentage, "%"), 
   main=paste("ESV abundances, blackfish", fish),
   col=smf$col,
   radius=rtdf$r[this_fish]
   )
 }
dev.off() 

write.csv(r6, "results_2020-03-26-1854.csv", row.names=FALSE)
save.image("2020-03-26-1854_workspace.RData")
```

### 1:00 pm early adopters’ data management meeting

  - Question: How can the metadata refer to a number of files? Answer:
    Under the distribution tab there will be a way to associated files /
    URLs, etc.

  - Hilmar commented on mdEditor files. Should not use the “product”
    resource type; use something more specific. Raw data means RAW data,
    e.g. data sheets, that have not been processed at all.

  - Collections. Question: I have thousands of files. Do I need to make
    a metadata entry for each file? Answer: No, there is a collection
    resource type for this kind of situation where there are many files
    in a standard format.

  - Data dictionaries. This tells the reader what each field is in your
    data file. Data dictionaries must be linked to resources.

  - Next steps are to complete three product medata records and to make
    a data dictionary.

I started writing an article on blackfish diet for the *AKES
Newsletter*.

## Friday, March 27

To do:

  - Blackfish analysis.
  - Publish blackfish sequence data.
  - Format Derek’s black spruce article.
  - Early adopters’ homework.

I worked on uploading images to the blackfish records on Arctos.

I uploaded pecimen data downloaded from Arctos and DNA extraction
methods, DNA sequencing methods, resulting raw metagenomic data, and
related files from RTL Genomics to Zenodo (Bowser and Bowser
[2020](#ref-bowser_raw_2020)).

# Bibliography

<div id="refs" class="references">

<div id="ref-ak_region_data_stewardship_team_alaska_2020">

Alaska Region Data Stewardship Team. 2019a. Alaska Region Interim Data
Management User Guide (v.3).
doi:[10.7944/P9JF0RT6](https://doi.org/10.7944/P9JF0RT6).

</div>

<div id="ref-alaska_region_data_stewardship_team_alaska_2020">

Alaska Region Data Stewardship Team. 2019b. Alaska Region Interim
Metadata Guide. Available from
<https://ak-region-dst.gitbook.io/alaska-region-mdeditor-interim-user-guide/>
\[accessed 27 February 2020\].

</div>

<div id="ref-bella_Vogel_2013">

Bella, E. 2013. Kenai national wildlife refuge invasive plant survey
2013. Vogel lake.
doi:[10.7299/X7V40VHG](https://doi.org/10.7299/X7V40VHG).

</div>

<div id="ref-bowmer_identification_1995">

Bowmer, K.H., Jacobs, S.W.L., and Sainty, G.R. 1995. Identification,
biology and management of *Elodea canadensis*, Hydrocharitaceae. Journal
of Aquatic Plant Management **33**: 13–19. Available from
<http://www.apms.org/wp/wp-content/uploads/2012/10/v33p13.pdf>.

</div>

<div id="ref-bowser_work_2019">

Bowser, M.L. 2019. Work journal 2019. Exported pdf version. U.S. Fish &
Wildlife Service, Kenai National Wildlife Refuge., Soldotna, Alaska.
doi:[10.7299/X7RB74XZ](https://doi.org/10.7299/X7RB74XZ).

</div>

<div id="ref-bowser_raw_2020">

Bowser, M.L., and Bowser, A.M. 2020, March. Raw metagenomic data from
gut contents of *Dallia pectoralis* specimens collected in Kenai,
Alaska, in 2019. Zenodo.
doi:[10.5281/zenodo.3731327](https://doi.org/10.5281/zenodo.3731327).

</div>

<div id="ref-bowser_et_al_2020">

Bowser, M.L., Brassfield, R., Dziergowski, A., Eskelin, T., Hester, J.,
Magness, D.R., McInnis, M., Melvin, T., Morton, J.M., and Stone, J.
2020. Towards conserving natural diversity: A biotic inventory by
observations, specimens, dna barcoding and high-throughput sequencing
methods. Biodiversity Data Journal **8**: e50124. Pensoft Publishers.
doi:[10.3897/BDJ.8.e50124](https://doi.org/10.3897/BDJ.8.e50124).

</div>

<div id="ref-braukmann_metabarcoding_2019">

Braukmann, T.W.A., Ivanova, N.V., Prosser, S.W.J., Elbrecht, V.,
Steinke, D., Ratnasingham, S., Waard, J.R. de, Sones, J.E., Zakharov,
E.V., and Hebert, P.D.N. 2019. Metabarcoding a diverse arthropod mock
community. Molecular Ecology Resources **19**(3): 711–727.
doi:[10.1111/1755-0998.13008](https://doi.org/10.1111/1755-0998.13008).

</div>

<div id="ref-costello_non-native_2008">

Costello, D.M., and Lamberti, G.A. 2008. Non-native earthworms in
riparian soils increase nitrogen flux into adjacent aquatic ecosystems.
Oecologia **158**(3): 499–510.
doi:[10.1007/s00442-008-1149-0](https://doi.org/10.1007/s00442-008-1149-0).

</div>

<div id="ref-costello_biological_2009">

Costello, D.M., and Lamberti, G.A. 2009. Biological and physical effects
of non-native earthworms on nitrogen cycling in riparian soils. Soil
Biology and Biochemistry **41**(10): 2230–2235.
doi:[10.1016/j.soilbio.2009.08.007](https://doi.org/10.1016/j.soilbio.2009.08.007).

</div>

<div id="ref-Criscuolo_2010">

Criscuolo, A., and Gribaldo, S. 2010. BMGE (block mapping and gathering
with entropy): A new software for selection of phylogenetic informative
regions from multiple sequence alignments. BMC Evolutionary Biology
**10**(1): 210. Springer Nature.
doi:[10.1186/1471-2148-10-210](https://doi.org/10.1186/1471-2148-10-210).

</div>

<div id="ref-Desper_2002">

Desper, R., and Gascuel, O. 2002. Fast and accurate phylogeny
reconstruction algorithms based on the minimum-evolution principle.
Journal of Computational Biology **9**(5): 687–705. Mary Ann Liebert
Inc.
doi:[10.1089/106652702761034136](https://doi.org/10.1089/106652702761034136).

</div>

<div id="ref-drake_american_1963">

Drake, C.J., and Lattin, J.D. 1963. American species of the lacebug
genus *Acalypta* (Hemiptera: Tingidae). Proceedings of the United States
National Museum **115**(3486): 331–345 +15 plates.
doi:[10.5479/si.00963801.115-3486.331](https://doi.org/10.5479/si.00963801.115-3486.331).

</div>

<div id="ref-fernandez_vagrant_2011">

Fernández, R., Almodóvar, A., Novo, M., Gutiérrez, M., and Díaz Cosín,
D.J. 2011. A vagrant clone in a peregrine species: Phylogeography, high
clonal diversity and geographical distribution in the earthworm
Aporrectodea trapezoides (Dugès, 1828). Soil Biology and Biochemistry
**43**(10): 2085–2093.
doi:[10.1016/j.soilbio.2011.06.007](https://doi.org/10.1016/j.soilbio.2011.06.007).

</div>

<div id="ref-froeschner_zoogeographic_1976">

Froeschner, R.C. 1976. Zoogeographic notes on the lace bug genus
*Acalypta* Westwood in the Americas with description of a new species
from Mexico (Hemiptera: Tingidae). The American Midland Naturalist
**96**(2): 257–269.
doi:[10.2307/2424067](https://doi.org/10.2307/2424067).

</div>

<div id="ref-gbiforg_gbif_2020-1">

GBIF.org. 2020a. GBIF Occurrence Download. GBIF.org.
doi:[10.15468/dl.bguubm](https://doi.org/10.15468/dl.bguubm).

</div>

<div id="ref-gbiforg_gbif_2020">

GBIF.org. 2020b. GBIF Occurrence Download. GBIF.org.
doi:[10.15468/dl.3dfs26](https://doi.org/10.15468/dl.3dfs26).

</div>

<div id="ref-gbiforg_gbif_2020-2">

GBIF.org. 2020c. GBIF Occurrence Download. GBIF.org.
doi:[10.15468/dl.deymas](https://doi.org/10.15468/dl.deymas).

</div>

<div id="ref-gbiforg_gbif_2020-3">

GBIF.org. 2020d. GBIF Occurrence Download. GBIF.org.
doi:[10.15468/dl.azbzw2](https://doi.org/10.15468/dl.azbzw2).

</div>

<div id="ref-gbif_secretariat_enterion_2019">

GBIF Secretariat. 2019. *Enterion castaneum* Savigny, 1826. *In* GBIF
Backbone Taxonomy. Checklist dataset. GBIF.org.
doi:[10.15468/39omei](https://doi.org/10.15468/39omei).

</div>

<div id="ref-gross_unterscheidung_2003">

Gross, E.M., Erhard, D., Glaschke, A., and Haid, V. 2003. Unterscheidung
unklarer morphotypen von *elodea* spp. Durch chemotaxonomische und
molekulartaxonomische methoden. Deutsche Gesellschaft für Limnologie
(DGL) - Tagungsbericht (Köln). Available from
<https://docplayer.org/7100923-Unterscheidung-unklarer-morphotypen-von-elodea-spp-durch-chemotaxonomische-und-molekulartaxonomische-methoden.html>.

</div>

<div id="ref-hajibabaei_coi_2019">

Hajibabaei, M., Porter, T.M., Wright, M., and Rudar, J. 2019a. COI
metabarcoding primer choice affects richness and recovery of indicator
taxa in freshwater systems. bioRxiv.
doi:[10.1101/572628](https://doi.org/10.1101/572628).

</div>

<div id="ref-hajibabaei_coi_2019b">

Hajibabaei, M., Porter, T.M., Wright, M., and Rudar, J. 2019b. COI
metabarcoding primer choice affects richness and recovery of indicator
taxa in freshwater systems. PLOS ONE **14**(9): e0220953.
doi:[10.1371/journal.pone.0220953](https://doi.org/10.1371/journal.pone.0220953).

</div>

<div id="ref-hedlin_cone_1974">

Hedlin, A.F. 1974. Cone and Seed Insects of British Columbia.
Environment Canada, Forestry Service, Pacific Forest Research Centre,
Victoria, British Columbia.

</div>

<div id="ref-huotari_comparative_2013">

Huotari, T., and Korpelainen, H. 2013. Comparative analyses of plastid
sequences between native and introduced populations of aquatic weeds
*Elodea canadensis* and *e*. *Nuttallii*. PLOS ONE **8**(4): e58073.
doi:[10.1371/journal.pone.0058073](https://doi.org/10.1371/journal.pone.0058073).

</div>

<div id="ref-huotari_population_2011">

Huotari, T., Korpelainen, H., Leskinen, E., and Kostamo, K. 2011.
Population genetics of the invasive water weed Elodea canadensis in
Finnish waterways. Plant Systematics and Evolution **294**(1): 27.
doi:[10.1007/s00606-011-0442-2](https://doi.org/10.1007/s00606-011-0442-2).

</div>

<div id="ref-jakubas_sandpiper_1984">

Jakubas, W.J., and Firman, A.S. 1984. Sandpiper Lake, 7/3/84. U.S. Fish
& Wildlife Service.
doi:[10.7299/X7JM29X1](https://doi.org/10.7299/X7JM29X1).

</div>

<div id="ref-Junier_2010">

Junier, T., and Zdobnov, E.M. 2010. The newick utilities:
High-throughput phylogenetic tree processing in the unix shell.
Bioinformatics **26**(13): 1669–1670. Oxford University Press (OUP).
doi:[10.1093/bioinformatics/btq243](https://doi.org/10.1093/bioinformatics/btq243).

</div>

<div id="ref-kadono_genetic_1997">

Kadono, Y., Nakamura, T., and Suzuki, T. 1997. Genetic uniformity of two
aquatic plants, *Egeria densa* Planch. And *Elodea nuttallii* (Planch.)
St. John, introduced in Japan. Japanese Journal of Limnology
(Rikusuigaku Zasshi) **58**(2): 197–203.
doi:[10.3739/rikusui.58.197](https://doi.org/10.3739/rikusui.58.197).

</div>

<div id="ref-Katoh_2013">

Katoh, K., and Standley, D.M. 2013. MAFFT multiple sequence alignment
software version 7: Improvements in performance and usability. Molecular
Biology and Evolution **30**(4): 772–780. Oxford University Press (OUP).
doi:[10.1093/molbev/mst010](https://doi.org/10.1093/molbev/mst010).

</div>

<div id="ref-langille_proposed_1904">

Langille, W.A. 1904. The proposed forest reserve on the Kenai Peninsula,
Alaska. United States Forest Service.
doi:[10.5962/bhl.title.124553](https://doi.org/10.5962/bhl.title.124553).

</div>

<div id="ref-Lefort_2015">

Lefort, V., Desper, R., and Gascuel, O. 2015. FastME 2.0: A
comprehensive, accurate, and fast distance-based phylogeny inference
program. Molecular Biology and Evolution **32**(10): 2798–2800. Oxford
University Press (OUP).
doi:[10.1093/molbev/msv150](https://doi.org/10.1093/molbev/msv150).

</div>

<div id="ref-Lemoine_2018">

Lemoine, F., Domelevo Entfellner, J.-B., Wilkinson, E., Correia, D.,
Dávila Felipe, M., De Oliveira, T., and Gascuel, O. 2018. Renewing
felsenstein’s phylogenetic bootstrap in the era of big data. Nature
**556**(7702): 452–456. Springer Nature.
doi:[10.1038/s41586-018-0043-0](https://doi.org/10.1038/s41586-018-0043-0).

</div>

<div id="ref-Letunic_et_al_2019">

Letunic, I., and Bork, P. 2019. Interactive Tree Of Life (iTOL) v4:
recent updates and new developments. Nucleic Acids Research **47**(W1):
W256–W259. doi:[10.1093/nar/gkz239](https://doi.org/10.1093/nar/gkz239).

</div>

<div id="ref-lipkin_investigation_2007">

Lipkin, R. 2007. Investigation of Potentially Sensitive Plant
Communities in the Old Dummy Burn, Kanuti National Wildlife Refuge,
Alaska, 2006. Alaska Natural Heritage Program, Environment; Natural
Resources Institute, University of Alaska Anchorage, Anchorage, Alaska.
Available from <https://www.arlis.org/docs/vol1/A/313839549.pdf>.

</div>

<div id="ref-marchand_oxygen_1985">

Marchand, P.J. 1985. Oxygen evolution by *Elodea canadensis* under ice
and snow cover: A case for winter photosynthesis in subnivean vascular
plants. Aquilo, Serie Botanica **23**: 57–61.

</div>

<div id="ref-martinsson_dna-barcoding_2015">

Martinsson, S., Cui, Y., Martin, P.J., Pinder, A., Quinlan, K., Wetzel,
M.J., and Erséus, C. 2015. DNA-barcoding of invasive European earthworms
(Clitellata: Lumbricidae) in south-western Australia. Biological
Invasions **17**(9): 2527–2532.
doi:[10.1007/s10530-015-0910-7](https://doi.org/10.1007/s10530-015-0910-7).

</div>

<div id="ref-mcalpine_manual_1983">

McAlpine, J.F., Peterson, B.V., Shewell, G.E., Teskey, H.J., Vockeroth,
J.R., and Wood, D.M. (*Editors*). 1983. Manual of Nearctic Diptera,
volume 1. Research Branch, Agriculture Canada, Ottawa, Ontario, Canada.

</div>

<div id="ref-morton_ghosts_2014">

Morton, J. 2014. Ghosts of fires past. *In* Refuge Notebook. USFWS Kenai
National Wildlife Refuge, Soldotna, Alaska. pp. 45–46. Available from
<https://www.fws.gov/uploadedFiles/Region_7/NWRS/Zone_2/Kenai/Sections/What_We_Do/In_The_Community/Refuge_Notebooks/2014_Articles/Refuge_Notebook_v16_n23.pdf>.

</div>

<div id="ref-morton_responding_2019">

Morton, J., Bowser, M., and Eskelin, T. 2019, July. Responding rapidly
to Elodea—the first freshwater invasive plant in Alaska. San Diego,
California. Available from
<https://www.fws.gov/uploadedFiles/Morton_JM_et_al_2019.pdf>.

</div>

<div id="ref-packer_bee_genera_2007">

Packer, L., Genaro, J.A., and Sheffield, C.S. 2007. The bee genera of
eastern canada. Canadian Journal of Arthropod Identification **3**.
doi:[10.3752/cjai.2007.03](https://doi.org/10.3752/cjai.2007.03).

</div>

<div id="ref-paez_alpine_1991">

Paez, C.E. 1991. Alpine vegetation of areas utilized by introduced
populations of caribou (*Rangifer tarandus*) on the Kenai Peninsula,
Alaska. Master’s thesis, University of Wisconsin-Madison, Madison,
Wisconsin.

</div>

<div id="ref-pinder_ecology_2013">

Pinder, R.A. 2013. Ecology of earthworms in riparian habitats. PhD
Thesis, University at Albany, State University of New York, Albany, New
York. Available from <https://search.proquest.com/docview/1419999664>.

</div>

<div id="ref-porco_reference_2018">

Porco, D., Chang, C.-H., Dupont, L., James, S., Richard, B., and
Decaëns, T. 2018. A reference library of DNA barcodes for the
earthworms from Upper Normandy: Biodiversity assessment, new records,
potential cases of cryptic diversity and ongoing speciation. Applied
Soil Ecology **124**: 362–371.
doi:[10.1016/j.apsoil.2017.11.001](https://doi.org/10.1016/j.apsoil.2017.11.001).

</div>

<div id="ref-pratt_notes_1980">

Pratt, G.K., and Pratt, H.D. 1980. Notes on Nearctic *Sylvicola*
(Diptera: Anisopodidae). Proceedings of the Entomological Society of
Washington **82**: 86–98.

</div>

<div id="ref-Sato_et_al_Rabdophaga_2020">

Sato, S., Harris, K.M., Collet, D.M., Kim, W., and Yukawa, J. 2020.
Genetic variation in intraspecific populations of *Rabdophaga rosaria*
(Diptera: Cecidomyiidae) indicating possible diversification scenarios
into sibling species along with host range expansion on willows
(Salicaceae: *Salix*). Zoological Journal of the Linnean Society.
doi:[10.1093/zoolinnean/zlz179](https://doi.org/10.1093/zoolinnean/zlz179).

</div>

<div id="ref-shorthouse_simplemappr_2010">

Shorthouse, D.P. 2010. SimpleMappr, an online tool to produce
publication-quality point maps. Available from
<https://www.simplemappr.net>.

</div>

<div id="ref-state_of_alaska_dec_2005">

State of Alaska Department of Environmental Conservation. 2005. Ambient
water quality monitoring system. Available from
<https://awqms2.goldsystems.com/Login.aspx>.

</div>

<div id="ref-stolz_nestling_2019">

Stolz, C. 2019. The nestling diet of Svalbard snow buntings identified
by DNA metabarcoding. Master’s thesis, The Arctic University of Norway.
Available from <https://hdl.handle.net/10037/15438>.

</div>

<div id="ref-wang_naive_2007">

Wang, Q., Garrity, G.M., Tiedje, J.M., and Cole, J.R. 2007. Naïve
Bayesian classifier for rapid assignment of rRNA sequences into the new
bacterial taxonomy. Applied and Environmental Microbiology **73**(16):
5261. doi:[10.1128/AEM.00062-07](https://doi.org/10.1128/AEM.00062-07).

</div>

</div>
